\begin{thebibliography}{99}

\bibitem[Ahmet et al.(2020)%
Ahmet, Basheer, Salem, Lu and Afarinkia]{Ahmet2020}
\textsc{Ahmet, D.S., Basheer, H.A., Salem, A., Lu, D. and Afarinkia, K.} (2020).
Application of small molecule FPR1 antagonists in the treatment of cancers.
Sci. Rep.
\textbf{10},
17249.

\bibitem[Cui et al.(2015)%
Cui, Li and  Zhong]{Cui2015}
\textsc{Cui, H., Li, R. and  Zhong, W.} (2015).
Model-free feature screening for ultrahigh dimensional discriminant analysis.
J. Am. Stat. Assoc.
\textbf{110},
630--641.

\bibitem[Fan et al.(2011)%
Fan, Feng and Song]{Fan2011}
\textsc{Fan, J., Feng, Y. and Song, R.} (2011).
Nonparametric independence screening in sparse ultra-high-dimensional additive models.
J. Am. Stat. Assoc.
\textbf{106},
544--557.

\bibitem[Fan and Li(2011)%
Fan and Li]{Fan2001}
\textsc{Fan, J. and  Li, R.} (2001).
Variable selection via nonconcave penalized likelihood and its oracle properties.
J. Am. Stat. Assoc.
\textbf{96},
1348--1360.

\bibitem[Fan and Lv(2008)%
Fan and Lv]{Fan2008}
\textsc{Fan, J. and Lv, J.} (2008).
Sure independence screening for ultrahigh dimensional feature space.
J. R. Stat. Soc. Ser. B (Methodological) 
\textbf{70},
849--911.

\bibitem[Fan et al.(2014)%
Fan, Ma and Dai]{Fan2014}
\textsc{Fan, J., Ma, Y. and Dai, W.} (2014).
Nonparametric independence screening in sparse ultra-high-dimensional varying coefficient models.
J. Am. Stat. Assoc.
\textbf{109},
1270--1284.

\bibitem[Fan et al.(2009)%
Fan, Samworth and Wu]{Fan2009}
\textsc{Fan, J., Samworth, R. and Wu, Y.} (2009).
Ultrahigh dimensional feature selection: Beyond the linear model.
J. Mach. Learn. Res.
\textbf{10},
2013--2038.

\bibitem[Fan and Song(2010)%
Fan and Song]{Fan2010}
\textsc{Fan, J. and Song, R.} (2010).
Sure independence screening in generalized linear models with NP-dimensionality.
Ann. Stat.
\textbf{38},
3567--3604.

\bibitem[Gendoo et al.(2019)%
Gendoo, Zon, Sandhu,  Manem, Ratanasirigulchai,  et al.]{Gendoo2019}
\textsc{Gendoo, D., Zon, M., Sandhu, V., Manem, V. S., Ratanasirigulchai, N., et al.} (2019)
MetaGxData: clinically annotated breast, ovarian and pancreatic cancer datasets and their use in generating a multi-cancer gene signature.
Sci. Rep.
\textbf{9},
8770.

\bibitem[Gorst-Rasmussen and Scheike(2013)%
Gorst-Rasmussen and Scheike]{Gorst2013}
\textsc{Gorst-Rasmussen, A. and Scheike, T.} (2013).
Independent screening for single-index hazard rate models with ultrahigh dimensional features.
J. R. Stat. Soc. Ser. B (Methodological)  
\textbf{75},
217--245.


\bibitem[Gretton et al.(2005)
Gretton, A.,  Bousquet, O.,  Smola, A. and Bernhard Sch{\"o}lkopf.]{Gretton2005}
\textsc{Gretton, A.,  Bousquet, O.,  Smola, A. and Bernhard Sch{\"o}lkopf.} (2005).
Measuring statistical dependence with Hilbert-Schmidt norms.
Algorithmic Learning Theory, International Conference, Alt, Singapore, October. Springer, Berlin, Heidelberg.






%\bibitem[Gui and Li(2005)%
%Gui and Li]{Gui2005}
%\textsc{Gui, J. and Li, H.} (2005).
%Penalized Cox regression analysis in the high-dimensional and low-sample size settings,
%with applications to microarray gene expression data.
%\emph{Bioinformatics},
%\textbf{21},
%3001--3008.





%
%\bibitem[Huang and Zhu(2016)%
%Huang and Zhu]{Huang2016}
%\textsc{Huang, Q. and Zhu, Y.} (2016).
%Model-free sure screening via maximum correlation.
%\emph{Journal of Multivariate Analysis},
%\textbf{148},
%89--106.


\bibitem[Karlan et al.(2014)%
Karlan B.Y., Dering J., Walsh C., et al.]{Karlan2014}
\textsc{Karlan B.Y., Dering J., Walsh C., et al.}(2014)
POSTN/TGFBI-associated stromal signature predicts poor prognosis in serous epithelial ovarian cancer.
Gynecol. Oncol.
\textbf{132},
334--342.



\bibitem[Kohn et al.(2014)%
Kohn, Zeeberg, Reinhold  and  Pommier]{Kohn2014}
\textsc{Kohn, K.W.,  Zeeberg, B.M.,  Reinhold, W.C.  and  Pommier, Y.} (2014)
Gene expression correlations in human cancer cell lines define molecular interaction networks for epithelial phenotype.
Plos One
\textbf{9},
 e99269.



%\bibitem[Li et al.(2005)%
%Li and Luan]{Li2005}
%\textsc{Li, H. and Luan, Y.} (2005).
%Boosting proportional hazards models using smoothing splines, with applications to
%high-dimensional microarray data.
%\emph{Bioinformatics},
%\textbf{21},
%2403--2409.


\bibitem[Li et al.(2012)%
Li, Zhong and Zhu]{Li2012}
\textsc{Li, R., Zhong, W. and Zhu, L.} (2012).
Feature screening via distance correlation learning.
J. Am. Stat. Assoc.
\textbf{ 107},
1129--1139.


\bibitem[Liu et al.(2014)
Liu, Li and  Wu]{Liu2014}
\textsc{Liu, J., Li, R., Wu, R.} (2014).
Feature selection for varying coefficient models with ultrahigh-dimensional covariates.
J. Am. Stat. Assoc.
\textbf{109},
266--274.


%
%\bibitem[Liu et al.(2013)%
%Liu, Liang, Xu, Zhang and  Leung]{Liu2013}
%\textsc{ Liu, X. Y., Liang, Y., Xu, Z. B., Zhang, H. and  Leung, K. S.} (2013).
%Adaptive $l_{1/2}$ shooting regularization method for survival analysis using gene expression data.
%\emph{Scientific World Journal},
%\textbf{2},
%Article ID: 475702.




\bibitem[Liu et al.(2018)%
Liu, Zhang and Zhao]{Liu2018}
\textsc{Liu, Y., Zhang, J. and Zhao, X.} (2018).
A new nonparametric screening method for ultrahigh-dimensional survival data.
 Comput. Stat. Data Anal.
\textbf{119},
74--85.




\bibitem[Mai and Zou(2015)%
Mai and Zou]{Mai2015}
\textsc{ Mai, Q. and Zou, H.} (2015).
The fused Kolmogorov filter: A nonparametric model-free screening method.
Ann. Stat.
\textbf{43},
1471--1497.

\bibitem[Minopoli et al.(2019)%
Minopoli, Botti, Gigantino, et al. ]{Minopoli2019}
\textsc{Minopoli, M., Botti, G., Gigantino, V. et al. }(2019)
Targeting the Formyl Peptide Receptor type 1 to prevent the adhesion of ovarian cancer cells onto mesothelium and subsequent invasion.
J. Exp. Clin. Cancer Res.
\textbf{38},
459.


\bibitem[Pan et al.(2019)%
Pan,  Wang, Xiao and Zhu]{Pan2019}
\textsc{Pan, W., Wang, X., Xiao, W. and Zhu, H.} (2019).
A generic sure independence screening procedure.
J. Am. Stat. Assoc.
\textbf{ 114},
928--937.


\bibitem[Rosenwald et al.(2002)%
Rosenwald, Wright,  Chan,  Connors,  Campo,  et al.]{Rosenwald2002}
\textsc{Rosenwald, A., Wright, G., Chan, W., Connors, J., Campo, E., et al.} (2002).
The use of molecular profiling to predict survival after chemotherapy for diffuse large-b-cell lymphoma.
N. Engl. J. Med.
\textbf{346},
1937--1947.

%\bibitem[Song et al.(2020)%
%Song, Park, Oh, Kim, Ryu, Park, Baek,
% Cheng, Lu and Jin]{Song2020}
%\textsc{Song, J., Park, S., Oh, J., Kim, D., Ryu, J. H., Park, W.C., Baek, I.J.,
% Cheng, X., Lu, X. and Jin, E. J.} (2020)
%Nudt7 loss promotes kras^{g12d} crc development.
%\emph{Cancers},
%\textbf{12},
%576.


\bibitem[Song et al.(2014)%
Song, Lu, Ma and Jeng]{Song2014}
\textsc{Song, R., Lu, W., Ma, S. and Jeng, X. J.} (2014).
Censored rank independence screening for high-dimensional survival data.
Biometrika
\textbf{101},
799--814.



\bibitem[Tibshirani(1996)%
Tibshirani]{Tibshirani1996}
\textsc{Tibshirani, R.} (1996).
Regression shrinkage and selection via the Lasso.
J. R. Stat. Soc. Ser. B (Methodological) 
\textbf{58},
267--288.

\bibitem[Tibshirani(2009)%
Tibshirani]{Tibshirani2009}
\textsc{Tibshirani, R.} (2009).
Univariate shrinkage in the Cox model for high dimensional data.
Stat. Appl. Genet. .Mol. Biol.
\textbf{8},
1--18.

\bibitem[Wu and Yin(2015)%
Wu and Yin]{Wu2015}
\textsc{Wu,  Y. and Yin, G.} (2015).
Conditional quantile screening in ultrahigh-dimensional heterogeneous data.
Biometrika
\textbf{102},
65--76.

\bibitem[Yan et al.(2020)%
Yan, Fang, Chen,  et al.]{Yan2020}
\textsc{Yan, S., Fang, J., Chen, Y. et al.}(2020)
Comprehensive analysis of prognostic gene signatures based on immune infiltration of ovarian cancer.
BMC Cancer
\textbf{20},
1205.


\bibitem[Yang et al.(2020)%
Yang,  He, Wu, Deng, Wang, Li and Liu]{Yang2020}
\textsc{Yang, D., He, Y., Wu, B., Deng, Y., Wang, N., Li, M. L. and Liu, Y.} (2020)
Integrated bioinformatics analysis for the screening of hub genes and therapeutic drugs in ovarian cancer.
J. Ovarian Res.
\textbf{13},
10.

\bibitem[Zhang(2010)%
Zhang]{Zhang2010}
\textsc{Zhang, C. H.} (2010).
Nearly unbiased variable selection under minimax concave penalty.
Ann. Stat.
\textbf{38},
894--942.

\bibitem[Zhang et al.(2021)%
Zhang, Liu and Cui]{Zhang2021}
\textsc{Zhang, J., Liu, Y. and Cui, H.} (2021).
Model-free feature screening via distance correlation for ultrahigh dimensional survival data.
Stat. Pap.
\textbf{62},
2711--2738.

\bibitem[Zhang et al.(2017)%
Zhang, Liu and Wu]{Zhang2017}
\textsc{Zhang, J., Liu, Y. and Wu, Y.} (2017).
Correlation rank screening for ultrahigh-dimensional survival data.
 Comput. Stat. Data Anal.
 \textbf{108},
 121--132.

\bibitem[Zhang et al.(2018)%
Zhang,  Yin,  Liu and Wu]{Zhang2018}
\textsc{ Zhang, J., Yin, G., Liu, Y. and Wu, Y.} (2018).
 Censored cumulative residual independent screening for ultrahigh-dimensional survival data.
Lifetime Data Anal.
 \textbf{ 24},
 273--292.

\bibitem[Zhao and Li(2012)%
Zhao and Li]{Zhao2012}
\textsc{Zhao, S. D. and Li, Y.} (2012).
Principled sure independence screening for Cox models with ultra-high-dimensional covariates.
J. Multivar. Anal.
\textbf{105},
397--411.

\bibitem[Zhou and  Zhu(2017)%
Zhou and  Zhu]{Zhou2017}
\textsc{Zhou, T. and  Zhu, L.} (2017).
Model-free feature screening for ultrahigh dimensional censored regression.
Stat. Comput. 
\textbf{27},
947--961.

\bibitem[Zhu et al.(2011)%
Zhu,  Li, Li and Zhu]{Zhu2011}
\textsc{Zhu, L. P., Li, L., Li, R. and Zhu, L. X.} (2011).
Model-free feature screening for ultrahigh-dimensional data.
J. Am. Stat. Assoc.
\textbf{ 106},
1464--1475.

\end{thebibliography}

