\section{Introduction}
% % protein sequence - structure - function
% Proteins play a crucial role in various parts of biological processes, and the ensemble of diverse functioning proteins is the basis of life's activities, such as immune response and metabolism. Such essential and versatile functions of proteins are encoded in protein sequences which are the arrangement of amino acid residues. The sequences determine their structures via complex biophysical interactions between residues and these structures are directly linked to the functions of proteins. Thus, optimizing the protein's function by changing amino acid residues of protein of interest, called protein engineering, has been of great interest in diverse industries such as biofuel~\citep{wen2009protein}, pharmaceuticals~\citep{h2014protein}, and agriculture~\citep{rao2008outlook}.

% % \begin{figure*}[h]
% % \vskip 0.2in
% % \begin{center}
% % \centerline{\includegraphics[width=420pt]{./figures/figure_1_context.pdf}}
% % \caption{An illustrative example of FIM protein engineering. The changed sequences for the target region are generated by generative pLMs or the like, and the structures are altered accordingly.}
% % \label{figure_fim}
% % \end{center}
% % \vskip -0.2in
% % \end{figure*}



% % previous approaches
% One of the representatives of protein sequence design methods is a mutagenesis technique, which gives evolutionarily plausible candidate protein sequence libraries with the help of genetic engineering~\citep{arnold1998design}. However, this approach requires substantial efforts in high-throughput screening experiments. Recently, machine learning-guided protein sequence design strategies have been proposed to achieve a more efficient sequence space search using experimentally acquired labeled data~\citep{yang2019machine}.

\begin{figure*}[ht]
\vskip 0.2in
\begin{center}
\centerline{\includegraphics[width=400pt]{./figures/figure2.pdf}}
\caption{An illustrative example of FIM protein engineering. The changed sequences for the target region are generated by generative pLMs or the like, and the structures are altered accordingly.}
\label{figure_fim}
\end{center}
\vskip -0.2in
\end{figure*}

 Optimizing the protein's function by changing amino acid residues of protein of interest, called protein engineering, has been of great interest in diverse industries such as biofuel~\cite{wen2009protein}, pharmaceuticals~\cite{h2014protein}, and agriculture~\cite{rao2008outlook}. One of the representatives of conventional protein sequence design methods is a mutagenesis technique, which gives evolutionarily plausible protein sequence libraries with the help of genetic engineering~\cite{arnold1998design}. However, this approach requires substantial efforts from high-throughput screening experiments. Recently, machine learning-guided protein sequence design strategies have been proposed to achieve a more efficient sequence space search using experimentally acquired labeled data~\cite{yang2019machine}.


With both advances in high-throughput sequencing technologies and language modeling in the field of natural language processing (NLP), protein language models (pLMs), which are trained in an unsupervised manner using tremendous sets of unlabeled protein sequences~\cite{uniprot2019uniprot}, have been developed for generating \textit{de novo} protein sequence~\cite{madani2020progen, hesslow2022rita, moffat2022design, ferruz2022protgpt2, nijkamp2022progen2}. Existing generative pLMs are trained using an auto-regressive (AR) strategy~\cite{radford2019language, brown2020language}, and generate sequences conditioning on the prefix protein sequences. 

Unfortunately, if the target region where we want to change amino acid residues is located at the front, existing pLMs use only a few preceding amino acid residues (``prompts'') for sequence generation. The interaction sites, positions that interact with other proteins or molecules to perform their functions and are mainly modified to improve functionality, are evenly located on the protein sequence. To prove this, we collect 3D protein structures from Protein Data Bank (PDB) database~\cite{sussman1998protein} and calculate the relative positions of protein-protein interaction sites on the protein sequences (see details in Appendix~\ref{appendix:a1}). As illustrated in Figure~\ref{interaction_dist}, interacting sites are evenly present on the protein sequence. This result suggests that in protein engineering, modifying the amino acid sequence will be done for the middle part of the sequence in many cases. In this case, existing pLMs may not effectively utilize the information behind them, which can result in poor quality of generation.

% \begin{figure*}[t]
%   \centering
%   \includegraphics[width=0.9\linewidth]{./figures/figure2.pdf}
%   \caption{An illustrative example of FIM protein engineering. The changed sequences for the target region are generated by generative pLMs or the like, and the structures are altered accordingly.}
%   \label{figure_fim}
% \end{figure*}





In this work, we regard middle protein engineering as a fill-in-middle (FIM) sequence generation problem as in Figure~\ref{figure_fim} and investigate the possibility of pLMs in FIM protein engineering framework. With the emergence of highly accurate protein structure predictors~\cite{jumper2021highly, baek2021accurate}, protein structures are predicted very quickly and accurately at a low cost. Using these advances, we propose a new evaluation scheme, Secondary structurE InFilling rEcoveRy, SEIFER, for FIM protein sequence generation. The secondary structures are usually desirable to be preserved~\cite{rubio2019redesigning} since the binding pockets of other interacting proteins or molecules are fixed to some extent. In SEIFER, models are tasked to recommend protein sequences and achieve two conditions: the new sequences must be different from the original sequences and their secondary structures must be fully maintained. So, SEIFER can assess both the diversity and structure of new sequences simultaneously and we believe that SEIFER is suitable for assessing generated sequences in the field of protein engineering which modifies the amino acid residues of original sequences to improve functions. Also, inspired by the latest research in the field of language models~\cite{bavarian2022efficient}, we propose a new Protein language model specialized for the Fill-In-Middle task, ProtFIM. Compared to existing pLMs, our proposed ProtFIM uses both front (``prefix'') and back (``suffix'') sequence information during training and inference. 


% Through SEIFER evaluation, we show that ProtFIM can generate diverse sequences while maintaining secondary structure, especially for $\alpha$-helix. Furthermore, ProtFIM outperforms when engineering on residues positioned in the front part of a protein sequence compared to existing pLMs, proving that the FIM training is more suitable for FIM engineering compared to AR pLMs. Finally, through analysis and visualization, we prove that ProtFIM has decent representations of protein sequences and can serve as a sequence optimization tool accompanied by AlphaFold2.  In summary, our contributions are:

Through SEIFER evaluation, we prove that ProtFIM outperforms AR-based pLM, ProtGPT2-C, in infilling protein sequence design. In addition, compared to state-of-the-art released models that were trained with 2-50 times more parameters, ProtFIM exhibits comparable performance, demonstrating the efficacy of infilling modeling in protein engineering. Intriguingly, we discovered that existing pLMs are proficient at only $\alpha$-helix engineering, whereas their performances on $\beta$ and coil structures are similar or worse compared to the random generator, which randomly generates protein sequences for target sites. Furthermore, we provide ablation studies of the $\alpha$-helix SEIFER performance on the relative positions and lengths of target sites, demonstrating that only ProtFIM consistently performs better than the random mutation.
In addition, using protein fitness prediction tasks, we demonstrate that infilling modeling can give a better representation than existing pLMs with a similar number of parameters. Lastly, case studies show that ProtFIM generates promising sequences, proving the effectiveness of infilling modeling in real-world protein engineering. In summary, our contributions are:

% \rian{Through SEIFER evaluation, we prove that ProtFIM outperforms AR-based pLM, ProtGPT2-C, in infilling protein sequence design. In addition, compared to state-of-the-art released models that were trained with 2-50 times more parameters, ProtFIM exhibits comparable performance, demonstrating the efficacy of infilling modeling in protein engineering. Intriguingly, we discovered that existing pLMs are proficient at only $\alpha$-helix engineering, whereas their performances on $\beta$ and coil structures are similar or worse compared to the random generator, which randomly generates protein sequences for target sites. Furthermore, we provide ablation studies of the $\alpha$-helix SEIFER performance on the relative positions and lengths of target sites, demonstrating that only ProtFIM consistently performs better than the random mutation.
% In addition, using protein fitness prediction tasks, we demonstrate that infilling modeling can give a better representation than existing pLMs with a similar number of parameters. Lastly, case studies show that ProtFIM generates promising sequences, proving the effectiveness of infilling modeling in real-world protein engineering. In summary, our contributions are:}


\begin{itemize}

\item {We define FIM protein engineering as protein sequence infilling tasks and provide the applicability of pLMs on the task.}

\item {We propose a new evaluation scheme, SEIFER, that can be used to evaluate the performance of pLMs on protein infilling sequence design tasks by considering structural conservation.}

% \item {We propose a new evaluation scheme, SEIFER, that can be used to evaluate the performance of pLMs on protein infilling sequence design tasks by considering structural conservation.}

\item {We propose a new type of pLM, ProtFIM, that has both AR and FIM capability. Comprehensive results show that ProtFIM has efficient and comparable performances in protein infilling and protein representations for protein engineering compared to other pLMs.}

\item {We show that the ProtFIM acts as a sequence optimizer, which generates novel sequences with high pLDDT of AlphaFold2 while maintaining the structures essential for the function of the protein.}

% parameter 적음에도 성능 좋다 ㄱ문 추가
\end{itemize}

