\documentclass[%
 preprint,
superscriptaddress,
 amsmath,amssymb,
 aps,
]{revtex4-2}

\usepackage{graphicx}%
\usepackage{dcolumn}%
\usepackage{bm}%
\usepackage{amsmath}
\usepackage{amssymb}
\DeclareMathOperator\erf{erf}
\DeclareMathOperator\erfc{erfc}
\usepackage{algorithm}
\usepackage{algpseudocode}

\renewcommand{\thesection}{S\arabic{section}}  
\renewcommand{\thetable}{S\arabic{table}}  
\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\theequation}{S\arabic{equation}}
\renewcommand{\figurename}{Supplemental Material, Figure} 

\begin{document}


\title{Supplemental Material:\\ Biological rhythms generated by a single activator-repressor loop with heterogeneity and diffusion.}

\author{Pablo Rojas}
\affiliation{ Theoretical Physics and Center for Interdisciplinary Nanostructure Science and Technology (CINSaT), University of Kassel, Kassel, Germany}
\author{Oreste Piro}%
\affiliation{Department of Ecology and Marine Resources, Institut Mediterrani d’Estudis Avançats, IMEDEA (CSIC–UIB), Balearic Islands, Spain}
\affiliation{Departament de F{\'i}sica, Universitat de les Illes Balears, Ctra. de Valldemossa, km 7.5, Palma de Mallorca E-07122, Spain}
\author{Martin E. Garcia}
\affiliation{ Theoretical Physics and Center for Interdisciplinary Nanostructure Science and Technology (CINSaT), University of Kassel, Kassel, Germany}%

\maketitle

\tableofcontents



\section{\label{sec:goodwin} Oscillations in Goodwin model }


The Goodwin model consists of a set of ordinary differential equations (ODEs), that represents a negative feedback loop in a chemical reaction network \cite{goodwin65,griffith68}. In this network, the protein at the end of the loop represses its own transcription. For each substance in the loop, the temporal evolution of its amount is described by production and degradation terms. In its simplest version, the only nonlinearity appears in the repression as a Hill function. The model assumes a well-mixed or compartmental condition. Eq. 1 and Eq. 2 from the main text are special cases of the more general $n$ reaction steps version, that reads as in Eq.\ref{eq:goodwin_n}. 

\begin{equation}
\label{eq:goodwin_n}
\begin{aligned}
%
	\frac{dx_1}{dt} & = \omega_{1} \frac{1}{1 + x_n^h} - \gamma_{1}x_1 \\
%
	\frac{dx_2}{dt} & = \omega_{2}x_1 - \gamma_{2}x_2 \\
	: &   \\
	\frac{dx_n}{dt} & = \omega_{n}x_{n-1} - \gamma_{n}x_n \\
\end{aligned}
\end{equation}

The secant condition, derived in \cite{tyson1978dynamics}, is the necessary condition that the exponent $h$ of the Hill function must fulfill to enable oscillatory solutions (Eq.\ref{eq:secant_condition}, Fig.1 in main text).
\begin{equation}
\label{eq:secant_condition}
h \geq \sec^n \left(\frac{\pi}{n} \right)  
\end{equation}

\section{\label{sec:deterministic} Deterministic model}


The modeled system is an extension of the Goodwin model to account for spatial heterogeneity, which transforms the set of ODEs into a set of partial differential equations (PDEs). Cell is represented by a symmetric spatial domain. For the sake of simplicity, no distinction between nucleus and cytoplasm transport properties is considered, i.e. homogenous diffusion coefficient. The membrane of the cell at the borders is impermeable.

The model first considers 2 species, mRNA and protein, whose spatial concentration are represented by  $m(\vec{r},t)$ and $p(\vec{r},t)$, respectively. Transcription of mRNA molecules and translation of proteins similarly to Goodwin model, but spatially restricted with functions $f_{GEN}(\vec{r})$ and $f_{RIB}(\vec{r})$, that represent the gene and ribosome locations, respectively. Linear degradation is considered to occur across the whole domain for both species, therefore it is equivalent to Goodwin model. We represented the molecular transport of mRNA and protein with Fickian diffusion. A parameter $p_{thresh}$ is used in the repression to control the scale of concentration of protein in the Hill function. The equations for the system are:
\begin{eqnarray}
\label{eq:deterministic2elem}
\dot{m}(\vec{r},t)   & = & \frac{\omega_m}{1+ \left(\dfrac{p(\vec{r},t) }{p_{thresh}} \right)^h} \, f_{GEN}(\vec{r}) -\gamma_m \, m(\vec{r},t)   + D_m \, \nabla^2 m(\vec{r},t)    \\
\dot{p}(\vec{r},t) & =& \omega_p \; f_{RIB}(\vec{r}) \, m(\vec{r},t)- \gamma_p \, p(\vec{r},t)   + D_p \nabla^2 \, p(\vec{r},t)    \nonumber 
\end{eqnarray}

This model represents our problem in the three dimensional space. To simplify our analysis we present our result with the reduction to one dimensional domain. Therefore, the equations read as:

\begin{eqnarray}
\label{eq:deterministic2elem1D}
\dot{m}(x,t)   & = & \frac{\omega_m}{1+    \left(\dfrac{p(x,t)  }{p_{thresh}} \right)^h} \, f_{GEN}(x) -\gamma_m \, m(x,t)  + D_m \, \nabla^2 m(x,t)   \nonumber \\
\dot{p}(x,t)   & = & \omega_p \; f_{RIB}(x) \, m(x,t) - \gamma_p \, p(x,t)  + D_p \nabla^2 \, p(x,t)      
\end{eqnarray}

The localization of gene and ribosomes is set to boxcar functions  $f_{GEN}(x)$ and $f_{RIB}(x)$ centered at $x_m$ and $x_p$, respectively, with radii $R_{m}$ and $R_{p}$.

\begin{equation}
\label{eq:fgen}
f_{GEN}(x) = 
    \begin{cases}       
        1 & \text{if } \; x \in [x_m - R_{m},x_m + R_{m} ]    \\
        0 & \text{else }
    \end{cases}
\end{equation}


\begin{equation}
\label{eq:frib}
f_{RIB}(x) = 
    \begin{cases}       
        1 & \text{if } \; x \in [x_p - R_{p},x_p + R_{p} ]    \\
        1 & \text{if } \; x \in [-x_p - R_{p},-x_p + R_{p} ]    \\
        0 & \text{else }
    \end{cases}
\end{equation}

The model as in Eq. \ref{eq:deterministic2elem1D} applies repression to the transcription of mRNA by comparing the values of $p(x,t)$ at each point $x$ with the respective parameter $p_{thresh}$. Due to the nonlinear nature of the Hill function, this differs from the well-mixed formulation in which the total amount is first computed and then compared to the respective parameter. We found these differences to lead to slight quantitative differences, only affecting the exact values for the onset of oscillations but preserving the structure of the solutions. Therefore, unless otherwise stated, the study shows results using Eq. \ref{eq:deterministic2elem1D}. However, for comparison with existing models we resorted to a formulation that compares the value of the repressing species, computed as the amount that is located within the gene location. We called this \emph{volumetric repression} (Eq. \ref{eq:deterministic2elem1Dvolum}).


\begin{eqnarray}
\label{eq:deterministic2elem1Dvolum}
\dot{m}(x,t)   & = & \frac{\omega_m}{1+    \left(\dfrac{P_{GEN}(t)  }{P_{thresh}} \right)^h} \, f_{GEN}(x) -\gamma_m \, m(x,t)  + D_m \, \nabla^2 m(x,t)   \nonumber \\
\dot{p}(x,t)   & = & \omega_p \; f_{RIB}(x) \, m(x,t) - \gamma_p \, p(x,t)  + D_p \nabla^2 \, p(x,t)      ,
\end{eqnarray}

where $ P_{GEN}(t) $ is the amount of $ p(x,t) $ integrated in the gene location (Eq. \ref{eq:Pgen_integral_value}). 

\begin{equation}
\label{eq:Pgen_integral_value}
	P_{GEN}(t)  =  \int_\Omega p(x,t) f_{GEN}(x) dx, 
\end{equation}
where $\Omega$ is the cell volume. 

The solution of the PDEs were obtained numerically using second order finite differences in the spatial domain, and Runge-Kutta-4 in the time domain. 


\section{\label{sec:three_molecules} Feedback loop with three species }

The extension of our model to 3 participating species is done similarly to 2 species. The equations for 3 participating species that is the corresponding to Eq. \ref{eq:deterministic2elem1D} for 2 species, are shown in Eq. \ref{eq:3elem}.


\begin{eqnarray}
\label{eq:3elem}
\dot{m}(x,t)   & = & \frac{\omega_m}{1+   \left(\dfrac{r(x,t)  }{r_{thresh}} \right)^h} \, f_{GEN}(x) -\gamma_m \, m(x,t)  + D_m \, \nabla^2 m(x,t)   \nonumber \\
\dot{p}(x,t)   & = & \omega_p \; f_{RIB}(x) \, m(x,t)- \gamma_p \, p(x,t)  + D_p \nabla^2 \, p(x,t)        \\
\dot{r}(x,t)   & = & \omega_r \; f_{REP}(x) \, p(x,t)- \gamma_p \, r(x,t)   + D_r \nabla^2 \, r(x,t)    \nonumber 
\end{eqnarray}

The function $f_{REP}(x)$ represents the site where the production of the repressor $r$ takes place. To simplify the analysis, $f_{REP}(x)$ is chosen to be identical to $f_{RIB}(x)$ (Eq.\ref{eq:frep}), so that only one distance remains as a parameter of the problem.
 
\begin{equation}
\label{eq:frep}
f_{REP}(x) = f_{RIB}(x) = 
    \begin{cases}       
        1 & \text{if } \; x \in [x_p - R_{p},x_p + R_{p} ]    \\
        1 & \text{if } \; x \in [-x_p - R_{p},-x_p + R_{p} ]    \\        
        0 & \text{else }
    \end{cases}
\end{equation}

The model with 3 species have a better comparison with ODE models when the formulation with volumetric repression is used. Under this framework, Eq. \ref{eq:3elem} is modified as:

\begin{eqnarray}
\label{eq:3elem_volum}
\dot{m}(x,t)   & = & \frac{\omega_m}{1+   \left(\dfrac{R_{GEN}(t)  }{R_{thresh}} \right)^h} \, f_{GEN}(x) -\gamma_m \, m(x,t)  + D_m \, \nabla^2 m(x,t)   \nonumber \\
\dot{p}(x,t)   & = & \omega_p \; f_{RIB}(x) \, m(x,t) - \gamma_p \, p(x,t)  + D_p \nabla^2 \, p(x,t)        \\
\dot{r}(x,t)   & = & \omega_r \; f_{REP}(x) \, p(x,t) - \gamma_p \, r(x,t)   + D_r \nabla^2 \, r(x,t)    \nonumber 
\end{eqnarray}

where 

\begin{equation}
\label{eq:Rgen_integral_value}
	R_{GEN}(t)  =  \int_\Omega r(x,t) f_{GEN}(x) dx, 
\end{equation}

In the examples shown for 3 species, we used volumetric repression.

\begin{figure}[h]
    \includegraphics[width=6.6in]{panel_3_elem_2_parameter_sets.pdf}
    \caption{ Amplitude of oscillations in an extension of the model for 3 species. Left panel: Parameter Set 1 described in the main text. Right panel: Parameter Set 2. Lines indicate minimum and maximum of the trajectories in the long term approximation. Shaded areas between minimum and maximum indicate amplitude of oscillations. Parameter Set 1 only displays oscillations when there is a separation between the sources. Parameter Set 2 shows oscillations when sources are co-localized (as predicted by the well-mixed model) and when sources are separated. Notice the intermediate region when no oscillation is present. Oscillations in the separation case present higher amplitude.  }
    \label{fig:3elem_loop}
\end{figure}

The problem is solved for 2 sets of parameters. In the Parameter Set 1, parameters for the transport of substances are selected according to the ones reviewed and used in \citep{fonkeu2019mrna,sturrock2011spatio}. These parameters represent experimental measurement of diffusion coefficients and kinetic rates for mRNA and protein. Since in this work reviews other sources of experimental data and reports a wide variability on the measurements, we simplified the choice of parameters by using the corresponding order of magnitude. In the Parameter Set 2, we selected kinetic parameters that would generate oscillations in the well-mixed Goodwin Model, inspired in Ref. \citep{gonze2013goodwin}, in order to induce an equivalence to the corresponding well-mixed version under low values of $x_p$. Full set of parameters in Table \ref{tab:param3elem}.


\section{\label{sec:stochastic} Stochastic model }

We constructed a stochastic model of the coupled reaction and diffusion of the systems, by treating each of the molecules as an individual agent, as shown in Fig. \ref{fig:stochastic_scheme}. The Eq. \ref{eq:SSA} represents the sets of equations that represent the creation, degradation and displacements of the molecules. In these equations, $\phi$ denotes chemical species that are of no interest, and is used in order to represent degradation and creation of molecules. To solve this system, we used the stochastic simulation algorithm (SSA) described in Ref. \cite{erban2020}. 
\begin{eqnarray}
\label{eq:SSA}
mRNA     & \rightarrow   & \phi        \nonumber  \\
protein  & \rightarrow   & \phi         \nonumber \\
\phi     & \rightarrow  & mRNA           \\
\phi     & \rightarrow    & protein     \nonumber \\
x_i(t+\Delta t) & =&  x_i(t ) + \sqrt{2D \Delta t} \xi , \nonumber 
\end{eqnarray}
In our SSA, we used a time step $\Delta t$, small enough so that the probability of a certain reaction ocurring during this interval of time is $k \Delta t$, $k$ being its reaction rate. We therefore generate random numbers at each time step to determine if an individual molecule is degraded. We also determine whether a new molecule of a certain species must be created, in which case it will be positioned at a random location within its own creation zone. During the same interval, an existing molecule can update its position by determining its stochastic displacement $\sqrt{2D \Delta t} \xi$, where $ \xi $ is a number drawn from a Gaussian distribution. We therefore determine, at each time step, whether an individual molecule is degraded, created and where its new position is, if corresponding (Alg. \ref{alg:SSA_steps}). The reaction rates are determined using the equivalent expresions from the deterministic model (Eq. \ref{eq:reaction_rates_SSA}).
\begin{eqnarray}
\label{eq:reaction_rates_SSA}
k_{degradation \, mRNA}     & =   &  \gamma_m        \nonumber  \\
k_{degradation \, protein}  & =   &  \gamma_p         \nonumber \\
k_{creation \, mRNA}     & =   & \frac{\omega_m}{1+    \left(\frac{protein_{transcription \; zone}  }{protein_{thresh}} \right)^h}       \\
k_{creation \, protein}  & =   &  \omega_p \quad {mRNA}_{translation \, zone} \,        \nonumber 
\end{eqnarray}


\begin{figure}
\begin{algorithm}[H]
\caption{Stochastic model steps. }\label{alg:SSA_steps}
\begin{algorithmic}
\State set initial condition and parameters
\While{$t \leq t_{max}$} 
	\State determine mRNA (protein) molecules in translation (transcription) zone
	\State compute reaction rates and probabilities
	\For{each molecule}
		\State generate random numbers	
		\State compute displacements
		\State update positions
	\EndFor
	\For{each molecule}
		\State generate random number
		\If{$random \quad number \leq probability \quad degradation$}
		    \State eliminate molecule
    		\EndIf
	\EndFor
	\For{each species (mRNA and Protein)}
		\State generate random number
		\If{$random \quad number \leq probability \quad creation$}
		    \State insert new molecule in random position within production zone
    		\EndIf
	\EndFor
\EndWhile
\end{algorithmic}
\end{algorithm}
\end{figure}



\begin{figure}[h!]
    \includegraphics[width=6.0in]{diagram_stoch_scheme_borders.png}
    \caption{ Scheme of the stochastic model. Molecules are modeled as individual particles. They are created either at gene (transcription zone) or ribosome (translation zone), similarly to deterministic model. To create molecules, the stochastic simulation algorithm (SSA) is implemented with a propensity function based on the number of molecules of the other species in their own production zone. Molecules travel following the Langevin equation, and are reflected at the borders of the cell. The results shown in Fig. 4 in the main text where obtained with a 1D version of this model. }
    \label{fig:stochastic_scheme}
\end{figure}

\section{\label{sec:relaxation} Derivation of relaxation times }

\subsection*{Reduced problem}
We first consider a symmetric one dimensional domain, in which for $t\geq0$ the production of a substance whose concentration is $c(x,t)$ is done at a constant rate $\omega$ in a punctual source centered in the interval $[-L,L]$, represented by $\delta(x)$. At $t=0$, $c(x,t=0)=0$ in the whole spatial domain. This substance is subjected to Fickian diffusion (with a constant diffusion constant $D$), and degradation (with $\gamma$ as degradation constant). The walls of the domain $[-L,L]$ are impermeable. For $t<0$ the system is at its equilibrium state $c(x)=0$ since no source is available, and this equilibrium is perturbed by the sudden appearance of the punctual source. This scenario is an idealized cartoon of the instant in which hypothetical mRNA (protein) initiates transcription (translation) inside a cell, without prior concentration. The new concentration as approaches equilibrium at long times. We will derived a characteristic time that describes the relaxation of the concentration at a certain spatial point, from its initial condition $c=0$ to the new established profile $c=c_{t \rightarrow \infty}$. The equation for the reduced problem reads:

\begin{equation}
\label{eq:finite_problem}
%
%
\left\{
\begin{aligned}
	\frac{\partial c}{\partial t} 			&= D  \nabla^2 c - \gamma c + \omega  \delta (x) 
													& \quad \forall x \mbox{ in } [-L,L]  \\
	\left.\frac{\partial c}{\partial x} \right|_L 	&= \left.\frac{\partial c}{\partial x} \right|_{-L} = 0  \\
	c(x,0) &= 0 & \forall x \mbox{ in } [-L,L]\\
\end{aligned} 
\right.
%
\end{equation}

Since the problem is symmetric, we can rewrite the system for the positive sub-interval, by transforming the production term into a boundary condition $\left( \left.\frac{\partial c}{\partial x} \right|_0 = -\frac{\omega}{2D}\right)$.

\begin{equation}
\left\{
\begin{aligned}
	\frac{\partial c}{\partial t} 			&= D  \nabla^2 c - \gamma c  
													& \quad \forall x \mbox{ in }[0,L]  \\
	\left.\frac{\partial c}{\partial x} \right|_0 	&= -\frac{\omega}{2D}  \\
	\left.\frac{\partial c}{\partial x} \right|_L 	&= 0  \\
	c(x,0) &= 0 & \forall x \mbox{ in }[0,L]\\
\end{aligned} 
\right.
\end{equation}

The solution of the problem is simpler if we assume the limit $L \rightarrow \infty$. Since our purpose is to derive estimates analytically, this assumption proves to be practical and does not affect the conclusions. For $L \rightarrow \infty$ the problem reads:


\begin{equation}
\label{eq:infinite_symm_problem}
\left\{
\begin{aligned}
	\frac{\partial c}{\partial t} 			&= D  \nabla^2 c - \gamma c  
													& \quad \forall x \mbox{ in }[0,\infty)  \\
	\left.\frac{\partial c}{\partial x} \right|_0 	&= -\frac{\omega}{2D}  \\
	\left.\frac{\partial c}{\partial x} \right|_{x \rightarrow \infty} 	&= 0  \\
	c(x,0) &= 0 & \forall x \mbox{ in }[0,\infty)\\
\end{aligned} 
\right.
\end{equation}


The solution to the steady state problem, i.e. $ \frac{\partial c}{\partial t} = 0$ is asymptotically approached by the transient solution at long times $t \rightarrow \infty$. The steady state solution becomes:
\begin{equation}
\label{eq:steady_state_solution}
c_{t \rightarrow \infty}= \frac{\omega}{2 \sqrt{D \gamma}} e^{-\sqrt{\frac{\gamma}{D}} x} 
\end{equation}

To obtain the transient solution, we will work with the Laplace transform $u(x,s) = \mathcal{L}[c(x,t)]$. The resulting differential equation is:

\begin{equation}
\label{eq:laplace_ODE}
	\frac{\partial^2 u}{\partial x^2} - \left( \frac{s+\gamma}{D} \right) u = 0 
\end{equation}
and we can propose a solution of the form:
\begin{equation*}
	u(x,s)= C_1 e^{-\sqrt{\frac{s +\gamma}{D}} x} 
\end{equation*}
 
By applying the boundary condition at $x=0$:

\begin{equation*}
	\frac{\partial u}{\partial x}=   \mathcal{L} \left[ \frac{\partial c}{\partial x}\right]  
\end{equation*}
\begin{equation*}
	C_1 \left( -\sqrt{\frac{s +\gamma}{D}} \right) = -\frac{\omega}{2D \, s}
\end{equation*}
\begin{equation*}
	\Rightarrow C_1  =  \frac{\omega}{2 \sqrt{D} \, s \, \sqrt{s +\gamma}} 
\end{equation*}

Then, the solution for $u$ is: 
\begin{equation}
\label{eq:solution_u}
	u(x,s)= \frac{\omega}{2 \sqrt{D} \, s \, \sqrt{s +\gamma}} e^{-\sqrt{\frac{s +\gamma}{D}} x} 
\end{equation}

To obtain $c(x,t)$, it is necessary to anti-transform the solution by $c(x,t) = \mathcal{L}^{-1}[ u(x,s)]$. 

\begin{equation}
\label{eq:antitransform_c}
	c(x,t)= \mathcal{L}^{-1} \left[ \frac{\omega}{2 \sqrt{D} \, s \, \sqrt{s +\gamma}} e^{-\sqrt{\frac{s +\gamma}{D}} x} \right]
\end{equation}

It is possible to antitransform first the factors $\mathcal{L}^{-1} \left[ \frac{1}{s} \right] $ and $\mathcal{L}^{-1} \left[ \frac{e^{-\sqrt{\frac{s +\gamma}{D}} x}}{\sqrt{s +\gamma}} \right] $, and use these results to arrive to the solution of $c(x,t)$.

\begin{equation}
\label{eq:solution_analytic_with_integral}
	c(x,t)=  \frac{\omega}{2 \sqrt{D \pi } } \int_{0}^{t} \frac{e^{- \left( \gamma y + \frac{x^2}{4Dy} \right)}}{\sqrt{y}} dy
\end{equation}

Solving the integral:

\begin{equation}
c(x,t) = \frac{\omega}{4\sqrt{D \gamma}} \left[ e^{-\sqrt{\frac{\gamma}{D}} x}  \erfc\left( \frac{x}{2\sqrt{D t}} - \sqrt{\gamma t} \right)   - e^{\sqrt{\frac{\gamma}{D}} x}  \erfc\left( \frac{x}{2\sqrt{D t}} + \sqrt{\gamma t} \right) \right] 
\label{solution_analytic}
\end{equation}

This solution is displayed in Fig.\ref{fig:delays_analytic}a for different times. 

\subsection*{Relaxation time}
The time at which $\frac{\partial c}{\partial t}$ maximizes can be obtained by the solution of:

\begin{equation}
\label{eq:maximal_c_rate}
	\frac{\partial }{\partial t} \left( \frac{\omega}{2 \sqrt{D \pi } }  \frac{e^{- \left( \gamma t + \frac{x^2}{4Dt} \right)}}{\sqrt{t}} \right) = 0 
\end{equation}

From which we can obtain:

\begin{equation*}
	\frac{\partial }{\partial t} \left(  \frac{e^{- \left( \gamma t + \frac{x^2}{4Dt} \right)}}{\sqrt{t}} \right) = \left( - \frac{e^{- \left( \gamma t + \frac{x^2}{4Dt} \right)}}{\sqrt{t}} \right) \frac{1 }{t^{3/2} } \left(      \gamma t - \frac{x^2}{4D t} + \frac{ 1 }{2} \right) 	 = 0
\end{equation*}

Therefore, the \emph{relaxation time} $t_{R}$ results as a solution of:
\begin{equation}
\label{eq:equation_relax_time}
t_{R}^2 +\frac{t_{R}}{2\gamma} - \frac{x^2}{4D\gamma} = 0
\end{equation}

Since we are interested in the interval $t\geq 0$, one of the solutions is ruled out and we obtain:
\begin{equation}
\label{eq:solution_relax_time}
t_{R} = \frac{1}{4\gamma}   \left(  \sqrt{1 + 4  \left(\frac{x}{\lambda}\right)^2 } -1  \right)
\end{equation}

where $ \lambda  = D/\gamma$ is the characteristic length. For values of $x/\gamma$ large enough, $t_{R}$ approaches a linear dependence on $x$ (Fig.\ref{fig:delays_analytic}b-c) (notice the difference with the $x^2/D$ dependence of time-scales in purely diffusive systems).


\subsection*{Dimensionless version of the reduced problem}

A change of variables that render the reduced problem as dimensionless can help the relevant parameters. The change of variables:

\begin{center}
$
\begin{array}{rllcrl}
	x^\ast & = \frac{x}{\lambda} & = \frac{x}{\sqrt{\frac{D}{\gamma}}} & \Rightarrow & \; x & =  x^* \sqrt{\frac{D}{\gamma}} \\
	t^* & = \frac{2 t \sqrt{D \gamma}}{x}  & = \frac{2 t }{ \frac{x}{\lambda \gamma  } } & \Rightarrow & \; t & =  \frac{t^* x^*}{2 \gamma}
\end{array}
$
\end{center}

yields the steady state solution ($t\rightarrow \infty$)  re-written as:

\begin{center}
$
\begin{array}{rlcrl}
	c^* & = \frac{c}{\frac{\omega}{2\sqrt{D \gamma}}} &  &  &  \\
	c_{t\rightarrow \infty}(x) & =  \frac{\omega}{2\sqrt{D \gamma}} e^{-\sqrt{\frac{\gamma}{D}} x} & \Rightarrow & \; c_{t\rightarrow \infty}^*(x^*) & =  e^{-x^*}
\end{array}
$
\end{center}

For the time-varying solution the dimensionless version yields:
\begin{equation}
\label{eq:dimensionless_solution}
c^*(x^*,t^*) = \frac{1}{2} \left[ e^{-x^*}  \erfc\left( \frac{1}{\sqrt{2}} \left[ \sqrt{\frac{x^*}{t^*}} - \sqrt{x^* t^* } \right] \right)   - e^{x^*}  \erfc\left( \frac{1}{\sqrt{2}} \left[ \sqrt{\frac{x^*}{t^*}} + \sqrt{x^* t^* } \right] \right) \right] 
\end{equation}

therefore, the problem scales with $x$ proportional to lambda $\lambda$, but in time $t$ it scales with $x/\lambda$ and inversely to $\gamma$.


\subsection*{Linear dependence of delay on distance in  oscillatory solutions}

The spatio-temporal distribution of $m(x,t)$ and $p(x,t)$ in oscillatory solutions (Fig.\ref{fig:heatmap_space_time}) described in the main text, shows an almost constant speed of propagation , in agreement with the derived dependence on $x$. 
For sufficiently small or sufficiently large separation of the sources, the spatial distribution of $m(x,t)$ and $p(x,t)$ remain constant over time (Fig.\ref{fig:heatmap_space_time}a,d).
However, for intermediate separations, these distributions display oscillatory behavior in time (Fig. \ref{fig:heatmap_space_time}b,d).
The period of oscillations increases as the distance between sources increases. Propagation in the spatial distribution of species are recognizable as the slanted brighter regions. The slope of the regions denotes the constant propagation speed. 


\begin{figure}[h]
    \includegraphics[width=6.6in]{delays_in_analytic.pdf}
    \caption{ Relaxation times in the reduced problem.  (a) Concentration of the substance $c$ as a function of $x$ at different times. The spatial profile converges asymptotically to the steady state. (b) Normalized concentration as a function of re-scaled time. Dots represent relaxation times, defined as the times in which the rate of increase of concentration is maximal. (c) Analytical solution for relaxation times show asymptotically linear dependence with distance to the source. }
    \label{fig:delays_analytic}
\end{figure}

\begin{figure}[h]
    \includegraphics[width=6.6in]{delays_in_PDE.pdf}
    \caption{ Propagation of concentration gradients arising from spatial separation of the sources. (a) Distribution of the species $m(x,t)$ and $p(x,t)$ in space and time, between center and borders of the cell, for different separation of sources. Concentrations are shown in normalized logarithmic scale. The two displayed oscillatory solutions ($x_p = 1.5 \lambda$ and  $x_p = 3 \lambda$) show different periods of oscillation, but the same dependence of diffusion-induced delays with distance (slope of the bright regions). The parameters for this figure correspond to the ones in Fig.4 in main text. $h= 10 $, $\lambda = \lambda_m = \lambda_p =1$, $ D_m = D_p = 0.1$ , $ \gamma_m = \gamma_p = 0.1 $, $ R_{m} = R_{p} = 0.5$, $ R_{cell} = 10$ , $\omega_m = \omega_p = 10 $  } 
    \label{fig:heatmap_space_time}
\end{figure}

\section{\label{sec:params} Parameters }

Parameters used in the models with two species are given in arbitrary units (Table \ref{tab:param2elem}), with the exception that the amounts can be interpreted in molecules. Parameters in models for three species are selected based on previous work with their corresponding units (Table \ref{tab:param3elem}). Nevertheless, parameters in models with two species could be interpreted as having the same units as the models for three species, i.e. $\mu m$ for sizes and distances, $h$ for time and their corresponding derived units. 

\begin{table}[h!]
\caption{\label{tab:param2elem}
Parameters used in the models with 2 species. Values are given in arbitrary units. }
\begin{ruledtabular}
\begin{tabular}{lcccccccccc}
 &$ D_{m}$  & $D_{p}$ &  $ \gamma_{m}$  & $ \gamma_{p}$& $ \omega_{m}$  & $ \omega_{p}$&$R_{cell}$ &$ R_{m}$  & $R_{p}$& $p_{thresh}$ \\
\hline
Fig. 2-3& 0.1 & 0.1 & 0.1 & 0.1 & 10.0 & 20.0 & 10.0 & 0.5 & 0.5 & 0.1\\
Fig. 4  & 0.1 & 0.1 & 0.1 & 0.1 & 10.0 & 10.0 & 10.0 & 0.5 & 0.5 & 0.1\\
\end{tabular}
\end{ruledtabular}
\end{table}



\begin{table}[h!]
\caption{\label{tab:param3elem}
Parameters used in the models with 3 species. }
\begin{ruledtabular}
\begin{tabular}{lcccccccccccccc}
 &$ D_{m}$  & $D_{p}$ &$ D_{r}$ &  $ \gamma_{m}$  & $ \gamma_{p}$ &  $ \gamma_{r}$ & $ \omega_{m}$  & $ \omega_{p}$ & $ \omega_{r}$  &$R_{cell}$ &$ R_{m}$  & $R_{p}$  &$ R_{r}$ & $r_{thresh}$ \\
 &$  \left(\frac{\mu m ^2}{h}\right)$  & $  \left(\frac{\mu m ^2}{h}\right)$ &$  \left(\frac{\mu m ^2}{h}\right)$ &  $  \left(\frac{1}{h}\right)$   & $  \left(\frac{1}{h}\right)$  &  $  \left(\frac{1}{h}\right)$  & $  \left(\frac{mol.}{h\,\mu m}\right)$   & $  \left(\frac{mol.}{h\,\mu m}\right)$ & $  \left(\frac{mol.}{h\,\mu m}\right)$  &$ (\mu m) $ & $ (\mu m) $   & $ (\mu m) $   & $ (\mu m) $ & $ (mol.) $  \\
\hline
Param. Set 1 & 10.0 & 10.0 & 10.0 & 1.0 & 1.0 & 1.0 & 50.0 & 100.0 & 100.0 & 10.0 & 0.5 & 0.5 & 0.5 & 2.0\\
Param. Set 2 & 0.01 & 0.01 & 0.01 & 0.1 & 0.1 & 0.1 & 5.0  & 10.0  & 10.0  & 4.0  & 0.5 & 0.5 & 0.5 & 1.0\\
\end{tabular}
\end{ruledtabular}
\end{table}

\bibliography{supp_text}

\end{document}
