% Template for PLoS
% Version 3.5 March 2018
%
% % % % % % % % % % % % % % % % % % % % % %
%
% -- IMPORTANT NOTE
%
% This template contains comments intended 
% to minimize problems and delays during our production 
% process. Please follow the template instructions
% whenever possible.
%
% % % % % % % % % % % % % % % % % % % % % % % 
%
% Once your paper is accepted for publication, 
% PLEASE REMOVE ALL TRACKED CHANGES in this file 
% and leave only the final text of your manuscript. 
% PLOS recommends the use of latexdiff to track changes during review, as this will help to maintain a clean tex file.
% Visit https://www.ctan.org/pkg/latexdiff?lang=en for info or contact us at latex@plos.org.
%
%
% There are no restrictions on package use within the LaTeX files except that 
% no packages listed in the template may be deleted.
%
% Please do not include colors or graphics in the text.
%
% The manuscript LaTeX source should be contained within a single file (do not use \input, \externaldocument, or similar commands).
%
% % % % % % % % % % % % % % % % % % % % % % %
%
% -- FIGURES AND TABLES
%
% Please include tables/figure captions directly after the paragraph where they are first cited in the text.
%
% DO NOT INCLUDE GRAPHICS IN YOUR MANUSCRIPT
% - Figures should be uploaded separately from your manuscript file. 
% - Figures generated using LaTeX should be extracted and removed from the PDF before submission. 
% - Figures containing multiple panels/subfigures must be combined into one image file before submission.
% For figure citations, please use "Fig" instead of "Figure".
% See http://journals.plos.org/plosone/s/figures for PLOS figure guidelines.
%
% Tables should be cell-based and may not contain:
% - spacing/line breaks within cells to alter layout or alignment
% - do not nest tabular environments (no tabular environments within tabular environments)
% - no graphics or colored text (cell background color/shading OK)
% See http://journals.plos.org/plosone/s/tables for table guidelines.
%
% For tables that exceed the width of the text column, use the adjustwidth environment as illustrated in the example table in text below.
%
% % % % % % % % % % % % % % % % % % % % % % % %
%
% -- EQUATIONS, MATH SYMBOLS, SUBSCRIPTS, AND SUPERSCRIPTS
%
% IMPORTANT
% Below are a few tips to help format your equations and other special characters according to our specifications. For more tips to help reduce the possibility of formatting errors during conversion, please see our LaTeX guidelines at http://journals.plos.org/plosone/s/latex
%
% For inline equations, please be sure to include all portions of an equation in the math environment.  For example, x$^2$ is incorrect; this should be formatted as $x^2$ (or $\mathrm{x}^2$ if the romanized font is desired).
%
% Do not include text that is not math in the math environment. For example, CO2 should be written as CO\textsubscript{2} instead of CO$_2$.
%
% Please add line breaks to long display equations when possible in order to fit size of the column. 
%
% For inline equations, please do not include punctuation (commas, etc) within the math environment unless this is part of the equation.
%
% When adding superscript or subscripts outside of brackets/braces, please group using {}.  For example, change "[U(D,E,\gamma)]^2" to "{[U(D,E,\gamma)]}^2". 
%
% Do not use \cal for caligraphic font.  Instead, use \mathcal{}
%
% % % % % % % % % % % % % % % % % % % % % % % % 
%
% Please contact latex@plos.org with any questions.
%
% % % % % % % % % % % % % % % % % % % % % % % %

\documentclass[10pt,letterpaper]{article}
\usepackage[top=0.85in,left=2.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8x]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

\usepackage{multirow}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
\raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in 
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\renewcommand{\figurename}{Fig}

% Use the PLoS provided BiBTeX style
\bibliographystyle{plos2015}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\usepackage{makecell}

% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
\fancyfootoffset[L]{2.25in}
\lfoot{\today}

%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}

%% END MACROS SECTION


\begin{document}
\linenumbers
\pagenumbering{arabic}
\vspace*{0.2in}

% Title must be 250 characters or less.
\begin{flushleft}
{\Large
\textbf\newline{Identifying promising candidate radiotherapy protocols via GPU-GA~\textit{in-silico} } % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).
}
\newline
% Insert author names, affiliations and corresponding author email (do not include titles, positions, or degrees).
\\
Wojciech Ozimek\textsuperscript{1},
Rafał Banaś\textsuperscript{2},
Paweł Gora\textsuperscript{3},
Simon D. Angus\textsuperscript{4,*},
Monika J. Piotrowska\textsuperscript{5},
%with the Lorem Ipsum Consortium\textsuperscript{\textpilcrow}
\\
\bigskip
\textbf{1} Ardigen SA, 76 Podole Street, 30-394 Krakow, Poland
\\
\textbf{2} NVIDIA Corporation, Al. Chmielna 73, 00-801, 00-001 Warsaw, Poland
\\
\textbf{3} Institute of Informatics, University of Warsaw,
Banacha 2, 02-097 Warsaw, Poland
\\
\textbf{4} Department of Economics and SoDa Laboratories, Monash Business School, Monash University, Wellington Rd, Clayton, 3800, Australia
\\
\textbf{5} Institute of Applied Mathematics and Mechanics, University of Warsaw, Banacha 2, 02-097 Warsaw, Poland
\bigskip

% Insert additional author notes using the symbols described below. Insert symbol callouts after author names as necessary.
% 
% Remove or comment out the author notes below if they aren't used.
%
% Primary Equal Contribution Note
%\Yinyang These authors contributed equally to this work.

% Additional Equal Contribution Note
% Also use this double-dagger symbol for special authorship notes, such as senior authorship.
%\ddag These authors also contributed equally to this work.

% Current address notes
%\textcurrency Current Address: Dept/Program/Center, Institution Name, City, State, Country % change symbol to "\textcurrency a" if more than one current address note
% \textcurrency b Insert second current address 
% \textcurrency c Insert third current address

% Deceased author note
%\dag Deceased

% Group/Consortium Author Note
%\textpilcrow Membership list can be found in the Acknowledgments section.

% Use the asterisk to denote corresponding authorship and provide email address in note below.
* simon.angus@monash.edu.au

\end{flushleft}
% Please keep the abstract below 300 words
\section*{Abstract}
Around half of all cancer patients, world-wide, will receive some form of radiotherapy (RT) as part of their treatment. And yet, despite the rapid advance of high-throughput screening to identify successful chemotherapy drug candidates, there is no current analogue for RT protocol screening or discovery at any scale. Here we introduce and demonstrate the application of a high-throughput/\-high-fidelity coupled tumour--irradiation simulation approach, we call `GPU-GA', and apply it to human breast cancer analogue -- EMT6/Ro spheroids. By analysing over 9.5 million candidate protocols, GPU-GA yields significant gains in tumour suppression versus prior state-of-the-art high-fidelity/\-low-throughput computational search under two clinically relevant benchmarks. By extending the search space to hypofractionated areas ($>2$~Gy/day) yet within total dose limits, further tumour suppression of up to 33.7\% compared to state-of-the-art is obtained. GPU-GA could be applied to any cell line with sufficient empirical data, and to many clinically relevant RT considerations.


% Please keep the Author Summary between 150 and 200 words
% Use first person. PLOS ONE authors please skip this step. 
% Author Summary not valid for PLOS ONE submissions.   
\section*{Author summary}
Radiotherapy remains a key tool the clinician's toolbox to fight cancer. But the pattern of radiotherapy application over a standard few week treatment period (i.e. dose and timing) can vary in millions or even billions of ways, far too numerous to test exhaustively given the high costs and risks associated with clinical studies. Here, we show how high performing candidate radiotherapy protocols can be discovered by the application of novel computational techniques to a high-fidelity numerical model of EMT6/Ro spheriods, a human breast cancer analogue. In effect, our method brings high throughput screening to the vast radiotherapy search problem. By analysing over 9.5 million candidate protocols, the GPU-GA approach identifies significant potential improvements in tumour suppression outcomes over state-of-the-art approaches. And, by extension to a larger search space, generates protocols with much larger gains. We also run a number of trials to identify the characteristics that are driving the best candidates identified. Our work shows how combining high-throughput screening with high-fidelity computational modelling can significantly advance candidate identification at a fraction of the costs.
\linenumbers

% Use "Eq" instead of "Equation" for equation citations.
\section*{Introduction}
Radiotherapy (RT), and specifically small-dose or fractionated RT, is widely considered to be a cost effective, and potent tool in the treatment of many varieties of cancer with 40 to 50\% of cancer patients receiving some form of RT during their treatment, regardless of the relative prosperity of their context~\cite{Thompson2018}. However, clinical treatment protocols -- the schedule of RT dose size and timing -- are remarkably \emph{singular} in practice despite the wide range of cancers and their pathologies in the human body: `2~Gy per day' being perhaps \emph{the} signature protocol of the last few decades~\cite{Rosenstein:2004vs,ORourke:2009,BoardoftheFacultyofClinicalOncology:2006wq}. Encouragingly, long-running and laborious clinical trials~\cite{Haviland2013,Whelan2010} of slight variations in protocol design (e.g. `3~Gy per day') have shown benefits leading to practice-changing adoptions in clinical settings. As such, given the high cost, and long delays inherent in gathering robust clinical evidence on candidate protocols, there is a pressing need for pre-clinical protocol candidate screening and discovery such that clinical trials examine the most promising alternative candidates. However, away from the tiny number of standard protocols, combinatorial considerations mean that there is an overwhelming number of ways to create a fractionated RT protocol. Consider that, with an alphabet of just five fractional dose sizes, and 10 possible wait-times for the next fraction, a 10 fraction protocol can be constructed in any one of almost $1\times 10^{17}$ ways ($(10\times 5)^{10}$). To make any progress in such a vast search space, any pre-clinical discovery strategy must deliver \emph{high-throughput} screening coupled to \emph{high-fidelity} modelling.	

Here we introduce and demonstrate the validation of a high-fidelity, high-performance computational model of human breast cancer analogue, EMT6/Ro spheroids, leveraging state-of-the-art computational approaches on Graphical Processor Unit (GPU) architectures coupled to GA search, an approach we call `GPU-GA', and apply this approach to optimal radiotherapy scheme discovery, at scale. By analysing over 9.5 million candidate protocols, the GPU-GA approach yields improved tumour suppression outcomes of between 4.0\% and 4.5\% versus prior state-of-the-art computational search outcomes under two clinically relevant benchmark constraints, respectively. Significantly, by extending the search space beyond the state-of-the-art constraints, we identify candidate protocols with improved fitness scores of as much as 33.7\% over the best protocols identified in the past work~\cite{Angus2014} under the benchmark conditions studied. By pooling all highly effective protocols discovered in the GPU-GA search activity, we enumerate, and model, the frequency surface of these potent [dose,delay] sub-modules, finding an imputed maximum potency frequency location of 1.4~Gy fraction administered with a 15.5~h delay (after interpolation). Finally, by leveraging the outcomes of the response surface activity, we demonstrate how hand-crafted candidate protocols composed of the most potent sub-modules can be formed and tested, in the framework of computationally-supported treatment design.

Our work extends a previous rather low throughput/high-fidelity methodology~\cite{Angus2014} and builds on the promising field of `computational oncology' (c.f.~\cite{Gerlee2007,Patel2001,Dormann2002,Kansal2000,Kansal2000a,Kempf2015}) which aims to offer a~better understanding of cancer dynamics {\it `in silico'} being the base for improving therapeutic outcomes {\it in vivo}. Importantly, we based on our previous work (c.f.~\cite{Piotrowska:2009gj,Angus:2010mo,Angus:2013id}) where we show that computational cellular automaton (CA)~models of EMT6/Ro spheroid growth can be successfully calibrated to a broad range of experimental data including 18 independent experimental multi-fraction studies~\cite{Otsuka2011,Sugie2006}. Achieving that designed model required  direct modelling of cell metabolism (depending on the local environmental conditions), cell division, DNA damage and repair processes following known biological systems. This provides an extremely promising baseline for the introduction of radiotherapy and/or chemotherapy modules to investigate therapeutic approaches. To this point, of the few authors that have attempted to implement a~computational tumour with a~therapeutic model 
(e.g.~\cite{Zacharaki2004,Stamatakos2006,Kempf2010, Powathil2012,Powathil2013,Kempf2013}) none of them have fully, if at all, calibrated the computational model to experimental data nor applied a~systematic search of the possible treatment protocols. On the other hand, our quasi-3D CA models of avascular tumour growth, built over several years and rigorously calibrated and tested at each stage of development, provide a rich, \emph{high-fidelity} computational model of tumour growth under irradiation.

Central to the high throughput approach, we develop in the present work, is the translation of an existing high-fidelity EMT6/Ro CA model of avascular tumour growth from its parallelised, classical CPU-based {\sc Matlab} language implementation into a parallelised GPU-architecture C++ implementation, delivering c. 717x speed up in candidate protocol evaluation times. As such, we leverage similar approaches in the rapidly emerging field of high-throughput screening (HTS), where a mixture of processor-, big-data-, statistical search/learning tools-, and novel computational architectures are converging to deliver rapid breakthroughs in molecular candidate identification~\cite{Liu2017,Lightbody2019}.

Taken together, we demonstrate the power of GPU-GA computational biology to rapidly and effectively search within a vast solution space for candidate protocols that could be explored in the clinic. Indeed, GPU-GA has much potential for computational oncology research, as it could be applied to any cell line with sufficiently available validation data, and could lead to the rapid identification of radiotherapy schemes to be tried in the clinical setting.

%\begin{eqnarray}
%\label{eq:schemeP}
%	\mathrm{P_Y} = \underbrace{H(Y_n) - H(Y_n|\mathbf{V}^{Y}_{n})}_{S_Y} + %\underbrace{H(Y_n|\mathbf{V}^{Y}_{n})- H(Y_n|\mathbf{V}^{X,Y}_{n})}_{T_{X\rightarrow Y}},
%\end{eqnarray}



\section*{Results}


\subsection*{Defining the search problem}

Following previous work \cite{Angus2014}, we focus our attention on two clinically relevant benchmark protocols \cite{Rosenstein:2004vs,Bentzen2008, Whelan2002}, both delivering 10~Gy total irradiation in multi-fractions~\cite{Rosenstein:2004vs,ORourke:2009,BoardoftheFacultyofClinicalOncology:2006wq}, over a virtual 5 day period (e.g. Mon-Fri): 
\begin{enumerate}
    \item a two fraction per day, 1.25~Gy per fraction protocol, with 6~h/18~h time delays between doses given at e.g. 9am/3pm, daily --- hereafter `Benchmark I', or BMI,
    \item a one fraction per day, 2~Gy per fraction protocol with doses given every 24~h e.g. at 9am ---  hereafter, `Benchmark II', or BMII.
\end{enumerate}

Let $\mathfrak{P}$ be a set of all protocols $p_i$ composed of a sequence of $k$ pairs of an irradiation dose, $d_i^j \in \mathfrak{D}$ (where $\mathfrak{D}$ is a set of allowed sizes of doses, in Gy, $j\in \{1,\ldots, k\}$), and a time delay  representing wait-times between the previous fraction and the current one, $t_i^j \in \mathfrak{T}$ (where $\mathfrak{T}$ is a set of allowed wait-times, in h), such that  $p_i = ((d_i^1,t_i^1),\dots,(d_i^k,t_i^k))\in \mathfrak{P}$ (for the first given dose, the corresponding wait-time should be interpreted as the time from the beginning of the simulation). Then, according to the definition, BMI and BMII protocols can be represented as $p_{BMI} \equiv ((1.25,0),(1.25,6),(1.25,18),(1.25,6),(1.25,18),(1.25,6),(1.25,18),(1.25,6))$ and $p_{BMII} \equiv ((2.0,0),(2.0,24),(2.0,24),(2.0,24),(2.0,24))$, respectively.

Now, given a function $f$ to determine the fitness of a protocol (see the \nameref{sec:materialsandmethods} section for the definition and the methodology applied to obtain the fitness over multiple synthetic tumours), the general search problem resolves to,
\[
\max_{p_i\in \mathfrak{P}} f(p_i), \quad \text{such that} \quad \sum^k_{m=1} d^m_i \leq 10.
\]
The constraint $\sum^k_{m=1} d^m_i \leq 10$ is added to limit the total sum of irradiation doses by a reasonable amount taking into account negative health consequences~\cite{Rosenstein:2004vs,ORourke:2009,BoardoftheFacultyofClinicalOncology:2006wq}. Note that for protocols with a sum lower than 10~Gy we can always improve (or not worsen, at least) the fitness function, but for $2$ protocols with a similar fitness function, the radiologists may prefer the one with a lower sum of doses, so in the search space we consider also protocols with a sum of doses lower than $10$~Gy.


In \cite{Angus2014}, for BMI, the fractional dose was constrained to $\mathfrak{D}' = \{1.25\}$, and time-delays were explored in 30 minute gradations with a minimum interval of 4~h, up to 26~h, $\mathfrak{T}' = \{4,4.5,\dots,25.5,26\}$, while for BMII $\mathfrak{D}'' = \{2.0\}$ and $\mathfrak{T}'' = \{10,10.5,\dots,31.5,32\}$, meaning that for each benchmark case the delay options considered had a 22~h span available and so $|\mathfrak{T}'|=|\mathfrak{T}''|=45$. It was also assumed that the first dose is always given in the first possible time slot, at the beginning of the simulation. The total dose constraint ($\leq$ 10~Gy) implies that $k \leq 8$ under BMI and $k\leq 5$ under BMII search conditions, respectively. To estimate the number of feasible protocols, under given constrains, we implemented and ran a simple program~\cite{SourceCUDA}, which for BMI and BMII gave as about $3.0\times 10^{11}$  and $4.1\times 10^6$ protocols, respectively, showing the size of the space for considered problem.


Nevertheless, the high-speed throughput gain of the present approach encourages a dramatic increase in the search space for a second phase of experiments, where in addition to searching over $\mathfrak{T}$ we also search over a range of irradiation \emph{doses} from 0.25~Gy up to a given maximal fractional dose, $d^{max}$, i.e. $\mathfrak{D} = \{0.25,0.50,\dots,d^{max}\}$. In the considered research, it was assumed that the minimum considered delay and wait-time is 6~h, which means that up to 20 fractions might be delivered over a 5 day treatment, implying that $k\leq 20$. It was also assumed that the total delivered dose must be lower or equal to 10~Gy. This leads to an enlarged search space size, for (say) $d^{max} = 5.0$~Gy, with at least $1.39 \times 10^{15}$ protocols, estimated as follows: first, we easily estimate the number of all protocols with doses summing up to $10$~Gy, without the limit for a single dose equal to $5$~Gy, by realising that such a number is just equal to the number of distributing 40 minimal doses ($0.25$~Gy) in 20 time slots which is $\binom{59}{19}$. Then, we subtract protocols that have a single dose greater than $5$~Gy. If such a dose is equal to $K$~Gy ($K \in \{5.25, 5.5, \ldots, 10\}$, so such a dose would have a number of minimal doses ($0.25$~Gy) from the set $\{21, 22,\ldots, 40\}$), then we have $2\times \binom{59 - 4\times K - 1}{40 - 4\times K}$ protocols in which such a dose is in the first or in the last time slot and $(59 - 4\times K - 1)\times \binom{59 - 4\times K - 2}{40 - 4\times K}$ protocols in which such a dose is in one of the time slots in the middle. Such combinatorial considerations imply that exhaustive, brute-force, search is not applicable under current computational architectures, but instead lend themselves to highly non-linear search strategies such as evolutionary search, as is undertaken here.

\subsection*{Establishing the fidelity of the GPU model}

First, the fidelity of the C++/GPU implementation of the underlying model to the original parallelised {\sc Matlab} version~\cite{Angus2014} was verified. An intermediate version of the code was written in C++~\cite{SourceCPU}, which accurately represented the dynamics of the original model, but was fast enough to permit baseline comparisons with the developed GPU code. Two forms of verification were conducted -- single step tests (unit tests) and whole run test (integration tests). Using a single tumour from the library, under two different irradiation protocols (a single 20~Gy dose, and a 5 day/2~Gy per day program) the output of the {\sc Matlab} and C++ code were compared and found to exactly match over both unit- and integration- testing regimes~\cite{BachelorThesis}. Next, the GPU implementation~\cite{SourceCUDA} was compared to the intermediate C++ implementation (run on CPU) by comparing the number of occupied sites at the end of an equivalent 5 day treatment period followed by 5 day regrowth period, for 60 randomly generated protocols having total dose less than or equal to 10~Gy. Each protocol was run over 10 test tumours (for details see \nameref{sec:lib} section), for a total of 100 (GPU) or 48 (C++/CPU) replications each. The GPU implementation was found to provide high-fidelity to the base C++ implementation (Fig~\ref{fig:CPUvsGPU_normal}, panel A). Two statistical tests were conducted. First, t-test~\cite{SciPyTTest} was carried out to compare the averages of the results of both (GPU and CPU) implementations (Fig~\ref{fig:CPUvsGPU_normal}, panel B). Second, Kolmogorov-Smirnov test~\cite{SciPyKSTest} was conducted to investigate the normality of the distribution of the new implementation (Fig~\ref{fig:CPUvsGPU_normal}, panels C). Finally, 2\,000 evaluations of a single randomly selected protocol were conducted and the distribution of the final number of occupied sites was visualised (see Fig~\ref{fig:CPUvsGPU_normal}, panel D).


\subsection*{Performance comparison of candidate protocols discovered by GPU-GA}

\subsubsection*{Search under BMI \& BMII}

To accurately assess the fitness of any candidate protocol against a standard benchmark, the fitness methodology (described in \nameref{sec:materialsandmethods} section below) was undertaken for both the clinical benchmarks (BMI and BMII), and the top three state-of-the-art candidate protocols discovered by a previous study~\cite{Angus2014} (i.e. BMI-1, -2, -3, BMII-1, -2, -3). For each protocol, the mean fitness scores and standard deviation was calculated based on fitness function~\eqref{eq:final_fit_fun} over 100 evaluations (random seeds) for each of 10 synthetic tumours from the study library (for details see \nameref{sec:materialsandmethods} section).

%% BMI - table
%%\input{bm1_table} % tab:ga_bmi_results

%% Results figure


Panel A of Fig~\ref{fig:benchmark_comparison} and Panel A of Table~\ref{tab:ga_bmi_results} present the results of the performance comparison between the BMI standard and the previous state-of-the-art~\cite{Angus2014} under the fitness methodology of the present work. In Panel B of Table~\ref{tab:ga_bmi_results} the best performing protocol candidates identified under GPU-GA search are reported where fractional doses of \emph{up to} 1.25~Gy were considered, with the top, average fitness outcome of 1153.62, representing a 4.0\% improvement over the prior BMI state-of-the-art, or 9.1\% gain over the BMI outcome (in terms of tumour suppression, based on formula \eqref{eq:supression}). In all cases, the top three GPU-GA protocols were discovered by a GA search strategy (see \nameref{sec:GA_approach} section for more details on GA methodology).

\begin{figure}[h]
    \centering
    \includegraphics[width=0.9\textwidth]{../figs/benchmark_comparison_top10_v0h.eps}
    \caption{{\bf Performance comparison of candidate protocols versus BMI, BMII and in the multi-parameter case.} Fitness scores (higher values indicate greater tumour suppression) for the clinical (circle markers) and state-of-the-art~\cite{Angus2014} (squares) protocols of prior work, and discovered via GPU-GA in the current study (open triangles), under the constraints of BMI (A), BMII (B) and the multi-parameter case (C). Markers for the GPU-GA represent the top 10 discovered protocols (n.b.: x-position randomly moved to prevent over-plotting). Note: in all cases $\sum^k_{m=1} d^m_i \leq 10$.}\label{fig:benchmark_comparison}
\end{figure}


\begin{table}[htp!]
    %\centering
    \caption{
    {\bf Performance comparison of candidate protocol fitness around {\bf BMI}.}
    Mean fitness $f(p)$ under the fitness methodology for each candidate protocol (standard deviation in parenthesis), either as given in the benchmark (BMI), or as discovered by limited (time-delay only) search by \cite{Angus2014} (Panel A); or, as found by this study (Panel B).}\label{tab:ga_bmi_results}
    \footnotesize 
    \begin{tabular}{|l|c|c|c|c|c|l|}
\hline
Id & $p$ & Total &  Max & Min time & GA & $f(p)$ \\
 & ([Gy], [h])  & dose  & dose & interval & type &   \\
 &  & [Gy] & [Gy] & [h] &  & \\ 
 
 \hline
 \multicolumn{7}{|c|}{\bf Panel A} \\
 \hline
 
 \multirow{4}{*}{BMI}  & ((1.25, 0) (1.25, 6), & \multirow{4}{*}{10} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{6$^*$ \& 18$^*$} &  \multirow{4}{*}{-} &  \\ %\cline{4-4}
                       & (1.25, 18), (1.25, 6), & & & & & \it 1118.95 \\ %\cline{4-4}
                       & (1.25, 18), (1.25, 6), & & & & & (40.45) \\ %\cline{4-4}
                       & (1.25, 18), (1.25, 6))  & & & & &  \\ \hline
\multirow{4}{*}{BMI-1} & ((1.25, 0), (1.25, 18.5), & \multirow{4}{*}{10} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Tmnt \cite{Angus2014}} &  \\ %\cline{4-4}
                       &  (1.25, 20), (1.25, 18.5), & & & & & 1138.55 \\ %\cline{4-4}
                       & (1.25, 17.5), (1.25, 16.5), & & & & & (33.60) \\ %\cline{4-4}
                       &(1.25, 13), (1.25, 10)) & & & & & \\ \hline
\multirow{4}{*}{BMI-2} & ((1.25, 0), (1.25, 18.5), & \multirow{4}{*}{10} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Top1 \cite{Angus2014}} &  \\ %\cline{4-4}
                       & (1.25, 20), (1.25, 18.5), & & & & & 1137.48\\ %\cline{4-4}
                       & (1.25, 17.5), (1.25, 16.5),  & & & & & (33.53) \\ %\cline{4-4}
                       & (1.25, 12.5), (1.25, 10)) & & & & & \\ \hline
\multirow{4}{*}{BMI-3} & ((1.25, 0), (1.25, 18.5)  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Tmnt \cite{Angus2014}} & \\ %\cline{4-4}
                       & (1.25, 20), (1.25, 18.5),  & & & & & 1139.26\\ %\cline{4-4}
                       & (1.25, 17.5), (1.25, 16.5), & & & & & (33.28)\\ %\cline{4-4}
                       &  (1.25, 14), (1.25, 10)) & & & & & \\
                       
\hline
\multicolumn{7}{|c|}{\bf Panel B} \\
\hline

\multirow{4}{*}{GA1.25-1} & ((1.25, 8.5), (1.25, 18.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{4}{*}{Smpl} &  \\ %\cline{4-4}
                       & (1.25, 18.5), (1.25, 15.5) & & & & &  \bf 1153.62$\dagger$ \\ %\cline{4-4}
                       & (1.25, 15.5), (1.25, 13.5) & & & & & (34.56)\\ %\cline{4-4}
                       & (1.25, 15), (1.25, 14)) & & & & & \\ \hline
\multirow{4}{*}{GA1.25-2} & ((0.75, 1.5), (1.25, 12.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{4}{*}{Smpl} &  \\ %\cline{4-4}
                       & (1, 19.5), (1.25, 15) & & & & & 1148.38$\dagger$ \\ %\cline{4-4}
                       & (1.25, 14.5), (1.25, 13.5) & & & & & (34.84) \\ %\cline{4-4}
                       & (1, 13), (1, 14), (1.25, 13.5)) & & & & & \\ \hline
\multirow{4}{*}{GA1.25-3} & ((1, 0), (1.25, 15.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{4}{*}{Smpl} &  \\ %\cline{4-4}
                       & (1.25, 15), (1.25, 16) & & & & & 1147.08$\dagger$ \\ %\cline{4-4}
                       & (1.25, 15), (1, 13.5) & & & & & (33.38) \\ %\cline{4-4}
                       & (1, 14), (1, 14), (1, 16.5)) & & & & & \\ 
\hline
%\multicolumn{7}{l}{\makecell{
%GA types indicate tournament (Tmnt), top-1 (Top1), or simple (Smpl) selection strategies.\\
%`*' indicates values fixed (not searched).\\ Italic and boldface $f$ values indicate minimum and maximum, respectively.\\ $\dagger$ significance ${p < 0.001}$, comparing to BMI
%}} \\
    \end{tabular}\\
GA types indicate tournament (Tmnt), top-1 (Top1), or simple (Smpl) selection strategies.\\
`*' indicates values fixed (not searched).\\ Italic and boldface $f$ values indicate minimum and maximum, respectively.\\ $\dagger$ significance ${p < 0.001}$, comparing to BMI
\end{table}

%% BMII - table
%\input{bm2_table} % tab:ga_bmi_results


Panel B of Fig~\ref{fig:benchmark_comparison} and Panels A and B of Table~\ref{tab:ga_bmii_results} give the equivalent results under the BMII environment. Here, fitness gains of 4.5\% and 8.2\% were obtained relative to the prior state-of-the-art and baseline BMII protocol, respectively (according to formula \eqref{eq:supression}). Under BMII settings the GPU-GA strategies with the best performance were found to be even distributed from tournament, simple or roulette selection.


\begin{table}[htp!]
    %\centering
    \caption{ 
    {\bf Performance comparison of candidate protocol fitness around {\bf BMII}.}
    Mean fitness $f(p)$ under the fitness methodology for each candidate protocol (standard deviation in parenthesis), either as given in the benchmark (BMII), or as discovered by limited (time-delay only) search by \cite{Angus2014} (Panel A); or, as found by this study (Panel B).}\label{tab:ga_bmii_results}
    \footnotesize
\begin{tabular}{|l|c|c|c|c|c|l|}
\hline
Id & $p$   & Total &  Max & Min time & GA &  \\
 & ([Gy], [h])  & dose  & dose & interval & type & $f(p)$ \\
 &  & [Gy] & [Gy] & [h] &  & \\
 \hline
  \multicolumn{7}{|c|}{\bf Panel A} \\
 \hline
\multirow{3}{*}{BMII} & ((2, 0), (2, 24), & \multirow{3}{*}{10} & \multirow{3}{*}{2$^*$} & \multirow{3}{*}{24$^*$} &  \multirow{3}{*}{-} & \it 1144.83 \\ %\cline{3-3}
                       & (2, 24), (2, 24), & & & & & (37.69)\\ %\cline{3-3}
                       & (2, 24)) & & & & & \\ \hline
\multirow{2}{*}{BMII-1} & ((2,0) (2, 15), (2, 22), & \multirow{2}{*}{10} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top5 \cite{Angus2014}} & 1158.54 \\ %\cline{2-2}
                       & (2, 15.5), (2, 23.5)) & & & & & (43.68) \\\hline

\multirow{2}{*}{BMII-2} &  ((2,0) (2, 15), (2, 19), & \multirow{2}{*}{10} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top5 \cite{Angus2014}} & 1150.39 \\ %\cline{2-2}
                       & (2, 15.5), (2, 18)) & & & & & (40.56)\\\hline

\multirow{2}{*}{BMII-3} & ((2,0) (2, 15), (2, 23), & \multirow{2}{*}{10} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top1 \cite{Angus2014}} & 1158.03 \\ %\cline{2-2}
                       & (2, 15.5), (2, 22.5)) & & & & & (44.14) \\

\hline
 \multicolumn{7}{|c|}{\bf Panel B} \\
 \hline

\multirow{4}{*}{GA2.0-1}   & ((0.25, 17.5), (1.75, 17),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Tmnt} &  \\ %\cline{4-4}
                       & (1.5, 20.5), (1.5, 15) & & & & & \bf 1173.96$\dagger$ \\ %\cline{4-4}
                       & (1.75, 16), (1.5, 18.5) & & & & &  (37.80)\\ %\cline{4-4}
                       & (1.75, 15)) & & & & & \\ \hline
\multirow{4}{*}{GA2.0-2}   & ((0.5, 11), (1.5, 18),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Smpl} & \\ %\cline{4-4}
                       & (1.75, 20), (1.75, 16.5) & & & & & 1172.55$\dagger$ \\ %\cline{4-4}
                       & (1.75, 16.5), (1.5, 16.5) & & & & & (38.71) \\ %\cline{4-4}
                       & (1.25, 16)) & & & & & \\ \hline
\multirow{4}{*}{GA2.0-3}   & ((0.25, 2), (0.25, 15.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Rlt} &  \\ %\cline{4-4}
                       & (1.5, 15.5), (1.5, 19) & & & & & 1172.44$\dagger$ \\ %\cline{4-4}
                       & (1.5, 15), (1.75, 16.5) & & & & & (36.74) \\ %\cline{4-4}
                       & (1.75, 16), (1.5, 18.5)) & & & & & \\
\hline
%\multicolumn{7}{c}{\makecell{GA types indicate top-1 (Top1), top-5 (Top5), tournament (Tmnt), simple (Smpl),\\ or roulette (Rlt) selection strategies. $\dagger$ significance ${p < 0.001}$, comparing to BMII\\ See caption to Table~\ref{tab:ga_bmi_results} for other notes.}} \\
    \end{tabular}\\
GA types indicate top-1 (Top1), top-5 (Top5), tournament (Tmnt), simple (Smpl), or roulette (Rlt) selection strategies. \\$\dagger$ significance ${p < 0.001}$, comparing to BMII.\\ See the caption to Table~\ref{tab:ga_bmi_results} for other notes.
\end{table}



\subsubsection*{Multi-parameter search beyond BMII}

In a second phase of search, the maximum fractional dose, $d^{max}$ was incrementally stepped beyond 2.0~Gy (as in BMII), in 0.5~Gy steps towards 4.5~Gy, whilst keeping the total, 5 day, treatment dose at or below 10~Gy. As noted earlier, by relaxing the constraints on $d^{max}$ the search space increases dramatically, potentially frustrating the GPU-GA technique in discovering more promising protocols. However, results presented in Table~\ref{tab:best_ga_larger_dose_results} and given in Panel C of Fig~\ref{fig:benchmark_comparison} demonstrate that the GPU-GA technique was highly effective in identifying potent candidate protocols under each set of search constraints provided.

%% BEST - larger doses - table
%\input{best_table} % tab:best_ga_larger_dose_results

{
\begin{table}[htp!]
    %\centering
    \caption{
    {\bf Performance comparison of candidate protocol fitness under multi-parameter search.}
    Mean fitness $f(p)$ under the fitness methodology for each candidate protocol (standard deviation in parenthesis), under varying maximum fractional dose constraints (column 4), holding the total treatment dose below or equal to 10 Gy.}\label{tab:best_ga_larger_dose_results}
    \footnotesize
    \begin{tabular}{|l|c|c|c|c|c|c|}
\hline
Id & $p$  & Total &  Max & Min time & GA &  \\
 & ([Gy], [h])  & dose  & dose & interval & type & $f(p)$  \\
 &  & [Gy] & [Gy] & [h] & & \\ \hline\hline
 
 
 \multirow{3}{*}{GA2.5-1} & ((0.25, 25.5), (1.75, 26.5),  & \multirow{3}{*}{10} & \multirow{3}{*}{2.5} & \multirow{3}{*}{12} & \multirow{3}{*}{Rlt} & 1193.99  \\ %\cline{3-3}
                       & (2, 16), (2, 19) & & & & &  (42.37)\\ %\cline{3-3}
                       & (2, 16.5), (2, 15.5)) & & & & &  \\ \hline
 \multirow{3}{*}{GA2.5-2} & ((2, 44.5), (2, 24.0),  & \multirow{3}{*}{10} & \multirow{3}{*}{2.5} & \multirow{3}{*}{12} & \multirow{3}{*}{Tmnt} & 1192.59  \\ %\cline{3-3}
                       & (2.25, 17.5), (2, 14.5) & & & & & (45.81) \\ %\cline{3-3}
                       & (1.75, 14)) & & & & &  \\ \hline\hline
\multirow{3}{*}{GA3.0-1} & ((0.25, 11.5), (3, 42), & \multirow{3}{*}{10} & \multirow{3}{*}{3} & \multirow{3}{*}{12} &  \multirow{3}{*}{Rlt} & 1213.25  \\ %\cline{3-3}
                       & (2.5, 23.5), (2.25, 17), & & & & & (50.57)\\ %\cline{3-3}
                       & (2, 18.5)) & & & & & \\ \hline
\multirow{3}{*}{GA3.0-2} & ((0.5, 22), (0.5, 14.5), & \multirow{3}{*}{10} & \multirow{3}{*}{3} & \multirow{3}{*}{12} &  \multirow{3}{*}{Tmnt} & 1212.63  \\ %\cline{3-3}
                       & (0.25, 17.5), (3, 16), & & & & & (50.92)\\ %\cline{3-3}
                       & (2.75, 23), (3, 20.5)) & & & & & \\ \hline\hline
\multirow{2}{*}{GA3.5-1} & ((2.5, 60.5), (2.5, 21), & \multirow{2}{*}{10} & \multirow{2}{*}{3.5} & \multirow{2}{*}{12} &  \multirow{2}{*}{Tmnt} & 1232.35   \\ %\cline{2-2}
                       & (2.5, 17.5), (2.5, 16.5)) & & & & & (60.85)\\ \hline
\multirow{3}{*}{GA3.5-2} & ((0.25, 2.5), (0.75, 57), & \multirow{3}{*}{10} & \multirow{3}{*}{3.5} & \multirow{3}{*}{12} &  \multirow{3}{*}{Smpl} & 1231.71   \\ %\cline{3-3}
                       & (3, 13.5), (2.75, 26.5), & & & & & (52.88) \\ %\cline{3-3}
                       & (3.25, 19.5)) & & & & & \\ \hline\hline
\multirow{2}{*}{GA4.0-1} & ((3.75, 75.5), (3.75, 24.5), & \multirow{2}{*}{10} & \multirow{2}{*}{4} & \multirow{2}{*}{12} &  \multirow{2}{*}{Smpl} & \bf 1273.7  \\ %\cline{2-2}
                       & (2.5, 19)) & & & & & (64.75)\\ \hline
\multirow{2}{*}{GA4.0-2} & ((3.25, 60), (3.5, 24), & \multirow{2}{*}{10} & \multirow{2}{*}{4} & \multirow{2}{*}{12} &  \multirow{2}{*}{Smpl} & 1248.25  \\ %\cline{2-2}
                       & (3.25, 19.5)) & & & & &(67.53) \\ \hline\hline
\multirow{2}{*}{GA4.5-1} & ((0.25, 53.5), (3.5, 22.5), & \multirow{2}{*}{10} & \multirow{2}{*}{4.5} & \multirow{2}{*}{12} &  \multirow{2}{*}{Tmnt} & 1264.54  \\ %\cline{2-2}
                       & (2.0, 24.5), (4.25, 15)) & & & & & (63.76) \\\hline
\multirow{2}{*}{GA4.5-2} & ((0.25, 3.5), (3.75, 69), & \multirow{2}{*}{10} & \multirow{2}{*}{4.5} & \multirow{2}{*}{12} &  \multirow{2}{*}{Smpl} & 1264.07    \\ %\cline{2-2}
                       & (4.25, 26), (1.75, 20)) & & & & & (61.92)\\
\hline
%\multicolumn{7}{c}{\makecell{GA type indicates roulette (Rlt), tournament (Tmnt), or simple (Smpl) selection strategies.\\ See caption to Table~\ref{tab:ga_bmi_results} for other notes.}}
    \end{tabular}\\
GA type indicates roulette (Rlt), tournament (Tmnt), or simple (Smpl) selection strategies.\\ See caption to Table~\ref{tab:ga_bmi_results} for other notes.
\end{table}
}% footnotesize



%Tumour suppression gains over the best performing GPU-GA candidate from the BMII ($d^{max} = 2.0$~Gy) from 6.1\% ($f(p)=1193.99$, $d^{max}=2.5$~Gy, GA2.5-1 protocol) to 30.6\% ($f(p)=1273.7$, $d^{max}=4$~Gy, GA4.0-1 protocol) were observed in the discovered candidate protocols. As the experiment settings moved from $d^{max} = 2.5$~Gy to $d^{max} = 4.0$~Gy, in reference to the best state-of-the-art BMII protocol (BMII-3, Table~\ref{tab:ga_bmii_results} Panel A) gains of 10.4\% and 33.7\% were observed. Comparing to the baseline BMII protocol (BMII Table~\ref{tab:ga_bmii_results} Panel A), gains are 13.8\% and 36.3\% respectively. A natural progression can be observed in the fitness outcomes discovered in the 10 best performing candidate protocols of the GPU-GA search in Fig.~\ref{fig:benchmark_comparison} from the $d^{max}$ 2.5~Gy to 4.0~Gy settings, before a decline is observed for the 4.5~Gy setting.

Tumour suppression gains over the best performing GPU-GA candidate from the BMII ($d^{max} = 2.0$~Gy, GA2.0-1 protocol) from 6.1\% ($f(p)=1193.99$, $d^{max}=2.5$~Gy, GA2.5-1 protocol) to 30.6\% ($f(p)=1273.7$, $d^{max}=4$~Gy, GA4.0-1 protocol) were observed in the discovered candidate protocols. 
As the experiment settings gradually moved from $d^{max} = 2.5$~Gy to $d^{max} = 4.0$~Gy, in reference to the best state-of-the-art BMII protocol (BMII-3, Table~\ref{tab:ga_bmii_results} Panel A) gains of 10.4\% (GA2.5-1 protocol) and 33.7\% (GA4.0-1 protocol) were observed. Comparing these GPU-GA results with the baseline clinical BMII protocol (BMII Table~\ref{tab:ga_bmii_results} Panel A), we can observe gains of 13.8\% and 36.3\% respectively. A natural progression can be observed in the fitness outcomes discovered in the 10 best performing candidate protocols of the GPU-GA search in Fig~\ref{fig:benchmark_comparison} from the $d^{max}$ 2.5~Gy to 4.0~Gy settings, before a decline is observed for the 4.5~Gy settings.

By inspection of the discovered protocols in Table~\ref{tab:best_ga_larger_dose_results} (GA4.5-1 and -2) it can be seen that a high degree of heterogeneity was present in the fractional doses for the 4.5~Gy setting, compared with the more even fractional doses of the GA4.0-1 and -2 candidates. It can be hypothesised that as the fractional dose approaches the total dose allowable, the optimisation surface becomes much more complex with a larger fraction of illegal protocols (i.e. failing the total dose constraint) adjacent to a legal, but well performing protocols.

\subsection*{Hand-crafted high-potential candidates through sub-modular pooling}

A natural question arises from the GPU-GA search phase described above -- what might be driving the success of the discovered protocols? and, could a \emph{hand-crafted} protocol, informed by the characteristics of the GPU-GA discovered protocols, obtain similar, or even better, performance? Previous work has already hinted at likely reasons for the value of certain protocol dose--delay sub-modules, finding that cell cycle synchronicity might play a role~\cite{Angus2014}. Following this intuition, we take the best 10 performing protocols from each of the seven $d^{max}$ constraints applied across the BMI, BMII and multi-parameter search phases $\{$1.25~Gy, 2.0~Gy, 2.5~Gy, $\dots$, 4.5~Gy$\}$ (the top 2 of which are shown in Table~\ref{tab:best_ga_larger_dose_results}), decompose them into dose--delay sub-modules and then pool all of these together to compute the occurrence density of each sub-module across these high performing protocols (see Fig~\ref{fig:density_composite}). As can be seen in the figure, a highly frequent peak arises around a fractional dose of 1.4~Gy and delay of 15.5~h, with a sharp, tightly defined peak at 0.7~Gy, 0.5~h also present. The existence of these peaks strongly suggests the presence of particular modalities of multi-fraction protocol design which are potent for the EMT6/Ro cell line, at the stage of growth modelled. If this were not the case, one might expect, for example, either a uniform or Gaussian-like density pattern (somewhat random relationship), or a density pattern matching a particular heuristic (e.g. dividing the given $d^{max}$ by a given number of fractional doses, over 120~h).

\begin{figure}
    \centering
    \includegraphics[width=0.9\textwidth]{../figs/density_composite.eps}
    \caption{{\bf Pooling density plots of $(t,d)$ pairs from the top 10 protocols from each of the seven $d^{max}$ constraints discovered via the pooled GPU-GA BMI, BMII \& multi-parameter search phases.} All sub-modules are pooled, and a 2-Dimensional kernel density surface is computed (blue indicates low density, red peak density, 3-D visualisation in (D)) (A), with 1-Dimensional smoothed densities provided for sequence delay only (B) and dose fraction only (C) also. The position of prominent peaks are labelled.}\label{fig:density_composite}
\end{figure}




%% Hand-crafted table
%\input{hc_table} % tab:manual_results

{
\begin{table}[htp!]
    %\centering
    \caption{
    {\bf Performance comparison of hand-crafted protocols under computationally-supported treatment design.}
    Mean fitness $f(p)$ under the fitness methodology for each candidate protocol tested (standard deviation in parenthesis), where protocol design was inspired by the results presented in Fig.~\ref{fig:density_composite}.}\label{tab:manual_results}
    \footnotesize
    \begin{tabular}{|l|c|c|c|}
\hline
Id & $p$ & Total dose &  $f(p)$ \\
 & ([Gy], [h]) & [Gy] & (sd)  \\
 \hline
 \multirow{2}{*}{HC 1} &  \multirow{2}{*}{$((1.4, 15.5) \dots (1.4, 15.5))$} & \multirow{2}{*}{9.8} & 1147.05 \\ %\cline{3-}
                       &  & & (36.31)\\ %\cline{3-3}
                       \hline
 \multirow{2}{*}{HC 2} &   \multirow{2}{*}{$((0.7, 0.5),(1.4, 15.5) \dots (1.4, 15.5))$}   & \multirow{2}{*}{9.1} & 1123.14 \\ %\cline{3-}
                       & & & (37.37)\\ %\cline{3-3}
                       \hline
 \multirow{2}{*}{HC 3} &  \multirow{2}{*}{$((0.7,0.5) ,(1.4, 15.5) \dots (1.4, 15.5),(0.7, 0.5))$}  & \multirow{2}{*}{9.8} & 1148.79\\ %\cline{3-}
                       &  & &  (38.29)\\ %\cline{3-3}
                       \hline
 \multirow{2}{*}{HC 4} &  \multirow{2}{*}{$((1.42, 17.1) \dots (1.42, 17.1))$}  & \multirow{2}{*}{9.94} & \bf 1157.83\\ %\cline{3-}
                       & & & (35.96)\\ %\cline{3-3}
                       \hline
%\multicolumn{4}{c}{\makecell{HC-1 repeats the (1.4,15.5) pair seven times; HC-2 is the same as HC-1,\\ but replaces the first fraction with the (0.7,0.5) pair;\\ HC-3 replicates HC-2, but adds a (0.7,0.5) pair to the end; and HC-4 is a dummy protocol\\ which aims to tightly fit a maximum fractional dose, 7-repeat protocol into a 120h (5 day) period.}} \\
    \end{tabular}\\
HC-1 repeats the (1.4,15.5) pair seven times; HC-2 is the same as HC-1,\\ but replaces the first fraction with the (0.7,0.5) pair;\\ HC-3 replicates HC-2, but adds a (0.7,0.5) pair to the end; and HC-4 is a dummy protocol\\ which aims to tightly fit a maximum fractional dose, 7-repeat protocol into a 120h (5 day) period. \\
\end{table}
% footnotesize
}


Taken together, a variety of hand-crafted protocols were developed to examine whether a protocol which repeats the high-frequency sub-module, or modules, would be as potent as candidate protocols discovered by GPU-GA (see Table~\ref{tab:manual_results}). Three protocols were developed on the basis of the (1.4~Gy, 15.5~h) sub-module peak: HC-1 repeats the module 7 times; protocols HC-2 and HC-3 replaces the first fraction with the (0.7~Gy, 0.5~h) module (HC-2, HC-3) and then additional adds the same to the end of the protocol (HC-3). Finally, HC-4 was also developed with a simple heuristic based on the findings of the sub-module density surface, namely to adjust the timing slightly from HC-1 so as to deploy uniform dose fractions evenly across the 120~h treatment window but sill close to the main density peak, reaching as close as possible to the 10~Gy total dose limit.

With a maximum dose of 1.4~Gy, the hand-crafted protocols can be compared to previous results lying between BMI (1.25~Gy) and BMII (2.0~Gy) constraints. There, GA1.25-1 (Table~\ref{tab:ga_bmi_results}) with fitness 1153.62 and GA2.0-1 (Table~\ref{tab:ga_bmii_results}) with fitness 1173.96 can be compared to the performance of the hand-crafted protocols HC-1 and HC-2 with fitness of 1147.05 and 1148.79, respectively. Linear interpolation between the performance of GA1.25-1 and GA2.0-1 would suggest that a fitness of 1157.7 would be an approximate ceiling for GPU-GA discovery at 1.4~Gy per fraction. Of the hand-crafted protocols, HC-1 and HC-3 reach close to this level, but the heuristically guided HC-4 achieves an almost identical, theorised GPU-GA like fitness ($f(p) = 1157.83$). These results lend support to the potential of computationally-supported treatment design to efficiently locate candidate protocols for future examination in the clinical setting.


\section*{Discussion}

% \mon{Include a Discussion that summarizes (but does not merely repeat) your conclusions and elaborates on their implications. There should be a paragraph outlining the limitations of your results and interpretation, as well as a discussion of the steps that need to be taken for the findings to be applied. Please avoid claims of priority.}

%% Summary of main findings
We have shown the development and application of a high-throughput, high-fidelity, computational screening approach to therapeutic irradiation design for EMT6/Ro spheroids, GPU-GA, which effectively combines state-of-the-art computational architectures with highly non-linear search technology to discover high potential candidates for further study. Through careful calibration of the underlying model, and then translation to GPU articulated form, high-fidelity irradiation protocol evaluations were achieved with c. 717x speed-up relative to the prior, high-fidelity, low-throughput, state-of-the-art~\cite{Angus2014}. The GPU-GA strategy thus enabled the facile exploration of the immense search space produced by even discretised irradiation protocol composition over a multi-day treatment window.

By searching near two clinically relevant benchmarks, the GPU-GA strategy yielded significant tumour suppression gains over both simulated clinical benchmark outcomes and prior, throughput-limited GA search. Beyond the clinical benchmarks, relaxed search constraints saw even larger potential gains. Furthermore, by pooling modular components of the most successful discovered candidates, simple, heuristic-led candidate protocols could be built and tested, yielding close to hypothetical optimal results, even though GPU-GA search was not conducted specifically near the hand-crafted protocol location. 

%% Relationship to other theoretical/in-silico -like studies
Our results thus contribute to, and advance, the emerging consensus from theoretical and simulation studies which have identified significant opportunities for treatment efficacy gains beyond the strictures of received, clinical practice. Be it via simulations of ordinary differential equations~\cite{Alfonso2020}, non-linear or linear surrogate modelling~\cite{Montaseri2020}, or agent-based \emph{in silico} modelling~\cite{Kempf2015}, studies that examine deviations from standard `2~Gy per (week) day' treatment modalities have shown that shorter total treatment regimes, hypofractionation, or harnessing immune-response dynamics with immunotherapy drug targeting each present opportunities for treatment optimisation. Whilst none of these related works present high-fidelity model \emph{validation} to a given cell-line, as has been achieved in the current work, all of them present the same conclusion despite their independent and differing model assumptions and approaches, namely that opportunities exist for treatment advance relative to the standard RT approach, and that these opportunities can be explored economically via \emph{in-silico} means.


%% Relationship to clinical work
Our results also align with the findings of the limited clinical experimentation in protocol design that has been conducted in recent years. Notable among these is that of the UK Standardisation of Breast Radiotherapy (START) trials conducted over 1999 and 2001 (START-A $n=2236$, START-B $n=2215$)~\cite{Bentzen2008, Whelan2002}. Ten year follow up studies of the cohorts confirmed that shorter, `more convenient', hypofractionated (RT fractions $> 2.0$~Gy) were no less effective than the world standard longer, 2.0~Gy RT fraction protocols, and were thus adopted by UK cancer treatment centres~\cite{Whelan2010, Haviland2013}. In our multi-parameter search phase, all of which was conducted within a fixed total dose hypofractionated setting, significant, and increasing, potential gains were discovered from 2.5~Gy to 4.0~Gy RT fractions (Fig~\ref{tab:best_ga_larger_dose_results}). Yet, interestingly, the GPU-GA protocols identified did not follow a regular, 24~h `daily' schedule as the clinical trials considered. Given the success of the computationally-supported protocol design (Table~\ref{tab:manual_results}), together with the clinical findings, a reasonable proposal for next-phase testing would be a hypofractionated, sub-24~h regular, protocol fashioned along the lines of HC-4.

%% Opportunities
With the benefit of high-throughput, high-fidelity numerical screening that GPU-GA affords, several promising lines of inquiry are now open. First, whilst the present study is built on the EMT6/Ro mammary mouse cell line, GPU-GA could effectively be extended to any cell-line where sufficient \emph{in vitro} empirical data are available. Given the widespread use of the rigid, accepted protocols in the clinical setting across cell lines~\cite{Thompson2018} there seems a high likelihood that similar potential tumour suppression gains will be achieved in other cell lines. Second, and of particular interest for clinical settings of low-compliance due to a variety of factors, GPU-GA could be used to explore the robustness of a given protocol under perturbation. For example, what is the relative impact of a single missed fraction? what if the fraction is missed early, or late in the treatment period? can down-stream impacts of protocol non-compliance be mitigated by adapting the later sequence of fractions? in what way? Whilst these questions are easy to pose, rigorous clinical exploration of these variations would be prohibitively expensive, and time-consuming, yet these are simply alternative specifications to run under GPU-GA and would be entirely feasible for future studies to tackle. Third, and in line with recent calls for more preclinical and clinical trials to `unravel' the best approach to combined RT and immunotherapy scheduling~\cite{Wang2018} GPU-GA could be extended by the development of a combined RT---immunotherapy module to enable rapid prototyping, screening and discovery of candidate joint therapy schedules to be taken to the next phase.

%% Limitations
Of course, the GPU-GA strategy outlined herein is not without limitations. First, high-fidelity model calibration depends strongly on the empirical evidence available. Not all cell lines have been studied so rigorously as EMT6/Ro, and, the calibration of the underlying mechanisms of the present model was only achieved under a relatively small (but clinically relevant) spheroidal scale. The dynamics of other cell lines, or growth dynamics under vascularisation will differ substantially and must be organically modelled prior to GPU-GA implementation. Second, the path from candidate protocol discovery to clinical translation will require all of the usual caveats, and lead to some additional challenges. For instance, the optimal fractional delay of the present work, in line with the findings of previous findings~\cite{Angus2014}, of around 15-17~h for EMT6/Ro cells, would translate to RT fractions being administered at variable times through the 24~h period over a treatment regime (e.g. 9~am, 1~am, 5~pm, 9~am, ...). Clinical implementation would require a substantial change to standard day-light hour practices in clinical settings, not to mention patients may require significant support to meet the timetable. Given the complex nature of human systems, irregular RT fractions could also see interactions with other circadian-rhythms of human biology. And, larger fractions, delivered as part of an irregular, quasi-optimal treatment program, may have deleterious side-effects, relative to regular, low-dose fractions. These considerations provide fertile ground for future modelling and discovery across a range of scales.



% Results and Discussion can be combined.
%\section*{Results}
%Nulla mi mi, venenatis sed ipsum varius, Table~\ref{table1} volutpat euismod diam. Proin rutrum vel massa non gravida. Quisque tempor sem et dignissim rutrum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi at justo vitae nulla elementum commodo eu id massa. In vitae diam ac augue semper tincidunt eu ut eros. Fusce fringilla erat porttitor lectus cursus, vel sagittis arcu lobortis. Aliquam in enim semper, aliquam massa id, cursus neque. Praesent faucibus semper libero.

% Place tables after the first paragraph in which they are cited.
%\begin{table}[!ht]
%\begin{adjustwidth}{-2.25in}{0in} % Comment out/remove adjustwidth environment if table fits in text column.
%\centering
%\caption{
%{\bf Table caption Nulla mi mi, venenatis sed ipsum varius, volutpat euismod diam.}}
%\begin{tabular}{|l+l|l|l|l|l|l|l|}
%\hline
%\multicolumn{4}{|l|}{\bf Heading1} & \multicolumn{4}{|l|}{\bf Heading2}\\ \thickhline
%$cell1 row1$ & cell2 row 1 & cell3 row 1 & cell4 row 1 & cell5 row 1 & cell6 row 1 & cell7 row 1 & cell8 row 1\\ \hline
%$cell1 row2$ & cell2 row 2 & cell3 row 2 & cell4 row 2 & cell5 row 2 & cell6 row 2 & cell7 row 2 & cell8 row 2\\ \hline
%$cell1 row3$ & cell2 row 3 & cell3 row 3 & cell4 row 3 & cell5 row 3 & cell6 row 3 & cell7 row 3 & cell8 row 3\\ \hline
%\end{tabular}
%\begin{flushleft} Table notes Phasellus venenatis, tortor nec vestibulum mattis, massa tortor interdum felis, nec pellentesque metus tortor nec nisl. Ut ornare mauris tellus, vel dapibus arcu suscipit sed.
%\end{flushleft}
%\label{table1}
%\end{adjustwidth}
%\end{table}


%\section*{Conclusion}

%CO\textsubscript{2} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque. Quisque augue sem, tincidunt sit amet feugiat eget, ullamcorper sed velit. 

%Sed non aliquet felis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris commodo justo ac dui pretium imperdiet. Sed suscipit iaculis mi at feugiat. Ut neque ipsum, luctus id lacus ut, laoreet scelerisque urna. Phasellus venenatis, tortor nec vestibulum mattis, massa tortor interdum felis, nec pellentesque metus tortor nec nisl. Ut ornare mauris tellus, vel dapibus arcu suscipit sed. Nam condimentum sem eget mollis euismod. Nullam dui urna, gravida venenatis dui et, tincidunt sodales ex. Nunc est dui, sodales sed mauris nec, auctor sagittis leo. Aliquam tincidunt, ex in facilisis elementum, libero lectus luctus est, non vulputate nisl augue at dolor. For more information, see \nameref{S1_Appendix}.

%\section*{Supporting information}

% Include only the SI item label in the paragraph heading. Use the \nameref{label} command to cite SI items in the text.
%\paragraph*{S1 Fig.}
%\label{S1_Fig}
%{\bf Bold the title sentence.} Add descriptive text after the title of the item (optional).

%\paragraph*{S2 Fig.}
%\label{S2_Fig}
%{\bf Lorem ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

%\paragraph*{S1 File.}
%\label{S1_File}
%{\bf Lorem ipsum.}  Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

%\paragraph*{S1 Video.}
%\label{S1_Video}
%{\bf Lorem ipsum.}  Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

%\paragraph*{S1 Appendix.}
%\label{S1_Appendix}
%{\bf Lorem ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

%\paragraph*{S1 Table.}
%\label{S1_Table}
%{\bf Lorem ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\section*{Materials and methods}\label{sec:materialsandmethods}

\subsection*{Underlying model: source, validation \& calibration}

The high throughput GPU-GA methodology introduced and demonstrated in this study implements exactly a previously published low-throughput, high-fidelity computational model of avascular EMT6/Ro spheroid growth~~\cite{Piotrowska:2009gj, Angus:2010mo, Angus:2013id, Angus2014}. The most recent version of the model~\cite{Angus2014} was re-implemented in C++ \cite{SourceCPU} and then prepared for GPU architectures~\cite{SourceCUDA} in this study.  The model was developed to replicate the biological dynamics of a specific mouse cancer cell line (EMT6/Ro), a widely used analogue of human breast cancer which also has abundant experimental studies available. The model takes into account several biologically important components of tumour growth and cellular response to irradiation such as: the diffusion of nutrients (oxygen, glucose) and metabolic waists; cell cycles with distinguishable cell phases;  differentiated cellular metabolism (aerobic or anaerobic, proliferating or quiescent) which is responsive to the cell's environmental nutrient concentrations and local pH.
Each site consists on $N$ cells. The fate of each cell staying in the site is determined by the fate of the site.
Each now born site is endowed with cell cycle duration time being a sum of the duration's of four distinguishable cycle phases ($G_1$, $S$, $G_2$ and $M$), Fig. XXX. Each duration is drawn from known normal distributions for EMT6/Ro~\cite{Angus2014} and
each side, each time step, try to progresses through its cell cycle towards mitosis.
In general all cells prefer to proliferate in the most efficient aerobic, rather when less efficient anaerobic, way. Nevertheless it is not always possible due to the local environmental conditions. Thus, upon each update each site use metabolism decision algorithm to determine (Fig. XXXX) its metabolism state. First, not available due to insufficient glucose concentration for the aerobic quiescent state ($n_i<n^{q,ox}$), local environmental hyper-acidity ($pH_i<pH_{death}$) or if site’s cycle limit time has been exceeded ($r_i>r_{max}$)) cells are removed from the system by assigning such a site into the site-death module. Next, the vacancy stopping condition at G1/S boundary is applied to the cells in the $G_1$ phase, being near the $G_1/S$ phase boundary (according to the metabolic time), which do not have sufficient free surrounding sites to place  daughter cells. In consequence such sites pass into a quiescent mode. The same is applied to the cells fro which the local pH is to low to actively proliferate but sufficient to maintain in quiescent mode. For the remaining cells the proliferate/quiescent criteria (level of local viable glucose and oxygen) are checked to determine the type of cell activity.


The implemented GPU multi-irradiation module works the same as the one previously developed in CPU approach~\cite{Angus2014}. Cell response to the irradiation is governed by the time dependent {\it reciprocal repair function}, which according to comparative curve-fitting studies \cite{Fowler2002}, \cite{Fowler1999}, \cite{Biedrmann1991}, \cite{Longo1997} provides a better fit to the data than either a mono-, bi-, or multi- exponential descriptions. The advantage of this approach is that the reciprocal repair function is parameterised by a single parameter, called {\it half-time of repair constant} ($\tau$), which should be estimated independently of the given cell line and temperature. Moreover, it allows to (each time step) model irradiation damage repair even for a complex irradiation schemes. For the detail description of the reciprocal repair function and estimation of half-time of repair constant, see \cite{Angus2014}.  

The model was calibrated at each step of development, including its response to 18 independent multi-fraction irradiation protocols tested in the laboratory~\cite{Otsuka2011,Sugie2006}. The calibration and validation of the model's performance was accomplished with reference to a~broad spectrum of tumour characteristics including: number of cells; saturation size; tumour volume; tumour doubling time; thickness of the proliferating rim; cell phase population fraction; onset and progression of necrosis; and effective dose induced by multi-fraction irradiation. For details, see~\cite{Angus2014} and Supplementary Information within.

% The previously established high-fidelity numerical simulation model of avascular EMT6/Ro growth and response to irradiation 

% In this work, we extend our high-fidelity numerical 
% simulation approach to examine yet untried protocols within the vast space 
% of available protocols to find candidate protocols that offer statistically significant improvements over benchmark protocols presently used in 
% radiotherapy clinics. Our simulation model described in~\cite{Angus2014}

%\subsection*{Benchmark protocols}
%-------------------------------------------------------------------------
\subsection*{The tumour case library}\label{sec:lib}

Due to the stochastic nature of the tumour growth and radiation response model, we follow the approach presented in~\cite{Angus2014} and perform evaluation of all considered protocols across a multi-tumour case library. The case library consists of ten 10-day old tumours developed {\it in-silico} as described in~\cite{Angus:2013id} from an initial seed population of 200 cancer cells (10 cellular automata lattice grid sites) placed in a well-mixed replenished substrate (glucose concentration: 5.5~mM, oxygen concentration: 0.28~mM, with pH level maintained at 7.4). The tumours were then grown {\it in silico} for 10 days without any radiation interference. For the full characterisation of the developed tumour library we refer the reader to~\cite[Supp. Inf., Table~S2]{Angus2014}. At the end of the 10-day growth period, the state of each tumour was saved, and stored in the case library for the future use.


\subsection*{GPU implementation: acceleration, fidelity \& normality} 

The key feature of the present study is to articulate the original EMT6/Ro {\sc Matlab} model implementation~\cite{Angus2014} to a new general-purpose C++/GPU implementation to obtain a step-change in protocol evaluation. Applying general purpose GPU programming to obtain high-throughput numerical simulation of candidate protocols was considered highly promising due to the parallelizable nature of cellular automata computation and the fact that evaluation of a radiation protocol requires multiple independent simulation runs. The architectural idea of the new implementation was to use leverage massive parallelism of the GPUs not only to parallelize the processing of single cells within a simulation but also to concurrently run a batch of simulations. With the new GPU implementation~\cite{SourceCUDA}, an average speed of 80\,000 simulations per hour was achieved using 4 NVIDIA V100 GPUs. This compares to about 111 simulations per hour using 13 10-core CPUs as reported in~\cite[Supp. Inf., Numerical Implementation: Details]{Angus2014}, giving c. 717 improvement.

To ensure fidelity of the new GPU implementation~\cite{SourceCUDA}, an intermediate version of the code written in standard C++ was used, which accurately copied the behaviour of the original {\sc Matlab} simulation but was fast enough to provide a convenient baseline to compare the GPU implementation with~\cite{SourceCPU}.  Given the importance of stochastic dynamics in the underlying model, the C++ implementation required replicating exactly the {\sc Matlab} random number generator (Mersenne Twister with 19937 bits) and associated state initiation. The C++ implementation was rigorously analysed for any biases or implementation errors, until an exact replication of the original {\sc Matlab} implementation under both unit (single simulation step) and integration (outcome over a series of steps) compliance was achieved. Integration testing obtained an exact state match between 14,400 reference points between the {\sc Matlab} and C++ implementations (i.e. the model state every 10 steps) over a simulated 10 day tumour growth period.  From there, the C++/GPU accelerated articulation of the C++/CPU code was prepared. A series of randomly generated protocols was evaluated with both CPU and GPU simulations and the results were statistically compared. For details, c.f.~Fig~\ref{fig:CPUvsGPU_normal} panel A together with ~Fig~\ref{fig:CPUvsGPU_normal} panel B where p-values of the two-tailed t-test comparing corresponding GPU and CPU distributions are reported.

\begin{figure}[ht]
    \centering
    \includegraphics[width = 0.9\textwidth]{../figs/CPU_GPU_comparison_v2.eps}
    \caption{{\bf Examining the fidelity and stability of the GPU implementation of the numerical model.} 60 randomly generated protocols (total dose $\leq$ 10~Gy) were developed and run through 48 and 100 iterations of the CPU and GPU code respectively, against each of 10 tumours in the library for 10 days. A: shows 95\% confidence intervals for CPU (blue) and GPU (orange) relative deviations from the mean number of occupied sites at the end of each replicate; B: provides t-test p-values (two sided) comparisons between the CPU and GPU implementations, with 0.1 (black) and 0.05 (red) significance lines shown; C: provides p-values for Kolmogorov-Smirnov test (KS-test) for the normality of the GPU 100 evaluations over each of 60 protocols, against each of $10$ tumours. D: the histogram arising from 2\,000 GPU evaluations of final occupied sites of a single randomly selected protocol (against a single randomly selected tumour) compared to the normal distribution with the same mean and standard deviation.}\label{fig:CPUvsGPU_normal}
\end{figure}


Finally, the normality of the GPU implementation was examined using the fast GPU implementation of the model. The final number of occupied grid sites was chosen as a representative measure of the tumour state because it is used to calculate the GA fitness function. To check if a data set can be well described by a normal distribution, a single irradiation protocol was randomly selected, and then 2\,000 evaluation of that protocol were run on the randomly selected tumour from the library. In Fig~\ref{fig:CPUvsGPU_normal} (D)  a histogram of the results is plotted which, together with the Kolmogorov-Smirnov test (p-value = 0.38), indicate that the null-hypothesis, i.e. that the distribution is normal, cannot be rejected. In Fig~\ref{fig:CPUvsGPU_normal} (C) p-values of Kolmogorov-Smirnov test for 60 randomly generated protocols comparing GPU and CPU implementations are presented.


\subsection*{The Genetic Algorithm (GA)}

\subsubsection*{GA Overview}\label{sec:GA_approach}

In order to find heuristically optimal protocols, we use genetic algorithms (GAs)~\cite{Holland:2012, goldberg} to explore the vast, combinatorial space of possible protocols. GAs are well suited to high dimensional, non-linear search, especially where quasi-optimal solutions are likely composed of modular sub-components.  In the GA approach irradiation protocols are encoded as genotypes represented as vectors. The quality (fitness) of genotypes is evaluated by a fitness function~\eqref{eq:final_fit_fun} for which the values are calculated using the main GPU implementation of the underlying computational model. The best genotypes are selected using a selection operator, and are later used to create new genotypes using crossover and mutation operators. This procedure is schematically presented in Fig~\ref{fig:GA_scheme}. Together, the aim of the GA approach is to amplify the abundance of the most effective protocols, and protocol sub-modules, whilst retaining (and introducing) sufficient novelty to each generation to avoid the population becoming `stuck' within a locally-optimal (but perhaps not globally optimal) basin of attraction. Since each problem will have its own solution landscape, it is common practice to experiment with a variety of GA operators (see below) within the main selection, crossover, and mutation steps.

\newpage

\begin{figure}[ht]
    \centering
    \includegraphics[width=0.9\textwidth]{../figs/GA_scheme_v5.eps}
    \caption{{\bf The Genetic Algorithm (GA) scheme.} A: A single generation consists of 40 candidate protocols. A new generation is formed by graduating 10\% (4) protocols directly without any alteration. A further 16 protocols are selected in one of three ways: Roulette (proportional chance by fitness), Simple (simple sort and take the top few), or Tournament (pair-wise competition on fitness). Next 36  protocols are formed from the 16 parents via applying the crossover and mutation operators. B: schematic representation of used selection operators. C: schematic representation of used crossover operators. D: schematic representation of used mutation operators.}
    \label{fig:GA_scheme}
\end{figure}

%\newpage

\subsubsection*{GA Resolution, constraints}
The CA model is designed with a resolution such that each hour between doses is split into 600 units of model time (a step). For our treatment protocols, we used a minimal step equal to 300 units, i.e. 30~min, given that smaller intervals between fractions may be difficult to deliver in the clinic. The full treatment time is set to 5~days (120~hours), which implies that each protocol should direct irradiation treatment over 240~time-steps. Additional constraints were established for possible fractional doses. First, following~\cite{Angus2014}, we assume that the sum of all dose values for a~single protocol should not be higher than 10~Gy to ensure that total delivered dose, affecting both the tumour and the healthy tissues, do not exceed a standard clinical practice for low-dose, multi-fraction, irradiation schemes \cite{Rosenstein:2004vs,ORourke:2009,BoardoftheFacultyofClinicalOncology:2006wq}. Moreover, the value of an acceptable single dose should be at a minimum 0.25~Gy and could be increased by 0.25~Gy per step, up to the maximal single dose ($d^{max}$), between 1.25~Gy and 4.5~Gy, depending on a particular experimental setting. In addition, to simulate realistic treatment protocols with doses given to the patient according to the scheduled intervals, we restricted the minimal time interval between doses to vary from 6 to 48 hours.

\subsubsection*{Protocol Representation}
In the implemented code, for GA, each protocol is represented by a sparse vector of 240 dose values, where each position corresponds to a specific point in time. Zeroed values in this representation indicate lack of dose at a corresponding time. A non-zero value in this representation indicates the fractional dose to deliver, at that time point. For example, a value of 1.25 in the fourth vector position would indicate that a 1.25~Gy fraction is delivered 2 hours (four times 30 minutes) after the initialisation of the treatment.

% For the GA model, a protocol is represented as a key-value pair of the absolute time-step and the value of the dose given at this time-step. This is a difference comparing to the original CA model, where the key-value pair consisted of dose value and time-step between previous and the current dose (time to the second dose), instead of the absolute time of application 
% The reason behind using two different representations is that the CA model was created first with the key-value pair representation. Now, during development of the GA model, performing various types of selection, crossover, and mutation operations might be non-trivial for such a representation. Thus, the new vector representation was introduced, which is more similar to standard candidates representation as an array of bits. Because conversion between two representations is very fast and computationally inexpensive, the CA model remained untouched and we convert protocols in each experiment.

\subsubsection*{The Main GA Loop}

% PoC, high-level concept:
% generation_0, G0
% step 1: COPY 4 'best' protocols from G0 via simple selection and take straight to G1. (leave the originals in the pool)
% step 2: SELECT 16 PARENTS from the G0 pool by some method.
% step 3: with 16 PARENTS, run Cross-over + Mutation to generate 36 CHILDREN which enter G1

Each GA generation $T$ consists of 40 protocols which, after fitness determination in the full EMT6/Ro GPU computational model, is used as the genetic material for the formation of the subsequent generation, $T+1$ (c.f. Fig~\ref{fig:GA_scheme}). In the initial step of the GA algorithm, 10\% (i.e., 4) of the best protocols in generation $T$ (according to simple selection) are \emph{copied} from generation $T$ into generation $T+1$ without modification (and retained in the genetic pool of generation $T$). Next, 40\% of the best protocols (i.e., 16) are selected from generation $T$ to become \emph{parents} using one of three \textit{selection} operators: \textit{roulette}, \textit{simple}, or \textit{tournament}. Note, \emph{selection} may cause one or more protocols from the initial copy step to be also present in the parent set. In a third step \emph{crossover} and \emph{mutation} operators are carried out in sequence to form 36 \emph{child} protocols. Crossover uses one of the available operators: \textit{single-point crossover}, \textit{two-point crossover}, or \textit{uniform crossover}. Mutation uses one of the available operators: \textit{swap mutation}, \textit{split mutation}, \textit{dose time mutation}, or \textit{dose value mutation}. Thus, the 36 generated \textit{child} protocols are combined together with the initial 4, copied protocols, to create the new generation $T+1$. Each protocol in the $T+1$ generation is evaluated using EMT6/Ro GPU computational model and \textit{fitness} function. After a prearranged number of iterations, representing a five-day treatment period, followed by a five-day re-growth period, the experiment is finished. The particular type of \textit{selection}, \textit{crossover} and \textit{mutation} operators used is set in the configuration of a given experiment.



%-------------------------------------------------------------------------

\subsubsection*{The fitness function \& experimental setup}
\label{sec:fitness_function}

One of the crucial elements of our experiments is to determine the value of a fitness function $f$, required for GA as the fitness indicator. We use
\begin{equation}\label{eq:final_fit_fun}
   f(p_{i,j}) = 1\,500 - n_{i,j}, 
\end{equation}
where $p_{i,j}$ denotes the considered protocol $i$ evaluated on a single tumour $j$, 1\,500 is the maximal number of possibly occupied sites, $n_{i,j}$ denotes the number of the occupied sites after testing protocol $i$ on tumour $j$ (at day 10). In the performed experiments, we pass the treatment protocol into the CA model and obtain the fitness score. In order to assess the robustness of these results, each protocol is evaluated on 10 different tumours from the library using 2 GPUs, 4 times per GPUs per tumour. This allows us to calculate 80 samples for every single treatment protocol. Thus, the fitness for the particular tested protocol is the mean from the fitnesses of all its evaluations.

\subsubsection*{Calculating tumour suppression improvement}

To estimate the improvement of tumour suppression for a considered protocol $i$ ($ITS_i$) versus a reference protocol evaluated over 10 tumours we calculate the relative difference between the average number of occupied sites after testing protocol $i$ on 10 tumours ($\bar n_{i}$) and the average number of occupied sites for a reference protocol (tested on the same 10 tumours, $\bar n_{ref}$)
\begin{equation}\label{eq:supression}
    ITS_i=\frac{\bar n_{ref}-\bar n_{i}}{\bar n_{ref}}.
\end{equation}


\section*{Acknowledgments}
This work was supported by grant no. 2015/19/B/ST1/01163 of National Science Centre, Poland, ``Mathematical models and methods in description of tumour growth and its therapies''. This research was carried out with the support of the Interdisciplinary Centre for Mathematical and Computational Modelling at the University of Warsaw (ICM UW ICM) under grants no. G74-17 and GR79-29.

\section*{Author contributions}
WO: Data Curation, Formal Analysis, Investigation, Methodology, Software, Validation, Visualization, Writing – Original Draft Preparation, Writing – Review & Editing; RB: Data Curation, Formal Analysis, Investigation, Software, Validation, Writing – Review & Editing; PG: Formal Analysis, Investigation, Methodology, Project Administration, Resources, Software, Validation, Writing – Review & Editing; SDA: Conceptualization, Formal Analysis, Investigation, Methodology, Validation, Visualization, Writing – Original Draft Preparation, Writing – Review & Editing; MJP: Conceptualization, Formal Analysis, Investigation, Methodology, Project Administration, Resources, Supervision, Validation, Visualization, Writing – Original Draft Preparation, Writing – Review & Editing .

\section*{Supporting information}

% Include only the SI item label in the paragraph heading. Use the \nameref{label} command to cite SI items in the text.
%\paragraph*{S1 Fig.}
%\label{S1_Fig}
%{\bf Bold the title sentence.} Add descriptive text after the title of the item (optional).

%\paragraph*{S2 Fig.}
%\label{S2_Fig}
%{\bf Lorem ipsum.} Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

%\paragraph*{S1 File.}
%\label{S1_File}
%{\bf Lorem ipsum.}  Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

%\paragraph*{S1 Video.}
%\label{S1_Video}
%{\bf Lorem ipsum.}  Maecenas convallis mauris sit amet sem ultrices gravida. Etiam eget sapien nibh. Sed ac ipsum eget enim egestas ullamcorper nec euismod ligula. Curabitur fringilla pulvinar lectus consectetur pellentesque.

\paragraph*{S1 Appendix.}
\label{S1_Appendix}
{\bf Details regarding GA experiments.} More detailed information on the infrastructure used in experiments, implemented GA operators, GA metrics and GA performed experiments.

\paragraph*{S1 Table.}
\label{S1_Table}
{\bf Configurations of batches in GA experiments.} %\textit{Number of runs} represents the number of GA runs for each configuration of settings. Starting from batch 6, we introduced the retaining best 10\% of protocols. Final evaluation on all configurations was conducted in the batch 8 (for robustness, as least 3 times for each configuration).

\paragraph*{S2 Table.}
\label{S2_Table}
{\bf Summary of the results for selection operators: \emph{simple selection}, \emph{roulette wheel selection} and \emph{tournament selection}.} %Reported values are means and standard deviations (for 3 best protocols). Bold values indicate the highest fitness function values for a particular time interval and maximum dose value out of available selection operators.


%\section*{Acknowledgments}
%Cras egestas velit mauris, eu mollis turpis pellentesque sit amet. Interdum et malesuada fames ac ante ipsum primis in faucibus. Nam id pretium nisi. Sed ac quam id nisi malesuada congue. Sed interdum aliquet augue, at pellentesque quam rhoncus vitae.

\nolinenumbers


\nolinenumbers

% Either type in your references using
% \begin{thebibliography}{}
% \bibitem{}
% Text
% \end{thebibliography}
%
% or
%
% Compile your BiBTeX database using our plos2015.bst
% style file and paste the contents of your .bbl file
% here. See http://journals.plos.org/plosone/s/latex for 
% step-by-step instructions.
% 

%\bibliographystyle{plos2015}
%\bibliography{scibib}

\begin{thebibliography}{10}

\bibitem{Thompson2018}
Thompson MK, Poortmans P, Chalmers AJ, Faivre-Finn C, Hall E, Huddart RA,
  et~al.
\newblock {Practice-changing radiation therapy trials for the treatment of
  cancer: where are we 150 years after the birth of Marie Curie?}
\newblock British Journal of Cancer 2018 119:4. 2018;119(4):389--407.
\newblock doi:{10.1038/s41416-018-0201-z}.

\bibitem{Rosenstein:2004vs}
Rosenstein B, Lymberis S, Formenti SC.
\newblock {Biologic comparison of partial breast irradiation protocols}.
\newblock International Journal of Radiation Oncology*Biology*Physics.
  2004;60(5):1393--1404.

\bibitem{ORourke:2009}
O'Rourke SFC, McAneney H, Hillen T.
\newblock Linear quadratic and tumour control probability modelling in external
  beam radiotherapy.
\newblock J Math Biol. 2009;58(4-5):799--817.
\newblock doi:{10.1007/s00285-008-0222-y}.

\bibitem{BoardoftheFacultyofClinicalOncology:2006wq}
{Board of the Faculty of Clinical Oncology}. {Radiotherapy Dose-Fractionation};
  2006.
\newblock Royal College of Radiologists.

\bibitem{Haviland2013}
Haviland J, JR O, JA D, RK A, J B, PJ BL, et~al.
\newblock {The UK Standardisation of Breast Radiotherapy (START) trials of
  radiotherapy hypofractionation for treatment of early breast cancer: 10-year
  follow-up results of two randomised controlled trials}.
\newblock The Lancet Oncology. 2013;14(11):1086--1094.
\newblock doi:{10.1016/S1470-2045(13)70386-3}.

\bibitem{Whelan2010}
Whelan T, JP P, MN L, JA J, R M, S P, et~al.
\newblock {Long-term results of hypofractionated radiation therapy for breast
  cancer}.
\newblock The New England journal of medicine. 2010;362(6):513--520.
\newblock doi:{10.1056/NEJMOA0906260}.

\bibitem{Angus2014}
Angus SD, Piotrowska MJ.
\newblock A Matter of Timing: Identifying Significant Multi-Dose Radiotherapy
  Improvements by Numerical Simulation and Genetic Algorithm Search.
\newblock PLoS ONE. 2014;9(12):e114098.
\newblock doi:{10.1371/journal.pone.0114098}.

\bibitem{Gerlee2007}
Gerlee P, Anderson ARA.
\newblock An evolutionary hybrid cellular automaton model of solid tumour
  growth.
\newblock J Theor Biol. 2007;246:583--603.

\bibitem{Patel2001}
Patel AA, Gawlinski ET, Lemieux SK, Gatenby RA.
\newblock A cellular automaton model of early tumor growth and invasion: The
  effects of native tissue vascularity and increased anaerobic tumour
  metabolism.
\newblock J Theor Biol. 2001;213:315--331.

\bibitem{Dormann2002}
Dormann S, Deutsch A.
\newblock Modeling of self-organized avascular tumor growth with a hybrid
  cellular automaton.
\newblock In Silico Biol. 2002;2:0035.

\bibitem{Kansal2000}
Kansal AR, Torquato S, Chioccaeb EA, Deisboeck TS.
\newblock Emergence of a subpopulation in a computational model of tumor
  growth.
\newblock J Theor Biol. 2000;207:431--441.

\bibitem{Kansal2000a}
Kansal AR, Torquato S, IV GRH, Chioccaeb EA, Deisboeck TS.
\newblock Simulated brain tumor growth dynamics using a three-dimensional
  cellular automaton.
\newblock J Theor Biol. 2000;203:367--382.

\bibitem{Kempf2015}
Kempf H, Bleicher M, Meyer-Hermann M.
\newblock Spatio-Temporal Dynamics of Hypoxia during Radiotherapy.
\newblock PLOS ONE. 2015;10(8):e0133357.
\newblock doi:{10.1371/JOURNAL.PONE.0133357}.

\bibitem{Piotrowska:2009gj}
Piotrowska MJ, Angus SD.
\newblock {A quantitative cellular automaton model of in vitro multicellular
  spheroid tumour growth}.
\newblock Journal of Theoretical Biology. 2009;258(2):165--178.

\bibitem{Angus:2010mo}
Angus SD, Piotrowska MJ.
\newblock The Onset of Necrosis in a 3D Cellular Automaton Model of EMT6
  Multi-Cellular Spheroids.
\newblock Applicationes Mathematicae (Warsaw). 2010;37(1):69--88.

\bibitem{Angus:2013id}
Angus SD, Piotrowska MJ.
\newblock {A numerical model of EMT6/Ro spheroid dynamics under irradiation:
  Calibration and estimation of the underlying irradiation-induced cell
  survival probability}.
\newblock Journal of Theoretical Biology. 2013;320:23--32.

\bibitem{Otsuka2011}
Otsuka S, Shibamoto Y, Iwata H, Murata R, Sugie C, Ito M, et~al.
\newblock Compatibility of the linear-quadratic formalism and biologically
  effective dose concept to high-dose-per-fraction irradiation in a murine
  tumor.
\newblock Int J Radiat Oncol Biol Phys. 2011;81(5):1538--1543.
\newblock doi:{10.1016/j.ijrobp.2011.05.034}.

\bibitem{Sugie2006}
Sugie C, Shibamoto Y, Ito M, Ogino H, Miyamoto A, Fukaya N, et~al.
\newblock Radiobiologic effect of intermittent radiation exposure in murine
  tumors.
\newblock Int J Radiat Oncol Biol Phys. 2006;64(2):619--624.
\newblock doi:{10.1016/j.ijrobp.2005.03.032}.

\bibitem{Zacharaki2004}
Zacharaki EI, Stamatakos GS, Nikita KS, Uzunoglu NK.
\newblock Simulating growth dynamics and radiation response of avascular tumour
  spheroids---model validation in the case of an EMT6/Ro multicellular
  spheroid.
\newblock Computer methods and programs in biomedicine. 2004;76(3):193--206.

\bibitem{Stamatakos2006}
Stamatakos GS, Antipas VP, Uzunoglu NK, Dale RG.
\newblock A four-dimensional computer simulation model of the in vivo response
  to radiotherapy of glioblastoma multiforme: studies on the effect of
  clonogenic cell density.
\newblock Brit J Radiol. 2006;79(941):389--400.
\newblock doi:{10.1259/bjr/30604050}.

\bibitem{Kempf2010}
Kempf H, Bleicher M, Meyer-Hermann M.
\newblock Spatio-temporal cell dynamics in tumour spheroid irradiation.
\newblock Eur Phys J D. 2010;60:177--193.

\bibitem{Powathil2012}
Powathil GG, Gordon KE, Hill LA, Chaplain MAJ.
\newblock Modelling the effects of cell-cycle heterogeneity on the response of
  a solid tumour to chemotherapy: Biological insights from a hybrid multiscale
  cellular automaton model.
\newblock Journal of Theoretical Biology. 2012;308:1--19.
\newblock doi:{10.1016/j.jtbi.2012.05.015}.

\bibitem{Powathil2013}
Powathil GG, Adamson DJA, Chaplain MAJ.
\newblock Towards Predicting the Response of a Solid Tumour to Chemotherapy and
  Radiotherapy Treatments: Clinical Insights from a Computational Model.
\newblock PLoS Computational Biology. 2013;9(7):e1003120.
\newblock doi:{10.1371/journal.pcbi.1003120}.

\bibitem{Kempf2013}
Kempf H, Hatzikirou H, Bleicher M, Meyer-Hermann M.
\newblock In Silico Analysis of Cell Cycle Synchronisation Effects in
  Radiotherapy of Tumour Spheroids.
\newblock PLoS Comput Biol. 2013;9(11):e1003295.
\newblock doi:{10.1371/journal.pcbi.1003295}.

\bibitem{Liu2017}
Liu R, Li X, Lam KS.
\newblock {Combinatorial chemistry in drug discovery}.
\newblock Current Opinion in Chemical Biology. 2017;38:117--126.
\newblock doi:{10.1016/J.CBPA.2017.03.017}.

\bibitem{Lightbody2019}
Lightbody G, Haberland V, Browne F, Taggart L, Zheng H, Parkes E, et~al.
\newblock {Review of applications of high-throughput sequencing in personalized
  medicine: barriers and facilitators of future progress in research and
  clinical application}.
\newblock Briefings in Bioinformatics. 2019;20(5):1795--1811.
\newblock doi:{10.1093/BIB/BBY051}.

\bibitem{Bentzen2008}
Bentzen S, RK A, EG A, JM B, PJ BL, SM B, et~al.
\newblock {The UK Standardisation of Breast Radiotherapy (START) Trial B of
  radiotherapy hypofractionation for treatment of early breast cancer: a
  randomised trial}.
\newblock Lancet (London, England). 2008;371(9618):1098--1107.
\newblock doi:{10.1016/S0140-6736(08)60348-7}.

\bibitem{Whelan2002}
Whelan T, R M, J J, M L, W S, L G, et~al.
\newblock {Randomized trial of breast irradiation schedules after lumpectomy
  for women with lymph node-negative breast cancer}.
\newblock Journal of the National Cancer Institute. 2002;94(15):1143--1150.
\newblock doi:{10.1093/JNCI/94.15.1143}.

\bibitem{SourceCUDA}
Bana\'s R. {EMT6-Ro CUDA implementation source code}; Online accessed
  11-May-2021.
\newblock \url{https://github.com/banasraf/EMT6-Ro}.

\bibitem{SourceCPU}
Bana\'s R, Bazi\'nska J, zewicz OL, Strza\l{}ka A. EMT6-Ro C++ implementation
  source code; Online accessed 11-May-2021.
\newblock \url{https://github.com/lamyiowce/tumor-ca}.

\bibitem{BachelorThesis}
Bana\'s R, Bazi\'nska J, zewicz OL, Strza\l{}ka A. Optimization of cancer
  treatment; 2019.

\bibitem{SciPyTTest}
E~Jones PP T~Oliphant. SciPy: Open source scientific tools for Python -
  \texttt{ttest\_ind} documentation; Online accessed 22-Dec-2021.
\newblock
  \url{https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_ind.html}.

\bibitem{SciPyKSTest}
E~Jones PP T~Oliphant. SciPy: Open source scientific tools for Python -
  \texttt{kstest} documentation; Online accessed 22-Dec-2021.
\newblock
  \url{https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html}.

\bibitem{Alfonso2020}
Alfonso JCL, Papaxenopoulou LA, Mascheroni P, Meyer-Hermann M, Hatzikirou H.
\newblock On the Immunological Consequences of Conventionally Fractionated
  Radiotherapy.
\newblock iScience. 2020;23(3):100897.
\newblock doi:{10.1016/J.ISCI.2020.100897}.

\bibitem{Montaseri2020}
Montaseri G, Alfonso JCL, Hatzikirou H, Meyer-Hermann M.
\newblock A minimal modeling framework of radiation and immune system synergy
  to assist radiotherapy planning.
\newblock Journal of Theoretical Biology. 2020;486:110099.
\newblock doi:{10.1016/J.JTBI.2019.110099}.

\bibitem{Wang2018}
Wang Y, Deng W, Li N, Neri S, Sharma A, Jiang W, et~al.
\newblock Combining Immunotherapy and Radiotherapy for Cancer Treatment:
  Current Challenges and Future Directions.
\newblock Frontiers in Pharmacology. 2018;0(MAR):185.
\newblock doi:{10.3389/FPHAR.2018.00185}.

\bibitem{Holland:2012}
Holland JH.
\newblock Genetic algorithms.
\newblock Scholarpedia. 2012;7(12):1482.

\bibitem{goldberg}
Goldberg DE.
\newblock Genetic Algorithms in Search, Optimization, and Machine Learning.
\newblock Addison-Wesley, Reading, Massachusetts; 1989.

\end{thebibliography}


\end{document}

