\chapter{Introduction}\label{chapter_1}

\section{Overview}


The identification of novel drug–target (DT) interactions is a substantial part of the drug discovery process. Most of the computational methods that have been proposed to predict DT interactions have focused on binary classification, where the goal is to determine whether a DT pair interacts or not. This particularly does not help much towards determining whether a certain drug is able to inhibit certain target protein. If the drug is able to inhibit the target, then what quantity of the drug is needed to do that cannot be determined by this binary classification. So, a binary classification does not help answering all of these questions. On the other hand, protein–ligand interactions assume a continuum of binding strength values, also called binding affinity. This value is good indication of drug–target (DT) interactions and also denotes how much of the drug is needed to inhibit the target protein. But predicting this value still remains a challenge. \Cref{visualization} shows how a drug interacts with a target protein.\\

\begin{figure}[!t]
  \centering
  \includegraphics[width=0.9\textwidth]{figures/figure1.png}
  \caption{Visualization of Drug-target interaction}
  \label{visualization}
\end{figure}

\section{Problem Statement}

In this work we wanted to address the problem of predicting the drug-target binding affinity value using computation. Normally we intake a drug as a cure for certain diseases.
What these drugs actually do is that they bind with very specific or certain target proteins making it unable to function and  cause the diseases. Most of the cases a single drug often bind with multiple type of protein as often the diseases are caused by multiple protein rather than a single protein. Besides, each drug has some toxicity value and solubility value. These values are important to denote whether the drug is actually usable. If a drug can inhibit a target protein but requires a large amount of intake then probably, we do not use that drug for trial and other phases. \Cref{problem_statement} illustrate the problem statement nicely,\\
\begin{figure}[!b]
  \centering
  \includegraphics[width=0.9\textwidth]{figures/figure2.png}
  \caption{Binding Affinity Prediction Problem}
  \label{problem_statement}
\end{figure}

\textbf{SMILES~\cite{weininger1988smiles}:} SMILES means simplified-molecular input line entry system. SMILES is a specification in the form of a line notation for describing the structure of chemical species using short ASCII strings. SMILES strings can be imported by most molecule editors for conversion back into two-dimensional drawings or three-dimensional models of the molecules. The original SMILES specification was initiated in the 1980s. It has since been modified and extended. In 2007, an open standard called OpenSMILES\footnote{https://en.wikipedia.org/wiki/Open\_standard} was developed in the open-source chemistry community. In our work we used the SMILES string for the drug molecule. The dataset we used is called the KIBA dataset~\cite{KIBA} and it has a total of 2,111 drug SMILES string. These drugs have a total of 64 unique characters.\\

\textbf{Protein Sequence:} Protein sequencing is the practical process of determining the amino acid sequence of all or part of a protein or peptide. This may serve to identify the protein or characterize its post-translational modifications. Typically, partial sequencing of a protein provides sufficient information (one or more sequence tags) to identify it with reference to databases of protein sequences derived from the conceptual translation of genes. In this problem we use such protein sequence of humans that we want to inhibit using drugs. These proteins cause health issues when certain type of disease occurs and inhibiting them becomes necessary to get rid of the diseases. In the KIBA dataset there are total of 229 such human proteins are available. The protein sequences are in string format and has a total of 25 unique characters.\\

\textbf{Binding Affinity:} Binding affinity is the strength of the binding interaction between a single biomolecule (e.g. protein or DNA) to its ligand/binding partner (e.g. drug or inhibitor). There are numerous metrics that denotes the drug-target binding affinity value. 
\begin{itemize}
    \item \textbf{IC50:} Half-maximal inhibitory concentration (IC50) is the most widely used and informative measure of a drug's efficacy. It indicates how much drug is needed to inhibit a biological process by half, thus providing a measure of potency of an antagonist drug in pharmacological research.
    \item \textbf{$K_D$:} $K_D$ is determined experimentally and is a measure of the affinity of a drug for a receptor. More simply, the strength of the ligand–receptor interaction. To determine $K_D$, a fixed mass of membranes (with receptor) are incubated with increasing concentrations of a radioligand until saturation occurs.
    \item \textbf{$K_i$:} For noncompetitive inhibition of enzymes, the $K_i$ of a drug is essentially the same numerical value as the IC50, whereas for competitive and uncompetitive inhibition the $K_i$ is about one-half that of the IC50's numerical value.
    

\end{itemize}

Each of these metrics have some merits and demerits from one another. So, in the KIBA dataset they tried to take the merits of all these metrics and combine them into a KIBA score and the predictor’s task is to predict this score. The lower the KIBA score the better the drug is to inhibit the protein with little amount of intake.\\

\section{Motivation}

Drug discovery is the process through which potential new medicines are identified. It involves a wide range of scientific disciplines, including biology, chemistry and pharmacology. Historically, drugs were discovered by identifying the active ingredient from traditional remedies or by serendipitous discovery, as with penicillin. More recently, chemical libraries of synthetic small molecules, natural products or extracts were screened in intact cells or whole organisms to identify substances that had a desirable therapeutic effect in a process known as classical pharmacology. After sequencing of the human genome allowed rapid cloning and synthesis of large quantities of purified proteins, it has become common practice to use high throughput screening of large compounds libraries against isolated biological targets which are hypothesized to be disease-modifying in a process known as reverse pharmacology. Hits from these screens are then tested in cells and then in animals for efficacy.\\

Modern drug discovery involves the identification of screening hits, medicinal chemistry and optimization of those hits to increase the affinity, selectivity (to reduce the potential of side effects), efficacy/potency, metabolic stability (to increase the half-life), and oral bioavailability. Once a compound that fulfills all of these requirements has been identified, the process of drug development can continue. If successful, clinical trials are developed.\\

Modern drug discovery is thus usually a capital-intensive process that involves large investments by pharmaceutical industry corporations as well as national governments (who provide grants and loan guarantees). Despite advances in technology and understanding of biological systems, drug discovery is still a lengthy, expensive, difficult, and inefficient process" with low rate of new therapeutic discovery. It takes about 10 years of time and \$2.6B dollars to develop a single drug. \\

\begin{figure}[H]
  \centering
  \includegraphics[width=0.9\textwidth]{figures/figure3.png}
  \caption{Drug Discovery Process in the USA}
  \label{drug_discovery_timeline}
\end{figure}

Currently we are going through a pandemic and we cannot wait ten years in a regular interval of lockdowns to have a drug. We need to find some way to discover drugs as quickly as possible. \\

Drug discovery is basically a search problem where we need to search a drug that cure a disease with little amount of intake. So, we can think that we are in a chemical maze where we need to find appropriate path to find the ideal drug. But doing that is not easy, there are $10^60$ ~\cite{kirkpatrick2004chemical} drug like molecule can be available and the whole USA industry can test about $10^5$ drugs per day. So, if we want to search the ideal drug for a disease in current setup it will be impossible if we want to brute force all the possible solution. So, we need to use computation in the drug discovery process.\\

The identification of novel drug–target (DT) interactions is a substantial part of the drug discovery process. Most of the computational methods that have been proposed to predict DT interactions have focused on binary classification, where the goal is to determine whether a DT pair interacts or not. However, protein–ligand interactions assume a continuum of binding strength values, also called binding affinity and predicting this value still remains a challenge. The increase in the affinity data available in DT knowledge-bases allows the use of advanced learning techniques such as deep learning architectures in the prediction of binding affinities. It will play an important role in the virtual screening for drug discovery.\\

\section{Limitation}
We know that the identification of drug-target binding affinity is a substantial part of the drug discovery process but it is not the only thing that solve the drug discovery pipeline and improve its timeline. There are several other factors that play a significant role in the drug discovery pipeline. For instance, there is Lipinski's rule of five, also known as Pfizer's rule of five or simply the rule of five (RO5), is a rule of thumb to evaluate drug likeness or determine if a chemical compound with a certain pharmacological or biological activity has chemical properties and physical properties that would make it a likely orally active drug in humans. The rule was formulated by Christopher A. Lipinski in 1997, based on the observation that most orally administered drugs are relatively small and moderately lipophilic molecules~\cite{lipinski1997experimental, lipinski2004lead}.\\

Lipinski analyzed all orally active FDA-approved drugs in the formulation of what is to be known as the Rule-of-Five or Lipinski's Rule. The Lipinski's Rule stated the following:

\begin{itemize}
    \item Molecular weight $< 500$ Dalton
    \item Octanol-water partition coefficient (LogP) $< 5$
    \item Hydrogen bond donors $< 5$
    \item Hydrogen bond acceptors $< 10$
\end{itemize}

Though our work predicts the binding affinity value as an end-to-end system but it does not consider these RO5 for a particular drug. So, for this the whole process is not completely automated at this moment. We can additionally address this constrain in future work of ours.\\

\section{Summary}

Predicting drug-target binding affinity is very problematic and time consuming for drug discovery process. Accurate and quick detection of binding affinity could increase the efficiency of this process. So, the predicting algorithms had to be computationally powerful and at the same time as precise as possible. The neural network approaches provided the desired level of computational power required for analyzing binding affinity. As for being precise, it was required to build the algorithm in such way that it would make the minimum number of mistakes. This is where the machine learning or deep learning algorithm came to help, as this model automatically traces back its own steps and adjusts itself to become more efficient and accurate. \\ 

\endinput

