\documentclass[]{article}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{blkarray}
\usepackage{caption}
\captionsetup[figure]{font=small,labelfont=small}
\usepackage{authblk}
\usepackage{graphicx, epstopdf}
\usepackage{lineno}
\usepackage{float}


\title{\Huge Simulations of 3D organoids suggest inhibitory neighbour-neighbour signalling as a possible growth mechanism in \textit{EGFR-L858R }mutant alveolar type II cells

\Large Short title: Early effects of oncogenic mutation on lung stem cell behaviour}


\author[1]{Helena Coggan}
\author[2]{Clare E. Weeden}
\author[3]{Philip Pearce}
\author[4]{Mohit P. Dalwadi}
\author[5]{Alastair Magness}
\author[6]{Charles Swanton}
\author[7]{Karen M. Page}
\affil[1]{University College London, Department of Mathematics; ORCID 0000-0002-7798-9318}
\affil[2]{Cancer Evolution and Genome Instability Laboratory, The Francis Crick Institute; ORCID 0000-0002-1561-1416}
\affil[3]{University College London, Department of Mathematics; UCL Institute for the Physics of Living Systems; ORCID 0000-0001-5788-3826}
\affil[4]{University College London, Department of Mathematics; UCL Institute for the Physics of Living Systems; ORCID 0000-0001-5017-2116}
\affil[5]{Cancer Evolution and Genome Instability Laboratory, The Francis Crick Institute; ORCID 0000-0001-9876-3863}
\affil[7]{University College London, Department of Mathematics; ORCID 0000-0003-4189-4664}
\affil[6]{Cancer Evolution and Genome Instability Laboratory, The Francis Crick Institute; Cancer Research UK Lung Cancer Centre of Excellence, University College London Cancer Institute; and Department of Oncology, University College London Hospitals; ORCID 0000-0002-4299-3018}



\begin{document}

\maketitle

\section{Abstract}

Mutations in the epidermal growth factor receptor (EGFR) are common in non-small cell lung cancer (NSCLC), particularly in never-smoker patients. However, these mutations are not always carcinogenic, and have recently been reported in histologically normal lung tissue from patients with and without lung cancer. To investigate the outcome of EGFR mutation in healthy lung stem cells, we grew murine alveolar type-II organoids monoclonally in a 3D Matrigel. Our experiments showed that the \textit{EGFR-L858R} mutation induced a change in organoid structure: mutated organoids displayed more `budding', in comparison to non-mutant controls, which were nearly spherical. On-lattice computational simulation suggested that this can be explained by an increase in reproductive fitness combined with inhibitory neighbour-neighbour signalling in mutated organoids. We predict that these effects prohibit division by cells in the interior of the organoid and boost the fitness of cells on the surface, since this differential growth is sufficient to cause `budding' structures in our simulations. These results suggest that the L858R mutation produces structures which expand quickly from surface protrusions. We suggest that the likelihood of L858R-fuelled tumorigenesis is affected not just by random fluctuations in cell fitness, but by whether the mutation arises in a spatial environment that allows mutant cells to reproduce without being forced to encounter each other. These data may have implications for cancer prevention strategies and for understanding NSCLC progression.

\section{Author Summary}
Cancer is driven by the development of genetic mutations. Some mutations which appear in aggressive lung cancers, particularly in people who have never smoked, have also been found to exist quite harmlessly in perfectly healthy people. Although inflammatory cytokines have been highlighted as important promoters of tumour formation, it is unclear what additional stimuli are required in order to drive a `normal cell' harbouring an oncogenic mutation into an invasive tumour. To examine this, we looked at the behaviour of stem cells with an activating mutation in EGFR, L858R, when they were given all the nutrients and space required to grow uninhibited in three dimensions. We used computational simulations to model their growth, and predicted that these cells seemed to be suppressing the division of any other cells they touch. We hypothesise that in the very early stages of cancer development, this mutation gives cells a reproductive advantage by preventing the division of non-mutant cells in their environment and driving down competition for space and resources. This also suggests that the success of these pre-cancerous cells depends on their spatial environment and surrounding cell ecology. We hope that this insight into early cancer development will drive more research into the consequences of cell-cell interaction dysfunction on early tumour initiation.
\section{Introduction}

Lung cancer is the leading cause of cancer death worldwide \cite{Sung2021}. Approximately a quarter of lung cancer patients are `never-smokers' (classified as those who have smoked fewer than 100 cigarettes in their lifetime) \cite{Sun2007}, and this proportion appears to be increasing \cite{Pelosof2017}. Never-smoking lung cancer is also more common in women than men \cite{Schabath2019}.

Lung adenocarcinomas are thought to arise from alveolar type II (AT2) cells within the healthy alveolar epithelium \cite{Sutherland2014}. The causes of lung cancer in never-smokers are unclear, although previous studies have highlighted germline genetics \cite{Zhang2021} and exposure to external factors such as infections, radon \cite{Corrales2020} and ambient air pollution \cite{Swanton2022}. Some specific lung cancer-associated mutations, in particular in EGFR, are known to occur more frequently amongst patients with no history of smoking \cite{Chapman2016}, and are found as clonal driver mutations in lung adenocarcinomas \cite{Jamal2017}. However, their presence alone is insufficient for tumorigenesis: Swanton and colleagues have recently reported that 18\% of normal lung tissue samples in patients both with and without lung cancer were found to carry EGFR mutations \cite{Swanton2022}. In order to determine the conditions necessary for lung cancer initiation, we must look further, to the cellular and environmental context in which carcinogenic mutations arise.

It is well-known that the emergence of cancer is probabilistic, and that tumours can take a variety of winding evolutionary paths to acquiring their six characteristic hallmarks \cite{Hanahan2000}. No mutation is guaranteed to cause cancer. The number of cells in the human body is such that every possible genetic mutation is likely to exist in at least one cell in everyone \cite{Traulsen2010}; the fact that some people do not develop cancer suggests that not all of these mutations lead to disease in every case. The reproductive fitness of a mutant cell depends not just on tissue context \cite{Haigis2019}, but also on spatial constraints \cite{West2021} and the ecology of the wider cell population \cite{Gatenby2014}. Subclonal cells interact with each other and compete for resources, giving rise to highly genotypically heterogeneous tumours \cite{Tabassum2015}, \cite{Gerlinger2012}. Within mathematics, the field of evolutionary game theory describes the effect of such interactions on population composition, and has seen a variety of recent applications to cancer modelling \cite{Coggan2022}. But to make full use of these theories we need strategic quantification: in order to predict a phenotype's success, we must determine how it interacts with other phenotypes, and thus how its population will rise and fall within a tumour population. To identify the evolutionary benefit of a particular mutation arising in a specific cell type, then, we must consider the effect on that cell's behaviour. Only then will we be able to predict its contribution to tumorigenesis.

The aim of this study is to analyse the phenotypic changes conferred by the \textit{EGFR-L858R }mutation on alveolar type-II cells. The single nucleotide substitution L858R in exon 21 comprises around 40\% of all EGFR mutations in lung cancer patients \cite{Shigematsu2006}. It is a missense mutation, affecting the intracellular kinase domain of the epidermal growth factor receptor (EGFR), which in turn affects a cell's response to many common growth factors. Biochemical and structural studies have shown that the \textit{EGFR-L858R} mutation promotes dimerisation \cite{Shan2012} and destabilises the inactive formation \cite{Yun2007} of EGFR, and thus causes abnormal levels of receptor activity. The function of this mutation is also highly clinically relevant. \textit{EGFR-L858R} mutations confer sensitivity to treatment with EGFR tyrosine kinase inhibitors, although acquired resistance can rapidly develop \cite{Hong2019}, \cite{Reita2021}. In addition, L858R has been shown to enhance invasiveness in adenocarcinomas \cite{Tsai2015}, and mice with L858R mutations induced in alveolar type-II cells rapidly develop diffuse carcinomas \cite{Politi2006}. It is this particular aspect of mutant behaviour that we address in the present study.

We infer an invasive mutant phenotype using an agent-based model (ABM) of organoid growth in a three-dimensional organoid culture, by comparing the structures of spheroids grown monoclonally from mutant and wild-type AT2 cells. Agent-based models have been common throughout systems biology for many years \cite{An2009}, and are useful for describing systems where individual `agents' (here AT2 cells) interact and reproduce probabilistically according to defined rules. Our focus is on the emergence of a characteristic `budding' structure in some mutant cell clusters, as opposed to their wild-type counterparts, which grow spherically. These organoids are composed of thousands or tens of thousands of cells, and are smaller than is required for the development of vascularity. The field of mathematical modelling of avascular tumour growth is larger than can be fully summarised here (see \cite{Roose2007, Byrne2009, Byrne2012} for an introduction). One common methodology is a `continuum' approach, where cell density is described as a continuous scalar field, which evolves with time. Cell-cell interactions and surface-dominated growth can be included in a continuum model \cite{Breward2002, Ciarletta2013}, and the resulting approach is amenable to mathematical analysis, which has shown surface-dominated growth to drive the formation of non-spherical structures. These models are particularly advantageous in the study of reaction-diffusion systems involving nutrient gradients which evolve with time \cite{Miura2008}, and when analysing the contribution of various biomechanical processes to general morphological instabilities \cite{Giverso2016}.  While these approaches are broadly useful, in many situations the problem we are modelling is three-dimensional, extremely asymmetric, and involves fully developed and morphologically specific protrusions, which require high spatial granularity to model. Here, where we wish to test nonlinear growth laws where the discrete number of neighbours a cell has is important, and where the number of cells involved is relatively small (on the order of thousands or tens of thousands), it is both conceptually and computationally easier to model the location and action of each cell directly. In recent years, ABMs have been increasingly used to shed light on complex systems of cell-cell interaction \cite{Sadhukhan2021, Sivakumar2022}, and we continue that approach in this study.

We use a novel on-lattice algorithm, in which space is divided into a fixed three-dimensional lattice whose points may be either occupied or empty, to simulate cells as they divide and push each other aside. This approach significantly speeds up computation times compared to off-lattice simulation (where the distance between of each pair of cells must be recalculated at each timepoint), which allows for thorough hypothesis testing and parameter sweeps. We test various growth hypotheses, including those where cell division is limited by differentiation and by the presence of surrounding cells, and compare the simulated organoids to experimentally observed morphologies. We find that the structural difference between mutant and non-mutant organoids can be explained by a model in which the oncogene increases cell fitness and allows them to transmit inhibitory signals to their neighbours. The overall effect is to suppress division in surrounded cells and to promote the reproduction of cells on the surface. We also find that the initial growth rates of non-mutant clusters follow a skewed-normal distribution, and that the growth of initially weakly-growing clusters slows much more rapidly than that of initially-fit clusters, suggesting an `evolutionary bottleneck' even in nutrient-rich \textit{in vitro} conditions. We hope this will shed light on the origin of invasive \textit{EGFR-L858R-}positive cancers, and the mechanisms by which they emerge from healthy tissue.

\section{Results}


We used a genetically engineered mouse model of EGFR-L858R-driven lung adenocarcinoma (Rosa26\textsuperscript{LSL-tTa/LSL-tdtomato}; \textit{TetO-huEGFR}-\textsuperscript{L858R mice} \cite{Swanton2022}; referred to as ET mice), where both the \textit{EGFR-L858R} transgene and tdTomato fluorescent reporter protein are only expressed upon delivery of Cre recombinase. Lox-stop-lox tdTomato mice were used as control (\textit{Rosa26}\textsuperscript{LSL-tdTomato/LSL-tdTomato}; referred to as T mice). Alveolar type II cells were purified from non-Cre treated T and ET mice, before inducing expression of the oncogene and/or tdTomato in vitro with adenoviral-CMV-Cre incubation using published methods \cite{Major2020},  \cite{Dost2020}. 10,000 AT2 cells were seeded in each organoid assay along with 50,000 supporting lung fibroblasts \cite{Choi2020}; in each experiment only some AT2 cells grew into organoids (with organoid forming efficiency of around 1-2\%). After 14 days, organoids were extracted from matrigel, stained with antibodies and analysed by 3D wholemount confocal microscopy \cite{Dekkers2019}. These analyses revealed a characteristic `budding' structure occurring in mutant organoids, but not wild-type organoids (Figure 1). We can quantify this effect by analysing the ratio of the perimeter of an organoid's cross-section, \(p\), to the square root of the area of its cross-section, \(\sqrt{a}\) (the 'stretching coefficient'). If all organoids were perfectly spherical, each organoid would have a \(p / \sqrt{a}\) ratio of \(2\sqrt{\pi}\), or roughly \(3.5\). Any protrusions or elliptical deviations will be a less `efficient' use of space and will increase this ratio. Using ImageJ to analyse the images in Figure 1, we find the average \(p / \sqrt{a}\) ratio is about 20 for imaged non-mutant clusters, but 44 for mutant clusters. Comparing these images and the distribution of `stretching coefficients' shows that the wild-type distribution is bimodal: one low-value peak for small, spherical clusters with low overall growth, and one peak at a slightly higher value for larger clusters, which have grown into slightly-imperfect spheroids. The mutant distribution makes clear the two phenotypes displayed by the EGFR-mutant organoids: they have one peak at a low stretching-value, corresponding to smaller clusters with little deformation, but the higher end of the distribution contains large, highly deformed clusters with secondary spheroids, never witnessed in wild-type organoids. The objective of this study is to determine the mechanism which would allow the development of these highly asymmetrical clusters.


\begin{figure}[H]
  \includegraphics[width=.8\linewidth]{newimage-eps-converted-to.pdf}
  \includegraphics[width=.8\linewidth]{violin-eps-converted-to.pdf}
  \caption{Top: Representative 3D confocal microscopy of AT2 organoids from T mice (top panel) and ET mice (bottom panel). Organoids stained with anti-surfactant protein C (SPC, cyan) and anti-keratin 8 (Krt8, magenta), endogenous tdTomato expression. Results shown from 1 experiment comprising 3 mice; n = 2 independent experiments. Scale bar represents 100 microns. Bottom: a violin plot of the 'stretching coefficients' of the clusters as determined from these images.}
\end{figure}

\subsection{Simulation design}
To simulate the development of these structures, we use an on-lattice algorithm. In this framework, 3D space is divided into a 30x30x30 grid. At the end of each simulation step, every lattice point is either occupied by one cell or by none. Each lattice point (except those on the grid's faces) has 26 Moore neighbours, each of which is considered equally `close' to it within the algorithm. This ensures that when cells are pushed aside by new cells in a simulation step, they are not constrained to move only in six directions (up, down, left, right, forwards or backwards) but can move in \textit{26}, allowing the simulation of much more realistic organoid structures. A cell may therefore have up to 26 occupied neighbours. Further model assumptions, and their experimental justification, are summarised in Table 1.


\begin{table}[H]
\centering
\begin{tabular}{|p{0.4\linewidth} | p{0.4\linewidth}|}
\hline
\textbf{Model assumption}                                                                                  & \textbf{Experimental justification}                                                                                                                                                                                                                                                \\ \hline
All cells have the same volume (approximately \(80 \mu m^3\))                                              & The Matrigel is soft enough that cells are not significantly compressed and may push each other aside when they divide to maintain constant volume. The figure is extrapolated from measurements of areas of individual cells, approximating cells as spherical                    \\ \hline
Cells remain in contact with their organoid of origin, and do not drift freely through the Matrigel        & Organoids are cohesive and contiguous; `clouds' of drifting fluorescent cells are not observed                                                                                                                                                                                      \\ \hline
Organoids grow independently and there is no crosstalk between them                                        & Top-down images of the well suggest that the organoids are seeded very far apart from each other relative to their diameter                                                                                                                                                        \\ \hline
External nutrient concentration does not deplete over time                                                          & Nutrients are regularly replenished by the experimenter                                                                                                                                                                                                                            \\ \hline
The probability that a cell will divide depends only on the number of cells in its immediate neighbourhood & The Matrigel is highly diffusive (eg. VEGF has a diffusion coefficient in it of the order of \(10^6 \mu m /h\) \cite{Miura2009}), so concentration gradients in nutrients, growth factors or any other relevant chemical cannot be maintained on the timescale of cellular reproduction. Communication between cells thus happens through mechanotransduction when cells touch \\ \hline
All clusters arise from a single cell                                                                      & Fusion of growing organoids is not experimentally witnessed                                                                                                                                                                                                                        \\ \hline
\end{tabular}

\caption{Assumptions made when modelling the system computationally, and their experimental justifications.}
\end{table}




The algorithm works as follows. At the start of the simulation only a single cell is occupied at the centre of the grid. At every timepoint, each existing cell either divides or does not, according to some probabilistic function of the number of cells in its neighbourhood (which we vary according to the hypothesis being tested). If a cell divides, that lattice point is instantaneously occupied by two cells. This is referred to as the \textit{reproduction} step. 

  The next step in the algorithm is \textit{spatial adjustment}, where cells on multi-occupied lattice points are shifted around until they find empty points to settle on, mimicking the physical process of cells pushing each other aside. This step is iterative and repeats until every cell has its own point. At each iteration of the adjustment step, one cell from every multi-occupied point is moved to one of its neighbouring points. The rules for choosing a lattice-point to move to are as follows.
\begin{enumerate}
\item If there is are any empty neighbour-points, choose between them at random with probability \(p \propto e^{\tau n_1}\), where \(n_1\)is the number of occupied lattice-points in the \textit{neighbour-point's} immediate neighbourhood, and \(\tau\) is a surface-tension parameter.
\item If there are no empty neighbour lattice-points, choose one at random with probability \(p \propto e^{-\gamma  n_0}\), where \(n_0\) is the number of cells currently occupying it and \(\gamma>0\) is a `repulsion parameter'.
\end{enumerate}
The repulsion parameter \(\gamma\) is included to dissuade cells from moving between double-occupied points. Cells are pushed through the cluster to minimise compression, and so will move preferentially to points with fewer cells occupying them. This parameter is designed to mimic the effect of a pressure gradient and is kept high (\(\gamma = 1\)). The `surface tension' parameter \(\tau\) controls the cohesion of cluster shapes and is designed to mimic the effect of cell-cell adhesion. When it is high, cells will preferentially move to empty points next to occupied points, to minimise cluster surface area. This is a variable parameter.

A key benefit of this algorithm is that whether or not two lattice-points are neighbours is only ever calculated once, when the grid is generated, and does not need to be worked out again with every simulation (or, worse, with every simulation step). The number of cells touching a focal cell can be calculated by looking up the list of its neighbouring lattice-points and checking whether each of them is occupied; it does not require calculating the distance between the focal cell and every other cell in the organoid. Its computational cost does not scale with the square of the current number of cells of the cluster, as it would in an off-lattice simulation, but linearly with the number of occupied cells in the system. This means that clusters of many tens of thousands of cells can be simulated in a few seconds, where off-lattice simulations struggle to simulate more than a few hundred cells efficiently. This computational efficiency allows for quick parameter sweeps.

Clusters are simulated until they reach a certain experimentally realistic physical size, comparable to the observed organoid sizes (a maximum diameter of roughly \(100 \mu m\), which generally requires around 5000 or 10000 cells, depending on the division probability function under investigation). The structure of the resulting organoid is examined and compared to experimental observations. 

\subsection{Nutrient absorption-based growth, anchorage-dependent growth and differentiation are insufficient to produce mutant organoid structures}

\subsubsection{Absorption-based growth}

A plausible hypothesis for the emergence of `budding' structures is that cells on the surface of the organoid (i.e. those with fewer neighbouring cells) have better access to nutrients and so have higher fitnesses. We assume here that all cells have sufficient access to nutrients to maintain a non-zero growth rate even when completely surrounded (see Table 1); later we will investigate a regime in which surrounded cells are assumed to have zero growth rate. 

We test a division probability which is a monotonically increasing function of the number of empty lattice-points in a cell's immediate environment, \(n\), which eventually saturates on physical grounds \cite{EUNGDAMRONG2004}. Within this framework, the probability \(p\) that a cell will divide in a given timestep \(dT\) is

\[p_{abs}(\alpha, \beta, h, c, n, dT) = \left(\alpha + \frac{\beta}{1+e^{-h(c-(26-n))}}\right)dT \]
where \(\alpha >0\) is a `baseline' division rate possessed by completely surrounded cells; \(\beta\) is the strength of the nutrient-derived fitness boost (such that the maximum possible division rate of an isolated cell is \(\alpha + \beta\)); \(c\) is the threshold number of occupied neighbours at which the proliferative capacity of a cell decreases; and \(h\) is the `steepness' of that saturation, and 26 is the maximum number of possible neighbours. The variation of this function with \(h\) and \(c\) is illustrated in Figure 2.

\begin{figure}[h!]
 
  \includegraphics[width=\linewidth]{pfig-eps-converted-to.pdf}
  \caption{An illustration of the nutrient-absorption-based probability function with \(\alpha = 1, \beta = 2\), i.e. the probability of reproducing in a time interval \(dT=0.01\) (days). On the top, \(c=13\) empty neighbours and the steepness \(h\) is varied. On the bottom, \(h=1\) and \(c\) is varied.}
\end{figure}


Throughout these experiments \(dT = 0.01\) days, or just under 15 minutes, to ensure the number of cells generated in any given step is not too high. In Figure 3, for example, we see that even completely isolated cells only ever have a 3\% probability of reproducing per timestep. In general we consider very large values of \(h\) to be biologically implausible, as they would imply sudden changes in a cell's ability to proliferate if it acquires one or two extra neighbours. We therefore restrict ourselves to values of \(h\) of order 1 or below. In a system where \(\beta = 0\) and \(\alpha > 0\), all cells would have equal ability to divide and the resulting organoid would be necessarily spheroidal. To test whether any other sets of parameters can result in the development of secondary spheroids, then, we fix \(\alpha\) and vary other parameters to test the possible results of the growth mechanism.

Varying \(\beta, c, h\) relative to \(\alpha\) (the baseline growth rate, kept at \(\alpha = 1\)), and altering the cell-cell adhesion parameter \(\tau\), results in uniformly spherical growth (with the surface rougher or smoother depending on whether \(\tau\) is high or low). Some representative samples are shown in Figure 3. The physical reason for this is that if all cells have a baseline fitness, then division will occur to some degree or another everywhere in the cluster. Any protrusions that form momentarily at the surface will be evened out by pressure from internal reproduction. Gaps at the surface, corresponding to unoccupied lattice-points, will be filled quickly by newly-produced cells squeezed out from the organoid. This is true even when the fitness boost is very large (ten times the baseline fitness) and when surface tension is low; in any structure where dividing cells remain in contact with each other, the number of mostly surrounded cells will be much larger than the number of completely isolated cells. Thus most reproduction will happen within the cluster, so instabilities at the surface will not develop and the organoid will remain spherical.

\begin{figure}[H]
 \includegraphics[width=.32\linewidth]{0-eps-converted-to.pdf}
  \includegraphics[width=.32\linewidth]{1-eps-converted-to.pdf}
   \includegraphics[width=.32\linewidth]{2-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{3-eps-converted-to.pdf}
     \includegraphics[width=.32\linewidth]{4-eps-converted-to.pdf}
      \includegraphics[width=.32\linewidth]{5-eps-converted-to.pdf}
       \includegraphics[width=.32\linewidth]{6-eps-converted-to.pdf}
        \includegraphics[width=.32\linewidth]{7-eps-converted-to.pdf}
         \includegraphics[width=.32\linewidth]{8-eps-converted-to.pdf}
          \includegraphics[width=.32\linewidth]{9-eps-converted-to.pdf}
           \includegraphics[width=.32\linewidth]{10-eps-converted-to.pdf}
            \includegraphics[width=.32\linewidth]{11-eps-converted-to.pdf}
  \caption{Illustrations of the effect of varying (top row) `boost' \(\beta\), (second row) steepness \(h\), (third row) cell-cell adhesion \(\tau\), and (fourth row) threshold \(c\). Default parameters are \(\beta=2, c=20, h=1, \tau=1\); all parameters are at these values unless stated otherwise above a graph. Very low surface tension gives a rough surface (not experimentally witnessed), so we use a default \(\tau=1\). No other parameter has a noticeable effect on the resulting structure. As only the relative values of the parameters affect structure, \(\alpha=1\) in every image. Each simulation is run until the first timestep in which the cluster contains more than 10,000 cells. Parameters kept constant for each row are displayed on the left-hand side. All axes are in \(\mu m\); here 10,000 cells gives an experimentally realistic diameter of \(100 \mu m\). Colour indicates the number of occupied neighbours a cell has: yellow cells are surrounded, whilst blue cells are isolated.}
\end{figure}

\subsubsection{Anchorage-dependent growth}

Another plausible hypothesis is `anchorage-dependent growth' \cite{Ruoslahti1994}, whereby cells require adhesion to extra-cellular matrix (ECM) in order to progress through the cell cycle. This suggests that reproductive fitness increases monotonically, with a physically motivated saturating effect, with the number of neighbouring cells. To model this, we use a division probability function


\[p_{anc}(\alpha, \beta, h, c, n, dT, dT) = p_{abs}(\alpha, \beta, -h, c, n, dT) =  \left(\alpha + \frac{\beta}{1+e^{h(c-(26-n))}}\right)dT \]where all parameters have the same meaning as before. Now \(c\) is the threshold number of occupied neighbours at which a cell's proliferative capacity \textit{increases}, and fitness \textit{declines} with \(n\), a cell's number of empty neighbouring lattice-points. We find that this leads to spherical growth since cells benefit from being as close to one another as possible, with protruding cells at a disadvantage (see Figure 4).

\begin{figure}[H]
   \includegraphics[width=.32\linewidth]{12-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{13-eps-converted-to.pdf}
     \includegraphics[width=.32\linewidth]{14-eps-converted-to.pdf}
  \caption{Four representative simulations of anchorage-dependent growth with varying `boost' \(\beta\), cell-cell adhesion \(\tau\), steepness \(h\), and threshold \(c\). If \(\alpha = 0\) (isolated cells never reproduce), then the steepness must be kept low, or the cluster never becomes larger than a single cell. If \(\alpha > 0\) then we can test hypotheses with very steep functions, i.e. the reproductive capacity of a cell doubles when it has 2 neighbours rather than 1 (bottom right). Secondary spheroids are never produced. All annotations are as Figure 3.}
\end{figure}

\subsubsection{Differentiation-based growth}
A further testable hypothesis is differentiation-based growth. A dividing progenitor cell might produce, for example, a progenitor cell (which can reproduce in turn) or a differentiated cell (which cannot). Could differences in the division patterns of progenitor cells create budding structures?

To test this, we make a slight alteration to our simulation, and divide cells into two types: `active' and `inactive'. Both cells can be pushed aside when other cells reproduce, but only active cells can divide. When an active cell divides, the new cell is active with probability \(q\), and inactive with probability \(1-q\). We also implement a `domino effect' during the adjustment step, to prevent newly-produced cells from ending up very far from their mother cells as soon as they are produced. We keep track of the type of cell last added to any lattice point. When a cell is moved from that point, we make sure to keep at least one cell of that type there after the movement step. The moved cell is chosen at random from those remaining. For example, if a point initially contains one active cell, and has an inactive cell moved to it during the adjustment step (so that it is double-occupied), then in the next step, the active cell will be moved. This ensures that cells are not propagated too far through the cluster during a single adjustment step.
  
Figure 5 shows the results of these simulations. (Here we assume that the division rate of all active cells is constant with respect to the number of empty lattice points, \(p_{\text{diff}}(n, dT) = \alpha dT\), to independently test the effect of differentiation.) 

\begin{figure}[H]
   \includegraphics[width=.32\linewidth]{16-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{17-eps-converted-to.pdf}
     \includegraphics[width=.32\linewidth]{18-eps-converted-to.pdf}
  \caption{Three representative simulations of differentiation-patterned growth, with varying stem-cell probabilities (increasing left to right). In all images, \(\tau=1.0, \alpha=1\).}
\end{figure}
  
This mechanism does not produce secondary spheroids, but low \(q\) (i.e. high levels of differentiation) do have a lengthening effect on the cluster, as can be seen in the leftmost image (\(q=0.3\) produces a shape that is more ellipsoidal than spherical). This is because active cells are relatively rare within the cluster, and so reproduction can easily become unevenly distributed. Active cells produce more active cells, so if by chance a spheroid ends up with more active cells in one half than another, that lopsidedness will increase over time. However, because reproduction is still happening within the cluster, new cells-- both active and inactive-- will push their neighbouring cells outward radially, and even out any surface protrusions, just as with nutrient-based growth. This makes intuitive sense: it would be difficult to biologically justify any hypothesis in which mutant budding structures resulted from their producing cells which were less likely to reproduce, given that the mutation under study is oncogenic. Cell differentiation may well be happening within this system (though we do not model it from this point forward), but our model suggests that it is not consistent for it to be the mechanism primarily responsible for `budding'. Adding nutrient-based growth to the system does not change this result (see Figure 6).

\begin{figure}[H]
   \includegraphics[width=.32\linewidth]{19-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{20-eps-converted-to.pdf}
     \includegraphics[width=.32\linewidth]{21-eps-converted-to.pdf}
  \caption{Three representative simulations of differentiation-patterned growth (\(q=0.3\)) with nutrient-based reproduction. Here \(\alpha=1, \beta=5, h=\tau=1\) and \(c\) is varied. Low \(c\) can produce `bumps' (see left) but not protrusions as seen experimentally (see Figure 1).}
\end{figure}



\subsection{Strong neighbour suppression is necessary to produce mutant organoid structures}

The simulations above establish that so long as surrounded cells are able to reproduce, clusters will remain largely spherical. What happens if surrounded cells are completely prevented from dividing?  To test this hypothesis, we use the division probability function 

\[p_{\text{NS}}(n, dT) = p_{abs}(0, \beta, h, c, n, dT) = \frac{\beta dT}{1+e^{-h(c-(26-n))}} \]
which is simply \(p_{abs}(n, dT)\) with \(\alpha = 0\). (The reader is referred to Figure 12 for graphical representation of all relevant division functions.) Here cells must have empty neighbour-points in order to reproduce; the baseline division probability of surrounded cells is zero. This may happen for a couple of mechanistic reasons. If cells are sufficiently densely packed that nutrients cannot reach cells at the centre of the cluster, we might expect them to stop dividing. This would only happen when cells are completely surrounded, however, and so would correspond to a high threshold \(c\). If cells were stopped from dividing when not completely surrounded, however, we might attribute this to active inhibition by neighbouring cells.

Using this growth mechanism, sustained protrusions can develop, and we observe structures which look remarkably similar to those seen in experiments. The set of parameters which produce the most visually similar structures to those seen in experiment are roughly \(\tau = 1.0, h=1, c=10\) (see Figure 7). This corresponds to a relatively high level of cell-cell adhesion, and a division probability which is saturated for a cell with fewer than 8 neighbours (i.e. when 30 \% of its surface is covered) and close to zero for a cell which has more than 12 (i.e. when just under half of its surface is covered). This low value of \(c\) is worth noting: cells that are still very much in contact with the nutrient-rich Matrigel cannot divide, so this corresponds to a hypothesis in which cells are actively inhibiting each other, not simply depriving each other of nutrients.

Cross-sectional analysis of the simulation shows that these organoids grow as solid structures, without `air pockets', which also matches experimental observation (when grown organoids are removed from the Matrigel and analysed at the end of the experiment, they are found to be solid). 

\begin{figure}[H]
    \includegraphics[width=.49\linewidth]{22-eps-converted-to.pdf}
     \includegraphics[width=.49\linewidth]{23-eps-converted-to.pdf}
      \includegraphics[width=.49\linewidth]{24-eps-converted-to.pdf}
       \includegraphics[width=.49\linewidth]{25-eps-converted-to.pdf}
  \caption{Four example structures of neighbour-suppressed clusters grown with \(\alpha=0, \beta = 10, \tau = 1.0, h = 1, c=10\). In the bottom row only half of the cluster is displayed, to show a cross-section: we can see that the organoid grows solidly. All axes are in \(\mu m\); here 5,000 cells gives an experimentally realistic diameter of \(100 \mu m\). Colour indications are as before.}
\end{figure}

We can see that the baseline division rate of surrounded cells, \(\alpha\), must be negligible compared to \(\beta=10\) to allow the development of budding growth from Figure 8, below, which shows the emergence of experimentally-observed organoid structures as \(\alpha\) is decreased from \(0.1\) (near-spheroidal growth) to zero (budding growth). The emergence of budding growth is gradual and takes the form of a slow increase in the number of protrusions, or the roughness of the organoid's surface, as the clusters become less cohesive. We can see that as \(\alpha\) approaches zero, the organoid becomes increasingly susceptible to budding instabilities as a result of surface-based growth. In the simulations which best replicate experimental observations, the baseline division rate of the cells is zero.

\begin{figure}[h!]
    \includegraphics[width=.32\linewidth]{alpha_45-eps-converted-to.pdf}
   \includegraphics[width=.32\linewidth]{alpha_47-eps-converted-to.pdf}
\includegraphics[width=.32\linewidth]{alpha_49-eps-converted-to.pdf}
\includegraphics[width=.32\linewidth]{alpha_51-eps-converted-to.pdf}
\includegraphics[width=.32\linewidth]{alpha_53-eps-converted-to.pdf}
\includegraphics[width=.32\linewidth]{alpha_55-eps-converted-to.pdf}
  \caption{Six example structures of neighbour-suppressed clusters grown with \(\beta = 10, \tau = 1.0, h = 1, c=10\) and varying \(\alpha\). All axes are in \(\mu m\); here 5,000 cells gives an experimentally realistic diameter of \(100 \mu m\). Colour indications are as before.}
\end{figure}

Figure 9 illustrates the effect of the simulation parameters. Decreasing the steepness \(h\) means that the benefit of neighbouring empty lattice points (i.e. the inhibitory effect of neighbour-neighbour signalling) becomes approximately linear, which allows surrounded cells to divide, leading to spherical growth. In order to generate secondary spheroids, the benefit to surface cells must be steeply nonlinear, so the behaviour of these cells is a `switchlike' function of their circumstances. Surface-dependent growth requires the `surface' to be distinctly and sharply advantaged, i.e. for the proliferative capacity of a cell to increase significantly when it has below a certain number of neighbours. 

As a result of this dependence, parameters which have little to no effect in the nutrient-dependent simulation are much more important here. Decreasing cell-cell adhesion \(\tau\) results in rougher surfaces; increasing it results in near-perfect spheroids. Increasing \(c\), the number of occupied neighbours required to induce inhibition, allows more reproduction from surrounded cells and again results in spherical growth. (Lowering it too far, e.g. to \(c=5\), `jams' the simulation, as almost every cell has more than 5 neighbours and so reproduction quickly stops.) We can see this effect clearly in the left hand side of Figure 10: if \(c>16\), clusters become approximately spherical. This suggests that wild-type organoids, and those mutant organoids which do not develop budding structures, may experience either much weaker active inhibition from their neighbours or simple nutrient depletion (in which only completely surrounded cells are prevented from dividing). As discussed above, a variety of mechanisms are able to produce spheroidal growth, but of our tested hypotheses only strong neighbour suppression is able to reproduce `budding' structures, regardless of the simulation parameters.

\begin{figure}[H]
    \includegraphics[width=.32\linewidth]{30-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{31-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{32-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{33-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{34-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{35-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{36-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{37-eps-converted-to.pdf}
    \includegraphics[width=.32\linewidth]{38-eps-converted-to.pdf}
    
  \caption{Illustrations of the effect of varying (first row) steepness \(h\), (second row) cell-cell adhesion \(\tau\), and (third row) threshold \(c\). For all simulations \(\alpha=0, \beta=10\).  Default parameters are \(c=10, h=1, \tau=1\); all parameters are at these values unless stated otherwise above a graph. All axes are in \(\mu m\); here 5,000 cells gives an experimentally realistic diameter of \(100 \mu m\). Colour indications are as before.}
\end{figure}
\begin{figure}[H]
   \includegraphics[width=.49\linewidth]{neighbour-eps-converted-to.pdf}
   \includegraphics[width=.49\linewidth]{21-eps-converted-to.pdf}
  \caption{Left: The variation of the expected number of neighbours of a cell in the organoid, \(n_{occ}\), with the suppression threshold \(c\), where one representative simulation is taken at each point. Low \(n_{occ}\) suggests more isolated cells and a higher surface-to-volume ratio; a value of \(n_{occ}\) approaching 26 means that the vast majority of cells are surrounded and the organoid has grown spherically. Increasing \(c\) allows more surrounded cells to reproduce and causes spherical growth to emerge cleanly from `budding' growth. Approximately spherical growth is achieved after roughly \(c = 16\). Clusters were simulated with \(\beta = 5, \tau = 1.0, h=1\), and to 5000 cells. 
  Right: A cluster grown to around 10,000 cells with \(\alpha=0, \beta = 1, h=1, c=25, \tau =1.0\).}
\end{figure}

   We make two further significant observations. Firstly, when neighbour suppression is high, the experimentally-witnessed diameter of about \(100 \mu m\) can be replicated with only 5,000 cells (versus roughly 10,000 without suppression, see previous section). Intuitively, this is the effect of protrusion-based growth: organoids with high neighbour suppression limit their division to surface cells and so grow outwards, as opposed to clustering together at the centre. In other words, neighbour suppression results in more invasive organoid behaviour. This could partly explain why carcinomas induced in mice with this mutation are so diffusive \cite{Politi2006}: neighbour suppression would force cells to grow away from each other, and a mutant cell at the boundary of an expanding clone would both be able to prevent the division of its wild-type neighbours and be sharply advantaged compared to a mutant cell at the clone's centre.
    We also note that when neighbour suppression is increased (i.e. \(c=10\)), clusters grow much more slowly. The reason for this is intuitive: most of the cells are being stopped from dividing, so the rate of division drops significantly. However, as can be seen from Figure 1, there seems not to be a difference in size between budding and spherical clusters at the end of the experiment. For this to be true, budding clusters must have a much higher \(\beta\) (baseline division rate for isolated cells) than spherical clusters. For example, the two clusters in the top row of Figure 7 grow in 13.4 and 13.6 days respectively with \(\beta = 10\). In the right hand side of Figure 10 we see a cluster with a very small amount of neighbour suppression (\(h=1, c=25\), so completely surrounded cells are very mildly inhibited), which grows to a similar diameter in 11.3 days with \(\beta = 1\). 
    \subsubsection{Accumulation of suppression also produces deformations}
    The picture that emerges is now clearer: cells in budding organoids are much fitter than cells in spheroidal organoids (represented by low \(\beta\) in our model), but surrounded cells are prevented from reproducing by contact with neighbouring cells (represented by a negligibly low baseline \(\alpha\)). Possible mechanisms for this suppression are discussed in the Conclusions. One important note is that the mechanism of suppression does not necessarily have to be exclusively current; cells which remember being surrounded can also produce deformations. To illustrate this, we can consider an alternative model, similar to the differentiation model in section 4.2.3, in which all cells are born active and become inactive in response to being surrounded. In this model all active cells have the same per-timestep division probability \(p_{\text{diff}}(n, dT) = \alpha dT\). However, in a given timestep, an active cell with \(n\) empty neighbouring lattice-points has a probability \(\frac{\sigma dT}{1+e^{h(c-(26-n))}}\) of becoming inactive, for some `inactivation rate' \(\sigma\). This probability is a saturating function of \(n\), which is zero when the cell is isolated and maximal when it is surrounded. A cell which is surrounded for a long time may initially be active, but is very likely to eventually be forced into inactivity through prolonged exposure to other cells. Even if it is later pushed to the surface, it will remain inactive. We may call this mechanism `suppression accumulation'. The results of simulations with similar `inhibition strength' parameters \(\alpha=10, c=10, h=1, \tau =1.0\) and various values of \(\sigma\) are shown in Figure 11. For high rates of inactivation, we observe non-spherical deformations, but there is a tradeoff; for very high rates, growth stops entirely. It would be feasible for a mixture of current and cumulative suppression to exist within the system.
\begin{figure}[h!]
  \includegraphics[width=.49\linewidth]{41-eps-converted-to.pdf}
  \includegraphics[width=.49\linewidth]{42-eps-converted-to.pdf}
  \includegraphics[width=.49\linewidth]{43-eps-converted-to.pdf}
  \includegraphics[width=.49\linewidth]{44-eps-converted-to.pdf}
  
  \caption{Four simulations of `accumulated-inactivity'-driven growth as described above. Here \(\alpha=10,  h=\tau=1, c=10\) and \(\sigma\) is varied. The first two simulations run to 10,000 cells; the third is run to 5,000; the figure on the bottom right stops growing before it reaches 5,000 cells, as all become inactive.}
  \end{figure}
  


\section{Discussion and conclusion}

In this study we have used a novel agent-based modelling approach to simulate the growth of mutant and non-mutant organoids, and have found that the secondary spheroids formed by mutant clusters can be explained if mutant cells have both a higher reproductive fitness than non-mutant cells and an ability to suppress the division of their neighbours. This study builds upon the body of theoretical and computational work establishing the existence of instabilities in organoid growth \cite{Ciarletta2013, Miura2008, Giverso2016}, and our agent-based modelling approach allows us to straightforwardly calculate the detailed morphology of the resulting developed protrusions. A summary of our key tested hypotheses and their results can be found in Figure 12 below. 

\begin{figure}[H]
  \includegraphics[width = .49\linewidth]{pabs-eps-converted-to.pdf}
  \includegraphics[width=.49\linewidth]{comp_0-eps-converted-to.pdf}
  \includegraphics[width = .49\linewidth]{panc-eps-converted-to.pdf}
    \includegraphics[width = .49\linewidth]{comp_12-eps-converted-to.pdf}
    \includegraphics[width = .49\linewidth]{pdiff-eps-converted-to.pdf}
      \includegraphics[width = .49\linewidth]{comp_15-eps-converted-to.pdf}
        \includegraphics[width = .49\linewidth]{pNS-eps-converted-to.pdf}
                \includegraphics[width = .49\linewidth]{comp_24-eps-converted-to.pdf}
        
    
  \caption{A summary of tested division probability functions and their results. On the left is a graph of the per-cell, per-timestep division probability function \(p(n)\), where  \(n\) is the number of empty lattice-point neighbours. On the right is a representative simulation for a given set of parameters. From top to bottom, we consider the `absorption-based growth' hypothesis (with division probability function \(p_{abs}\) and \(\alpha=1, c=20, h=1, \tau=1.0\)); the `anchorage-dependent growth' hypothesis (\(p_{anc})\); the `differentiation-based growth' hypothesis (\(p_{\text{diff}})\), which has slightly different simulation rules to incorporate the frequency of differentiation- see section 4.2; and `neighbour-suppression-based' growth (\(p_{\text{NS}}\) and \(\alpha=0, \beta=10, c=10, h=1, \tau=1.0\)) .}
\end{figure}


The mechanism of this suppression is currently unclear. Contact-inhibited proliferation (CIP), the phenomenon whereby cells stop dividing in areas of high local density \cite{Levine1965}, is known to be mediated by EGF concentration \cite{Kim2009} and so might in theory be affected by the presence of an EGFR mutation. Variations in CIP were found to produce morphologically distinct organoid structures in the simulations of Karolak \textit{et. al.} \cite{Karolak2019}. However, CIP is generally lost during the development of cancer \cite{Hanahan2000}, and so it would be highly surprising if such a cancer-associated mutation as \textit{EGFR-L858R} were found to induce it. It seems more likely that mutant cells are secreting some antagonist which halts the division of their immediate neighbours, wild-type or mutant. A similar phenomenon has recently been observed amongst intestinal stem cells with mutated \textit{Apc} by Flanagan and colleagues \cite{Flanagan2021}. If this is happening in our system, then this should give \textit{EGFR-L858R} mutant AT2 cells a significant competitive advantage over non-mutant cells \textit{in vivo} and when grown in co-culture, and at least partially explain the invasiveness of L858R-mutant carcinomas. Further experimental work is needed to verify this. If a signalling pathway which leads to neighbour suppression could be identified, this might be pharmacologically targetable and have implications for the prevention and treatment of NSCLC amongst never-smokers. 

Through quantitative analysis of the growth of non-mutant organoids (see Appendix 8.2) one may also observe that evolutionary pressure acts even on healthy cells grown in nutrient-rich \textit{in vitro} conditions, selecting out initially-fitter monoclonal organoids and slowing the growth of initially-weak clusters. This may stem from a number of mechanisms, including differences in the tendency of progenitor cells to produce differentiated cells. More detailed longitudinal observations than were possible in this work are necessary to clarify these explanations. In particular, quantitative measurement of surface-area-to-volume ratios would allow closer fitting of the simulation parameters discussed in section 4.2; as it is, we can confirm that the diameters of the simulated clusters, and their qualitatively observed structures, match experimental observation. Of particular value would be movies formed from time series data, over timescales of hours or days, that show the movement and adjustment of the organoids as they grow, which would allow us to directly verify many of our experimental assumptions and incorporate dynamic fluctuations of cell positions into our model. We hope to address this in further work.

Our work emphasises the importance of considering spatial structure and cell-cell interaction in tumorigenesis, and to consider the circumstances of cell growth when determining whether or not a particular mutation will lead to cancer. Only by considering mutant cells within their full context, ecological and otherwise, will we be able to untangle the mechanisms that lead to cancer.
  
\section{Materials and Methods}

\subsection{Animal Procedures}

Animals were housed in ventilated cages with access to food and water ad libitum. All animal procedures were approved by The Francis Crick Institute Biological Research Facility Strategic Oversight Committee, incorporating the Animal Welfare and Ethical Review Body, conforming with UK Home Office guidelines and regulations under the Animals (Scientific Procedures) Act 1986 including Amendment Regulations 2012. Both male and female animals aged 6-15 weeks were used.

\textit{EGFR}-L858R [Tg(tet-O-EGFR-L858R)56Hev] mice were obtained from the National Cancer Institute Mouse Repository. Rosa26tTA and Rosa26-LSL-tdTomato mice were obtained from Jackson laboratory and backcrossed as previously described \cite{Swanton2022}, \cite{Politi2006}. After weaning, the mice were genotyped (Transnetyx, Memphis, USA), and placed in groups of one to five animals in individually ventilated cages with a 12-hour daylight cycle. 

\subsection{Fluorescence-activated cell sorting} 
For flow cytometry sorting of alveolar type II cells, minced lung tissue was digested with Liberase TM and TH (Roche Diagnostics) and DNase I (Merck Sigma-Aldrich) in HBSS (Gibco) for 30 minutes at 37 $^{\circ}$ C in a shaker at 180 rpm. Samples were passed through a 100 \(\mu\) m filter, centrifuged (300 x g, 5 min, 4 degrees) and bed blood cells were lysed for 5 min on ice using ACK buffer (Life Technologies). Cells were blocked with anti-CD16/32 antibody (BD) for 10 min. Extracellular antibody staining was then performed for 30 min on ice (see table), followed by incubation in DAPI (Sigma Aldrich) to label dead cells. Cell sorting was performed on Influx, Aria Fusion or Aria III machines (BD). AT2 cells were defined as DAPI-CD45-CD31-Ter119-EpCAM+MHC Class II+ CD49f- as previously described \cite{Major2020}.

\begin{table}[]
\centering
\begin{tabular}{|l|l|l|l|l|}
\hline
\textbf{antigen} & \textbf{fluorochrome} & \textbf{Vendor} & \textbf{Cat \#} & \textbf{dilution} \\ \hline
CD45             & BV421                 & Biolegend       & 103134          & 1/150             \\ \hline
CD31             & BV421                 & Biolegend       & 102423          & 1/150             \\ \hline
Ter119           & BV421                 & Biolegend       & 1162          & 1/150             \\ \hline
EpCAM            & APC-Fire750           & Biolegend       & 118230          & 1/150             \\ \hline
MHC Class II     & FITC                  & Biolegend       & 107606          & 1/150             \\ \hline
CD49f            & PE-Cy7                & eBiosciences    & 25-0495-82      & 1/150             \\ \hline
\end{tabular}
\end{table}
\subsection{Organoid forming assay}

AT2 cells were isolated from control T or ET mice, without \textit{in vivo} Cre induction, incubated in vitro with 6 x 107 PFU/ml of Ad5-CMV-Cre in 100 \(\mu\)L per 100,000 cells 3D organoid media (DMEM/F12 with 10 percent FBS, 100 U ml-1 penicillin-streptomycin, insulin/transferrin/selenium, L-glutamine (all GIBCO) and 1mM HEPES (in-house)) for 1hr at 37$^{\circ}$ C as detailed in Dost \textit{et. al.} \cite{Dost2020}. Cells were washed three times in PBS, before 10,000 cells were mixed with a murine lung fibroblast cell line (MLg2908, ATCC, 1:5 ratio) and resuspended in growth factor reduced Matrigel (Corning) at a ratio of 1:1. 100 \(\mu\) l of this mixture was pipetted into a 24-well transwell insert with a 0.4 \(\mu\)m pore (Corning). After incubating for 30 min at 37 $^{\circ}$ C, 500 \(\mu\) l of organoid media was added to the lower chamber and media changed every other day, following previous methods \cite{Choi2020}. Bright-field and fluorescent images were acquired after 14 days using an EVOS microscope (Thermo Fisher Scientific) and quantified using FiJi (.2.0.0-rc-69/1.52r, ImageJ). For wholemount staining of organoids, organoids were prepared according to previous methods \cite{Dekkers2019} and stained with anti-proSPC (Abcam, clone EPR19839) and anti-keratin 8 (DSHB Iowa, clone TROMA-1). 3D confocal images were acquired using an Olympus FV3000 and analysed in FiJI.



\section{Acknowledgements}


H. C. is supported by a grant from the Engineering and Physical Sciences Research Council [EP/W523835/1]. P. P. is supported by a UKRI Future Leaders Fellowship [MR/V022385/1]. M. P. D. is supported by the UK Engineering and Physical Sciences Research Council [EP/W032317/1]. C.S. acknowledges grant support from AstraZeneca, Boehringer-Ingelheim, Bristol Myers Squibb, Pfizer, Roche-Ventana, Invitae (previously Archer Dx Inc - collaboration in minimal residual disease sequencing technologies), and Ono Pharmaceutical. He is an AstraZeneca Advisory Board member and Chief Investigator for the AZ MeRmaiD 1 and 2 clinical trials and is also Co-Chief Investigator of the NHS Galleri trial funded by GRAIL and a paid member of GRAILs Scientific Advisory Board. He receives consultant fees from Achilles Therapeutics (also SAB member), Bicycle Therapeutics (also a SAB member), Genentech, Medicxi, Roche Innovation Centre  Shanghai, Metabomed (until July 2022), and the Sarah Cannon Research Institute C.S has received honoraria from Amgen, AstraZeneca, Pfizer, Novartis, GlaxoSmithKline, MSD, Bristol Myers Squibb, Illumina, and Roche-Ventana. C.S. had stock options in Apogen Biotechnologies and GRAIL until June 2021, and currently has stock options in Epic Bioscience, Bicycle Therapeutics, and has stock options and is co-founder of Achilles Therapeutics. 
Kate Gowers, Matthew A. Clarke, Yuxin Sun, Daniel Jacobson, Francesco Moscato, Pedro Victori Rosa, Jasmin Fisher, Sasha Bailey and Erik Sahai are thanked for their advice. 


\bibliography{main.bib}
\bibliographystyle{vancouver}


\section{Supplementary Information}

\subsection{S1: Data availability statement}
All data and code used in this work is available at https://github.com/hcoggan/L858R. The EGFR-condition images have been previously published in \cite{Swanton2022}; all other data is original.

 \subsection{S2: Appendix 1: Analysis of quantitative experimental data reveals `fitness bottleneck'}

As discussed above, various mechanistic models can produce spherical organoid growth, including models in which a cell's division rate is independent of how many neighbours it has. In addition to the simulation-based work discussed above, we performed a quantitative analysis of experimentally obtained non-mutant organoid sizes to attempt to distinguish between these models. 

Top-down images of the fluorescent organoids were taken at day 7 and day 14 of the experiment. This permits analysis of the cross-sections of the fluorescent clusters using a custom ImageJ script \cite{Schindelin2012}. (The script, and all relevant images, are accessible on Github; for details see the Data Availability statement.) For non-mutant clusters, which are almost perfectly spherical, we calculate their volumes, on the assumption that the cross-sectional area \(a_i\) of a cluster is related to its volume \(v_i\) by \(v_i = \frac{4}{3}\frac{a_i^{\frac{3}{2}}}{\sqrt{\pi}}\).
Since we have assumed that all cells have the same volume, \(v_0 = 80 \mu m^3\), we can also calculate the number of cells \(n_i\) in the \(i\)th cluster from its volume as \(v_i = n_i v_0\). We assume that each cluster grows monoclonally (from a single cell), so its volume at the start of the experiment is \(v_0\). The manner in which the images were taken makes it impossible to match clusters between timepoints, so we must assume that we know the volume of each cluster at either day 7 or day 14, but not both. We do however have samples of the \textit{distribution} of cluster sizes at three timepoints-- \(T=0, 7, 14\) days-- and so can test whether these distributions are consistent with well-known models of tumour growth. We consider three, each with a clear mechanistic justification: exponential growth, which implies each cell has a constant probability of division; logistic growth, which implies finite resources are divided equally amongst every cell in the cluster; and the classic von Bertalanffy model, which describes spherical growth in which only surface cells can reproduce. The fitting procedure is described in  Appendix 2. None of these models give consistent or physical parameters across all three experiments.

We can instead consider a model where a cell's division rate is explicitly dependent on the passing of time:

\[\frac{d v_i}{dt} = v_i f_i (t)\]
for some cluster-specific function \(f_i (t) \), and given that the volume of the cluster is assumed to be linearly dependent on the number of cells it contains. This model implies that

\[\ln \frac{v_i (T)}{v_0} = \int^{T}_{0} f_i(t) dt \]i.e. that the logarithms of these volumes correspond in some sense to the \textit{total fitness of the cluster over time}. We can therefore calculate the time-averaged fitness of the \(i\)th cluster, \(\frac{1}{T}\int^{T}_{0} f_i(t) dt = \frac{1}{T}\ln \frac{v_i (T)}{v_0}\), for both day-7 and day-14 data. The ensemble averages of the time-averaged fitnesses are displayed in Table 2, and their distributions are illustrated in Figure 13. These fitness distributions have a rough `bell-curve' shape, clustered around a central mean, as one might expect for genetically identical clusters growing in experimentally identical conditions. They are extremely consistent between experiments (see Table 2).

\begin{table}[H]
\begin{center}
\begin{tabular}{ | l | l | l | l | } 
  \hline
    \(<\frac{1}{T}\int^{T}_{0} f_i(t) dt> \) /day & E1 & E2 & E3 \\ 
  \hline
\(T=7\)  & 0.76 & 0.84 & 0.83 \\ 
  \hline
  \(T=14\) & 0.42 & 0.44 & 0.45 \\ 
  \hline
\end{tabular}
\end{center}
\caption{Ensemble averages of time-averaged cluster fitness, measured after 7 and 14 days for each of three identical experiments on non-mutant clusters (labelled E1, E2, E3). }
\end{table}


\begin{figure}[H]
  \includegraphics[width = .49\linewidth]{day7-eps-converted-to.pdf}
  \includegraphics[width=.49\linewidth]{day14-eps-converted-to.pdf}
  \caption{A histogram of the time-averaged fitnesses, \(\frac{1}{T}\int^{T}_{0} f_i(t) dt \), for \(T=7\) days (top) and \(T=14\) days (bottom), for non-mutant clusters.}
\end{figure}
We can make two observations from these figures. The first is that cluster fitness decreases significantly over time, with the vast majority of growth happening in the first week. There are a number of possible explanations for this. Neighbour suppression, even when mild enough to result in spherical clusters, naturally produces a slowdown in growth over time. As the organoid grows, the proportion of surrounded cells increases, and so even if those cells' division rate is only mildly inhibited, this will decrease the average fitness of the cluster. This cannot be the only effect at play, however, or the clusters would naturally fit a classic von Bertalanffy model, which describes surface-dominated growth with a constant birth or death rate ascribed to every cell; we see in the Appendix that this does not work. 
  Cell differentiation, as described in section 3.1, would also produce a growth slowdown. We can see this mathematically: if we denote the number of active cells in a cluster by \(x_i\) and the number of inactive cells by \(y_i\), and say that active cells divide at rate \(\alpha\) and produce active cells during that division with probability \(q\), then
\[\dot{x_i} = \alpha q x_i\]\[\dot{y_i} = \alpha (1-q) x_i\]
and so, given \(x_i(0) = 1, y_i(0)=0\)-- i.e. we begin each simulation with a single active cell- we have \(x_i = \alpha e^{\alpha q t}\), \(y_i = \frac{1-q}{q} (e^{\alpha q t}-1)\). The average fitness of the cluster is proportional to the fraction of cells within the cluster which are active; this is \(\frac{x_i}{x_i + y_i} = \frac{\alpha q e^{\alpha q t}}{(1-q)(e^{\alpha q t}-1) + qe^{\alpha q t}}\), which decreases with time to reach the constant value \(q\).
Another hypothesis, motivated by Weeden \textit{et. al.}, \cite{Weeden2017}, is that adult lung epithelial cells \textit{in vitro} accumulate damaging mutations as they age. These affect their reproductive fitness, and decrease the overall division rate of the cluster over time. Further study is needed to distinguish between these effects; detailed monitoring of the shape and size of the organoids at daily intervals, which was not possible during these experiments, would help to clarify the mechanisms at work here. 

Another interesting aspect of the distributions above is that the leftward skew of the fitnesses-- the bias towards less fit clusters-- increases over time. We can describe this as an `evolutionary bottleneck'-- a distribution which is initially clustered around a central mean stretches out over time, with a higher proportion of the clusters trapped on the `less fit' side of the graph whilst a few very-fit clusters proceed unhindered. The simplest interpretation of this is that `weak clusters get weaker faster': clusters which are less fit in the first week decline rapidly in fitness and so grow barely at all in the second, whilst initially-fit organoids weaken more slowly. This cannot be explained by neighbour suppression alone, which would produce the opposite effect: a cluster that initially grows should rapidly become mostly-comprised of surrounded cells and so slow down more quickly than a cluster that starts off slowly. Some other process must be occurring here.

If cell differentiation is at work, a higher initial probability of differentiation (i.e. a low \(q\)) will lead to slower initial growth and a more rapid decline in fitness, leading as observed to an increasing skew. It may also be that cells which are initially less inclined to divide are also more susceptible to accumulating deleterious mutations-- that these two quantities are tied to a deeper underlying determinant of `fitness'. Regardless of the underlying mechanism, it is noteworthy that we can see natural selection at play here even in nutrient-rich \textit{in vitro} conditions and in non-mutant, `healthy' cells: random fluctuations in fitness between supposedly identical cells have increasing effects over time, leading to a small number of very large clusters and gradually stopping the growth of less-fit cells. 

When this analysis is carried out on mutant clusters, the figures obtained are extremely similar; the only effect we can pick out is a slight decrease in average fitnesses (see Table 3 below). The distributions (see Figure 14) look similar to the non-mutant graphs above, but show significantly more internal variation and are generally less `clean'. The development of secondary spheroids means that the fundamental assumption behind the calculation-- that they are essentially spherical, and so their volumes can be deduced from their cross-sectional areas-- has certainly broken down during the second week and may well be tenuous by the end of the first. We note only that the similarity of the day-7 graphs between mutants and non-mutants suggests that, at least initially, mutant growth strategies (i.e. the suppression of surrounded cells and a high division rate amongst surface cells) do not result in \textit{larger} clusters. The fitness effects of the mutation are, as we have seen, much more complex than simply giving an overall fitness advantage to mutant clusters.)
\begin{table}
\begin{center}
\begin{tabular}{ | l | l | l | l | } 
  \hline
 \(<\frac{1}{T}\int^{T}_{0} f_i(t) dt> \) /day & E4 & E5 & E6 \\ 
  \hline
\(T=7\)  & 0.76 & 0.72 & 0.75 \\ 
  \hline
  \(T=14\) & 0.40 & 0.39 & 0.39 \\ 
  \hline
\end{tabular}
\end{center}
\caption{Ensemble averages of time-averaged cluster fitness, measured after 7 and 14 days for each of three identical experiments on mutant clusters (labelled E4, E5, E6).}
\end{table}

\begin{figure}[H]
  \includegraphics[width = .49\linewidth]{day7m-eps-converted-to.pdf}
  \includegraphics[width=.49\linewidth]{day14m-eps-converted-to.pdf}
  \caption{A histogram of the time-averaged fitnesses, \(\frac{1}{T}\int^{T}_{0} f_i(t) dt \), for \(T=7\) days (top) and \(T=14\) days (bottom), for mutant clusters.}
\end{figure}

\subsection{S3: Appendix 2: Classical models of tumour growth fail to describe organoid distribution}


We describe here the procedure used to attempt to fit quantitative models to non-mutant organoid growth. We seek a growth law that will explain the distribution of cluster sizes at two timepoints, \(T=7\) days and \(T=14\) days. We have these distributions for each of three experiments on genetically identical mice, but we cannot match clusters between timepoints.


We first consider single-parameter models of growth. If we hypothesise that every cell within a cluster is equally likely to divide at all points in time, then the \(i\)th cluster will have a volume \(v_i\) given by

\[\frac{d v_i}{d t} = \alpha_{i} v_ i\]
where \(\alpha _i\) is its cluster-specific division rate, and \(v_i (0) = v_0\). We can calculate the average cluster-specific division rate as \(\alpha_i = \frac{\log\frac{v_i(T)}{v_0}}{T}\) from cluster sizes at \(T=7\) days and \(T=14\) days, for all three experiments (see Table 4).

\begin{table}[H]
\begin{center}
\begin{tabular}{ | l | l | l | l | } 
  \hline
 average \(\alpha_i\)   /day & E1 & E2 & E3 \\ 
  \hline
\(T=7\)  & 0.76 & 0.84 & 0.83 \\ 
  \hline
  \(T=14\) & 0.42 & 0.44 & 0.45 \\ 
  \hline
\end{tabular}
\end{center}
\caption{Ensemble averages of hypothetical cluster-specific division rates, taken from data after 7 and 14 days of organoid growth, for each of three identical experiments on non-mutant clusters (labelled E1, E2, E3).}
\end{table}

This is very consistent between experiments but not between timepoints. Clearly there is some mechanism causing a slowdown in growth, which this model cannot capture. 
Similarly, we can attempt to fit a model of surface-based growth, where the rate of growth is proportional to surface area, i.e.

\[\frac{d v_i}{d t} = \alpha_i v_ i^{\frac{2}{3}}\]But this model fares even worse, losing even the inter-experimental consistency between growth rates (see Table 5).

\begin{table}[H]
\begin{center}
\begin{tabular}{ | l | l | l | l | } 
  \hline
  \(\alpha_i\) (surface-based) & E1 & E2 & E3 \\ 
  \hline
  T = Day 7 & 3.6 & 4.7 & 5.5 \\ 
  \hline
  T = Day 14 & 2.5 & 3.0 & 3.3 \\ 
  \hline
\end{tabular}
\end{center}
\caption{Ensemble averages of hypothetical cluster-specific \textit{absorption-dependent} division rates, taken from data after 7 and 14 days of organoid growth, for each of three identical experiments on non-mutant clusters (labelled E1, E2, E3).}
\end{table}

A two-parameter model is necessary to incorporate slowing growth. A logistic growth model can capture both cell division and inter-cellular interaction/competition:
\[\frac{d v_i}{d t} = \alpha_{i}(v_ i - \frac{1}{K_i} v_i ^2)\]
However, the introduction of a second parameter leaves us with a determination problem. Since we only have two datapoints for each cluster (one by observation at either day 7 or day 14, and one by assumption at the start of the experiment) we cannot fit two cluster-specific parameters. We may assert that one parameter is global (i.e. \(K_i = K\), the more natural choice), fit the \(\alpha_i\)s accordingly to specific clusters, and choose the value of \(K\) which gives the best agreement between cluster-specific parameters using a grid-search method. This leads to nonsensical values (i.e. making \(K\)very small forces all the \(\alpha_i\) values to zero, which is technically a case of perfect agreement but makes no biological sense here). There are no other values of \(K\)which produce agreement. Crucially, all positive values of \(K\) produce negative \(<\alpha_i>\). This is because the clusters are so large that the model fits a positive quadratic term and a negative linear term-- i.e. positive intercellular interaction and a constant death rate. This is a substantial change in our hypothesis, and not one we can support on biological grounds. 


\begin{figure}[H]
  \includegraphics[width=.49\linewidth]{carrying-eps-converted-to.pdf}
  \caption{An attempt to fit the logistic model of growth to Experiment 1, carrying capacity first. Figures for Experiments 2 and 3 produce nearly identical results.}
\end{figure}

This result is not all that surprising ecologically. We have assumed in general that nutrients and space are always more than sufficient for growth, so there should not be competition within clusters for resources. In any case, we have no reason to think that cells within a cluster are well-mixed (i.e. moving around freely and no likelier to interact with one cell than another). Once cells are born, we might naturally expect them to stay roughly in place, be pushed aside when other cells are born, and generally interact only with their neighbours. 

A more intuitive model is the classic von Bertalanffy model, which suggests that growth is driven by surface-based absorption and a constant death rate:

\[\frac{d v_i}{dt} = \alpha_i v_i ^{\frac{2}{3}} - \beta_i v_i\]
If we assume that the death rate of each cluster is constant, \(\beta_i = \beta\), then we can solve this equation and rearrange to find the death rate of each cluster in terms of its observed volume at time \(T\):

\[\alpha_i = \beta\frac{v_i(T)^{\frac{1}{3}} -v_0^{\frac{1}{3}} e^{-\frac{\beta T}{3}} }{1- e^{-\frac{\beta T}{3}}}\]Once again we can observe the average \(<\alpha_i>\) at each value of \(\beta\) and choose the \(\beta\) at which the day-7 and day-14 averages match. Here, for each experiment, we obtain valid parameter values (see Fig 15).

\begin{figure}[H]
  \includegraphics[width=0.5 \linewidth]{abs-eps-converted-to.pdf}
  \caption{An attempt to fit the von Bertalanffy model of growth to Experiment 1. Figures for Experiments 2 and 3 produce similar results.}
\end{figure}

These `crossing point' values are outlined in Table 6, and are reasonably consistent between experiments.

\begin{table}[H]

\begin{center}
\begin{tabular}{ | l | l | l | l | } 
  \hline
  Matching parameters & E1 & E2 & E3 \\ 
  \hline
  <\( \alpha_i \)> / day & 20 & 27 & 35 \\ 
  \hline
  \(\beta\) & 0.45 & 0.53 & 0.67 \\ 
  \hline
\end{tabular}
\end{center}
\caption{Fit parameters to classic von Bertalanffy model, for each of three identical experiments on non-mutant clusters (labelled E1, E2, E3).}
\end{table}

Our problem comes when we look beyond averages. What is the distribution of death rates around each \(<\alpha_i>\) at the `correct' \(\beta\)? Observing the histogram, we find three issues. One is that the distribution is monotonically decreasing, not Gaussian as one might expect, given that all clusters are genetically identical and grown in identical circumstances. It is difficult to biologically justify such a distribution of death rates. Another issue is that the distribution is noticeably different between day-7 and day-14 data; a rightward shift happens in the second week, suggesting the death rates increase: i.e. that we have not actually captured the mechanism by which growth slows in the second week of development, but merely reflected it. We turn instead to the time-dependent model discussed in Appendix 1.

\begin{figure}[H]
  \includegraphics[width=.49\linewidth]{absdist-eps-converted-to.pdf}
  \caption{The distribution of death rates produced at the correct \(\beta\) for Experiment 3, from both day-7 (blue) and day-14 (orange) data. Figures for Experiments 1 and 2 produce similar results, though there is significant variation in the size of the largest cluster.}
\end{figure}




 



\end{document}

