\documentclass[fleqn,10pt]{wlscirep}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{algorithm2e}
\usepackage{tikz}
\title{Biologically-primed deep neural network improves colorectal Cancer Molecular subtypes prediction from H\&E stained images}

\author[1,*]{Hadar Hezi}
\author[2]{Daniel Shats}
\author[3,4]{Daniel Gurevich}
\author[3,4]{Yosef E. Maruvka}
\author[1]{Moti Freiman}
\affil[1]{Faculty of Biomedical Engineering, Technion IIT, Haifa, Israel}
\affil[2]{Faculty of Computer science, Technion IIT, Haifa, Israel}
\affil[3]{Faculty of Biotechnology and Food Engineering, Technion IIT, Haifa, Israel}
\affil[4]{Lokey Center for life science and engineering,  Technion IIT, Haifa, Israel}
\affil[*]{hadar.hezi@campus.technion.ac.il}
%\affil[+]{these authors contributed equally to this work}


\keywords{Colorectal cancer, Digital pathology, Convolutional neural networks}

\begin{abstract}
Colorectal cancer (CRC) molecular subtypes play a crucial role in determining treatment options. Immunotherapy is effective for the microsatellite instability (MSI) subtype of CRC, but not for the microsatellite stability (MSS) subtype. Recently, convolutional neural networks (CNNs) have been proposed for automated determination of CRC subtypes from H\&E stained histopathological images. However, previous CNN architectures only consider binary outcomes of MSI or MSS, and do not account for additional biological cues that may affect the histopathological imaging phenotype. 
In this study, we propose a biologically-primed CNN (BP-CNN) architecture for CRC subtype classification from H\&E stained images. Our BP-CNN accounts for additional biological cues by casting the binary classification outcome into a biologically-informed multi-class outcome.
We evaluated the BP-CNN approach using a 5-fold cross-validation experimental setup for model development on the TCGA-CRC-DX cohort, comparing it to a baseline binary classification CNN. Our BP-CNN achieved superior performance when using either single-nucleotide-polymorphism (SNP) molecular features (AUC: 0.824$\pm$0.02 vs. 0.761$\pm$0.04, paired t-test, p$<$0.05) or CpG-Island methylation phenotype (CIMP) molecular features (AUC: 0.834$\pm$0.01 vs. 0.787$\pm$0.03, paired t-test, p$<$0.05). A combination of CIMP and SNP models further improved classification accuracy (AUC: 0.847$\pm$0.01 vs. 0.787$\pm$0.03, paired t-test, p$=$0.01).
Our BP-CNN approach has the potential to provide insight into the biological cues that influence cancer histopathological imaging phenotypes and to improve the accuracy of deep-learning-based methods for determining cancer subtypes from histopathological imaging data.

\end{abstract}
\begin{document}

\flushbottom
\maketitle

\thispagestyle{empty}



\section*{Introduction}
The remarkable success of immunotherapy in treating solid tumors such as melanoma and lung cancer has led to an increased interest in using this approach for other types of cancer.\cite{Ganesh2019-mw,Barzaman2021-cb} However, high inter- and intra-tumor heterogeneity present a challenge for the success of immunotherapy in colorectal cancer (CRC).\cite{molinari2018heterogeneity,Li2020-pt} 
One promising outcome is the success of immune-checkpoint inhibitors (ICIs) in the subgroup of CRC patients with the microsatellite instability (MSI) phenotype.\cite{le2017mismatch}  This highlights the importance of molecular subtyping in CRC to select the appropriate immunotherapy protocol in order to achieve the best treatment response.\cite{Hu2021PersonalizedStand}


The gold standard for CRC subtyping is DNA sequencing using a Polymerase Chain Reaction (PCR) test.\cite{Baudrin2018-tq}  However, this test is expensive, time-consuming, and has limited availability to patients.
%Immunohistochemistry (IHC) staining and analysis has been proposed as an alternative for CRC subtyping,\cite{banias2020immunohistochemical} but it requires specialized staining and can lead to inconclusive results. Additionally, not all pathogenic mismatch repair (MMR) mutations result in the loss of the protein presented by IHC, and interpretation of IHC images requires an experienced pathologist.
Recently, convolutional neural networks (CNNs)\cite{he2016deep,simonyan2014very} have been proposed for automatically determining CRC subtypes from common Hematoxylin and Eosin (H\&E) stained histopathological images.\cite{Kather2019DeepCancer,kuntz2021gastrointestinal,wagner2023fully} This approach, which utilizes images that are already generated as part of the clinical routine, has the potential to provide a cheap and accurate tool for determining CRC subtypes within the current clinical setting. However, due to the large size of the H\&E images, they cannot be directly processed by conventional graphics processing units (GPUs) for model training. 

Previous studies have attempted to overcome this limitation by dividing the images into small patches, performing patch-level classification, and aggregating the results. Specifically, Kather et al.\cite{Kather2019DeepCancer} was the first to infer CRC molecular sub-types MSI and microsatellite stability (MSS) from H\&E images using CNNs by dividing the images into small patches, performing  patch-level classification, and aggregating the classification results. Their approach achieved moderate success on the TCGA-CRC-DX\cite{crc-data} database (Area under the curve (AUC) per patient of 0.77, n=360,18\% MSI). 
Echle et al.\cite{Echle2020}, used a similar method but further improved the overall classification accuracy by increasing the dataset size through the combination of several databases. More recently, Bilal et al. \cite{Bilal2021DevelopmentStudy} used an iterative `draw and rank' method to eliminate less-informative patches from the final decision by performing a patch selection as a pre-processing step. For a comprehensive analysis of recent techniques utilizing deep learning for the classification of CRC subtypes from standard H\&E stained histopathological images, we refer to the study by Kuntz et al. \cite{kuntz2021gastrointestinal}. 
However, these studies have utilized CNNs as a black-box classification system, neglecting to take into account the unique ways in which molecular features can shape the H\&E imaging phenotype. 

In this study, we aim to investigate the interplay between molecular and morphological levels using CNNs. Our main hypothesis is that molecular features beyond the conventional subtypes of MSI and MSS may impact the H\&E image phenotype, thus, the classification task should account for these molecular features. To address this, we introduce a novel biologically-primed CNN (BP-CNN) architecture and training approach that considers molecular variations within a tumor's subtype.
Unlike previous methods, our BP-CNN approach goes beyond the traditional black-box classification system by taking into account specific ways in which molecular features influence the H\&E imaging phenotype. This not only has the potential to improve the accuracy of deep-learning-based methods for determining cancer subtypes, but also provides a deeper understanding of the underlying biology.
It is crucial to underscore that despite leveraging molecular subtypes information during the training phase, our inference procedure solely relies on the H\&E images without any supplementary data. Leveraging additional information during the training phase only was previously used by Su et al.\cite{Su2022-xh} who trained a CNN using H\&E and Immunohistochemistry (IHC) stained images to classify tumor vs. normal patches and tested it on H\&E-stained images solely.

In our experiments, we will utilize the TCGA-CRC-DX cohort\cite{crc-data} (n=360), previously pre-processed by Kather et al.\cite{Kather2019DeepCancer}. Through our analysis, we have identified a high single-nucleotide-polymorphism (SNP) rate and a high CpG-Island methylation phenotype (CIMP) as the dominant molecular features of the MSI class \cite{Liu2018ComparativeAdenocarcinomas}. Not only are they dominant but they also present a range of values among the MSI class. We hypothesize that the diversity within these features in the MSI class limits the learning capacity of the CNN. Figures~\ref{fig:patches_view} and ~\ref{fig:patches_cimp} present examples of MSI H\&E patches, along with their baseline binary classification predictions, SNP levels, and methylation types. Although all patches are MSI patches there is an evident visual difference between them. 


We tested our hypothesis by introducing the BP-CNN architecture for CRC subtype classification. Our approach transforms the binary classification outcome proposed by Kather et al. \cite{Kather2019DeepCancer} into a three-class classification. Specifically, we divided the MSI group into two subgroups based on the values of their molecular features and modified the CNN architecture to predict these three classes instead of the binary classification of MSI/MSS. We then aggregated the classification of the MSI subtypes to produce the final binary classification. 
 Additionally, to account for the potential lack of correlation between classification based on SNP and CIMP, we developed a combined model that merges these two molecular features into a single classification model.
To evaluate the performance of our BP-CNN approach, we conducted a 5-fold cross-validation experiment on the TCGA-CRC-DX cohort \cite{crc-data} and compared it to a baseline binary classification CNN. \cite{Kather2019DeepCancer}


%snp patches
\begin{figure}[htbp]
  \centering
   \begin{tabular}[t]{cc}
\raisebox{1cm}{true MSI} &
   \includegraphics[width=0.5\textwidth]{images/true_msi_snp.png}\\
\raisebox{1cm}{false MSS} &
    \includegraphics[width=0.5\textwidth]{images/false_mss_snp.png}\\ 
\end{tabular}
\caption{Visualization of patches with their SNP rate(number of SNPs for the entire DNA sample) from the baseline model's classified results. True MSI are patches that were correctly classified to MSI; False MSS are MSI patches that were mistakenly classified to MSS.}% caption for whole figure
\label{fig:patches_view}% label for whole figure
\end{figure}
%cimp patches
\begin{figure}[htbp]
  \centering
   \begin{tabular}[t]{cc}
\raisebox{1cm}{true MSI} &
   \includegraphics[width=0.6\textwidth]{images/true_msi_cimp.png}\\
\raisebox{1cm}{false MSS} &
    \includegraphics[width=0.6\textwidth]{images/false_mss_cimp.png}\\ 
\raisebox{1cm}{true MSS} &
    \includegraphics[width=0.6\textwidth]{images/true_mss_cimp.png} \\
\raisebox{1cm}{false MSI} &
    \includegraphics[width=0.6\textwidth]{images/false_msi_cimp.png}
\end{tabular}
\caption{Visualization of patches with their methylation type from the baseline model's classified results. True MSI are patches that were correctly classified to MSI, False MSS are MSI patches that were mistakenly classified to MSS and vice versa.}% caption for whole figure
\label{fig:patches_cimp}% label for whole figure
\end{figure}





\section*{Results}
\subsection*{Baseline model}
\begin{figure}
    \centering
    \begin{tabular}[t]{c}
        \includegraphics[width=0.7\textwidth]{images/roc_test_baseline.png} 
    \end{tabular}
    \caption{Average ROC of our baseline model}
    \label{fig:roc_test_baseline}
\end{figure}
We began by evaluating the performance of a baseline model, which replicated the results of Kather et al.\cite{Kather2019DeepCancer} with slight modifications.
Our baseline model achieved an average AUC of 0.8 (95\% CI, 0.75–0.83) on the 100 patients test set over the 5 folds training sessions as shown in Figure \ref{fig:roc_test_baseline}. This outcome is comparable to the results of Kather et al.\cite{Kather2019DeepCancer} (median bootstrapped AUC of 0.77,  95\% CI: 0.62–0.87). The difference could be attributed to several factors that varied between our study and Kather's study; such as the use of python implementation as opposed to Matlab\textsuperscript{\textregistered}, the use of patient-level aggregation using MSI probabilities instead of the predicted label, and the use of Inception v3 model\cite{szegedy2015going} instead of Resnet18 \cite{he2016deep}.

\subsection*{Molecular features analysis}
We examined the distribution of various molecular characteristics in relation to the initial CNN classification. Specifically, we selected two prevalent features, namely SNPs and methylation type. It is well-established that tumors with MSI often exhibit a high prevalence of SNPs as a result of a malfunctioning DNA mismatch-repair mechanism. Additionally, they are frequently classified as being highly methylated \cite{Liu2018ComparativeAdenocarcinomas}.
As a control experiment, we also examined the copy-number variation (CNV) feature, which is commonly observed in MSS tumors and infrequently in MSI tumors.

Figure~\ref{fig:feature_boxplot}a, depicts a boxplot of the SNP rates in relation to the initial classification of the test set, as determined by the baseline model.
The MSS class, whether it is a true or a false classification, has a low rate of SNPs as these patients do not present a deficiency in the DNA mismatch repair mechanism. In contrast, the MSI class exhibits a distinct variation in SNP distribution between the true-positive classifications, where MSI is the positive class, and the false-negative classifications, which were identified as MSS. The true-positive group has a slightly higher median value and a low variance around it, whereas the false-negative group displays a higher degree of variation.

The methylation types distribution in relation to the baseline classification is displayed in Figure~\ref{fig:feature_boxplot}b. CIMP-H is rare in MSS and common in MSI. CIMP-low has two subtypes: GEA CIMP-low (upper GI tract) and CRC CIMP-low (lower GI tract), with the latter mostly present in MSS. In MSI, CRC CIMP-low is more frequent in false negatives than true-positive classifications. Non-CIMP type is mostly found in MSS compared to MSI.

The distribution of the CNV feature in patches that were classified by the baseline model is summarized in Figure~\ref{fig:feature_boxplot}c. The MSS patches show higher rates of CNV with large variability among them, whereas the MSI patches display low rates of CNV with minimal variation in this feature.

\begin{figure}[htbp]
 \centering
     \begin{tabular}[t]{ccc}
  \includegraphics[width=0.3\linewidth]{images/boxplot_snp.png} &
  \includegraphics[width=0.3\linewidth]{images/cimp_diagram.png} &
   \includegraphics[width=0.3\textwidth]{images/boxplot_cnv.png} \\
   (a) & (b) & (c)
  
    \end{tabular}
    \caption{The distribution of molecular features within the test set, separated according to the baseline classification of the patches is presented as follows:
(a) A boxplot of the SNP rate at the patch level is shown, with the y-axis representing the number of SNPs for the entire DNA sample.
(b) A boxplot of the methylation type at the patch level.
(c) A boxplot of the CNV rate at the patch level, with the y-axis representing the fraction of the entire DNA sample that has a CNV.}
     \label{fig:feature_boxplot}
\end{figure}



\subsection*{Biologically-primed models}
We developed 3 biologically-primed models. The first, BP-CNN\textsubscript{SNP}, divided MSI patients into two subgroups based on their SNP rate using a threshold of 1200. The second, BP-CNN\textsubscript{CIMP}, separated the MSI group into CIMP-H vs. non-CIMP-H subgroups. The third, BP-CNN\textsubscript{CNV}, divided the MSI group into two subgroups based on each patient's CNV rate with a threshold of 0.005.


The BP-CNN\textsubscript{SNP} model achieved a significantly higher AUC compared to the baseline model on the 100 patients test set over 5 stratified folds training sessions (0.824$\pm$0.02 (95\% CI 0.79-0.86) vs. 0.761$\pm$0.04 (95\% CI 0.68-0.8), paired t-test, p$<$0.05). Figure~\ref{fig:roc_test_vs}a presents the ROC curves (average and the 95\% CI) for the test set over the different training sessions. Figure~\ref{fig:roc_boxplot} presents the distribution of the AUC on the test set over the different training sessions. The BP-CNN\textsubscript{SNP} model achieved a significantly higher F1 score compared to the baseline model (0.65$\pm$0.04 vs. 0.61$\pm$0.05, paired t-test, p$<$0.1). 

 \begin{figure}
    \centering
    \begin{tabular}[t]{ccc}
        \includegraphics[width=0.3\textwidth]{images/roc_test_vs_snp.png} &
        \includegraphics[width=0.3\textwidth]{images/roc_test_vs_cimp.png} &
        \includegraphics[width=0.3\textwidth]{images/roc_test_vs_cnv.png} \\
         (a) & (b) & (c)\\
       
    \end{tabular}
    \caption{Average and 95\% CI ROC curves for per-patient classification using: (a) the BP-CNN\textsubscript{SNP} and the baseline models,  (b) the BP-CNN\textsubscript{CIMP} and the baseline models, and; (c) the BP-CNN\textsubscript{CNV} and the baseline models.}
    \label{fig:roc_test_vs}
\end{figure}

  \begin{figure}
    \centering
    \begin{tabular}[t]{cc}
        \includegraphics[width=0.5\textwidth]{images/boxplot_roc_BP.png} &
        \includegraphics[width=0.5\textwidth]{images/boxplot_f1.png} \\
        (a) & (b)
    \end{tabular}
    \caption{Box-plot visualization of (a) AUC results, (b) F1-scores for per-patient classification for the biologically primed vs. the baseline models on the test set over the different training sessions. Note that since we used a stratified k-fold approach to divide the training data over the sessions, the baseline model performance can vary between experiments.} 
    \label{fig:roc_boxplot}
\end{figure}

 \begin{figure}
    \centering
    \begin{tabular}[t]{cc}
        \includegraphics[width=0.5\textwidth]{images/cm_SNP.png} &
        \includegraphics[width=0.5\textwidth]{images/cm_CIMP.png}\\
        (a)  & (b)\\ \\         
    \end{tabular}
    \caption{Patient level confusion matrices. Each matrix is an average on the test set over the different training sessions. BP-CNN SNP model (a) BP-CNN CIMP model (b). Threshold for MSI prediction is the best F1 score threshold.}
    \label{fig:confusion}
\end{figure}

Similarly, the BP-CNN\textsubscript{CIMP} model achieved a significantly higher AUC compared to the baseline model on the 100 patients test set over 5 stratified folds training sessions  (0.834$\pm$0.01 (95\% CI 0.81-0.85) vs. 0.787$\pm$0.03 (95\% CI 0.72-0.81), paired t-test, p$<$0.05). Figure~\ref{fig:roc_test_vs}b presents the ROC curves (average and the 95\% CI) for the test-set over the different training sessions. Figure~\ref{fig:roc_boxplot} presents the distribution of the AUC on the test set over the different training sessions.  The BP-CNN\textsubscript{CIMP} model achieved a significantly higher F1 score compared to the baseline model (0.71$\pm$0.03 vs. 0.63$\pm$0.06, paired t-test, p$<$0.06).

In contrast, the BP-CNN\textsubscript{CNV} model underperformed the baseline model on the 100 patients test set over 5 stratified folds training sessions (AUC: 0.793$\pm$0.03 (95\% CI 0.75-0.85) vs. 0.801$\pm$0.02 (95\% CI 0.75-0.83) . 
Figure~\ref{fig:roc_test_vs}c presents the ROC curves (average and the 95\% CI) on the test-set over the different training sessions and figure~\ref{fig:roc_boxplot} presents the distribution of the AUC on the test set over the training sessions. The differences were not statistically significant.
The F1 score for the BP-CNN\textsubscript{CNV} model was slightly better than this of the baseline model (0.64$\pm$0.03 vs. 0.62$\pm$0.04, paired t-test, p$<$0.6).



Figure~\ref{fig:confusion} depicts the confusion matrices of per-patient classification on the test set averaged over the training sessions for the BP-CNN\textsubscript{SNP} and the BP-CNN\textsubscript{CIMP} models.
The BP-CNN\textsubscript{SNP} model was better at classifying MSI patients while the BP-CNN\textsubscript{CIMP} was better at classifying MSS patients. 



\subsection*{Combined model}

We leveraged the uncorrelated performance presented by the BP-CNN\textsubscript{SNP} and BP-CNN\textsubscript{CIMP} models by developing a combined architecture that combines the results of the BP-CNN\textsubscript{SNP} and BP-CNN\textsubscript{CIMP} through a multi-layer perceptron model (BP-CNN\textsubscript{combined}).
The BP-CNN\textsubscript{combined} achieved a significantly higher AUC compared to the baseline model on the 100 patients test set over 5 stratified folds training sessions (0.847$\pm$0.01 (95\% CI 0.82-0.87 vs. 0.787$\pm$0.03 (95\% CI 0.72-0.81) paired t-test, p$<$0.01). The average F1 score on the test set, over the different training sessions for the BP-CNN\textsubscript{combined} was 0.71$\pm$0.02 compared to 0.63$\pm$0.06 for the baseline model . Figure \ref{fig:res_combined} presents the ROC curves and the boxplots of the AUC for the baseline and the BP-CNN\textsubscript{combined} models on the test set over the different training sessions.



\begin{figure}
    \centering
    \begin{tabular}[t]{cc}
        \includegraphics[width=0.5\textwidth]{images/roc_test_vs_comb.png} &
        \includegraphics[width=0.5\textwidth]{images/boxplot_roc_comb.png} \\
         (a) & (b)\\        
    \end{tabular}
    \caption{Average ROC of baseline (CIMP folds) and Combined model (a), 5-folds AUC results comparison (b)}
    \label{fig:res_combined}
\end{figure}



\section*{Discussion}
Distinguishing CRC subtypes through analysis of H\&E stained histopathological images is crucial for cost-effective wide-scale implementation of personalized treatment plans for patients.\cite{Hu2021PersonalizedStand} Recently, utilizing CNN-based techniques has been proposed as an automated method for classifying H\&E stained histopathological images of CRC.\cite{Kather2019DeepCancer}
The current study highlights the significance of taking into account additional biological factors that may impact the CNN-based histopathological imaging phenotype for CRC sub-typing. The incorporation of these biological cues not only enhances classification accuracy but also serves as a systematic method for investigating whether and potentially how molecular-level biological factors influence the histopathological imaging phenotype.

Our findings revealed a significant improvement in the AUC results for the classification of CRC into MSI and MSS subtypes using our BP-CNN approach in conjunction with CNNs. Specifically, our findings suggest that the SNP and CIMP molecular characteristics have an impact on the phenotype of H\&E stained histopathological images, whereas the CNV molecular feature does not. An intriguing discovery is that incorporating the SNP molecular feature into the CNN enhances the classification of MSI patients while incorporating the CIMP molecular feature improves the classification of MSS patients. This implies that incorporating both SNP and CIMP molecular features into a combined model would result in an increased overall accuracy in the classification of CRC subtypes. Direct integration of multiple molecular features in the BP-CNN approach proves challenging due to the overlap of patients across classes. Notably, a majority (75\%) of CIMP-H patients have high SNP rates (above the threshold of 1200). However, the remaining methylation types exhibit a wide range of SNP values.
We, therefore, tested our hypothesis by merging the classification results of the SNP and CIMP-based models using a feed-forward multi-layer perceptron model. Our experiments confirm that the combined model surpasses the baseline model in terms of CRC subtypes classification accuracy.

Our results show a considerable increase in classification accuracy, as measured by the average AUC, when compared to previous works that have utilized CNN models and transfer-learning for CRC subtypes classification, such as the work by Kather et al.
\cite{Kather2019DeepCancer}
This improvement not only reflects better performance but also highlights the impact of molecular subtypes such as SNP and CIMP on the phenotype of H\&E stained histopathological imaging.
Although the state-of-the-art `draw and rank' method by Bilal et al.\cite{Bilal2021DevelopmentStudy} demonstrates a higher classification accuracy (average AUC of 0.9$\pm$0.01 using a repeated cross-validation setup with 3 repetitions and 4 folds), it treats the CNN model for image classification as a `black-box', failing to provide insight into the impact of molecular features on the phenotype of H\&E stained histopathological images.

%limitations
It's worth noting that this study has several limitations. One limitation of this study is that we solely utilized the CRC TCGA dataset as pre-processed by Kather et al.\cite{Kather2019DeepCancer} As such, extrapolating the findings of this study to other datasets should be exercised with caution. Additionally, the use of alternative pre-processing methods may also affect the outcome of the study.
An additional limitation is that we only examined a small number of molecular features. Motivated by  Liu's et al. work \cite{Liu2018ComparativeAdenocarcinomas}, we focused on two specific features, SNP and CIMP, as potential factors that could affect the appearance of H\&E stained histopathological images of CRC. We also examined the effect of CNV as a control. It would be beneficial to investigate the influence of additional molecular features on the phenotype of H\&E stained histopathological images of CRC.
Additionally, factors such as age, gender, and tumor location may also play a role in the appearance of H\&E images and could be utilized to enhance the capability of CNN-based models to effectively classify CRC subtypes using these images.


In conclusion, We introduced a BP-CNN approach that accounts for the potential impact of molecular features on the appearance of H\&E stained histopathological images of CRC. Our study showed the impact of SNP and CIMP molecular features on these images as demonstrated by the improved accuracy of CNN-based models for classifying CRC into subtypes. The results and approach of this study may be of significance to researchers exploring the relationship between genetic mutations and image characteristics in various types of cancer. Additionally, these findings can be utilized by engineers working to enhance the accuracy and interpretability of CNN-based methods for classifying cancer subtypes using H\&E stained histopathological images.


\section*{Methods}

\subsection*{Data and pre-processing}
\begin{figure}
    \centering
    \begin{tabular}[t]{c}
        \includegraphics[width=0.8\textwidth]{images/data_split.pdf}
    \end{tabular}

\caption{Entire cohort of TCGA COAD and READ datasets is n=632. Kather et al. pre-processed and published part of the data, n=360. They split the patients to training (n=260) and testing (n=100) sets. The training set was balanced on patch level. In our work we used stratified folds at the patient level for cross-validation. The folds were determined by the new sub-labels. }
\label{fig:data_split}
\end{figure}

We used the TCGA-CRC-DX\cite{crc-data} cohort in all of our experiments. The dataset contains N=360 patients diagnosed with CRC (CRC-DX). Samples are formalin-fixed paraffin-embedded (FFPE) diagnostic slides, H\&E stained.\\The dataset is a subset of TCGA published data\cite{tcga} and includes, in addition to the H\&E images, DNA mutation, RNA expression, and clinical annotations.
The database went through pre-processing as described in detail by Kather et al.\cite{Kather2019DeepCancer} 
100 out of 360 patients were randomly selected by Kather et al. and set aside as a test set. Multiple image patches were extracted from each H\&E image as described in Kather et al. \cite{Kather2019DeepCancer}.
The training set was balanced at the patch level by randomly discarding MSS patches.
15\% of the training set patients are MSI, 26\% of the test set (n=100) patients are MSI. Figure \ref{fig:data_split} depicts the patient split. 
True labels are assigned with respect to Liu et al. \cite{Liu2018ComparativeAdenocarcinomas} (Supplementary Table 2 in Kather et al. \cite{Kather2019DeepCancer}).

\subsection*{CNN architecture}
\input{figures/net_description}
\input{figures/snp_algo}
\paragraph{Baseline model}
Following the works of Kather et al. \cite{Kather2019DeepCancer} and Bilal et al. \cite{Bilal2021DevelopmentStudy}  we first implemented a baseline CNN model for patch-level binary classification (MSI/MSS) using a transfer-learning approach. We used an Inception-v3\cite{DBLP:journals/corr/SzegedyVISW15} network as our feature-extractor model. The Inception-v3 model was pre-trained on the ImageNet (\url{www.image-net.org}) dataset. We fine-tuned the model by retraining the last 3 inception blocks and the fully-connected layers on our dataset. 
The CNN model provides a patch-level classification. We aggregate the patch-level classifications into a patient-level classification by averaging patch classification probabilities. Formally, for a classifier $F$, the output for some patch $x$ is the probability to belong to the class MSI or to the class MSS, $\left(0 \leq F(x) \leq 1\right)$. 
For an H\&E image $W$, $N$ patches are extracted from $W$, such that $\left((x_1,x_2, \hdots, x_N) \in W\right)$. Then, a patient-level MSI probability is computed as follows:
\begin{equation} 
P_w(MSI) = \frac{\sum_{i=1}^{N}F(x_i)}{N} 
\label{eq:agg}
\end{equation}
Figure \ref{fig:net_arch}(a) presents the baseline model architecture used in our study. 
\\
\paragraph{Molecular features analysis}
We examined the possibility that there is a diversity in the imaging phenotype within the classes which interferes with the CNN learning and generalizing well. To examine that, we used the baseline model patches results and visualized some of their biological feature rates. From the visualization we were able to suggest a type-cast for the incorrectly classified patches. The incorrectly classified patches might have shared features of the opposite class or a spread range of values of the feature. This leads us to an assumption that the CNN learned a narrow phenotype of the classes and we can guide it to look for a wider phenotype by dividing the classes according to the type-casts we concluded. 
%This leads us to establishing a threshold of the biologic feature so that the labels will be further divided by it. 

\paragraph{Biologically-primed models}
 We developed the biologically-primed model classification architecture as follows. We replaced the binary classification layer of the baseline model with a three-class classification layer. One class represents the MSS, and the other two represent two subclasses of the MSI group divided according to some molecular feature. i.e for example the  CIMP-high). Alg. ~\ref{alg:snp} lists the steps for generating the sub classification. It is important to emphasize that while we use the molecular subtypes information during training, our inference process is similar to the baseline architecture. The only difference is the number of classes. The classification to 3 classes in the inference phase requires only the H\&E images without any additional information. Of the three inferred classes two are equivalent in terms of being MSI, i.e. the MSI\&CIMP-high and the MSI of CIMP-low and non-CIMP. 

The CNN here classifies the patches to 3 classes - $MSI_1$, $MSI_2$ and MSS. To perform patient level classification we consider the patch probability of MSI as the higher probability among $MSI_1$, $MSI_2$. Formally,  for patch x, and a classifier F, the MSI probability will be \(P_x(MSI) = max(F(x=MSI_1),F(x=MSI_2)) \). Similarly to the baseline model we calculate the patient-level classification for a H\&E image with equation \ref{eq:agg}. 
Figure \ref{fig:net_arch}(b) portrays the BP-CNN model architecture description.\\

Specifically, we developed 3 biologically-primed models. The first, BP-CNN\textsubscript{SNP}, divided MSI patients into two subgroups based on their SNP rate. We determined the SNP threshold for sub classification by testing various thresholds in the range of 800 to 1500 on the first fold of the training data. We chose the threshold that yielded the highest AUC on the first training fold validation dataset.

The second, BP-CNN\textsubscript{CIMP}, separated the MSI group into CIMP-H vs. non-CIMP-H subgroups in the MSI class. We excluded MSS patches with CIMP-H, resulting in a 3-class classification, MSS, MSI-CIMP-H and MSI-NON-CIMP-H.

We verified that the potential improvement in classification accuracy is not just due to random division by implementing a third model, BP-CNN\textsubscript{CNV}. This model divided the MSI group into two subgroups based on each patient's CNV rate with a threshold of 0.005 chosen using qualitative assessment of the CNV distribution (Figure \ref{fig:feature_boxplot}).

Finally, based on the performance of the BP-CNN models with a single molecular feature, BP-CNN\textsubscript{SNP} and BP-CNN\textsubscript{CIMP}, we constructed a combined model accounting from the uncorrelated improvements achieved by the BP-CNN\textsubscript{SNP} and BP-CNN\textsubscript{CIMP}.
\begin{figure}
    \centering
        \includegraphics[width=0.6\textwidth]{images/combined_pic.pdf}
    \caption{Combined model architecture. Models A and B are biologically-primed models based on 2 different biological features. Network output from trained and fixed model A and model B is concatenated and inserted to a linear layer. Outputs are propagated to a softmax layer for probabilities. N is number of patient patches, $P_i$ are patches MSI probabilities. $P_w$, MSI score for each patient is the average over its MSI probabilities. This figure uses code from D. Stutz repo \cite{latex-repo}. }
    \label{fig:combined}
\end{figure}


 Specifically, we used the BP-CNN\textsubscript{SNP} and BP-CNN\textsubscript{CIMP} to extract class probabilities. Then, we trained a multi-layer perceptron (MLP) with the 6 dimension input vector (3 class probabilities from each model) to predict the binary classification as MSI or MSS. Figure~\ref{fig:combined} presents the architecture of the combined model.
 
 The MLP performs a patch-level classification. Patient-level aggregation is done by taking the average of the patch probabilities (Equation~\ref{eq:agg}).

 \paragraph{Training details}
 All models used Cross-entropy loss function, Adam optimizer, Initial learning rate of $10^{-4}$,
 15 training epochs, 64 images batch size. Saved models are by best validation AUC. Cross-validation folds were divided with Sklearn stratified-folds. A random weighted sampler was utilized to balance the dataset in the patch-level. Our code was implemented in pyTorch 1.9 and executed with Nvidia A100 graphics-processing units (GPUs) through a container image of version 21.04. 
 
\bibliography{sample}


\section*{Acknowledgements}
Y.E.M. acknowledge funding from the Israel science foundation (ISF, grant number 2794/21) and from the Israel cancer association (ICA, grant number 20210132).

\section*{Author contributions statement}

%Must include all authors, identified by initials, for example:
%A.A. conceived the experiment(s),  A.A. and B.A. conducted the experiment(s), C.A. and D.A. analysed the results.  All authors reviewed the manuscript. 
H.H., Y.E.M. and M.F. conceived the experiment(s), H.H conducted the experiments, H.H, , D.G, Y.E.M. and M.F. analysed the results. All authors reviewed the manuscript. 

\section*{Additional information}
%To include, in this order: \textbf{Accession codes} (where applicable);
\textbf{Accession codes} Our source codes are freely available at \url{https://github.com/TechnionComputationalMRILab/MSI_MSS_BP-CNN} \cite{code}.\\
\textbf{Competing interests} The authors declare no competing interests. 

\end{document}