%Version 2.1 April 2023
% See section 11 of the User Manual for version history
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%\documentclass[referee,sn-basic]{sn-jnl}% referee option is meant for double line spacing

%%=======================================================%%
%% to print line numbers in the margin use lineno option %%
%%=======================================================%%

%%\documentclass[lineno,sn-basic]{sn-jnl}% Basic Springer Nature Reference Style/Chemistry Reference Style

%%======================================================%%
%% to compile with pdflatex/xelatex use pdflatex option %%
%%======================================================%%

%%\documentclass[pdflatex,sn-basic]{sn-jnl}% Basic Springer Nature Reference Style/Chemistry Reference Style


%%Note: the following reference styles support Namedate and Numbered referencing. By default the style follows the most common style. To switch between the options you can add or remove Numbered in the optional parenthesis. 
%%The option is available for: sn-basic.bst, sn-vancouver.bst, sn-chicago.bst, sn-mathphys.bst. %  
 
%%\documentclass[sn-nature]{sn-jnl}% Style for submissions to Nature Portfolio journals
%%\documentclass[sn-basic]{sn-jnl}% Basic Springer Nature Reference Style/Chemistry Reference Style
\documentclass[sn-mathphys,Numbered]{sn-jnl}% Math and Physical Sciences Reference Style
%%\documentclass[sn-aps]{sn-jnl}% American Physical Society (APS) Reference Style
%%\documentclass[sn-vancouver,Numbered]{sn-jnl}% Vancouver Reference Style
%%\documentclass[sn-apa]{sn-jnl}% APA Reference Style 
%%\documentclass[sn-chicago]{sn-jnl}% Chicago-based Humanities Reference Style
%%\documentclass[default]{sn-jnl}% Default
%%\documentclass[default,iicol]{sn-jnl}% Default with double column layout

%%%% Standard Packages
%%<additional latex packages if required can be included here>

\usepackage{graphicx}%
\usepackage{multirow}%
\usepackage{amsmath,amssymb,amsfonts}%
\usepackage{amsthm}%
\usepackage{mathrsfs}%
\usepackage[title]{appendix}%
\usepackage{xcolor}%
\usepackage{textcomp}%
\usepackage{manyfoot}%
\usepackage{booktabs}%
\usepackage{algorithm}%
\usepackage{algorithmicx}%
\usepackage{algpseudocode}%
\usepackage{listings}%
%%%%

%%%%%=============================================================================%%%%
%%%%  Remarks: This template is provided to aid authors with the preparation
%%%%  of original research articles intended for submission to journals published 
%%%%  by Springer Nature. The guidance has been prepared in partnership with 
%%%%  production teams to conform to Springer Nature technical requirements. 
%%%%  Editorial and presentation requirements differ among journal portfolios and 
%%%%  research disciplines. You may find sections in this template are irrelevant 
%%%%  to your work and are empowered to omit any such section if allowed by the 
%%%%  journal you intend to submit to. The submission guidelines and policies 
%%%%  of the journal take precedence. A detailed User Manual is available in the 
%%%%  template package for technical guidance.
%%%%%=============================================================================%%%%

%\jyear{2021}%

%% as per the requirement new theorem styles can be included as shown below
\theoremstyle{thmstyleone}%
\newtheorem{theorem}{Theorem}%  meant for continuous numbers
%%\newtheorem{theorem}{Theorem}[section]% meant for sectionwise numbers
%% optional argument [theorem] produces theorem numbering sequence instead of independent numbers for Proposition
\newtheorem{proposition}[theorem]{Proposition}% 
%%\newtheorem{proposition}{Proposition}% to get separate numbers for theorem and proposition etc.

\theoremstyle{thmstyletwo}%
\newtheorem{example}{Example}%
\newtheorem{remark}{Remark}%

\theoremstyle{thmstylethree}%
\newtheorem{definition}{Definition}%

\raggedbottom
%%\unnumbered% uncomment this for unnumbered level heads

\begin{document}

\title[Article Title]{Analysis of DNA sequences through local distribution of nucleotides in strategic neighborhoods}

%%=============================================================%%
%% Prefix	-> \pfx{Dr}
%% GivenName	-> \fnm{Joergen W.}
%% Particle	-> \spfx{van der} -> surname prefix
%% FamilyName	-> \sur{Ploeg}
%% Suffix	-> \sfx{IV}
%% NatureName	-> \tanm{Poet Laureate} -> Title after name
%% Degrees	-> \dgr{MSc, PhD}
%% \author*[1,2]{\pfx{Dr} \fnm{Joergen W.} \spfx{van der} \sur{Ploeg} \sfx{IV} \tanm{Poet Laureate} 
%%                 \dgr{MSc, PhD}}\email{iauthor@gmail.com}
%%=============================================================%%

\author[1]{\fnm{Probir} \sur{Mondal}}\email{probir.008@gmail.com}

\author*[2]{\fnm{Pratyay} \sur{Banerjee}}\email{pratyaybanerjeesinp@gmail.com}
%\equalcont{These authors contributed equally to this work.}

\author[3]{\fnm{Krishnendu} \sur{Basuli}}\email{krishnendu.basuli@gmail.com}
%\equalcont{These authors contributed equally to this work.}

\affil[1]{\orgdiv{Department of Computer Science}, \orgname{P. R. Thakur Govt. College}, \orgaddress{\city{Thakurnagar}, \postcode{743287}, \state{West Bengal}, \country{India}}}

\affil*[2]{\orgdiv{Department of Physics}, \orgname{P. R. Thakur Govt. College}, \orgaddress{ \city{Thakurnagar}, \postcode{743287}, \state{West Bengal}, \country{India}}}

 \affil[3]{\orgdiv{Department of Computer Science}, \orgname{West Bengal State University}, \orgaddress{ \city{Barasat}, \postcode{700126}, \state{West Bengal}, \country{India}}}

%%==================================%%
%% sample for unstructured abstract %%
%%==================================%%

\abstract{We propose a new alignment-free algorithm by constructing a compact vector representation on $\mathbb{R}^{24}$ of a DNA sequence of arbitrary length. Each component of this vector is obtained from a representative sequence, the elements of which are the values realized by a  function $\Gamma$. $\Gamma$ acts on neighborhoods of arbitrary radius that are located at strategic positions within the DNA sequence and carries complete information about the local distribution of frequencies of the nucleotides as a consequence of the  uniqueness of prime factorization of integer. The algorithm exhibits linear time complexity and turns out to consume significantly small memory. The two natural parameters characterizing the radius and location of the neighbourhoods are fixed by comparing the phylogenetic tree with the benchmark for  full genome sequences of fish mtDNA datasets. Using these fitting parameters, the method is applied to analyze a number of genome sequences from benchmark and other standard datasets. The algorithm proves to be computationally efficient compared to Co-phylog and CD-MAWS when applied over a certain range of a simulated dataset. }



\keywords{Alignment-free, Compact representation, Prime factorization, Time complexity}

%%\pacs[JEL Classification]{D8, H51}

%%\pacs[MSC Classification]{35A01, 65L10, 65L12, 65L20, 65L70}

\maketitle

\section{Introduction}\label{sec1}

Recent advancement in molecular biology stems primarily from the inclusion of Information Science and Technology into this field. So far, various computational tools have been developed and applied to analyze biological sequences. There exist algorithms to identify the origin of viruses, the quantum of similarity present among species, the mutation occurring inside them etc. The pioneering work in the field of sequence alignment was done by Needleman-Wunsch \cite{nid:wun} in 1970 followed by Smith-Waterman \cite{ smith:wat}. Following them, alignment-based searching tools, viz., BLAST \cite{al:miller, mark:keith}, FASTA \cite{fasta1,fasta2} etc. were developed.

Molecular biology continued to become increasingly interesting with the enormous growth of its database generated by various initiatives. Despite their accuracy, most of the alignment-based algorithms that have been implemented so far admit quadratic time complexity and, thus, do not seem to have proved useful for analyzing long sequences. Consequently, it seemed customary to develop time-efficient algorithms. A new variety of alignment-free (AF) algorithms \cite{Vinga_38, Zielezinski_AF, Brian_doi_1} arose thereafter as an alternative description of biological sequences with a focus to represent them through concrete mathematical object amenable to further treatment. AF algorithms, in general, admit linear time complexity. Moreover, their global nature is expected to play a vital role while comparing DNA sequences of unequal length.

A typical DNA sequence comprises of four nucleotide bases, viz., Adenine (A), Guanine (G), Cytosine (C) and Thymine (T). To compare two such sequences, there exist in the literature a large number of AF approaches that fall in two broad categories: viz., the word-based method \cite{word_based, Oliver_doi_2} which is based on the frequencies of subsequences of certain length and information-theory based method \cite{info_theory1,info_theory2} which quantifies the informational content between pair of sequences. In addition, there exist AF methods belonging to neither of these two categories. This includes, for example, techniques relying on the length of matching words viz., average common word \cite{avg_common}, shortest unique substring \cite{unique}, the minimal absent words between sequences \cite{minimal_absent}, iterated maps \cite{Almeida}, graphical representation \cite{randic1,zhang:zhang}, chaos game representation \cite{Chaos}: all in the interest to extract information about the distribution of nucleotides within sequences.


Here, in this article, we adopt a new AF approach. Given a DNA sequence, instead of looking at individual nucleotide,
we consider its corresponding course-grained form which provides a `compact' representation of the same in the sense that the length of its corresponding representative sequence is half or even smaller compared to that of the sequence itself. This is a clear indication that the practical running time of our algorithm will decrease.

The arrangement of the paper is as follows: In Sec. 2, we propose our algorithm in detail regarding the association of a scalar with a DNA sequence of arbitrary length and calculation of the euclidean metric $\rho$ between a pair of such sequences. In Sec. 3, we apply our algorithm on full genome sequence of fish mtDNA to fix the values of the two fitting parameters. To do so, we calculate the normalized Robinson-Foulds (nRF) distance of the phylogenetic tree obtained from our algorithm with respect to the standard tree in the benchmark datasets \cite{Benchmark}. To check the accuracy of the algorithm we run it on four other genome sequences from benchmark. In the next section, we explicitly demonstrate the efficiency of our method on six complete genome sequences by comparing running times with four well-known AF algorithms. Moreover, we compare graphically the running time and peak memory consumed by our method with two other standard algorithms on simulated datasets. In Sec. 5, we finally draw conclusion and state possible future direction.

\section{Construction of the representative vector}\label{sec2}

Consider a typical string $\xi$ consisting of $N$ nucleotides. We denote such a string as $\xi=s_1 s_2 \cdots s_N$, where each $s_i$ is one of the four nucleotides, viz., A, C, G and T. Let P be the ordered set containing the first four prime numbers, i.e., P$=\{2,3,5,7\}$ and let $\sigma_i$ be the $i^{\text{th}}$ permutation of the set P. Clearly, we have $4!=24$ such permutations, viz., $\sigma_0$, $\sigma_1$, $\ldots$, $\sigma_{23}$; where we have chosen $\sigma_0$ to be the identity permutation.
%i.e., $\sigma_0(\alpha)=\alpha$ for any prime $\alpha \in P$.
Now we assign to each of the nucleotides, corresponding to a particular permutation $\sigma_i$ of the set $P$, the first four prime numbers as follows: $A=\sigma_i(1), C=\sigma_i(2), G=\sigma_i(3)$ and $T=\sigma_i(4)$, where $\sigma_i(j)$ is the $j^{\text{th}}$ element corresponding to the $i^{\text{th}}$ permutation of the set P. For example, in the case of identity permutation, we have $i=0$ and thus~ $A=\sigma_0(1)=2, C=\sigma_0(2)=3, G=\sigma_0(3)=5, T=\sigma_0(4)=7$.  Next, we define a $l$-neighbourhood $U_l(s_i)$ of the nucleotide $s_i$ with radius $l$ and centre $s_i$ to be the set $U_l(s_i)\equiv  \{s_{i-l}, \ldots, s_i, \ldots, s_{i+l} \}$. Let $\Gamma_j$ be the function associating with each neighbourhood $U_l(s_i)$, a positive integer $\Gamma_j(U_l(s_i))$ with respect to the permutation $\sigma_j$ through the following way:
\begin{equation}
\label{a1}
 \Gamma_j(U_l(s_i))=[\sigma_j(1)]^{f_1}\cdotp [\sigma_j(2)]^{f_2}  \cdotp[ \sigma_j(3)]^{f_3} \cdotp [\sigma_j(4)]^{f_4}
\end{equation}
where the exponents $f_1, f_2,f_3$ and $f_4$ are the frequencies of the four nucleotides A, C, G and T, respectively in the neighbourhood $U_l(s_i)$. In other words, the value of $\Gamma$ at a neighbourhood is simply the product of the prime numbers (PPN) assigned in a particular permutation to the nucleotides appearing in that neighbourhood. Note there is no nucleotide $s_i$ for $i 
< 1$ or $i > N$. Clearly, $0 \leq  f_r \leq 2l+1$ for $r\in \{1,2,3,4\}$. It is useful to note that the function $\Gamma_j(U_l(s_i))$ contains complete information about the frequencies of the nucleotides in the neighbourhood $U_l(s_i)$. This follows from the uniqueness of the prime factorization of a positive integer. However, it is observed that given the function $\Gamma_j(U_l(s_i))$, one can reconstruct the associated neighbourhood $U_l(s_i)$ with the exact ordering of nucleotides only in case the degeneracy (in the ordering of nucleotides) arising out of permutation is lifted depending upon the values of $\Gamma$ at the two adjacent neighbourhoods on either side of $U_l(s_i)$. This point is illustrated at the end of the present section.

\vspace{.1cm}

Let 

\begin{equation}
\small
\label{a2}
 \zeta=\bigg\{ \Gamma_j(U_l(s_1)), \Gamma_j(U_l(s_{t+2})), \Gamma_j(U_l(s_{2t+3})), \ldots, \Gamma_j(U_l(s_w))\bigg\}
\end{equation}
 be a sequence consisting of $n$ numbers 
 $\Gamma_j(U_l(s_i))$, where 
\begin{equation}
\label{a3}
 n=1+\left\lfloor \frac{N-1}{t+1}\right\rfloor
\end{equation}
and $w=(n-1)t+n$. Here $\lfloor z \rfloor$ denotes the integer part of $z$ and $t$ is the distance (i.e., number of nucleotides) between the centres of two successive neighbourhoods. In Sec. 3, for a given value of $l$, we shall set $1 \leq t \leq l$. Otherwise, two adjacent neighbourhoods will no longer overlap and may lead to loss of information. We claim the sequence $\zeta$ in Eq. (\ref{a2}) to be a compact representation of the string $\xi$ since from Eq. (\ref{a3}) we find $ n \lesssim \frac{N}{2}$. Next, our objective is to associate a scalar with the sequence (\ref{a2}) in a way such that the scalar is sufficiently sensitive under the transformation of string $\xi$ through point mutation, insertion and deletion. To this end, we propose the scalar $\eta_j$ (with respect to the permutation $\sigma_j$) associated with the sequence $\zeta$  by adding  all the entries of $\zeta$  as follows:
\begin{equation}
\label{a4}
 \eta_j=\sum\limits_{i=1}^n \Gamma_j(U_l(s_{i+t(i-1)}))
\end{equation}

In order not to give preference to any particular permutation 
we assign prime numbers to the nucleotides A, C, G and T corresponding to every permutation $\sigma_j$ of the set P. Thus, for every $\sigma_j$ we obtain a scalar $\eta_j$ from Eq. (\ref{a4}).  In this way a representative vector $\vec{\eta}=(\eta_0, \eta_1, \cdots, \eta_{23})$ is constructed corresponding to the string $\xi$. As each $\eta_j$ is real, we presume the vector $\vec{\eta} $ resides in the $24$-dimensional real euclidean space $\mathbb{R}^{24}$ endowed with the euclidean metric $\rho$. Thus two DNA sequences are compared by computing $\rho$ between the corresponding representative vectors.


% \begin{equation}
% \label{a4}
%  E=\sum\limits_0^k {[P(i)]}^2
% \end{equation}




 
As an example, let us consider a typical string of nucleotides
\begin{equation}
\label{a5}
 ACTGCCTCGATAA
\end{equation}
Here N=13. Choose, say, the identity permutation 
$\sigma_0$. Then it follows that A=2, C=3, G=5 and T=7. We choose the two parameters as $l=1$ and $t=1$. Thus we consider neighbourhoods with centre at every alternate nucleotide comprising only of nearest neighbours. The neighbourhoods for the string (\ref{a5}) (see Fig. \ref{fig_1}) turns out to be $U_1(s_1)=\{A, C \}, U_1(s_3)=\{C, T, G \}, U_1(s_5)=\{G, C, C \}, U_1(s_7)=\{C, T, C \}, U_1(s_9)=\{C, G, A \}, U_1(s_{11})=\{A, T, A \}$ and $U_1(s_{13})=\{A, A \}$.

 \begin{figure}[h]

\begin{center}

\includegraphics[keepaspectratio=false,width=7cm,height=2cm,scale=0.2,angle=0]{fig1}

 \end{center}
 \caption{\scriptsize A typical string of $13$ nucleotides. The collection of nucleotides lying inside an over/under brace forms a neighborhood with centre at the middle nucleotide and radius $l=1$. However, the  two extreme nucleotides are the centres of the neighborhoods $U_1(s_1)$ and $U_1(s_{13})$, respectively as $U_1(s_1)$ has no nucleotide on the left of its centre at $s_1$ and $U_1(s_{13})$ has no nucleotide on the right of $s_{13}$. The distance $t$, i.e., the number of nucleotides between successive neighbourhoods, is $1$. }
 \label{fig_1}
 \end{figure}

Note that the neighbourhoods at the two extreme positions naturally contain fewer nucleotides. Using Eq. (\ref{a1}), we calculate, say, the function $ \Gamma_0(U_1(s_5))$ with respect to the identity  permutation $\sigma_0$ as
$\Gamma_0(\{G,C,C\})=2^0 \cdotp 3^2 \cdotp 5^1 \cdotp 7^0=45$. From Eq. (\ref{a2}) we construct the sequence as 
\begin{equation}
\label{a6}
 \zeta=\bigg\{ 6,105,45,63,30,28,4 \bigg\}.
\end{equation}
 Using Eq. (\ref{a3}), we find $n=7$. Finally, from Eq. (\ref{a4}), the scalar $\eta_0$ associated with the string (\ref{a5}), which is simply the sum of the elements of the sequence (\ref{a6}), turns out to be $\eta_0=281$. As stated earlier, we call the sequence (\ref{a2}) to be a compact representation of the string $\xi$ as the former contains around half or even smaller number of elements compared to the latter. In addition, the uniqueness of prime factorisation of integer reproduces the local distribution of various nucleotides with exact frequencies. However, the invertibility of this representation depends heavily on the entries of the sequence (\ref{a2}). As an illustration, choose a subsequence $\zeta'=\{6,105\}$ of the sequence $\zeta$ in (\ref{a6}) by selecting the first two elements. As $6=2^1 \cdotp 3^1 \cdotp 5^0 \cdotp 7^0$, the corresponding neighbourhood, which is the inverse image of the function $\Gamma$,  is either \{A, C\} or \{C, A\}. Similarly for the second element of $\zeta'$, we find $105=2^0 \cdotp 3^1 \cdotp 5^1 \cdotp 7^1$, so that the corresponding neighbourhood is one of the 6 permutations of C, G and T, viz., \{C, G, T\} , \{C, T, G\}, \{G, C, T\}, \{T, C, G\}, \{G, T, C\} and \{T, G, C\}. Now, observe that the nucleotide A is absent in  all of the six neighbourhoods. Thus, looking at the overlap region, it is clear that \{A, C\} is the right choice of neighbourhood corresponding to the element 6 of $\zeta'$. Consequently,  \{C, G, T\} and \{C, T, G\} are the remaining possibilities as the inverse image of $\Gamma$ of the second element 105 of $\zeta'$. Thus from the subsequence $\zeta'$, we reproduce (up to a degeneracy in the ordering of the last two nucleotides) the corresponding string of nucleotides to be either $ACGT$ or $ACTG$. Notice that the example (\ref{a5}) that we have chosen here is perfectly reproducible following this procedure from its representative sequence (\ref{a6}).

\section{Similarity analysis}\label{sec4}
In order to test the usefulness of our algorithm, proposed in Sec. 2 and denoted as PPN hereafter, we apply it on the benchmark dataset of fish mtDNA from AFproject \cite{Benchmark}. The corresponding phylogenetic tree obtained through PPN is shown in Fig. \ref{fish_mtDNA} by setting the parameters at $l=4$ and $t=1$. Upon comparison with the benchmark, we find that PPN ranks $11^{\text{th}}$ for fish mtDNA with nRF=0.64 while the normalized Quartet Distance  (nQD) is 0.2602. Although we find the same nRF value by implementing the Manhattan metric (in place of euclidean distance) between the representative vectors, the nQD value is 0.2723. It is to be noted that PPN is expected to operate upto $l = 4$ as the fluctuation in the distribution of frequencies of different nucleotides in a neighborhood becomes smaller with increasing radius.

\begin{figure}[h!]
\begin{center}
\includegraphics[width=8.5cm]{fish_mtDNA} %fbox to put boundary
\caption{{\scriptsize UPGMA tree for fish mtDNA sequences having 25 species drawn in software MEGA 11 with parameter values $l=4$, $t=1.$ }}
\label{fish_mtDNA}
\end{center}
\end{figure}  

\begin{figure}[h!]
\begin{center}
\label{mammals_own}
\includegraphics[width=8.5cm]{mammals_own} %fbox to put boundary
\caption{{\scriptsize UPGMA tree for mammals mtDNA sequences having 41 species drawn in software MEGA 11 with parameter values $l=4$, $t=1.$ }}
\end{center}
\end{figure} 
Choosing $l=4, t=1$, we apply PPN on the assembled 29 E. Coli/Shigella strain taken from the Genome-based phylogeny (GBP) section and on Yersinia, plants and unsimulated 27 E. Coli/Shigella strain  from the Horizontal Gene Transfer (HGT) section of the benchmark datasets. We record the nRF, nQD and rank for each of these datasets in Table \ref{nrf_distance}  as obtained from AFproject \cite{Benchmark}.



\begin{table}[h]
\caption{The nRF, nQD and rank using PPN for five genome sequences. }\label{nrf_distance}
\begin{tabular*}{\textwidth}{@{\extracolsep\fill}lcccccc}
\toprule%

Genome sequence & nRF & nQD & Rank  \\
\midrule

Fish mtDNA GBP                                 & 0.64         & 0.2602       & 11            \\ 
Unsimulated 27 E.Coli/Shigella strain HGT      & 0.79         & 0.4369       & 15            \\ 
Assembled 29 E.Coli/Shigella strain GBP        & 0.73         & 0.3337       & 16            \\ 
Plants HGT                                     & 0.73         & 0.3077       & 9             \\ 
Yersinia HGT                                   & 0.80         & 0.6043       & 5             \\ 
\botrule
\end{tabular*}

\end{table}






In addition, using the datasets from ref. \cite{fast_vector}, we generate the phylogenetic tree shown in Fig. \ref{mammals_own} through PPN for the mammals mtDNA. Implementation of our algorithm is available  at \url{https://github.com/Workspace-PM/PPN}.


\section{Performance analysis }\label{sec5}
In this section we demonstrate the efficiency of PPN proposed in Sec. 2 by comparing the running time against standard AF tools. We have conducted the computation on a machine with Intel(R) Core(TM) i7-8700 CPU @ 3.20 GHz x 12, 16 GB RAM and driven by Ubuntu 22.04.03 OS. The running times taken by PPN for complete genome sequences of mammals, influenza A virus, rhinovirus, ebolavirus, coronavirus and bacteria are shown in the first row of Table \ref{tab:Running_time}. The running times of four other standard algorithms viz., CD-MAWS \cite{CD_MAWS}, Co-phylog \cite{co_phylog}, KSNP-3 \cite{kSNP3} and Skmer \cite{skmer} (applied on the same sequences) are reproduced from (Table 5 of) ref. \cite{CD_MAWS} in the next four rows of the Table  \ref{tab:Running_time}. It is apparent from this comparison that PPN,  which admits a linear time-complexity, is, indeed, fast. Note the value of the parameter $t$ provides an estimate of the extent to which one  can compactify a given DNA sequence. For example, when $t$ is set to be $1$, i.e., at its minimum value, we find the length $n$ in Eq. (\ref{a3}) of the sequence $\zeta$ becomes approximately half that of the original DNA sequence $\xi$.





\begin{table}[h]
\caption{Running time comparison of six complete genome sequences}\label{tab:Running_time}
\begin{tabular*}{\textwidth}{@{\extracolsep\fill}lcccccc}
\toprule%
& \multicolumn{6}{@{}c@{}}{Running time in minutes}   \\\cmidrule{2-7}%
Method & Mammals & Influenza & Rhino & Ebola & Corona & Bacteria \\
\midrule
% Element 3  & 990 A & 1168 & $1547\pm12$\\
% Element 4  & 500 A & 961  & $922\pm10$  \\
PPN & 0.048 & 0.005 & 0.064 &0.082 & 0.065  & 16.105   \\ 
CD-MAWS &0.181   & 0.008     & 0.200 & 0.076 & 0.088  & 3.318    \\ 
Co-phylog   & 0.055  & ERR       & 0.157 & 0.076 &ERR   & 11.483   \\ 
kSNP3    & 1.133   & 0.500     & 7.683 & 1.416 & 0.867  & 1.716    \\ 
Skmer    & 0.035   & ERR       & 0.203 & 0.061 & 0.050  & 0.608    \\ 
\botrule

\end{tabular*}
\begin{center}
\footnotetext{ERR: Shows an error.}    
\end{center}
\end{table}





\begin{figure}[!ht]
\begin{center}
\includegraphics[width=\linewidth]{running_time}\\ %fbox to put boundary
\caption{{\scriptsize Average running time (in minute) of PPN for simulated datasets with number of species ranging from 100 to 900. Each species contains 50,000 nucleotides.}}
\label{running_time}
\end{center}
\end{figure}  


\begin{figure}[ht]
\begin{center}
\includegraphics[width=\linewidth]{PeakMemoryConsumption}\\ %fbox to put boundary
\caption{{\scriptsize Peak memory consumption in MB by PPN for simulated datasets with number of species varying between 100 and 900. Here, each species contains 50,000 nucleotides.}}
\label{PeakMemoryConsumption}
\end{center}
\end{figure}  



Additionally, we run PPN on simulated datasets generated by Seq-gen Monte Carlo simulation tool \cite{seq_gen} where each sequence representing a particular species contains 50,000 nucleotides. In Fig. \ref{running_time} we draw the variation of the average running time taken by three algorithms (including PPN) with the number of species. The average running time is calculated by taking the average of 10 runs conducted on a machine with Intel(R) Core(TM) i7-8700 CPU @ 3.20 GHz x 12, 16 GB RAM and driven by Ubuntu 22.04.03 OS. The number of species is varied over a wide range from 100 to 900. Similarly, in Fig. \ref{PeakMemoryConsumption}, the peak memory consumption is recorded by choosing the maximum value from 10 trials. On this simulated dataset, PPN seems to be computationally efficient both in terms of runtime and peak memory consumption compared to Co-phylog and CD-MAWS within  the range shown in Fig. \ref{running_time} and Fig. \ref{PeakMemoryConsumption}. It is to be noted that the combinatoric factor representing the number of actual pairs of sequences to be compared within a dataset increases quadratically with the number of species.  

Moreover, to unravel the truly alignment free nature of PPN, we apply it to compare two sequences differing drastically in the number of nucleotides. One chromosome of Zea mays (GCF 000005005.2) contains 30,70,41,717 nucleotides while that of Oryza sativa (GCF 001433935.1) contains 4,32,70,923 nucleotides. PPN runs successfully and takes 33.68 minute to determine the distance between these sequences and consumes maximum 20.24 GB memory with Intel(R) Xeon(R) GOLD 6134 CPU @3.20GHz x 16, 64GB RAM and Ubuntu 22.04 OS.


\section{Conclusion}\label{sec6}

In this article, we have adopted a new alignment-free approach to trace the amount of similarity/dissimilarity present within a pair of DNA sequences. PPN associates a scalar in a simple way to the representative sequence $\zeta$ corresponding to the string $\xi$. The problem regarding comparing sequences of unequal length is circumvented through the construction of the said scalar. The significant reduction in running time has occurred in two steps. As a first step, we have considered the course-grained form of the original DNA sequence  $\xi$ to reduce its length to a considerable extent. Further reduction happens due to the asymptotically linear nature of the time complexity arising in our algorithm. Comparison with standard AF algorithms clearly reveals that PPN is reasonably fast and also its peak memory consumption is significantly small. We believe the simplicity inherent in our algorithm will make it accessible to  a larger group of users. Moreover, PPN, being fast and memory efficient, can be used to extract features from the sequences in a dataset which is essential to construct model using machine learning techniques for further analysis. 



\begin{thebibliography}{20}
%1
\bibitem{nid:wun}
S. B. Needleman and C. D. Wunsch, "A general method applicable to the search for similarities in the amino acid sequence of two proteins", \textit{J. Mol. Biol.}, vol. 48, pp. 443–453, 1970.

%2
\bibitem{smith:wat}
T. F. Smith and M. S. Waterman, "Identification of common molecular subsequences", \textit{J. Mol. Biol.}, vol. 147, no. 1, pp. 195–197, 1981.

%3
\bibitem{al:miller}
S. F. Altschul, W. Gish, W. Miller, E. W. Myers and D. J. Lipman, "Basic local alignment search tool", \textit{J. Mol. Biol.}, vol. 215, pp. 403–410, 1990.

%4
\bibitem{mark:keith}
K. J. Menlove, M. Clement and K. A. Crandall, "Similarity searching using BLAST", \textit{Methods Mol. Biol.}, vol. 537, pp. 1-22,    2009.

\bibitem{fasta1}
W. R. Pearson, "Rapid and sensitive sequence comparison with FASTP and FASTA", \textit{Methods Enzymol.}, vol. 183, pp. 63-98, 1990.
%5
\bibitem{fasta2}
W. R. Pearson and D. J. Lipman, "Improved tools for biological sequence comparison", \textit{Proc. Natl. Acad. Sci. USA}, vol. 85, pp. 2444-2448, 1988.



\bibitem{Vinga_38}
 S. Vinga and J. Almeida, "Alignment-free sequence comparison - a review", \textit{Bioinformatics}, vol. 19, pp. 513–523, 2003.



\bibitem{Zielezinski_AF}
 A. Zielezinski, S. Vinga, J. Almeida and W. M. Karlowski, "Alignment-free sequence comparison: benefits, applications, and tools", \textit{Genome Biol.}, vol. 18, Article no. 186, 2017.

%Oct 3;18(1):186. doi: 10.1186/s13059-017-1319-7. PMID: 28974235; PMCID: PMC5627421.

\bibitem{Brian_doi_1}
 B. B. Luczak, B. T. James and H. Z. Girgis, "A survey and evaluations of histogram-based statistics in alignment-free sequence comparison", \textit{Brief. Bioinform.}, vol. 20(4), pp. 1222–1237, 2019.

%https://doi.org/10.1093/bib/bbx161

\bibitem{word_based}
 B. E. Blaisdell, "Average values of a dissimilarity measure not requiring sequence alignment are twice the averages of conventional mismatch counts requiring sequence alignment for a computer-generated model system", \textit{J. Mol. Evol.}, vol. 29(6), pp. 538–547, 1989.


\bibitem{Oliver_doi_2}
 O. Bonham-Carter, J. Steele and D. Bastola, "Alignment-free genetic sequence comparisons: a review of recent approaches by word analysis",  \textit{Brief. Bioinform.}, vol. 15, pp. 890–905, 2013.

\bibitem{info_theory1}
 S. Vinga, "Information theory applications for biological sequence analysis", \textit{Brief. Bioinform.}, vol. 15, pp. 376–389,
 2014.
 
\bibitem{info_theory2} 
 Y. Gao and L. Luo, "Genome-based phylogeny of dsDNA viruses by a novel alignment-free method", \textit{Gene}, vol. 492(1), 309–314, 2012.

\bibitem{avg_common}
 I. Ulitsky, D. Burstein, T. Tuller and B. Chor, "The average common substring approach to phylogenomic reconstruction", \textit{J. Comput. Biol.}, vol. 13, pp. 336-350, 2006.

\bibitem{unique}
 B. Haubold, N. Pierstorff, F. Möller and T. Wiehe, "Genome comparison without alignment using shortest unique substrings", \textit{BMC Bioinformatics}, vol. 6, Article no. 123, 2005.

\bibitem{minimal_absent}
A. J. Pinho, P. JSG Ferreira, S. P. Garcia and J. MOS Rodrigues, "On finding minimal absent words", \textit{BMC Bioinformatics}, 
 vol. 10, Article no. 137, 2009.


\bibitem{Almeida}
 J. S. Almeida, "Sequence analysis by iterated maps, a review", \textit{Brief Bioinform.}, vol. 15(3), pp. 369-375, 2014.


\bibitem{randic1}
M. Randić, M. Vraćko, N. Lerś and P. Dejan, "Analysis of similarity/dissimilarity of DNA sequences based on novel 2-D graphical representation", \textit{Chem. Phys. Lett.}, vol. 371, pp. 202-207, 2003.


\bibitem{zhang:zhang}
 Z. Zhang, S. Wang, X. Zhang and Z. Zhang, "Similarity analysis of DNA sequences based on a compact representation", \textit{IEEE  Fifth International Conference on Bio-Inspired Computing: Theories and Applications (BIC-TA).}, pp. 1143-1146, 2010.



\bibitem{Chaos}
 H. J. Jeffrey, "Chaos game representation of gene structure", \textit{Nucleic Acids Res.}, vol. 18(8), pp. 2163–2170, 1990.


 \bibitem{Benchmark}
  A. Zielezinski, H. Z. Girgis, G. Bernard \textit{et al.}, "Benchmarking of alignment-free sequence comparison methods", \textit{Genome Biol.}, vol. 20, Article no. 144, 2019.

\bibitem{fast_vector} Y. Li, L. He, R. L. He \textit{et al.}, “A novel fast vector method for genetic sequence comparison”, 
\textit{Sci. Rep.}, vol. 7, Article no. 12226, 2017.

\bibitem{CD_MAWS}

N. Anjum, R. L. Nabil, R. I. Rafi, Md. S. Bayzid and M. S. Rahman, "CD-MAWS: An alignment-free phylogeny estimation method using cosine distance on minimal absent word sets. \textit{IEEE/ACM Trans. Comput. Biol. Bioinform.} vol. 20(1), pp. 196-205, 2023.



\bibitem{co_phylog}
 H. Yi and L. Jin, "Co-phylog: an assembly-free phylogenomic approach for closely related organisms", \textit{Nucleic Acids Res.}, vol. 41(7), pp. e75, 2013. 

 
\bibitem{kSNP3}S. N. Gardner, T. Slezak and B. G. Hall, "kSNP3.0: SNP detection and phylogenetic analysis of genomes without genome alignment or reference genome", \textit{Bioinform.}, vol. 31, no. 17, pp. 2877–2878, 2015.

\bibitem{skmer}
 S. Sarmashghi, K. Bohmann, M. T. P. Gilbert \textit{et al.}, "Skmer: assembly-free and alignment-free sample identification using genome skims", \textit{Genome Biol.}, vol. 20, Article no. 34, 2019.

 \bibitem{seq_gen}
 A. Rambaut and N. C. Grassly, "Seq-Gen: an application for the Monte Carlo simulation of DNA sequence evolution along phylogenetic trees", \textit{Comput. Appl. Biosci.}, vol. 13, issue 3, pp. 235-238, 1997.

\end{thebibliography}






%%===========================================================================================%%
%% If you are submitting to one of the Nature Portfolio journals, using the eJP submission   %%
%% system, please include the references within the manuscript file itself. You may do this  %%
%% by copying the reference list from your .bbl file, paste it into the main manuscript .tex %%
%% file, and delete the associated \verb+\bibliography+ commands.                            %%
%%===========================================================================================%%

%\bibliography{sn-bibliography}% common bib file
%% if required, the content of .bbl file can be included here once bbl is generated
%%\input sn-article.bbl


\end{document}
