
%% bare_jrnl_compsoc.tex
%% V1.4b
%% 2015/08/26
%% by Michael Shell
%% See:
%% http://www.michaelshell.org/
%% for current contact information.
%%
%% This is a skeleton file demonstrating the use of IEEEtran.cls
%% (requires IEEEtran.cls version 1.8b or later) with an IEEE
%% Computer Society journal paper.
%%
%% Support sites:
%% http://www.michaelshell.org/tex/ieeetran/
%% http://www.ctan.org/pkg/ieeetran
%% and
%% http://www.ieee.org/

%%*************************************************************************
%% Legal Notice:
%% This code is offered as-is without any warranty either expressed or
%% implied; without even the implied warranty of MERCHANTABILITY or
%% FITNESS FOR A PARTICULAR PURPOSE! 
%% User assumes all risk.
%% In no event shall the IEEE or any contributor to this code be liable for
%% any damages or losses, including, but not limited to, incidental,
%% consequential, or any other damages, resulting from the use or misuse
%% of any information contained here.
%%
%% All comments are the opinions of their respective authors and are not
%% necessarily endorsed by the IEEE.
%%
%% This work is distributed under the LaTeX Project Public License (LPPL)
%% ( http://www.latex-project.org/ ) version 1.3, and may be freely used,
%% distributed and modified. A copy of the LPPL, version 1.3, is included
%% in the base LaTeX documentation of all distributions of LaTeX released
%% 2003/12/01 or later.
%% Retain all contribution notices and credits.
%% ** Modified files should be clearly indicated as such, including  **
%% ** renaming them and changing author support contact information. **
%%*************************************************************************


% *** Authors should verify (and, if needed, correct) their LaTeX system  ***
% *** with the testflow diagnostic prior to trusting their LaTeX platform ***
% *** with production work. The IEEE's font choices and paper sizes can   ***
% *** trigger bugs that do not appear when using other class files.       ***                          ***
% The testflow support page is at:
% http://www.michaelshell.org/tex/testflow/


\documentclass[10pt,journal,compsoc]{IEEEtran}
%
% If IEEEtran.cls has not been installed into the LaTeX system files,
% manually specify the path to it like:
% \documentclass[10pt,journal,compsoc]{../sty/IEEEtran}





% Some very useful LaTeX packages include:
% (uncomment the ones you want to load)


% *** MISC UTILITY PACKAGES ***
%
%\usepackage{ifpdf}
% Heiko Oberdiek's ifpdf.sty is very useful if you need conditional
% compilation based on whether the output is pdf or dvi.
% usage:
% \ifpdf
%   % pdf code
% \else
%   % dvi code
% \fi
% The latest version of ifpdf.sty can be obtained from:
% http://www.ctan.org/pkg/ifpdf
% Also, note that IEEEtran.cls V1.7 and later provides a builtin
% \ifCLASSINFOpdf conditional that works the same way.
% When switching from latex to pdflatex and vice-versa, the compiler may
% have to be run twice to clear warning/error messages.






% *** CITATION PACKAGES ***
%
\ifCLASSOPTIONcompsoc
  % IEEE Computer Society needs nocompress option
  % requires cite.sty v4.0 or later (November 2003)
  \usepackage{cite}
  \usepackage[colorlinks, citecolor=blue, linkcolor=blue]{hyperref}
\else
  % normal IEEE
  \usepackage{cite}
\fi
% cite.sty was written by Donald Arseneau
% V1.6 and later of IEEEtran pre-defines the format of the cite.sty package
% \cite{} output to follow that of the IEEE. Loading the cite package will
% result in citation numbers being automatically sorted and properly
% "compressed/ranged". e.g., [1], [9], [2], [7], [5], [6] without using
% cite.sty will become [1], [2], [5]--[7], [9] using cite.sty. cite.sty's
% \cite will automatically add leading space, if needed. Use cite.sty's
% noadjust option (cite.sty V3.8 and later) if you want to turn this off
% such as if a citation ever needs to be enclosed in parenthesis.
% cite.sty is already installed on most LaTeX systems. Be sure and use
% version 5.0 (2009-03-20) and later if using hyperref.sty.
% The latest version can be obtained at:
% http://www.ctan.org/pkg/cite
% The documentation is contained in the cite.sty file itself.
%
% Note that some packages require special options to format as the Computer
% Society requires. In particular, Computer Society  papers do not use
% compressed citation ranges as is done in typical IEEE papers
% (e.g., [1]-[4]). Instead, they list every citation separately in order
% (e.g., [1], [2], [3], [4]). To get the latter we need to load the cite
% package with the nocompress option which is supported by cite.sty v4.0
% and later. Note also the use of a CLASSOPTION conditional provided by
% IEEEtran.cls V1.7 and later.





% *** GRAPHICS RELATED PACKAGES ***
%
\ifCLASSINFOpdf
  % \usepackage[pdftex]{graphicx}
  % declare the path(s) where your graphic files are
  % \graphicspath{{../pdf/}{../jpeg/}}
  % and their extensions so you won't have to specify these with
  % every instance of \includegraphics
  % \DeclareGraphicsExtensions{.pdf,.jpeg,.png}
\else
  % or other class option (dvipsone, dvipdf, if not using dvips). graphicx
  % will default to the driver specified in the system graphics.cfg if no
  % driver is specified.
  % \usepackage[dvips]{graphicx}
  % declare the path(s) where your graphic files are
  % \graphicspath{{../eps/}}
  % and their extensions so you won't have to specify these with
  % every instance of \includegraphics
  % \DeclareGraphicsExtensions{.eps}
\fi
% graphicx was written by David Carlisle and Sebastian Rahtz. It is
% required if you want graphics, photos, etc. graphicx.sty is already
% installed on most LaTeX systems. The latest version and documentation
% can be obtained at: 
% http://www.ctan.org/pkg/graphicx
% Another good source of documentation is "Using Imported Graphics in
% LaTeX2e" by Keith Reckdahl which can be found at:
% http://www.ctan.org/pkg/epslatex
%
% latex, and pdflatex in dvi mode, support graphics in encapsulated
% postscript (.eps) format. pdflatex in pdf mode supports graphics
% in .pdf, .jpeg, .png and .mps (metapost) formats. Users should ensure
% that all non-photo figures use a vector format (.eps, .pdf, .mps) and
% not a bitmapped formats (.jpeg, .png). The IEEE frowns on bitmapped formats
% which can result in "jaggedy"/blurry rendering of lines and letters as
% well as large increases in file sizes.
%
% You can find documentation about the pdfTeX application at:
% http://www.tug.org/applications/pdftex






% *** MATH PACKAGES ***
%
%\usepackage{amsmath}
% A popular package from the American Mathematical Society that provides
% many useful and powerful commands for dealing with mathematics.
%
% Note that the amsmath package sets \interdisplaylinepenalty to 10000
% thus preventing page breaks from occurring within multiline equations. Use:
%\interdisplaylinepenalty=2500
% after loading amsmath to restore such page breaks as IEEEtran.cls normally
% does. amsmath.sty is already installed on most LaTeX systems. The latest
% version and documentation can be obtained at:
% http://www.ctan.org/pkg/amsmath





% *** SPECIALIZED LIST PACKAGES ***
%
%\usepackage{algorithmic}
% algorithmic.sty was written by Peter Williams and Rogerio Brito.
% This package provides an algorithmic environment fo describing algorithms.
% You can use the algorithmic environment in-text or within a figure
% environment to provide for a floating algorithm. Do NOT use the algorithm
% floating environment provided by algorithm.sty (by the same authors) or
% algorithm2e.sty (by Christophe Fiorio) as the IEEE does not use dedicated
% algorithm float types and packages that provide these will not provide
% correct IEEE style captions. The latest version and documentation of
% algorithmic.sty can be obtained at:
% http://www.ctan.org/pkg/algorithms
% Also of interest may be the (relatively newer and more customizable)
% algorithmicx.sty package by Szasz Janos:
% http://www.ctan.org/pkg/algorithmicx




% *** ALIGNMENT PACKAGES ***
%
%\usepackage{array}
% Frank Mittelbach's and David Carlisle's array.sty patches and improves
% the standard LaTeX2e array and tabular environments to provide better
% appearance and additional user controls. As the default LaTeX2e table
% generation code is lacking to the point of almost being broken with
% respect to the quality of the end results, all users are strongly
% advised to use an enhanced (at the very least that provided by array.sty)
% set of table tools. array.sty is already installed on most systems. The
% latest version and documentation can be obtained at:
% http://www.ctan.org/pkg/array


% IEEEtran contains the IEEEeqnarray family of commands that can be used to
% generate multiline equations as well as matrices, tables, etc., of high
% quality.




% *** SUBFIGURE PACKAGES ***
%\ifCLASSOPTIONcompsoc
%  \usepackage[caption=false,font=footnotesize,labelfont=sf,textfont=sf]{subfig}
%\else
%  \usepackage[caption=false,font=footnotesize]{subfig}
%\fi
% subfig.sty, written by Steven Douglas Cochran, is the modern replacement
% for subfigure.sty, the latter of which is no longer maintained and is
% incompatible with some LaTeX packages including fixltx2e. However,
% subfig.sty requires and automatically loads Axel Sommerfeldt's caption.sty
% which will override IEEEtran.cls' handling of captions and this will result
% in non-IEEE style figure/table captions. To prevent this problem, be sure
% and invoke subfig.sty's "caption=false" package option (available since
% subfig.sty version 1.3, 2005/06/28) as this is will preserve IEEEtran.cls
% handling of captions.
% Note that the Computer Society format requires a sans serif font rather
% than the serif font used in traditional IEEE formatting and thus the need
% to invoke different subfig.sty package options depending on whether
% compsoc mode has been enabled.
%
% The latest version and documentation of subfig.sty can be obtained at:
% http://www.ctan.org/pkg/subfig




% *** FLOAT PACKAGES ***
%
%\usepackage{fixltx2e}
% fixltx2e, the successor to the earlier fix2col.sty, was written by
% Frank Mittelbach and David Carlisle. This package corrects a few problems
% in the LaTeX2e kernel, the most notable of which is that in current
% LaTeX2e releases, the ordering of single and double column floats is not
% guaranteed to be preserved. Thus, an unpatched LaTeX2e can allow a
% single column figure to be placed prior to an earlier double column
% figure.
% Be aware that LaTeX2e kernels dated 2015 and later have fixltx2e.sty's
% corrections already built into the system in which case a warning will
% be issued if an attempt is made to load fixltx2e.sty as it is no longer
% needed.
% The latest version and documentation can be found at:
% http://www.ctan.org/pkg/fixltx2e


%\usepackage{stfloats}
% stfloats.sty was written by Sigitas Tolusis. This package gives LaTeX2e
% the ability to do double column floats at the bottom of the page as well
% as the top. (e.g., "\begin{figure*}[!b]" is not normally possible in
% LaTeX2e). It also provides a command:
%\fnbelowfloat
% to enable the placement of footnotes below bottom floats (the standard
% LaTeX2e kernel puts them above bottom floats). This is an invasive package
% which rewrites many portions of the LaTeX2e float routines. It may not work
% with other packages that modify the LaTeX2e float routines. The latest
% version and documentation can be obtained at:
% http://www.ctan.org/pkg/stfloats
% Do not use the stfloats baselinefloat ability as the IEEE does not allow
% \baselineskip to stretch. Authors submitting work to the IEEE should note
% that the IEEE rarely uses double column equations and that authors should try
% to avoid such use. Do not be tempted to use the cuted.sty or midfloat.sty
% packages (also by Sigitas Tolusis) as the IEEE does not format its papers in
% such ways.
% Do not attempt to use stfloats with fixltx2e as they are incompatible.
% Instead, use Morten Hogholm'a dblfloatfix which combines the features
% of both fixltx2e and stfloats:
%
% \usepackage{dblfloatfix}
% The latest version can be found at:
% http://www.ctan.org/pkg/dblfloatfix




%\ifCLASSOPTIONcaptionsoff
%  \usepackage[nomarkers]{endfloat}
% \let\MYoriglatexcaption\caption
% \renewcommand{\caption}[2][\relax]{\MYoriglatexcaption[#2]{#2}}
%\fi
% endfloat.sty was written by James Darrell McCauley, Jeff Goldberg and 
% Axel Sommerfeldt. This package may be useful when used in conjunction with 
% IEEEtran.cls'  captionsoff option. Some IEEE journals/societies require that
% submissions have lists of figures/tables at the end of the paper and that
% figures/tables without any captions are placed on a page by themselves at
% the end of the document. If needed, the draftcls IEEEtran class option or
% \CLASSINPUTbaselinestretch interface can be used to increase the line
% spacing as well. Be sure and use the nomarkers option of endfloat to
% prevent endfloat from "marking" where the figures would have been placed
% in the text. The two hack lines of code above are a slight modification of
% that suggested by in the endfloat docs (section 8.4.1) to ensure that
% the full captions always appear in the list of figures/tables - even if
% the user used the short optional argument of \caption[]{}.
% IEEE papers do not typically make use of \caption[]'s optional argument,
% so this should not be an issue. A similar trick can be used to disable
% captions of packages such as subfig.sty that lack options to turn off
% the subcaptions:
% For subfig.sty:
% \let\MYorigsubfloat\subfloat
% \renewcommand{\subfloat}[2][\relax]{\MYorigsubfloat[]{#2}}
% However, the above trick will not work if both optional arguments of
% the \subfloat command are used. Furthermore, there needs to be a
% description of each subfigure *somewhere* and endfloat does not add
% subfigure captions to its list of figures. Thus, the best approach is to
% avoid the use of subfigure captions (many IEEE journals avoid them anyway)
% and instead reference/explain all the subfigures within the main caption.
% The latest version of endfloat.sty and its documentation can obtained at:
% http://www.ctan.org/pkg/endfloat
%
% The IEEEtran \ifCLASSOPTIONcaptionsoff conditional can also be used
% later in the document, say, to conditionally put the References on a 
% page by themselves.




% *** PDF, URL AND HYPERLINK PACKAGES ***
%
%\usepackage{url}
% url.sty was written by Donald Arseneau. It provides better support for
% handling and breaking URLs. url.sty is already installed on most LaTeX
% systems. The latest version and documentation can be obtained at:
% http://www.ctan.org/pkg/url
% Basically, \url{my_url_here}.





% *** Do not adjust lengths that control margins, column widths, etc. ***
% *** Do not use packages that alter fonts (such as pslatex).         ***
% There should be no need to do such things with IEEEtran.cls V1.6 and later.
% (Unless specifically asked to do so by the journal or conference you plan
% to submit to, of course. )


% correct bad hyphenation here
\hyphenation{op-tical net-works semi-conduc-tor}


\usepackage{graphicx}
\usepackage{multirow}
\usepackage{amsmath,amssymb,amsfonts}
\usepackage{amsthm}
\usepackage{mathrsfs}
\usepackage[figuresright]{rotating}
%\usepackage[title]{appendix}
\usepackage{xcolor}
\usepackage{textcomp}
\usepackage{manyfoot}
\usepackage{booktabs}
\usepackage{algorithm}
\usepackage{algorithmicx}
\usepackage{algpseudocode}
\usepackage{program}
\usepackage{listings}
\usepackage{multirow}
\begin{document}
%
% paper title
% Titles are generally capitalized except for words such as a, an, and, as,
% at, but, by, for, in, nor, of, on, or, the, to and up, which are usually
% not capitalized unless they are the first or last word of the title.
% Linebreaks \\ can be used within to get better formatting as desired.
% Do not put math or special symbols in the title.
\title{Analysis of DNA sequences through local distribution of nucleotides in strategic neighborhoods}


\author{Probir~Mondal,~\IEEEmembership{}
        Pratyay~Banerjee ~\IEEEmembership{}
        and~Krishnendu~Basuli ~\IEEEmembership{}% <-this % stops a space
        
\IEEEcompsocitemizethanks{\IEEEcompsocthanksitem P. Mondal is with the Department
of Computer Science, P. R. Thakur Govt. College, West Bengal, India,
e-mail: probir.mondal@prtgc.ac.in.\protect
% note need leading \protect in front of \\ to get a newline within \thanks as
% \\ is fragile and will error, could use \hfil\break instead.

\IEEEcompsocthanksitem P. Banerjee (Corresponding author) is with the Department
of Physics, P. R. Thakur Govt. College, West Bengal, India,
e-mail: pratyay.banerjee@prtgc.ac.in.

\IEEEcompsocthanksitem K. Basuli is with the Department
of Computer Science, West Bengal State University, India,
e-mail: krishnendu.basuli@gmail.com.
}% <-this % stops an unwanted space
\thanks{}}




% note the % following the last \IEEEmembership and also \thanks - 
% these prevent an unwanted space from occurring between the last author name
% and the end of the author line. i.e., if you had this:
% 
% \author{....lastname \thanks{...} \thanks{...} }
%                     ^------------^------------^----Do not want these spaces!
%
% a space would be appended to the last name and could cause every name on that
% line to be shifted left slightly. This is one of those "LaTeX things". For
% instance, "\textbf{A} \textbf{B}" will typeset as "A B" not "AB". To get
% "AB" then you have to do: "\textbf{A}\textbf{B}"
% \thanks is no different in this regard, so shield the last } of each \thanks
% that ends a line with a % and do not let a space in before the next \thanks.
% Spaces after \IEEEmembership other than the last one are OK (and needed) as
% you are supposed to have spaces between the names. For what it is worth,
% this is a minor point as most people would not even notice if the said evil
% space somehow managed to creep in.



% The paper headers
%\markboth{}%
%{Shell \MakeLowercase{\textit{et al.}}: Analysis of DNA sequences through local distribution of nucleotides in strategic neighborhoods}
% The only time the second header will appear is for the odd numbered pages
% after the title page when using the twoside option.
% 
% *** Note that you probably will NOT want to include the author's ***
% *** name in the headers of peer review papers.                   ***
% You can use \ifCLASSOPTIONpeerreview for conditional compilation here if
% you desire.




% If you want to put a publisher's ID mark on the page you can do it like
% this:
%\IEEEpubid{0000--0000/00\$00.00~\copyright~2015 IEEE}
% Remember, if you use this you must call \IEEEpubidadjcol in the second
% column for its text to clear the IEEEpubid mark.



% use for special paper notices
%\IEEEspecialpapernotice{(Invited Paper)}




% make the title area
\maketitle
\thispagestyle{empty}
% As a general rule, do not put math, special symbols or citations
% in the abstract or keywords.
\begin{abstract}
We propose a new alignment free algorithm by constructing a compact vector representation on $\mathbb{R}^{24}$ of a DNA sequence of arbitrary length. Each component of this vector is obtained from a representative sequence, the elements of which are the values realized by a  function $\Gamma$. The function $\Gamma$, so defined, acts on neighborhoods of arbitrary radius that are located at strategic positions within the DNA sequence. $\Gamma$ carries complete information about the local distribution of frequencies of the nucleotides as a consequence of the  uniqueness of prime factorization of integer. The algorithm exhibits linear time complexity and turns out to be space efficient. The two natural parameters characterizing the radius and location of the neighbourhoods are fixed by comparing the phylogenetic tree with the benchmark for  full genome sequences of fish mtDNA datasets. Using these fitting parameters, the method is further applied to analyze a number of genome sequences from benchmark and other standard datasets.
\end{abstract}

% Note that keywords are not normally used for peerreview papers.
\begin{IEEEkeywords}
Alignment-free, Compact representation, Prime factorization, Time complexity.
\end{IEEEkeywords}






% For peer review papers, you can put extra information on the cover
% page as needed:
% \ifCLASSOPTIONpeerreview
% \begin{center} \bfseries EDICS Category: 3-BBND \end{center}
% \fi
%
% For peerreview papers, this IEEEtran command inserts a page break and
% creates the second title. It will be ignored for other modes.
\IEEEpeerreviewmaketitle



\section{Introduction}


Recent advancement in molecular biology stems primarily from the inclusion of Information Science and Technology into this field. So far, various computational tools have been developed and applied to analyze biological sequences. There exist algorithms to identify the origin of viruses, the quantum of similarity present among species, the mutation occurring inside them etc. The pioneering work in the field of sequence alignment was done by Needleman-Wunsch \cite{nid:wun} in 1970 followed by Smith-Waterman\cite{ smith:wat}. Following them, alignment-based searching tools, viz., BLAST\cite{al:miller, mark:keith}, FASTA \cite{fasta1,fasta2} etc. were developed.

Molecular biology continued to become increasingly interesting with the enormous growth of its database generated by various initiatives. Despite their accuracy, most of the alignment-based algorithms that have been implemented so far admit quadratic time complexity $\mathcal{O}(N^2)$ and, thus, do not seem to have proved useful for analyzing long sequences. Consequently, it seemed customary to develop time-efficient algorithms. A new variety of alignment free (AF) algorithms \cite{Vinga_38, Zielezinski_AF, Brian_doi_1} arose thereafter as an alternative description of biological sequences with a focus to represent them through concrete mathematical object amenable to further treatment. AF algorithms, in general, admit linear time complexity. Moreover, their ‘global’ nature is expected to play a vital role while comparing DNA sequences of unequal length.

A typical DNA sequence comprises of four nucleotide bases, viz., Adenine (A), Guanine (G), Cytosine (C) and Thymine (T). To compare two such sequences, there exist in the literature a large number of AF approaches that fall in two broad categories: viz., the word-based method \cite{word_based, Oliver_doi_2} which is based on the frequencies of subsequences of certain length and information-theory based method \cite{info_theory1,info_theory2} which quantifies the informational content between pair of sequences. In addition, there exist AF methods belonging to neither of these two categories. This includes, for example, techniques relying on the length of matching words viz., average common word\cite{avg_common}, shortest unique substring\cite{unique}, the minimal absent words between sequences \cite{minimal_absent}, iterated maps \cite{Almeida}, graphical representation \cite{randic1,zhang:zhang}, chaos game representation \cite{Chaos}: all in the interest to extract information about the distribution of nucleotides within sequences.


Here, in this article, we adopt a new AF approach. Given a DNA sequence, instead of looking at individual nucleotide,
we consider its corresponding course-grained form which provides a `compact' representation of the same in the sense that the length of its corresponding representative sequence is half or even smaller compared to that of the sequence itself. This is a clear indication that the practical running time of our algorithm will decrease.

The arrangement of the paper is as follows: In Sec. 2, we propose our algorithm in detail regarding the association of a scalar with a DNA sequence of arbitrary length and calculation of the euclidean metric $\rho$ between a pair of such sequences. In Sec. 3, we apply our algorithm on full genome sequence of fish mtDNA to fix the values of the two fitting parameters. To do so, we calculate the normalized Robinson-Foulds (nRF) distance of the phylogenetic tree obtained from our algorithm with respect to the standard tree in the benchmark datasets \cite{Benchmark}. To check the accuracy of the algorithm we run it on four other genome sequences from benchmark. In the next section, we explicitly demonstrate the efficiency of our method on six complete genome sequences by comparing running times with four well-known AF algorithms. Moreover, we show our performance graphically regarding running time and peak memory consumption on simulated datasets. In Sec. 5, we finally draw conclusion and state possible future direction.

\section{Construction of the representative vector}
Consider a typical string $\xi$ consisting of $N$ nucleotides. We denote such a string as $\xi=s_1 s_2 \cdots s_N$, where each $s_i$ is one of the four nucleotides, viz., A, C, G and T. Let P be the ordered set containing the first four prime numbers, i.e., P$=\{2,3,5,7\}$ and let $\sigma_i$ be the $i^{\text{th}}$ permutation of the set P. Clearly, we have $4!=24$ such permutations, viz., $\sigma_0$, $\sigma_1$, $\ldots$, $\sigma_{23}$; where we have chosen $\sigma_0$ to be the identity permutation.
%i.e., $\sigma_0(\alpha)=\alpha$ for any prime $\alpha \in P$.
Now we assign to each of the nucleotides, corresponding to a particular permutation $\sigma_i$ of the set $P$, the first four prime numbers as follows: $A=\sigma_i(1), C=\sigma_i(2), G=\sigma_i(3)$ and $T=\sigma_i(4)$, where $\sigma_i(j)$ is the $j^{\text{th}}$ element corresponding to the $i^{\text{th}}$ permutation of the set P. For example, in the case of identity permutation, we have $i=0$ and thus~ $A=\sigma_0(1)=2, C=\sigma_0(2)=3, G=\sigma_0(3)=5, T=\sigma_0(4)=7$.  Next, we define a $l$-neighbourhood $U_l(s_i)$ of the nucleotide $s_i$ with radius $l$ and centre $s_i$ to be the set $U_l(s_i)\equiv  \{s_{i-l}, \ldots, s_i, \ldots, s_{i+l} \}$. Let $\Gamma_j$ be the function associating with each neighbourhood $U_l(s_i)$, a positive integer $\Gamma_j(U_l(s_i))$ with respect to the permutation $\sigma_j$ through the following way:
\begin{equation}
\label{a1}
 \Gamma_j(U_l(s_i))=[\sigma_j(1)]^{f_1}\cdotp [\sigma_j(2)]^{f_2}  \cdotp[ \sigma_j(3)]^{f_3} \cdotp [\sigma_j(4)]^{f_4}
\end{equation}
where the exponents $f_1, f_2,f_3$ and $f_4$ are the frequencies of the four nucleotides A, C, G and T, respectively in the neighbourhood $U_l(s_i)$. In other words, the value of $\Gamma$ at a neighbourhood is simply the product of the prime numbers (PPN) assigned in a particular permutation to the nucleotides appearing in that neighbourhood. Note there is no nucleotide $s_i$ for $i 
< 1$ or $i > N$. Clearly, $0 \leq  f_r \leq 2l+1$ for $r\in \{1,2,3,4\}$. It is useful to note that the function $\Gamma_j(U_l(s_i))$ contains complete information about the frequencies of the nucleotides in the neighbourhood $U_l(s_i)$. This follows from the uniqueness of the prime factorization of a positive integer. However, it is observed that given the function $\Gamma_j(U_l(s_i))$, one can reconstruct the associated neighbourhood $U_l(s_i)$ with the exact ordering of nucleotides only in case the degeneracy (in the ordering of nucleotides) arising out of permutation is lifted depending upon the values of $\Gamma$ at the two adjacent neighbourhoods on either side of $U_l(s_i)$. This point is illustrated at the end of the present section.

\vspace{.1cm}

Let 

\begin{equation}
\small
\label{a2}
 \zeta=\bigg\{ \Gamma_j(U_l(s_1)), \Gamma_j(U_l(s_{t+2})), \Gamma_j(U_l(s_{2t+3})), \ldots, \Gamma_j(U_l(s_w))\bigg\}
\end{equation}
 be a sequence consisting of $n$ numbers 
 $\Gamma_j(U_l(s_i))$, where 
\begin{equation}
\label{a3}
 n=1+\left\lfloor \frac{N-1}{t+1}\right\rfloor
\end{equation}
and $w=(n-1)t+n$. Here $\lfloor z \rfloor$ denotes the integer part of $z$ and $t$ is the distance (i.e., number of nucleotides) between the centres of two successive neighbourhoods. In Sec. 3, for a given value of $l$, we shall set $1 \leq t \leq l$. Otherwise, two adjacent neighbourhoods will no longer overlap and may lead to loss of information. We claim the sequence $\zeta$ in Eq. (\ref{a2}) to be a compact representation of the string $\xi$ since from Eq. (\ref{a3}) we find $ n \lesssim \frac{N}{2}$. Next, our objective is to associate a scalar with the sequence (\ref{a2}) in a way such that the scalar is sufficiently sensitive under the transformation of string $\xi$ through point mutation, insertion and deletion. To this end, we propose the scalar $\eta_j$ (with respect to the permutation $\sigma_j$) associated with the sequence $\zeta$  by adding  all the entries of $\zeta$  as follows:
\begin{equation}
\label{a4}
 \eta_j=\sum\limits_{i=1}^n \Gamma_j(U_l(s_{i+t(i-1)}))
\end{equation}

In order not to give preference to any particular permutation 
we assign prime numbers to the nucleotides A, C, G and T corresponding to every permutation $\sigma_j$ of the set P. Thus, for every $\sigma_j$ we obtain a scalar $\eta_j$ from Eq. (\ref{a4}).  In this way a representative vector $\vec{\eta}=(\eta_0, \eta_1, \cdots, \eta_{23})$ is constructed corresponding to the string $\xi$. As each $\eta_j$ is real, we presume the vector $\vec{\eta} $ resides in the $24$-dimensional real euclidean space $\mathbb{R}^{24}$ endowed with the euclidean metric $\rho$. Thus two DNA sequences are compared by computing $\rho$ between the corresponding representative vectors.
 
As an example, let us consider a typical string of nucleotides
\begin{equation}
\label{a5}
 ACTGCCTCGATAA
\end{equation}
Here N=13. Choose, say, the identity permutation 
$\sigma_0$. Then it follows that A=2, C=3, G=5 and T=7. We choose the two parameters as $l=1$ and $t=1$. Thus we consider neighbourhoods with centre at every alternate nucleotide comprising only of nearest neighbours. The neighbourhoods for the string (\ref{a5}) (see Fig. 1) turns out to be $U_1(s_1)=\{A, C \}, U_1(s_3)=\{C, T, G \}, U_1(s_5)=\{G, C, C \}, U_1(s_7)=\{C, T, C \}, U_1(s_9)=\{C, G, A \}, U_1(s_{11})=\{A, T, A \}$ and $U_1(s_{13})=\{A, A \}$.

 \begin{figure}[h!]
 
\begin{center}

\includegraphics[keepaspectratio=false,width=7cm,height=2cm,scale=0.2,angle=0]{fig1}
\label{figure1}
 \end{center}
 \caption{\scriptsize A typical string of $13$ nucleotides. The collection of nucleotides lying inside an over/under brace forms a neighborhood with centre at the middle nucleotide and radius $l=1$. However, the  two extreme nucleotides are the centres of the neighborhoods $U_1(s_1)$ and $U_1(s_{13})$, respectively as $U_1(s_1)$ has no nucleotide on the left of its centre at $s_1$ and $U_1(s_{13})$ has no nucleotide on the right of $s_{13}$. The distance $t$, i.e., the number of nucleotides between successive neighbourhoods, is $1$. }
 \end{figure}

Note that the neighbourhoods at the two extreme positions naturally contain fewer nucleotides. Using Eq. (\ref{a1}), we calculate, say, the function $ \Gamma_0(U_1(s_5))$ with respect to the identity  permutation $\sigma_0$ as
$\Gamma_0(\{G,C,C\})=2^0 \cdotp 3^2 \cdotp 5^1 \cdotp 7^0=45$. From Eq. (\ref{a2}) we construct the sequence as 
\begin{equation}
\label{a6}
 \zeta=\bigg\{ 6,105,45,63,30,28,4 \bigg\}.
\end{equation}
 Using Eq. (\ref{a3}), we find $n=7$. Finally, from Eq. (\ref{a4}), the scalar $\eta_0$ associated with the string (\ref{a5}), which is simply the sum of the elements of the sequence (\ref{a6}), turns out to be $\eta_0=281$. As stated earlier, we call the sequence (\ref{a2}) to be a compact representation of the string $\xi$ as the former contains around half or even smaller number of elements compared to the latter. In addition, the uniqueness of prime factorisation of integer reproduces the local distribution of various nucleotides with exact frequencies. However, the invertibility of this representation depends heavily on the entries of the sequence (\ref{a2}). As an illustration, choose a subsequence $\zeta'=\{6,105\}$ of the sequence $\zeta$ in (\ref{a6}) by selecting the first two elements. As $6=2^1 \cdotp 3^1 \cdotp 5^0 \cdotp 7^0$, the corresponding neighbourhood, which is the inverse image of the function $\Gamma$,  is either \{A, C\} or \{C, A\}. Similarly for the second element of $\zeta'$, we find $105=2^0 \cdotp 3^1 \cdotp 5^1 \cdotp 7^1$, so that the corresponding neighbourhood is one of the 6 permutations of C, G and T, viz., \{C, G, T\} , \{C, T, G\}, \{G, C, T\}, \{T, C, G\}, \{G, T, C\} and \{T, G, C\}. Now, observe that the nucleotide A is absent in  all of the six neighbourhoods. Thus, looking at the overlap region, it is clear that \{A, C\} is the right choice of neighbourhood corresponding to the element 6 of $\zeta'$. Consequently,  \{C, G, T\} and \{C, T, G\} are the remaining possibilities as the inverse image of $\Gamma$ of the second element 105 of $\zeta'$. Thus from the subsequence $\zeta'$, we reproduce (up to a degeneracy in the ordering of the last two nucleotides) the corresponding string of nucleotides to be either $ACGT$ or $ACTG$. Notice that the example (\ref{a5}) that we have chosen here is perfectly reproducible following this procedure from its representative sequence (\ref{a6}).



\section{Similarity analysis}

In order to test the usefulness of our algorithm, proposed in Sec. 2 and denoted as PPN hereafter, we apply it on the benchmark dataset of fish mtDNA from AFproject \cite{Benchmark}. The corresponding phylogenetic tree obtained through PPN is shown in Fig. \ref{fish_mtDNA} by setting the parameters at $l=4$ and $t=1$. Upon comparison with the benchmark, we find that PPN ranks $11^{\text{th}}$ for fish mtDNA with nRF=0.64 while the normalized Quartet Distance  (nQD) is 0.2602. Although we find the same nRF value by implementing the Manhattan metric (in place of euclidean distance) between the representative vectors, the nQD value is 0.2723. However, the Cosine distance, in this case, do not produce expected result. Moreover, it is observed that PPN is not effective for $l > 4$; possibly because the fluctuation in the distribution of frequencies of different nucleotides in a neighborhood appears to be smaller with increasing radius.

\begin{figure}[h!]
\begin{center}
\includegraphics[width=8.5cm]{fish_mtDNA} %fbox to put boundary
\caption{{\scriptsize UPGMA tree for fish mtDNA sequences having 25 species drawn in software MEGA 11 with parameter values $l=4$, $t=1.$ }}
\label{fish_mtDNA}
\end{center}
\end{figure}  

\begin{figure}[h!]
\begin{center}
\label{mammals_own}
\includegraphics[width=8.5cm]{mammals_own} %fbox to put boundary
\caption{{\scriptsize UPGMA tree for mammals mtDNA sequences having 41 species drawn in software MEGA 11 with parameter values $l=4$, $t=1.$ }}
\end{center}
\end{figure} 
After optimizing the two fitting parameters at $l=4, t=1$, we apply PPN on the assembled 29 E. Coli/Shigella strain chosen from the Genome-based phylogeny (GBP) section and on Yersinia, plants and unsimulated 27 E. Coli/Shigella strain  from the Horizontal Gene Transfer (HGT) section of the benchmark datasets. We record the nRF, nQD and rank for each of these datasets in TABLE \ref{nrf_distance}  as obtained from AFproject \cite{Benchmark}.

\begin{table}[h]

\caption{The nRF, nQD and rank using PPN for five genome sequences. }
\centering
\scriptsize
\label{nrf_distance}

\begin{tabular}{|l|c|c|c|}
\hline
\multicolumn{1}{|c|}{\textbf{Genome sequence}} & \textbf{nRF} & \textbf{nQD} & \textbf{Rank} \\ \hline
Fish mtDNA GBP                                 & 0.64         & 0.2602       & 11            \\ \hline
Unsimulated 27 E.Coli/Shigella strain HGT      & 0.79         & 0.4369       & 15            \\ \hline
Assembled 29 E.Coli/Shigella strain GBP        & 0.73         & 0.3337       & 16            \\ \hline
Plants HGT                                     & 0.73         & 0.3077       & 9             \\ \hline
Yersinia HGT                                   & 0.80         & 0.6043       & 5             \\ \hline
\end{tabular}
\end{table}

In addition, using the datasets from ref. \cite{fast_vector}, we generate the phylogenetic tree shown in Fig. \ref{mammals_own} through PPN for the mammals mtDNA. The corresponding nRF distances of this tree is computed (using ETE3 toolkit package) from each of the trees obtained through CD-MAWS\cite{CD_MAWS}, PASTA\cite{PASTA}, Skmer\cite{skmer} and Mash\cite{Mash}. They are  0.6, 0.57, 0.6 and 0.57, respectively. Thus, PPN shows an 'overall' agreement with the top methods at the cost of some accuracy. Note that PPN fails to place the cat at the 'right' location in the phylogeny. Implementation of our algorithm is available  at https://github.com/probir-mondal/PPN.

\section{Performance analysis }
In this section we demonstrate the efficiency of PPN proposed in Sec. 2 by comparing the running time against standard AF tools. We have conducted the computation on a machine with Intel(R) Core(TM) i7-8700 CPU @ 3.20 GHz x 12, 8 GB RAM and driven by Ubuntu 22.04 OS. The running times taken by PPN for complete genome sequences of mammals, influenza A virus, rhinovirus, ebolavirus, coronavirus and bacteria are shown in the first row of TABLE \ref{tab:Running_time}. The running time of four other standard algorithms viz., CD-MAWS, Co-Phylog\cite{co_phylog}, KSNP-3\cite{kSNP3} and Skmer  (applied on the same sequences) are reproduced from ref. \cite{CD_MAWS} in the next four rows of the TABLE  \ref{tab:Running_time}. It is apparent from this comparison that PPN,  which admits a linear time-complexity, is, indeed, fast. Note the value of the parameter $t$ provides an estimate of the extent one  can compactify a given DNA sequence. For example, even for $t=1$, the length $n$ in Eq. (\ref{a3}) of the sequence $\zeta$ is approximately half that of the original DNA sequence $\xi$.



\begin{table}[!ht]
\caption{Running time comparison of six complete genome sequences}
\centering

 \label{tab:Running_time}
  \begin{center}
   
 \scalebox{0.85}{
\begin{tabular}{|l|cccccr|}
\hline
\multicolumn{1}{|c|}{\multirow{2}{*}{\textbf{Method}}} & \multicolumn{6}{c|}{\textbf{Running time in minutes}}                                                                                                            \\ \cline{2-7} 
\multicolumn{1}{|c|}{}                                 & \multicolumn{1}{c|}{Mammals} & \multicolumn{1}{c|}{Influenza} & \multicolumn{1}{c|}{Rhino} & \multicolumn{1}{c|}{Ebola} & \multicolumn{1}{c|}{Corona} & Bacteria \\ \hline
PPN                                                    & \multicolumn{1}{c|}{0.048}   & \multicolumn{1}{c|}{0.005}     & \multicolumn{1}{c|}{0.064} & \multicolumn{1}{c|}{0.082} & \multicolumn{1}{c|}{0.065}  & 16.105   \\ \hline
CD-MAWS                                                & \multicolumn{1}{c|}{0.181}   & \multicolumn{1}{c|}{0.008}     & \multicolumn{1}{c|}{0.200} & \multicolumn{1}{c|}{0.076} & \multicolumn{1}{c|}{0.088}  & 3.318    \\ \hline
Co-Phylog                                              & \multicolumn{1}{c|}{0.055}   & \multicolumn{1}{c|}{ERR}       & \multicolumn{1}{c|}{0.157} & \multicolumn{1}{c|}{0.076} & \multicolumn{1}{c|}{ERR}    & 11.483   \\ \hline
kSNP3                                                  & \multicolumn{1}{c|}{1.133}   & \multicolumn{1}{c|}{0.500}     & \multicolumn{1}{c|}{7.683} & \multicolumn{1}{c|}{1.416} & \multicolumn{1}{c|}{0.867}  & 1.716    \\ \hline
Skmer                                                  & \multicolumn{1}{c|}{0.035}   & \multicolumn{1}{c|}{ERR}       & \multicolumn{1}{c|}{0.203} & \multicolumn{1}{c|}{0.061} & \multicolumn{1}{c|}{0.050}  & 0.608    \\ \hline
\end{tabular}}\\
 \end{center}
ERR: Shows an error.
\end{table}




\begin{figure}[!ht]
\begin{center}
\includegraphics[width=\linewidth]{running_time}\\ %fbox to put boundary
\caption{{\scriptsize Average running time (in second) of PPN for simulated datasets with number of species ranging from 100 to 900. Each species contains 10,000 nucleotides.}}
\label{running_time}
\end{center}
\end{figure}  


\begin{figure}[ht]
\begin{center}
\includegraphics[width=\linewidth]{PeakMemoryConsumption}\\ %fbox to put boundary
\caption{{\scriptsize Peak memory consumption in MB by PPN for simulated datasets with number of species varying between 100 and 900. Here, each species contains 10,000 nucleotides.}}
\label{PeakMemoryConsumption}
\end{center}
\end{figure}  



Additionally, we run PPN on simulated datasets where each sequence representing a particular species contains 10,000 nucleotides while the number of such species range from 100 to 900. The average running time (Fig. \ref{running_time}) ) is calculated by taking the average of 10 such runs conducted on a machine with Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz x 4, 6 GB RAM and Ubuntu 22.04 OS. Similarly, the peak memory consumption (Fig. \ref{PeakMemoryConsumption}) is recorded by choosing the maximum value from 10 trials.

Moreover, we apply PPN for comparing one chromosome of Zea mays (GCF\_000005005.2) having 30,70,41,717 nucleotides with that of Oryza sativa (GCF\_001433935.1) having 4,32,70,923 nucleotides. Although these two sequences differ drastically in size, PPN determines the distance between them in 33.68 minutes and consumes maximum 20.24 GB memory with Intel(R) Xeon(R) GOLD 6134 CPU @3.20GHz x 16, 64GB RAM and Ubuntu 22.04 OS.


\section{Conclusion}
\vspace{-0.05cm}

In this article, we have adopted a new alignment-free approach to trace the amount of similarity/dissimilarity present within a pair of DNA sequences. PPN associates a scalar in a simple way to the representative sequence $\zeta$ corresponding to the string $\xi$. The problem regarding comparing sequences of unequal length is circumvented through the construction of the said scalar. The significant reduction in running time has occurred in two steps. As a first step, we have considered the course-grained form of the original DNA sequence  $\xi$ to reduce its length to a considerable extent. Further reduction happens due to the asymptotically linear nature of the time complexity arising in our algorithm. Comparison with standard AF algorithms clearly reveals that PPN is a reasonably fast algorithm and consumes significantly small memory. Thus, in spite of the 'limited' accuracy that PPN has exhibited so far, we believe that it will play a leading role specially in the case of analyzing very long  sequences with unknown characteristics where the promise of the existing algorithms turns out to be insufficient.

\begin{thebibliography}{20}
%1
\bibitem{nid:wun}
S. B. Needleman and C. D. Wunsch, "A general method applicable to the search for similarities in the amino acid sequence of two proteins", \textit{J. Mol. Biol.}, vol. 48, pp. 443–453, 1970.

%2
\bibitem{smith:wat}
T. F. Smith and M. S. Waterman, "Identification of common molecular subsequences", \textit{J. Mol. Biol.}, vol. 147, no. 1, pp. 195–197, 1981.

%3
\bibitem{al:miller}
S. F. Altschul, W. Gish, W. Miller, E. W. Myers and D. J. Lipman, "Basic local alignment search tool", \textit{J. Mol. Biol.}, vol. 215, pp. 403–410, 1990.

%4
\bibitem{mark:keith}
K. J. Menlove, M. Clement and K. A. Crandall, "Similarity searching using BLAST", \textit{Methods Mol. Biol.}, vol. 537, pp. 1-22,    2009.

\bibitem{fasta1}
W. R. Pearson, "Rapid and sensitive sequence comparison with FASTP and FASTA", \textit{Methods Enzymol.}, vol. 183, pp. 63-98, 1990.
%5
\bibitem{fasta2}
W. R. Pearson and D. J. Lipman, "Improved tools for biological sequence comparison", \textit{Proc. Natl. Acad. Sci. USA}, vol. 85, pp. 2444-2448, 1988.



\bibitem{Vinga_38}
 S. Vinga and J. Almeida, "Alignment-free sequence comparison - a review", \textit{Bioinformatics}, vol. 19, pp. 513–523, 2003.



\bibitem{Zielezinski_AF}
 A. Zielezinski, S. Vinga, J. Almeida and W. M. Karlowski, "Alignment-free sequence comparison: benefits, applications, and tools", \textit{Genome Biol.}, vol. 18, Article no. 186, 2017.

%Oct 3;18(1):186. doi: 10.1186/s13059-017-1319-7. PMID: 28974235; PMCID: PMC5627421.

\bibitem{Brian_doi_1}
 B. B. Luczak, B. T. James and H. Z. Girgis, "A survey and evaluations of histogram-based statistics in alignment-free sequence comparison", \textit{Brief. Bioinform.}, vol. 20(4), pp. 1222–1237, 2019.

%https://doi.org/10.1093/bib/bbx161

\bibitem{word_based}
 B. E. Blaisdell, "Average values of a dissimilarity measure not requiring sequence alignment are twice the averages of conventional mismatch counts requiring sequence alignment for a computer-generated model system", \textit{J. Mol. Evol.}, vol. 29(6), pp. 538–547, 1989.


\bibitem{Oliver_doi_2}
 O. Bonham-Carter, J. Steele and D. Bastola, "Alignment-free genetic sequence comparisons: a review of recent approaches by word analysis",  \textit{Brief. Bioinform.}, vol. 15, pp. 890–905, 2013.

\bibitem{info_theory1}
 S. Vinga, "Information theory applications for biological sequence analysis", \textit{Brief. Bioinform.}, vol. 15, pp. 376–389,
 2014.
 
\bibitem{info_theory2} 
 Y. Gao and L. Luo, "Genome-based phylogeny of dsDNA viruses by a novel alignment-free method", \textit{Gene}, vol. 492(1), 309–314, 2012.

\bibitem{avg_common}
 I. Ulitsky, D. Burstein, T. Tuller and B. Chor, "The average common substring approach to phylogenomic reconstruction", \textit{J. Comput. Biol.}, vol. 13, pp. 336-350, 2006.

\bibitem{unique}
 B. Haubold, N. Pierstorff, F. Möller and T. Wiehe, "Genome comparison without alignment using shortest unique substrings", \textit{BMC Bioinformatics}, vol. 6, Article no. 123, 2005.

\bibitem{minimal_absent}
A. J. Pinho, P. JSG Ferreira, S. P. Garcia and J. MOS Rodrigues, "On finding minimal absent words", \textit{BMC Bioinformatics}, 
 vol. 10, Article no. 137, 2009.


\bibitem{Almeida}
 J. S. Almeida, "Sequence analysis by iterated maps, a review", \textit{Brief Bioinform.}, vol. 15(3), pp. 369-375, 2014.


\bibitem{randic1}
M. Randić, M. Vraćko, N. Lerś and P. Dejan, "Analysis of similarity/dissimilarity of DNA sequences based on novel 2-D graphical representation", \textit{Chem. Phys. Lett.}, vol. 371, pp. 202-207, 2003.


\bibitem{zhang:zhang}
 Z. Zhang, S. Wang, X. Zhang and Z. Zhang, "Similarity analysis of DNA sequences based on a compact representation", \textit{IEEE  Fifth International Conference on Bio-Inspired Computing: Theories and Applications (BIC-TA).}, pp. 1143-1146, 2010.



\bibitem{Chaos}
 H. J. Jeffrey, "Chaos game representation of gene structure", \textit{Nucleic Acids Res.}, vol. 18(8), pp. 2163–2170, 1990.


 \bibitem{Benchmark}
  A. Zielezinski, H. Z. Girgis, G. Bernard \textit{et al.}, "Benchmarking of alignment-free sequence comparison methods", \textit{Genome Biol.}, vol. 20, Article no. 144, 2019.

\bibitem{fast_vector} Y. Li, L. He, R. L. He \textit{et al.}, “A novel fast vector method for genetic sequence comparison”, 
\textit{Sci. Rep.}, vol. 7, Article no. 12226, 2017.

\bibitem{CD_MAWS}

N. Anjum, R. L. Nabil, R. I. Rafi, Md. S. Bayzid and M. S. Rahman, "CD-MAWS: An alignment-free phylogeny estimation method using cosine distance on minimal absent word sets. \textit{IEEE/ACM Trans. Comput. Biol. Bioinform.} vol. 20(1), pp. 196-205, 2023.

\bibitem{PASTA} S. Mirarab, N. Nguyen, S. Guo, L. Wang, J. Kim and T. Warnow, “PASTA: Ultra-large multiple sequence alignment for nucleotide and amino-acid sequences”, \textit{J. Comput. Biol.}, vol. 22, no. 5, pp. 377–386, 2015.

\bibitem{skmer}
 S. Sarmashghi, K. Bohmann, M. T. P. Gilbert \textit{et al.}, "Skmer: assembly-free and alignment-free sample identification using genome skims", \textit{Genome Biol.}, vol. 20, Article no. 34, 2019.

 


 \bibitem{Mash}
  B. D. Ondov, T. J. Treangen, P. Melsted \textit{et al.}, "Mash: fast genome and metagenome distance estimation using MinHash", \textit{Genome Biol.}, vol. 17, Article no. 132, 2016.

\bibitem{co_phylog}
 H. Yi and L. Jin, "Co-phylog: an assembly-free phylogenomic approach for closely related organisms", \textit{Nucleic Acids Res.}, vol. 41(7), pp. e75, 2013. 

 
\bibitem{kSNP3}S. N. Gardner, T. Slezak and B. G. Hall, “kSNP3.0: SNP detection and phylogenetic analysis of genomes without genome alignment or reference genome”, \textit{Bioinform.}, vol. 31, no. 17, pp. 2877–2878, 2015.

\end{thebibliography}

% biography section
% 
% If you have an EPS/PDF photo (graphicx package needed) extra braces are
% needed around the contents of the optional argument to biography to prevent
% the LaTeX parser from getting confused when it sees the complicated
% \includegraphics command within an optional argument. (You could create
% your own custom macro containing the \includegraphics command to make things
% simpler here.)
%\begin{IEEEbiography}[{\includegraphics[width=1in,height=1.25in,clip,keepaspectratio]{mshell}}]{Michael Shell}
% or if you just want to reserve a space for a photo:

% \begin{IEEEbiography}{Michael Shell}
% Biography text here.
% \end{IEEEbiography}

% % if you will not have a photo at all:
% \begin{IEEEbiographynophoto}{John Doe}
% Biography text here.
% \end{IEEEbiographynophoto}

% % insert where needed to balance the two columns on the last page with
% % biographies
% %\newpage

% \begin{IEEEbiographynophoto}{Jane Doe}
% Biography text here.
% \end{IEEEbiographynophoto}

% You can push biographies down or up by placing
% a \vfill before or after them. The appropriate
% use of \vfill depends on what kind of text is
% on the last page and whether or not the columns
% are being equalized.

%\vfill

% Can be used to pull up biographies so that the bottom of the last one
% is flush with the other column.
%\enlargethispage{-5in}



% that's all folks
\end{document}


