\documentclass[aps,superscriptaddress]{revtex4-2}
%\documentclass[prx,aps,superscriptaddress]{revtex4}
\usepackage{floatrow}
\usepackage{epsfig,wrapfig}
\usepackage{amssymb,amsfonts,amsmath}
\usepackage{afterpage}
\usepackage{soul}

\usepackage{bm}% bold math
\usepackage{siunitx}
\usepackage{xcolor}

\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage[francais]{babel}

\pagenumbering{gobble} 

\newcommand{\be}{\begin{equation}}
\newcommand{\ee}{\end{equation}}
\newcommand{\bea}{\begin{eqnarray}}
\newcommand{\eea}{\end{eqnarray}}

\newcommand{\orange}{\color{orange}}
\newcommand{\blue}{\color{blue}}
\newcommand{\red}{\color{red}}
\newcommand{\cbl}{\color{black}}

\newcommand{\Ecoli}{{\it Escherichia coli}}
\newcommand{\ecoli}{{{\it E.~coli}}}
\newcommand{\Bsub}{{\it Bacillus subtilis}}
\newcommand{\bsub}{{\it B.~subtilis}}
\newcommand{\Ccres}{{\it Caulobacter crescentus}}
\newcommand{\ccres}{{\it C.~crescentus}}
\newcommand{\Pseudo}{{\it Pseudomonas aeruginosa}}
\newcommand{\pseudo}{{\it P.~aeruginosa}}

\newcommand{\Lk}{\text{Lk}}
\newcommand{\Tw}{\text{Tw}}
\newcommand{\Wr}{\text{Wr}}
\newcommand{\s}{\sigma}


\begin{document}

\title{DNA supercoiling in bacteria: state of play and challenges from a modeling viewpoint}

\author{Ivan Junier}
\email{ivan.junier@univ-grenoble-alpes.fr}
\affiliation{Univ. Grenoble Alpes, CNRS, UMR 5525, VetAgro Sup, Grenoble INP, TIMC, 38000 Grenoble, France}
\author{Elham Ghobadpour}
\affiliation{Univ. Grenoble Alpes, CNRS, UMR 5525, VetAgro Sup, Grenoble INP, TIMC, 38000 Grenoble, France}
\affiliation{Université de Lyon, École Normale Supérieure (ENS) de Lyon, CNRS, Laboratoire de Physique and Centre Blaise Pascal de l’ENS de Lyon, F-69342 Lyon, France}
\author{Olivier Espeli} 
\affiliation{Center for Interdisciplinary Research in Biology (CIRB), Coll\`ege de France, CNRS, INSERM, Universit\'e PSL, Paris, France}
\author{Ralf Everaers}
\affiliation{Université de Lyon, École Normale Supérieure (ENS) de Lyon, CNRS, Laboratoire de Physique and Centre Blaise Pascal de l’ENS de Lyon, F-69342 Lyon, France}

\keywords{DNA supercoiling $|$ Bacterial DNA $|$ Physical modeling} 


\begin{abstract} 
DNA supercoiling is central to fundamental processes of living organisms. Its average level along the chromosome and over time reflects the dynamic equilibrium of opposite activities of topoisomerases, which are required to relax mechanical stresses that are inevitably produced during DNA replication and gene transcription. Supercoiling affects all scales of the spatio-temporal organization of bacterial DNA, from the base pair to the large scale chromosome conformation. Highlighted {\it in vitro} and {\it in vivo} in the 1960s and 1970s, respectively, the first physical models were proposed concomitantly in order to predict the deformation properties of the double helix. About fifteen years later, polymer physics models demonstrated on larger scales the plectonemic nature and the tree-like organization of supercoiled DNA. Since then, many works %based on equilibrium and out-of-equilibrium statistical mechanics
have tried to establish a better understanding of the multiple structuring and physiological properties of bacterial DNA in thermodynamic equilibrium and out of equilibrium. The purpose of this essay is to discuss upcoming challenges by discussing the relevance, the predictive capacity and the limitations of current physical models, focusing on structural properties above the scale of the double helix. We discuss more particularly four fundamental aspects: gene transcription, DNA replication, nucleoid formation and the large-scale structure of chromosomes. The review being intended for both biologists and physicists, we have tried to reduce the respective jargon to a minimum.	
\end{abstract}

\maketitle 

With respect to DNA, efficient growth and division of bacteria rely on two major processes: (i) an appropriate expression of the genetic program allowing the generation in the right amounts and proportions of the proteins and enzymes necessary for the duplication of cells; (ii) a faithful replication of DNA and a reliable segregation of the replicated chromosomes during cell division. %A large part of modeling issues of the bacterial genome therefore concern the relationship between DNA structure and gene expression, on the one hand, and mechanisms associated with the correct replication and cellular distribution of chromosomes, on the other hand.
Research over the last fifty years or so has shown that the analysis of the topological constraints inherent in the double-helix nature of DNA is crucial for a quantitative understanding of these problems. Topological constraints are more particularly responsible for the supercoiling of bacterial DNA~\cite{wang_dna_1983} and are known to impact all levels of chromosome structure, from the base pair to its large scale conformation. In this context, we aim to review supercoiling-based models developed to explain {\it in vivo} properties of bacterial DNA~\cite{badrinarayanan_bacterial_2015,dorman_dna_2016,dame_chromosome_2020,lioy_multiscale_2021} occurring above the double helix scale.
%Models that deal with a finer resolution, which are useful to precisely investigate sequence effects (see e.g.~\cite{pyne_base-pair_2021} for a recent work), are reviewed in~\cite{manghi_physics_2016}.

A dramatic acceleration in the production of experimental results has, in fact, been at work since the 2000s as a consequence of low-cost DNA sequencing, new genome engineering techniques and the development of visualization methods of increasing resolution. One of the consequences of this access to comprehensive data, some of which, like high-throughput chromosome conformation capture (Hi-C) data~\cite{lieberman-aiden_comprehensive_2009}, cover almost all the scales of a chromosome~\cite{lieberman-aiden_comprehensive_2009,le_new_2014}, is the possibility to build data-driven ``models'' of chromosomal organization~\cite{rosa_computational_2014,junier_demultiplexing_2015,imakaev_modeling_2015}. This type of approach is particularly useful from a visualization perspective, i.e., a {\it three-dimensional
 representation} of two-dimensional data like
those produced by Hi-C methods.
%the basis for a batter exploration which has employed 
%i.e., a three-dimensional representation of two-dimensional data like those produced by Hi-C methods.
%By including the notion of entropy, it can be properly formalized within the framework of polymer physics, as e.g.~recently implemented in bacteria~\cite{messelink_learning_2021}. 
In order to avoid an often tacit, yet misleading association with the notion of ``chromosome model'' used in this context,
%i.e., of a three-dimensional representation of two-dimensional data like those produced by Hi-C methods,
we recall here the physical meaning — adopted in this review — of the word {\it model}: an idealized mathematical representation of a certain reality (physical, chemical and/or biological) including a set of parameters, whose emergent behavior is studied within the framework of the fundamental laws of physics. In the case of DNA, the employed models often come from the neighboring fields of polymer physics and of soft and active matter~\cite{marko_biophysics_2015}. %They are then studied in appropriate equilibrium and non-equilibrium statistical mechanics frameworks.

%In the case of DNA behavior, these models are often issued from polymer physics and the framework of thermodynamics or active processes, depending on the focus of the study -- equilibrium or non-equilibrium properties.

The review is divided into six sections. In section~\ref{sec:supercoiling}, we revisit essential concepts of DNA topology, we introduce the molecular machines central to the problem and we discuss {\it in vivo} measurements.  In section~\ref{sec:thermo}, we present the main steps marking the development of models aiming at capturing the {\it equilibrium properties} of supercoiled DNA, with a discussion of their relevance for {\it in vivo} situations. In the two subsequent sections we focus on transcription (section~\ref{sec:transcription}) and replication (section~\ref{sec:replication}), stressing the necessity to build {\it non-equilibrium models} that involve not only the transcription and replication machineries but also the action of topoisomerases. In section~\ref{sec:nucleoid}, we discuss the formation of the nucleoid, i.e., the membrane-free region of the bacterial cells where DNA is found. In the final section~\ref{sec:scaleup}, we review the attempts to model the structuring of bacterial chromosomes at the largest scales. %In the discussion, we summarize the progress made so far and delineate the challenges to overcome to obtain realistic, physics-based models of bacterial DNA.

\section{DNA supercoiling: connecting the multiple scales of a chromosome}
\label{sec:supercoiling}

\subsection{Fundamental concepts of supercoiling DNA}

The supercoiling level of a DNA circle indicates the {\it average} over- or under-winding of the double helix compared to the average winding of the double helix if the DNA circle was in a torsionally relaxed state. It is quantified by the so-called supercoiling density is given by $\s=\frac{n_0}{n_\s}-1$  with: $n_0\simeq10.5$ the number of base pairs of the relaxed B-DNA helix; $n_\s$ the theoretical average number of base pairs of the helix if the molecule were linearized and fully stretched while preventing the ends from rotating. Negative (positive) DNA supercoiling is thus characterized by $n_\s>n_0$ ($n_\s<n_0$) such that the corresponding double helix contains more (less) base pairs than that of B-DNA and, hence, is longer (shorter) (Fig.~\ref{fig:linking}).

Supercoiling induces mechanical stress in DNA as it affects, from a thermodynamic viewpoint, the most favorable geometry of base pairing associated with B-DNA. This stress is relaxed at multiple scales under the effect of a topological constraint that is specific to any system of concatenated curves such as the two strands of a circular DNA molecule (Fig.~\ref{fig:linking}). Specifically, for two interlacing strands as those of a DNA circle, their so-called linking number is equal to the number of times the two strands cross itself in space. Remarkably, for any continuous deformation of such a DNA circle, the linking number is unchanged~\cite{calugareanu_lintegrale_1959, white_self-linking_1969}. This {\it conservation law} implies two important properties for a DNA molecule. First, the supercoiling density $\s$ is constant unless DNA is cut by (dedicated) enzymes such as topoisomerases (see below). Second, any corresponding conformation of the molecule verifies  $t+w=C^{te}=\frac{1}{n_\s}$, where $t$, the average twist per base pair, or average helicity, reflects the number of times the two strands cross each other following the main axis of the double helix, and $w$, the writhe per base pair, reflects the number of times the main axis crosses itself~\cite{fuller_decomposition_1978}. More specifically, the average helicity $t$ is equal to $1/n$ with $n$ the average number of base pairs per helix when moving along the main axis of the molecule. Conservation of the linking number therefore implies $w=\frac{1}{n_\s}-\frac{1}{n}$. When DNA is supercoiled ($n_\s \neq n_0$), the writhe $w$ may thus become large as minimal torsional stress is obtained for $n=n_0$. As a result, braided super-structures such as  super-helices, also known as plectonemes, may spontaneously form (Fig.~\ref{fig:linking}). Note that by convention, a positive (negative) value of the twist corresponds to a right-handed (left-handed) helix. In this context, a positive (negative) value of the writhe corresponds, instead, to a left-handed (right-handed) super-helix (Fig.~\ref{fig:linking}).

\begin{figure}[b]
\floatbox[{\capbeside\thisfloatsetup{capbesideposition={right,top},capbesidewidth=0.5\linewidth}}]{figure}[\FBwidth]
{\caption{Cartoon to apprehend the conservation of the linking number of a circular DNA molecule consisting of 20 base pairs (indicated in black). For the sake of graphical representation, we have considered the case of a relaxed double helix composed of only four base pairs. {\it Upper left part}: the relaxed molecule has $n_0=5$ helices and a twist of $t=5/20=1/4=0.25$. The writhe $w=0$ (planar molecule) and $t+w=0.25$. {\it Rest of the figure}: we removed one helix from the molecule in the upper left panel to obtain $n_\s=4$, which corresponds to a negative supercoiling $\s=\frac{n_0}{n_\s}-1=-0.2$. The resulting molecule (top right) therefore has four helices with a half-helix containing six base pairs (likely to denaturate in real situations). In this case, the average number of base pairs per helix becomes $20/4=5$, so that $t=1/5=0.2$ and $t+w=0.2$. Two prototypical possibilities for conserving the value of the linking number can then occur: i) the base pairs redistribute uniformly along the molecule (bottom right conformation); ii) the molecule buckles to form a right-handed crossing (bottom left conformation): the writhe varies by $-1$ and the molecule recovers its average number of base pairs per helix at rest (four). One can then check that $t+w$ remains equal to $0.2$. The numbers in the yellow boxes indicate the corresponding crossings between the two strands of the DNA molecule, and the green circles indicate the points where a new cross is created, leading to an additional helix.}\label{fig:linking}}
{\includegraphics[width=\linewidth]{linking.pdf}}
\end{figure}

\subsection{Topoisomerases: defining the average supercoiling level...}

Bacteria have evolved enzymes known as topoisomerases that can change the linking number of DNA by adding supercoils into it~\cite{forterre_origin_2007,mckie_dna_2021}. These supercoils can be helices (type I topoisomerases) or super-helices (type~II), depending on whether the enzymatic reaction involves the cut of a single strand or of the full helix, in which case the topoisomerase makes another helix pass trough the cut~\cite{gellert_dna_1976}. Importantly, the {\it average} supercoiling level of bacterial DNA results from the {\it overall activity} of these topoisomerases. Evidence in various bacteria~\cite{drlica_control_1992,rovinskiy_rates_2012} points more particularly on a major role of Topo~I and DNA gyrase, which are type~I and type~II topoisomerases adding positive and negative supercoils, respectively. These two topoisomerases play a crucial role in gene transcription (section~\ref{sec:transcription}). In addition, DNA gyrase is involved in DNA replication together with Topo~IV (type~II) and Topo~III (type~I). Compared to Topo~I and DNA gyrase, the main activity of the latter is to resolve the topological interlacing of replicated DNA molecules (section~\ref{sec:replication}). While additional topoisomerases exist~\cite{forterre_origin_2007,mckie_dna_2021}, DNA gyrase, Topo~I, Topo~III and Topo~IV are believed to play a major role in most mesophilic bacteria, that is, in bacteria living under mild conditions of temperature, pressure and pH.

\subsection{...by relaxing transient torsional stresses produced by RNA and DNA polymerases}

Most of the time, topoisomerases do not act indiscriminately along the chromosome. Instead, they are used by the cell to specifically relax torsional stresses that are produced during gene transcription and DNA replication. Namely, in both situations, associated macromolecular complexes including the RNA and DNA polymerases (i) locally open bacterial DNA and (ii) proceed along it in a specific direction. Multiple protein complexes are bound to this DNA, and the expected situation {\it in vivo} is that of a chromosome organized into DNA domains whose ends are prevented from rotating by topological barriers~\cite{liu_supercoiling_1987} (see section~\ref{sec:transcription} for details). Consider, in this case, a piece of DNA such that the Watson and Crick strands of the double helix are held in a rotationally fixed position at the borders (Fig.\ref{fig:twin_scheme}A). Just as in a circular molecule, these constraints impose the conservation of the linking number between the two strands. Consider, then, an idealized machine locally opening the DNA and advancing along it (Fig.\ref{fig:twin_scheme}BC).
%The consequences depend on whether the machine can freely rotate or not (Figs.\ref{fig:twin_scheme}B and C), 
%or more generally on whether the machine and the DNA can freely rotate relative to each other.
To the extent that the local opening is associated with a local unwinding of the strands (not represented in Figs.\ref{fig:twin_scheme} for clarity), the conservation of the linking number implies that the remaining double helical parts have to overwind in compensation. Next, the torsional stresses induced by the progressing machine depend on whether it can freely rotate around the DNA  (Figs.\ref{fig:twin_scheme}C) or not (Figs.\ref{fig:twin_scheme}B). 
% and guides each strand through a specific point. 
In the former case, the machine rotates clockwise while advancing along the right-handed DNA double helix and no additional torsional stresses are exerted beyond those due to the initial opening. In the latter case, the double helix becomes increasingly overwound downstream and underwound upstream. This means that the number of base pairs per turn decreases or increases correspondingly. The progression thus induces respectively positive downstream and negative upstream {\it twin} DNA supercoiling~\cite{liu_supercoiling_1987}, although {\it no net overall supercoiling} has been introduced. As discussed extensively in this review, the {\it in vivo} situation is expected to lie between these two extreme cases. As a consequence, upstream replicated DNA molecules tend to form super-helices (section~\ref{sec:precat}). In addition, for both DNA replication and gene transcription, the resulting DNA supercoiling produced on each side of the RNA and DNA polymerases, respectively, %generates mechanical stress on the DNA and
exerts a restoring torque on the complex itself, hindering its progression. Topoisomerases are used by the cell to relax these torsional stresses.
%which facilitates the progression of DNA and RNA polymerases. 
In doing so, {\it they, and only they, change the average level} of supercoiling.

\begin{figure}
\centering
\includegraphics[width=0.75\linewidth]{twin_scheme.pdf}
\caption{Schematic representation of torsional stresses generated during translocation of a DNA unwinding machine. Four base pairs (vertical lines) are indicated to facilitate reading. A) The DNA ends are prevented from rotating, mimicking the effect of a topological barrier. Two extreme possibilities can then be considered: B) If the unwinding machine does not rotate around the DNA, the double helix becomes increasingly overwound downstream and underwound upstream, respectively generating positive and negative supercoiling. The latter can lead to DNA denaturation, as indicated by the breaking of the base pair. C) If the unwinding machine freely rotates around the DNA, the machine rotates clockwise while advancing along the undeformed right-handed DNA double helix.}
\label{fig:twin_scheme}
\end{figure}


\subsection{{\it In vivo} measured supercoiling and expected structural consequences}

The level of DNA supercoiling in bacteria is not measured on chromosomes, but on plasmids. These are small circular DNA molecules of about 2-5 kilobase pairs (kb) that coexist with chromosomes and can be easily extracted from cells to quantify their supercoiling density. The tacit assumption that plasmids are good proxies for chromosomes in this respect is justified by the fact that topoisomerases are expected to behave similarly on both molecules. In practice, migration properties of plasmids on gels are quantified, which reflect their compaction status and, therefore, their supercoiling density. Reported values thus also correspond to quantities that are averaged over time. In mesophilic bacteria, these values are negative, not exceeding in intensity 0.1~\cite{bliska_use_1987}. Chromosomes are therefore {\it in average} (along the genome and over time) underwound, with e.g.~$n_{\s=-0.05}\;\simeq 11.1$ base pairs per helix for the corresponding linear molecule.

As shown in the 1960s by the pioneering work of Vinograd and his collaborators~\cite{vinograd_early_1968}, negative supercoiling can actually strongly deform the double helix (Fig.~\ref{fig:linking}). Knowing that the intensity of supercoiling-induced mechanical stress depends on the local DNA sequence, various non-uniform phenomena can take place at the scale of a few base pairs, including DNA denaturation~\cite{vinograd_early_1968}, generation of DNA forms alternative to B-DNA~\cite{mirkin_discovery_2008} and generation of alternative secondary DNA structures such as cruciforms~\cite{mizuuchi_cruciform_1982}. Importantly, some of these structural motifs have a functional role, making their {\it physical prediction} an important {\it biological problem}~\cite{du_genome-wide_2013}. More generally, the question of the respective balance between these local deformations of the double helix and the formation of super-structures is a major open question. 

\section{Equilibrium models of supercoiled DNA}
\label{sec:thermo}


Even though molecular tools have been developed to probe the topology of DNA {\it in vivo} at multiple scales~\cite{lagomarsino_structure_2015}, with the recent possibility of obtaining information on the distribution of torsional stress along the genome~\cite{guo_high-resolution_2021,visser_psoralen_2022}, the {\it in vivo} occurrence and permanence of associated structural phenomena remain poorly quantified. Difficulties lie in the small size (of the order of the nm) of the structures involved but also in the often transient nature of phenomena. Namely, transcription and replication operate out of equilibrium, consuming both energy (ATP) and mass (nucleotides). From a modeling viewpoint, this raises the question of the relevance of equilibrium approaches to predict {\it in vivo} features associated with DNA supercoiling. Validity of these approaches is in fact often a consequence of the ``near-equilibrium'' nature of phenomena, as we explain below.

\subsection{1D models: predicting sequence effects}
\label{sec:1dmodel}

An illustrative example of the power of equilibrium statistical mechanics methods concerns the question of the alternative forms of DNA that are generated during the deformation of the double helix. The most common approaches are based on two assumptions. First, they neglect the effects of writhe (i.e., of super-structuring) to make the problem one-dimensional~\cite{fye_exact_1999}, which is similar to consider a molecule that is stretched by a few pN (Fig.~\ref{fig:thermo}). In doing so, analytical calculations are possible, making it possible to establish mathematical relationships between observables (i.e., measurements performed on the system) and system parameters (e.g., supercoiling level). It is then possible to predict behaviors without resorting to simulations which are often time-consuming and limited from the viewpoint of exhaustivity. Second, the most common approaches tacitly assume that supercoiling constraints are relaxed much faster than they are produced ({\it near-equilibrium condition}). This hypothesis appears to be reasonable, for example, in the case of transcription, whose initiation step requires the formation of a DNA denaturation bubble~\cite{murakami_bacterial_2003}. Namely, the twist and writhe relaxation times (below $\SI{1}{ms}$) for a 10 kb long molecule are typically 5 orders of magnitude smaller than the time for synthesizing a 1 kb long messenger RNA (100 s)~\cite{joyeux_requirements_2020,fosado_nonequilibrium_2021,wan_two-phase_2022} and 2 orders of magnitude with respect to the time for synthesizing a single base pair. Thus, questions concerning the energy required to denature DNA have been systematically addressed in the context of the equilibrium statistical mechanics of one-dimensional systems ~\cite{benham_torsional_1979,fye_exact_1999,jost_bubble_2011}. In particular, efficient semi-analytical approaches allow to predict the most probable sites of denaturation at the scale of a genome~\cite{jost_bubble_2011,jost_twist-dna_2013}. Despite simple assumptions with respect to the complexity of {\it in vivo} phenomena, including the neglect of super-structuring, such approaches are sufficiently predictive to be used, for example, in the analysis of the sensitivity of transcription initiation to supercoiling~\cite{forquet_role_2021}. This suggests in particular that strong deformations of the double helix is often dominant {\it in vivo}, which can be rationalized by considering the forces on the pN range that are expect to act on DNA as well~\cite{strick_stretching_2003} (Fig.~\ref{fig:thermo}).

\subsection{3D models: towards the large scales}
\label{sec:3dmodel}


Another illustrative example of the power of equilibrium statistical mechanics approaches concerns the problem of the three-dimensional structure adopted {\it in vivo} by bacterial DNA. In the 1970s, pioneering electron microscopy experiments on DNA extracted from \Ecoli\ cells revealed the presence of plectonemes~\cite{delius_electron_1974,kavenoff_electron_1976} but also of numerous loops~\cite{delius_electron_1974}. In 1990, {\it in vitro} experiments, still visualized by electron microscopy, showed for {\it in vivo} relevant values of supercoiling density a systematic tendency of bacterial DNA to form plectonemes at small scales and trees at large scales~\cite{boles_structure_1990}. Knowing that an excess of writhe can manifest itself in the form of solenoids or plectonemes, the question of the respective frequency of these two types of super-structures remained unanswered for many years. The question was partially solved in the early 90s with the help of the first polymer models of DNA including both self-avoidance properties and supercoiling constraints, with a resolution of a few tens base pairs~\cite{klenin_computer_1991,vologodskii_conformational_1992}. These models, which are still at the basis of current works, account for the electrostatic repulsion of DNA (self-avoidance), the energies of DNA curvature and torsion, which result from a coarse-grained description of DNA that neglects fine atomic details, as well as the global constraint of the conservation of the linking number.
% -- the latter can be implemented locally~\cite{carrivain_silico_2014,lepage_modeling_2017}, i.e., efficiently through the notion of parallel transport~\cite{bergou_discrete_2008}.
In 1994, the question of the prevalence of plectonemes was definitively resolved by Marko and Siggia on the basis of a quasi-analytical description of the equilibrium properties of these models~\cite{marko_fluctuations_1994}, showing that under physiological conditions of temperature, salt and supercoiling density, plectonemes are thermodynamically favored compared to solenoids. The reason lies in  the ``large'' energy of curvature of solenoids, which can be reduced drastically in plectonemes while keeping similar torsional stresses~\cite{marko_fluctuations_1994}. Single-molecule magnetic tweezers experiments combined with fluorescent labelling of DNA~\cite{van_loenhout_dynamics_2012} and polymer simulations~\cite{lepage_thermodynamics_2015} have then shown that the length of plectonemes {\it in vitro} are on the order of $\SI{1}{kb}$. Electron microscopy~\cite{boles_structure_1990} and statistical mechanics of plectonemes~\cite{marko_statistical_1995,barde_energy_2018} also revealed a diameter of the plectoneme varying between $\simeq\SI{30}{nm}$ at $\s\simeq-0.025$ and $\simeq\SI{5}{nm}$ at $\s\simeq-0.1$.
% — the same work also provided a rationale for reported ratio of 3 to 1 for the excess of writhe with respect to the excess of twist~\cite{boles_structure_1990}.
In addition, due to the entropic contribution of branches, it was shown that plectonemic structures become branched and form trees at large scales~\cite{marko_statistical_1995}, in accord with both experiments~\cite{boles_structure_1990} and numerical simulations~\cite{vologodskii_conformational_1992}. 


\begin{figure}[t]
\centering
\includegraphics[width=0.75\linewidth]{thermodynamics.pdf}
\caption{Example of diversity of structures obtained using a coarse-grain model~\cite{lepage_polymer_2019} of DNA at a resolution of $\SI{10}{bp}$, including the possibility of forming alternative DNA structures such as denaturation bubbles. A) Comparison between experiments~\cite{vlijm_experimental_2015} (colored curves) and simulations~\cite{lepage_polymer_2019} (black curves) for a $\SI{21}{kb}$ long molecule manipulated by a magnetic tweezer. The $x$-axis indicates the imposed supercoiling density on the molecule, and the $y$-axis shows the measured extension of the molecule. The experiments were conducted at two forces ($\SI{0.5}{pN}$ in blue and $\SI{4.5}{pN}$ in red). The inner panels show the typical conformations of the molecule obtained in the simulations for different experimental parameters (black dots). For example, the top panel indicates that when the molecule is stretched at $\SI{4.5}{pN}$ and undergoes a negative supercoiling of $\sim -0.04$, a denaturation bubble forms (indicated in red). The other conformations indicate the presence of plectonemes. B) For some force and supercoiling density, conformations can display denaturation bubbles (in red) located at the apex of plectonemes. These were initially predicted to occur using a coarse-grained polymer model of DNA at the nucleotide level~\cite{matek_plectoneme_2015}.}
\label{fig:thermo}
\end{figure}


The pioneering work of Vologodskii and his collaborators in the early 1990s, which focused on the development of Monte-Carlo simulations for topologically constrained polymer chains~\cite{vologodskii_conformational_1994}, sparked intense and ongoing research on the thermodynamic properties of supercoiled DNA at the scale of a few kb (see Fig.~\ref{fig:thermo} for an example). The so-called self-avoiding rod-like chain model~\cite{vologodskii_conformational_1992}, also known as the twistable worm-like chain model~\cite{nomidis_twist-bend_2019}, is typical of this approach and has been instrumental in analyzing the folding properties of both positively and negatively supercoiled DNA molecules without strong deformation of the B-DNA double helix. These properties include molecular extensions~\cite{vologodskii_extension_1997}, torques~\cite{schopflin_probing_2012,lepage_thermodynamics_2015}, and conformation details of super-structures~\cite{vologodskii_conformational_1992,bednar_twist_1994,klenin_modulation_1995}, as obtained in magnetic tweezer experiments and cryo-electron microscopy. The models can be extended to include DNA denaturation and the formation of alternative forms that occur at high negative supercoiling levels~\cite{lepage_polymer_2019}. Brownian dynamic simulations of supercoiled DNA were also developed in the early 1990s by Langowski and his collaborators, enabling the study of the kinetic properties of DNA loci~\cite{chirico_kinetics_1994}.

It is not possible to cover all the research that has been conducted in this field within the scope of this review. Therefore, we will only highlight a few results that we consider particularly relevant for understanding DNA structuring {\it in vivo}. For instance, Brownian dynamics simulations have predicted that plectonemes move along the DNA not only through (slow) diffusion but also by disappearing at one location and reappearing at a distant location, as early as in 1996~\cite{chirico_brownian_1996}. This ``hopping'' type of motion has been observed years later in fluorescent-labelling single-molecule experiments~\cite{van_loenhout_dynamics_2012}. Brownian dynamics simulations have also shown, for molecules of a few kb, that loci tend to make contacts through intra-plectoneme slithering (secondary-type of contacts) rather than through inter-plectoneme random collisions (tertiary type of contacts)~\cite{huang_dynamics_2001} -- tendency that may be reinforced by the hopping motion of plectonemes. The genomic range for which secondary contacts are expected to be more frequent {\it in vivo} nevertheless remains open. From a modeling viewpoint, this would require in particular to properly investigate finite size effects knowing that the size of molecules are at most on the order of a few tens kb, i.e., 2 to 3 orders of magnitude smaller than e.g.~the chromosome of \ecoli.

Next, Brownian dynamics simulations have been recently used to address the conditions for proteins that are able to bridge DNA~\cite{dillon_bacterial_2010} to act as topological barriers. Namely, to topologically insulate a genomic domain from its neighbor, it has been shown that not only bridges must block the diffusion of twist but they must also prevent DNA segments to rotate with respect to each other, i.e., they must block the diffusion of writhe, too~\cite{joyeux_requirements_2020}. This insight is all the more important that both \ecoli~\cite{postow_topological_2004} and {\it Salmonella}~\cite{deng_organization_2005} genomes have been shown to be organized into topologically independent domains, while comparative genomics predicts this organization to be ubiquitous in bacteria and to be associated with the basal coordination of transcription~\cite{junier_conserved_2016,junier_universal_2018}. Yet, the exact nature of topological barriers have remained mostly unknown. It also raises the question of how a transcribing RNA polymerase (RNAP) actually creates a barrier and contributes to the organization of the bacterial genome into topologically independent domains, as it has been shown experimentally~\cite{deng_transcription-induced_2004,higgins_rna_2014}. An interesting possibility might come, again, from polymer simulations of supercoiled DNA, but this time in the context of active processes. Namely, recent Brownian dynamics simulations have shown that the DNA supercoiling introduced by a transcribing RNAP might relax under the form of plectonemes that form far from the RNAP~\cite{fosado_nonequilibrium_2021}. The absence of any plectoneme that embeds the RNAP is indeed in accord with the capacity of this RNAP to block the diffusion of writhe. It is nevertheless worth mentioning that other modeling works have reported the tendency for a transcribing RNAP to locate at the apex of plectonemes~\cite{racko_transcription-induced_2018,joyeux_models_2022}, which is in opposition with its functioning as a topological barrier but consistent with previous {\it in vitro} experimental results~\cite{ten_heggeler-bordier_apical_1992}.

Lastly, we would like to highlight a recent study that utilized a combination of analytical approaches, Monte-Carlo simulations, and single-molecule experiments to investigate the ability of bridging proteins to form topological domains using fluctuation properties of supercoiled molecules~\cite{skoruppa_equilibrium_2022,vanderlinden_dna_2022}. This innovative type of analysis is expected to not only aid in the investigation of protein binding to DNA but also provide critical new insights into the molecular actors involved in the topological organization of bacterial genomes.

In summary, thermodynamic equilibrium approaches are a useful tool to study the structuring properties of DNA within a well-established framework, particularly when the relaxation speed of topological constraints is much larger than the rate at which they are produced, which is often the case {\it in vivo}. Using this approach, properties and sequence effects that occur at multiple scales can be studied as a function of parameters such as the supercoiling density or mechanical forces acting on DNA (Fig.~\ref{fig:thermo}).

\section{Supercoiling constraints and transcription}
\label{sec:transcription}

\subsection{The twin transcriptional-loop model and its topoisomerases}

Awareness of the central role of DNA supercoiling in transcription dates from the 1970s~\cite{wang_interactions_1974}, with the seminal work of James C. Wang, who discovered the first topoisomerase~\cite{wang_interaction_1971} known today as Topo~I. In particular, Liu and Wang hypothesized that the most frequent situation in bacteria for a transcribing RNAP is to generate supercoiling stresses on each side of it because of the impossibility of the RNAP to rotate around DNA~\cite{liu_supercoiling_1987}. More precisely, because the transcribing RNAP and its associated mRNA interact with other macromolecules (ribosomes, regulatory factors, other RNAPs or DNA itself through e.g.~the formation of R-loops~\cite{thomas_hybridization_1976}), the resulting macro-complex experiences torsional friction. This hinders the rotation of the RNAP around DNA. In addition, DNA itself interacts with various macromolecules (e.g. membrane~\cite{lynch_anchoring_1993}, clusters of RNAPs~\cite{stracy_live-cell_2015}), which is expected to hinder its global rotation, too. As a consequence of the difficulty of both RNAPs and DNA molecules to rotate and according to the topological considerations of Fig.~\ref{fig:twin_scheme}, Liu and Wang surmised that the transcription of genes is most often associated with negative (positive) DNA supercoiling upstream (downstream) the transcribing RNAPs, which they demonstrated for a particular case on a plasmid~\cite{wu_transcription_1988}. The corresponding biological ``model'' is known as the twin transcriptional-loop (TTL) model~\cite{liu_supercoiling_1987} (Fig.~\ref{fig:twin}).

An often overlooked ingredient of the TTL model is the presence of topoisomerases. Liu, Wang and collaborators indeed demonstrated that Topo~I and DNA gyrase were responsible for relaxing the upstream negative supercoils and the downstream positive supercoils, respectively~\cite{wu_transcription_1988}. They then anticipated that for gene expression to be properly predicted, one would need to include the activity of these topoisomerases~\cite{liu_supercoiling_1987,wu_transcription_1988}. 35 years later, experiments have convincingly demonstrated that gene context is indeed an ingredient at least as important as the action of transcription factors to rationalize gene expression~\cite{nagy-staron_local_2021,scholz_genetic_2022}. Moreover, several experiments over the last decade have not only corroborated the relevance of the TTL model but demonstrated the necessity to consider Topo~I and DNA gyrase to quantitatively apprehend the transcription process~\cite{chong_mechanism_2014, ahmed_transcription_2017, kim_long-distance_2019, ferrandiz_genome-wide_2021, sutormin_interaction_2022, boulas_assessing_2022}.

\begin{figure}[t]
\floatbox[{\capbeside\thisfloatsetup{capbesideposition={right,top},capbesidewidth=0.5\linewidth}}]{figure}[\FBwidth]
{\caption{Schematic representation of the twin transcriptional-loop (TTL) model. In this model, a transcribing RNAP (in pink) both generates torsional stresses on each side and behaves as a topological barrier due to the presence of, for example, ribosomes (in yellow) translating the mRNA into proteins and forming a large transcription-translation complex. This induces two twin domains of supercoiling with: i) upstream, a negatively generated supercoiling relaxed by TopoI (in green) and ii) downstream, a positively generated supercoiling relaxed by DNA gyrase (in blue).}\label{fig:twin}}
{\includegraphics[width=\linewidth]{twin.pdf}}
\end{figure}

\subsection{Transcriptional bursting and DNA topology}

The transcription of many genes in bacteria (and eukaryotes~\cite{coulon_eukaryotic_2013}) is bursty~\cite{golding_real-time_2005}: it is governed by a non-Poissonian process of transcript production involving at least two distinct characteristic times. Namely, single-cell experiments have revealed that the dynamics of expression alternate slowly between active and inactive phases of transcription, with a characteristic time on the order of ten minutes~\cite{golding_real-time_2005,so_general_2011}. In particular, this characteristic time is much larger than those associated with the mechanisms of transcription during the active phase, whether it be the time required to transcribe the entire gene ($\sim 1$  minute) or the time between two supercoil removals by the topoisomerases (a few seconds)~\cite{boulas_assessing_2022}. Interestingly, this slow modulation of transcription depends on the activity of DNA gyrases, with a characteristic time that is all the smaller that the concentration of DNA gyrase is high~\cite{chong_mechanism_2014}. To understand this phenomenon, let us recall that RNAPs stall when the positive downstream supercoiling becomes too intense~\cite{ma_transcription_2013,ma_interplay_2014}, that is, when the supercoiling density is on the order of $+0.06$ (as shown below). In the absence of DNA gyrase, transcription is therefore hindered up to the point where a DNA gyrase binds downstream and relaxes the positive supercoils. Note that in \ecoli, ``only'' about 300 gyrases have been shown to be bound along the genome~\cite{stracy_single-molecule_2019}, that is, one gyrase every $\SI{15}{kb}$, which is consistent with DNA gyrase being a limiting factor for transcription. Altogether, this shows again the importance of DNA supercoiling and of topoisomerases for the transcription process.


\subsection{Getting quantitative: some numbers}

To get a quantitative apprehension of the constraints at play during the transcription of a gene, let us recall that in \ecoli,  RNAPs transcribe at a rate between $25$ and $\SI{100}{bp}$.s$^{-1}$, depending on the growth rate of the bacterium~\cite{bremer_modulation_2008}. In the extreme case of an absence of rotation of the RNAP, this means that the DNA unwinding associated with transcription generates between $\sim 2$ to $\sim 10$ positive (negative) supercoils per second upstream (downstream) the RNAP --  considering one supercoil per transcription of $\sim 10$ base pairs or one turn of the DNA double helix.
Considering the presence of topological barriers located at a distance of $\sim\SI{10}{kb}$ ($\sim\SI{1000}{supercoils}$) that prevent the dissipation of these supercoils~\cite{postow_topological_2004}, according to the TTL model, transcription activity is expected to make DNA supercoiling density $\s$ vary by an amount of at least $0.01$ every second on each side of the transcription complex. With respect to DNA, the effects of supercoiling become significant for $|\s|=0.01$ and highly disruptive for $|\s|=0.1$~\cite{strick_stretching_2003}. With respect to RNAPs, studies have suggested that they stall {\it in vivo} for torques ($\Gamma$) on the order of $\SI{18}{pN}$~\cite{ma_transcription_2019} or equivalently, $|\sigma| \simeq 0.06$ -- using $\sigma = \Gamma/A$ where $A= \SI{300}{pN}$ is an average of the values estimated from single-molecule experiments for the regime where plectonemes are present ($\SI{200}{pN}$) and for the regime where no super-structure is present ($\SI{400}{pN}$)~\cite{marko_torque_2007}. RNAP translocation along the DNA can then resume only if the associated torques are released, which can occur {\it in vivo} only through two mechanisms: i) another RNAP compensates the supercoiling, which however does not solve the problem upstream and downstream the train of RNAPs; ii) the action of topoisomerases. In other frequent situations such as those involving divergent genes, supercoiling densities may actually vary even more abruptly. Namely, for two divergent promoters separated by a distance of $\simeq \SI{200}{bp}$, the transcription of the upstream gene would create a transitory barrier and the total variations of supercoiling would be on the order of $0.1$ every second.

\subsection{Physical models of the TTL biological model}

{\it Physical} models have then been developed in the context of the {\it biological} TTL model with the aim to quantitatively rationalize transcription {\it in vivo}. %~\cite{sevier_mechanical_2016,ancona_transcriptional_2019,klindziuk_mechanochemical_2020,tripathi_dna_2022,boulas_assessing_2022}. 
They include the interplay between DNA, RNAPs and topoisomerases and they are based on a coarse-grained description of the motion of RNAPs at a resolution of typically one or a few base pairs. Associated torques can indeed vary dramatically as soon as the RNAP transcribes a few base pairs: for two consecutive RNAPs separated by e.g.~$100$ (500) base pairs,
%(as in RNAP convoys observed in eukaryotic cells~\cite{tantale_single-molecule_2016}),
it only requires the transcription of one (five) base pair(s) to make the supercoiling density vary by an amount of $\sim 0.01$. Physical models do not yet include the explicit structure of DNA. They nevertheless display a rich phenomenology that still needs to be fully understood.

More precisely, some of the models have endeavored to quantify the downstream accumulation of positive supercoiling and the impact of gyrase on the bursty behavior of transcription~\cite{sevier_mechanical_2016,ancona_transcriptional_2019,klindziuk_mechanochemical_2020}. Others have focused more specifically on the collective behavior of RNAPs~\cite{brackley_stochastic_2016,chatterjee_dna_2021,sevier_collective_2022,tripathi_dna_2022,geng_spatially_2022}. In particular, several scenarios have been proposed for the observation of non-trivial long-distance effects associated with transcription. Namely, opposite tendencies for the translocation speed of an RNAP in the presence of other RNAPs have been observed, depending on whether the promoter is active or not, with more rapid, slower respectively, translocation rates~\cite{kim_long-distance_2019}. These phenomena cannot be explained by a simple cancelation of the supercoiling between successive RNAPs. Additional mechanisms have thus been hypothesized. These include (i) the  velocity of an RNAP that depends on the net torque that is exerted on it, i.e., the downstream torque minus the upstream torque~\cite{chatterjee_dna_2021,tripathi_dna_2022,sevier_collective_2022,geng_spatially_2022}, ii) a supercoiling stress that increases with the number of bound RNAPs~\cite{chatterjee_dna_2021}, iii) a DNA-bound TF acting as a topological barrier~\cite{chatterjee_dna_2021} and iv) a slow diffusion of the linking number~\cite{brackley_stochastic_2016,geng_spatially_2022}.

Hypothesis (i) deserves experimental testing since single-molecule experiments have thus far examined the impact of downstream and upstream torques on elongating RNAPs {\it separately}~\cite{ma_transcription_2013,ma_interplay_2014}. It also remains to be determined whether this hypothesis is consistent with an elongating RNAP's ability to act as a topological barrier. Lastly, it should be noted that quantitative modeling of transcription by separately considering downstream and upstream stalling torques is feasible~\cite{boulas_assessing_2022}.
Hypothesis (ii) echoes the observation of RNAPs that cluster when the most downstream one stalls~\cite{fujita_transcriptional_2016}, which should indeed exert a higher torsional friction. Hypothesis (iii) could be tested experimentally. Nevertheless, both experiments~\cite{leng_dividing_2011} and polymer simulations~\cite{joyeux_requirements_2020} already suggest that for TF to act as a topological barrier, it must at least bridge DNA. Finally, hypothesis (iv) was made by considering the relaxation speed of the linking number as given by the diffusion speed of plectonemes. However, both single-molecule experiments~\cite{crut_fast_2007,van_loenhout_dynamics_2012} and polymer simulations~\cite{matek_plectoneme_2015,joyeux_requirements_2020,fosado_nonequilibrium_2021,wan_two-phase_2022} have  shown that the former, which is responsible for the formation of plectonemes, is much higher than the latter. In other words, supercoiling establishment during transcription can be regarded as a quasi-static process~\cite{wan_two-phase_2022}.

\subsection{Towards a quantitative understanding of topoisomerases}

Recently, two physical implementations of the TTL model have, for the first time, {\it separately} considered the actions of Topo~I and DNA gyrase~\cite{geng_spatially_2022,boulas_assessing_2022}. In particular, the model proposed in~\cite{boulas_assessing_2022} has a minimal number of parameters and, coupled with an experimental realization of the TTL model in \ecoli, has provided novel, quantitative insights into the involvement of topoisomerases~\cite{boulas_assessing_2022}. Specifically, it predicts that Topo~I and DNA gyrase systematically accompany gene transcription by respectively removing negative and positive turns at rates of approximately 1 to 2 (negative) supercoils per second and at least 2 (positive) supercoils per second. These rates are consistent with in vitro activities reported for both Topo~I~\cite{terekhova_bacterial_2012} and DNA gyrase~\cite{ashley_activities_2017}. Moreover, the model predicts that the positive supercoils introduced by Topo~I have antagonistic effects on the different stages of transcription. On the one hand, they allow the release of negative torque upstream of the RNAP so that it can properly translocate~\cite{ma_transcription_2013,ma_interplay_2014}. On the other hand, they hinder the opening of the double helix, thereby tending to repress the formation of the so-called open complex~\cite{murakami_bacterial_2003} at the initiation stag.

\subsection{Cooperative effects between genes}

The global nature of the conservation of the linking number (section~\ref{sec:supercoiling}) and the quick relaxation of twist and writhe compared to the speed of supercoil generation (section~\ref{sec:thermo}) suggest that there is a long-range coupling of supercoiling-induced mechanical stresses that extends to topological barriers. Accordingly, changes in supercoiling around highly transcribing genes can extend up to tens of kb~\cite{visser_psoralen_2022}. Multiple experimental studies have, {\it de facto}, demonstrated that supercoiling-induced coupling affects the transcription of neighboring genes~\cite{lilley_dna_1996,opel_dna_2001}, with an impact observed at distances of several kb~\cite{hanafi_activation_2002, moulin_topological_2004}.
%Notably, this possibility was predicted by Liu and Wang in their seminal paper~\cite{liu_supercoiling_1987}.

Physical models have been developed in order to better understand these effects ~\cite{meyer_torsion-mediated_2014,johnstone_supercoiling-mediated_2022,sevier_collective_2022,geng_spatially_2022} and to understand the impact of this coupling on the organization of genomes~\cite{sobetzko_transcription-coupled_2016,geng_spatially_2022} and their possible evolution~\cite{grohens_genome-wide_2022}. So far, models have not included effects from topoisomerases, except in a very recent work~\cite{geng_spatially_2022}. Yet, the necessity to include them to understand the coupling between neighbor genes was stressed (already) 30 years ago in an analysis of the non-trivial transcriptional properties of the leucine biosynthetic operon in {\it Salmonella} Typhimurium~\cite{lilley_local_1991}. The latter has become a prototypical system of the supercoiling-based coupling of the transcription of divergent genes~\cite{lilley_dna_1996,rhee_transcriptional_1999,opel_dna_2001,hanafi_activation_2002}.

\subsection{What about DNA folding?}
\label{sec:DNAfold}

All physical models of transcription developed so far either neglect the effects of DNA super-structuring or include them in an effective way, for example by using appropriate conversion of supercoiling densities into associated torques that reflects the presence of super-structures~\cite{marko_torque_2007}. Experimentally, the effect of local DNA folding on transcription is actually not known, except in the specific case of small DNA loops involving transcription factors~\cite{cournac_dna_2013}. Interestingly, Wang nevertheless suspected that for large values of supercoiling density, folding effects would limit the accessibility of RNAP to DNA~\cite{wang_interactions_1974}. His reasoning came from the comparison of two phenomena, whose behaviors as a function of the supercoiling density were similar. Namely, on the one hand, he observed that the transcriptional activity of an RNAP, and more specifically of the core enzyme (i.e., without the ability of the RNAP to recognize specific promoters), is a non-monotonic function of supercoiling density with a maximum at values between $-0.05$ and $-0.04$. On the other hand, he observed a change in the sedimentation properties of plasmids in migration gels around $-0.035$ that he interpreted as a ``higher twisting of one double helix around the other''~\cite{wang_interactions_1974}. Years later, equilibrium studies of polymer physics models of $\SI{10}{kb}$ long supercoiled molecules confirmed this conformational effect~\cite{krajina_large-scale_2016}: when the supercoiling density decreases below $\sim -0.03$, branches become longer and tighter, which could indeed hinder accessibility to DNA. We note, here, that this structural effect could actually contribute to the systematic non-monotonic behavior of gene expression level as a function of supercoiling density observed for different promoters {\it in vitro}~\cite{pineau_what_2022}, although the ``maximal'' supercoiling values differ substantially between promoters~\cite{pineau_what_2022}.
%, with a maximum value on the order of $-0.05$. Variations around this value between promoters might then come from different sentivities to supercoiling of the denaturation process involved during the formation of the open-complex at initiation~\cite{pineau_what_2022}.


%Finally, let us note that recent high-resolution (on the order of 500 bp) data  obtained in a context where transcription of a single gene is active~\cite{bignaud_transcriptional_2022} have provided crucial information on the expected properties of local DNA organization associated with transcription. Any relevant polymer physics model associated with the transcription process should thus be able to provide a rationale for the specific patterns found in these experiments, such as the presence of arrows for highly expressed genes.

\section{Supercoiling constraints and DNA replication}
\label{sec:replication}

\subsection{Topological constrains downstream the replisome}

The DNA polymerase of mesophilic bacteria duplicates DNA at a rate of about $\SI{1000}{bp}$ per second. Composed of a large number of proteins and, hence, expected to be constrained by a high torsional friction with the surrounding biomolecules of the cytoplasm, the associated replication complex, also known as the replisome, is unlikely to rotate as quick as it introduces supercoils in DNA. Supposing no rotation at all, the replisome would thus introduce on the order of $100$ positive supercoils per second in the downstream DNA. Considering the presence of topological barriers located at a distance on the order of $\SI{10}{kb}$ (section~\ref{sec:transcription}), the replisome would thus make downstream DNA supercoiling density vary by an amount of $0.1$ every second  -- see below for the discussion of a rotating replisome. Since DNA replication is directly linked to the ability of bacteria to multiply, it is therefore not surprising that replisome's advancing is accompanied by a high activity of topoisomerases~\cite{khodursky_analysis_2000, mckie_dna_2021}, and more specifically downstream by DNA gyrases. In this regard, high-speed single-molecule fluorescence imaging has revealed the presence in \ecoli\ of clusters containing an average of 12 gyrases (ranging from 2 to $\sim100$) and concomitant with the onset of replication~\cite{stracy_single-molecule_2019}.
%, which are used to relax the positive supercoiling generated along the yet-to-be-replicated DNA~\cite{mckie_dna_2021}.
Also, the DNA gyrase of \Bsub\ has been shown to relax up to 100 supercoils per second in single-molecule experiments~\cite{ashley_activities_2017}. In any case, the effective rate of positive supercoils removal {\it in vivo} remains unknown. We also remind that the removal of positive supercoiling by DNA gyrases is ATP-dependent with an enzymatic cycle involving the hydrolysis of two ATP molecules to remove two supercoils~\cite{wang_moving_1998}.

\subsection{Upstream topological constrains: the precatenane problem}
\label{sec:precat}

The topological problems upstream and downstream of the advancing replisome are of a different nature. Downstream, they involve a single molecule: the unreplicated DNA. Upstream, they involve the intermingling of two molecules: the replicated DNAs. Specifically, unwinding of the two DNA strands during replication does not generate upstream mechanical stress that would destabilize the system, as it does in transcription. The two resulting single-stranded DNA molecules are instead managed simultaneously by dedicated enzymes associated with the replication complex to build new double helices~\cite{reyes-lamothe_chromosome_2012}. However, super-structuring between replicated DNA is known to occur behind the replisome~\cite{peter_structure_1998}. To understand this phenomenon, it must be realized that although the replication complex is large, it can rotate in principle, especially because of the large torques generated downstream (as described earlier). From a topological viewpoint, the two replicated DNA molecules extend the Watson and Crick strands of the unreplicated DNA (Fig.\ref{fig:precat}A). The situation is thus identical to the generation of twin supercoils described in Fig.\ref{fig:twin_scheme}, with the possibility of rotation of the unwinding machine. According to that figure, the replisome rotates in the clockwise sense, and the replicated DNA forms a right-handed superhelix (Fig.\ref{fig:precat}A), known {\it in vivo} as precatenanes, and in single-molecule experiments as braids. Importantly, precatenanes impede replicated chromosomes from diffusing away from each other. As a consequence, precatenane release is necessary for replicated chromosomes to properly segregate. Multiple lines of evidence over the last 25 years have revealed that this is primarily performed by the topoisomerase Topo~IV~\cite{zechiedrich_topoisomerase_1997,charvin_single-molecule_2003,stone_chirality_2003,wang_modulation_2008,lesterlin_sister_2012}, with additional specific contributions from Topo~III~\cite{mckie_dna_2021}.

\begin{figure}
\centering
\includegraphics[width=\linewidth]{precatenane.pdf}
\caption{A) Schematic representation of the formation of precatenanes during DNA replication. The replisome, indicated in light purple, moves along the unreplicated DNA double helix with the Watson and Crick strands shown in green and orange, respectively. Downstream, these strands give rise to two replicated molecules indicated by the thick green and orange lines, respectively. Upper panels: during the unwinding of the unreplicated DNA double helix, if the replisome rotates, it transfers the inter-strand crossings (black circles on the left) to the replicated DNA (circles on the right), which then form a superhelix (precatenane) with the same chirality. The red circle indicates a crossing that has not yet been unwound by the replisome. Lower panel: the net result of this operation is the formation of precatenanes. B) Possible conformations of precatenanes (i.e., the green and orange lines represent replicated DNA molecules) leading to left-handed crossings. On the left, the precatenanes buckle to form a plectonemic structure generating two left-handed crossings indicated in red (adapted from~\cite{charvin_single-molecule_2003}). On the right, the intrinsic negative supercoiling of each replicated DNA leads to a left-handed crossing indicated by the red arrow (adapted from~\cite{rawdon_how_2016}).}
\label{fig:precat}
\end{figure}


%A puzzling aspect, at first sight, of the decatenation by
Note, then, while precatenanes are right-handed, single-molecule experiments have shown that Topo~IV decatenates left-handed braided structures much more efficiently ~\cite{crisona_preferential_2000,charvin_single-molecule_2003,stone_chirality_2003}, raising the question of how Topo~IV would remove precatenanes {\it in vivo}. From a polymer physics modeling perspective, three non-exclusive scenarios have been proposed. 
%as a consequence of the folding properties of braided molecules studied in the context of polymer physics at equilibrium (Fig.~\ref{fig:precat}B). 
First, equilibrium statistical mechanics analysis of braided molecules have shown that precatenanes, just as DNA, also buckle to form left-handed plectonemes (of precatenanes) when the density of precatenanes is sufficiently high. More precisely, defining the density of precatenanes as the ratio between the number of crossing of the two molecules, on the one hand, and the number of double helices along a single molecule, on the other hand, Marko predicted buckling to occur at a value around $0.045$~\cite{marko_supercoiled_1997}. This has been confirmed by polymer simulations of braided molecules that are stretched by pN-range forces relevant to {\it in vivo} conditions ~\cite{stone_chirality_2003,charvin_braiding_2005,forte_plectoneme_2019}. The decatenation of a right-handed precatenane could therefore occur inside a left-handed plectoneme of precatenanes (Fig.~\ref{fig:precat}B). 
%This would require in particular two enzymatic reactions instead of a single one for the left-handed precatenane, which is in accord with a reported two-fold difference in the decatenation rate between left-handed precatenanes and right-handed precatenanes in the buckling regime~\cite{charvin_single-molecule_2003}.
Second, for a number of precatenanes much below their buckling regime,
% and forces much lower than $\SI{1}{pN}$, 
%polymer simulations have shown that the angles between two braided molecules are distributed very similarly, i.e., rather broadly  with a peak close to $90^\circ$~\cite{neuman_mechanisms_2009}. In parallel, 
single-molecule experiments revealed that the chiral asymmetry in Topo~IV activity resulted from a difference in the processivity of the enzymes with respect to the chirality of the braid, with a high (low) processivity for left-handed (right-handed) precatenanes. Topo~IV could thus punctually remove right-handed precatenanes similarly to left-handed ones. %, but simply with a much less efficiency.
Third, polymer simulations of catenated DNA molecules at equilibrium revealed specific left-handed crossing between the two catenanes when the molecules are negatively supercoiled~\cite{rawdon_how_2016} (Fig.~\ref{fig:precat}B). Accordingly, the decatenation of sister chromatids by Topo~IV could be enhanced by negative supercoiling.

\subsection{An inefficient process selected by Nature?}

Why would nature select an inefficient Topo~IV decatenation activity? A common response is that Topo IV {\it should not   affect the average level of supercoiling}, with the idea that there exists some optimal value of average supercoiling~\cite{menzel_regulation_1983}. Thus, Topo IV should not intervene in the resolution of right-handed plectonemes generated upstream of RNAPs. However, this answer fails to explain why Topo~IV and DNA gyrase have overlapping activities~\cite{mckie_dna_2021,hirsch_what_2021}. Moreover, these two enzymes mainly differ at their C-terminal domain only~\cite{hirsch_what_2021}, making their inter-conversion a rather easy process from an evolutionary perspective. Instead, we surmise that the inefficiency of Topo~IV to remove right-handed plectonemes {\it allows not to interfere with the dynamics of the transcription initiation stage}. Namely, as recently shown by a quantitative modeling of transcription~\cite{boulas_assessing_2022} (see section~\ref{sec:transcription} for more details), the punctual action of Topo~I-like enzymes can generate strong variations of the upstream supercoiling, which may have drastic impact on the initiation step~\cite{lilley_local_1991}.
%Accordingly, the inefficiency of Topo~IV for right-handed precatenanes could thus avoid interfering with the regulation of transcription initiation, only.

\subsection{The cohesion-segregation problem}

Segregation of chromosomal loci, which occurs quickly after being replicated~\cite{reyes-lamothe_chromosome_2012,possoz_bacterial_2012,wang_organization_2013,kleckner_bacterial_2014,badrinarayanan_bacterial_2015}, requires replicated chromosomes to properly drift away from each other. Yet, replicated loci are known to remain close-by in space for at least a few minutes after the passage of the replication machinery. This so-called cohesion stage depends on multiple factors, including the type of bacteria, their growth conditions but also the timing along the cell cycle~\cite{possoz_bacterial_2012}. Precatenanes play a major role in this process~\cite{conin_extended_2022} as they topologically prevent replicated chromosomes to drift away from each other. Specific systems such as an increased activity of Topo~IV~\cite{el_sayyed_mapping_2016} or the action of molecular motors pulling on the replicated DNAs~\cite{bigot_ftsk_2007} are thus known to participate in the resolution of topological problems at the end of replication, when the density of precatenanes is a priori the highest or when only catenanes remain, i.e., when replication is finished. Nevertheless, several fundamental aspects of cohesion remain to be understood. For instance, are chromatid cohesion and precatenane formation a unique process, or can chromatids be cohesive without being topologically intermingled? Also, what are the expected respective trajectories of replicated loci once precatenanes are removed? Do they spontaneously segregate? In which directions?

In the early 2000s, the possibility of spontaneous, thermodynamically favorable segregation of intermingled sister chromatids due to the plectonemic structure of each chromatid was proposed~\cite{postow_topological_2001}. This was inspired by polymer physics modeling work showing that the probability of catenation between circular DNA and linear cyclizing DNA decreases exponentially with the supercoiling density of circular DNA~\cite{rybenkov_effect_1997} — as a consequence of a volume exclusion from the DNA compacted by the supercoiling and of the reduction of the possibilities to insert the linear DNA into the circular DNA. Yet, the formation of replication precatenanes is qualitatively different from this problem. A few years later, similar ideas were investigated in the context of the equilibrium statistical mechanics of catenated DNA molecules that are individually supercoiled, asking in particular the question of the amount of energy to provide to add/remove a supercoil to one chromatid {\it versus} add/remove a hypercoil from the pair of concatenated sister chromatids~\cite{martinez-robles_interplay_2009}. Two observations were discussed in particular: (i) intra-molecule negative supercoiling under the form of plectonemes make the addition of catenanes more difficult, which may hinder the production of precatenanes; (ii) segregation of the two molecules is favored by plectonemes, very likely as the result of volume-exclusion effects. Knowing that the diffusion of DNA supercoiling stresses is very fast compared to, for example, transcription rates~\cite{ivenso_simulation_2016,joyeux_requirements_2020,fosado_nonequilibrium_2021} (see section~\ref{sec:thermo} for details), the time scale associated with the structuring of freshly replicated DNA into plectonemes would therefore be dominated by the transcription reinitiation time (i.e., the slowest time scale). 

\subsection{Challenges ahead: out-of-equilibrium models involving long molecules}

As discussed above, the mechanisms adopted by Topo~IV and, hence, its efficiency to decatenate replicated DNA {\it in vivo} remain unknown. In particular, the spatial conformations of the precatenanes remain unknown, with at least three types of conformations that could occur (Fig.~\ref{fig:precat}B). Moreover, as far as segregation is concerned, it remains to be demonstrated that both volume-exclusion effects and entropic forces similar to those invoked to explain large-scale segregation of chromosomes ~\cite{jun_entropy-driven_2006,jun_entropy_2010} are sufficient to explain the rapid segregation of replicated chromosomes throughout the cell cycle.
%— in~\cite{jun_entropy-driven_2006}, e.g., the authors used an unrealistic outer cylinder close to the cell wall to facilitate this mechanism.
Altogether, this suggests that novel theoretical studies must be performed in order to better understand the disentangling and segregation of freshly replicated chromosomes. In this regard, a minimal model has been recently analyzed in the absence of volume exclusion effects~\cite{sevier_mechanical_2020}. It is composed of three distinct molecules (unreplicated DNA and the two copies of replicated DNA), of a converter that transforms unreplicated DNA double helices into precatenanes as well as the respective downstream and upstream actions of DNA gyrase and Topo~IV. The objective of this work was to identify very general properties associated with the fundamental constrains on how replisomes and their associated topoisomerases process DNA. The system was analyzed in the simplifying context of a replisome that freely rotates such that the upstream and downstream torques acting on each side of it are equal. Two important results are then worth mentioning. First, it was found that the unreplicated DNA fully collapse into plectonemes before the precatenanes buckle in the absence of topoisomerases.%, suggesting that the low precatenanes regime discussed in section~\ref{sec:precat} might be a relevant regime {\it in vivo}. 
Second, to avoid this downstream plectonemic collapse that would trap the replisome, topoisomerases (i.e., DNA gyrase) must remove at least $\sim 1$ positive supercoil per second.

To further progress in the problem of the disentanglement and the segregation of replicated DNA molecules, it will be necessary to include the explicit structure of DNA, without which the phenomena of volume exclusion are difficult to quantify. The cost to be paid is the absence of analytical solutions as in~\cite{sevier_mechanical_2020} and the need to resort to simulations in order to study the non-equilibrium properties of the system. The numerical challenge is significant because the scales involved in the cohesion of sister chromatids (a few hundreds kilo base pairs~\cite{el_sayyed_mapping_2016}) are at least one order of magnitude greater than the typical lengths of molecules studied in Brownian dynamics (a few tens kb at most) and two orders of magnitudes greater than the lengths used in the most recent studies of precatenane-like braiding phenomena~\cite{forte_plectoneme_2019}. Methods like those used in the dynamics of rigid body~\cite{carrivain_silico_2014} thus need to be contemplated in order to improve the efficiency of the simulations.

These approaches should then give useful information in combination with data about contact frequencies between chromosomal loci, be it those allowing to differentiate sister chromatids as in the recently developed  Hi-SC2 method~\cite{espinosa_high-resolution_2020} or those resulting from standard Hi-C methods~\cite{lieberman-aiden_comprehensive_2009,le_new_2014}. Predictions should be tested in the context of topoisomerase mutants, whose effects on contact properties can be precisely quantified~\cite{conin_extended_2022}, and the activity of DNA gyrase and Topo~IV hopefully be estimated (at least for various rates of precatenane production). These approaches are also expected to provide crucial insights about how Topo~IV actually removes precatenanes {\it vivo} by quantifying the relative occurrence of the three mechanisms discussed in section~\ref{sec:precat} (Fig.~\ref{fig:precat}B). These models should also make it possible to validate or refute the spontaneous nature of the segregation of freshly disentangled replicated DNA.

Finally, let us mention that just as eukaryotes, bacteria contain condensins whose activity is crucial to the proper organisation and segregation of chromosomes~\cite{hirano_condensin-based_2016}. Interestingly, some of the phenomena associated with the segregation of replicated chromosomes are reminiscent of the problem of the organization and segregation of mitotic chromosomes in eukaryotes~\cite{nasmyth_disseminating_2001}. Namely, Brownian dynamics simulations in the context of molecular motors extruding DNA have clarified the crucial role of condensins for chromatid segregation during prophase. The proposed mechanism relies on an effective repulsion between topologically unlinked loops~\cite{halverson_melt_2014} facilitated in this particular case by the active extrusion of intra-chromatid DNA loops by the condensins~\cite{goloborodko_compaction_2016}.%, i.e., according to an out-of-equilibrium process acting on polymer chains.
Transposed to the problem of bacteria, these approaches offer a promising modeling framework for studying the phenomenology associated with condensins, which are known to play a fundamental role in the segregation of chromosomes~\cite{gruber_interlinked_2014,wang_smc_2014,lioy_distinct_2020} and to functionally interact with topoisomerases like Topo~IV~\cite{hayama_physical_2010,li_escherichia_2010}.

\section{Supercoiling and nucleoid formation}
\label{sec:nucleoid}

Contrary to eukaryotes, bacterial DNA is localized in a membrane-free region of the cell called the nucleoid, which was first highlighted in the 1940s -- see~\cite{robinow_bacterial_1994} for an historical review.
Recent live imaging techniques have confirmed this phenomenon, revealing more particularly the exclusion of most ribosomes from the nucleoid so that they localize at the poles of the cells (when these are cylindrical as in many bacteria) -- see~\cite{chai_organization_2014} and references therein.
In \ecoli, live fluorescence imaging indicates that, independently of the time point along the cell cycle, the nucleoid occupies approximately half the main axis of the cell and the majority of the cell section, leaving only a thin layer close to the cell wall~\cite{junier_polymer_2014,wu_cell_2019}. Super-resolution techniques have reported smaller and more structured regions~\cite{spahn_super-resolution_2014}, in accord with large internal rearrangements occurring at short time scales (i.e., below 1 minute)~\cite{wu_direct_2019}. A puzzling aspect of nucleoids has concerned their specific cellular localization during the cell cycle~\cite{possoz_bacterial_2012,reyes-lamothe_chromosome_2012,wang_organization_2013,kleckner_bacterial_2014,badrinarayanan_bacterial_2015}. In \ecoli\, for instance, just after cell division the nucleoid is localized at the center of the cell. As replication proceeds, it quickly splits into two (replicated) nucleoids which localize at the quarters of the cell until cell division occurs.

The physical mechanisms responsible for nucleoid formation have fueled numerous theoretical studies (see~\cite{benza_physical_2012,joyeux_compaction_2015} for not too old reviews), with a recurring question: what is the precise role of DNA supercoiling in this matter? The latter is indeed often mentioned as contributing to DNA compaction. However, more than 30 years ago, Cozzarelli and colleagues noted that ``the extended thin form of plectonemically supercoiled DNA offers little compaction for cellular packaging, but promotes interaction between cis-acting sequence elements that may be distant in primary structure''~\cite{boles_structure_1990}. So, does supercoiling really participate in genome compaction? More specifically, is it a key factor of nucleoid formation?

\subsection{Spatial extension of a supercoiled DNA versus confinement: scaling arguments}

First and foremost, let us address the question of the spatial extension of a supercoiled circular DNA molecule under conditions of temperature and salinity equivalent to those {\it in vivo}, but without the confinement of the cell. In polymer physics, the spatial extension of a chain is quantified by its radius of gyration, i.e., the root-mean-square distance between the center of mass of the chain and each of its monomers. It is then customary to describe the large-scale behaviors of polymer chains by assessing how their radius of gyration varies with their molecular length $L$ as the latter becomes large, also known as scaling laws~\cite{de_gennes_scaling_1979}. For example, the radius of gyration of both linear and circular self-avoiding chains has been shown to scale as $L^{0.59}$\cite{de_gennes_exponents_1972,baumgartner_statistics_1982}. Knowing that a circular chain of 30 kb has a radius of gyration on the order of $\SI{325}{nm}$ (see e.g.~\cite{walter_supercoiled_2021}), this means that a genome of $\SI{5}{Mb}$ (genomic length typical of many bacteria, including \ecoli) is predicted to have an equivalent radius of gyration of approximately $325\times(5000/30)^{0.59}\simeq\SI{6.6}{\um}$.
For comparison, an \ecoli\ cell with a length of $\SI{2}{\um}$ and a radius of $\SI{0.5}{\um}$ has a much smaller equivalent gyration radius of $\simeq \SI{0.67}{\um}$.
In particular, the volume of the bacterium is $(6.6/0.67)^3 \approx1000$ times smaller than the typical volume spanned by its thermally fluctuating, unconstrained circular DNA.
% which is much larger than the radius of bacterial cells ($\simeq \SI{0.5}{\um}$ in \ecoli).

As discussed in section~\ref{sec:thermo}, a supercoiled circular DNA molecule adopts tree-like conformations, which is expected to strongly affect these results. Interestingly, by neglecting the details of this tree, such as the distribution of branch sizes, one can estimate the corresponding scaling law. Namely, scaling arguments~\cite{daoud_conformation_1981,khokhlov_array_of_obstacles_1985,gutin_annealed_rings_1993,everaers_flory_review_2017}, analytical approaches~\cite{parisi_sourlas_1981} and numerical simulations~\cite{rensburg_nonlocal_1992,cui_chen_tree_MC_1996,rosa_tree_simulation_2016,rosa_beyond_2017} have shown that the radius of gyration of self-avoiding trees scales as $L^{0.5}$. Knowing that a circular chain of $\SI{30}{kb}$ has a radius of gyration on the order of $\SI{200}{nm}$ in the plectonemic phase~\cite{walter_supercoiled_2021}, the corresponding extension for our $\SI{5}{Mb}$ long bacterial genome is equal to~\cite{cunha_polymer-mediated_2001} $200\times(5000/30)^{0.5}\simeq\SI{2.6}{\um}$. While this is a significant reduction compared to topologically unconstrained circular DNA, the corresponding volume is still 60 times larger than the volume of the cells. 

\subsection{Adding (large) molecular crowders: segregative phase separation}

The scaling arguments outlined above suggest that supercoiling alone cannot account for the formation of the nucleoid, as the unconfined resulting tree would occupy a much larger volume than the bacterial cell itself. Furthermore, these arguments do not address the specific issue of the nucleoid's location within the cell, whether it is located at the center or the quarters of the cell. One significant factor missing from these arguments is the physical nature of the cytoplasm and the potential for microcompartmentalization caused by liquid-liquid phase separation~\cite{walter_brooks_phase_separation_from_crowding_1995,Hyman_LLPS_2014}.
In particular, the nucleoid might form due to depletion interactions~\cite{Asakura_Oosawa_1954,Asakura_Oosawa_1958,Lekkerkerker2011}
%https://link.springer.com/chapter/10.1007/978-94-007-1223-2_2,
%https://aip.scitation.org/doi/abs/10.1063/1.1740347?journalCode=jcp,
%https://onlinelibrary.wiley.com/doi/abs/10.1002/pol.1958.1203312618
between the bacterial DNA and ``crowders'' contained in the cellular solvent in which it is immersed \cite{odijk_osmotic_1998,mondal_entropy-based_2011,joyeux_bacterial_2020}. 

Molecular crowders are typically identified with small, $\sim\SI{5}{nm}$ sized proteins, which are present in the cytoplasm in large concentrations.
Their presence affects the mobility of biomolecules, protein folding and stability, and the association of macromolecules with each other~\cite{van_den_Berg_crowding_homeostasis_2017} as well as the structure and stability of DNA~\cite{Miyoshi_crowding_DNA_2008}.
%https://www.nature.com/articles/nrmicro.2017.17
%https://www.sciencedirect.com/science/article/abs/pii/S0300908408000436
However, the formation of the nucleoid might owe more to the presence of larger crowders like ribosomes or  polysomes (small polymers of ribosomes connected by the messenger RNA they are sitting on)~\cite{mondal_entropy-based_2011,joyeux_bacterial_2020}.

Quite generally,  large structures (like spheres, plates or rods) can be pushed together by smaller molecules, as this reduces the total volume inaccessible to the crowders and hence maximizes their translational entropy and the total disorder in the system.  In a nutshell, the compressing forces arise, because the osmotic pressure of crowders in open spaces cannot be balanced due to their absence from inaccessible spaces.
Depletion interactions are particularly effective for rod-like particles, where nematic ordering can arise for similar reasons~\cite{frenkel_onsager_revisited_1987}
and mixtures of spheres and rod display a rich phase diagram~\cite{adams_separation_of_rods_and_spheres_1998,dogic_layering_for_spheres_and_rods_2000,urakami_spheres_and_rods_2003} as a function of their relative size and concentration. Of special interest for this review are the implications of DNA supercoiling and, in particular, the importance of the length and the stiffness of the rod-like plectonemic regions in between branch points.

Crowding-induced segregation of plectonemic DNA into a nucleoid was first invoked in 1998 for physiological concentrations of small proteins~\cite{odijk_osmotic_1998}. However, the equally predicted nematic ordering of the supercoiled DNA has never been observed. Instead, supercoiled DNA appears to mix with small crowders in {\it in vitro} experiments~\cite{gupta_compaction_2017} and even with $\SI{15}{nm}$ crowders in Brownian dynamics simulations~\cite{joyeux_bacterial_2020}. In 2011, a cell-scale model suggested that plectonemic DNA and polysomes undergo segregative phase separation, resulting in a similar phenomenon to that of the nucleoid in \ecoli: the plectonemes do not exhibit nematic ordering, and the chromosome tends to be localized in the center of the cell, with the polysomes congregating at the poles and in a thin layer between the chromosome and the cell walls~\cite{mondal_entropy-based_2011}. The simulation assumed that the crowders had a diameter of $\SI{20}{nm}$, slightly larger than in~\cite{joyeux_bacterial_2020}, and modeled the bacterial chromosome as a self-avoiding random tree with braided supercoiled DNA branches, approximately $\SI{1}{kb}$ ($\SI{200}{nm}$) in size. Notably, the branches were assumed to be {\it straight, i.e., very stiff}. In this context, the absence of nematic ordering is consistent with previous findings \cite{urakami_spheres_and_rods_2003} where mixtures of rods and spheres with similar diameters exhibited such a phenomenology for a certain concentration of the spheres. Interestingly, in a mixture of rods of spheres of different sizes, there also exists a regime where the smallest spheres freely mixed with the rods, while the largest spheres may induce the nematic ordering anticipated in \cite{odijk_osmotic_1998}.

Interestingly, in this model, the chromosome avoids the cell wall to preserve the orientational entropy of the stiff plectonemes~\cite{mondal_entropy-based_2011}. Even more remarkably, it predicted that once activated, through the physical coupling of transcription and translation (section~\ref{sec:transcription}), transcribed genes should migrate to the surface of the nucleoid. This was experimentally demonstrated a few years later using live cell super-resolution imaging~\cite{stracy_live-cell_2015}. An important question nevertheless remains: are straight plectonemes of $\SI{200}{nm}$ in size biologically relevant, knowing that their persistence length is on the order of $\SI{100}{nm}$, i.e., that they can actually bend rather easily below $\SI{200}{nm}$? Should one interpret the good agreement as indirect evidence for the association of plectonemes into stiffer bundles? If not, how would this affect the observed nucleoid phenomenology? Which additional ingredient would be necessary to add in this case? A physical coupling between part of the chromosome and the polysomes to include active genes?

Finally, recent visualization of the nucleoid in single non-dividing cells with a growing membrane have shown that a single nucleoid diffuses slowly compared to its internal dynamics, regardless of the cell length. Additionally, it diffuses slowly enough compared to the rate of cell division that it remains at the center of the cell, even when the cell becomes artificially very large~\cite{wu_cell_2019}. To understand this effect, let us first mention that experiments of \ecoli\ chromosome micromanipulation have shown that it behaves {\it in vivo} like a highly compressed spring, meaning that the pressure exerted by the cytoplasm is much greater than that required to fit the chromosome inside the cell~\cite{pelletier_physical_2012}. Thus, in a first approximation, the chromosome can be seen as a double-piston for which the cytoplasm exerts strong pressure on each side~\cite{wu_cell_2019}. As the volumes on each side of this piston contain on average equal amounts of proteins, they exert comparable pressure. Nevertheless, the proteins can pass from one side to the other through e.g.~the thin layer between the chromosome and the cell wall. The question then is to know the time scale associated with these fluctuations. An interesting insight comes from the modeling work accompanying the experiments of~\cite{wu_cell_2019}. Namely, the authors implemented molecular dynamics simulations of a brushed polymer, i.e., of a polymer composed of a (rather stiff) ring to which loops, which could be plectonemes, are attached (Fig.~\ref{fig:scaleup}). This brushed polymer was then immersed in a medium mimicking a cytoplasm crowded by ribosomes. Their results then support the idea that under these conditions, the chromosome diffuses slowly \cite{wu_cell_2019}, very likely because of rare exchanges of ribosomes between the two sides of the polymer. Accordingly, in the presence of two nucleoids, they showed that the continuous addition of ribosomes distributed equally on either side of the corresponding polymers led to a cellular arrangement with two nucleoids located at the quarters of cells.

\section{Scaling up models of supercoiled DNA}
\label{sec:scaleup}

\subsection{DNA fiber models}

Structural details of DNA can be coarse-grained, meaning that they can be summarized at the scale of a single nucleotide to build almost atomistic resolution polymer models~\cite{harris_modelling_2006,ouldridge_structural_2011,manghi_physics_2016}. However, simulating such single-nucleotide resolution models is typically limited to less than $\SI{1}{kb}$ long molecules.  
At a slightly lower resolution, rigid base~\cite{lavery_ABC_2009,gonzalez2013sequence} or base-pair~\cite{olson1998_rigid_base_pair_model,becker2006indirect_readout,lavery_ABC_2009,becker2009dna_nanomechanics} models of B-DNA allow to preserve the sequence-dependent structure and elasticity of the canonical double-helix. Further coarse-graining \cite{becker2007rigid,Maddocks_cgDNA_2014,sakaue_coarse_graining_DNA_2023} leads to tens-of-base-pairs resolution models like the self-avoiding rod-like chain model which still preserve the microscopic physics of DNA bending, DNA torsion, and self-avoiding properties (section\ref{sec:3dmodel}). So, can these single chain models be used to simulate the folding of an entire bacterial chromosome?

The question at hand is how long a simulation must run to reach thermodynamic equilibrium for a supercoiled DNA molecule -- we tacitly suppose that thermodynamic equilibrium is relevant for the large scale organization of chromosomes, which should be the case for sufficiently slow cell growth. To that end, we must consider the most effective methods for forming and equilibrating supercoiled DNA structures, which involve chain deformations that are particularly well-suited to relaxing plectonemic structures~\cite{liu_efficient_2008}. Simulations suggest that the characteristic number of iterations required to reach equilibrium in this context is of the order of the chain length ($L$)~\cite{liu_efficient_2008}. Suppose, then, that the topological constraint of the conservation of the linking number is implemented locally\cite{carrivain_silico_2014,lepage_modeling_2017} through the notion of parallel transport~\cite{bergou_discrete_2008}, which avoids to calculate the writhe of the molecule (which scales as $L^2$). In this context, simulations show that $K$ elementary Monte Carlo moves (whose subchain sizes range from 1 to $L$) take a CPU time that scales as $K\times L^{1.2}$~\cite{lepage_thermodynamics_2015}. Assuming that this time can be reduced to $K\times L$ (the exponent $1.2$ reflects the management of the self-avoiding constraint), since $L$ moves are necessary to reach equilibrium, the characteristic simulation time for the most efficient simulations should scale as $L^2$ -- note that these simulations are challenging to parallelize due to non-trivial self-avoidance constraints~\cite{krajina_large-scale_2016}.

Knowing that it takes about $5$ hours on a 3.5~Ghz processor to reach equilibrium for a chain of $\SI{20}{kb}$ when the supercoiling density is not too intense (e.g., for $\sigma=-0.03$)~\cite{lepage_thermodynamics_2015}, the time to reach equilibrium for a $\sim\SI{500}{kb}$ long genome, such as the JCVI-syn3A synthetic minimal genome for which Hi-C data is available\cite{gilbert_generating_2021}, is of the order of $5\times(500/20)^2\simeq3000$ hours, or approximately $130$ days. For \ecoli, the time is approximately $35$ years. To scale up to chromosomes, particularly those with a length of a few Mb as that of \ecoli, coarser-graining methods that neglect the details of plectonemes are necessary. In the following section, we discuss two main types of models resulting from these procedures: trees and bottle brushes.

\subsection{On trees and bottle brushes}

One way to scale up is to consider that braided structures such as plectonemes behave like self-avoiding linear polymers with, for example, an equivalent diameter of the order of $\sim 10 nm$ for $\sigma=-0.05$\cite{boles_structure_1990}. It is then possible to use a classical linear chain modeling without topological constraints (such as a wormlike chain) to address the problem of large-scale polymer folding. In this case, one must nevertheless ask how these pieces of linear chain are connected together. Two possibilities have particularly caught the attention of researchers: tree structures and bottle brush organizations (Fig.\ref{fig:scaleup}).

The tree-like structures are observed in vitro without the action of enzymes and proteins acting on DNA~\cite{boles_structure_1990} as well as in polymer simulations (see e.g.~\cite{krajina_large-scale_2016,walter_supercoiled_2021} for molecules above $\SI{30}{kb}$ in length). Tree-like models are therefore good candidates to predict behaviors at large scales, i.e.~when the details of the trees, such as the length of their branches, do not have an impact on the studied properties -- see~\cite{daoud_conformation_1981,everaers_flory_review_2017} and references therein for the physics of trees. A characteristic example is the behavior of the average contact frequency between loci as a function of genomic distance ($s$), generally called the ``contact law'' and denoted by $P(s)$~\cite{mirny_fractal_2011}. Specifically, in situations of {\it high polymer concentrations}, simulations of trees lead to contact laws of the form $P(s)\sim s^{-1.1}$~\cite{rosa_conformational_2019}. Interestingly, this law seems to be compatible with observations in very different bacteria, namely \Ccres~\cite{le_high-resolution_2013}, \ecoli~\cite{lioy_multiscale_2018}, \Pseudo~\cite{varoquaux_computational_2022}, or {\it Streptomyces}~\cite{lioy_dynamics_2021}. A thorough study of the relevance, or not, of tree-like models for contact properties at various scales is ongoing~[Ghobadpour et al., in prep]. 

Remarkably, $P(s)\sim s^{-1.1}$ is actually also compatible with large-scale contact properties of chromosomal loci in  several eukaryotes such as Human~\cite{lieberman-aiden_comprehensive_2009}. While it is tempting to ascribe this here as well to DNA supercoiling known to occur in eukaryotes~\cite{Giaever_supercoiling_in_eukaryotes_1988,Corless_effects_2016,Corless_investigating_2017},
the commonly invoked explanations of crumpling~\cite{cremer_territories_2001,grosberg_crumpled_DNA_1993,rosa_interphase_chromosomes_2008,lieberman-aiden_comprehensive_2009,mirny_fractal_2011,halverson_melt_2014}
and active loop extrusion~\cite{goloborodko_compaction_2016} also lead to double-folded branching structures~\cite{khokhlov_array_of_obstacles_1985,grosberg_ring_melt_2014,rosa_crumpling_2014,rosa_tree_melt_2016,everaers_flory_review_2017,rosa_conformational_2019}.
Note, also, that the high concentration nature of the polymers for bacteria is a consequence of an {\it in vivo} concentration to be considered that is not that of DNA, which is a few percent, but that of plectonemes bound by multiple proteins. Namely, a rough calculation assuming beads of diameter $\SI{30}{nm}$ (consisting of $\sim 10$ nm in diameter and $\sim 20$ nm of protein complexes) with $\SI{200}{bp}$ per bead results in a volumetric fraction of beads of approximately $0.25$ for a $\SI{5}{Mb}$ genome folded within a nucleoid with a cross-section of $\SI{800}{nm}$ and a length of $\SI{1}{\um}$.


Regarding the organization in bottle brush, it should be mentioned firstly that based on biochemical and biophysical analyses of nucleoids extracted from cells, a rosette structure was predicted 50 years ago for the \ecoli\ chromosome~\cite{worcel_structure_1972}. In this structure, long plectonemes (of approximately $\SI{100}{kb}$) emanate from a central core made of proteins and RNA. This structure was later confirmed by electron microscopy observations of nucleoids extracted from cells~\cite{kavenoff_electron_1976}. However, {\it in vivo} evidence for such a rosette structure has remained elusive so far. Interestingly, recent experiments in which DNA replication and cell growth were decoupled led to widened cell geometries inside which a toroidal geometry of the circular chromosome of \ecoli\ could be clearly identified~\cite{wu_direct_2019}. This structure is compatible with a circular bottle brush polymer model, which is a polymer model made of a circular backbone along which plectonemes are attached (Fig.~\ref{fig:scaleup}B).

Interestingly, the chromosome of \ccres\ has been modeled using such a bottle brush polymer model in order to provide a rationale for the patterns observed in the first bacterial Hi-C data produced 10 years ago~\cite{le_high-resolution_2013}. In this model, the plectonemes were stochastic structures whose length was adjusted along with $5$ other parameters (such as the stiffness of the plectonemes or their distance along the backbone) to reproduce the Hi-C data. Interestingly, the plectonemes in the obtained model had an average length of 15 kb, which is compatible with the length of topologically independent domains predicted to partition bacterial genomes (see section~\ref{sec:3dmodel}). Furthermore, the introduction of plectoneme-free zones blocking the diffusion of plectonemes allowed for the reproduction of the phenomenology of so-called chromosome interaction domains, or CIDs, inside which interaction between any pair of loci is enhanced compared with external loci located at a similar genomic distance~\cite{le_high-resolution_2013}. Finally, some of the large-scale conformations of this model adopted a loose helix conformation, a property that has been reported for the \ecoli\ chromosome~\cite{hadizadeh_yazdi_variation_2012,fisher_four-dimensional_2013}. This is in contrast to the early data-driven ``models'' of \ccres\ chromosomes presenting a marked helix~\cite{umbarger_three-dimensional_2011} but whose origin was not physical, as demonstrated in a version including the fundamental concept of entropy~\cite{messelink_learning_2021}. Note also that several theoretical studies have been carried out on these bottle brushes, highlighting helical structures in a regime where the backbone persistence length is at least of the order of the cell diameter~\cite{chaudhuri_spontaneous_2012,jung_confinement_2019}. The relevance of this hypothesis for {\it in vivo} situations remains to be demonstrated. Finally, it is noteworthy that the bottle brush polymer model, which was developed for \ccres~\cite{le_high-resolution_2013}, has recently inspired a data-driven approach aimed at creating a three-dimensional representation of the current knowledge on the structuring of bacterial chromosomes~\cite{hacker_features_2017}.

\subsection{On-lattice models}

The simulation of tree-like models is commonly performed on a lattice. Lattice simulations are preferred due to the ease of managing discrete elementary movements as compared to continuous movements involved in off-lattice approaches. This leads to higher efficiency of lattice simulations. In fact, lattice simulations are particularly suitable when the properties under study occur on a much larger scale than the lattice mesh, i.e., when the properties studied do not depend on the geometric parameters of the lattice. The possibility of performing non-local movements, such as cutting a branch at one point and randomly reintroducing it at another point~\cite{SeitzKlein1981,rensburg_nonlocal_1992,rosa_tree_simulation_2016,rosa_tree_melt_2016}, or the exchange sections of overlapping chains~\cite{Theodorou2002PhysRevL} or trees are particularly effective on a lattice~\cite{svaneborg2016multiscale,svaneborg2023multiscale} and allow to reach thermodynamic equilibrium very efficiently. Elastic chain methods on the lattice are also very effective for exploring polymer dynamics in situations where the polymer concentration is very high~\cite{EvansEdwards1981-Part1,Barkema.Book1999,Barkema2003The.JofCh.Phy,Kolb2009Macromolecules,Schram2013TheJ.ofCh.Ph}. In a nutshell, the principle is based on the ability to redistribute monomers along a given spatial conformation, with several consecutive monomers being able to overlap. This then allows for the exploration of new conformations that would be inaccessible without this prior redistribution. Finally, simulation techniques can be adopted to reproduce realistic dynamic properties of polymers~\cite{ghobadpour_monte_2021}, as well as simulate active processes such as the action of condensins~\cite{miermans_lattice_2020}.

To our knowledge, no work has reported on the properties of a lattice-based physical model that covers the multiple scales of the bacterial chromosome. However, it is worth mentioning that a computational method has been developed to efficiently construct lattice-based conformations of a bottle brush polymer with a backbone to which plectonemes are attached~\cite{goodsell_lattice_2018}. The plectoneme modeling used in this study bears resemblance to the double-folding lattice polymer models, where linear chains fold back on themselves to form overlapping double-chain structures. In this regard, we believe that the range of methods developed in this specific area of polymer physics  should allow for a precise and quantitative analysis of the physical nature of bacterial chromosomes. In particular, an important open question to us is the folllowing: is it possible to distinguish between tree-like and bottle brush-type phenomenologies based solely on contact properties between loci as provided by Hi-C data, knowing that the latter can be generated in principle for any type of bacteria cultivable in the laboratory~\cite{marbouty_generation_2017}.

\begin{figure}[t]
\floatbox[{\capbeside\thisfloatsetup{capbesideposition={right,top},capbesidewidth=0.35\linewidth}}]{figure}[\FBwidth]
{\caption{Two types of polymer models including the effects of supercoiling can be contemplated to study the large-scale structure of bacterial chromosomes: A) tree-like models where plectonemes are abstracted by simple linear branches (right panel). B) bottle brush models where plectonemes are attached along a ring or backbone, indicated in blue. This model is therefore composed of two {\it a priori} independent entities and, hence, is more complex than the tree-like model. At large scales, the details of these entities can nevertheless be discarded (right panel). It should be noted that if the bottle brush structure is relevant {\it in vivo}, as suggested by chromosome visualization data in \ecoli~\cite{wu_direct_2019}, the mechanisms of its formation remain an open question.}\label{fig:scaleup}}
{\includegraphics[width=\linewidth]{scaleup.pdf}}
\end{figure}

\section{Discussion}

\paragraph*{The need for coarse-grain models.} In principle, molecular dynamics simulations of atomistic models~\cite{Karplus_McCammon_biomolecular_simulation_2002,brooks_CHARMM_2009,case2021amber} are ideal tools for studying the complexities of biomolecular systems. With steady advances in available computer power and the performance of employed codes~\cite{shaw2014anton,abraham2015gromacs,eastman2017openmm,phillips2020scalableNAMD,plimpton_LAMMPS_2022}, they provide an ever more powerful ``computational microscop''~\cite{schulten_computational_microscope_2009,shaw_computational_microscope_2012} into biomolecular structures and processes. Of particular interest for this review is their ability to help rationalize the structural diversity of supercoiled DNA molecules~\cite{mitchell_atomistic_supercoiled_DNA_2011}, which has been found to be more significant than initially thought~\cite{Irobalieva_supercoiled_DNA_2015}. However, extending the domain of application of molecular dynamics simulations, which currently concern molecules of a few hundred base pairs, to the bacterial scale is not feasible. Namely, using a single GPU for a system composed of $10^6$ atoms, one can currently simulate on the order of 10 nanoseconds per day. While this allows reaching the microsecond scale in $100$ days, simulating an entire $5$ Mb long bacterial genome, which comprises on the order of $10^9$ atoms, over biologically relevant time scales remains elusive. For instance, simulating a 100-minute-long cell cycle would require the time elapsed since the extinction of the dinosaurs. Coarse-grained models~\cite{Saunders_Voth_coarse_graining_2013,dans_multiscale_2016,jewett_Moltemplate_2021}, which consist of dropping fine details below a given resolution to build simpler descriptions that capture properties above this resolution, are thus inevitably needed to rationalize and predict the structuring properties of DNA {\it in vivo}.

\paragraph*{Caveats.} The situation just described also neglects the interactions of DNA with proteins and molecular machines, as well as with all the small molecules and ions that make up the cytoplasm. Although coarse-grained models of (supercoiled) DNA have proven successful in single-molecule experiments, it is therefore reasonable to question how well these models capture the behavior of DNA in a living cell. Even more worrying for a rational approach to the phenomena at play, proteins and molecular machines often have their own specificity that arises from the molecular tinkering induced by natural selection~\cite{jacob_evolution_1977}. Many of their properties therefore escape the universality feature of physical phenomena.

\paragraph*{Statistical universality and ubiquitous phenomena.} The relevance of coarse-grained models actually arises from two realities. First, in many situations, the conditions are equivalent to those of a system with a large number of particles or in the limit of a very large size of the entities involved. In this case, statistical physics approaches become relevant. For example, while the plectonemic structure of supercoiled DNA may be a simplifying average view of the dynamics of DNA interacting with many proteins and molecular machines, this average behavior becomes probably relevant at much larger scales, such as the chromosome, and a tree-like description of the problem should capture a good part of the associated phenomena. Second, evolutionary conserved phenomena are often associated with generic physical properties~\cite{junier_conserved_2014}. For instance, the double helix nature of DNA necessarily creates topological problems that require dedicated enzymes to resolve. This has two consequences: first, topoisomerases are ubiquitous in living organisms; and second, generic physical models for handling topological constraints can be considered, regardless of the mechanisms involved. Variations in behavior between bacteria should then reflect the possible range of physiologically relevant parameters. In all cases, proposed physical models should be evaluated not only for their descriptive (i.e., postdictive) capacity but also, and perhaps most importantly, for their predictive power.

\bibliography{bib}

\end{document}