%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                     file: 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\documentclass[aps,twocolumn,nofootinbib]{revtex4}
% \documentclass[aps,twocolumn]{revtex4}
% \documentclass[aps,nofootinbib]{revtex4}
\usepackage{adjustbox}
\usepackage{graphicx} 
\usepackage{caption,subcaption}
\captionsetup{justification=raggedright,singlelinecheck=false}
\usepackage{amsmath}
\usepackage{amsthm}
\usepackage{framed}

%\usepackage{boondox-cal}
\usepackage{comment}
\usepackage{bbm}
\usepackage[version = 4]{mhchem}
\usepackage{tikz-cd} 
\usepackage{tikz}
\usepackage{amssymb}
\usepackage{mathtools}
\newtheorem{theorem}{Theorem}[section]
\newtheorem{corollary}{Corollary}[theorem]
\newtheorem{lemma}[theorem]{Lemma}
\newtheorem{proposition}[theorem]{Proposition}
\newtheorem{remark}[theorem]{Remark}
\newtheorem{example}[theorem]{Example}


\usepackage{physics}
\usepackage{xcolor}
%\renewcommand{\thealgorithm}{}
% \usepackage{ulem}
\usepackage{bm}
\usepackage{mathtools}
\usepackage[hidelinks]{hyperref}
%\usepackage{algorithm}
\usepackage{algpseudocode}
\usepackage{xcolor}
\usepackage{xfrac}
\usepackage{bm}
\usepackage[T1]{fontenc}
\usepackage{upquote}\newcommand{\vn}{\varnothing}
\DeclareMathOperator{\Ima}{im}
\DeclareMathOperator{\ord}{\mathcal{O}}
%\DeclareMathOperator{\var}{\overline{}}
\DeclareMathOperator{\ch}{ch}
\DeclareMathOperator{\sh}{sh}
\DeclareMathOperator{\Th}{th}
\DeclareMathOperator{\diag}{diag}
\DeclarePairedDelimiter\ceil{\lceil}{\rceil}
\DeclarePairedDelimiter\floor{\lfloor}{\rfloor}
%\DeclareMathOperator{\Tr}{Tr}
\DeclareMathOperator{\supp}{supp}
\newcommand{\rn}{{\rm n}}
\newcommand{\rx}{{\rm x}}
\usepackage[utf8]{inputenc}
\usepackage[english]{babel}
\newcommand{\dsum}{\displaystyle\sum}
\usepackage[lined,boxed]{algorithm2e}


\begin{document} 


\title{The geometry and combinatorics of an autocatalytic ecology in chemical and cluster chemical reaction networks}

%\author{Praful Gagrani$^{1,3}$, David Baum$^{2,3}$, Eric Smith$^{4,5,6,7}$}
\author{Praful Gagrani$^{1,2}$, Victor Blanco$^{2,3,4}$, Eric Smith$^{1,5,6,7,8}$, David Baum$^{2,9}$}

%\begin{comment}
\affiliation{$^1$Department of Physics, University of Wisconsinâ€“Madison, Madison, WI 53706, USA\\
$^2$Wisconsin Institute for Discovery, University of Wisconsin-Madison, Madison, WI 53715, USA\\
$^3$Institute of Mathematics (IMAG), Universidad de Granada, Spain\\
$^4$ Department of Quant. Methods for Economics \& Business, Universidad de Granada, Spain\\
$^5$ Department of Biology, Georgia Institute of Technology, Atlanta, GA 30332, USA\\
$^6$ Earth-Life Science Institute, Tokyo Institute of Technology, Tokyo 152-8550, Japan\\
$^7$ Santa Fe Institute, 1399 Hyde Park Road, Santa Fe, NM 87501, USA\\
$^8$ Ronin Institute, 127 Haddon Place, Montclair, NJ 07043, USA\\
$^9$ Department of Botany, University of Wisconsin-Madison, Madison,
WI 53706, USA}
%\end{comment}
\date{\today}
\begin{abstract}
Developing a mathematical understanding of autocatalysis in chemical reaction networks has both theoretical and practical implications. For a class of autocatalysis, which we term \textit{stoichiometric autocatalysis}, we show that it is possible to classify them in equivalence classes and develop mathematical results about their behavior. We also provide a linear-programming algorithm to exhaustively enumerate them and a scheme to visualize their polyhedral geometry and combinatorics. We then define cluster chemical reaction networks, a framework for coarse-graining realistic chemical reactions using conservation laws. We find that the list of minimal autocatalytic subnetworks in a maximally connected cluster chemical reaction network with one conservation law grows exponentially in the number of species. We end our discussion with open questions concerning autocatalysis and multidisciplinary opportunities for future investigation.
\end{abstract}
\keywords{Chemical reaction networks}
\maketitle
\begingroup
  \hypersetup{hidelinks}
  \tableofcontents
\endgroup

\section{Introduction}
Chemical reaction network (CRN) theory offers a versatile mathematical framework in which to model complex systems, ranging from biochemistry and game theory to the origins of life \cite{veloz2014reaction,smith2016origin}. The usefulness of CRNs in modelling these phenomena stems from its ability to exhibit a wide range of nonlinear dynamics and it is widely recognized that \textit{autocatalysis} can be seen as a basis for many of them \cite{epstein1998introduction,schuster2019special}. Broadly, autocatalysis is framed as the ability of a given chemical species to make more copies of itself or otherwise promote its own production. Thus, from the perspective of kinetics, one would expect autocatalysts to be able to show super-linear growth \cite{hordijk2004detecting,andersenadefining}. For some examples of nonlinear kinetics obtained by composing two autocatalytic cycles and their relevance to ecological dynamics, see Fig.\ \ref{fig:Peng_summary}.    



\begin{figure}
    \centering
    \includegraphics[width = \linewidth]{figures/peng_sumary.png}
    \caption{Ecologies obtained by composing autocatalytic cycles. The member species of each autocatalytic cycle are shown to consume food and produce waste resources. Two cycles can be composed in several ways such as by sharing food (line one), the member species of one can be the food of another (line two), or the waste of one can deplete the food of another (line three). For a detailed account of such interactions and their ecological counterpart, see \cite{peng2020ecological}. }
    \label{fig:Peng_summary}
\end{figure}



A comprehensive understanding of autocatalysis in CRNs has great practical values, but its full mathematical treatment remains to be developed. In \cite{andersenadefining}, it is conjectured that it is impossible for a CRN to show a temporary speed-up of the reaction before settling down to reach an equilibrium if, at least, it is not \textit{formally} autocatalytic. Also, it is suggested in \cite{deshpande2014autocatalysis} that understanding the interactions between autocatalytic and drainable subnetworks might provide a basis for obtaining a proof for the long-standing \textit{persistence conjecture} \cite{feinberg1989necessary}. 

In recent work \cite{blokhuis2020universal}, Blokhuis et al.\ defined a notion of autocatalysis strictly in terms of the stoichiometric matrix without referring to the underlying CRN. While their notion is more restrictive than the notion of autocatalysis discussed in \cite{andersenadefining,deshpande2014autocatalysis}, it permits richer mathematical analysis. Using their stoichiometric criterion for autocatalysis, Blokhuis et al.\ show that there are only five types of minimal autocatalytic motifs. They also found that autocatalysis is more abundant than previously thought, and simple reaction networks with a few reactions can contain very many minimal autocatalytic motifs in them. 

In this paper, we extend the definition of Blokhuis et al.\ to subnetworks, and provide tools for exhaustively enumerating autocatalytic subnetworks and understanding their combinatorics. The layout of the paper is as follows. In Sec.\ \ref{sec:CRN_AC}, we briefly review chemical reaction networks (CRNs) and a few different notions of autocatalysis. In Sec.\ \ref{sec:organization}, we develop a formalism for analyzing minimal autocatalytic subnetworks (MASs) that are detectable from the stoichiometric matrix and provide a linear-programming algorithm to exhaustively enumerate them. We also propose a visualization scheme in which to represent MAS combinatorics and geometry. In Sec.\ \ref{sec:applications}, we present the cluster chemical reaction network (CCRN) framework and argue that it is provides a natural coarse-graining of realistic chemical and biochemical reaction networks. We then explore the combinatorics of MASs in CCRN framework and provide a worked-out example to demonstrate the geometry of autocatalysis in the stoichiometric subspace of species concentration. Multiple examples from the CCRN framework are also spread throughout this work. We conclude, in Sec.\ \ref{sec:discussion}, with an overview of our contributions and possible avenues for future investigation. 


%----------------section end -----------------------------
\section{Chemical reaction networks and autocatalytic cycles}
\label{sec:CRN_AC}
The mathematical theory of chemical reaction networks (CRNs), pioneered by Horn, Jackson, and Feinberg \cite{horn1972general,feinberg2019foundations,yu2018mathematical}, has ubiquitous applications in understanding nature. In particular, for biological and ecological applications, the formalism allows for a notion of self-replication or \textit{autocatalysis} where certain species increase the population of these same species. Consequently, there is value in rigorously understanding the organization of a collection (\textit{ecology}) of autocatalytic cycles.   

In \cite{feinberg2019foundations}, Feinberg defines a CRN as a triple of the set of species, complexes, and reactions $(\mathcal{S},\mathcal{C},\mathcal{R})$, which we review in Appendix \ref{app:deficiency}. It has been argued in \cite{deshpande2014autocatalysis}, however, that the data of the set $\mathcal{C}$ of complexes is redundant as it can be recovered from the reaction set $\mathcal{R}$ and, thus, a CRN can be defined more economically as $(\mathcal{S},\mathcal{R})$. Since the two representations are equivalent, we will pick either, dictated by convenience. The concept of autocatalysis, dating back to the late 1800s \cite{ostwald1890autokatalyse,peng2022wilhelm}, has several distinct formulations in CRN theory, for a review of which we refer the readers to Andersen et al.\ \cite{andersenadefining}. In our work, we closely follow the formulation of Blokhuis et al.\ \cite{blokhuis2020universal}, while acknowledging its relationship to other formulations of autocatalysis, as shown in Sec.\ \ref{sec:AC-motif}.  



\subsection{Hypergraphs, stoichiometric matrix, conservation laws, and null flows}
\label{sec:hypergraphs}

A \textit{chemical reaction network} (CRN) is defined by the pair $(\mathcal{S},\mathcal{R})$ where $\mathcal{S}$ and $\mathcal{R}$ denote the finite sets of species and reactions, respectively. 
\begin{align*}
    \mathcal{S} &= \{ s_1, \ldots, s_{\mathcal{S}} \} \\
    \mathcal{R} &= \{ \ce{$y$ -> $y'$}  \}
\end{align*}
The \textit{reaction hypergraph} of the CRN $(\mathcal{S},\mathcal{R})$ is the directed graph with \textit{complex} set $\mathcal{C}:= \{y,y' | y\to y'\in \mathcal{R}\}$ and edge set $\mathcal{R}$. The complexes, are multisets of \textit{species} (vertices), and are denoted by a column vector $y$ representing the stoichiometry or multiplicity of each species in the complex. Note that this is purely topological data, and no reference is made to the kinetics or dynamics of a system specified by the CRN.    

Let (hyper)graph $\mathcal{G} = (\mathcal{S},\mathcal{R})$ be a CRN. Then the \textit{stoichiometric subspace} $S$ is defined as the linear span of the reaction vectors $\{y'-y|y \to y' \in \mathcal{R}\}$. The \textit{stoichiometric matrix} $\mathbb{S}$ is defined as the matrix whose columns are the reaction vectors, 
\begin{align}
    \text{cols}(\mathbb{S}) &= \{y'-y|y\to y' \in \mathcal{R}\}. \label{eq:stoich_mat}
\end{align}


We denote the restriction of the rows of the stoichiometric matrix $\mathbb{S}$ to a species set $\mathcal{S}'\subset \mathcal{S}$ by $\mathbb{S}_{\mathcal{S}'}$, and the restriction of the columns of $\mathbb{S}$ to a reaction set
$\mathcal{R}' \subset \mathcal{R}$ by $\mathbb{S}^{\mathcal{R}'}$. In general, for a subgraph $\mathcal{G}' = (\mathcal{S}',\mathcal{R}')\subset \mathcal{G}$, we denote the restriction of $\mathbb{S}$ to $\mathcal{G}'$ by $\mathbb{S}|_{\mathcal{G}'} := \mathbb{S}^{\mathcal{R}'}_{\mathcal{S}'}$. 

We define a \textbf{subnetwork} of a graph $\mathcal{G}$ to be the subgraph $(\mathcal{S}|_{\mathcal{R}'},\mathcal{R}')$ where $\mathcal{R}' \subset \mathcal{R}$ and there is no omission of species in the subnetwork that participates in the set of reactions $\mathcal{R}$ of the original graph $\mathcal{G}$. 

Let vector $\textbf{n} = [n_1, \ldots, n_\mathcal{S}]^T \in \mathbb{Z}_{\geq 0}^\mathcal{S}$ ($\in \mathbb{R}_{\geq 0}^\mathcal{S}$) denote the population (concentration) of each species in the system, and $\textbf{v} = [v_1, \ldots, v_\mathcal{R}]^T$ denote a (hyper)flow on $\mathcal{G}$. Then the change of species population (concentration) $\Delta \textbf{n}$ resulting from the flow $\textbf{v} \in \mathbb{Z}_{\geq 0}^\mathcal{R}$ ($\textbf{v} \in \mathbb{R}_{\geq 0}^\mathcal{R}$ ) on the graph $\mathcal{G}$ is given by 
\begin{align*}
    \Delta \textbf{n} &= \mathbb{S} \cdot \textbf{v}.
\end{align*}
The span of the columns of the stoichiometric matrix is termed as the \textit{stoichiometric subpsace}, and is the subspace of the change in species concentration under an arbitrary flow on $\mathcal{G}$. 

A flow on a graph can also be used to create a linear combination of the reactions, also called a composite reaction or $\mathcal{G}-$dilution in the terminology of \cite{andersenadefining} or \cite{deshpande2014autocatalysis,cardelli2011strand}, respectively. For example, a flow $\textbf{v}$ on graph $\mathcal{G}$ would result in the composite reaction
\begin{align*}
    \sum^{\mathcal{R}}_{r=1} v_r    y_r &\to \sum^{\mathcal{R}}_{r=1} v_r y'_r. 
\end{align*}


The left null space of $\mathbb{S}$, i.e.\ vectors $\textbf{x}$ such that $\textbf{x}\cdot\mathbb{S} = 0$, correspond to \textit{conservation laws} $$\textbf{x} \cdot \Delta \textbf{n} = 0.$$ Equivalently, the inner product of the conservation law with the population $\textbf{x} \cdot \textbf{n}$ is an invariant for the network under an arbitrary flow. The vector $\textbf{x}$ will be called a \textit{positive conservation law} if all its coefficients are nonnegative. 

Similarly, the right null space of $\mathbb{S}$, i.e. vectors $\textbf{w}$ such that $\mathbb{S}\cdot \textbf{w} = 0$, correspond to \textit{null flows} which leave the population (concentration) of the species unchanged ($\Delta \textbf{n} = 0$). A careful accounting of the right null space of $\mathbb{S}$ provides a topological classification of a CRN, which we review in Appendix \ref{app:deficiency}. 


\begin{figure*}[t]
\centering
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=0.8\linewidth]{figures/E_graph_example_2_1.png}
  %\caption{E-graph}
  
\end{subfigure}%
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=.8\linewidth]{figures/flow_example_2_1.png}
  %\caption{Change in concentration}
  
\end{subfigure}
\caption{In the left panel, the Euclidean embedded graph (E-graph, see \cite{craciun2019polynomial}) of the CRN in Example \ref{eg:CRN-1} is shown. The E-graph is obtained by representing the complexes as lattice points in a Euclidean lattice and representing the reactions as edges between them. The flows on the reactions are labelled by $\kappa$ and the change in concentration they induce, $\Delta n = \mathbb{S} \kappa$, is schematically represented in the right panel. }
\label{fig:test}
\end{figure*}




\begin{example}
\label{eg:CRN-1}
Consider the CRN $\mathcal{G} = (\mathcal{S},\mathcal{R})$ given as
\begin{align*}
    \ce{A + B <=> 2 A},\\
    \ce{A + B -> 2B}.
\end{align*}
For a visual representation of this example, see Fig.\ \ref{fig:test}. Then the species set $\mathcal{S} = \{ A, B\}$ and the reaction set $\mathcal{R} = \{ A + B \to 2 A , 2A \to A + B ,A + B \to 2B \}$. As can be read from $\mathcal{R}$, the complex set $\mathcal{C} = \{A + B, 2A, 2B \}$, and their corresponding vectors are  
\[y_{A+B}=\begin{bmatrix}
    1\\
    1
\end{bmatrix}, 
y_{2A} = \begin{bmatrix}
    2\\
    0
\end{bmatrix},
y_{2B} = \begin{bmatrix}
    0\\
    2
\end{bmatrix}.
\]
The stoichiometric matrix $\mathbb{S}$ for $\mathcal{G}$ is 
\[ \mathbb{S} = \begin{bmatrix}
    \phantom{-}1 & -1 & -1\\
    -1 & \phantom{-}1 &  \phantom{-}1
\end{bmatrix},
\] with left null space (conservation laws) spanned by $\textbf{x} = [1, 1]$ and right null space (null flows) spanned by the basis $\{ [1, 1, 0 ]^T, [1, 0, 1 ]^T\}$. Notice that the stoichiometric subspace is one-dimensional and perpendicular to the conservation law. 
\end{example}

%------------------
\subsection{Formal, exclusive and stoichiometric autocatalysis, and autocatalytic cores}
\label{sec:AC-motif}

\subparagraph{Formal autocatalysis:} Following the formulation in \cite{king1978autocatalysis}, Andersen et al.\ in \cite{andersenadefining}, define a CRN $\mathcal{G}$ to be \textit{formally autocatalytic} in the set $\mathcal{M}$ if: 
\begin{enumerate}
    \item there exists a flow on $\mathcal{G}$ such that the resulting composite reaction is of the form
    \begin{align*}
        \text{(F)} + \textbf{m} \mathcal{M} \to \textbf{n} \mathcal{M} + \text{(W)}, 
    \end{align*}
    where $\textbf{0} \prec \textbf{m} \prec \textbf{n}$\footnote{$A \prec B$, read as \textit{A precedes B}, if all entries of $A-B$ are strictly negative. Analogously, $A \succ B$, read as \textit{A succeeds B}, if all entries of $A-B$ are strictly positive.}. Here $\textbf{m}$ and $\textbf{n}$ are the stoichiometries of the set $\mathcal{M}$ in the input and output of the composite reaction, respectively, and $\textbf{o} \mathcal{M} = \sum_i \textbf{o}_i \mathcal{M}_i$. 
\end{enumerate}

\begin{example}
    \label{eg:formal_autocat}
    Consider the reaction network $\mathcal{G}_f = (\{A,B\},\{A \to B, A + B \to  B \})$ under the flow $\textbf{v} = [1,1]^T$. The resulting composite reaction is 
    \begin{align*}
        2A + B &\to 2B.
    \end{align*}
    Thus the reaction network is formally autocatalytic for the set $\mathcal{M} = \{B\}$.
\end{example}



\subparagraph{Exclusive autocatalysis:} In \cite{andersenadefining}, Andersen et al.\ define a CRN $\mathcal{G}$ to be \textit{exclusively autocatalytic} in the set $\mathcal{M}$ if:  
\begin{enumerate}
    \item there exists a flow on $\mathcal{G}$ such that the population of each species in the set $\mathcal{M}$ strictly increases.
    \item for every reaction $y \to y' \in \mathcal{G}$, the input complex $y$ and the output complex $y'$ contain at least one species from the set $\mathcal{M}$. This ensures that the flow is inadmissible, or there is no flow, if the population of any species in the set $\mathcal{M}$ is zero. 
\end{enumerate}
While the first condition is identical, it is the second condition that makes exclusive autocatalysis a more restrictive notion than formal autocatalysis.
\begin{example}
\label{eg:CRN-exclusive}
    The CRN $\mathcal{G}_f$ in Example \ref{eg:formal_autocat} is not exclusively autocatalytic since the flow proceeds even if the population of $B$ is $0$. That is because the reaction $A\to B$ does not require the presence of $B$ to proceed.\\    
   However, in Example \ref{eg:CRN-1}, $\mathcal{G}$ is exclusively autocatalytic in the set $\mathcal{M} = \{A\}$ since:
   \begin{enumerate}
       \item the flow $\textbf{v}=[1,0,0]^T$ increases the population of $A$, $\textbf{n}_A$, by $1$ if $\textbf{n}_A>0$.
       \item the reaction $A+B \to 2 A$ requires the presence of at least one $A$ and cannot occur if $\textbf{n}_A=0.$
   \end{enumerate}
   Similarly, one can show that $\mathcal{G}$ is also exclusively autocatalytic in the set $\mathcal{M} = \{B\}$.
\end{example}

Note that this definition is consistent with an intuitive notion of autocatalysis and is cognate with those of prior works \cite{deshpande2014autocatalysis,gopalkrishnan2011catalysis,barenholz2017design}. 



\subparagraph{Stoichiometric autocatalysis:} In \cite{blokhuis2020universal}, Blokhuis et al.\ generalize the notion of autocatalysis for subgraphs rather than complete networks. They define a \textit{motif} to be any subgraph $\mathcal{G}'$ of a CRN, or equivalently any possible submatrix $\mathbb{S}|_{\mathcal{G}'}$ of the stoichiometric matrix $\mathbb{S}$. Then, an \textit{autocatalytic motif} is defined to be a subgraph $\mathcal{A} = (\mathcal{A}_S, \mathcal{A}_R)$ such that:
\begin{enumerate}
    \item There exists a nonnegative flow $\textbf{g}$ on $\mathcal{A}$ such that the change in concentration of each species in the set $\mathcal{A}_S$ is strictly positive or  $\Delta \textbf{n}_{\mathcal{A}_S} = \overline{\mathbb{S}} \cdot \textbf{g} \succ 0$.
    \item For every reaction $y \to y' \in \mathcal{A}_R$, the complexes $y$ and $y'$ contain at least one \textbf{distinct} species from the set $\mathcal{A}_S$. Equivalently, each column in $\overline{\mathbb{S}}:=\mathbb{S}|_{\mathcal{A}}$ has at least one positive and one negative coefficient.
\end{enumerate}
Notice that the first condition is identical to that of \textit{exclusive autocatalysis}. However, the second condition, also termed \textit{autonomy} of submatrix $\overline{\mathbb{S}}$, ensures that the production of any species \textit{consumes} at least one other species of the motif, disallowing unconditional growth. The condition that there needs to be at least one autocatalytic species in the reactants distinct from the products makes this definition more restrictive than exclusive autocatalysis, due to which we term their criterion as \textbf{stoichiometric autocatalysis}. A useful feature of stoichiometric autocatalysis is that it can be inferred from the stoichiometric matrix without referring to the underlying CRN. In the rest of our work we will adhere to this criterion and will use \textit{autocatalysis} to mean stoichiometric autocatalysis, unless specified otherwise. 

Following the terminology in \cite{deshpande2014autocatalysis}, if there exists a nonnegative flow such that $\overline{\mathbb{S}} \cdot \textbf{g} \prec 0$, we would term the subgraph as a \textit{drainable motif}, where the change in concentration of all the species in the subset is strictly negative. While we focus our attention on autocatalytic motifs, the results in our paper are easily extended to their drainable counterparts. In \cite{deshpande2014autocatalysis}, the authors prove that a network is persistent if it has no drainable motifs and also remark on the importance of autocatalytic and drainable motifs for understanding the dynamics of a network in ecological terms. 


\begin{figure*}[t]
\centering
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=0.9\linewidth]{figures/graphics_4cluster_1.png}
  %\caption{E-graph}
  
\end{subfigure}%
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=.9\linewidth]{figures/Fig_IntersectionPartitionCones_L4.png}
  %\caption{Change in concentration}
  
\end{subfigure}
\caption{In the left panel, all minimal autocatalytic subnetworks (MASs) for the complete 1-constituent CCRN with $L=4$ are shown. The core types are also color coded using the typology of Blokhuis et al.\ in \cite{blokhuis2020universal}. In a CCRN, each species is denoted by a vector of positive integers with an overbar, and each reaction satisfies certain conservation laws. For a definition of a CCRN, see Sec.\ \ref{sec:applications}. The list of reactions can be found in Appendix \ref{sec:MAS_list} Table \ref{t:react4}. In the right panel, the list of the FWMC classes with the information of their intersection is shown.}
    \label{fig:cluster_n4}        
\end{figure*}



\subparagraph{Autocatalytic core:} An \textit{autocatalytic core} is a minimal autocatalytic motif that does not contain a smaller autocatalytic motif. In Ref.\ \cite{blokhuis2020universal}, it is shown that if $\mathcal{A} = (\mathcal{A}_C, \mathcal{A}_R)$ is an autocatalytic core, it has the following properties:
\begin{enumerate}
    \item $\overline{\mathbb{S}}$ is a square and invertible matrix, and the number of reactions $|\mathcal{A}_R|$ must be equal to the number of autocatalytic species in the core $|\mathcal{A}_C|$. 
    \item Every forward reaction in $\mathcal{A}_R$ contains only one species from the core set $\mathcal{A}_C$ (possibly with stoichiometry higher than one).
\end{enumerate}
Furthermore, the authors provide a geometric characterization of the autocatalytic cores and show that they can only be of five types. We label the autocatalytic subnetworks in Fig.\ \ref{fig:cluster_n4} using their taxonomy, and for details we refer the readers to \cite{blokhuis2020universal}. 

\begin{example}
\label{eg:CRN-extension}
    Recall from Example \ref{eg:CRN-1}, the stoichiometric matrix was given by
\[ \mathbb{S} = \begin{bmatrix}
    \phantom{-}1 & -1 & -1\\
    -1 & \phantom{-}1 &  \phantom{-}1
\end{bmatrix}.
\] 
Notice that there is no autocatalytic core in this network, and thus a naive characterization by the Blokhuis et al.\ criteria would regard the network to be non-autocatalytic. On the other hand, from Example \ref{eg:CRN-exclusive}, we know that the network is exclusively autocatalytic. To remedy this, we modify the network by replacing each reaction whose reactant and product complex shares a species with two new reactions with a fictitious distinct intermediate species. This then yields a new network $\mathcal{G}^*$ given by 
\[
\ce{   2 A  <=>  C <=>  A + B  ->  D -> 2 B  },
\]
where $C,D$ are the newly added fictitious species. The stoichiometric matrix for $\mathcal{G}^*$ is given by
\[ \mathbb{S}^* = \begin{bmatrix}
      \phantom{-}2  &    -2        & -1              & \phantom{-}1 & -1         & \phantom{-}0\\
    \phantom{-}0 & \phantom{-}0 &-1              &    \phantom{-}1 & -1         & \phantom{-}2\\
       -1   & \phantom{-}1 & \phantom{-}1   & -1           & \phantom{-}0 &\phantom{-}0\\
       \phantom{-}0 & \phantom{-}0 & \phantom{-}0   &  \phantom{-} 0           & \phantom{-}1 & -1
\end{bmatrix}.
\] 
Notice that the restriction of the stoichiometric matrix to the species set $\mathcal{S}' = \{A,C\}$ and the reactions set $\mathcal{R}' = \{A+B \to C, C\to 2A\}$, denoted by $\mathbb{S}^{*\mathcal{R}'}_{\mathcal{S}'}$, indeed satisfies the properties for an autocatalytic core, and thus $\mathcal{G}^*$ is indeed autocatalytic with core species $\{A,C\}$. A similar construction shows that the network is also autocatalytic in the set $\{B,D\}$. (In the typology of Blokhuis et al.\, both of these cores are of type I.)
\end{example}

%----------------section end -----------------------------
\section{Organizing an autocatalytic ecology}
\label{sec:organization}

Even mildly complicated CRNs can exhibit an abundance of autocatalytic cores \cite{personalconv,peng2022hierarchical}. Recall from Sec.\ \ref{sec:definitions} that we define a \textit{subnetwork} of a graph $\mathcal{G} = (\mathcal{S},\mathcal{R})$ to be a subgraph $(\mathcal{S}|_{\mathcal{R}'},\mathcal{R}')$ where $\mathcal{R}' \subset \mathcal{R}$ and there is no omission of species in the subnetwork that participate in $\mathcal{R}$ in the original graph $\mathcal{G}$. Then, we define a \textit{minimal autocatalytic subnetwork} (MAS) to be a subnetwork that contains exactly one autocatalytic core. In Sec.\ \ref{sec:definitions}, we demonstrate first, these subnetworks organize within equivalence classes, and, second, that the subnetworks within an equivalence class can also be organized by analyzing, what we call their \textit{species-productive cones}. We prove some results about autocatalytic subnetworks in Sec.\ \ref{sec:results}, and provide algorithms to exhaustively enumerate the MASs and to identify their species-productive cones in Sec.\ \ref{sec:algorithms}. Finally, we discuss the polyhedral geometry of autocatalytic ecologies and develop a visualization scheme in Sec.\ \ref{sec:visualization}. It must be noted that, to simplify construction, we have assumed that the graph $\mathcal{G}$ is reversible throughout this section, i.e.\ for every reaction $y_1\to y_2 \in \mathcal{G}$, the reverse reaction $y_2 \to y_1$ is also in the reaction set of $\mathcal{G}$.  



\subsection{Definitions}
\label{sec:definitions}


Let $\mathcal{G} = (\mathcal{S},\mathcal{R})$ be a CRN. Let $A = (\mathcal{S}_A,A_R)$ be a minimal autocatalytic subnetwork (MAS) containing exactly one core, where $\mathcal{S}_A$ is the set of species with non-zero stoichiometry in the reaction set $A_R \subseteq \mathcal{R}$. From property 1 of an autocatalytic core, there must be a subset $A_C \subseteq \mathcal{S}_A$ containing the core species that has the same size as $A_R$. In general, however, there will also be disjoint subsets of $\mathcal{S}_A$ of species that only occur as co-reactants, co-products, and both co-reactants and co-products in $A_R$. Correspondingly, if we consider the restriction of the stoichiometric matrix $\mathbb{S}$ to the reaction set $A_R$, denoted by $\overline{\mathbb{S}}:=\mathbb{S}|_{A}$, for the MAS $A$ we define:
\begin{enumerate}
    \item the species set corresponding to all rows of $\overline{\mathbb{S}}$ with nonpositive coefficients as the \textit{food set} and denote it by $A_F$.
    \item the species set corresponding to all rows of $\overline{\mathbb{S}}$ with nonnegative coefficients as the \textit{waste set} and denote it by $A_W$.
    \item the species set corresponding to all rows of $\overline{\mathbb{S}}$ with both positive and negative coefficients as the \textit{member set} and denote it by $A_M$.
    \item the species set corresponding to all rows of autocatalytic core as the \textit{core member set} and denote it by $A_C$.     
\end{enumerate}
We denote the exclusion of a subset $K$ from the species set by $A_{/K}$. For example, all species in the species set of an autocatalytic subnetwork $A$ except the core member species will be denoted by $A_{/C}$. Note that the core member set, or simply \textit{core set,} is a subset of the member set, or $A_C \subseteq A_M$. We refer to the species in the member set that are not in the core set as \textit{non-core member species} and denote their set as $A_{M/C}$. For an example of the partitioning process on a reaction network where each set is non-empty, see Fig.\ \ref{fig:FWMC_example}.

\subsubsection{Food-waste-member-core partition of an autocatalytic subnetwork}
\begin{figure}
    \centering
    \includegraphics[width=\linewidth]{figures/FWMC_partition.png}
    \caption{Food-waste-member-core and resource-member partition of a minimal autocatalytic subnetwork in the 1-constituent Cluster CRN (CCRN) with $L=6$. Using the typology of Blokhuis et al.\, this subnetwork has a type two core. }
    \label{fig:FWMC_example}
\end{figure}


Henceforth we will refer to the above partition of the species set for a MAS as the \textit{food-waste-member-core} (FWMC) partition of the MAS. Moreover, we will refer to the submatrices generated by restricting to the food, waste, member, and core species, as $\overline{\mathbb{S}}_F$, $\overline{\mathbb{S}}_W$, $\overline{\mathbb{S}}_M$, and $\overline{\mathbb{S}}_C$, respectively. Finally, notice that MASs with identical FWMC partitions form an equivalence relation. Thus we will call MASs $A$ and $B$ to be equivalent if they have the same FWMC partition,
\begin{align*}
A \equiv B &\iff \text{FWMC}(A) = \text{FWMC}(B).    
\end{align*}

Our \textit{quadpartite} partition of the species set can be coarse-grained to a bipartite \textit{resource-member} partition where the \textit{resource} set is the union of food and waste sets (and member set is the same as above) (see Fig.\ \ref{fig:FWMC_example}). Our partitioning of the species set by examining the coefficients of the stoichiometric matrix is cognate with the partitioning done by Avanzini et al.\ in \cite{avanzini2022circuit}, however, since their interest is in null cycles, they partition the set of species into a \textit{resource-member} partition. It is also aligned with, and is in fact a refinement of, the \textit{food-waste-member} partitioning by Peng et al.\ in \cite{peng2020ecological}, where it is shown that different compositions of distinct autocatalytic cycles can lead to various ecological dynamics (see Fig.\ \ref{fig:Peng_summary}). 

\begin{example}
\label{eg:FWMC}
In the network $\mathcal{G}^*$ from Example \ref{eg:CRN-extension}, the MASs are:
\begin{enumerate}
    \item $A_1 = (\{A,B,C\}, \{A+B \to C, C \to 2 A \}$ with FWMC partition $\{\{B\},\{\},\{\{A,C\}\}\}$.
    \item $A_2 = (\{A,B,D\}, \{A+B \to D, D \to 2 B \}$ with FWMC partition $\{\{A\},\{\},\{\{B,D\}\}\}$.
\end{enumerate}
Notice that both these subnetworks have empty waste and non-core member species sets.

\end{example}



\begin{figure*}[t]
\centering
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=0.8\linewidth]{figures/flow_productive_eg_3_2.png}
  %\caption{Flow-productive cone}
  %
\end{subfigure}%
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=.8\linewidth]{figures/species_productive_eg_3_2.png}
  %\caption{Species-productive cone}
  %
\end{subfigure}
\caption{The flow and species productive cones for the Example \ref{eg:all-cones} are shown in the left and right panel, respectively. The dimensions of the embedding space of the flow and species productive cones are the number of reactions and species in the minimal autocatalytic subnetwork, respectively. Notice that the food species $B$ is strictly consumed (nonpositive) and the core species $A,C$ are strictly produced (nonnegative) in the species-productive region. For this example, the partition-productive cone is identical to the species-productive cone.  }
\label{fig:test}
\end{figure*}




\subsubsection{Flow, species and partition productive cones}\label{cones}
\subparagraph{Flow-productive cone:} Recall from property 1 of an autocatalytic core that $\overline{\mathbb{S}}_C$ is invertible. The chemical interpretation of the inverse is that the $k^\text{th}$ column is a flow vector that increases the $k^\text{th}$ species by exactly one unit, making it an elementary mode of production. We define the \textit{flow-productive cone}, denoted by $\mathcal{F}$, as the cone generated by the elementary modes of production of a minimal autocatalytic subnetwork (MAS),
\begin{align}
    \mathcal{F}(A) &= \text{cone}(\text{cols}(\overline{\mathbb{S}}_C^{-1})).
\end{align}
By definition, an element in the interior of the flow-productive cone, the \textit{flow productive region}, of $A$ corresponds to a flow vector for which $A$ is productive, i.e.\ the core member species of $A$ are strictly produced. It should be noted that the inverse of $\overline{\mathbb{S}}_C$ might contain negative coefficients. Since we are only considering reversible networks, negative flows on a reaction simply correspond to the reverse reactions. In general, however, for non reversible networks the productive cone may be defined as \[\mathcal{F}(A) = \text{cone}(\text{cols}(\overline{\mathbb{S}}_C^{-1})) \cap \mathbb{R}_{\geq 0}^{A_{C}}.\]

\subparagraph{Species-productive cone:} We term the image of the flow-productive cone of a MAS under its stoichiometric matrix $\overline{\mathbb{S}}$ as the \textit{species-productive cone} and denote it by $\mathcal{P}$,
\begin{align}
    \mathcal{P}(A) &= \overline{\mathbb{S}} \cdot \mathcal{F}(A) \nonumber \\
    &= \text{cone}(\text{cols}(\overline{\mathbb{S}} \cdot \overline{\mathbb{S}}_C^{-1}).
\end{align}
Therefore, each vector in the interior of the species-productive cone for a MAS, which we refer to as the \textit{species-productive region}, describes a change in concentration of the participating species for which the subnetwork is productive in the core member species. We remark that the \textit{species-consumptive cone} for the \textit{drainable subnetwork} $A^\text{op}$, which is obtained by reversing the edges of the autocatalytic subnetwork $A$, is the opposite of the species-productive cone:
\[\mathcal{P}(A^\text{op})=-\mathcal{P}(A).\]   


\subparagraph{Partition-productive cone:} Consider a MAS with species set $\mathcal{S}$. Let us denote the FWMC partition of $\mathcal{S}$ as $T = \{T_F, T_W, T_{M/C}, T_C \}$. Let $e_i$ be a column vector of size $\mathcal{S}$ with $1$ at the $i^\text{th}$ position and $0$ elsewhere. We define the set of basis vectors $\mathcal{B}_T$ for the partition $T$ to be
\begin{align}
    \mathcal{B}_T &= \mathcal{B}_T^F \cup \mathcal{B}_T^W \cup \mathcal{B}_T^{M/C} \cup \mathcal{B}_T^C  \label{eq:part_prod_cone}
\end{align}
where 
\begin{align*}
    \mathcal{B}_T^F &= \{ -e_f | f \in T_F \},\\
    \mathcal{B}_T^W &= \{ e_w | w \in T_W \},\\
    \mathcal{B}_T^{M/C} &= \{ -e_m \cup e_m | m \in T_{M/C} \},\\
    \mathcal{B}_T^C &= \{ e_c | c \in T_C \}.\\
\end{align*}
Thus the basis $\mathcal{B}_T$ for a partition $T$ spans the negative half-space of each species in the food set, the positive half-space of each species in the waste and core set, and both negative and positive half-spaces of each species in the non-core member set. Let the reaction network have conservation laws $\{\textbf{x}_i\}$, such that $\textbf{x}_i\cdot \mathbb{S} = 0$. We define the \textit{partition-productive cone} for the partition $T$, denoted by $\mathcal{Q}(T)$, to be the cone generated by the basis vectors restricted to the stoichiometric subspace
\begin{align*}
    \mathcal{Q}(T) &= \text{cone}(\mathcal{B}_T) \perp \{\textbf{x}_i\}.
\end{align*}

As noted in the previous subsection, belonging to an FWMC partition is an equivalence relation. By definition, the species-productive cones of all equivalent autocatalytic subnetworks belonging to the partition $T$ will lie in the partition-productive cone $\mathcal{Q}(T)$.  

\begin{example}
\label{eg:all-cones}
    In this example, we will identify the flow-productive, species-productive and partition-productive cones for the MAS $A_1$ from Example \ref{eg:FWMC} (shown in Fig.\ \ref{fig:test}). For $A_1 = (\{A,B,C\}, \{A+B \to C, C \to 2 A \}$,
    \begin{align*}
        \overline{\mathbb{S}} &= 
        \begin{bmatrix}
            -1  & \phantom{-}2 \\
            -1 & \phantom{-}0 \\
            \phantom{-}1 & -1  
        \end{bmatrix}\\
        \overline{\mathbb{S}}_C &= 
        \begin{bmatrix}
            -1  & \phantom{-}2 \\
            \phantom{-}1 & -1 
        \end{bmatrix}\\
         \overline{\mathbb{S}}_C^{-1} &= 
       \begin{bmatrix}
           1 & 2\\
           1 & 1
       \end{bmatrix}.
    \end{align*}
    The flow-productive cone is 
    \begin{align*}
        \mathcal{F}(A_1) &= \text{cone}\left( \begin{bmatrix}
            1\\
            1
        \end{bmatrix}, \begin{bmatrix}
            2\\
            1
        \end{bmatrix}  \right),
    \end{align*}
        and the species-productive cone is 
    \begin{align*}
        \mathcal{P}(A_1) &= \text{cone}\left( \begin{bmatrix}
            \phantom{-}1\\
            -1\\
            \phantom{-}0
        \end{bmatrix}, \begin{bmatrix}
            \phantom{-}0\\
            -2\\
            \phantom{-}1
        \end{bmatrix}  \right).
    \end{align*}
    Notice that the graph $\mathcal{G}^*$ (from Example \ref{eg:CRN-extension}) has a positive conservation law $\textbf{x} = [1,1,2]$. The partition-productive cone for the partition $T_1=\{\{B\},\{\},\{\{A,C\}\}\}$ is 
    \begin{align*}
        \mathcal{Q}(T_1) &= \text{cone}\left(  \begin{bmatrix}
            \phantom{-}0\\
            -1\\
            \phantom{-}0
        \end{bmatrix}, 
         \begin{bmatrix}
            1\\
            0\\
            0
        \end{bmatrix},
         \begin{bmatrix}
            0\\
            0\\
            1
        \end{bmatrix}
        \right) \perp \textbf{x}\\
        &= \text{cone}\left( \begin{bmatrix}
            \phantom{-}1\\
            -1\\
            \phantom{-}0
        \end{bmatrix}, \begin{bmatrix}
            \phantom{-}0\\
            -2\\
            \phantom{-}1
        \end{bmatrix}  \right)\\ 
        &=\mathcal{P}(A_1).
    \end{align*}
For this example, the species-productive and partition-productive cones are identical, which is an instance of a general result proved in Theorem \ref{theorem:partition-prod}.    
\end{example}

\subsection{Mathematical results}
\label{sec:results}
To understand the geometry of the different cones defined in the previous subsection, we prove some results about their behavior. In order to organize the partitions, we give the conditions under which two minimal autocatalytic subnetworks (MASs) with different partitions will have non-intersecting species-productive regions in Proposition \ref{prop:food-set}. Next, we explore conditions under which the species-productive cone is identical with the partition-productive cone for a MAS in Theorem \ref{theorem:partition-prod}. Under such conditions, the partition itself contains the information of the species-productive regions of all the MASs belonging to it. Finally, in Proposition \ref{prop:null-flows}, we understand the topological properties a CRN must posses in order for the species-productive cones of two MASs to intersect. In Sec.\ \ref{sec:visualization} we will use these results to construct a visualization scheme for the list of MASs in any CRN.

\begin{proposition}
\label{prop:food-set}
Two autocatalytic subnetworks with different food sets have disjoint species-productive regions if their non-core member species sets are empty.
\end{proposition}

\begin{proof}
We will show that if two autocatalytic subnetworks $A$ and $B$ have distinct food sets and their non-core member species sets are empty, then the interiors of their species-productive cones (species-productive regions) do not intersect, 
\[\text{int}(\mathcal{P}(A)) \cap \text{int}(\mathcal{P}(B)) = \vn.\]
Let $A_{F}/B_{F}\neq \vn$. Recall that that the species-productive cone of the subnetworks lies within their partition-productive cone $\mathcal{Q}(\text{FWMC}(A))$ and $\mathcal{Q}(\text{FWMC}(B))$. Let $\textbf{y}$ be the vector of length $\mathcal{S}$ such that
\begin{align*}
    \bigr[ \textbf{y} \bigr]_i &=
    \begin{cases}
   -1   \text{ if } i \in A_{F}/B_{F}\\
    \phantom{-}0 \text{ otherwise.}
    \end{cases}
\end{align*}
Then, by definition of the partition-productive cones, $\textbf{y} \cdot \text{int}(\mathcal{Q}(\text{FWMC}(A))) >0 $ and $\textbf{y} \cdot \text{int}(\mathcal{Q}(\text{FWMC}(A))) \leq 0 $. Thus, $\textbf{y}$ defines a hyperplane separating the two species-productive regions, and the two regions do not intersect.
\end{proof}
\begin{remark}
If the non-core member species set is not empty, then the result is no longer true. For instance, see Example \ref{eg:4species-partitionproductive}.
\end{remark}

\begin{example}
    \label{eg:food-intersect}
    Using Examples \ref{eg:FWMC} and \ref{eg:all-cones}, the species-productive cones for $A_1$ and $A_2$ in the species set $\{A,B,C,D\}$ are
    \begin{align*}
        \mathcal{P}(A_1) &=
        \text{cone}\left( \begin{bmatrix}
            \phantom{-}1\\
            -1\\
            \phantom{-}0\\
            \phantom{-}0
        \end{bmatrix}, \begin{bmatrix}
            \phantom{-}0\\
            -2\\
            \phantom{-}1\\
            \phantom{-}0
        \end{bmatrix}  \right),\\
        \mathcal{P}(A_2) &=
        \text{cone}\left( \begin{bmatrix}
            -1\\
            \phantom{-}1\\
            \phantom{-}0\\
            \phantom{-}0
        \end{bmatrix}, \begin{bmatrix}
            -2\\
            \phantom{-}0\\
            \phantom{-}0\\
            \phantom{-}1
        \end{bmatrix}  \right).
    \end{align*}
    If we let $\textbf{y} = [0, -1, 0, 0] $, then $\textbf{y}\cdot\text{int}(\mathcal{P}(A_1)) >0$ and $\textbf{y}\cdot\text{int}(\mathcal{P}(A_2)) <0$. Thus $\textbf{y}$ defines a separating hyperplane, and the two species-productive regions are disjoint. 
\end{example}

\begin{lemma}
\label{lemma:core_identity}
The restriction of the species-productive cones of all MASs to their core (member) species is the positive orthant in the core species.
\end{lemma}
\begin{proof}
    Recall that the species-productive cone of a MAS is defined as 
    \begin{align*}
         \mathcal{P}(A) &= \text{cone}(\text{cols}(\overline{\mathbb{S}} \cdot \overline{\mathbb{S}}_C^{-1}).
    \end{align*}
    By definition of the matrix inverse, the restriction of the product to the core species is the identity matrix, \[\overline{\mathbb{S}} \cdot \overline{\mathbb{S}}_C^{-1}\big|_C = I_C.\]
    Thus, the species-productive cone restricted to the core species is the positive orthant.
\end{proof}

\begin{lemma}
\label{lemma:food}
    If a CRN has any positive conservation laws, the union of the food set and the non-core member set of any MAS must be non-empty.
\end{lemma}
\begin{proof}
    Let $A$ be a MAS, with associated stoichiometric matrix $\overline{\mathbb{S}}$, of a CRN with a positive conservation law given by the vector $\textbf{x}$. Suppose $A$ does not have any element in the food set $A_F$ or the non-core member set $A_{M/C}$. For any flow $\textbf{v}$ in the productive region of $A$, we know that $\overline{\mathbb{S}} \cdot \textbf{v} \succ 0$. Also, since $\textbf{x}$ is a positive conservation law, $\textbf{x}\cdot \overline{\mathbb{S}} \cdot \textbf{v} \overset{!}{=} 0$. Since the sum of positive values cannot add up to zero, we have a contradiction. Thus, there must be at least one element in the food set or the non-core member set of a MAS with positive conservation laws. 
\end{proof}

\begin{theorem}
\label{theorem:partition-prod}
    For a CRN with positive conservation laws, the species-productive cone of any MAS with exactly one more species than the core set is identical to its partition-productive cone.  
\end{theorem}

\begin{proof}
    Let $A$ be a MAS of a CRN with positive conservation laws $\{\textbf{x}_i\}$ with exactly one more species than the core set, denoted by $C = \{ C_1, \ldots, C_C \}$. From Lemma \ref{lemma:food}, we know that the extra species must be either in the food set or non-core member set of $A$. In either case, the species must be net consumed by the subnetwork to respect the positive conservation laws, and we denote it by $f$. \\
    Let us label the species set as $\{f, C_1, \ldots, C_C\}$. Recall that the partition-productive cone is defined as $\mathcal{Q}(\text{FWMC}(A)) = \text{cone}(\mathcal{B})\perp \{\textbf{x}_i\}$, where
    \begin{align*}
        \mathcal{B} &= \left\{ 
        \begin{bmatrix}
            -1 \\
            \phantom{-} 0 \\
            \phantom{-}\vdots \\
            \phantom{-}0
        \end{bmatrix} ,
        \begin{bmatrix}
            0 \\
            1 \\
            \vdots \\
            0
        \end{bmatrix} ,
        \ldots ,
                \begin{bmatrix}
            0 \\
            0 \\
            \vdots \\
            1
        \end{bmatrix} 
        \right\}.   
    \end{align*}
    Note that the additional basis vector for the non-core member species with positive entries is omitted since we know that it must be consumed in the species-productive region to respect the positive conservation laws. Moreover, using Lemma \ref{lemma:core_identity}, we know that the restriction of the species-productive cone to the core species is the positive orthant in the core species. In particular, for the species-productive cone $\mathcal{P}(A) = \text{cone}(\text{cols}(\overline{\mathbb{S}} \cdot \overline{\mathbb{S}}_C^{-1})$, we have 
   \begin{align*}
       \overline{\mathbb{S}} \cdot \overline{\mathbb{S}}_C^{-1}   
       &= 
                        \begin{bmatrix}
                            \textbf{f}\\
                            I_C
                        \end{bmatrix},
    \end{align*}
    where $\textbf{f}$ is a row vector $[\textbf{f}_1, \ldots, \textbf{f}_C]$ and $I_C$ is the identity matrix of size $C$. Since the CRN also has conservation laws, these coefficients must satisfy
    \begin{align*}
        \bigr[\textbf{x}_i\bigr]_f \cdot \textbf{f}_1 +  \bigr[\textbf{x}_i\bigr]_{C_1} \cdot 1 &= 0 \\
        \vdots \\
        \bigr[\textbf{x}_i\bigr]_f \cdot \textbf{f}_C +  \bigr[\textbf{x}_i\bigr]_{C_C} \cdot 1 &= 0, \\
    \end{align*}
    for every conservation law indexed by $i$. But these are simply the basis vectors of the partition-productive cone $\mathcal{B}$ when made orthogonal to the conservation laws. Thus the partition-productive cone is identical to the species-productive cone. 
\end{proof}

\begin{example}
    \label{eg:4species-partitionproductive}
    Consider two MASs (from Sec.\ \ref{sec:CCRN-1constituent} Fig.\ \ref{fig:cluster_n4}) 
    \begin{align*}
        A_1 &= (\{ \overline{1}, \overline{2}, \overline{3}, \overline{4}\},
        \{ \overline{1} + \overline{3} \to \overline{4}, \\
        & \phantom{=((} \overline{1} + \overline{4} \to \overline{2} + \overline{3},
         \overline{2}+ \overline{2} \to \overline{3} + \overline{1}\}  ),\\
             A_2 &= (\{ \overline{1}, \overline{2}, \overline{3}, \overline{4}\},
    \{ \overline{1} + \overline{3} \to \overline{4}, \\
    & \phantom{=((} \overline{1} + \overline{4} \to \overline{2} + \overline{3},
     \overline{2}+ \overline{2} \to \overline{4}\}  ).
    \end{align*}
    The partitions of $A_1$ and $A_2$ are $\{\{\},\{\},\{\overline{1},\{\overline{2},\overline{3},\overline{4}\}\}\}$ and $\{\{\overline{1}\},\{\},\{\{\overline{2},\overline{3},\overline{4}\}\}\}$, respectively. A simple calculation yields that their species-productive cones are in fact identical, and
    \begin{align*}
    \mathcal{P}(A_1) = \mathcal{P}(A_2) &=
         \text{cone}\left( \begin{bmatrix}
            -2\\
            \phantom{-}1\\
            \phantom{-}0\\
            \phantom{-}0
        \end{bmatrix}, \begin{bmatrix}
            -3\\
            \phantom{-}0\\
            \phantom{-}1\\
            \phantom{-}0
        \end{bmatrix}, \begin{bmatrix}
            -4\\
            \phantom{-}0\\
            \phantom{-}0\\
            \phantom{-}1
        \end{bmatrix}  \right).
    \end{align*}
\end{example}


\begin{proposition}
\label{prop:null-flows}
    If the species-productive cones of two MASs have an intersection, the reversible graph of their union has a null flow.
\end{proposition}
\begin{proof}
    Let the productive cones of two MASs $A$ and $B$, have a non-empty intersection, $\mathcal{P}(A)\cap \mathcal{P}(B)\neq \vn$. Let $A\cup B$ be the subnetwork obtained by taking the union of the two subnetworks, and let $\mathbb{S}_{A\cup B}$ be the associated stoichiometric matrix. Then, there must be non-identical flows $\textbf{v}_A$ and $\textbf{v}_B$ with support in $A$ and $B$, respectively, such that
    \[ \mathbb{S}_{A\cup B} \cdot \textbf{v}_A = \mathbb{S}_{A\cup B} \cdot \textbf{v}_B.\]
    This implies that $\mathbb{S}_{A\cup B} \cdot (\textbf{v}_A - \textbf{v}_B) = 0$, and thus the kernel of the stoichiometric matrix is non-empty. In general, the difference of their flows $(\textbf{v}_A - \textbf{v}_B)$ can have negative coefficients, however, it yields the interpretation of a null flow on a network if the union of the two subnetworks $A\cup B$ is made reversible.  
\end{proof}

\begin{remark}
    The converse statement, if the reversible graph of the union of two MASs contains a null cycle then their species-productive cones must intersect, is not true. For example, consider the MASs (from Sec.\ \ref{sec:CCRN-1constituent} Fig.\ \ref{fig:cluster_n4})
    \begin{align*}
        A_1 &= (\{ \overline{1}, \overline{2}, \overline{3}\},
        \{ \overline{1} + \overline{2} \to \overline{3},  \overline{1} + \overline{3} \to \overline{2}+ \overline{2}\}),  \\
        A_1 &= (\{ \overline{1}, \overline{2}, \overline{3}\},
        \{ \overline{2} \to \overline{1} +  + \overline{1},  \overline{3} + \overline{1} \to \overline{2}+ \overline{2}\}).  
    \end{align*}
    Notice that $\text{FWMC}(A_1) = \{\{\overline{1}\},\{\},\{\{ \overline{2},\overline{3}\}\}\}$ and $\text{FWMC}(A_2) = \{\{\overline{3}\},\{\},\{\{ \overline{1},\overline{2}\}\}\}$. Thus, using Proposition \ref{prop:food-set}, the two species-productive cones do not intersect. However, the union of the two subnetworks yields 
    \begin{align*}
        A_1\cup A_2 &= (\{ \overline{1}, \overline{2}, \overline{3}\},  \{ \overline{1} + \overline{2} \to \overline{3},\\
        &\phantom{=((}
         \overline{1} + \overline{3} \to \overline{2}+ \overline{2}, \overline{2} \to \overline{1} +  \overline{1}\}).
    \end{align*}
    which is a null cycle (with null flow $[1,1,1]^T$).
\end{remark}

\begin{figure}[t]
\includegraphics[scale=0.5]{figures/flowchart.jpg}
\caption{Flowchart of our procedure to construct minimal autocatalytic subnetworks.\label{fig:flowchart}}
\end{figure}


\subsection{Algorithms}

\label{sec:algorithms}
We propose a mathematical optimization-based approach to enumerate MASs in a CRN and to identify their intersecting productive cones. A flowchart of our procedure to construct the MAS is shown in Fig. \ref{fig:flowchart}. Given the stoichiometric matrix, the algorithm, in its first iteration computes an autocatalytic subnetwork by adequately selecting the core species and the reactions taking part in a subnetwork with minimum number of reactions. The obtained subnetwork is saved, and its FWMC partition is computed. Its tuple of  reactions is added to the set of \emph{incompatible} reactions, \texttt{Incomp}, and the procedure is repeated, by filtering the reactions in the subsequent obtained subnetworks to not contain any of the tuples in \texttt{Incomp}, until no feasible subnetworks with the \emph{compatibility} conditions are possible. Minimality is assured by the minimization of the number of reactions in the subnetwork in each run, and also by the compatibility conditions.

The identification of the FWMC partition for a given autocatalytic subnetwork results from checking both the signs of the restricted stoichiometric matrix, $\overline{\mathbb{S}}$ and the production vector $\overline{\mathbb{S}} v^*$ for the flow $v^*$ also obtained after each run. Specifically, for each species $i$ in the subnetwork: if all the components in $\overline{\mathbb{S}}_i$ are non positive (resp. non negative), $i$ is classified as a food (resp. waste) species; in case $\bar S_i$ has negative and positive entries, $i$ is classified as a member species; and if $\overline{\mathbb{S}}_i$ has negative and positive entries and $(\overline{\mathbb{S}} v^*)_i$ is strictly positive, then $i$ is a core species.

Once the set of minimal autocatalytic subnetworks is obtained, they are classified by means of their FWMC partitions. Within each FWMC class, we check the \textit{pairwise intersection} of the species-productive cones. For each subnetwork $A_1, A_2$ in the same class, we first check if the cones $\mathcal{P}(A_1)$ and $\mathcal{P}(A_2)$ intersect by finding a nonzero vector shared by both cones. In case the cones intersect, we then normalize the generators of the cones, and compute the bi-directional minimum distances between each of the generators of one cone and the other cone. If all these distances  are zero, the cones are identical; if one of the sets of distances is zero but the other is not, one of the cones is contained in the other; but if in both sets there are positive distances, the cones partially intersect. Using the same procedure, we also check the pairwise intersection of the partition-productive cones of each equivalence class. Further details on the described approaches are provided in Appendix \ref{app:alg}.



\begin{figure*}
    \centering
    \includegraphics[width=0.95\textwidth]{figures/Fig_IntersectionPartitionCones_L5.png}
    \caption{The intersection data of partition-productive cones for all partitions of the complete $L=5$ 1-constituent CCRN (described in Sec.\ \ref{sec:CCRN-1constituent}). }
    \label{fig:intersection_partition_5}
\end{figure*}



\subsection{Geometry and visualization}
\label{sec:visualization}


In the last subsection, we proposed an algorithm that, given a stoichiometric matrix of a CRN, outputs a list of the MASs it contains. In this section, we will explain how to take the list of MASs and visualize their combinatorics and geometry. Recall that for each MAS, we define a flow-productive cone in the flow space of the graph where each core member species is strictly produced, and a species-productive cone on the space of changes in concentration (population). Geometrically, the space of changes in chemical concentration (i.e., the velocity space of chemical concentrations) is the stoichiometric subspace of the CRN. Thus, the list of MASs can be seen as yielding a partial polyhedral decomposition of the stoichiometric space induced from the flows on the hypergraph (CRN). We remark that, for a more complete decomposition one must also consider the consumptive cones of the minimal drainable subnetworks. However, it is not necessary that the union of the autocatalytic and drainable subnetworks span the complete stoichiometric space (for example, see Sec.\ \ref{sec:L_3_example} Fig.\ \ref{fig:cluster_L3}). 





In subsec.\ \ref{sec:definitions}, we showed that each subnetwork can be assigned to its FWMC equivalence class. As explained in Proposition \ref{prop:food-set}, while there are cases where the species-productive cones of different equivalence classes can be shown to not intersect, in general the productive regions of different equivalence classes can intersect. To visualize the list of equivalence classes to which the MASs belong, we run the algorithm for finding an intersection between each pair of partition-productive cones and obtain a two-dimensional square matrix,  $C_{\text{pp}}$, of dimension equal to the number of equivalence classes. Let us denote the list of equivalence classes by $\mathcal{L}$. The entry of $C_{\text{pp}}$ in the $i^{\text{th}}$ row and $j^{\text{th}}$ columns is given by,
\begin{align*}
    [C_{\text{pp}}]_i^j & = 
    \begin{cases}
    \phantom{-}2 \text{ if } Q(\mathcal{L}_i) = Q(\mathcal{L}_j),\\
        \phantom{-}1 \text{ if } Q(\mathcal{L}_i) \subsetneq Q(\mathcal{L}_j),\\
        \phantom{-}0 \text{ if } \varnothing \neq Q(\mathcal{L}_i) \cap Q(\mathcal{L}_j) \subsetneq Q(\mathcal{L}_i), Q(\mathcal{L}_j),\\
        -1 \text{ if }  Q(\mathcal{L}_i) \cup Q(\mathcal{L}_j) = \varnothing.
    \end{cases}
\end{align*}
Notice that any asymmetry in entries across the diagonal indicates that only one of the partition-productive cones completely contain the other. This matrix can then be visualized as a heat map, for example see Fig.\ \ref{fig:intersection_partition_5}.



\begin{figure*}[t]
\centering
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=0.9\linewidth]{figures/food_waste_L_5.png}
  %\caption{E-graph}
  
\end{subfigure}%
\begin{subfigure}{.5\textwidth}
  \centering
  \includegraphics[width=.9\linewidth]{figures/Fig_Class27_n5.png}
  %\caption{Change in concentration}
  
\end{subfigure}
\caption{All MASs belonging to the $\{\{\overline{4}\},\{\overline{1}\},\{\{\overline{2},\overline{3},\overline{5}\}\}\}$ class in the complete $L=5$ 1-constituent CCRN are considered. The projections of their species-productive cones on the food and waste species are shown in the left panel. The right panel displays the intersection information of MASs within a class analogous to Fig.\ \ref{fig:intersection_partition_5}. Note that there are no MASs in the class that are disjoint, and thus the `disjoint' label is omitted. }
\label{fig:cone_intersection}    
\end{figure*}


Each equivalence class can contain several MASs. The species-productive cones of the MASs in a class are not always identical. While they will share the same projection on the core species, the productive regions in the non-core species can be very different. For example, let us pick the equivalence class $\{\{\overline{4}\},\{\overline{1}\},\{\{\overline{2},\overline{3},\overline{5}\}\}\}$ from the list of classes shown in Fig.\ \ref{fig:intersection_partition_5}. From the figure, we know that it contains $5$ MASs. We plot the projection of the species-productive region in the non-core species in the top panel of Fig.\ \ref{fig:cone_intersection}. In higher dimensions when more non-core members are involved, this representation can get rather cumbersome. Thus, we employ a similar visualization as $C_\text{pp}$ to depict the intersection of MASs within an equivalence class. Whether or not there is an intersection between the species-productive cones can be ascertained using our algorithm outlined in the previous subsection. For an example of the resulting visualization for the same equivalence class considered above, see the bottom panel of Fig.\ \ref{fig:cone_intersection}. In the same manner, a visualization for the information of pairwise intersection of the productive cones of all MASs for a CRN can also be obtained, for example see Fig.\ \ref{fig:all_intersect}. 

\begin{figure*}
    \includegraphics[width = \textwidth]{figures/Fig_AllIntersections_L5.png}
    \caption{The pairwise intersection information for all the species-productive cones of the $L=5$ 1-constituent complete CCRN of order two. The reaction network and the list of MASs can be found in Appendix \ref{sec:MAS_list}. \label{fig:all_intersect}}
\end{figure*}


\begin{example}
    Consider the complete $L=5$ 1-constituent CCRN of order two. The list of all the FWMC classes, the number of MASs they contain, and their intersection information is summarized in Fig.\ \ref{fig:intersection_partition_5}. Consider the class $\{\{\overline{4}\},\{\overline{1}\},\{\{\overline{2},\overline{3},\overline{5}\}\}\}$. It contains the MASs:
    \begin{align*}
        \text{AS1} =\{&  \overline{2}+\overline{2}\to \overline{3} + \overline{1},\\
        &\overline{5}\to \overline{2} + \overline{3},\\
        &\overline{4}+\overline{3}\to \overline{2} + \overline{5}\},\\
        \text{AS2} =\{&  \overline{4}+\overline{2}\to \overline{1} + \overline{5},\\
        &\overline{5}\to \overline{2} + \overline{3},\\
        &\overline{4}+\overline{3}\to \overline{2} + \overline{5}\},
    \end{align*}
    \begin{align*}
        \text{AS3} =\{&  \overline{3}+\overline{3}\to \overline{5} + \overline{1},\\
        &\overline{5}\to \overline{2} + \overline{3},\\
        &\overline{4}+\overline{2}\to \overline{3} + \overline{3}\},\\ 
        \text{AS4} =\{&  \overline{3}+\overline{3}\to \overline{5} + \overline{1},\\
        &\overline{4}+\overline{2}\to \overline{5} + \overline{1},\\
        &\overline{5}\to \overline{2} + \overline{3}\},\\
        \text{AS5} =\{&  \overline{3}\to \overline{2} + \overline{1},\\
        &\overline{4}+\overline{2}\to \overline{5} + \overline{1},\\
        &\overline{5}\to \overline{2} + \overline{3}\}.        
    \end{align*}
    The projection of the species-productive cones for the above MASs and their intersection information is summarized in Fig.\ \ref{fig:cone_intersection}.
\end{example}


%----------------section end -----------------------------
\begin{figure*}
    \centering
    \includegraphics[width = \textwidth]{figures/clusterization.png}
    \caption{Consider a CRN of polymers whose species set is arranged as a binary tree in two letters, shown to the left. Then the \textit{cluster} resulting from each binary string is obtained by counting the number of occurrences of each alphabet. The clusters thus obtained form a lattice structure, shown to the right.}
    \label{fig:clusterization}
\end{figure*}
\section{Cluster chemical reaction networks (CCRN) framework}
\label{sec:applications}

All realistic chemical reactions, ignoring nuclear reactions, conserve the number of atoms of each type. Often, but not always, reactions involving polymers also conserve the number of each monomer. This motivates taking a \textit{coarse-grained} perspective, where we forget the bond structure of each molecule or polymer and only count the number of atoms or monomers, respectively. Following terminology from Chapter 7 of \cite{kelly2011reversibility}, we will refer to this coarse-grained description of each species as a \textit{cluster} (see Fig.\ \ref{fig:clusterization}). This also allows us to coarse-grain a realistic CRN into a new CRN which has clusters as species and reactions between multisets of clusters induced from the reactions in the original CRN. We will call the resulting CRN, a \textit{cluster chemical reaction network} (CCRN) obtained from the original CRN.


There are three advantages of considering CCRN descriptions of CRNs. First, consider a CRN whose species set consists of all linear polymers formed of $D$ monomers up to size $N$. Then, the CCRN induced by the CRN would reduce the number of species from exponential ($D^N$) to polynomial ($N^D$) in the size of the polymer, providing a significant computational advantage. Secondly, since each reaction in the original CRN induces a reaction in the CCRN with the same topology, the mapping would preserve the autocatalytic property. In other words, if a CRN is autocatalytic, its induced CCRN will also be autocatalytic. However, since it is a many-to-one map, multiple autocatalytic subnetworks of the CRN might map to the same autocatalytic subnetwork in its CCRN or new autocatalytic subnetworks in the CCRN might emerge (see Example \ref{eg:CCRN_polymer}). Thirdly, one can later re-introduce structure into the CCRN by adding more species with the same cluster counts (conserved quantities). In this way, the coarse-graining can be systematically refined to recover the original CRN, and allows a gradual complexification of the model.

\begin{example}
\label{eg:CCRN_polymer}
    Consider the CRN 
    \begin{align*}
        \mathcal{G} &= (\mathcal{S},\mathcal{R})\\
        \mathcal{S} &= \{ A, B, BAA, ABB, ABBA, BAAB, ABBABAAB\}\\
        \mathcal{R} &= \{ ABBA + BAA + B \to ABBABAAB,\\
        & \phantom{000.} BAAB + ABB + A \to ABBABAAB,\\
        & \phantom{0000} ABBABAAB \to ABBA + BAAB. \}
    \end{align*}
    The CCRN obtained by counting the number of $A$s and $B$s in each string and representing the resulting clusters as $\overline{n_A, n_B}$ is
    \begin{align*}
        \text{CCRN}(\mathcal{G}) &= (\mathcal{S}_C,\mathcal{R}_C)\\
        \mathcal{S}_C &= \{ \overline{1,0}, \overline{0,1}, \overline{2,1}, \overline{1,2}, \overline{2,2}, \overline{4,4}\}\\
        \mathcal{R}_C &= \{ \overline{2,2} + \overline{2,1} + \overline{0,1} \to \overline{4,4},\\
        & \phantom{000} \overline{2,2} + \overline{1,2} + \overline{1,0} \to \overline{4,4},\\
        & \phantom{000} \overline{4,4} \to \overline{2,2} + \overline{2,2}. \}
    \end{align*}
Notice that while the CRN has one minimal autocatalytic subnetwork (MAS), the CCRN has two MASs. This can be remedied by remembering that $BAAB$ and $ABBA$ are distinct by introducing a new species $\overline{2,2}^*$. A `finer' CCRN would then be 
    \begin{align*}
        \text{CCRN}(\mathcal{G})^* &= (\mathcal{S}_C^*,\mathcal{R}_C^*)\\
        \mathcal{S}_C^* &= \{ \overline{1,0}, \overline{0,1}, \overline{2,1}, \overline{1,2}, \overline{2,2},\overline{2,2}^*, \overline{4,4}\}\\
        \mathcal{R}_C^* &= \{ \overline{2,2} + \overline{2,1} + \overline{0,1} \to \overline{4,4},\\
        & \phantom{000} \overline{2,2}^* + \overline{1,2} + \overline{1,0} \to \overline{4,4},\\
        & \phantom{000} \overline{4,4} \to \overline{2,2} + \overline{2,2}^*. \}
    \end{align*}
    Note that $\text{CCRN}(\mathcal{G})^*$ has identical information to the original CRN and is the finest CCRN that can be obtained from the original CRN. 
\end{example}


We formally define a CCRN in Sec.\ \ref{sec:CCRN-formalism}, and systematically explore the properties of CCRNs with one conserved quantity (type of atom or monomer) in Sec.\ \ref{sec:CCRN-complete}. We also provide the statistics of autocatalytic subnetworks found in such reaction networks and present a worked-out-example in Sec.\ \ref{sec:L_3_example}. We briefly discuss the computational challenges in scaling the algorithm for fully connected models in Sec.\ \ref{sec:comp_challenge} and introduce rule-generated CCRNs in Sec.\ \ref{sec:rule_gen}.  


\subsection{Formalism}
\label{sec:CCRN-formalism}
A \textit{cluster chemical reaction network (CCRN)} is a CRN with the additional structure that, upon excluding the inflow and outflow reactions, the network has at least one nonnegative integer conservation law (for the definition of a conservation law, see Sec.\ \ref{sec:hypergraphs}). Each conservation law corresponds to a type of \textit{constituent} that is conserved. We denote the number of distinct constituents by $D$ and label them as $A_1, \ldots, A_D$. The species of a CCRN, termed \textit{clusters}, are multisets of constituents and denoted by an overline (notation chosen to be consistent with \cite{liu2018mathematical}) over a vector of nonnegative integers representing the number of each constituent in the cluster. For instance, \[\overline{\textbf{n}} := \overline{n_1,\ldots,n_D}\] denotes the cluster comprising $n_1, \ldots, n_D$ of constituents $A_1, \ldots, A_D$, respectively. Notice that a unit of a constituent represents a unit of conserved quantity, thus a cluster is labelled by its set of conserved quantities. In case of multiple clusters with the same conserved quantities, we distinguish them with an asterisk $^*$. For example, in Table \ref{table:cluster-paradigm-rules}, $\overline{1}$ and $\overline{1}^*$ refer to a monomer and an activated monomer, respectively. Finally, we define the \textit{length} of a cluster to be the sum of its conserved quantities, \[l(\textbf{n}) := \sum_{i=1}^D n_i.\]       


In a CCRN, \textit{complexes} are multisets of clusters and denoted by \[c^\alpha := \sum_\textbf{n} c^\alpha_\textbf{n} \cdot \overline{\textbf{n}}.\] As an abuse of notation, we denote the stoichiometry of complex $c^\alpha$ also by $c^\alpha$, where now it is the column vector with entries $c^\alpha_\textbf{n}$. We define the \textit{size} of a complex to be its vector sum of conserved quantities of each type, and denote it as \[|c^\alpha| = \sum_\textbf{n} c^\alpha_\textbf{n} \cdot \textbf{n}.\] We define the \textit{width} of a complex to be total number of clusters in the multiset, and denote it by \[w(c^\alpha):= \sum_\textbf{n} c^\alpha_\textbf{n}.\] In the CCRN, every reaction must be such that the size of the source and target complexes are identical. Thus, a reaction $c^\alpha \to c^\beta$ is allowed only if \[|c^\alpha| = \sum_\textbf{n} c^\alpha_\textbf{n} \cdot \textbf{n}= \sum_\textbf{n} c^\beta_\textbf{n} \cdot \textbf{n} = |c^\beta|.\] This restriction consistently ensures that the number of constituents across each reaction are conserved, as 
\[\sum_\textbf{n} \textbf{n}\cdot(c^\beta - c^\alpha)_\textbf{n} = 0.\] 

Let $\mathcal{H}_D = (\mathcal{S}, \mathcal{R})$ denote a CCRN with $D$ constituents. Then for any finite CCRN, 
\begin{align*}
    \mathcal{S} &\subset \{\overline{\textbf{n}} | \textbf{n} \in \mathbb{Z}_{\geq 0}^D  \},\\
    \mathcal{R} &\subseteq \{c^\alpha \to c^\beta | |c^\alpha| = |c^\beta|  \}.
\end{align*}
We define the length of a CCRN by the maximum length of its clusters, and denote it by \[l(\mathcal{H}_D) = \max_{\mathcal{S}} |\textbf{n}|.\] We also define the \textit{order} of a CCRN to be the maximum width of its complexes, denoted by \[o(\mathcal{H}_D) = \max_{\mathcal{R}} w(c^\alpha).\] 

We define a \textit{complete CCRN} of length $L$ and order $w$ to consist of all species up to length $L$ and all possible reactions that are allowed by the conservation laws with complexes of width up to $w$. Since any reaction consisting of more than two reactants or products can be written as chains of second-order reactions, for our analysis \textit{we will restrict to CCRNs of order two}.


\begin{figure}
    \centering
    \includegraphics[width = 0.5\textwidth]{figures/1_const_MASs_data.png}
    \caption{Number of minimal autocatalytic subnetworks (MASs) varying with $L$ for the complete 1-constituent CCRN of order two. For each $L$, the distribution of MAS with the number of reactions is also shown. Notice that total MASs for each $L$, plotted as black dots, increases exponentially with $L$.  }
    \label{fig:1_const_MASs_data}
\end{figure}


\begin{table}[h]
\begin{tabular}{c|ccccc|ccccc}
   Max.\ size & \multicolumn{5}{c|}{CCRN properties}  & \multicolumn{5}{c}{$\#$ reactions in a MAS}  \\
   $L$ & $|\mathcal{C}|$ & $|\mathcal{R}|$ & $\ell$ & $\imath$ & $\delta$ & 2 & 3 & 4 & 5 &6  \\
   \hline 
    3 & 6 & 6 & 3 & 3 & 1 & 2 & 0 & 0 & 0 &0\\
    4 & 11 & 14 & 5 & 8 & 3 & 9 & 11 & 0 & 0 &0 \\
    5 & 17 & 26 & 7 & 16 & 6 & 24 & 98 & 48 & 0&0\\
    6 & 24 & 44  & 9 & 29 & 10 & 48 & 461 & 768 & 331&0\\
    7 & 32 & 68 & 11 & 47 & 15 & 85 & 1549& 6028& 5673& 1709
%    8 & 41 & 100 & 13 & 72 & 21 & & & &&\\
%    9 & 51 & 140 & 15 & 104 & 28 & & & &&\\
%    10 & 62 & 190 & 17 & 145 & 36 & & & &&\\
\end{tabular}
\caption{\label{table:1cluster} The complete 1-constituent CCRN of size $L$ and order two is considered for $L$ from $3-7$. The topological properties of the CCRN, such as number of linkage classes $\ell$ and deficiency $\delta$, are shown in the left half (for calculations, see Appendix \ref{app:kernel}). The number of MASs for a given number of reactions is shown in the right half (for a visual representation, see Fig.\ \ref{fig:1_const_MASs_data}).}
\end{table}

\subsection{Complete CCRN}
\label{sec:CCRN-complete}



\begin{figure}
    \centering
    \includegraphics[width=0.45\textwidth]{figures/cluster_CCRN_1d.png}
    \caption{The first few reactions in the complete 1-constituent CCRN of order two. The reactions containing clusters only up to length four ($L=4$) are shown in yellow boxes. }
    \label{fig:CCRN_1d}
\end{figure}






\subsubsection{1 constituent CCRN}
\label{sec:CCRN-1constituent}
Let $\mathcal{H}_1^L = (\mathcal{S},\mathcal{C},\mathcal{R})$ denote a complete 1-constituent CCRN of length $L$ and order two. 1-constituent CCRN of length $L$ means that the species set consists of clusters up to length $L$, denoted as $\mathcal{S} = \{ \overline{1}, \overline{2}, \ldots, \overline{L}\}$. By order two, we mean that all complexes are at most of width two, and the set of complexes is denoted as $\mathcal{C} = \{ \overline{a} \}\cup\{ \overline{a} + \overline{b}\}$ for $\overline{a},\overline{b} \in \mathcal{S}$ and $b\leq a \leq L$. By \textit{a complete CCRN}, we mean that the set of reactions contains all the possible reactions that are allowed by the conservation law, i.e.\ 
\begin{align*}
   \mathcal{R} =& \{ \ce{$\overline{a} + \overline{b}$ <=> $\overline{c} + \overline{d}$} | a+b = c+d \} \cup \\
&   \{ \ce{$\overline{a} $ <=> $\overline{c} + \overline{d}$} | a = c+d \}. 
\end{align*}
The first few linkage classes of the complete 1-constituent CCRN of order two are shown in Fig.\ \ref{fig:CCRN_1d} and the complete CCRN for $L=4$ is highlighted in yellow. The information about the reaction networks and the MASs that they contain for the complete 1-constituent CCRNs of $L$ ranging from $3$ to $7$ is collected in Table \ref{table:1cluster} and Fig.\ \ref{fig:1_const_MASs_data}. 



Consider the complete 1-constituent CCRN with $L=4$ of order two, the reaction network for which is shown in Fig.\ \ref{fig:CCRN_1d}. Notice that this subnetwork has $5$ linkage classes, $11$ complexes, and $14$ reactions. An application of the algorithm for enumerating all the MASs of the CCRN discussed in Sec.\ \ref{sec:algorithms} yields a list of $20$ subnetworks, shown in Fig.\ \ref{fig:cluster_n4}. In the list, there are $9$ MASs with two reactions and $11$ MASs with three reactions. Since the CCRN has $4$ species, and $1$ conservation law, the stoichiometric subspace is of dimension $3$. Moreover, due to the conservation law, from Lemma \ref{lemma:food} each MAS must possess at least one non-core species. This means that there cannot be any $4$ reaction MAS as it would require at least $5$ distinct species, which would be a contradiction. 

Upon obtaining a list of MASs for a given CRN, we would like to understand how their species-productive and partition-productive cones intersect. For the $L=4$ CCRN considered above, using Theorem \ref{theorem:partition-prod}, any MASs within a partition have identical species-productive cones and it is identical to the partition-productive cone. Thus, we do not show any plot for the species-productive cone intersection data. Moreover, any two partitions consisting of the same core of $3$ species must have the same partition-productive cone, as the stoichiometric subspace is of dimension $3$. For example, the partition-productive cones of $\{\{\overline{1}\},\{\},\{\{\overline{2},\overline{3},\overline{4}\}\}\}$ and $\{\{\},\{\},\{\overline{1},\{\overline{2},\overline{3},\overline{4}\}\}\}$ are identical. The information of the intersection of partition-productive cones for different FWMC partitions are shown in the bottom panel of Fig.\ \ref{fig:cluster_n4}.  

Unlike the $L=4$ case, for the complete 1-constituent CCRN with $L=5$ of order two, there can be MASs in the same equivalence class whose species-productive cones do not intersect. Moreover, it is not a priori clear whether or not two partition-productive cones of different partitions will intersect. Thus we employ our visualization scheme and display this information in Fig.\ \ref{fig:intersection_partition_5}. As described in Sec.\ \ref{sec:visualization}, for a particular equivalence class we also show the intersection information of the species-productive cones in Fig.\ \ref{fig:cone_intersection}.   


\subsubsection{L=3 1-constituent CCRN}
\label{sec:L_3_example}
\begin{figure}
    \centering
    \includegraphics[width=0.4\textwidth]{figures/L_3_basis.png}
    \caption{The 2-dimensional stoichiometric subspace along with the autocatalytic productive cones for the $L=3$ complete 1-constituent CCRN are shown. The $6$ rays emanating from the origin, labelled by their directions in the $3-$dimensional space, form the edges of productive cones where one species is consumed and another is produced. For instance, the ray $[-3,0,1]$ corresponds to the direction in the stoichiometric space where $\overline{1}$ is consumed, $\overline{3}$ is produced, and $\overline{2}$ remains constant.}
    \label{fig:cluster_L3}
\end{figure}




Consider the complete 1-constituent CCRN with $L=4$ of order two, given by 
\begin{align*}
    \mathcal{H}_1^3 &= (\mathcal{S},\mathcal{R})\\
    \mathcal{S} &= \{ \overline{1},\overline{2},\overline{3}\}\\
    \mathcal{R} &= \{ \ce{$\overline{1} + \overline{1}$ <=>[k_11][k_2] $\overline{2}$},\\
    & \phantom{--} \ce{$\overline{1} + \overline{2}$ <=>[k_12][k_3] $\overline{3}$},\\
    & \phantom{--} \ce{$\overline{1} + \overline{3}$ <=>[k_13][k_22] $\overline{2}+ \overline{2}$}\}.
\end{align*}
This network has two MASs, namely
\begin{align*}
    A_1 =& \{ \overline{1} + \overline{2} \to \overline{3},\\
         & \phantom{0} \overline{1} + \overline{3} \to \overline{2} + \overline{2}\},\\
     A_2 =&  \phantom{...} \{ \phantom{...} \overline{2} \to \overline{1} + \overline{1},\\
     & \phantom{0} \overline{3} + \overline{1} \to \overline{2} + \overline{2}\}.
\end{align*}

Since the CCRN has one conservation law, the stoichiometric subspace is of dimension $2$ and perpendicular to $[1,2,3]^T$. The two-dimensional stoichiometric subspace is shown in Fig.\ \ref{fig:cluster_L3}. The $6$ rays emanating from the origin, labelled by their directions in the $3-$dimensional space, form the edges of productive cones where one species is consumed and another is produced. The species-productive cones of the two autocatalytic cycles are also labelled. Note that the species-productive regions is the complete cone within the bounding rays, and the finite boundary is drawn simply to enhance visualization. For example, the species-productive cone of the MAS $A_1$ is bounded by the rays $[-2,1,0]$ and $[-3,0,1]$. Notice that the FWMC partition of $A_1$ is $\{\{\overline{1}\},\{\},\{\{\overline{2},\overline{3}\}\}\}$, consistent with the species-productive region that strictly consumes species $\overline{1}$ and strictly produces species $\overline{2}$ and $\overline{3}$.

We also want to make a few remarks about the details that are absent in Fig.\ \ref{fig:cluster_L3}. Firstly, for every MAS, there is a minimal drainable subnetwork obtained by reversing the reaction edges. The species-productive cone of these drainable networks will be the negative of their autocatalytic counterparts. Secondly, notice that there is no MAS with the partition $\{\{\overline{2}\},\{\},\{\{\overline{1},\overline{3}\}\}\}$. In case there was one, as will be if we allow reactions of order $3$, then its species-productive cone would be bound by the rays $[0,-3/2,1]$ and $[1,-1/2,0]$.

Notice from Table \ref{table:1cluster} that this CCRN has a deficiency of one. This means that when taken with mass-action kinetics, it has the potential to exhibit multistability. To investigate this, we used Feinberg's deficiency one algorithm \cite{feinberg1988chemical} in Appendix \ref{app:deficiency-1} and obtained rate constants for which the CCRN has multiple steady states. An example of multiple steady states and the rate constants using which they are obtained are reported in Fig.\ \ref{fig:cluster_L3_fp}.
\begin{figure}
    \centering
    \includegraphics[width=0.4\textwidth]{figures/L_3_fixed_points.png}
    \caption{Multiple fixed points for the complete 1-constituent CCRN with $L=3$ and order two obtained using Feinberg's deficiency one algorithm (see Appendix \ref{app:deficiency-1}). The parameters for the algorithm are chosen to be $\mu_{\overline{1}}=1,\mu_{\overline{2}}=3,\mu_{\overline{3}}=4=\eta$, and the rate constants for the model are $k_{11} = 1, k_2 = 1.001, k_{12} =1.1 , k_{3}=1.011, k_{13} = 0.146, k_{22} = 0.027$. }
    \label{fig:cluster_L3_fp}
\end{figure}

\subsubsection{Number theoretic result for 1-constituent CCRN of order two}

Since CCRNs are defined by number conservation laws, finding autocatalytic subnetworks in CCRN corresponds to number-valued solutions of systems of equations. In this subsection we will show that if there exists a two-reaction MAS in a partition of a 1-constituent CCRN, then it must be unique.

\begin{lemma}
    Let $A$ be a $2$-reactions MAS in a 1-constituent CCRN of order two. Then, there does not exist any other MASs in the CCRN with the same FWMC partition as $A$.
\end{lemma}
\begin{proof}
    Let $A_C = \{\overline{i}, \overline{j}\}$ be the core species in $A$ (assume without loss of generality that $i>j$). 
    The possible 2-reaction MASs with that set of core species are given by reactions of the form
    \begin{align*}
        \begin{cases}
            \overline{i} + \overline{\ell_1} \to& \overline{j} + \overline{\ell_2},\\
            \overline{j} + \overline{\ell_3} \to& 2\overline{i}.
        \end{cases}\\
        \begin{cases}
            \overline{i} + \overline{m_1} \to & 2\overline{j}, \\
            \overline{j} + \overline{m_2} \to & \overline{i} + \overline{m_3}.
        \end{cases}
    \end{align*}
    Note that each non-core species can take the value $0$, indicating the absence of a reactant. However, due to Lemma \ref{lemma:food}, the conservation law will require that at least one food species is non zero. Observe, also, that in both types of potential MASs there is a reaction which is uniquely determined by $\bar i$ and $\bar j$ (being then $\bar \ell_3$ and $\bar m_1$ also unique for this choice of the core set). Concretely $\bar \ell_3 = \overline{2i-j}$ and $\bar m_1 = \overline{2j-i}$. Note that these reactions are different since it is assumed that $i\neq j$. Thus, the above reactions reduce to:
        \begin{align*}
        M_1: \begin{cases}
            \overline{i} + \overline{j+\ell-i} \to& \overline{j} + \overline{\ell},\\
            \overline{j} + \overline{2i-j} \to& 2\overline{i}.
        \end{cases}\\
        M_2: \begin{cases}
            \overline{i} + \overline{2j-i} \to & 2\overline{j}, \\
            \overline{j} + \overline{i+m-j} \to & \overline{i} + \overline{m}.
        \end{cases}
    \end{align*}
     Now we will do a case-by-case analysis to show that each FWMC-partition will contain at most one solution of the systems of equations. 

     Note that the only possible partition of the species induced by the reactions in $M_1$ is:
     $FWMC(M_1) = 
     \{\bar j+\bar\ell-\bar i, \overline{2i-j}\}, \{\bar\ell\}, \{\{\bar i, \bar j\}\}
     $
     whereas the partition for $M_2$ is:
     $
     FWMC(M_2)=      \{\overline{2j-i}, \bar i+\bar m-\bar j, \}, \{\bar m\}, \{\{\bar i, \bar j\}\}
     $
     From both types of partition, the only possibility that both coincide is that $\bar l = \bar m$ and $\bar i = \bar j$ contradicting the assumption that $i>j$.
    
    

    % In the sets of reactions, by the conservation laws $\ell_3$ and $m_1$ are fixed to be $\ell_3=2i-j$ and $m_1=2j-1$, respectively. It is also clear for a given $\ell_1 \geq 0$ and $m_2 \geq i-j$, $\ell_2$ and $m_3$ are fixed uniquely. 

    % Consider the first set of equations. The only way there could be two MASs with the same partition if for a solution $(\ell_1,\ell_2,\ell_3)$ the interchanging of $\ell_1$ and $\ell_3$ still yields a solution $(\ell_3,\ell_2,\ell_1)$. However, we know that $\ell_3$ is fixed by $i,j$. Thus, this cannot be the case unless $\ell_3$ and $\ell_1$ are identical. In any case, however, there is a unique MASs in a given partition. An analysis of the second set of equations yields a similar conclusion about the second set.

    % There could, however, be another way in which two different MASs are in the same partition, namely if $(\ell_1,\ell_3) = (m_1,m_2)$ or $(\ell_1,\ell_3) = (m_2,m_1)$. The second case is ruled out since we know that $\ell_3=2i-j$ and $m_1=2j-i \neq \ell_3$. Suppose $(\ell_1,\ell_3) = (m_1,m_2)$. A simply calculation by plugging in the expressions for $\ell_3=m_2$ and $\ell_1 = m_1$ in the sets of equations yields that the resulting system of equations reduces to a single set
    % \begin{align*}
    %     \begin{cases}
    %         \overline{i} + \overline{m_1} \to& 2\overline{j},\\
    %         \overline{j} + \overline{\ell_3} \to& 2\overline{i}
    %     \end{cases}
    % \end{align*}
    % for which a solution is uniquely determined by $i,j$. 
    
    % Thus we conclude for all the cases that the species in the FWMC-partition uniquely specify a 2-reaction MAS in a 1-constituent CCRN. 
\end{proof}
% \textcolor{red}{What about the case
% $\begin{array}{rrl}
%     R_1:  &  \overline{i} + \overline{2j-i} &\to 2\overline{j} \\
%     R_2:   & \overline{j} + \overline{i+k - j} &\to \overline{i} + \overline{k}
% \end{array}$
% }
Given two different species $\overline{i}$, $\overline{j}$ with $i>j\geq 1$, the above result also provides a way to construct all the 2-reactions MASs with those species being the core species. The explicit reactions taking part of the MASs are in the form:
\begin{itemize}
    \item In case $i \leq \frac{L}{2}$:
$$
MAS_1 = \left\{\begin{array}{rrl}
     R_1: &       \overline{i} + \overline{j+\ell-i} \to& \overline{j} + \overline{\ell},\\
     R_2: &       \overline{j} + \overline{2i-j} \to& 2\overline{i}.
\end{array}\right.
$$
for all $\ell \in [i-j,L] \cap \mathbb{Z}$.
\item In case $\frac{i}{2} \leq j \leq \frac{L}{2}$:
$$
MAS_2 = \left\{\begin{array}{rrl}
      R_1:  &      \overline{i} + \overline{2j-i} \to & 2\overline{j}, \\
     R_2:  &       \overline{j} + \overline{i+m-j} \to & \overline{i} + \overline{m}.
\end{array}\right.
$$
for all $m \in [0,L+j-i] \cap \mathbb{Z}$.
\end{itemize}

\subsubsection{Computational challenges in scaling}
\label{sec:comp_challenge}
Consider the complete 2-constituent CCRN of order two, defined as
\begin{align*}
\mathcal{H}_2 =& (\mathcal{S},\mathcal{R})  \\
\mathcal{S} = & \{ \overline{a,b} | a,b \in \mathbb{Z}_{\geq 0} \}\\
\mathcal{R} =& \{ \overline{a,b} + \overline{c,d} \to \overline{e,f} + \overline{g,h} |\\
              & \phantom{--}    a+c = e+g \text{ and } b+d = f+h \}.
\end{align*}
We wish to remark that for $\mathcal{H}_2$ and other CRNs with a large number of reactions, generating the whole set of autocatalytic subnetworks is computationally challenging. As the number of reaction increases, the dimension of the space where the MASs are to be found also increases, and even finding a  single MAS in the network implies solving a difficult optimization problem that might be computationally costly. Even being each of the cycles polynomial-time solvable (which does not seem to be the case), the complexity of the enumeration algorithm may turn into exponential (see, e.g., \cite{garey1979computers}) since the number of subnetworks increases considerable with the number of species and reactions. Thus for practical reasons, it maybe more feasible to consider sparser rule-generated networks, as explained in the next subsection and shown in Fig.\ \ref{table:cluster-paradigm-rules}.



% {\color{blue} Victor: Column $|\mathcal{C}|$ is is $(r^2 + 5r - 2)/2$ (where $r$ is max. size). One can compute how large can be for larger $r$s. Column $\mathcal{R}$ can be also identified with a sequence where term $n$ can be computed from term $n-1$ as $a(n) = 3a(n-1)-2a(n-2)-2a(n-3)+3a(n-4)-a(n-5)$. }

\subsection{Rule generated CCRNs}
\label{sec:rule_gen}
\begin{table}[]
    \centering
    \begin{tabular}{cc}
    \hline\\
 Monomer activation &           \ce{$\overline{1}$ <=>  $\overline{1}^*$}    \\
Catalyzed activation &    \ce{$\overline{C} + \overline{1}$ <=>  $\overline{1}^*+\overline{C}$}       \\ \\
    \hline\\
    Polymerization & \ce{$\overline{1}+\overline{1}^*$ <=>  $\overline{2}$}\\
                    & \ce{$\overline{2}+\overline{1}^*$ <=>  $\overline{3}$}\\
                    & \vdots \\
    1-constituent   & \ce{$\overline{j}+\overline{1}^*$ <=>  $\overline{j+1}$}\\ \\
  2-constituent     & 
        $\begin{cases}
            \ce{$\overline{j,k}+\overline{1,0}^*$ <=>  $\overline{j+1,k}$}\\
            \ce{$\overline{j,k}+\overline{0,1}^*$ <=>  $\overline{j,k+1}$} 
        \end{cases}$
  \\ 
  \vdots  & \vdots \\ \\
  \hline \\
  \begin{tabular}{@{}c@{}}Templated \\ polymerization\end{tabular}  & 
  $\begin{cases}
      \ce{$\overline{j,k}+ \overline{k-1,j} +\overline{1,0}^*$ <=>  $\overline{j,k}+ \overline{k,j}$}\\
      \ce{$\overline{j,k}+ \overline{k,j-1} +\overline{0,1}^*$ <=>  $\overline{j,k}+ \overline{k,j}$}
  \end{cases}$\\ \\
  \hline \\
  Fusion & \ce{$\overline{j} + \overline{k}$ <=>  $\overline{j+k}$ }\\
  Fission & \ce{$\overline{j}$ <=>  $\overline{j-k} + \overline{k}$ } \\
  \\
  \hline
    \end{tabular}
    \caption{Several examples of rules for generating a sparse CCRN relevant to biochemistry. Here, in accordance with the definition of a CCRN, the variables $j,k$ must be chosen such that the resulting clusters have a non-negative integer quantity of each constituent. }
    \label{table:cluster-paradigm-rules}
\end{table}


A naive computational modelling of artificial chemistry \cite{banzhaf2015artificial} often runs into issues of memory, storage and processing due to a combinatorial explosion of chemical species and reactions involved in the CRN. For instance, if one uses string chemistry \cite{moyer2020stoichiometric} (or polymer sequence chemistry) to model polymers of size up to $N$ with $D$ distinct monomers ($D$ constituents), a straightforward calculation shows that one is required to track $\sum_{i=1}^N D^i > D^N$ species. Since for any physically relevant model $N\gg D$, so it is clear that the computation will soon become intractable as one increases the size of the string (polymer) $N$ while also considering their relevant network of interactions. On the other hand, even a simple graph-grammar for generating artificial chemistry, with the application of \cite{andersen2016software}, can generate a reaction network that is computationally unfeasible. Moreover, even if one restricts the reaction network in the computationally feasible regime, an enumeration of autocatalytic subnetworks or motifs will generically be an extremely long list. In such a scenario, it might be unclear how to simplify the model without stripping away the essential details or adding artifacts. 



We argue that the \textit{Cluster framework}, or considering CCRNs induced by CRNs, can help alleviate some of the issues raised above. Any rule generated CRN, in string chemistry or built using graph-grammar methods, must induce rules on the CCRN. Since the induced CCRN can have an exponentially reduced species set, it offers a way to constrain the model to a more computationally feasible regime. Since autocatalytic networks are preserved under this coarsening (with some caveats mentioned in the introduction to this section), one can use the CCRN to obtain a smaller list of MASs. One can then gradually complexify the CCRN by adding more species with the same conserved quantities until the required behavior of the CRN is captured by the CCRN.

For example, consider polymer chemistry with two monomers $A$ and $B$. For an arbitrary polymer $\omega$, the addition of an $A$ or $B$ yields $\omega \cdot A$ or $\omega \cdot B$, respectively. In the cluster framework, if we map $A$, $B$, and $\omega$ to $\overline{1,0}$, $\overline{0,1}$, and $\overline{j,k}$, respectively, the polymerization reactions in the CCRN become
\begin{align*}
    \ce{$\overline{j,k}+\overline{1,0}$ <=>  $\overline{j+1,k}$},\\
        \ce{$\overline{j,k}+\overline{0,1}$ <=>  $\overline{j,k+1}$}.
\end{align*}
In realistic chemistry, however, a monomer addition to a polymer is only done by an activated monomer and not an unactivated monomer. Thus, we may want to distinguish activated and unactivated monomers in our model, for which we can add extra species $\overline{1,0}^*$ and $\overline{0,1}^*$. The resulting polymerization reactions and other examples of rule generated CCRNs are shown in Table \ref{table:cluster-paradigm-rules}. 


%----------------section end -----------------------------
\section{Discussion and future research}
\label{sec:discussion}



In this work, we began by reviewing the notions of autocatalysis in literature in Sec.\ \ref{sec:AC-motif}. We term the autocatalysis that can be detected from the stoichiometric matrix as \textit{stoichiometric autocatalysis}, and show that it is more constrained than \textit{formal} or \textit{exclusive} autocatalysis. Any system showing stoichiometric autocatalysis is formally and exclusively autocatalytic, which means that enumeration of stoichiometrically autocatalytic subnetworks can only underestimate the autocatalytic structure in a CRN. 

In \cite{blokhuis2020universal}, it was shown that there are exactly five types of minimal stoichiometrically autocatalytic cores, where a core is obtained by only selecting a few species from the complete set of species that participate in the autocatalytic reactions. In Sec.\ \ref{sec:organization}, we extend their work by defining stoichiometric autocatalysis for autocatalytic subnetworks defined on the complete set of participating species in the reaction network. We show that for any such subnetwork, the species set can be uniquely assigned a \textit{food-waste-member-core} partition, and define geometrically relevant quantities for them. We then derive mathematical results about these quantities, and provide an algorithm for exhaustively enumerating all minimal autocatalytic subnetworks (MASs). We also show how to take the list of MASs and obtain a visual summary of their combinatorics. 

In Sec.\ \ref{sec:applications}, we define the \textit{cluster chemical reaction network} (CCRN) framework. Since CCRNs are defined using conservation laws, we argue that this provides a natural framework for coarse-graining real CRNs. We then use the tools developed in Sec.\ \ref{sec:organization} to organize a list of MASs for maximally connected 1-constituent CCRNs with $L$ species and show that the number of MASs increases exponentially with $L$. While the fully connected CCRNs might not have physical relevance, in Sec.\ \ref{sec:rule_gen} we comment on how sparse CCRNs maybe obtained using rules induced by real polymer or biological chemistry (see Table \ref{table:cluster-paradigm-rules}).  


Our work opens up a few lines of future research, namely:
\begin{enumerate}
    \item For a simple network in Example \ref{eg:CRN-1}, proven to be exclusively autocatalytic in Example \ref{eg:CRN-exclusive}, we proposed a modification such that the resulting graph was stoichiometrically autocatalytic in Example \ref{eg:CRN-extension}. Is there a general algorithm to modify any exclusively autocatalytic motif to a stoichiometrically autocatalytic motif? Having such an algorithm would allow us to use the typology of \cite{blokhuis2020universal} and the results in this work to understand the autocatalytic properties of any CRN. 
    \item Within the CCRN framework, we restricted the clusters to have nonnegative constituents. However, one can also consider the clusters defined on the complete integer lattice. The resulting reaction network could then be used to model real nuclear reactions. For example, if the positive conserved quantity corresponds to positive charge or subatomic particle type, the negative conserved quantity would refer to a negative charge or anti-particle, respectively.    
    \item The generation of MASs, for general CRNs, and especially for CCRNs, is computationally costly. Our optimization-based approach requires solving a series of binary mathematical programming problems with an increasing number of linear constraints. Further research includes the particular study of these optimization problems, in particular, its polyhedral properties that will allow to strengthen the formulation and solve them more efficiently. Exploiting the algebraic properties behind conservation laws of sequences of integer numbers would also lead us to understand the mathematical insights of these networks.
    \item Realistic CRNs will not coarse-grain to a fully connected CCRN, thus appreciably reducing the number of reactions in the network, as well as the MASs that they contain. As a future investigation, similar to \cite{anderson2022prevalence}, can we consider randomly generated CCRNs in an Erdos-Renyi framework and compute a bound on the number of autocatalytic cycles which persist as a function of the probability of a reaction being allowed in the network? 
    \item CCRNs are defined in terms of their conservation laws, and have more degrees of symmetry than generic CRNs. Having completely enumerated the stoichiometrically autocatalytic ecology of a complete CCRN, can we better understand and characterize their kinetics?        
\end{enumerate}


%----------------section end -----------------------------

\section*{Acknowledgements}
We want to thank Connor Simpson for help with combinatorics and programming, Alex Plum, Tymofii Sokolskyi, and Zhen Peng for biological discussions, and Asvin G., Dave Auckly, Diego Rojas, Gautam Neelakantham, Jeff Linderoth, and Vladimir Sotirov for mathematical discussions. This work was partially funded by the National Science Foundation, Division of Environmental Biology (Grant No: DEB-2218817). V. Blanco acknowledges the financial support by the Spanish Ministerio de Ciencia e Innovaci\'on AEI/FEDER grant number PID2020-114594GBC21 and IMAG-Maria de Maeztu grant CEX2020-001105-M /AEI /10.13039/501100011033, Junta de AndalucÃ­a projects P18-FR-1422/2369, AT 21\_00032, FEDERUS-1256951, B-FQM-322-UGR20, and UE-NextGenerationEU (ayudas de movilidad para la recualificaci\'on del profesorado universitario).

\appendix

\section{Deficiency theory and multistability in CRNs}
\label{app:deficiency}
In \cite{feinberg2019foundations}, a CRN is defined by the triple $\{\mathcal{S},\mathcal{C},\mathcal{R}\}$, where $\mathcal{S}, \mathcal{C},$ and $\mathcal{R}$ are the set of species, complexes, and reactions respectively. Recall from Sec.\ \ref{sec:hypergraphs} that a CRN is a hypergraph with hypervertices in the set $\mathcal{C}$ and hyperedges in $\mathcal{R}$. The stoichiometric matrix $\mathbb{S}$ is defined such that (see Eq. \ref{eq:stoich_mat})
\begin{align*}
    \text{cols}(\mathbb{S}) &= \{y'-y|y\to y' \in \mathcal{R}\}. \nonumber
\end{align*}


Alternatively, $\mathbb{S}$ can be written as the product $Y \cdot \mathbb{M}$, where the columns of $Y$ are the stoichiometries of the hypervertices 
\begin{align*}
    \text{cols}(Y) = \{ \phantom{.} y \phantom{.} | y \in \mathcal{C}\}
\end{align*}
and $\mathbb{M}$ is the oriented vertex-edge incidence matrix of the hypergraph (see \cite{balandin1940structural,feinberg2019foundations})
\begin{align*}
    \text{cols}(\mathbb{M}) = \{e_{y'} - e_y |y\to y' \in \mathcal{R}\},
\end{align*}
where $e_y$ is the vector with $1$ at hypervertex $y$ and $0$ otherwise. Note that $Y$ and $\mathbb{M}$ are of sizes $|\mathcal{S}|\times|\mathcal{C}|$ and $|\mathcal{C}|\times|\mathcal{R}|$, respectively, yielding $\mathbb{S}$ of size $|\mathcal{S}|\times|\mathcal{R}|$. 

Since $\mathbb{S}$ is a linear operator, using the rank-nullity theorem we get
\begin{align}
    \text{dim}(\text{domain}(\mathbb{S})) &= \text{dim}(\text{im}(\mathbb{S})) + \text{dim}(\text{ker}(\mathbb{S})). 
    \label{eq:rank_null_stoic}
\end{align}
Let $E, V, s$ denote the number of reactions (hyperedges), complexes (hypervertices) and the dimension of the stoichiometric subspace $S$, respectively. Then, by definition,
\begin{align*}
    \text{dim}(\text{domain}(\mathbb{S})) &= E,\\
    \text{dim}(\text{im}(\mathbb{S})) &= s.
\end{align*}
Using $\mathbb{S} = Y \cdot \mathbb{M}$, we have
\begin{align}
    \text{dim}(\text{ker}(\mathbb{S})) &= \text{dim}(\text{ker}(\mathbb{M})) + \text{dim}(\text{ker}(Y) \cap \text{im}(\mathbb{M})). \label{eq:ker_S}
\end{align}
Notice that the complexes and reactions, without referring to the underlying species, form a graph, the vertex-edge incidence matrix of which is given by $\mathbb{M}$. Using properties of the vertex-edge incidence matrix, if the number of independent loops in the graph are $\imath$, then
\[ \text{dim}(\text{ker}(\mathbb{M})) = \imath. \]
Denoting $\text{dim}(\text{ker}(Y) \cap \text{im}(\mathbb{M}))$ by $\delta$, also referred to as the \textit{deficiency} of the CRN, substituting the above and Eq. \ref{eq:ker_S} in Eq. \ref{eq:rank_null_stoic}, we get
\begin{align}
    \delta &= E - \imath - s. \label{eq:delta_1}
\end{align}

Furthermore, let us denote the number of connected components or \textit{linkage classes} of the complex-reaction graph by $\ell$. Notice that $\ell$ and $\imath$ are the zeroth and first Betti numbers of a graph, respectively \cite{hatcher2005algebraic}. Using the formula for the Euler characteristic of a graph, we have the relation
\begin{align}
    V - E & = \ell - \imath, \label{eq:euler}
\end{align}
where $V$ is the number of vertices in the graph. Substituting the above in Eq. \ref{eq:delta_1}, we obtain the more widely used expression for the deficiency
\begin{align}
    \delta & = V - \ell - s. \label{eq:delta_2}
\end{align}
For direct derivations of Eq. \ref{eq:delta_2} without resorting to a counting of independent loops, see \cite{smith2017flows, feinberg2019foundations}.


Intuitively, deficiency is a topological property of the reaction network that informs about the steady state dynamics. In particular, a non-zero deficiency is a necessary, but not sufficient, condition for the dynamics to exhibit \textit{multistability} or multiple steady states. In certain cases, such as for deficiency one networks \cite{feinberg1988chemical}, there exist algorithms that can help classify whether or not multiple steady states can exist for a given CRN and identify the rate constants such that under mass-action kinetics it exhibits multistability. Once a model exhibiting multistability is finalized, one can also use algorithms from \cite{gagrani2022action,dykman1994large} to estimate the most-likely path the system will take, along with its probability, to escape from one steady state to another. 


\section{Kernel of the stoichiometric matrix for the complete 1-constituent CCRN}
\label{app:kernel}
\begin{figure}
    \centering
    \includegraphics[width=0.5\textwidth]{figures/clust_1d.png}
    \caption{Representation of the complete 1-constituent CCRN with $L=6$. Complex label $(a,b)$ on a lattice point corresponds to the complex $\overline{a} + \overline{b}$. All the complexes connected by a red line correspond to a fully connected graph in the reaction network. For example, the red line connecting $\{(5,0),(4,1),(3,2)\}$, labelled $a+b=5$, corresponds to the reactions $\{\overline{5} \iff \overline{4} + \overline{1} \iff \overline{3} + \overline{2} \iff \overline{5}\}$ in the CCRN. }
    \label{fig:1-cluster-complexes}
\end{figure}

Let $\mathcal{H}_1^L = (\mathcal{S},\mathcal{C},\mathcal{R})$ denote a complete 1-constituent CCRN of length $L$ and order two. 1-constituent CCRN of length $L$ means that the species set consists of clusters up to length $L$, denoted as $\mathcal{S} = \{ \overline{1}, \overline{2}, \ldots, \overline{L}\}$ (for a detailed definition, see Sec.\ \ref{sec:CCRN-1constituent}.) For a diagrammatic representation of the reaction network, see Fig.\ \ref{fig:1-cluster-complexes}. Here, following Appendix \ref{app:deficiency}, we will perform the necessary calculations to calculate the kernel of the stoichiometric matrix for a general length $L$.

First, we will count the number of complexes in a complete 1-constituent CCRN of length $L$. Let us denote the set of complexes of size $N$ as $\mathcal{C}^N_L$. Recall that size of a complex is defined as the sum of the conserved quantities in each cluster. For $2\leq N\leq L$, 
\[
\mathcal{C}^N_L=\{\overline{N},\overline{N-1}+\overline{1},\ldots, \overline{\floor{N/2}{}} +\overline{\ceil{N/2}{}}\}.
\] 
For $L < N < 2L -2$, 
\[
\mathcal{C}^N_L = \{ \overline{N-L} + \overline{L}, \ldots, \overline{\floor{N/2}{}} +\overline{\ceil{N/2}{}}\}.
\]
Thus, the number of complexes as a function of $N$ and $L$ are 
\begin{align*}
    |\mathcal{C}^N_L| &= 
    \begin{cases}
        \floor{N/2} + 1 \text{ for } 2\leq N\leq L,\\
        L + 1 - \ceil{N/2}  \text{ for } L < N \leq 2L-2.
    \end{cases}
\end{align*}
The total number of complexes for a CCRN of length $L$ is thus 
\[|\mathcal{C}_L| = \sum_N |\mathcal{C}^N_L| = \frac{(L+1)(L+2)}{2}-4.
\]
This can be found by either summing the individual contributions or by observing from Fig. \ref{fig:1-cluster-complexes} that the complexes occupy the lower triangle including the diagonal for a lattice with $L+1$ rows and columns excluding $4$ points. 

Let us denote the set of reactions of complexes of size $N$ by $\mathcal{R}^N_L$. Since the reaction network of the complete CCRN is obtained by connecting all allowed reactions, for complexes of size $N$, the number of reactions are 
\begin{align*}
    |\mathcal{R}^N_L| &=
    \begin{cases}
        2{ \floor{N/2} + 1 \choose 2} \text{ for } 2\leq N\leq L,\\
        2{L + 1 - \ceil{N/2}  \choose 2} \text{ for } L < N \leq 2L-2,        
    \end{cases}
\end{align*}
where the factor of $2$ is multiplied since all reactions are reversible.
Since, each $N$ contributes exactly one linkage class (or connected component), the total number of linkage classes $\ell$ for the complete CCRN is $2L - 3$.

Recall from Appendix \ref{app:deficiency}, the kernel of the stoichiometric matrix is the sum of the independent loops $\imath$ and the deficiency $\delta$, where using Equations \ref{eq:euler} and \ref{eq:delta_2} we have
\begin{align}
    \imath &= |\mathcal{R}_L| - |\mathcal{C}_L| + \ell\\ \nonumber 
    &= \sum _{N=2}^{2L-2}|\mathcal{R}^N_L| - \frac{L^2-5L-2}{4} \nonumber \\
    \delta &= |\mathcal{C}_L| - \ell - s \\\nonumber
        &= \frac{(L+1)(L+2)}{2}-4 - (2L - 3) - (L-1) \\
        &= \frac{(L-1)(L-2)}{2}. \nonumber %\label{eq:delta_3}
\end{align}


The intersection $\text{Im}(\mathbb{M}) \cap \text{Ker}(Y)$, the dimension of which is the deficiency $\delta$, counts the null flows of the CCRN which are not simple loops in the network. While an explicit basis can be obtained using subnetworks of the form
\begin{align*}
    \overline{m} \to \overline{m-k} + \overline{k}\\
    \overline{n} + \overline{k} \to \overline{n+k} \\
    \overline{n+k} + \overline{m-k} \to \overline{n} + \overline{m},
\end{align*}
we will not be pursuing it since we do not need it for any result in this work. 

\section{Deficiency-one algorithm for L=3 1-constituent CCRN}
\label{app:deficiency-1}
The complete $L=3$ 1-constituent CCRN is the simplest complete CCRN, and as we will see, it possesses non-trivial dynamical properties. In particular, from Table \ref{table:1cluster}, it can be seen that the deficiency $\delta$ for the CCRN is one. This hints at the possibility that the CRN, when taken with mass-action kinetics, exhibits multistability. In this section, using Feinberg's deficiency-one algorithm from \cite{feinberg1988chemical,feinberg2019foundations}, we will confirm that the network indeed has the capability of exhibiting multiple steady states, and find a relation between the rate constants and concentrations where that is the case. The algorithm works by converting the CRN into a system of linear inequalities, and if they have a solution, they can be used to obtain a relation between the rate constants and multiple steady states. 

The $L=3$ 1-constituent CCRN, denoted by $\mathcal{G} = (\mathcal{S},\mathcal{C},\mathcal{R})$, is given by the species set $\mathcal{S} = \{ \overline{1},\overline{2},\overline{3}\}$, complex set $\mathcal{C} = \{\overline{1} + \overline{1},\overline{2},\overline{1} + \overline{2},\overline{3},\overline{1} + \overline{3},\overline{2} + \overline{2}\} $ and reaction set
\begin{align*}
    \mathcal{R} &= \{ \ce{$\overline{1} + \overline{1}$ <=>[k_11][k_2] $\overline{2}$},\\
    & \phantom{--} \ce{$\overline{1} + \overline{2}$ <=>[k_12][k_3] $\overline{3}$},\\
    & \phantom{--} \ce{$\overline{1} + \overline{3}$ <=>[k_13][k_22] $\overline{2}+ \overline{2}$}\}.
\end{align*}
Notice that the above reaction network is a regular \cite{feinberg1988chemical} deficiency-one reaction network, and is thus a valid candidate for the deficiency-one algorithm. To proceed with our calculation, we will follow closely the example in Chapter 17 of \cite{feinberg2019foundations} and Sec.\ 5 in \cite{feinberg1988chemical}.

The confluence vector $g\in \mathbb{R}^\mathcal{C}$ for our network can be calculated to be (up to sign)
\begin{align*}
    g_{\overline{2}} &= -1, &     g_{\overline{1} + \overline{1}} &= 1,\\
     g_{\overline{1} + \overline{2}} &= -1, &    g_{\overline{3}} &= 1,\\
    g_{\overline{1} + \overline{3}} &= -1, &     g_{\overline{2}+\overline{2}} &= 1.\\
\end{align*}
For our construction, we will use the following upper-middle-lower partition
\begin{align*}
    U := & \{\overline{1} + \overline{3},\overline{2} + \overline{2} \},\\
    M := & \{\overline{1} + \overline{2},\overline{3} \},\\
    L := & \{  \overline{1} + \overline{1},\overline{2}\}.
\end{align*}
The partition induces the following system of inequalities on the chemical potential,
\begin{align*}
    2 \mu_{\overline{2}} > \mu_{\overline{1}} + \mu_{\overline{3}} > \mu_{\overline{3}} = \mu_{\overline{1}}+\mu_{\overline{2}} > \mu_{\overline{2}} > 2 \mu_{\overline{1}}.
\end{align*}
The inequalities are satisfied by 
\begin{align*}
     \mu_{\overline{1}} & > 0,\\
      \mu_{\overline{2}} & > 2  \mu_{\overline{1}},\\
    \mu_{\overline{3}} &= \mu_{\overline{1}}+\mu_{\overline{2}},
\end{align*}
which clearly have a solution, for e.g.\ $ \mu_{\overline{1}} = 1,  \mu_{\overline{2}} = 3,  \mu_{\overline{3}} = 4$. Thus, the CCRN can exhibit multiple steady states. 

To find the rate constants at which the system exhibits multistability, we use subsection 5.3 in \cite{feinberg1988chemical}. Following the reference, let us denote the mass-action kinetics rate vector by $\kappa_{y\to y'} := k_{y\to y'} x^y $, where $x$ is the concentration vector and $x^y:= \prod_{i\in \mathcal{S}} x_i^{y_i}$. Choosing the monotonically increasing function to be the exponential $\phi(x) = e^x$, and choosing $\eta$ so that $\mu_{\overline{1}} + \mu_{\overline{3}} < \eta < \mu_{\overline{2}} $ we get 
\begin{align*}
    \kappa_{\overline{1} + \overline{3} \to \overline{2} + \overline{2}}
    &= 
    1 \cdot \frac{e^{2\mu_{\overline{2}}}- e^{\eta}}{e^{2\mu_{\overline{2}}} - e^{\mu_{\overline{1}}}e^{\mu_{\overline{3}}}}  = k_{13} x_{\overline{1}}  x_{\overline{3}},
    \\
        \kappa_{ \overline{2} + \overline{2} \to \overline{1} + \overline{3} }
    &= 
    1 \cdot \frac{e^{\mu_{\overline{1}+\overline{3}}}- e^{\eta}}{e^{2\mu_{\overline{2}}} - e^{\mu_{\overline{1}}}e^{\mu_{\overline{3}}}} 
    =
    k_{22} x_{\overline{2}}^2 ,\\
    \kappa_{\overline{1} + \overline{2} \to \overline{3}}
    &= 
     k_{12} x_{\overline{1}}  x_{\overline{2}},\\
        \kappa_{ \overline{3} \to \overline{1} + \overline{2} }
    &= 
     k_{12} x_{\overline{1}}  x_{\overline{2}} - 1  = k_{3} x_{\overline{3}} ,\\
    \kappa_{\overline{2} \to \overline{1} + \overline{1}}
    &= 
    1 \cdot \frac{e^{2\mu_{\overline{1}}}- e^{\eta}}{e^{2\mu_{\overline{1}}} - e^{\mu_{\overline{2}}}} = k_{2}  x_{\overline{2}},\\
        \kappa_{ \overline{1} + \overline{1} \to \overline{2} }
    &= 
    1 \cdot \frac{e^{\mu_{\overline{2}}}- e^{\eta}}{e^{2\mu_{\overline{1}}} - e^{\mu_{\overline{2}}}} = k_{11} x_{\overline{1}}^2. 
\end{align*}
From \cite{feinberg1988chemical}, it is also known that if $x := (x_{\overline{1}},x_{\overline{2}},x_{\overline{3}})$, then so is $x^* := (x_{\overline{1}} e^{\mu_{\overline{1}}}, x_{\overline{2}} e^{\mu_{\overline{2}}}, x_{\overline{3}} e^{\mu_{\overline{3}}})$, thus yielding multiple steady states for our system. 

\section{Details of the algorithm for detecting autocatalytic motifs}\label{app:alg}

In what follows we detail the optimization-based approach that we have designed to generate the set of MASs for a given CRN. Using the notation in this work, $\mathcal{S} = \{\bar 1, \ldots, \bar L\}$ is the set of species, $\mathcal{R}$ is the set of reactions, and $\mathbb{S}= (a_{ij})_{i\in \mathcal{S}, j\in \mathcal{R}} \in \mathbb{Z}^{n\times m}$ is the stoichiometric matrix of the CRN. It is worth mentioning that our approach is valid for any CRN, in particular for CCRN with one or more constituents.  

The goal of the algorithm that we propose is to determine sets of core species and reactions conforming a MAS by solving, sequentially, a series of nested mathematical optimization problems.
We use the following sets of variables in our \textit{master} integer linear optimization problem:

\begin{align*}
y_i &= \begin{cases}
1 & \mbox{if species $i\in$ Core set,}\\
0 & \mbox{otherwise}
\end{cases},\quad\\
q_i &= \begin{cases}
1 & \mbox{if species $i$ is a non-core present species}\\
& \mbox{in the MAS,} \\
0 & \mbox{otherwise}
 \end{cases}.
\end{align*}
for all $i \in \mathcal{S}$.

$\mathbf{v} = [v_1, \ldots, v_{|\mathcal{R}|}] \in [0,1]^{|\mathcal{R}|}_+$: flow inducing a MAS.

$z_j = \begin{cases}
1 & \mbox{if $v_j>0$},\\
0 & \mbox{otherwise}
 \end{cases}$,
for all $j \in \mathcal{R}$.

The {\sc Master} optimization problem to construct MASs is the following:
\begin{align}
\min & \dsum_{j\in \mathcal{R}} z_j\label{c:0}\\
& v_j \leq z_j, \forall j \in \mathcal{R}, \label{c:1}\\
& \dsum_{j \in \mathcal{R}} z_j \geq 2, \label{c:11}\\
& y_i + q_i \leq 1, \forall i \in \mathcal{S}, \label{c:2}\\
& y_i + q_i\geq z_j, \forall j \in \mathcal{R}, i \in \mathcal{S} \text{ with } a_{ij}\neq 0,\label{c:3}\\
& \dsum_{i \in \mathcal{S}:\atop a_{ij}>0} y_i \geq z_j, \forall j \in \mathcal{R}, \label{c:5}\\
& \dsum_{i \in \mathcal{S}:\atop a_{ij}<0} y_i \geq z_j, \forall j \in \mathcal{R}, \label{c:6}\\
 & \dsum_{j \in \mathcal{R}:\atop |a_{ij}|>0} z_j \geq q_i, \forall i \in \mathcal{S}, \label{c:7}\\
 & \dsum_{j \in \mathcal{R}:\atop a_{ij}>0} z_j \geq y_i, \forall i \in \mathcal{S}, \label{c:8}\\
 & \dsum_{j \in \mathcal{R}:\atop a_{ij}<0} z_j \geq y_i, \forall i \in \mathcal{S}, \label{c:9}\\
& \dsum_{j \in \mathcal{R}} a_{ij} x_j \geq \varepsilon - \Delta_i (1-y_i), \forall i \in \mathcal{S}, \label{c:12}\\
& y_i, q_i\in \{0,1\}, \forall i \in \mathcal{S}, \label{c:13}\\
& x_j \in [0,1], z_j \in \{0,1\}\forall j \in \mathcal{R}.
\end{align}

There, the objective function accounts for minimizing the number of reactions in the MAS. This objective assures, that each of the generated autocatalytic subnetworks is support-inclusion minimal. Constraints \eqref{c:1} allow the correct representation  of variable $z_j$ (if $v_j>0$, then $z_j$ is forced to take value one, if $v_j=0$, by the minimization criterion, $z_j$ will also take value $0$). Constraint \eqref{c:11} avoids solutions with a single reaction. Constraints \eqref{c:2} assure that species are either core species or present non-core species but not both (it is also possible to not being part of the MAS). In case $y_i+q_i=1$, species $i$ is involved in the MASs either as food, member, core or waste species. Otherwise, species $i$ is not part of the MAS. Constraints \eqref{c:3} assure that in case a reaction $j$ is selected to be part of the MAS (being then $z_j=1$), the reactants and product species of $j$ must be present (either as core species or present non-core species). Since $q_i$ and $y_i$ are binary variables, the sum of them being greater than one (but because of the previous constraint the equality holds) implies that either $i$ is a core species ($y_i=1$), or non-core present species ($q_i=1$). Otherwise, the constraint is redundant.  Constraints \eqref{c:5} and \eqref{c:6} ensure that in case a reaction is in the MAS, there must be a core species as reactant and other as product in the restricted stoichiometric matrix. Observe that in case $z_j=1$, the sum of the binary variables $y_i$ with $i$ being reactant/product of $j$ must be at least one, implying that at least one of the reactant/product species must be activated as core species. In case $z_j=0$, the constraint is redundant.  Similarly, Constraints \eqref{c:7} enforces that in case a species is select to be a present non-core species ($q_i=1$), there must be a reaction where it is present.
\eqref{c:12} is the exclusive autocatalysis condition. There, in case $y_i=1$ ($i$ is a core species in the motif), the right-hand side in the constraint becomes $1$, implying that $\dsum_{j \in M} a_{ij} v_j \geq \varepsilon >0$ (here $\varepsilon$ is a given positive but small enough constant to assure that the production is positive). 

Constants $\Delta_i$, for $i\in \mathcal{S}$, are assumed to be big enough constants assuring that in case $i$ is a non-core (present or not) species ($y_i=0$), the \textit{production} of species $i$ with flow vector $x$ must always greater that $\varepsilon - \Delta_i$ (being the constraint redundant). Specifically, for those non-core species whose production is negative (as food species), the absolute value production is upper bounded by $\Delta_i$. We take $\Delta_i = -\sum_{j\in M:\atop a_{ij}<0} a_{ij}$, for all $i\in \mathcal{S}$.

The above model belongs to the class of discrete optimization problems, which are known to be computationally challenging~\cite{papadimitriou1981complexity}, but at the same time tremendously useful to determine solutions of complex decision problems in many different fields~\cite{taha2014integer}. Although the problem of detecting MAS is known to be a NP-complete \cite{andersen2012maximizing}, there are several available softwares with implemented routines to obtain solutions of integer programming problems (as Gurobi, CPLEX, ...) in reasonable CPU time for medium size instances. 

In order to speed up the solution approach, the above master problem can be strengthened by adding valid inequalities that reduce the search of solutions. For instance, since the number of reactions in an MAS coincides with the number of core species taking part of it, one can add this (reduntant) constraint to the model:
$$
\dsum_{i \in \mathcal{S}} y_i = \dsum_{j \in \mathcal{R}} z_j
$$
Furthermore, in reverse CRN, denoting by 
\[\mathrm{Rev}=\{(j,j') \in \mathcal{R}\times \mathcal{R}: \mbox{reaction $j$ is reverse of reaction $j'$}\},\] 
one must also impose the following constraints that avoid using a reaction and its reverse in the same subnetwork:
$$
z_j + z_{j'} \leq 1, \forall (j, j')\in \mathrm{Rev}.
$$
That is, if $j$ and $j'$ are reverse reaction of each other, the $z$-variables of those reactions cannot be both one.

The solution of the problem above is a MAS with minimum number of reactions. Once the problem is solved, with optimal values in the variables $(\bar y, \bar q, \bar z, \bar x)$, the set of core species is obtained as:
\[{\rm Core} =\{i\in \mathcal{S}: \bar y_i=1\},\]
and the set of reactions involved in the MAS is:
$$
A_R= \{j \in \mathcal{R}: \bar z_j=1\}.
$$
Among the set of present non-core species, ${\rm Q}=\{i \in N: \bar q_i=1\}$, one can classify them based on the sign of the coefficients of the restricted stoichiometric matrix:
\begin{align*}
{\rm Food} &=\{i\in {\rm Q}: a_{ij}\leq 0 \forall j \in \mathcal{R}\}, \\
{\rm Waste} &=\{i\in {\rm Q}: a_{ij}\geq 0 \forall j \in \mathcal{R}\},,
\\
{\rm Member} &=({\rm Q} \cup {\rm Core})\backslash ({\rm Food} \cup {\rm Waste})
\end{align*}

The algorithm to generate th whole set of MASs is described in Algorithm \ref{alg:ac}. There, at each iteration, $it$, the above Master problem is solved. Once a solution is obtained, say $(y^{it},   q^{it}, z^{it}, x^{it})$, the tuple of reactions taking pat of the MAS is considered to restrict the next MASs to use the same reactions tuples (assuring then minimality of the generated MASs). This restriction is incorporated to the model by means of Constraint \eqref{incomp}. The process is repeated until no more autocatalytic subnetwork can be found respecting the pool of incompatibilities that have been added during the procedure. With this algorithm the MASs are generated sorted by number of reactions involved in the subnetwork since each time the problem is solved the number of reactions is minimized.

 \begin{algorithm}[H]
  \SetAlgoLined
 \KwData{$\mathcal{S}$, $\mathcal{R}$, $\mathbb{S}$, $it=0$}

 \While{{\sc Feasible}}{
  Solve {\sc Master}: $\textsc{Mas}^{it} := (y^{it},   q^{it}, z^{it}, x^{it})$.
  
  \If{{\sc Master} is feasible}{
   Save the solution and add to {\sc Master} the constraint:
   \begin{equation}\label{incomp}
\dsum_{j \in \mathcal{R}:\atop z_j=1} z^{it}_j \leq |\{j\in M: z_j^{it} =1\}|-1
\end{equation}
$it +1 \leftarrow it$.}
 }
  \KwResult{Set of MASs of a CRN: $\textsc{Mas}^{1}, \ldots, \textsc{Mas}^{it}$.}
 \caption{Optimization-based approach to enumerate MASs.\label{alg:ac}}
\end{algorithm}
 
One of the advantages of using mathematical optimization-based approaches to determine \textit{special} sets from an underlying set (as MASs from a CRN) is its flexibility to be adapted to different featrues. For instance, one can easily filter the obtained MASs to have a most a number of reactions, {\sc MaxReactions}, involved by adding the following constraint to the model:
$$
\dsum_{j\in \mathcal{R}} z_j \leq \text{\sc MaxReactions}.
$$
Analogously, one can also generate MASs where just a given set of species takes part of the core species, $\mathcal{S}_0 \subseteq \mathcal{S}$, set by adding:
$$
\dsum_{i\in \mathcal{S}_0} y_i =0.
$$

Computing the entire set of MASs is computationally chanllenguing. Each iteration in the proposed procedure requires solving a integer linear optimization problem with the constraints in the initial Master problem, but also those in the shape of \eqref{incomp} that are added each time a MASs is found. Thus, the difficulty of solving the integer optimization problem increases with the number of MASs for a given CRN. In Figure \eqref{times} we show the performance of the cumulative CPU time required to generate the MASs for 1-constituent CCRNS with $L=4, 5,$ and $6$. As can be observed from the plots, the consumed CPU time increases linearly in the number of MASs, with a slope strictly greater than one.

\begin{figure}[h]
\includegraphics[scale=0.5]{figures/timeL4.png}\\\includegraphics[scale=0.5]{figures/timeL5.png}\\\includegraphics[scale=0.5]{figures/timeL6.png}
\caption{CPU time vs. \# MASs for 1-constituent CCRN.\label{times}}
\end{figure}

% One can also filter the MASs in the generation process by the number of species taking part (as reactant or products) of a reaction in a cycle can be also upper bounded by fixing to zero those reactions with more species involved:
% $$
% z_j = 0, \forall j \in M \text{ with } |\{i \in N: a_{ij} <0 \}| > \text{\sc MaxReactants}.
% $$
% $$
% z_j = 0, \forall j \in M \text{ with } |\{i \in N: a_{ij} >0 \}| > \text{\sc MaxProducts}.
% $$

\subsection*{Checking intersection of cones}

Once the whole set of MASs have been generated, $\mathcal{M} = \{{\sc Mas}_1, \ldots, {\sc Mas}_K\}$, we hae also developed a optimization-based methodology to check the pairwise intersection of the different polyhedral flow-productive cones and partition-productive cones induced from the set $\mathcal{M}$. For two of these cones in $\mathbb{R}^d_+$, say $C_1(v_1, \ldots, v_s)$ and $C_2(w_1, \ldots, w_r)$, described by means of its generating vectors (see Section \ref{cones}). We first check if they the interior of the cones intersect or not by finding a nonnegative vector that belongs to the intersection of the closed cones maximising the sum of the coefficients of that vector with respect to each of the generators of the cones, i.e.:
\begin{alignat*}{4}
\rho^*(C_1,C_2) := & \max &&\sum_{l=1}^s \lambda_l + \sum_{l=1}^r \mu_l\\
& \mbox{s.t.} && x = \dsum_{l=1}^s \lambda_l v_l,\\
& && x = \dsum_{l=1}^r \mu_l w_l,\\
&&& x \in \mathbb{R}^d_+,\\
&&& \lambda_l \in [0,1], \forall l=1, \ldots, s,\\
&&& \mu_l \in [0,1], \forall l=1, \ldots, r.
\end{alignat*}
If the optimal solution of the above linear programming problem is $\rho^*(C_1, C_2)=0$, then, the cones do not intersect (otherwise there should be another vector with greatest coefficients in the representation). If $\rho^*(C_1,C_2)>0$, then, the cones intersect. 

In case the cones intersect, different situations are possible: the cones are identical, one of the cones is strictly contained in the other, or the cones partially intersect. To detect each of the above possibilities, we compute the distances between each of the  normalized generating vectors of one of the cones to the generating vectors in the other cones:
$$
\phi^1_l = D(v_l, \{w_1, \ldots, w_r\}) = \min_{j=1, \ldots, r} \{\|v_l-w_j\|\}
$$
for $l=1, \ldots, s,$
and
$$
\phi^2_l = D(w_l, \{v_1, \ldots, v_s\}) = \min_{j=1, \ldots, s} \{\|w_l-v_j\|\}
$$
for $l=1, \ldots, r$.

With these values, in case the cones intersect:
\begin{itemize}
\item If $\sum_{l=1}^s \phi^1_l = \sum_{l=1}^r \phi^2_l =0$, then for each of the generating vectors in both cones, there is normalized generating vector of the other cone at distance zero from it. Thus, both cones are identical.
\item If $\sum_{l=1}^s \phi^1_l = 0$, but $ \sum_{l=1}^r \phi^2_l >0$, then, $C_1 \subset C_2$ since for all the generating vectors in $C_1$, there exist a generating vector in $C_2$ at distance zero from it.
\item If $\sum_{l=1}^s \phi^1_l >0$, but $ \sum_{l=1}^r \phi^2_l =0$, then, $C_2 \subset C_1$ by the same reasoning than above.
\item If $\sum_{l=1}^s \phi^1_l >0$ and $ \sum_{l=1}^r \phi^2_l >0$, then, the cones intersect but only partially.
\end{itemize}


\section{Minimal Autocatalytic Subnetworks for CCRN with $L=4$ and $L=5$}
\label{sec:MAS_list}

Tables \ref{t:react4} and \ref{t:react5} show the set of feasible reactions for 1-constituent CCRNs with $L=4$ and $L=5$, respectively. For those CCRNs we report in Tables \ref{t:mas4}, \ref{t:mas5a}, and \ref{t:mas5b} the list of MASs obtained with our approach. Each row in those tables represent a MAS. In the first column of those tables (id) indicates the identifier of the MAS. The second column (Reactions (Flow)) provides the list of reactions taking part of the MAS. In parenthesis we provide a feasible flow for autocatalysis. In the third column (FWMC Partition), we report the FWMC partition for such a MAS. Finally, in the fourth column (Production) we show the production vector for the feasible flow in the second column, for each of the core species in the MAS.

\begin{table}[h]
\begin{tabular}{rc}
Reaction id & Reaction\\\hline
R1/R2 & \ce{$2{\bar 1}$ <=> $ {\bar 2}$}\\
R3/R4 & \ce{${\bar 1} +{\bar 2}$ <=> $ {\bar 3}$}\\
R5/R6 & \ce{${\bar 1} +{\bar 3}$ <=> $ {\bar 4}$}\\
R7/R8 & \ce{${\bar 1} +{\bar 4}$ <=> $ {\bar 2} + {\bar 3}$}\\
R9/R10 & \ce{${\bar 1} +{\bar 3}$ <=> $ 2{\bar 2}$}\\
R11/R12 & \ce{$2{\bar 2}$ <=> $ {\bar 4}$}\\
R13/R14 & \ce{${\bar 2} +{\bar 4}$ <=> $ 2{\bar 3}$}\\\hline
\end{tabular}
\caption{List of reactions for a CCRN with 1 constinuent and $L=4$.\label{t:react4}}
\end{table}

\begin{table}[h]
\begin{tabular}{rc}
Reaction id & Reaction\\\hline
R1/R2 & \ce{$2 {\bar 1}$ <=> $ {\bar 2}$}\\
R3/R4 & \ce{${\bar 1} +{\bar 2}$ <=> $ {\bar 3}$}\\
R5/R6 & \ce{${\bar 1} +{\bar 3}$ <=> $ {\bar 4}$}\\
R7/R8 & \ce{${\bar 1} +{\bar 4}$ <=> $ {\bar 5}$}\\
R9/R10 & \ce{${\bar 1} +{\bar 5}$ <=> $ 2{\bar 3}$}\\
R11/R12 & \ce{${\bar 1} +{\bar 5}$ <=> $ {\bar 2} + {\bar 4}$}\\
R13/R14 & \ce{${\bar 1} +{\bar 4}$ <=> $ {\bar 2} + {\bar 3}$}\\
R15/R16 & \ce{${\bar 1} +{\bar 3}$ <=> $ 2{\bar 2}$}\\
R17/R18 & \ce{$2 {\bar 2}$ <=> $ {\bar 4}$}\\
R19/R20 & \ce{${\bar 2} +{\bar 3}$ <=> $ {\bar 5}$}\\
R21/R22 & \ce{${\bar 2} +{\bar 5}$ <=> $ {\bar 3} + {\bar 4}$}\\
R23/R24 & \ce{${\bar 2} +{\bar 4}$ <=> $ 2{\bar 3}$}\\
R25/R26 & \ce{${\bar 3} +{\bar 5}$ <=> $ 2{\bar 4}$}\\
\end{tabular}
\caption{List of reactions for a CCRN with 1 constinuent and $L=5$.\label{t:react5}}
\end{table}

\begin{table}[h]
\begin{adjustbox}{width=0.42\textwidth}
\begin{tabular}{ccccc}
MAS id & Reactions (Flow) & FWMC Partition &  Production\\\hline
1 & $\{ {\rm R2} (1),  {\rm R9} (1)\}$   & $\{\{\bar 3\}, \{\},\{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
2 & $\{ {\rm R9} (1),  {\rm R13} (1)\}$   & $\{\{\bar 1,\bar 4\}, \{\},\{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
3 & $\{ {\rm R4} (3),  {\rm R13} (2)\}$   & $\{\{\bar 4\}, \{\bar 1\}, \{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
4 & $\{ {\rm R3} (3),  {\rm R9} (2)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
5 & $\{ {\rm R8} (3),  {\rm R13} (2)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4\}\}$  & $(1,  1)$\\
6 & $\{ {\rm R5} (3),  {\rm R13} (2)\}$   & $\{\{\bar 1,\bar 2\}, \{\},\{\{\bar 3,\bar 4\}\}$  & $(1,  1)$\\
7 & $\{ {\rm R8} (3),  {\rm R9} (2)\}$   & $\{\{\bar 3\}, \{\bar 4\}, \{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
8 & $\{ {\rm R2} (2),  {\rm R7} (3)\}$   & $\{\{\bar 4\}, \{\bar 3\}, \{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
9 & $\{ {\rm R8} (3),  {\rm R12} (2)\}$   & $\{\{\bar 3\}, \{\bar 1\}, \{\{\bar 2,\bar 4\}\}$  & $(1,  1)$\\
10 & $\{ {\rm R4} (2),  {\rm R7} (2),  {\rm R10} (1)\}$   & $\{\{\bar 4\}, \{\},\{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  2,  1)$\\
11 & $\{ {\rm R3} (2),  {\rm R6} (1),  {\rm R8} (2)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
12 & $\{ {\rm R2} (2),  {\rm R5} (3),  {\rm R12} (2)\}$   & $\{\{\bar 3\}, \{\},\{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  2,  1)$\\
13 & $\{ {\rm R3} (2),  {\rm R5} (4),  {\rm R7} (3)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
14 & $\{ {\rm R5} (4),  {\rm R7} (5),  {\rm R11} (2)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
15 & $\{ {\rm R3} (5),  {\rm R5} (4),  {\rm R12} (3)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
16 & $\{ {\rm R3} (6),  {\rm R7} (3),  {\rm R14} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
17 & $\{ {\rm R5} (6),  {\rm R7} (5),  {\rm R10} (2)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
18 & $\{ {\rm R7} (5),  {\rm R9} (4),  {\rm R11} (6)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
19 & $\{ {\rm R3} (9),  {\rm R12} (3),  {\rm R14} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
20 & $\{ {\rm R7} (9),  {\rm R11} (6),  {\rm R14} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$
 \end{tabular}
 \end{adjustbox}
\caption{List of MASs for 1-constituent CCRN $L=4$.\label{t:mas4}}
\end{table}

\begin{table}[h]
\begin{adjustbox}{width=0.4\textwidth}{
\begin{tabular}{ccccc}
MAS id & Reactions (Flow) & FWMC Partition &  Production\\\hline
1 & $\{ {\rm R2} (1),  {\rm R15} (1)\}$   & $\{\{\bar 3\}, \{\},\{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
2 & $\{ {\rm R23} (1),  {\rm R25} (1)\}$   & $\{\{\bar 2,\bar 5\}, \{\},\{\{\bar 3,\bar 4\}\}$  & $(1,  1)$\\
3 & $\{ {\rm R15} (1),  {\rm R23} (1)\}$   & $\{\{\bar 1,\bar 4\}, \{\},\{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
4 & $\{ {\rm R13} (3),  {\rm R25} (2)\}$   & $\{\{\bar 1,\bar 5\}, \{\bar 2\}, \{\{\bar 3,\bar 4\}\}$  & $(1,  1)$\\
5 & $\{ {\rm R22} (3),  {\rm R23} (2)\}$   & $\{\{\bar 4\}, \{\bar 5\}, \{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
6 & $\{ {\rm R2} (2),  {\rm R13} (3)\}$   & $\{\{\bar 4\}, \{\bar 3\}, \{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
7 & $\{ {\rm R12} (3),  {\rm R25} (2)\}$   & $\{\{\bar 2,\bar 3\}, \{\bar 1\}, \{\{\bar 4,\bar 5\}\}$  & $(1,  1)$\\
8 & $\{ {\rm R7} (3),  {\rm R25} (2)\}$   & $\{\{\bar 1,\bar 3\}, \{\},\{\{\bar 4,\bar 5\}\}$  & $(1,  1)$\\
9 & $\{ {\rm R9} (2),  {\rm R22} (3)\}$   & $\{\{\bar 1,\bar 4\}, \{\bar 2\}, \{\{\bar 3,\bar 5\}\}$  & $(1,  1)$\\
10 & $\{ {\rm R6} (3),  {\rm R25} (2)\}$   & $\{\{\bar 5\}, \{\bar 1\}, \{\{\bar 3,\bar 4\}\}$  & $(1,  1)$\\
11 & $\{ {\rm R14} (3),  {\rm R15} (2)\}$   & $\{\{\bar 3\}, \{\bar 4\}, \{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
12 & $\{ {\rm R5} (3),  {\rm R23} (2)\}$   & $\{\{\bar 1,\bar 2\}, \{\},\{\{\bar 3,\bar 4\}\}$  & $(1,  1)$\\
13 & $\{ {\rm R14} (3),  {\rm R23} (2)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4\}\}$  & $(1,  1)$\\
14 & $\{ {\rm R15} (2),  {\rm R21} (3)\}$   & $\{\{\bar 1,\bar 5\}, \{\bar 4\}, \{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
15 & $\{ {\rm R22} (3),  {\rm R25} (2)\}$   & $\{\{\bar 3\}, \{\bar 2\}, \{\{\bar 4,\bar 5\}\}$  & $(1,  1)$\\
16 & $\{ {\rm R12} (3),  {\rm R15} (2)\}$   & $\{\{\bar 3,\bar 4\}, \{\bar 5\}, \{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
17 & $\{ {\rm R14} (3),  {\rm R18} (2)\}$   & $\{\{\bar 3\}, \{\bar 1\}, \{\{\bar 2,\bar 4\}\}$  & $(1,  1)$\\
18 & $\{ {\rm R2} (2),  {\rm R11} (3)\}$   & $\{\{\bar 5\}, \{\bar 4\}, \{\{\bar 1,\bar 2\}\}$  & $(1,  1)$\\
19 & $\{ {\rm R4} (3),  {\rm R9} (2)\}$   & $\{\{\bar 5\}, \{\bar 2\}, \{\{\bar 1,\bar 3\}\}$  & $(1,  1)$\\
20 & $\{ {\rm R9} (2),  {\rm R14} (3)\}$   & $\{\{\bar 2,\bar 5\}, \{\bar 4\}, \{\{\bar 1,\bar 3\}\}$  & $(1,  1)$\\
21 & $\{ {\rm R18} (2),  {\rm R21} (3)\}$   & $\{\{\bar 5\}, \{\bar 3\}, \{\{\bar 2,\bar 4\}\}$  & $(1,  1)$\\
22 & $\{ {\rm R3} (3),  {\rm R15} (2)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
23 & $\{ {\rm R4} (3),  {\rm R23} (2)\}$   & $\{\{\bar 4\}, \{\bar 1\}, \{\{\bar 2,\bar 3\}\}$  & $(1,  1)$\\
24 & $\{ {\rm R9} (2),  {\rm R19} (3)\}$   & $\{\{\bar 1,\bar 2\}, \{\},\{\{\bar 3,\bar 5\}\}$  & $(1,  1)$\\
25 & $\{ {\rm R11} (1),  {\rm R14} (2),  {\rm R22} (2)\}$   & $\{\{\bar 3\}, \{\},\{\bar 1, \{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
26 & $\{ {\rm R13} (2),  {\rm R21} (2),  {\rm R24} (1)\}$   & $\{\{\bar 1,\bar 5\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  2,  1)$\\
27 & $\{ {\rm R11} (2),  {\rm R19} (2),  {\rm R22} (1)\}$   & $\{\{\bar 1,\bar 3\}, \{\},\{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
28 & $\{ {\rm R3} (2),  {\rm R6} (1),  {\rm R14} (2)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
29 & $\{ {\rm R12} (2),  {\rm R20} (2),  {\rm R22} (1)\}$   & $\{\{\bar 4\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$\\
30 & $\{ {\rm R4} (2),  {\rm R13} (1),  {\rm R21} (2)\}$   & $\{\{\bar 5\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
31 & $\{ {\rm R6} (2),  {\rm R11} (2),  {\rm R14} (1)\}$   & $\{\{\bar 5\}, \{\},\{\bar 2, \{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
32 & $\{ {\rm R3} (2),  {\rm R20} (1),  {\rm R22} (2)\}$   & $\{\{\bar 1,\bar 4\}, \{\},\{\{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$\\
33 & $\{ {\rm R4} (1),  {\rm R12} (2),  {\rm R13} (2)\}$   & $\{\{\bar 4\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
34 & $\{ {\rm R4} (2),  {\rm R13} (2),  {\rm R16} (1)\}$   & $\{\{\bar 4\}, \{\},\{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  2,  1)$\\
35 & $\{ {\rm R5} (2),  {\rm R8} (1),  {\rm R12} (2)\}$   & $\{\{\bar 2,\bar 3\}, \{\},\{\{\bar 1,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
36 & $\{ {\rm R5} (2),  {\rm R13} (2),  {\rm R21} (1)\}$   & $\{\{\bar 1,\bar 5\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
37 & $\{ {\rm R11} (2),  {\rm R18} (1),  {\rm R19} (3)\}$   & $\{\{\bar 1,\bar 3\}, \{\},\{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
38 & $\{ {\rm R19} (3),  {\rm R21} (2),  {\rm R23} (1)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
39 & $\{ {\rm R3} (3),  {\rm R18} (2),  {\rm R25} (2)\}$   & $\{\{\bar 1,\bar 5\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  2)$\\
40 & $\{ {\rm R7} (2),  {\rm R19} (2),  {\rm R21} (3)\}$   & $\{\{\bar 1,\bar 2\}, \{\},\{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
41 & $\{ {\rm R8} (2),  {\rm R14} (2),  {\rm R22} (3)\}$   & $\{\{\bar 3\}, \{\bar 1\}, \{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
42 & $\{ {\rm R12} (2),  {\rm R19} (2),  {\rm R21} (3)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
43 & $\{ {\rm R11} (2),  {\rm R17} (2),  {\rm R22} (3)\}$   & $\{\{\bar 1,\bar 3\}, \{\},\{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
44 & $\{ {\rm R5} (2),  {\rm R12} (3),  {\rm R21} (2)\}$   & $\{\{\bar 2\}, \{\},\{\bar 3, \{\bar 1,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
45 & $\{ {\rm R18} (2),  {\rm R19} (3),  {\rm R25} (2)\}$   & $\{\{\bar 3\}, \{\},\{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  2,  1)$\\
46 & $\{ {\rm R12} (3),  {\rm R13} (2),  {\rm R20} (2)\}$   & $\{\{\bar 4\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 5\}\}$  & $(1,  1,  1)$\\
47 & $\{ {\rm R19} (3),  {\rm R21} (3),  {\rm R26} (1)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
48 & $\{ {\rm R7} (2),  {\rm R11} (3),  {\rm R19} (2)\}$   & $\{\{\bar 1,\bar 3\}, \{\},\{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
49 & $\{ {\rm R2} (2),  {\rm R5} (3),  {\rm R18} (2)\}$   & $\{\{\bar 3\}, \{\},\{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  2,  1)$\\
50 & $\{ {\rm R16} (2),  {\rm R20} (2),  {\rm R22} (3)\}$   & $\{\{\bar 4\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$\\
51 & $\{ {\rm R4} (2),  {\rm R6} (3),  {\rm R11} (4)\}$   & $\{\{\bar 5\}, \{\bar 2\}, \{\{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
52 & $\{ {\rm R14} (4),  {\rm R20} (2),  {\rm R22} (3)\}$   & $\{\{\}, \{\bar 1\}, \{\bar 3, \{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
53 & $\{ {\rm R4} (2),  {\rm R12} (4),  {\rm R20} (3)\}$   & $\{\{\bar 4\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$\\
54 & $\{ {\rm R7} (4),  {\rm R11} (3),  {\rm R14} (2)\}$   & $\{\{\bar 3\}, \{\},\{\bar 1, \{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
55 & $\{ {\rm R11} (3),  {\rm R13} (2),  {\rm R19} (4)\}$   & $\{\{\bar 1\}, \{\},\{\bar 3, \{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
56 & $\{ {\rm R13} (2),  {\rm R19} (4),  {\rm R21} (3)\}$   & $\{\{\bar 1\}, \{\},\{\bar 2, \{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
57 & $\{ {\rm R12} (4),  {\rm R14} (2),  {\rm R21} (3)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
58 & $\{ {\rm R6} (4),  {\rm R11} (3),  {\rm R21} (2)\}$   & $\{\{\bar 5\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  1,  1)$\\
59 & $\{ {\rm R8} (4),  {\rm R19} (2),  {\rm R22} (3)\}$   & $\{\{\bar 3\}, \{\bar 1\}, \{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
60 & $\{ {\rm R7} (4),  {\rm R14} (2),  {\rm R21} (3)\}$   & $\{\{\bar 2\}, \{\},\{\bar 1, \{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
61 & $\{ {\rm R4} (4),  {\rm R6} (2),  {\rm R21} (3)\}$   & $\{\{\bar 5\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
62 & $\{ {\rm R5} (2),  {\rm R7} (4),  {\rm R21} (3)\}$   & $\{\{\bar 1,\bar 2\}, \{\},\{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
63 & $\{ {\rm R3} (4),  {\rm R11} (2),  {\rm R22} (3)\}$   & $\{\{\bar 1\}, \{\},\{\bar 4, \{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$\\
64 & $\{ {\rm R5} (2),  {\rm R14} (3),  {\rm R22} (4)\}$   & $\{\{\bar 3\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  1,  1)$\\
65 & $\{ {\rm R4} (3),  {\rm R11} (2),  {\rm R21} (4)\}$   & $\{\{\bar 5\}, \{\bar 4\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
66 & $\{ {\rm R3} (2),  {\rm R5} (4),  {\rm R13} (3)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
67 & $\{ {\rm R8} (2),  {\rm R12} (3),  {\rm R13} (4)\}$   & $\{\{\}, \{\bar 3\}, \{\bar 4, \{\bar 1,\bar 2, \bar 5\}\}$  & $(1,  1,  1)$\\
68 & $\{ {\rm R6} (2),  {\rm R19} (4),  {\rm R21} (3)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
69 & $\{ {\rm R3} (4),  {\rm R12} (2),  {\rm R14} (3)\}$   & $\{\{\bar 2\}, \{\bar 5\}, \{\{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
70 & $\{ {\rm R5} (2),  {\rm R8} (3),  {\rm R22} (4)\}$   & $\{\{\bar 3\}, \{\bar 2\}, \{\{\bar 1,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
71 & $\{ {\rm R7} (2),  {\rm R12} (3),  {\rm R20} (4)\}$   & $\{\{\bar 4\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 5\}\}$  & $(1,  1,  1)$\\
72 & $\{ {\rm R12} (4),  {\rm R13} (3),  {\rm R22} (2)\}$   & $\{\{\bar 4\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
73 & $\{ {\rm R3} (4),  {\rm R14} (5),  {\rm R26} (2)\}$   & $\{\{\bar 2\}, \{\bar 5\}, \{\{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
74 & $\{ {\rm R11} (5),  {\rm R19} (4),  {\rm R26} (2)\}$   & $\{\{\bar 1\}, \{\},\{\bar 3, \{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
75 & $\{ {\rm R8} (4),  {\rm R17} (2),  {\rm R22} (5)\}$   & $\{\{\bar 3\}, \{\bar 1\}, \{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
76 & $\{ {\rm R10} (2),  {\rm R12} (4),  {\rm R13} (5)\}$   & $\{\{\bar 4\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
77 & $\{ {\rm R10} (2),  {\rm R12} (4),  {\rm R20} (5)\}$   & $\{\{\bar 4\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$
\end{tabular}}
\end{adjustbox}
\caption{List of MASs for 1-constituent CCRN $L=5$.\label{t:mas5a}}
\end{table}

\begin{table}
\begin{adjustbox}{width=0.4\textwidth}{
\begin{tabular}{ccccc}
MAS id & Reactions (Flow) & FWMC Partition &  Production\\\hline
78 & $\{ {\rm R1} (2),  {\rm R12} (5),  {\rm R20} (4)\}$   & $\{\{\bar 4\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 5\}\}$  & $(1,  1,  1)$\\
79 & $\{ {\rm R7} (4),  {\rm R10} (2),  {\rm R21} (5)\}$   & $\{\{\bar 2\}, \{\},\{\bar 1, \{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
80 & $\{ {\rm R1} (2),  {\rm R14} (5),  {\rm R22} (4)\}$   & $\{\{\bar 3\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  1,  1)$\\
81 & $\{ {\rm R5} (4),  {\rm R13} (5),  {\rm R17} (2)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
82 & $\{ {\rm R6} (4),  {\rm R11} (5),  {\rm R16} (2)\}$   & $\{\{\bar 5\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  1,  1)$\\
83 & $\{ {\rm R6} (5),  {\rm R11} (4),  {\rm R24} (2)\}$   & $\{\{\bar 5\}, \{\bar 2\}, \{\{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
84 & $\{ {\rm R10} (2),  {\rm R12} (4),  {\rm R21} (5)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
85 & $\{ {\rm R2} (3),  {\rm R9} (5),  {\rm R24} (4)\}$   & $\{\{\bar 5\}, \{\bar 4\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  2)$\\
86 & $\{ {\rm R5} (5),  {\rm R9} (3),  {\rm R12} (4)\}$   & $\{\{\bar 2\}, \{\},\{\bar 1, \{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
87 & $\{ {\rm R2} (3),  {\rm R5} (5),  {\rm R22} (4)\}$   & $\{\{\bar 3\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  1,  1)$\\
88 & $\{ {\rm R2} (3),  {\rm R7} (5),  {\rm R20} (4)\}$   & $\{\{\bar 4\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 5\}\}$  & $(1,  1,  1)$\\
89 & $\{ {\rm R8} (4),  {\rm R15} (3),  {\rm R19} (5)\}$   & $\{\{\bar 3\}, \{\bar 4\}, \{\{\bar 1,\bar 2, \bar 5\}\}$  & $(1,  1,  1)$\\
90 & $\{ {\rm R11} (4),  {\rm R19} (5),  {\rm R23} (3)\}$   & $\{\{\bar 1\}, \{\},\{\bar 2, \{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
91 & $\{ {\rm R5} (5),  {\rm R7} (4),  {\rm R9} (3)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
92 & $\{ {\rm R3} (5),  {\rm R5} (4),  {\rm R18} (3)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
93 & $\{ {\rm R16} (4),  {\rm R18} (5),  {\rm R25} (3)\}$   & $\{\{\bar 5\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 4\}\}$  & $(2,  1,  1)$\\
94 & $\{ {\rm R8} (4),  {\rm R19} (5),  {\rm R23} (3)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
95 & $\{ {\rm R8} (4),  {\rm R18} (3),  {\rm R19} (5)\}$   & $\{\{\bar 3\}, \{\bar 1\}, \{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
96 & $\{ {\rm R6} (6),  {\rm R11} (5),  {\rm R17} (2)\}$   & $\{\{\bar 5\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  1,  1)$\\
97 & $\{ {\rm R7} (6),  {\rm R11} (5),  {\rm R17} (2)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
98 & $\{ {\rm R13} (3),  {\rm R16} (4),  {\rm R22} (6)\}$   & $\{\{\bar 4\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
99 & $\{ {\rm R4} (6),  {\rm R21} (5),  {\rm R26} (2)\}$   & $\{\{\}, \{\bar 1\}, \{\bar 5, \{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
100 & $\{ {\rm R1} (2),  {\rm R4} (5),  {\rm R21} (6)\}$   & $\{\{\bar 5\}, \{\bar 4\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
101 & $\{ {\rm R3} (6),  {\rm R13} (3),  {\rm R24} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
102 & $\{ {\rm R5} (6),  {\rm R13} (5),  {\rm R16} (2)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
103 & $\{ {\rm R12} (6),  {\rm R21} (5),  {\rm R24} (2)\}$   & $\{\{\}, \{\bar 1\}, \{\bar 2, \{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
104 & $\{ {\rm R6} (5),  {\rm R10} (2),  {\rm R11} (6)\}$   & $\{\{\}, \{\bar 2\}, \{\bar 5, \{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
105 & $\{ {\rm R12} (6),  {\rm R20} (5),  {\rm R24} (2)\}$   & $\{\{\}, \{\bar 1\}, \{\bar 4, \{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$\\
106 & $\{ {\rm R5} (6),  {\rm R21} (3),  {\rm R26} (4)\}$   & $\{\{\bar 1,\bar 2\}, \{\},\{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
107 & $\{ {\rm R12} (6),  {\rm R13} (5),  {\rm R24} (2)\}$   & $\{\{\}, \{\bar 5\}, \{\bar 4, \{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
108 & $\{ {\rm R4} (3),  {\rm R11} (6),  {\rm R16} (4)\}$   & $\{\{\bar 5\}, \{\bar 4\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
109 & $\{ {\rm R6} (6),  {\rm R21} (3),  {\rm R24} (4)\}$   & $\{\{\bar 5\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
110 & $\{ {\rm R7} (6),  {\rm R21} (5),  {\rm R24} (2)\}$   & $\{\{\bar 1\}, \{\},\{\bar 2, \{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
111 & $\{ {\rm R14} (6),  {\rm R21} (3),  {\rm R26} (4)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
112 & $\{ {\rm R11} (5),  {\rm R16} (6),  {\rm R18} (4)\}$   & $\{\{\bar 5\}, \{\bar 3\}, \{\{\bar 1,\bar 2, \bar 4\}\}$  & $(1,  1,  1)$\\
113 & $\{ {\rm R10} (6),  {\rm R21} (5),  {\rm R23} (4)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
114 & $\{ {\rm R6} (5),  {\rm R9} (4),  {\rm R24} (6)\}$   & $\{\{\bar 5\}, \{\bar 2\}, \{\{\bar 1,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
115 & $\{ {\rm R13} (5),  {\rm R15} (4),  {\rm R17} (6)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
116 & $\{ {\rm R10} (6),  {\rm R13} (5),  {\rm R23} (4)\}$   & $\{\{\bar 4\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
117 & $\{ {\rm R10} (6),  {\rm R20} (5),  {\rm R23} (4)\}$   & $\{\{\bar 4\}, \{\bar 1\}, \{\{\bar 2,\bar 3, \bar 5\}\}$  & $(1,  1,  1)$\\
118 & $\{ {\rm R5} (9),  {\rm R9} (3),  {\rm R26} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
119 & $\{ {\rm R3} (9),  {\rm R18} (3),  {\rm R24} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
120 & $\{ {\rm R10} (6),  {\rm R13} (9),  {\rm R16} (4)\}$   & $\{\{\bar 4\}, \{\bar 5\}, \{\{\bar 1,\bar 2, \bar 3\}\}$  & $(1,  1,  1)$\\
121 & $\{ {\rm R10} (6),  {\rm R21} (9),  {\rm R26} (4)\}$   & $\{\{\bar 2\}, \{\bar 1\}, \{\{\bar 3,\bar 4, \bar 5\}\}$  & $(1,  1,  1)$\\
122 & $\{ {\rm R13} (9),  {\rm R17} (6),  {\rm R24} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4\}\}$  & $(1,  1,  1)$\\
123 & $\{ {\rm R2} (1),  {\rm R9} (1),  {\rm R18} (1),  {\rm R25} (1)\}$   & $\{\{\bar 5\}, \{\},\{\{\bar 1,\bar 2, \bar 3, \bar 4\}\}$  & $(1,  1,  1,  1)$\\
124 & $\{ {\rm R3} (3),  {\rm R8} (3),  {\rm R12} (2),  {\rm R19} (2)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
125 & $\{ {\rm R9} (2),  {\rm R16} (3),  {\rm R18} (2),  {\rm R24} (3)\}$   & $\{\{\bar 5\}, \{\},\{\{\bar 1,\bar 2, \bar 3, \bar 4\}\}$  & $(1,  1,  1,  1)$\\
126 & $\{ {\rm R4} (3),  {\rm R7} (3),  {\rm R16} (2),  {\rm R20} (2)\}$   & $\{\{\bar 4\}, \{\},\{\{\bar 1,\bar 2, \bar 3, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
127 & $\{ {\rm R3} (5),  {\rm R8} (3),  {\rm R10} (2),  {\rm R12} (2)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
128 & $\{ {\rm R3} (3),  {\rm R6} (2),  {\rm R8} (3),  {\rm R19} (4)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
129 & $\{ {\rm R7} (3),  {\rm R10} (3),  {\rm R16} (2),  {\rm R20} (5)\}$   & $\{\{\bar 4\}, \{\},\{\{\bar 1,\bar 2, \bar 3, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
130 & $\{ {\rm R3} (4),  {\rm R6} (2),  {\rm R10} (4),  {\rm R21} (3)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
131 & $\{ {\rm R3} (5),  {\rm R8} (3),  {\rm R10} (4),  {\rm R23} (2)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
132 & $\{ {\rm R3} (3),  {\rm R5} (2),  {\rm R7} (5),  {\rm R11} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
133 & $\{ {\rm R3} (3),  {\rm R8} (5),  {\rm R19} (4),  {\rm R26} (2)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
134 & $\{ {\rm R3} (5),  {\rm R8} (3),  {\rm R12} (4),  {\rm R24} (2)\}$   & $\{\{\}, \{\},\{\bar 2, \{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
135 & $\{ {\rm R3} (5),  {\rm R7} (5),  {\rm R11} (4),  {\rm R24} (2)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
136 & $\{ {\rm R3} (7),  {\rm R6} (2),  {\rm R8} (3),  {\rm R10} (4)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
137 & $\{ {\rm R3} (3),  {\rm R5} (6),  {\rm R7} (5),  {\rm R20} (4)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
138 & $\{ {\rm R3} (5),  {\rm R7} (5),  {\rm R10} (2),  {\rm R11} (6)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
139 & $\{ {\rm R3} (7),  {\rm R8} (5),  {\rm R10} (4),  {\rm R26} (2)\}$   & $\{\{\bar 2\}, \{\},\{\{\bar 1,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
140 & $\{ {\rm R3} (3),  {\rm R5} (7),  {\rm R11} (4),  {\rm R26} (5)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
141 & $\{ {\rm R5} (2),  {\rm R7} (8),  {\rm R11} (7),  {\rm R16} (3)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
142 & $\{ {\rm R3} (5),  {\rm R7} (5),  {\rm R9} (4),  {\rm R24} (6)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
143 & $\{ {\rm R7} (5),  {\rm R9} (4),  {\rm R15} (7),  {\rm R17} (6)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
144 & $\{ {\rm R3} (3),  {\rm R5} (11),  {\rm R20} (4),  {\rm R26} (5)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
145 & $\{ {\rm R5} (6),  {\rm R7} (8),  {\rm R17} (3),  {\rm R20} (7)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
146 & $\{ {\rm R3} (9),  {\rm R7} (5),  {\rm R20} (4),  {\rm R24} (6)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
147 & $\{ {\rm R5} (7),  {\rm R11} (7),  {\rm R17} (3),  {\rm R26} (8)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
148 & $\{ {\rm R3} (10),  {\rm R11} (4),  {\rm R24} (7),  {\rm R26} (5)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
149 & $\{ {\rm R7} (10),  {\rm R11} (9),  {\rm R16} (5),  {\rm R24} (2)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
150 & $\{ {\rm R7} (9),  {\rm R9} (8),  {\rm R17} (3),  {\rm R24} (7)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  2,  1,  1)$\\
151 & $\{ {\rm R5} (9),  {\rm R7} (8),  {\rm R16} (3),  {\rm R20} (7)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
152 & $\{ {\rm R7} (10),  {\rm R10} (2),  {\rm R11} (11),  {\rm R16} (5)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
153 & $\{ {\rm R5} (10),  {\rm R11} (7),  {\rm R16} (3),  {\rm R26} (8)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
154 & $\{ {\rm R3} (10),  {\rm R10} (7),  {\rm R11} (6),  {\rm R13} (5)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
155 & $\{ {\rm R3} (10),  {\rm R9} (4),  {\rm R24} (11),  {\rm R26} (5)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
156 & $\{ {\rm R7} (8),  {\rm R15} (6),  {\rm R17} (9),  {\rm R20} (7)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
157 & $\{ {\rm R9} (4),  {\rm R15} (12),  {\rm R17} (11),  {\rm R26} (5)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(2,  1,  1,  1)$\\
158 & $\{ {\rm R11} (7),  {\rm R15} (7),  {\rm R17} (10),  {\rm R26} (8)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
159 & $\{ {\rm R5} (14),  {\rm R17} (3),  {\rm R20} (7),  {\rm R26} (8)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
160 & $\{ {\rm R3} (15),  {\rm R10} (7),  {\rm R11} (6),  {\rm R18} (5)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
161 & $\{ {\rm R3} (10),  {\rm R10} (7),  {\rm R11} (11),  {\rm R26} (5)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
162 & $\{ {\rm R3} (14),  {\rm R20} (4),  {\rm R24} (11),  {\rm R26} (5)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
163 & $\{ {\rm R7} (10),  {\rm R9} (9),  {\rm R16} (5),  {\rm R24} (11)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
164 & $\{ {\rm R5} (17),  {\rm R16} (3),  {\rm R20} (7),  {\rm R26} (8)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
165 & $\{ {\rm R10} (7),  {\rm R11} (6),  {\rm R13} (15),  {\rm R17} (10)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
166 & $\{ {\rm R7} (14),  {\rm R17} (9),  {\rm R20} (13),  {\rm R24} (6)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
167 & $\{ {\rm R11} (14),  {\rm R17} (10),  {\rm R24} (7),  {\rm R26} (15)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
168 & $\{ {\rm R15} (14),  {\rm R17} (17),  {\rm R20} (7),  {\rm R26} (8)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
169 & $\{ {\rm R10} (7),  {\rm R11} (21),  {\rm R17} (10),  {\rm R26} (15)\}$   & $\{\{\}, \{\},\{\bar 1, \{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$\\
170 & $\{ {\rm R9} (14),  {\rm R17} (10),  {\rm R24} (21),  {\rm R26} (15)\}$   & $\{\{\bar 1\}, \{\},\{\{\bar 2,\bar 3, \bar 4, \bar 5\}\}$  & $(1,  1,  1,  1)$
 \end{tabular}
} 
\end{adjustbox}
\caption{List of MASs for 1-constituent CCRN $L=5$ (cntd.).\label{t:mas5b}}
\end{table}

\bibliographystyle{unsrt} 
\bibliography{bibliography}
\end{document}



\begin{table}[h]
\begin{tabular}{c|cccc}
   & \multicolumn{4}{c}{$\#$ constituents}\\
   Type & 1 & 2 & 3 & 4  \\
   \hline 
    I & 100 & 100 & 100 & 100\\
    II & 100 & 100 & 100 & 100\\
    III & & & &\\
    IV & & & &\\
    V & & & &\\
\end{tabular}
\caption{\label{table:Type_class} The number of smallest minimal autocatalytic subnetworks for cluster species of maximum length $5$ classified by Blokhuis types.}
\end{table}


\begin{table}[h]
\begin{tabular}{c|cccc}
   & \multicolumn{4}{c}{$\#$ constituents}\\
   $\#$ reactions & 1 & 2 & 3 & 4  \\
   \hline 
    2 & 9 & 30 &  & \\
    3 & 11 & 51& &\\
    4 & 0 & 1 & &\\
    5 & 0 & 0 & &\\
   \hline
   $\#$ partitions &  &  &  &   \\
\end{tabular}
\caption{\label{table:2-4cluster} The number of MASs for cluster species of maximum length $5$.}
\end{table}


Thus, 
    
    Since the $\ell_3$ and $m_1$ are uniquely fixed by $i$ and $j$, 
    $\{\overline{i} + \overline{k_1} \to \overline{j} + \overline{k_2}, \overline{j} + \overline{k_3} \to 2\overline{i}\}$ or $\{\overline{i} + \overline{\ell_1} \to 2\overline{j},  \overline{j} + \overline{\ell_2} \to 2\overline{i}\}$, for some species $\overline{k_1}, \overline{k_2}, \overline{k_3}, \overline{\ell_1}$, and  $\overline{\ell_2}$ different from $\overline{i}$ and $\overline{j}$.
    In the first set of reactions, $k_1$ and $k_2$ are fixed by the conservation law to be $k_1 = 2j - i$ and $k_2=2i - j$. The set of equations is only feasible if $i=2j$ (because $i>j$), being then $k_1=0$ and $k_2=2i-j=3\, j$.    
    By the conservation laws, species $\overline{k_3}, \overline{\ell_1}$ and $\overline{\ell_2}$ are uniquely determined by $\overline{i}$ and $\overline{j}$ ($\overline{k_3}=\overline{\ell_2} = \overline{2i - j}$ and $\overline{\ell_1}=\overline{2j - i}$). {\color{blue}The first set of reactions, once $\overline{i}$ and $\overline{j}$ are fixed, and either $\overline{k_1}$ or $\overline{k_2}$, the rest of the species and reactions involved in the MASs are  uniquely determined by the conservation law. On the other hand, the second set of reactions is only feasible if $i=2j$ (because $i>j$), being then $\ell_1=0$ and $\ell_2=2i-j=3\, j$, being all the species involved in the partition uniquely determined by $i$ and $j$. Thus, there is no other MASs with the same partition as $A$.}