\section{DNA}
\paragraph
\indent Deoxyribonucleic acid (DNA) stores the genetic information
that determines all activities of every living organism. The
genetic information stored in DNA is passed from one generation to
the next. DNA is made up of four \textit{nucleotides:} guanine,
adenine, thymine and cytosine, often referred as G,A,T,C.
Nucleotides are organic structures made up of three subunits:
phosphate, deoxyribose sugar and a nitrogenous base. The four
nucleotides G,A,T and C have the same phosphate and sugar group
but differ in their nitrogenous bases (fig 2.1).
\subsection{Structure}
\paragraph
\indent Nucleotides are linked to each  other by the phosphate
group of one nucleotide with the deoxyribose sugar of another
nucleotide forming a strand. The hydroxyl groups on the 5'(5th
carbon)- and 3'(3rd carbon) of deoxyribose sugar link to the
phosphate groups to form the DNA backbone. DNA is a double
stranded molecule with the two strands in \textit{anti-parallel}
directions. The 5' end of one strand corresponds to the 3' end of
the complementary strand and vice versa \cite{Krane2003}. The
strands are hydrogen bonded together by the base pairs A-T and
G-C, the nucleotide A in one strand is hydrogen bonded with
nucleotide T in the other strand and similarly nucleotide C in one
strand is hydrogen bonded with nucleotide G on the other. For
example, if one strand is 5'-ACTG-3' then the other strand is
3'-TGAC-5'. These strands twist together to form a double helical
structure.
\begin{figure}
\begin{center}
\includegraphics[scale=0.50]{DNA.eps}
\caption{DNA; A, T, G, C - nucleotides; Anti-parallel strands (5'
to 3' and 3' to 5') bonded by base pairs A-T and G-C;}
\label{fig:2.1}
\end{center}
\end{figure}
\subsection{ Genetic code}
\label{sect:gencode}
\paragraph
\indent A strand of DNA is composed of coding and non-coding
regions. A coding region refers to part of a DNA strand that
contains the genetic information necessary for producing the amino
acid chains of proteins responsible for performing many cellular
functions. The non-coding regions on the other
hand do not participate in amino acid chain formation. The process
of generating proteins from DNA is know as \textit{gene
expression} and this process involves two stages,
\textit{transcription} and \textit{translation} (fig 2.2).
\begin{figure}
\begin{center}
\includegraphics[scale=0.50]{Trans.eps}
\caption{Transcription and translation} \label{fig:2.2}
\end{center}
\end{figure}
\paragraph
\indent In \textit{transcription}, the nucleotide sequence that
ultimately encodes a protein is used as a template to code for RNA (ribonucleic
acid), known as mRNA (\textit{messenger RNA}). RNA molecules are
similar to DNA except the deoxyribose sugar in DNA is replaced by
ribose in RNA, the base Thymine (T) in DNA is replaced by the base
Uracil (U) in RNA and RNA is a single stranded structure. The
process by which mRNA codes for protein is translation. The
translation process is performed by a special type of RNA called
tRNA(\textit{transfer RNA}) that deciphers triplet nucleotide code of
mRNA to specific amino acids. These triplet nucleotides are
referred as \textit{codons}. The relationship between codon and
amino acid is referred as \textit{Genetic code} (fig 2.3). Since
DNA is composed of four nucleotides, there are {$4^{3}$=64}
possible codons. The beginning of translation is signaled by a
special codon called \textit{start codon}. There are three codons
that do not encode any amino acid but instead signal the end of
translation, they are called \textit{Stop Codons}. Since there
are only twenty amino acids that make up proteins, more than one
codon may refer to a particular amino acid.
\begin{figure}
\begin{center}
\includegraphics[scale=0.45]{geneticcode.eps}
\caption{Genetic code - Triplet codon (example: UUU); 3 letter
representation of amino acid (example: Phe) and corresponding 1
letter representation (example: F)} \label{fig:2.3}
\end{center}
\end{figure}
\section{Proteins}
\paragraph
\indent Proteins are long chain molecules built from twenty amino
acids encoded by \textit{Codons}. The twenty amino acids are
Alanine(A), Arginine(R), Asparagine(N), Aspartic acid(D),
Cysteine(C), Glutamic acid(E), Glutamine(Q), Glycine(G),
Histidine(H), Isoleucine(I), Leucine(L), Lysine(K),
Methionine(M),Phenylalanine(F), Proline(P), Serine(S),
Threonine(T), Tryptophan(W), Tyrosine(Y) and Valine(V). \paragraph
\indent Amino acids are small molecules containing an amino group,
carboxyl group, hydrogen atom and a side chain (or R group)
attached to the central carbon (or alpha carbon)
\cite{Zimmermann2003}. Amino acids differ only in the side chain
R. The amino acids are linked to one another by the central carbon
of one amino acid with the amino group of the other amino acid
forming a peptide bond. The amino acids in the long chains are
referred as \textit{residues}.
\begin{figure}
\begin{center}
\includegraphics{aminoacid.eps}
\caption{The basic structure of an amino acid} \label{fig:2.4}
\end{center}
\end{figure}
\subsection{Amino acid classification}
\paragraph
\indent The similarity of amino acids is classified based on the
chemical properties of the side chain ( R group). Two major
classifications are Hydrophobic (non-polar) - not soluble in
water: A, I, L, M, F, P, W and V and Hydrophilic (polar) - soluble
in water: R, H, K, D, E, N, C, Q, G, S, T and Y. The polar class
is further divided into: positively charged: R, H and K,
negatively charged: D and E and uncharged: N, C, Q, G, S, T and Y.
Amino acids are also grouped together based on their
physio-chemical properties by which they could be substituted for
one another in protein sequences with minimal apparent affect on
the functionality of the proteins, known as \textit{Conservative
substitutions} \cite{Basu1997}. The following are some of the
groupings: ILVM, RK, DE, ST, AG and FY.
\subsection{Structure}
\paragraph
\indent The sequence of amino acids forming a protein is known as
the primary structure of the protein. In nature, protein molecules
collapse and fold into a unique structure known as \textit{native
structure}. There are some patterns in the native structure that
are quite common and found in many proteins, the location and
direction of these patterns are called \textit{secondary
structures}. The three main secondary structures are
$\alpha$-helix, $\beta$-sheet and random coil.  The $\alpha$-helix
is formed by the hydrogen bonds between the carbonyl group of the
$ith$ residue and the nitrogen group of the $(i+4)th$ residue (fig
2.5). An $\alpha$-helix on average has 10 residues having 3.6
residues per turn \cite{Krane2003}. $\beta$-sheets are strands of
amino acid sequences forming hydrogen bonds between them. The
bonds are formed between the carbonyl oxygen of amino acids from
one strand with nitrogen groups of the other strand. The strands
could be parallel or anti-parallel to each other. A $\beta$-sheet
can consist of all parallel strands or all anti-parallel strands
or can contain both. $\beta$-sheets usually consist of 5 to 10
residues \cite{Philip2003}(fig 2.6). Random coils are sequences of
amino acids that connect  $\alpha$- helices and $\beta$-sheets and
they are not regular structures, both in shape and size.
\begin{figure}
\begin{center}
\includegraphics[width=0.4\textwidth, height=0.4\textheight]{helix.eps}
\caption{Protein secondary structure: $\alpha$- Helix}
\label{fig:2.5}
\end{center}
\end{figure}
\subsection{Motif and domain}
\paragraph
\indent A motif is a combination of a few secondary structures
\cite{Philip2003}. For example, \textit{helix - random coil -
helix} is a motif. A domain is a more complex combination of
secondary structures having a specific function by binding to
external molecules (i.e, DNA), therefore referred to as active
site. A domain could maintain its characteristic structure even if
separated from the original protein \cite{Philip2003}. A protein
can have several motifs, which can combine to form specific
domains, and one or more domains together form the protein's
tertiary structure.
\begin{figure}
\begin{center}
\includegraphics[width=0.4\textwidth, height=0.4\textheight]{para_antipara.eps}
\caption{Protein secondary structure: $\beta$- Sheets}
\label{fig:2.6}
\end{center}
\end{figure}
\subsection{Sequence alignment}
\subsubsection{Pairwise alignment}
\paragraph
\indent The number of proteins that exists in nature is very large
but these proteins could be classified  based on the sequence
pattern, structure and their functionality. Proteins that have
similarities in their sequences are believed to be derived from an
common ancestor. Therefore, determining the similarities of
sequences would help to understand the evolution of proteins.
Sequence alignment is a technique used to compare biological
sequences in order to study the similarities and differences to
find the origin of evolution between them. Sequences that share
sequence similarity would differ from one another in some sequence
positions due to
\begin{enumerate}
    \item substitution that replaces one nucleotide/amino acid with another.
    \item an insertion that adds one or more nucleotides/amino acids.
    \item deletion that deletes one or more nucleotides/amino acids.
    \item inversion that reverses the orientation of subsequences.
\end{enumerate}
\paragraph
\indent Given two sequences $a = a_{1}\ldots a_{m}$ and $b=
b_{1}\ldots b_{n}$ over the alphabet $\Sigma$. An alignment of the
sequences $a$ and $b$ is a pair of sequences $a^{'}_{1} \ldots
a^{'}_{l}$ and $b^{'}_{1} \ldots b^{'}_{l}$ of equal lengths
defined over the extended alphabet $\Sigma'$ = $\Sigma \cup $
\{-\} containing blank character '-' such that the string $a'$ is
derived from $a$ and string $b'$ is derived from $b$.
The alignment is denoted by\\
$$a{'}_{1} a^{'}_{2}\ldots a^{'}_{l}$$
$$b{'}_{1} b^{'}_{2} \ldots b^{'}_{l}$$\\
The length l of an alignment ($a^{'}$,$b^{'}$) is restricted to
$max \{m,n\}\le l \le m+n$, since column pairs  are not allowed.
In sequence alignment, the blank character '-' is referred as a
gap denoting insertion/deletion referred to as an \textit{indel}
(fig 2.7).
\begin{figure}
\begin{center}
\includegraphics[scale=0.45]{InsDel.eps}
\caption{Alignment between two sequences} \label{fig:2.7}
\end{center}
\end{figure}
\subsubsection{Alignment score}
\paragraph
\indent Scoring schemes are used to evaluate the alignment between
sequences. There are two scores that are used in alignment
evaluation:
 \begin{enumerate}
    \item substitution score
    \item insertion and deletion score.
\end{enumerate}
\subsubsection{Substitution score}
\paragraph
\indent Substitution scores are matrices developed based on
experimental data that encode the expected evolutionary change at
the amino acid level. One of the widely used substitution scores
in amino acid alignment is Point Accepted Mutation Matrix (PAM)
developed by Dayhoff in 1978 \cite{Krane2003}. The PAM matrix $M$
contains the probability of amino acid $i$ replaced by amino acid
$k$ in a certain evolutionary time period \cite{Krane2003}. For
example, 1PAM represents, 1 substitution per 100 residues
therefore, $n$PAM is $n$ accepted substitutions in 100 residues
(i.e, probability that amino acid $i$ will be replaced by amino
acid $k$ in sequences separated by $n$PAMs of evolutionary
distance). 1PAM is generally used for closely related sequences
and higher PAM matrices are used for distantly related sequences
(highly divergent). 1PAM was obtained by calculating the
substitution probabilities based on 71 groups of sequences with $>
80 \%$ sequence identity \cite{Eidhammer2004}.
The entries of 1PAM matrix ${M^1}$ is calculated as\\
$$M_{ij} = \log{\frac{\frac{m_j * F_{ij}}{\Sigma_i F_{ij}}} {f_i}}$$\\
where, the relative mutability $m_j$ is the number of times the
amino acid $i$ is substituted, $F_{ij}$ is the number of times
amino acid $i$ is substituted by amino acid $j$ and $f_i$ is the
frequency of amino acid $i$. Once $M$ is known, the matrix $M^n$
gives the probability of any amino acid mutating to any other
amino acid in $n$PAM units. The PAM matrix $M^n$ for $n
> 1$ can be obtained by matrix multiplication of $M^1$.
$$M^2 = M^1 * M^1$$
$$M^3 = M^2 * M^1$$
$$\vdots$$
$$M^n = M^{(n-1)} * M^1$$
\paragraph
\indent Another substitution matrix widely used is BLOSUM (Blocks
Substitution Matrix), developed by Henikoff and Henikoff in 1992
based on known alignments of more diverse sequences
\cite{Eidhammer2004}. The matrix is based on the ungapped
alignment (block) from the sequence alignment. Like the PAM
matrix, different BLOSUM scoring matrices are obtained for
different evolutionary distances. For example, BLOSUM80 matrix
represents sequences with approximately $80\%$ identity in
sequence alignment.
\paragraph
\indent The relationship between the two substitution matrices is
given as, BLOSUM with low percentage corresponds to PAM with large
evolutionary distances (i.e PAM250 $\rightarrow$ BLOSUM45, PAM120
$\rightarrow$ BLOSUM80). Lower numbered BLOSUM matrices are
appropriate for more distantly related sequences and  lower
numbered PAM matrices are appropriate for more closely related
sequences.
\subsubsection{Insertion/deletion score}
\paragraph
\indent Insertion and deletion scores are calculated based on the
gap opens (single insertion/deletion) and gap extensions (long
insertion/deletion)(fig 2.7). Since long insertions and deletions
are expected less than single insertion and deletion, they are
penalized less.
\begin{figure}
\begin{center}
\includegraphics[scale=0.45]{align.eps}
\caption{Three possible alignments of two sequences}
\label{fig:2.8}
\end{center}
\end{figure}
\paragraph
\indent There could be more than one possible alignment between
the sequences (fig 2.8) but, the best alignment reflects the
evolutionary relationship between homologous sequences. In order
to find the best alignment, exhaustive search of all possible
alignments are not feasible. Therefore, alignment algorithms use a
dynamic program approach to break the problem into subproblems and
using partial results to compute the final answer
\cite{Krane2003}.
\subsubsection{Multiple sequence alignment}
\paragraph
\indent Multiple sequence alignment is an extension of pairwise
alignment to align more than two sequences simultaneously.
Multiple sequences are aligned in order to provide insight
into\begin{enumerate}
    \item characteristics of protein families
    \item identify motifs in sequences with a conserved biological
    function
    \item identify motifs of new proteins that would help to
    determine biological function.
\end{enumerate}
There are several multiple sequence alignment algorithms.
Algorithms with a heuristic approach are more commonly used than
the ones that give optimal alignment because optimal alignments
are practical only for a handful of sequences. Heuristic
algorithms are rapid, require less memory space and offer good
performance when used on relatively well conserved homologous
sequences. One of the most common heuristic approaches is
Progressive alignment used by ClustalW \cite{Zimmermann2003}. The
Progressive alignment algorithm works as follows:
\begin{enumerate}
    \item determine pairwise alignment between all pairs of
    sequences and their alignment scores.
    \item construct a guide tree (phylogenetic tree - see section ~\ref{phyl:phyltree}) using the
    alignment score.
    \item Align sequences according to the guide tree by aligning
    the most closely related sequences using sequence-sequence
    alignment first, then profile-sequence alignment( between
    an alignment and a sequence) and finally, profile-profile
    alignment( between alignments).
\end{enumerate}
In ClustalW substitution matrices and gap penalties vary at
different stages of alignment depending on the divergences of the
sequences to be aligned. Gap penalties depend on the substitution
matrices, the similarity of the sequences, and the length of the
sequences in order to introduce new gaps in the coil region rather
than in secondary structure regions \cite{Zimmermann2003}. One the
drawbacks of progressive alignment is that it is unreliable when
highly divergent sequences are aligned.
\subsubsection{Protein classification}
\paragraph
\indent A set of proteins that share a common evolutionary origin
reflected by their relatedness in function, which is usually
demonstrated by similarities in sequence, or in primary,
secondary, or tertiary structure is known as Protein
Family\cite{MCW}. Similarly, \textit{superfamiliy} is collection
of protein familes that have same overall domain structure (i.e,
same domain in same order) \cite{PIR}. There are several protein
family databases such as Prosite and Pfam. Proteins are also
classified based on the secondary structure similarities. Some
databases that group proteins based on structure classifications
are SCOP (Structural Classification of Proteins) and CATH (Class,
Architecture, Topology and Homologous superfamily).
\subsection{Phylogenetic tree}
\label{phyl:phyltree}
\paragraph
\indent Evolutionary relationships between species/sequences
(taxa) are called phylogenies and they are graphically represented
by trees known as \textit{Phylogenetic trees}. A phylogenetic tree
is made up of nodes and branches. Nodes represent distinct
taxonomical units. Nodes at the tips of the branches are
\textit{terminal nodes} and the internal nodes represent an
inferred common ancestor (fig 2.9). Branch lengths indicate the
amount of divergence between different species/sequences, longer
the lines between two species/sequences, the greater the
difference between them. Branch order refers to the genealogy of
the organism. If two species/sequences are closer to the branch
then closer their relationship.
\begin{figure}
\begin{center}
\includegraphics[scale=0.45]{Phylo.eps}
\caption{Phylogenetic tree - (B,C,D,E) - internal nodes; snake,
lizard, bird, mouse, fish, frog - leaf nodes} \label{fig:2.9}
\end{center}
\end{figure}
 \subsubsection{Tree construction}
 \paragraph
 \indent One of the widely used tree constructions is the
 neighbor-joining Method based on distance matrices. The distance
 matrix consists of estimated distance between all pairs
 of taxas or operational taxon (OTU) (fig 2.10a) calculated from any method (say sequence alignment)
 used to find similarities and differences between sequences. The neighbor-joining method
 starts with a star tree having central node X of degree m ( number of neighbor of
 X). The new internal nodes are successively created and the degree of X
is reduced by 1 in each cycle. The iteration stops when the degree
of X becomes 3.
\begin{figure}[!hbp]
\begin{center}
\subfigure[Distance matrix of fig 2.11a]{\label{fig:2.10
a}\includegraphics[scale=0.40]{DM1.eps}} \subfigure[(i) Distance
Matrix for fig 2.11a based on neighbor-joining method  (ii)
Distance matrix for fig 2.11b]{\label{fig:2.10
b}\includegraphics[scale=0.40]{DM2.eps}}
\end{center}
\caption{Distance matrices}\label{fig:2.10}
\end{figure}
\begin{figure}
\begin{center}
\includegraphics[scale=0.50]{NJT.eps}
\caption{(a) A star tree  (b) a tree with nodes A and B clustered}
\label{fig:2.11}
\end{center}
\end{figure}
\paragraph
 \indent The construction of phylogenetic tree  by neighbor-joining method (\cite{Saitou1987}) using distance matrix
 is explained using the following steps.
  \begin{enumerate}
    \item  start with a star tree (fig 2.11a).
    \item  calculate the net divergence $r(i)$ for each of the OTU's from all the
 OTU's using the distance matrix from fig 2.10a, net divergence for A is given as
 \begin{center}$r(A) = 5+4+7+6+8 = 30$ \end{center}
    \item  calculate a new distance matrix (fig 2.10b(i)) using the formula
 \begin{center} $M(ij) = d(ij)- [r(i)+r(j)] /(N-2)$ \end{center}
 where N is the number of OTU's, N=6 for fig 2.11a and $d(ij)$ is the distance between i and j, $d(AB) = 5$.
    \item  Choose two OTU's (A and B from fig:2.11a) from the distance matrix (fig:2.10b(i)) that
 has the smallest distance and create a new internal node Y that connects A, B and X.
    \item  calculate new branch length for A and B from Y using
    \begin{center} $S(AY) = d(AB)/2 - [r(A)-r(B)]/2(N-2), S(BY) =
    d(AB) - S(AY)$ \end{center} also, calculate the distance
    between Y to all the other nodes. The new distance matrix (fig 2.10 b(ii)) is
    created for fig 2.11b.
    \item  repeat process from step 2 until the degree of X
    becomes 3.
 \end{enumerate}
