\paragraph
\indent  Protein sequence analysis is the key tool for
understanding the evolution of proteins, sequence classification
and for identifying conserved (they remain same across all
species) positions crucial for the function and structure of
proteins. This thesis is intended to study the protein sequence
similarity using a holistic approach differing from the
traditional sequence alignment one which is based on subsequences.
The tool used for studying the sequence as a whole is known as
\textit{Chaos Game Representation} (CGR).
\paragraph
\indent Jeffrey in 1990 \cite{Jeffrey1990} introduced the new tool
CGR to visually represent DNA sequences. This new tool stimulated
interest among researchers and CGR has since been used to explore
the primary sequence organization of DNA and proteins. Although
research on CGR has been widely explored, it has been limited to a
two dimension representation. This thesis goes a step further to
represent the CGR in three dimensions and to understand its
potential as a tool in analysing protein sequence similarity.
\paragraph
\indent Following the introduction, Chapter 2 gives the basics of
molecular biology: The basic notions of DNA and protein sequences,
the synthesis of protein from DNA and the representation of
sequence/species relatedness through phylogenetic trees. It also
looks into bioinformatic techniques such as sequence alignment and
multiple sequence alignment with CLUSTALW, needed for the
understanding of the thesis.
\paragraph
\indent Chapter 3 looks into the mathematics behind the chaos game
representation. This chapter explains what the chaos game is, how
it could reveal the underlying patterns of any
sequence. Also, this chapter gives a brief introduction into the mathematics of
generating fractals.
\paragraph
\indent In Chapter 4, the usefulness of the CGR explored in the
past is explained. The literature on the CGR is grouped into two
sections, one on DNA sequences and the other on protein sequences.
Since the emphasis of the thesis is on protein sequences, a
detailed analysis of all the previous work of the CGR of protein
has been provided.
\paragraph
\indent Chapter 5 deals with the new three dimensional approach to
the CGR and results of the analysis of protein sequences using
3D-CGR. In the beginning, the objectives of the new approach are
presented: to detect protein homology using 3D-CGR, to understand
the impact of dinucleotide bias at the amino acid level on the
3D-CGR derived protein homology and to study the sequence
relatedness to detect shuffled motifs. Following this, a
description of the geometric solid icosahedron used for playing
the chaos game, the method of representing amino acids on the
icosahedron, and the chaos game in three dimensions are explained.
Also, various distance measures used in the thesis and the spatial
subdivision method used for determining the fractal dimension are
explained in this chapter.
\paragraph
\indent Next, the experimental objectives  of the thesis are
discussed. They are: (i) the validation of the phylogenetic trees
obtained using 3D-CGR for detecting sequence relatedness, (ii)
detection of the impact of dinucleotide bias at the amino acid
level on the 3D-CGR derived protein homology, (ii) comparison of
the trees generated by the 3D-CGR and CLUSTALW for sequence
relatedness and shuffled motif detection, (iv) comparison of the
effect of using various distance measures on the phylogenetic
trees and (v) study the sequence relatedness using fractal
patterns.
\paragraph
\indent Following this, the methodologies used for performing the
experiments are presented in detail. Our experiments reveal that
the 3D-CGR can distinguish protein families and species
relatedness within the families of the sequences. The 3D-CGR can
detect shuffled motifs that cannot be detected by CLUSTALW. The
detection of shuffled motifs by 3D-CGR could be a useful tool in
studying protein evolution due to exon shuffling. Also, the
significant difference in branch length between closely related
sequences on comparison with CLUSTALW indicate that 3D-CGR could
be used for measuring the amount of divergence between sequences
within a family. The impact of dinucleotide bias at the amino acid
level was seen in the branch length between some of the closely
related sequences and in the branch order of the families.
Finally, the sequence relatedness assessed using fractal curves
and its limitation in studying protein homology is explained.
\paragraph
\indent Lastly, Chapter 6 concludes the thesis by briefly presenting
the major concepts discussed in each chapter, the novel outcome of the
new approach and few words on the future work using 3D-CGR.
