\paragraph
\indent The chaos game representation of proteins in two
dimensions discussed in the previous chapter helped to identify
motifs in the protein databases and to test secondary structure
prediction methods \cite{Fiser1994}, reveal patterns that
distinguish protein families \cite{Basu1997}, understand the
sequentiality and composition of amino acids \cite{Pleibner1997}
and better understanding of the bacterial family homology
\cite{Yu2004}. In this chapter, a new three dimensional approach
to CGR (3D-CGR) as an analysis tool of protein sequence is
proposed, with the following objectives:
\begin{itemize}
 \item   use the three dimensional approach to detect protein homology
 \item   assess the impact of dinucleotide bias at the amino acid level on 3D-CGR derived protein homology and
 \item   use the three dimensional approach to detect shuffled motifs.
\end{itemize}
\section{Three dimensional structure and amino acid mapping}
\paragraph
\indent In order to play the chaos game for protein sequences in
three dimensions an \textit{icosahedron} was chosen to be the
geometric model. An icosahedron is a geometric solid that has
twelve vertices, thirty edges and twenty faces. An icosahedron was
chosen because the  twenty amino acids of a protein can be
represented by the twenty faces of an icosahedron.
\begin{figure}[b]
\begin{center}
\subfigure[2D Representation; Amino acid K represents the face at
the
back]{\label{fig:5.1-a}\includegraphics[scale=0.40]{Ico_2D.eps}}
\subfigure[3D
Representation]{\label{fig:5.1-b}\includegraphics[scale=0.75]{Icosahedron.eps}}
\end{center}
 \caption{a) 2D and  b) 3D Representation of an icosahedron with first and second nucleotide position of codon  and its amino acid mapping respectively \label{fig:5.1} }
\end{figure}
\subsection{Mapping}
\label{map}
\paragraph
\indent The amino acids are mapped onto the faces of an
icosahedron in an order based on the dinucleotide relatedness of
codons (see Chapter \ref{chap:Bioint}, section
 ~\ref{sect:gencode}). The amino acids that differ by a single
nucleotide in the codons are mapped onto the faces of an
icosahedron that are closer in three dimensional space and the
rest of the amino acids were mapped onto the faces that are
further apart. The mapping of amino acids onto two dimensional and
three dimensional representations of an icosahedron is shown in
figure 5.1a and 5.1b.
\section{Chaos game on an icosahedron}
\paragraph \indent The chaos game was played by taking the center of the
icosahedron as the starting point. When the  first amino acid is
read from a protein sequence, a point is plotted halfway between
the center of the icosahedron and the center of the face of the
corresponding amino acid. Subsequent points are plotted halfway between
the previous point plotted and the center of the
face of the amino acid read.
\paragraph
\indent Figure 5.2a shows the chaos game of a sample sequence
'MSDEFGHR' plotted. The first point is plotted halfway between the
center of the icosahedron and the center of the face corresponding
to the amino acid 'M', the second point is plotted halfway between
the first point plotted and the center of the face corresponding
to the amino acid 'S', the third point is plotted halfway between
the second point and the center of the face corresponding to the
amino acid 'D'; similarly E, F, G, H and R are plotted.
\begin{figure}[!htp]
\begin{center}
\subfigure[Chaos game on icosahedron for protein sequence
'MSDEFGHR']{\label{fig:5.2-a}\includegraphics[width=4in,height=3in]{SDCGR.eps}}
\subfigure[Chaos game on icosahedron for protein sequence - DNA
Polymerase Human Alpha
chain]{\label{fig:5.2-b}\includegraphics[width=4in,height=3in]{DCGR.eps}}
\end{center} \caption{ Chaos game of protein sequence in three dimension} \label{fig:5.2}
\end{figure}
\paragraph
\indent The chaos game on a protein sequence produces a
\textit{cloud of points} in space. The \textit{cloud of points}
did not reveal any obvious patterns to the naked eye. This could
be due to the length of protein sequences (approx. $< 2000$) and
also due the points being in 3D space. Figure 5.2b is the output
of chaos game using an icosahedron when played on the protein
sequence of DNA Polymerase human alpha chain (DPOA$\_$HUMAN) of length 1462. Twenty unique colors have been used to represent the twenty amino acids.
\paragraph
\indent 3D-CGR was expected to reveal useful patterns in protein sequences, since CGR has been a visual representation technique to study sequence similarities.
But, due to the points being in space no patterns were visible to the naked eye. Therefore, we reduced the number of amino acids by grouping them based on
conservative substitutions. The groupings were then mapped onto the 12 vertices of the icosahedron and chaos game was played. The points generated by the
chaos game showed empty regions near certain groups of aminoacids and more points towards other groups of amino acids indicating the frequency of occurences of the
amino acid groups but, failed to reveal any visible patterns to the naked eye. Also, we tried to visually study the points for patterns by rotating the three dimensional
figure and projecting the points onto their corresponding faces but, they were not helpful in revealing any visible patterns. Therefore, we decided to quantitatively analyse
the similarities and differences between the points produced by different sequences using phylogenetic trees.
\section{Distance measure} \paragraph \indent In this section we quantitatively analyse the relationship
between protein sequences  as well as the membership of a given protein to a protein family
by calculating the distance between the clouds of points
generated by protein sequences using 3D-CGR. In order to
define a distance measure between the clouds of points produced by any
two sequences, the points produced by the sequences were enclosed
inside a cube. The cube was then subdivided into $n \times n\times
n$ small cubes and the density of points in each of the small
cubes was calculated. Let $P$ and $Q$ be any two sequences, and $s$ and
$t$ be length of the sequences. The density of points in each of the $n
\times n\times n$ cubes of the sequences $P$ and $Q$ are represented in matrices\\
\\Let $A_{n \times n \times n}$ and $B_{n \times n \times n}$ be the 3 dimensional matrices
 \begin{center} Density $ D_A$ of $\left(a_{ijk}\right)$ = (no. of points that fall into the cube $a_{ijk}) \times  1 / s$ \\
 Density  $D_B$ of $\left(b_{ijk}\right)$ = (no. of points that fall into the cube $b_{ijk}) \times  1 / t$ \end{center}
Dividing by the length of the sequence is to normalize as the
protein sequences compared are of various length. The \textit{image distance} \cite{Wang2004} between two sequences $P$ and $Q$ is defined as\\
  $$\sum_{i,j,k=1}^n  \left| D_A \left(a_{ijk}\right ) - D_B \left(b_{ijk}\right)\right |$$
Two other distances used for sequence comparison in this thesis are the \textit{Euclid and Pearson distance}. The Euclid and Pearson distance between sequences $P$ and $Q$  using density matrices $A$ and $B$ is given by
\\The Euclid distance
  $$ \sqrt{\sum_{i,j,k=1}^n \left(D_A \left(a_{ijk}\right ) -
  D_B  \left(b_{ijk}\right)\right)^{2}}$$
\\The Pearson distance
$$ \frac{\sum_{i,j,k=1}^n D_A \left(a_{ijk}\right ) \times D_B \left(b_{ijk}\right) - \frac{\sum_{i,j,k=1}^n D_A \left(a_{ijk}\right ) \times \sum_{i,j,k=1}^n D_B \left(b_{ijk}\right )}{\mathcal{N}}}{\sqrt{(\sum_{i,j,k=1}^n D_A \left(a_{ijk}\right )^2 - \frac{(\sum_{i,j,k=1}^n D_A \left(a_{ijk}\right ))^2}{\mathcal{N}}) \times (\sum_{i,j,k=1}^n D_B \left(b_{ijk}\right )^2 - \frac{(\sum_{i,j,k=1}^n D_B \left(b_{ijk}\right
))^2}{\mathcal{N}})}}$$
$$\mathcal{N} = n \times n \times n$$
\section{Experimental objectives}
\paragraph \indent To detect protein homology using the 3D-CGR approach, to assess the
impact of dinucleotide bias at amino acid sequence level on 3D-CGR
derived protein homology and to detect shuffled  motifs, the
following experiments were performed :
\begin{itemize}
    \item \textbf{validate tree:} The goal of this experiment was to test if the phylogenetic tree constructed using 3D-CGR can detect protein sequence homology.
    \item \textbf{effect of mapping change: } The goal of this experiment was to investigate whether or not varying the mapping of the 20 amino acids on to the 20 faces of the icosahedron has an effect on the quality of the phylogenetic trees.
    \item \textbf{compare trees: } The goal of this experiment was to compare the phylogenetic trees generated by 3D-CGR with an alignment technique CLUSTALW used for studying sequence relatedness.
    \item \textbf{compare distance measures: } The goal of this experiment was to compare the phylogenetic trees generated by three distance measures and decide which one gives better results in describing the protein sequence homology.
    \item \textbf{assess fractal pattern: } The goal of this experiment was to compare the fractal patterns produced by protein sequences using their fractal dimension.
\end{itemize}
\section{ Dataset for protein sequence analysis using 3D-CGR}
\paragraph \indent The test data  was obtained from the SWISS-PROT Database. Table
5.1 and 5.2 lists all the protein sequences, their length and
SWISS-PROT ID used for the test analysis. The protein sequences
were selected such that they were of various lengths, and from
protein families of diverse functionalities.
\begin{table}[!htp]
\begin{center}
\begin{tabular}{|l|l|l|}
\hline
Protein Family & $SWISS\_PROT ID$ & Length\\
\hline
Myoglobin & $MYG\_ALLMI$(Alligator)   &   154\\
&$MYG\_CHICK$   &   153\\
&$MYG\_HUMAN$   &   153\\
&$MYG\_MOUSE$   &   153\\
\hline
Hemoglobin & $HBA\_ALLMI$   &   141\\
&$HBA\_CHICK$   &   141\\
&$HBA\_HUMAN$   &   141\\
&$HBA\_MOUSE$   &   141\\
&$HBA\_XENTR$(Frog)   &   141\\
&$HBA\_BRARE$(Fish)   &   141\\
\hline
Superoxide dismutase & $SOD1\_ORYSA$(Rice)  &   151\\
&$SODC\_DROME$(Fruit Fly)  &   152\\
&$SODC\_CHICK$  &   153\\
&$SODC\_NEUCR$(Fungus)  &   153\\
&$SODC\_XENLA$  &   150\\
&$SODC\_BRARE$  &   154\\
&$SODC\_HUMAN$  &   153\\
&$SODC\_MOUSE$  &   153\\
&$SODC\_CAEEL$(Worm)  &   158\\
&$SODC\_YEAST$  &   153\\
\hline
Alcohol dehydrogenase & $ADH1\_YEAST$  &   347\\
&$ADH1\_NEUCR$  &   353\\
&$ADH1\_BACST$  &   337\\
&$ADH1\_CAEEL$  &   349\\
&$ADH1\_ORYSA$  &   376\\
&$ADHA\_HUMAN$  &   374\\
&$ADHA\_MOUSE$  &   374\\
&$ADHA\_CHICK$  &   375\\
&$ADH1\_ALLMI$  &   374\\
\hline
\end{tabular}
\caption{Protein Family, Swiss-Prot ID and Length of test protein
sequences} \label{Table: 5.1}
\end{center}
\end{table}
\begin{table}[!htp]
\begin{center}
\begin{tabular}{|l|l|l|}
\hline
Protein Family & $SWISS\_PROT ID$ & Length\\
\hline
Catalase & $CAT1\_CAEEL$  &   524\\
&$CATA\_BACSU$(Bacteria)  &   482\\
&$CATA\_DROME$  &   506\\
&$CATA\_HUMAN$  &   526\\
&$CATA\_MOUSE$  &   526\\
&$CATA\_BRARE$  &   526\\
&$CATA\_ORYSA$  &   491\\
&$CAT1\_NEUCR$  &   736\\
&$CATA\_YEAST$  &   515\\
\hline
Methionine adenosyltransferase & $METK\_BACSU$  &   400\\
&$METK\_CAEEL$  &   404\\
&$METK\_DROME$  &   408\\
&$METK\_HUMAN$  &   395\\
&$METK\_RAT$    &   395\\
&$METK\_ORYSA$  &   396\\
&$METK\_NEUCR$  &   395\\
&$METK\_YEAST$  &   381\\
\hline
6 phosphogluconate dehydrogenase & $6PGD\_BACSU$  &   468\\
&$6PGD\_CANAL$(Yeast)  &   517\\
&$6PG1\_YEAST$  &   489\\
&$6PGD\_DROME$  &   481\\
&$6PGD\_HUMAN$  &   482\\
&$6PGD\_MOUSE$  &   482\\
\hline
DNA Polymerase &$DPO1\_BACST$  &   876\\
&$DPOA\_DROME$  &   1488\\
&$DPOA\_HUMAN$  &   1462\\
&$DPOA\_MOUSE$  &   1465\\
&$DPOA\_YEAST$  &   1468\\
&$DPOA\_ORYSA$  &   1243\\
&$DPOD\_CAEEL$  &   1081\\
\hline
\end{tabular}
\caption{Protein Family, Swiss-Prot ID and Length of test protein
sequences (continued from Table 5.1)} \label{Table: 5.2}
\end{center}
\end{table}
\section{Software}
\paragraph \indent The plotting of amino acids and evaluation of distance
measures were performed using Maple 9. X Windows was used for
running Maple 9 under the Unix Operating System. Phylip 3.63
package was used for generating phylogenetic trees to determine
the sequence similarity and differences. Fractal analysis was
performed using Java. CLUSTALW for the test data was run using the
default parameter, gap open penalty of 10 and the BLOSUM matrix.
\section {Results and discussion}
\subsection{Tree validation}
\paragraph \indent This experiment was performed to test whether the three
dimensional CGR could generate a phylogenetic tree that can
identify relatedness of sequences and distinguish differences
between sequences. The method was to use a set of sequences for
which the true phylogenetic tree was known and compare the tree
obtained by using 3D-CGR with the true phylogenetic tree. As in
nature the true phylogenetic tree is never known for sure, we used
a starting sequence and simulated its evolution through mutations
such that the relatedness of the subsequent sequences was
completely transparent.In order to perform this experiment 15
simulated protein sequences of length 352 with known percentage of
relatedness between them were created. Sequence 1 was assumed to
be the root, sequences 2 and 3 were derived from sequence 1 by 16
amino acid substitutions (in the first half for sequence 2 and
second half for sequence 3), sequences 4 and 6 were derived from
sequence 2 and sequences 5 and 7 were derived from sequence 3 by
the same method, and similarly, sequences 8 and 10 from sequence
4, sequences 12 and 14 from sequence 6, sequences 9 and 11 from
sequence 5 and sequences 13 and 15 from sequence 7. At each stage
of derivation additional substitutions were made to the derived
sequences as in the first step and the substitutions were made
such that they do not replace an earlier substitution. Figure 5.3a
represents the sample sequence derivation explained above and
figure 5.3b represents the true phylogenetic tree that the
simulated sequences were expected to generate. In parallel, the
chaos game was played on the icosahedron and the image distance
was calculated for all pair of sequences from the cloud of points
generated by their CGR's. The distances were represented in a
distance matrix and the phylogenetic tree was generated.
\subsubsection{Result and interpretation}
\paragraph \indent Figure 5.3c represents the phylogenetic tree generated by
3D-CGR. The phylogenetic tree generated by the simulated protein
sequences using 3D-CGR was able to establish sequence relatedness
based on the mutational difference in the sequences. The
hierarchical structure of ancestor and children was exact to that
of the known tree. The result shows that the 3D-CGR can identify
related sequences and distinguish differences between them.
Therefore, 3D-CGR as an analysis tool can be used to study protein
sequence relatedness.
\begin{figure}[!htp]
\begin{center}
\subfigure[SEQUENCE02 and SEQUENCE03 derived from first and second
half of SEQUENCE01, SEQUENCE04 and SEQUENCE06 derived from
SEQUENCE02; substitutions are represented in lowercase letters
]{\label{Fig:5.3a}\includegraphics[scale=0.48]{TrueTree.eps}}
\subfigure[True Tree]{\label{Fig:5.3b}\includegraphics[scale
=0.70]{sptree.eps}} \subfigure[Validated Tree by
3DCGR]{\label{Fig:5.3c}\includegraphics[scale=0.40]{NSIM.eps}}
\end{center}
\caption{Phylogenetic tree validation -  a) Sequence derivation b)
Known tree c) Tree generated by 3D-CGR} \label{Fig:5.3}
\end{figure}
\subsection{Effect of mapping change}
\paragraph \indent In order to obtain meaningful results we wanted to map the amino acids onto the
faces of an icosahedron in a way that is biologically meaningful. Therefore, for our working mapping, we
maped the amino acids the differ by a single nucleotide in the codons onto the neighboring faces of an
icosahedron and rest of the amino acids were mapped onto the faces that are further apart.
We wanted to test whether varying the mapping would change the results of our analysis i.e we wanted
the impact of dinucleotide bias at the amino acid level. Consequently, the chaos game was played on the icosahedron
with the above mentioned mapping of amino acids and the image
distance was calculated between the cloud of points for all pairs
of sequences. The image distances between all pairs of sequences
was represented as a distance matrix and a phylogenetic tree was
generated based on the distance matrix. Similarly, phylogenetic
trees were generated for four other random mappings of amino acids
onto the faces of an icosahedron. Figures 5.4, 5.5 and 5.6
represents the phylogenetic trees obtained using the dinucleotide
related mapping and two random mappings.
\subsubsection{Results and interpretation}
\paragraph \indent All the trees generated distinguished the protein families
of the test sequences from one another. Also, the trees displayed
species relatedness within families. The comparison between the
three trees is provided in Table 5.3. The dinucleotide related  mapping differs
from random mapping 1 and random mapping 2 in the branching of ADH sequences,
in the branch order of the families and in branch length between closely related species. The difference in branch
length and branch order between the mapping based on dinucleotide
relatedness of codons and the two random mappings can be attributed to the minor
effect of dinucleotide biases at amino acid level.
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.75]{GECeps.eps}
\caption{ Phylogenetic  Tree generated by dinucleotide relatedness
mapping using the Image distance} \label{Fig:5.4}
\end{center}
\end{figure}
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.75]{GECM2eps.eps}
\caption{ Phylogenetic  Tree generated by random mapping 1 using
the Image distance} \label{Fig:5.5}
\end{center}
\end{figure}
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.75]{GECM3eps.eps}
\caption{ Phylogenetic  Tree generated by random mapping 2 using
the Image distance} \label{Fig:5.6}
\end{center}
\end{figure}
\begin{landscape}
\begin{table}
\begin{tabular}{|p{2.5in}|p{2.5in}|p{2.7in}|}
\hline
& Random Mapping 1 & Random Mapping 2\\
\hline
Dinucleotide Mapping&difference in branching of ADH sequences, difference in the evolution of protein families and difference in branch length between closely related sequences - $(ADHA\_HUMAN, AHDA\_MOUSE)$, $(MYG\_ALLMI, MYG\_CHICK)$& difference in branching of ADH sequences, difference in branch length between closely related sequences - $(ADHA\_HUMAN, AHDA\_MOUSE)$, $(MYG\_ALLMI, MYG\_CHICK)$ and evolution of protein families \\
\hline
Random Mapping 1 & identical & difference in branching of ADH sequences\\
\hline
\end{tabular}
\caption{Pairwise comparisons of the three mapping strategies} \label{Table 5.3}
\end{table}
\end{landscape}
\subsection{3D-CGR and CLUSTALW phylogenetic tree comparison}
\paragraph \indent In order to evaluate the three dimensional CGR method the phylogenetic tree generated by 3D-CGR was
compared with the tree generated by the well known multiple sequence alignment program
CLUSTALW. CLUSTALW and 3D-CGR follow different approaches to study sequence similarity.
In CLUSTALW, the similarity is initially assessed by pairwise alignment, the alignment of any pair of amino acid depends on the alignment score and is irrespective of the
pair of amino acid that preceeds it as well as the pair of amino acid that follows it. In contrast, in 3D-CGR a holistic approach is used,
every point in the image depends on the preceding sequence of points.
Phylogenetic trees were obtained for multiple sequence
alignment and 3D-CGR from the dataset. Figure 5.3 and 5.7 represent the
phylogenetic trees generated by 3D-CGR and CLUSTALW.
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.75]{clust_gene.eps}
\caption{Phylogenetic tree generated by CLUSTALW} \label{Fig:5.7}
\end{center}
\end{figure}
\subsubsection{Branch length and protein family evolution}
\paragraph \indent The trees when compared showed that they both were
able to distinguish protein families and identify species
relatedness within the familes. But the following differences were
also noted: $a$) In the CLUSTALW tree, when two sequences are
closely related (example: ADHA\_HUMAN
 and ADHA\_MOUSE), the branch lengths of the two sequences are
almost equal, whereas in the 3D-CGR tree significant differences
between the branch lengths can be seen; $b$) the evolution of
protein families in CLUSTALW tree indicate they had diverged long
time back whereas in 3D-CGR tree the evolution of protein families
indicate recent divergence and; $c$) the superoxide dismuatse
family branches off closely to myoglobin and hemoglobin in the
3D-CGR tree, whereas in CLUSTALW tree, superoxide dismutase does
not closely branch off from myoglobin and hemoglobin families.
\paragraph
\indent The branch lengths in 3D-CGR indicates 3D-CGR could be
used measure the amount of divergence of sequences within protein
families. However, we could not identify the biological
significance of the protein family evolution obtained by 3D-CGR.
\subsubsection{Shuffled Motif detection} \paragraph \indent The power of CGR relies on its holistic
approach to biological sequences. Every point on a 3D-CGR depends
on its previous point therefore it has a long memory of the
preceding amino acids in a protein sequence, whereas in sequence
alignment the alignment of any pair of amino acid does not depend
on the alignment of its preceding pair of amino acids. Therefore,
when two sequences are to be aligned and there are two different
motifs present at different positions in both the sequences (fig
5.8a), the sequence alignment would align based on the best
alignment and may not detect one of the motifs. In contrast, in
the case of 3D-CGR, the comparison of protein sequences is based on
the frequency of points in each cubic region, therefore, 3D-CGR is
expected to identify motifs better than sequence alignment. In
order to test this hypothesis, a protein sequence - SEQUENCE01 of
length 300  was selected  and a new sequence - SEQUENCE02 was
obtained from it by shuffling various regions in the SEQUENCE01.
Similarly, another sequence SEQUENCE03 was derived from SEQUENCE01
by performing several insertion/deletions and substitutions,
SEQUENCE07 was derived by interchanging two big
subsequences, and SEQUENCE06 was made to be identical to
SEQUENCE01. Fig 5.8b depicts the above example. Phylogenetic trees
for the test sequences together with two other unrelated sequences
SEQUENCE04 and SEQUENCE05 were obtained using 3D-CGR (figure 5.9a)
and CLUSTALW (figure 5.9b). In the tree obtained from CLUSTALW,
SEQUENCE03 was identified more closely to SEQUENCE01 than sequence
SEQUENCE02 whereas in 3D-CGR shuffled sequences, SEQUENCE02 and
SEQUENCE07 were identified more closely to SEQUENCE01 than
SEQUENCE02 with mutations.
\begin{figure}[!htp]
\begin{center}
\subfigure[Interchanged
subsequence]{\label{Fig:5.8a}\includegraphics[scale
=0.75]{Interch.eps}} \subfigure[Positions marked 1-8 in SEQUENCE01
are interchanged in SEQUENCE02, positions marked in SEQUENCE03 are
mutated from SEQUENCE01, SEQUENCE04 with two big interchanged
regions
marked]{\label{Fig:5.8b}\includegraphics[scale=0.50]{SeqJum.eps}}
\end{center}
\caption{Shuffled motif detection}\label{Fig:5.8}
\end{figure}
\begin{figure}[!htp]
\begin{center}
\subfigure[Phylogenetic tree generated by Chaos for Jumbled
Sequences]{\label{Fig:5.9a}\includegraphics[scale=0.30]{MYSIM.eps}}
\subfigure[Phylogenetic tree generated by CLUSTALW for Jumbled
Sequences]{\label{Fig:5.9b}\includegraphics[scale=0.50]{clust_MYSIM.eps}}
\end{center}
\caption{Shuffled motif detection by chaos and CLUSTALW}\label{Fig:5.9}
\end{figure}
\paragraph
\indent The above result indicates that 3D-CGR can identify shuffled motifs/domains better than CLUSTALW due to its long memory
of preceeding sequences. Therefore, 3D-CGR does not require the motifs/domains to be in the same order between sequences in contrast to CLUSTALW which requires the sequences to have same motifs/domains in the same order when assessing sequence similarity.
The ability of 3D-CGR to identify shuffled motifs/domians between any pair of sequences indicate it can be used to study protein evolution due to exon shuffling. Exon shuffling is a process by which motifs/domains have been shuffled to form new proteins.
\subsection{Distance measure comparison}
\paragraph \indent All the results obtained in section 5.7.1, 5.7.2 and 5.7.3 used the image distance.
We performed some experiments using another two different
distances in order to determine the best distance measure for the
output produced by 3D-CGR. The three distance measures used for
the comparison were the Image distance, the Euclid Distance and
the Pearson distance. The three distance matrices containing the
distances between all pairs of protein sequences were calculated
using Image, Euclid and Pearson distances and their corresponding
phylogenetic trees were generated. Figures 5.3, 5.10 and 5.11
represent the phylogenetic trees generated by the Image distance,
Euclid distance and Pearson distance measures.
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.75]{EGEC.eps}
\caption{Phylogenetic tree generated by the Euclid distance}
\label{fig:5.10}
\end{center}
\end{figure}
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.75]{PGEC.eps}
\caption{Phylogenetic tree generated by the Pearson distance}
\label{Fig:5.11}
\end{center}
\end{figure}
\subsubsection{Result and interpretation}
\paragraph \indent The trees generated by the Image distance, Euclid distance and
Pearson Distance using 3D-CGR are all able to identify the related
species and their families. The only difference was that the tree
generated by the Pearson Distance was different from the trees generated by the Image
distance and the Euclid distance in branch order and branch length.
Therefore, it is not easy to conclude that one distance is better than
the other.
\section{Fractal analysis on protein sequences}
\paragraph \indent Fractal analysis assesses the fractal patterns produced by
a cloud of points generated by a sequence. The fractal dimension (Box Counting Dimension) is usually
used in assessing the fractal patterns. Since there was no
pattern visible to the naked eye in the cloud points produced by the
3D-CGR, the relationship between sequences was assessed by
calculating box counting dimensions for the cloud of points
generated. The fractal dimensions for various box sizes for the
sequences were calculated and the implementation was done using a
Spatial Subdivision Algorithm - Octree.
\subsection{Octree}
\paragraph
\indent
An Octree is a tree data structure with eight children used to
represent spatial subdivision. Each node of the octree holds the
physical position of the boxes. Figure 5.12 represents the
subdivided cube and its Octree. Initially, a single cube is needed
to cover the point cloud, therefore, it is the root node in the
Octree. Divide the cube into eight smaller cubes and determine if
any of the cube covers a portion of the point cloud. If they are,
then subdivide those cubes into further smaller cubes. The process
continues for some $n$ times to obtain a better approximation of the
fractal dimension.
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.60]{Octree.eps}
\caption{Spatial Subdivision using Octree} \label{fig:5.12}
\end{center}
\end{figure}
\subsection{Fractal dimensions of test sequences}
\paragraph
\indent A log-log plot
of the number of boxes needed to cover the cloud of points and the box
sizes for each sequence in the data set was performed to analyze
the fractal nature of the sequences. Fig 5.13 shows the log-log
plot of the sequences in the data set.
\begin{figure}[!htp]
\begin{center}
\includegraphics[scale=0.65]{fractal.eps}
\caption{Fractal curve for the test data set} \label{Fig:5.13}
\end{center}
\end{figure}
\subsubsection{Result and interpretation} \paragraph \indent Though the fractal
curves generated were similar for protein families, the curves were
dependent on their sequence lengths. Therefore, fractal analysis is not
good method to analyze the relationship between the
sequences.
