\section{ Chaos game representation of DNA sequences}
\paragraph
\indent Chaos game representation (or CGR) is a visual
representation technique used, among others to study the patterns
in gene structures \cite{Jeffrey1990}. The chaos game is played on
a square using IFS. The nucleotide bases (A,C,G,T) correspond to
the four vertices; the first point is plotted halfway between the
center of the square and the corresponding vertex of the first
nucleotide base in the sequence, and each subsequent point is
plotted halfway between the previous point and the vertex of the
subsequent nucleotide base from the sequence (fig 4.1a). The chaos
game when applied to DNA sequences showed fractal structures (fig
4.1b , fig 4.1c). Figure 4.1b represents the attractor of chaos
game of (\textit{Human Beta Globin-HUMBB}). The 'double
scoop'(sparse regions on fig 4.1b) structure in the attractor is
due to the paucity of points in various regions of the square
\cite{Jeffrey1990}. Similarly, when A $\&$ T and C $\&$ G were
plotted opposite to one another, the paucity of points is
represented by squares instead of 'double scoop' (fig 4.1c). The
patterns are repeated on various scales in the attractor
exhibiting the property of self-similarity. These patterns
revealed the non-random nucleotide composition in DNA sequences.
In a CGR, any $ith$ point in the attractor uniquely represents the
$ith$ long initial subsequence of the sequence. The attractor
depicts the base composition of the gene sequence
\cite{Jeffrey1990}; the square, when divided into four, sixteen,
sixty four sub-quadrants and so on represents the mono, di- and
tri- etc. nucleotides subsequences (fig 4.1d).
\begin{figure}[!hbp]
\subfigure[CGR-AGAAT; Plotting "AGAAT"- 'A' is plotted half-way
between the center and the vertex representing A, 'G' is plotted
half-way between the previous point plotted and the vertex
representing G, 'A' is plotted half-way between the previous point
and the vertex representing A, next 'A' is plotted halfway between
the previous point  and the vertex representing A and similarly
'T' is
plotted]{\label{fig:4.1-a}\includegraphics[scale=0.55]{DNA_SCGR.eps}}
\subfigure[CGR-HumanBetaGlobin; Nucleotides A,C,G,T are
represented as red, violet, green and blue; paucity of
dinucleotide CG forms a 'Double Scoop'
pattern]{\label{fig:4.1-b}\includegraphics[scale=0.65]{Dscoop.eps}}
\subfigure[CGR-HumanBetaGlobin; paucity of dinucleotide CG forms a
square pattern as vertices C and G are opposite to each
other.]{\label{fig:4.1-c}\includegraphics[scale=0.65]{ScoopDiag.eps}}
\subfigure[Mono-, di-, and Tri- nucleotide
configuration]{\label{fig:4.1-d}\includegraphics[scale=0.65]{Neucgrid.eps}}
\caption {Properties of CGR} \label{fig:4.1}
\end{figure}
\paragraph
\indent Using a CGR, the presence or absence of a sequence of
nucleotides in any DNA sequence can be mathematically
characterized \cite{Dutta1992}. Dutta et.al \cite{Dutta1992} gave algorithms to find the
subsequences corresponding to any given point in CGR, and to
simulate CGR patterns of a sequence by predicting the order of
nucleotides using the probability of occurrences of di or tri
nucleotides \cite{Dutta1992}. These algorithms are presented in the
appendices A and B.
\paragraph
\indent Hill et.al in 1992 \cite{Hill1992} examined the coding region of the CGR's of seven
\textit{globin} genes from human and the CGRs of 29 closely related \textit{alcohol
dehydrogenase} genes from phylogenetically divergent species. CGRs of Human \textit{globin} coding
regions and the CGR of the entire Human Globin gene (coding and
non-coding) are visually similar to each other but the
self-similarity was not readily visible in the individual sequences
due to smaller number of points (2000 nucleotides). Also, the
di-nucleotide frequencies of the \textit{globin} genes from human
are not significantly different from one another. Therefore, the
di-nucleotide frequencies partially accounted for the
self-similarity in the CGR patterns \cite{Hill1992}. CGRs from
coding regions of \textit{alcohol dehydrogenase} gene from the
same species (ADH1, ADH2 etc) are similar to one another. Also,
ADH CGRs of closely related species such as human, rodent, primate
were similar to one another\cite{Hill1992}. CGRs of unrelated
genes from the same species are more similar to one another than
sequences from unrelated species. Therefore,  Hill et.al in 1992 \cite{Hill1992} said that
the CGR patterns reflect genome type specificity which could be
the result of mutation rates of  mono, di, tri nucleotide bases
and so on and said the evolution of a gene and its coding sequences
should not be examined in isolation, genome specific differential
mutation in di-nucleotides or oligonucleotide should be taken into
account \cite{Hill1992}. Hill et.al in 1997 \cite{Hill1997} studied 28 complete mitochondrial genomes using CGR. They said, the global DNA sequence organizationof mitochondrial genomes is species-type specific. The species-type specific patterns appear primarily due to the dinucleotide composition.
\paragraph
\indent CGRs generated from simulated sequences using a
first-order Markov-chain probability matrix for \textit{Human Beta
Globin} and second-order Markov-chain probability matrix for
\textit{Bacteriophage Lambda} were similar to CGRs of the original
Human Beta Globin and Bacteriophage Lambda sequences \cite{Goldman1993}. The first-order and second-order
Markov probability matrices were obtained from calculating the
dinucleotide and trinucleotide frequencies directly from the DNA
sequences without reference to the CGR. Therefore Goldman in 1993
\cite{Goldman1993}, suggested that CGR is a particular case of
Markov-Chain model and CGR is only limited to represent mono, di
and tri nucleotide representation of the sequences
\cite{Goldman1993}.
\paragraph
\indent The Markov chain model is limited to produce only integer
number of bases whereas the frequency matrix obtained from CGR
(FCGR) can produce non-integer number of bases \cite{Almedia2001}
in contradiction to statement by Goldman in 1993 \cite{Goldman1993}. The frequency matrix for
oligonucleotides of length $n_{c}$ is obtained by dividing the CGR
into a $2^{n_{c}} * 2^{n_{c}}$ grid. Then the Markov chain
probability matrix could be obtained from
FCGR only if the quadrant $k$ satisfies the condition in the following equation to produce an integer order\\
\begin{center} $k = 2^{2{n_{c}}} , n_{c}$ is an integer $ \ge 1$ \end{center}
But, if the condition '$n_{c}$ is an integer $\ge 1$' is removed
then
$$n_{c} = \log_{2}(k) /2,$$\\
i.e FCGR can track the frequency of oligonucleotide of non-integer
order. Therefore, CGR enables the determination of the frequency
of redundant fractionary sequences also, FCGR of non-integer order
can be used to calculate global distance and local similarities
between sequences \cite{Almedia2001}.
\paragraph
\indent If $k$=3, then the patterns in CGR are determined by
mononucleotide, dinucleotide and trinucleotide frequencies but if
$k > 3$, then longer oligonucleotides may influence the CGR
patterns. Therefore, Wang et.al \cite{Wang2004} said, a CGR of 1/$2^k$ resolution is completely
determined by all the frequencies of length $k$ when the length of
the DNA sequence is longer than $k$ . They also analyzed the relationship between dinucleotide
relative abundance profile \footnote{ratio of the dinucleotide
frequency to the frequency of two single nucleotide composing this
dinucleotide}(DRAP) and CGR. DRAP can be computed from second-order or
dinucleotide frequency FCGR, but second order FCGR cannot be
computed from DRAP. Therefore, DRAP or rFCGR (relative FCGR) is a
special case of FCGR.  Wang et.al \cite{Wang2004} said, an $n$
th order FCGR provides more info than DRAP. However, the
second-order FCGR or DRAP is a good choice of genomic signature \footnote{The whole set of short oligonucleotide frequencies observed in a DNA sequence is species-specific and is thus considered as a GENOMIC SIGNATURE} as
the computational cost is higher for higher order FCGRs
\cite{Wang2004}. A new distance measure called \textit{image
distance} used to calculate the distance between genomic
signatures of two DNA sequences \cite{Wang2004} is given by\\
\begin{center} $dI_\mathcal{R}\left(\bar{\mathcal{A}},\bar{\mathcal{B}}\right) = 1/4^k * \sum_{i=1}^{2^k}\sum_{j=1}^{2^k} \left| density_\mathcal{R}\left(\bar{\mathcal{A}}\right)_{i,j}- density_{\mathcal{R}}\left(\bar{\mathcal{B}}\right)_{i,j}\right|$\end{center}
\begin{center} $\bar{\mathcal{A}} = 4^k/\sum_i\sum_j a_{i,j} * \mathcal{A}$ and \end{center}
\begin{center} $\bar{\mathcal{B}} = 4^k/\sum_i\sum_j a_{i,j} * \mathcal{B}$ \end{center}
where $\mathcal{A}$ and $\mathcal{B}$ are frequency matrices of
$k$th order, $\mathcal{R}$ is the radius of the neighborhood
centered at $(i,j)$ and \textit{$density_\mathcal{R}$}. The phylogenetic trees built using the Euclid
distance, Pearson distance and Image distance between two CGRs have
proven to be more compatible with phylogenetic relatedness of
species than the tree obtained from ClustalW \cite{Wang2004}.
\section{Chaos game representation of protein sequences}
\label{sec:CGRPro}
\subsection{Chaos game using a 20 sided polygon}
\paragraph
\indent The chaos game representation of protein sequences was
used to find the motifs in the sequence, describe regularities in
structure elements, and evaluate various secondary structure
prediction algorithms \cite{Fiser1994}.
\subsubsection{Method}
\indent Fiser et.al in 1994 \cite{Fiser1994} applied Chaos Game
Representation to protein sequences to investigate the motifs in
the protein database and protein sequences. A 20-sided regular
polygon was used to represent the 20 different amino acids. The
$(x,y)$ coordinates of
each of the vertices were given as\\
$$v_{i,x}   = cos(2\pi* i/n)$$
$$v_{i,y} = sin(2\pi *i/n)$$
\subsubsection{Plotting}
\paragraph
\indent The coordinates of the 0th point are [0,0] and the $m$th
point was given by dividing the distance
 between the $(m-1)$th point and the vertex representing the $m$th amino acid using the dividing ratio
 $s_1$ and $s_2$. \\ The coordinates of the points are\\
 $$p_{m,x}  = (v_{m,x} - p_{m-1,x}) * s_{2}+ p_{m-1,x }$$
 $$p_{m,y} = (v_{m,x} - p_{m-1,y}) * s_{2} + p_{m-1,y}$$\\
The dividing ratio $s_1$: $s_2$ is 0.135:0.865 calculated
from\\
$$s_{1}  = sin(2\pi * i/n) / ( 1 + sin(2\pi * i/n))$$
$$s_{2} = 1 / (1 + sin(2\pi * i/n))$$ \\
A lower dividing ratio was used in order to obtain an unambiguous
and decodable fractal for an attractor.
\begin{figure}[!hbp]
\begin{center}
\includegraphics[scale=0.75]{fiser.eps}
\caption{CGR using 20-gon of DNA Polymerase Human Alpha Chain:
Length = 1462} \label{fig:4.4 Motif Detection}
\end{center}
\end{figure}
\subsubsection{Properties}
\paragraph
\indent The attractor produced was a 20-gon in which there are
small separate 20-gons at every vertex. The small 20-gon can
further contain smaller 20-gons in their vertices etc (fig 4.2).
For example, an amino acid subsequence IDEAL can be decoded by zooming the
20-gon at vertex L followed by the  20-gon at vertex A of the L
polygon, the 20-gon at vertex E of the A polygon, the 20-gon at
vertex D polygon and the 20-gon at vertex I of the D polygon.
Theoretically, each point represents the preceding sequence motif.
Eventhough the attractor can be used for identifying subsequences
or motifs, they are indistinguishable as the sequence length
increases. Fiser et.al extended
the chaos game to study the regularities in secondary structure
elements of proteins. The major secondary structures helix, sheet
and turn were represented as vertices of a triangle and the random
coil as the center. The attractor produced was used to study the
frequency of attachment of various secondary structure and
evaluate structure prediction methods. Therefore, CGR could be
used to study both primary and 3D structures of proteins
\cite{Fiser1994}.
\subsubsection{Limitation}
\paragraph
\indent The major drawback of this approach is that as the
sequence length becomes larger all the polygons looks equally
filled. Therefore, various sequence motifs become
indistinguishable.
\subsection{Chaos game using a rectangle}
\subsubsection{Method}
\paragraph
\indent A rectangle was used to represent the sequentiality and composition
of amino acids in a sequence. The rectangle was divided into 5 x 4
sub-rectangles representing 20 different amino acids.
\begin{figure}[!hbp]
\begin{center}
\includegraphics[scale=0.75]{2DPlot.eps}
\caption{2D point representation using rectangle for sequentiality
and composition of amino acids} \label{fig:4.5 Point Pattern}
\end{center}
\end{figure}
\subsubsection{Plotting}
\paragraph
\indent The chaos game is played as follows: the first point is
plotted in the middle of the sub-rectangle labelled with the first
amino acid in the sequence. The \textit{ith} point is plotted by
scaling the \textit{(i-1)th} point by 1/5 in $x$ -direction and
1/4 in $y$-direction and moving the point to the sub-rectangle
labelled with the \textit{ith} amino acid (fig 4.3).
\subsubsection{Properties}
\paragraph
\indent Some characteristic properties noted were that the points
that follow after the insertion and deletion of amino acids are
shifted but the degree of shifting is reduced by $5^n$ in the
$x$-direction and $4^n$ in the $y$-direction, where $n$ is the
number of letters after the inserted/deleted position. Therefore, insertion/deletion does not change the
overall visual impression of the point pattern
. The reduction in shift was also noted when there were repeats in amino acids.
\cite{Pleibner1997}.
\subsection{Chaos game using a 12 sided polygon}
\subsubsection{Method}
\paragraph
\indent CGR can be used to study characteristic patterns of
protein families \cite{Basu1997}. A 12 sided regular polygon was
used to plot a concatenated amino acid sequence of proteins from
protein family (fig 4.4). Each vertex of the polygon corresponded
to a group of amino acid residues of conservative substitutions
(section 3.3) and the amino acids along the vertices of the
polygon were placed in the order of decreasing normalized
hydrophobicity.
\subsubsection{Plotting}
\paragraph
\indent The chaos game was played as in the case of DNA sequences
using the square \cite{Jeffrey1990}. A grid counting algorithm was
used to quantify the CGR.  The 12 sided regular polygon was
divided into 24 segments and the number of points in each segment
was calculated, the percentage of points in each segment was given
by $C_{j}$ / N* 100(j=1..24).
\subsubsection{Properties}
\paragraph
\indent Even though no fractals were detected in the CGRs, there
were specific statistical biases in the distribution of different
amino acids, mono-, di-, tri- or higher order peptides in
functional classes of proteins \cite{Basu1997}. The CGRs of a
protein family were dependent on the relative order of the
residues. The patterns were insensitive to the shuffling of less
abundant residues along the vertices of the polygon, but
sensitive to the shuffling of more abundant residues along the
vertices. The plots are visually similar for a protein class for
many different orientations of the residues. The grid count was
also invariant for a particular family of proteins and particular
orientation of the residue group along the vertices of the CGR
irrespective of the number of sequences concatenated and the order
of concatenation. Therefore, the grid count can be used as a
diagnostic signature of a protein family for identifying new
members of the family and CGR has a potential to reveal
evolutionary and functional relationship between proteins having
no significant homology \cite{Basu1997}.
\begin{figure}
\begin{center}
\includegraphics[scale=0.60]{HSP.eps}
\caption{CGR using 12 sided polygon; HEAT SHOCK PROTEIN 90 (hsp90)
family} \label{fig:4.6}
\end{center}
\end{figure}
\subsection{Chaos game using a square}
\paragraph
\indent Multi-fractal and correlation analysis were performed on
CGR of bacteria families to study the phylogenetic relationship
between sequences and sub-families \cite{Yu2004}.
\subsubsection{Method}
\indent A detailed hydrophobic or non-polar and hydrophilic or
polar (HP) model is used to represent the four classes of amino
acids - non-polar, uncharged, positively charged and negatively
charged as four vertices of the square.
\subsubsection{Plotting}
\paragraph
\indent The chaos game is played as in the case of DNA sequences
\cite{Jeffrey1990}. The square is divided into equal meshes and a
measure $\mu$ for each mesh was calculated by dividing the number
of points lying in the subset of the CGR  by the length of the
sequence. This is represented as a measure matrix $\mathcal{A}$.
Also, a symbolic sequence is created based on the probability of
amino acids in the position of the original sequence and a measure
matrix is calculated for it and referred to as a measure of
fractal background $\mathcal{A}^{f}$. A new measure matrix
$\mathcal{A}^{d}$ is obtained by subtracting the measure matrix of
the fractal background from the measure matrix of the original
sequence and used for calculating the correlation distance.\\
\subsubsection{Properties}
\paragraph
\indent The phylogenetic tree based on correlation distance is
more precise as the mesh size increases \cite{Yu2004}.
Multifractal analysis (\cite{Harte2001}) of the test sequences
exhibited multi-fractal like forms indicating that the protein
sequences of a complete genome are non-random. Also, said, the
correlation analysis is more precise than the multi-fractal analysis for the phylogenetic problem\cite{Yu2004}.
\subsection{Summary}
\paragraph
\indent The following table summarizes the results of Chaos Game
Representation on Protein Sequences in two dimensions.
\begin{landscape}
\begin{table}[!hbp]
\begin{tabular}{|p{3in}|p{2.5in}|p{1.5in}|p{0.5in}|p{0.3in}|}
\hline
 Approach & Novel Advances& Limitations & Authors & Year\\ \hline
Chaos Game using 20 sided polygon, 20 vertices represents 20
amino acids & motif detection in protein database,regularities in
secondary structure elements and evaluation of secondary structure
prediction methods& for large sequences, motif detection is not easy due to resolution of the monitor& Fiser et.al & 1994\\
\hline
Chaos Game using a 5 x 4 rectangle &sequentiality and composition of amino acids & & Pleibner & 1997\\
\hline Chaos Game using 12 sided polygon, 12 vertices represent
12 groups of amino acids & Characteristic patterns of Protein family, measure to detect protein family for a given protein & individual representation of amino acid is lost & Basu et.al & 1997\\
\hline Chaos Game using a square, vertices represent non-polar,
uncharged, positively charged and negatively charged amino acids
&evaluate phylogenetic tree of bacteria &individual representation of amino acid is lost & Yu et.al & 2004\\
\hline
\end{tabular}
\caption{Summary - Chaos game representation of protein sequences
in two dimension} \label{Table: 4.1}
\end{table}
\end{landscape}
