\documentclass{article}

\usepackage{booktabs}
\usepackage{graphics}
\usepackage{graphicx}
\usepackage{listings}
\lstset{basicstyle=\footnotesize\ttfamily,breaklines=true}
\usepackage{nameref}
\usepackage{algorithm}
\usepackage{algpseudocode}
\algrenewcommand\algorithmicrequire{\textbf{Input:}}
\algrenewcommand\algorithmicensure{\textbf{Output:}}
\usepackage{bm}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{caption} 
\usepackage{multirow}
\usepackage{color}
\usepackage[table]{xcolor}
\newcommand*{\red}{\textcolor{red}}
\captionsetup[table]{skip=10pt}
\setlength{\heavyrulewidth}{1.5pt}


\newcommand{\name}{PhaMer}
\newcommand{\Cherry}{PhaMer}
\usepackage{arxiv}

\usepackage[utf8]{inputenc} % allow utf-8 input
\usepackage[T1]{fontenc}    % use 8-bit T1 fonts
\usepackage{hyperref}       % hyperlinks
\usepackage{url}            % simple URL typesetting
\usepackage{booktabs}       % professional-quality tables
\usepackage{amsfonts}       % blackboard math symbols
\usepackage{nicefrac}       % compact symbols for 1/2, etc.
\usepackage{microtype}      % microtypography
\usepackage{lipsum}
\usepackage{graphicx}
\graphicspath{ {./images/} }



\title{PhaBOX: A web server for identifying and characterizing phage contigs in metagenomic data}


\author{
 Jiayu Shang \\
  Dept. of Electrical Engineering\\
  City University of Hong Kong\\
  Kowloon, Hong Kong SAR, China\\
  \texttt{jyshang2-c@my.cityu.edu.hk} \\
  \And
 Cheng Peng \\
  Dept. of Electrical Engineering\\
  City University of Hong Kong\\
  Kowloon, Hong Kong SAR, China\\
  \texttt{cpeng29-c@my.cityu.edu.hk} \\
  \And
 Herui Liao \\
  Dept. of Electrical Engineering\\
  City University of Hong Kong\\
  Kowloon, Hong Kong SAR, China\\
  \texttt{heruiliao2-c@my.cityu.edu.hk} \\
  \And
 Xubo Tang \\
  Dept. of Electrical Engineering\\
  City University of Hong Kong\\
  Kowloon, Hong Kong SAR, China\\
  \texttt{xubotang2-c@my.cityu.edu.hk} \\
  \And
 Yanni Sun \\
  Dept. of Electrical Engineering\\
  City University of Hong Kong\\
  Kowloon, Hong Kong SAR, China\\
  \texttt{yannisun@cityu.edu.hk} \\
}

\begin{document}

\maketitle
\begin{abstract}
\textbf{Motivation:} There is accumulating evidence showing the important roles of bacteriophages (phages) in regulating the structure and functions of microbiome. However, lacking an easy-to-use and integrated phage analysis software hampers microbiome-related research from incorporating phages in the analysis.  \\
\textbf{Results:} In this work, we developed a web server, PhaBOX, to comprehensively identify and analyze phage contigs in metagenomic data. To our best knowledge, this is the first web server that supports integrated phage analysis, including detecting phage contigs from the metagenomic assembly, lifestyle prediction, taxonomic classification, and host prediction. Instead of treating the algorithms as a black box, PhaBOX also supports visualization of the essential features for making predictions. With the user-friendly graphical interface, users with or without informatics training can easily use the web server for analyzing phages in microbiome data. \\
\textbf{Availability:} The web server of PhaBOX is available via: \href{https://phage.ee.cityu.edu.hk}{https://phage.ee.cityu.edu.hk}. The source code of PhaBOX is available via: \href{https://github.com/KennthShang/PhaBOX}{https://github.com/KennthShang/PhaBOX}. \\
\textbf{Contact:} \href{yannisun@cityu.edu.hk}{yannisun@cityu.edu.hk}\\
\end{abstract}



\section{Introduction}
\label{sec:intro}
As viruses mainly infect bacteria, bacteriophages (phages) are the most widely distributed and abundant biological entities in the biosphere \cite{cobian2016viruses}. With an estimated population of more than $10^{31}$ particles \cite{lyon2017phage}, phages play an important role in modulating microbial system dynamics by lysing bacteria \cite{fernandez2018phage}. Recently, accumulating studies show that phages have an important impact on multiple applications, such as disease diagnostics \cite{roth2021developing}, phage therapy \cite{petrovic2020safety}, and food industry \cite{cristobal2021phages}. 


Although there are tools available for different tasks such as phage taxonomic classification and host prediction, these tools are often published as open-source codes. When the users want to characterize phages in metagenomic data, they need to install several tools and parse the intermediate files from different methods. However, installing some of these software requires informatics training and takes substantial computational resources. For example, some open-source codes are not accompanied with detailed user manuals, making the installation tedious and error-prone. Thus, it is preferred to have an integrated web server that can conduct comprehensive phage analysis while sparing the users the time and trouble of installing multiple open-source tools. 
%Because phages ubiquitously exist in many different ecosystems, such as soil and marine samples, we develop this web server to accommodate the needs of users from different fields and help advance the field of phage discovery in different types of ecosystems.


In this work, we present a toolbox for phage analysis (PhaBOX), a comprehensive web service for phage identification, lifestyle prediction, taxonomic classification, and host prediction. In PhaBOX, we optimize the integrated tools to improve the running speed and efficiency. In addition, PhaBOX can visualize the essential features that are important for making the final predictions, such as the similarity-based relationships between the input sequence and other phages, predicted proteins on the input sequences, and protein homology. To our best knowledge, this is the first web server for comprehensive phage  analysis, which can take either metagenomic assemblies or whole genome sequencing assemblies as inputs. All the predictions and intermediate results are provided for downstream analysis.



\begin{figure*}[h!]
    \centering
    \includegraphics[width=0.95\linewidth]{figure.jpg}
    \caption{Overview of workflow in PhaBOX. (A) The pipelines of PhaBOX. The input of PhaBOX is the FASTA file containing assembled contigs. Then, PhaMer \cite{shang2022accurate} is applied for phage identification. Only the contigs predicted as phages will be used for lifestyle prediction (PhaTYP \cite{shang2023phatyp}), taxonomic classification (PhaGCN \cite{shang2021bacteriophage}), and host prediction (CHERRY \cite{shang2022cherry}). Finally, we provided the predictions and visualization for users on the result page. (B) An example of the result page. The visualization can be accessed by clicking the visualize button. (C) All the results can be downloaded from the top-right download panel. The meaning of the head will be shown when moving the cursor onto the icon of the header. (D) The visualization of the homology search on the input contigs. The position of the translated proteins with high alignments score will be shown, and the detailed alignment results (E) can be accessed by clicking the visualize button. (F) We will show the one-step neighbors on the sequence-similarity graph to reveal the relationships between the input contigs and the phages/bacteria in the database. The complete sequence-similarity graph is provided in the ``Network'' panel in (C). They are in the standard format for most network-based visualization tools. One example using Gephi to visualize the graph is shown in (G).}
    \label{fig:model}
\end{figure*}


\subsection{Overview}

PhaBOX is a web server developed with Python/R, providing integrative identification and characterization for phage contigs in metagenomic data. Our algorithms behind PhaBOX were peer-reviewed and published, including PhaMer \cite{shang2022accurate} for phage identification, PhaTYP \cite{shang2023phatyp} for lifestyle prediction, PhaGCN \cite{shang2021bacteriophage} for taxonomic classification, and CHERRY \cite{shang2022cherry} for host prediction. All these tools combined the strength of alignment-based strategies and deep learning models to learn different sequence-based features, including protein organizations, sequence homology, and protein-protein associations. Our methods outperform the available programs in each task based on our rigorous tests on highly diverged phages, short contigs, mock metagenomic data, and real metagenomic data. For example, according to a third-party view \cite{tang2022phage}, the earlier version of CHERRY, named HostG \cite{shang2021predicting} has the best performance on predicting the phage-host relationship on the genus level. In CHERRY, we not only further improved the accuracy but also supported host prediction at the species level.


In PhaBOX, we develop a web server for a one-stop shop for phage identification and analysis. PhaBOX has a modular design and thus supports two modes of running. The default mode allows users to conduct end-to-end analysis including all the supported analyses. The other mode allows users to only run the modules they need, e.g. predicting the hosts for recently sequenced phages. As shown in Fig. \ref{fig:model} A, the input of PhaBOX is the FASTA file containing assembled contigs from metagenomic data or whole genome sequencing data. Then, PhaMer is applied for phage identification. By default, the identified phage contigs will be used as inputs for all modules including lifestyle prediction, taxonomic classification, and host prediction. For each module, we updated the database and optimized the program to save computing resources. The running time of PhaBOX is \textasciitilde70\% of the total time of running each program individually. We also provided a multi-threads local version of PhaBOX for users who want to analyze phages in a large amount of metagenomic data.

In addition to the advantages of faster running speed, we provide detailed visualization of how our programs make predictions, which is not available in the previous works. As shown in Fig. \ref{fig:model} B to G, we visualized the essential components in the model, such as the similarity-based relationships between the contigs and other phages, predicted proteins on the contigs, and protein homology, to show evidence for generating predictions. 

In the following sections, we will showcase how to use PhaBOX for identifying and characterizing phage contigs in metagenomic data. We will demonstrate the utility of PhaBOX and detail the visualization functions provided on the result page.


\vspace{-0.2cm}
\section{Case study}

There is accumulating evidence showing the need of virus analysis in microbiome data analysis \cite{rasmussen2020bacteriophage}. In this case study, we demonstrate that PhaBOX can simplify and speed up phage analysis in human microbiome data. Specifically, we will use PhaBOX to investigate whether phages can be potential biomarkers for type 2 diabetes (T2D). 

Over the past few decades, observations have suggested that gut microbes may contribute to the pathogenesis of various common metabolic disorders. T2D is one of the most well-studied diseases, and several studies \cite{fan2021gut, qin2012metagenome} have shown that some bacteria genera are positively associated with T2D. In this case study, we downloaded the data provided by \cite{qin2012metagenome} and used PhaBOX to identify and characterize the phages from the gut metagenomic samples. 


\subsection{Example of the result page}

The dataset contains eight groups of cohorts, including 71 T2D patients and 74 controls. We downloaded the assembled contigs provided by the authors and uploaded the FASTA files to the PhaBOX server. To maintain high-quality predictions, we set the minimum length of contigs as 10kbp as suggested in \cite{qin2012metagenome}. One example result page of cohorts with the prefix ``DLF'' is shown in Fig. \ref{fig:model} B. There are totally six modules on the result page, including the tab for downloading results,  pie graphs of the prediction, contig results, phage family classification results, phage host prediction results, and low confidence results. Because of the limitation of the image size, we only show the first four modules in Fig. \ref{fig:model} B. The meaning of each module is listed below.


\paragraph{The module for downloading results} As shown in Fig. \ref{fig:model} C, this module contains all the predictions and intermediate files for users to download. The standard outputs of PhaBOX are four CSV files containing the results of identified phages, their taxonomy, lifestyles, and hosts. PhaBOX also provides intermediate files of the essential features, including the homology of phage-related proteins output by PhaMer/PhaTYP and knowledge network output by PhaGCN/CHERRY. The homology files containing phage-related proteins in FASTA format and BLAST alignment results in CSV format. The knowledge network contains several types of interactions between metagenomic assemblies and phages/bacteria in the database, such as gene-sharing information, CRIPSRs, and sequence similarity. The network files are in standard input formats for most network-based visualization tools, such as Gephi \cite{bastian2009gephi} and Cytoscape \cite{shannon2003cytoscape}.

\paragraph{Pie graphs of prediction}  In the PhaBOX web server, pie graphs are employed to visualize the final prediction results. There are four pie charts in total, including the results of the percentage of identified phages, taxonomic composition, lifestyle composition, and host prediction. As shown in the bottom-right of Fig. \ref{fig:model} B, the default pie graph shows the percentage of identified phages. 

\paragraph{Contig results} As shown in Fig. \ref{fig:model} B, the top-left table contains all the identified phages and their lifestyles predicted by PhaMer and PhaTYP. The visualization of the homology search on the input contigs can be accessed by clicking the ``Visualize'' buttons. Then, Fig. \ref{fig:model} D will be presented to show the position of the translated proteins with their alignments score. Finally, each protein's detailed alignment results can be accessed as shown in Fig. \ref{fig:model} E.



\paragraph{Phage family classification results and host prediction results} As shown in Fig. \ref{fig:model} B, the bottom-left tables contain the family-level taxonomic classification results predicted by PhaGCN and host prediction results predicted by CHERRY. Because both methods utilize sequence similarity and gene organization as features to construct phage-phage and phage-bacterium relationship networks, we provide the visualization of these networks. However, the complete similarity network is too large to be visualized on a web page. Thus, we only show the one-step neighborhood of each contig as shown in Fig. \ref{fig:model} F. The visualized network is interactive, and users can drag the nodes to check their accessions. As mentioned in \textit{download results} module, the complete graph can be conveniently generated using our provided file and any of the commonly used network visualization tools such as Gephi. An example of a visualized complete graph by Gephi is shown in Fig. \ref{fig:model} G.



\paragraph{Low confidence results} This module is not shown in Fig. \ref{fig:model} but is similar to the \textit{contig result} module. It is a table containing all contigs that have alignments with the database, but the confidence scores given by the model are too low to be predic-ted as phages. There are two possible reasons for the low confidence. First, there may exist assembly errors in the contigs. Second, the contigs contain regions that are shared by phage and bacteria. Thus, we provide the corresponding results and visualization for the user who wants to further analyze these sequences.


\section{Conclusion}
Because phages ubiquitously exist in many different ecosystems, such as soil and marine samples, we develop this web server to accommodate the needs of users from different fields and help advance the field of phage discovery in different types of ecosystems. The integration of phage identification, lifestyle prediction, taxonomic classification, and host prediction of our platform provides not only comprehensive analysis for metagenomic assemblies but also detailed  visualizations for users. The case study demonstrates that PhaBOX can facilitate users with fast and convenient phage characterization in metagenomic data. Future improvements include adding more functions, such as novel protein annotation and protein structural analysis. We will also upgrade the hardware of our server to provide a faster user experience.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%     please remove the " % " symbol from \centerline{\includegraphics{fig01.eps}}
%     as it may ignore the figures.
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\section*{Method and materials}
The architecture of the PhageBOX server consists of two major components: a client web interface and a server backend. The client web interface is responsible for submitting the tasks and displaying the output. It was implemented by JS, CSS, jQuery, Bootstrap, and their extension packages. Specifically, the sequence similarity was visualized by BlasterJS, the protein sequence viewer was presented using pViz, and the topological graph structure was drawn using Plotly in R. The server backend is responsible for interacting with users through the web interface, handling usersâ€™ input, and executing the whole prediction process. The former interface was implemented by the fast and lightweight python-based Flask framework and the extension python packages. The server backend employs a lite SQL database that stores and updates the job information and status. The scheduling method also allows the architecture to be added to add new computational facilities to meet the increasing demand in predicting ever-accumulating genome-scale data. More detailed information can be found under the ``Home'' tab on the PhaBOX web page.



\section*{Funding}
City University of Hong Kong (Project 9678241 and 7005453) and the Hong Kong Innovation and Technology Commission (InnoHK Project CIMDA).
\vspace{-0.2cm}
\bibliographystyle{unsrt}


\bibliography{references}

\end{document}
