% !TeX spellcheck = en_GB
\PassOptionsToPackage{x11names}{xcolor}
\documentclass[12pt,a4paper,twoside]{article}

\usepackage[colorinlistoftodos, textwidth=30mm, shadow]{todonotes}
%\renewcommand\familydefault{\sfdefault}

%%%%%%% packages
\usepackage{array}
\usepackage{amsmath}
\usepackage{tabularx}
\usepackage{txfonts}
\usepackage{enumerate}
%\usepackage{cite}
\usepackage{float}
\usepackage{xspace}
\usepackage[x11names]{xcolor}
\usepackage{multirow}
\usepackage{algpseudocode,algorithmicx}
\usepackage{booktabs}
\usepackage{siunitx}
\usepackage{makecell}

\usepackage{subcaption}
%\usepackage{subfig}

\usepackage{pdflscape}

%\usepackage{lineno}

%\usepackage{tikz}
%\usepgflibrary{shapes}
%\usetikzlibrary{spy}
%\usepackage{pgfplots}
%\pgfplotsset{compat=1.11}

%\usepackage{showkeys}
%%\usepackage{showlabels}

\usepackage[T1]{fontenc}
\usepackage[cp1250]{inputenc}

\usepackage[margin=2.5cm]{geometry}
%\usepackage[margin=2cm]{geometry} %optimal
%\usepackage[margin=2.5cm,head=15pt]{geometry}

\usepackage[colorlinks=true,citecolor=red,linkcolor=DarkOrange1]{hyperref}

%\usepackage{showlabels}

\topmargin 0.0cm
\oddsidemargin 0.2cm
\textwidth 16cm 
\textheight 21cm
\footskip 1.0cm



\usepackage{listings}
\lstset{basicstyle=\footnotesize\ttfamily,breaklines=true, tabsize=2}

\usepackage{placeins} % FloatBarrier

% More pictures allowed per page
\renewcommand{\topfraction}{0.9}
\renewcommand{\bottomfraction}{0.8}
\renewcommand{\floatpagefraction}{0.8}
\renewcommand{\dblfloatpagefraction}{0.8}
\setcounter{topnumber}{3}
\setcounter{bottomnumber}{3}
\setcounter{totalnumber}{4}
% More pictures allowed per page (end)

%\renewcommand\thesection{S\arabic{section}}
%\renewcommand{\thefigure}{S\arabic{figure}}

%\usepackage[T1,plmath]{polski} % pakiet ustawiajcy polskie wzorce 

\usepackage{changes}
\colorlet{cmonika}{RoyalBlue3}
\colorlet{cpaw}{Orchid4}
\colorlet{cwojtek}{Aquamarine4}
\colorlet{csimon}{Tomato3}
\colorlet{crav}{DeepSkyBlue4}


\definechangesauthor[name={Monika}, color=cmonika]{Monika}
\definechangesauthor[name={Pawel}, color=cpaw]{Pawel}
\definechangesauthor[name={Wojtek}, color=cwojtek]{Wojtek}
\definechangesauthor[name={Simon}, color=csimon]{Simon}
\definechangesauthor[name={Rafal}, color=crav]{Rafal}

\newcommand{\madd}[2][]{\added[id=Monika,remark=#1]{#2}}
\newcommand{\mdel}[2][]{\deleted[id=Monika,remark=#1]{#2}}

\newcommand{\padd}[2][]{\added[id=Pawel,remark=#1]{#2}}
\newcommand{\pdel}[2][]{\deleted[id=Pawel,remark=#1]{#2}}

\newcommand{\wadd}[2][]{\added[id=Wojtek,remark=#1]{#2}}
\newcommand{\wdel}[2][]{\deleted[id=Wojtek,remark=#1]{#2}}

\newcommand{\sadd}[2][]{\added[id=Simon,remark=#1]{#2}}
\newcommand{\sdel}[2][]{\deleted[id=Simon,remark=#1]{#2}}

\newcommand{\radd}[2][]{\added[id=Rafal,remark=#1]{#2}}
\newcommand{\rdel}[2][]{\deleted[id=Rafal,remark=#1]{#2}}

\newcommand{\mon}[1]{\todo[inline,color=cmonika]{\color{black}#1}}
\newcommand{\paw}[1]{\todo[inline,color=cpaw]{\color{black}#1}}
\newcommand{\woj}[1]{\todo[inline,color=cwojtek]{\color{black}#1}}
\newcommand{\simo}[1]{\todo[inline,color=csimon]{\color{black}#1}}
\newcommand{\rav}[1]{\todo[inline,color=crav]{\color{black}#1}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%  
% Authors' definitons
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\newcommand{\abs}[1]{\left\vert#1\right\vert}
\newcommand{\norm}[1]{\left\Vert#1\right\Vert}
\newcommand{\scal}[1]{\langle #1\rangle}
\DeclareMathOperator{\sign}{\mathrm{sign}}
\DeclareMathOperator{\e}{\mathrm{e}}
\DeclareMathOperator{\dd}{\mathrm{d\!}}
\DeclareMathOperator{\Int}{\mathrm{Int}}
%\DeclareMathOperator{\mod}{\mathrm{mod}}

\newcommand*{\dwd}[1][]{{\par\noindent\bfseries {\em Proof#1: }}}
\newcommand*{\cbdo}{\hspace*{\fill} $\rule{1.5ex}{1.5ex}${\par}\vspace{2em}}
\newcommand{\etal}{\textit{et al.}\xspace}



\renewcommand{\Re}{\textrm{Re}}
\renewcommand{\Im}{\textrm{Im}}

\newcommand{\N}{\varmathbb{N}}
\newcommand{\R}{\varmathbb{R}}
\newcommand{\C}{\varmathbb{C}}
\newcommand{\Cb}{\mathcal{C}}
\newcommand{\Kb}{\mathcal{K}}

\newcommand*{\kryt}{\textrm{cr}}
\newcommand*{\vp}{\mathbf{p}}
\newcommand*{\vq}{\mathbf{q}}

\newtheorem{theorem}{Theorem}[section]
\newtheorem{lemma}[theorem]{Lemma}
\newtheorem{proposition}[theorem]{Proposition}
\newtheorem{cor}[theorem]{Corollary}
\newtheorem{rem}[theorem]{Remark}
\newtheorem{defs}[theorem]{Definition}

\newenvironment{proof}[1][]{\par\noindent\textbf{Proof#1: }}{\hfill\rule{1.3ex}{1.3ex}\vspace{0.5cm}\par}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
% end of authors' definitions
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\numberwithin{equation}{section}


\begin{document}
	\pagestyle{myheadings}
	\markright{xxxx et al.}

	% ------------------------------------------------------------------------- %
	\noindent\begin{tabular}{|p{\textwidth}}
		\Large\bf  
		Identifying Promising Candidate Radiotherapy Protocols via GPU-GA \textit{in-silico}\\(Candidate Radiotherapy Protocols via GPU-GA)
		%\mon{Full titles can be a maximum of 135 characters. Short titles can be a maximum of 50 characters.}
		\\\vspace{0.01cm}

        \it  W. Ozimek$^{1}$,  R. Bana{\'s}$^{2}$, P. Gora$^{*,3}$, S.D. Angus$^{4}$, M.J. Piotrowska$^{5,\dagger,\ddagger}$ 
		\\\vspace{0.02cm}
		\it\small $^4$Dept. of Economics, Monash University; and SoDa Laboratories, Monash Business School, Monash University,\\\vspace{0.01cm}
		\it\small $^5$Institute of Applied Mathematics and Mechanics; Faculty of Mathematics, Informatics and Mechanics,\\
		\it\small University of Warsaw, Banacha 2, 02-097 Warsaw, Poland,\\\vspace{0.01cm}


		\small $^\dagger$\texttt{monika@mimuw.edu.pl},\\
		\small $^{1}$\texttt{p.gora@mimuw.edu.pl}\\
		\small $\ddagger$ equal contribution\\
		\small $*$ corresponding author\\
		%\multicolumn{1}{|r}{\large\color{orange} Research Article} \\
		\\
		%\hline
	\end{tabular}
% ------------------------------------------------------------------------- %
% 	\thispagestyle{empty}

% %\tableofcontents
% %\addtocontents{toc}{\protect\setcounter{tocdepth}{1}}

% \noindent\begin{tabular}{p{\textwidth}}
% 	\\
% 	\hline
% \end{tabular}
% \vspace{1em}\\
%-------------------------------------------------------------------------

\begin{abstract}

% \mon{The abstract should be a single paragraph (no more than 150 words) written in plain language that a general reader can understand. It should include:
% An opening sentence that states the question/problem addressed by the research AND
% Enough background content to give context to the study AND
% A brief statement of primary results AND
% A short concluding sentence.}

In this paper we introduce computational techniques applied to high-fidelity modelling and optimisation of EMT6/Ro tumour growth under multi-day radiotherapy treatment protocols. The original model, utilising cellular automata, implemented in MATLAB, and applying genetic algorithm (GA) search techniques to only a single protocol parameter to identify candidate irradiation protocols, was re-implemented in C++ and enhanced with GPUs to achieve large execution speedups, enabling the search of a much larger, multi-parameter search space. Further, new operators were developed in the GA re-implementation to conduct selection, crossover and mutation. Together these computational design and implementation developments resulted in improvements of c. 760x solution execution speedup  compared to the original architecture, and 6-8\% protocol fitness improvement.
\end{abstract}

%-------------------------------------------------------------------------


% \section*{TEST}
% \mon{
% The manuscript should be a maximum of 15,000 words.\\
% There is only one reference list for all sources cited in the main text, figure and table legends, and Supplementary Materials, and this main list should not exceed 40 citations.\\
% from template file: You may include up to a total of six figures and/or tables (combined) throughout the manuscript.\\ web page info: You may include up to ten figures and/or tables and about 60 references.
% }%mon

%See \cite{Angus:2013id}. Text \padd{and results} \mdel{for this section}, \madd{as per the individual} \wadd{journal's} \wdel{instructions for authors.} \sadd{hnasicac} \sdel{asdcbwieuvc} \radd{hnasicac} \rdel{asdcbwieuvc}
%\cite{koon,oreg,khar,zvai,xjon,schn,pond,smith,marg,hunn,advi,koha,mouse}
%\mon{test}\paw{test2}\woj{test3}\simo{test4}\rav{test5}

%-------------------------------------------------------------------------
\section{Introduction}

Radiotherapy protocols in cancer treatment -- the schedule of dose size and timing -- are fundamental to radiotherapy treatment design, but the choice set for clinicians is immense due to combinatorial considerations. Consider that, with an alphabet of just five fractional dose sizes, and 10 possible wait-times for the next fraction, a 10 fraction protocol can be constructed in any one of almost $1\times 10^{17}$ ways ($(10\times 5)^{10}$).  At present, the most commonly used radio- and chemo- therapy protocols in clinical settings represent a tiny fraction of possible protocols that could be used, being more a product of trial-and-error and received wisdom, than a~systematic search (REF). In contrast, calibrated, high-fidelity, cellular automata (CA) models coupled to highly non-linear search via genetic algorithms (GAs) have demonstrated promising opportunities to efficiently identify more effective candidate protocols~\cite{Angus2014}. However, the full potential of the GA approach requires a very fast execution time, and previous methods were highly constrained in this regard, requiring over 40~min for a single candidate radiotherapy scheme to be tested. As such, prior work fixed the dose fraction (1.25 or 2.0 Gy), and explored only the fractional delay dimension space. To truly leverage the power of high-fidelity numerical simulation, a step-change in candidate protocol testing throughput is needed.

Here we introduce and demonstrate the validation of a high fidelity, high performance computational model of EMT6/Ro spheroids leveraging state-of-the-art computational approaches on Graphical Processor Unit (GPU) architectures coupled to GA search, an approach we call `GPU-GA', and apply this approach to optimal radiotherapy scheme discovery, at scale. By analysing over XX candidate protocols, the GPU-GA approach yields improved suppression (?fitness) scores in the range XX to YY, and XX to YY, versus state-of-the-art outcomes under two clinically relevant benchmark constraints respectively. Significantly, by extending the search beyond the state-of-the-art constrains (to fractional doses greater than 2.0 Gy), we identify improved fitness scores as much as XX\% over even the best protocols identified in the present work under the benchmark conditions studied. By pooling all highly effective protocols discovered in the GPU-GA search activity, we enumerate, and model, the frequency surface of these potent [dose,delay] sub-modules, finding an imputed maximum potency frequency location of 1.4~Gy fraction administered with a 15.5~h delay. Finally, by leveraging the outcomes of the response surface activity, we demonstrate how hand-crafted candidate protocols composed of the most potent sub-modules can be formed and tested, in a form of computationally-supported treatment design.

Our work extends the promising field of `computational oncology' (see e.g.~\cite{Gerlee2007,Patel2001,Dormann2002,Kansal2000,Kansal2000a})  which aims to offer a~better understanding of cancer dynamics {\it `in silico'} which can be used to improve therapeutic outcomes {\it in vitro}. Importantly, we build on our previous work (cf.~\cite{Piotrowska:2009gj,Angus:2010mo,Angus:2013id}) where we show that computational CA~models of EMT6/Ro spheroid growth can be successfully calibrated to a broad range of experimental data. This provides an extremely promising baseline for the introduction of 
radiotherapy and/or chemotherapy modules to investigate therapeutic approaches. To this point, of the few authors that have attempted to implement a~computational tumour with a~therapeutic model 
(e.g.~\cite{Zacharaki2004,Stamatakos2006,Kempf2010, Powathil2012,Powathil2013,Kempf2013}) none have fully, if at all, calibrated the computational model to experimental data nor applied a~systematic search of the possible treatment protocols. On the other hand, our quasi-3D CA models of avascular tumour growth, built over several years and rigorously calibrated and tested at each stage of development, provide a rich, \emph{high-fidelity} computational model of tumour growth under irradiation.

Central to the high throughput approach we develop in the present work is the translation of an existing high-fidelity EMT6/Ro CA model of avascular tumour growth from its parallelised, classical CPU-based {\sc Matlab} language implementation into a parallelised GPU-architecture C++ implementation. Here, we leverage similar approaches in the rapidly emerging field of high-throughput screening (HTS) where a mixture of processor-, big-data-, statistical search/learning tools- and novel computational architectures are converging to deliver rapid breakthroughs in molecular candidate identification~\cite{Liu2017, Lightbody2019}. The underlying coupled EMT6/Ro tumour growth --- multi-irradiation 
model~\cite{Angus2014}, was calibrated across 18 independent experimental multi-fraction studies, requiring the direct modelling of DNA damage and repair processes following known biological systems.

Taken together, we demonstrate the power of GPU-GA computational biology to rapidly and effectively search within a vast solution space for candidate protocols that could be explored in the clinic.  The approach demonstrated here has major implications for computational oncology research, as the demonstrated approach could be applied to any cell line with sufficiently available validation data, and could lead to the rapid identification of radiotherapy schemes to be tried in the clinical setting.

% Next, using GAs, we explored thousands of alternative multi-fraction protocols 
% and compared their performance to clinical baseline approaches. The protocols obtained by this method gave an average improvement of 9.4\% (max benefit 16.5\%) and 7.1\% (13.3\%) improvement (reduction) on tumour cell count compared to the two considered benchmark protocols, respectively, over just a~ single week of equivalent treatment. Hence, by implementing and analysing calibrated computational CA~models of tumours, we are well placed to discover novel clinically relevant, treatment protocols in this innovative research area.
% \madd{
% Nevertheless, due to the stochastic nature of our CA,~every treatment or benchmark protocol should be 
% tested over varying digital tumours and with several random seeds and the results 
% should be compiled and reduced using standard statistical methods (including 
% statistical tests), generating computational costs.  
% Moreover, even the use of cellular 
% automata models to investigate a~vast space of possible protocols is 
% challenging. For instance, to define the multi-fraction protocol one needs to 
% give a~number of fractions per day, timing of each fraction within a~day and 
% intensity (dosage) of each fraction. So, considering only a~`10 fraction' 
% program with fixed dosage, with time-gaps between fractions in 
% \{18,18.5,...,29.5,30\} hours, one obtains 95 trillion different protocols. In the presented work, we rise beyond previous restrictions regarding the investigated protocols (fixed fractional doses) and considered much bigger protocol space. To reach that aim, a very fast simulator is required. Thus, we decided to convert our existing CPU-based paralleled code implemented in the {\sc Matlab} programming environment (that can be downloaded from http://dx.doi.org/10.6084/m9.figshare.1227531) to GPU-based paralleled code implemented in C++.}

%-------------------------------------------------------------------------

\section{Results}

%-------------------------------------------------------------------------
\subsection*{Evaluation of the base(?) protocols} A potency to discovered better schemes (regarding the tumour size) according to benchmarks, presentation of benchmarks [MJP],  present results for new fitness function for benchmarks and best protocols from Plos One paper ({\bf table 1})

\madd{
Benchmark info: \\
In the following paper we proposed protocols that essentially bring larger benefits to the patients comparing to clinically used (benchmark) protocols, \cite{Rosenstein:2004vs, Angus2014}, at the same time keeping the maximal dose of 10 Gy (per week) to ensure that radiotherapy side-effects are at acceptable level. Both benchmark protocols, (BMI, BMII), characterised in Table \ref{tab:ga_results}, are based on a low (single) dose multi-fraction treatment approach \cite{Rosenstein:2004vs, ORourke:2009, BoardoftheFacultyofClinicalOncology:2006wq} and, in the same way as for all tested protocols, their effectiveness has been estimated using newly defined fitness function (see formula \ref{eq:final_fit_fun} for details) \pdel{described in Section~\ref{sec:GPU:implem}}. In Table~\ref{tab:ga_results}, due to the update of fitness function definition comparing to our previous study~\cite{Angus2014}, we also report the best sub-optimal protocols (for which we varied only the timing of the
fractions not doses) found in~\cite{Angus2014}.    
}%madd

% \begin{table}[hp!]
%     \centering
%     \caption{Characterisation of benchmark protocols (BMI and BMII), best sub-protocols found in~\cite{Angus2014} (BMI 1-BMI 3 and BMII 1-BMII 3), and best protocols found using GA algorithm 
%     (GA1-GA4). Mean fitness scores and standard deviations (sd) calculated based on fitness function~\eqref{eq:final_fit_fun} defined in Section~\ref{sec:GPU:implem} evaluated 100 times for each of 10 tumours from study library, for details see Section~\ref{sec:lib}. Results show that even small differences in dose intervals or dose values have a significant impact on protocol performance. To correctly compare GA algorithm with benchmark protocols we established constraints of minimum 12 and 24~hours between doses and 1.25 and 2.0~Gy maximum single dose. Min time interval and max dose indicate the constrains for the GA except the values indicated with $^*$ which were fixed.
%     The p-value (compared to BMII) indicates the probability that the GA algorithm results (GA1-GA4) are from the same set as the BMII results. The formula used for p-value calculation is available in \cite{SciPyTTest}.
%     }\label{tab:ga_results}
%     \small
%     \hspace*{-1cm}
%     \begin{tabular}{|l|c|c|c|c|c|c|c|}
% \hline
% Id & Protocol schedule (Dose,   & Total &  Max & Min time & Selection & GPU mean & p-value\\
%  & inter-fraction time-gap)  & dose  & dose & interval & type & fitness score (sd) & (compared\\
%  & ([Gy], [h]) & [Gy] & [Gy] & [h] & (only for GA) & & to BMII)\\ \hline\hline
 
 
%  \multirow{4}{*}{BMI}  & (1.25, 6), (1.25, 18), & \multirow{4}{*}{10} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{6$^*$ and 18$^*$} &  \multirow{4}{*}{-} & \multirow{4}{*}{1120.16 (39.71)} & \multirow{4}{*}{-} \\ %\cline{4-4}
%                       & (1.15, 6), (1.25, 18), & & & & & & \\ %\cline{4-4}
%                       & (1.25, 6), (1.25, 18),  & & & & & & \\ %\cline{4-4}
%                       & (1.25, 6), (1.25, 18)  & & & & & & \\ \hline
% \multirow{4}{*}{BMI 1} & (1.25, 18.5), (1.25, 20), & \multirow{4}{*}{8.75} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Tournament \cite{Angus2014}} & \multirow{4}{*}{1142.19 (33.92)} & \multirow{4}{*}{-} \\ %\cline{4-4}
%                       & (1.25, 18.5), (1.25, 17.5), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 16.5), (1.25, 13), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 10) & & & & & &\\ \hline
% \multirow{4}{*}{BMI 2} & (1.25, 18.5), (1.25, 20), & \multirow{4}{*}{8.75} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Top 1 \cite{Angus2014}} & \multirow{4}{*}{1140.48 (33.52)} & \multirow{4}{*}{-}\\ %\cline{4-4}
%                       & (1.25, 18.5), (1.25, 17.5), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 16.5), (1.25, 12.5), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 10) & & & & & &\\ \hline
% \multirow{4}{*}{BMI 3} & (1.25, 18.5), (1.25, 20), & \multirow{4}{*}{8.75} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Tournament \cite{Angus2014}} & \multirow{4}{*}{1141.57 (32.82)} & \multirow{4}{*}{-}\\ %\cline{4-4}
%                       & (1.25, 18.5), (1.25, 17.5), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 16.5), (1.25, 14), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 10) & & & & & &\\ \hline\hline
% \multirow{3}{*}{BMII} & (2, 24), (2, 24), & \multirow{3}{*}{10} & \multirow{3}{*}{2$^*$} & \multirow{3}{*}{24$^*$} &  \multirow{3}{*}{-} & \multirow{3}{*}{1148.68 (37.05)} & \multirow{3}{*}{-}\\ %\cline{3-3}
%                       & (2, 24), (2, 24), & & & & & &\\ %\cline{3-3}
%                       & (2, 24) & & & & & &\\\hline
% \multirow{2}{*}{BMII 1} & (2, 15), (2, 22), & \multirow{2}{*}{8} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top 5 \cite{Angus2014}} & \multirow{2}{*}{1158.66 (42.25)} & \multirow{2}{*}{-} \\ %\cline{2-2}
%                       & (2, 15.5), (2, 23.5) & & & & & & \\\hline

% \multirow{2}{*}{BMII 2} & (2, 15), (2, 19), & \multirow{2}{*}{8} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top 5 \cite{Angus2014}} & \multirow{2}{*}{1155.87 (43.92)} & \multirow{2}{*}{-}\\ %\cline{2-2}
%                       & (2, 15.5), (2, 18) & & & & & &\\\hline

% \multirow{2}{*}{BMII 3} & (2, 15), (2, 23), & \multirow{2}{*}{8} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top 1 \cite{Angus2014}} & \multirow{2}{*}{1159.33 (41.82)} & \multirow{2}{*}{-}\\ %\cline{2-2}
%                       & (2, 15.5), (2, 22.5) & & & & & & \\\hline\hline
 
% \multirow{4}{*}{GA 1}   & (1.25, 8.5), (1.25, 18.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{4}{*}{Simple} & \multirow{4}{*}{1153.62 (34.56)} & \multirow{4}{*}{0.002}\\ %\cline{4-4}
%                       & (1.25, 18.5), (1.25, 15.5) & & & & & & \\ %\cline{4-4}
%                       & (1.25, 15.5), (1.25, 13.5) & & & & & &\\ %\cline{4-4}
%                       & (1.25, 15), (1.25, 14) & & & & & &\\ \hline
% \multirow{4}{*}{GA 2}   & (0.25, 17.5), (1.75, 17),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Tournament} & \multirow{4}{*}{1173.96 (37.80)} & \multirow{4}{*}{0} \\ %\cline{4-4}
%                       & (1.5, 20.5), (1.5, 15) & & & & & & \\ %\cline{4-4}
%                       & (1.75, 16), (1.5, 18.5) & & & & & & \\ %\cline{4-4}
%                       & (1.75, 15) & & & & & &\\ \hline
% \multirow{4}{*}{GA 3}   & (1.25, 8), (1.25, 25),  & \multirow{4}{*}{6.25} & \multirow{4}{*}{1.25} & \multirow{4}{*}{24} & \multirow{4}{*}{Tournament} & \multirow{4}{*}{1034.66 (37.11)} & \multirow{4}{*}{0}\\ %\cline{4-4} 
%                       & (1.25, 24), (1.25, 26.5), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 25) & & & & & &\\ %\cline{4-4}
%                       & & & & & & &\\ \hline
% \multirow{4}{*}{GA 4}   & (2, 8), (2, 24.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{24} & \multirow{4}{*}{Simple} & \multirow{4}{*}{1149.04 (35.99)} & \multirow{4}{*}{0.826}\\ %\cline{4-4}
%                       & (2, 24.5), (2, 24.55), & & & & & & \\ %\cline{4-4}
%                       & (2, 24.5) & & & & & & \\ %\cline{4-4} 
%                       & & & & & & &\\ \hline
%     \end{tabular}
    

%     % \caption{Best protocols found using GA algorithm. For comparison purposes with protocols in \ref{tab:bench}, presented protocols have constraints of minimum 12 and 24~hours between doses and 1.25 and 2.0~Gy maximum single dose. Fitness scores are calculated based on fitness function~\eqref{eq:final_fit_fun} defined in Section~\ref{sec:GPU:implem} evaluated 100 times for each of 10 tumours from study library, for details see Section~\ref{sec:lib}}.
%     %\mon{consider to change it -- shall we include CPU, GPU results?}
% \end{table}

\begin{table}[htp!]
    \centering
    \caption{
    Characterisation of benchmark protocol (BMI), best sub-protocols found in~\cite{Angus2014} (BMI 1-BMI 3), and best protocols with the limit of 1.25~Gy dose found using GA algorithm (GA1.25 1-GA1.25 3), see Figure~\ref{fig:benchmark_comaprison} A. Mean fitness scores and standard deviations calculated based on fitness function~\eqref{eq:final_fit_fun} defined in Section~\ref{sec:GPU:implem} evaluated 100 times for each of 10 tumours from study library, for details see Section~\ref{sec:lib}. Results show the superiority in protocol performance of GA algorithm. To correctly compare GA algorithm with benchmark protocols we established constraint of minimum 12~hours between doses and 1.25~Gy maximum single dose. Min time interval and max dose indicate the constrains for the GA except the values indicated with $^*$ which were fixed.
    The p-value, compared to BMI, indicates the probability that the GA algorithm results (GA1.25 1-GA1.25 3) are from the same set as the BMI results. The formula used for p-value calculation is available in \cite{SciPyTTest}.
    }\label{tab:ga_bmi_results}
    \small
    \hspace*{-1cm}
    \begin{tabular}{|l|c|c|c|c|c|c|c|}
\hline
Id & Protocol schedule (Dose,   & Total &  Max & Min time & Selection & GPU mean & p-value\\
 & inter-fraction time-gap)  & dose  & dose & interval & type & fitness score & (compared\\
 & ([Gy], [h]) & [Gy] & [Gy] & [h] & (only for GA) & & to BMI)\\ \hline\hline
 
 
 \multirow{4}{*}{BMI}  & (1.25, 6), (1.25, 18), & \multirow{4}{*}{10} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{6$^*$ and 18$^*$} &  \multirow{4}{*}{-} & \multirow{4}{*}{1120.16 (39.71)} & \multirow{4}{*}{-} \\ %\cline{4-4}
                       & (1.15, 6), (1.25, 18), & & & & & & \\ %\cline{4-4}
                       & (1.25, 6), (1.25, 18),  & & & & & & \\ %\cline{4-4}
                       & (1.25, 6), (1.25, 18)  & & & & & & \\ \hline
\multirow{4}{*}{BMI 1} & (1.25, 18.5), (1.25, 20), & \multirow{4}{*}{8.75} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Tournament \cite{Angus2014}} & \multirow{4}{*}{1142.19 (33.92)} & \multirow{4}{*}{-} \\ %\cline{4-4}
                       & (1.25, 18.5), (1.25, 17.5), & & & & & &\\ %\cline{4-4}
                       & (1.25, 16.5), (1.25, 13), & & & & & &\\ %\cline{4-4}
                       & (1.25, 10) & & & & & &\\ \hline
\multirow{4}{*}{BMI 2} & (1.25, 18.5), (1.25, 20), & \multirow{4}{*}{8.75} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Top 1 \cite{Angus2014}} & \multirow{4}{*}{1140.48 (33.52)} & \multirow{4}{*}{-}\\ %\cline{4-4}
                       & (1.25, 18.5), (1.25, 17.5), & & & & & &\\ %\cline{4-4}
                       & (1.25, 16.5), (1.25, 12.5), & & & & & &\\ %\cline{4-4}
                       & (1.25, 10) & & & & & &\\ \hline
\multirow{4}{*}{BMI 3} & (1.25, 18.5), (1.25, 20), & \multirow{4}{*}{8.75} & \multirow{4}{*}{1.25$^*$} & \multirow{4}{*}{4} &  \multirow{4}{*}{Tournament \cite{Angus2014}} & \multirow{4}{*}{1141.57 (32.82)} & \multirow{4}{*}{-}\\ %\cline{4-4}
                       & (1.25, 18.5), (1.25, 17.5), & & & & & &\\ %\cline{4-4}
                       & (1.25, 16.5), (1.25, 14), & & & & & &\\ %\cline{4-4}
                       & (1.25, 10) & & & & & &\\ \hline\hline

\multirow{4}{*}{GA1.25 1} & (1.25, 8.5), (1.25, 18.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{4}{*}{Simple} & \multirow{4}{*}{1153.62 (34.56)} & \multirow{4}{*}{+(0.00)}\\ %\cline{4-4}
                       & (1.25, 18.5), (1.25, 15.5) & & & & & & \\ %\cline{4-4}
                       & (1.25, 15.5), (1.25, 13.5) & & & & & &\\ %\cline{4-4}
                       & (1.25, 15), (1.25, 14) & & & & & &\\ \hline
\multirow{4}{*}{GA1.25 2} & (0.75, 1.5), (1.25, 12.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{4}{*}{Simple} & \multirow{4}{*}{1148.38 (34.84)} & \multirow{4}{*}{+(0.00)}\\ %\cline{4-4}
                       & (1, 19.5), (1.25, 15) & & & & & & \\ %\cline{4-4}
                       & (1.25, 14.5), (1.25, 13.5) & & & & & &\\ %\cline{4-4}
                       & (1, 13), (1, 14), (1.25, 13.5) & & & & & &\\ \hline
\multirow{4}{*}{GA1.25 3} & (1, 0), (1.25, 15.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{4}{*}{Simple} & \multirow{4}{*}{1147.08 (33.38)} & \multirow{4}{*}{+(0.00)}\\ %\cline{4-4}
                       & (1.25, 15), (1.25, 16) & & & & & & \\ %\cline{4-4}
                       & (1.25, 15), (1, 13.5) & & & & & &\\ %\cline{4-4}
                       & (1, 14), (1, 14), (1, 16.5) & & & & & &\\ \hline
    \end{tabular}
\end{table}

\begin{table}[htp!]
    \centering
    \caption{ Characterisation of benchmark protocol (BMII), best sub-protocols with the limit of 2.0~Gy dose found found using GA algorithm (GA2.0 1 - GA2.0 3), see Figure \ref{fig:benchmark_comaprison} B. The p-value indicates the probability that the GA algorithm results are from the same set, is compared to BMII fitness value. Similarly to Table~\ref{tab:ga_bmi_results}, the performance of GA algorithm protocol with 12~h time interval is beyond the BMII 1-BMII 3.
    }\label{tab:ga_bmii_results}
    \small
    \hspace*{-1cm}
    \begin{tabular}{|l|c|c|c|c|c|c|c|}
\hline
Id & Protocol schedule (Dose,   & Total &  Max & Min time & Selection & GPU mean & p-value\\
 & inter-fraction time-gap)  & dose  & dose & interval & type & fitness score & (compared\\
 & ([Gy], [h]) & [Gy] & [Gy] & [h] & (only for GA) & & to BMII)\\ \hline\hline
\multirow{3}{*}{BMII} & (2, 24), (2, 24), & \multirow{3}{*}{10} & \multirow{3}{*}{2$^*$} & \multirow{3}{*}{24$^*$} &  \multirow{3}{*}{-} & \multirow{3}{*}{1148.68 (37.05)} & \multirow{3}{*}{-}\\ %\cline{3-3}
                       & (2, 24), (2, 24), & & & & & &\\ %\cline{3-3}
                       & (2, 24) & & & & & &\\\hline
\multirow{2}{*}{BMII 1} & (2, 15), (2, 22), & \multirow{2}{*}{8} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top 5 \cite{Angus2014}} & \multirow{2}{*}{1158.66 (42.25)} & \multirow{2}{*}{-} \\ %\cline{2-2}
                       & (2, 15.5), (2, 23.5) & & & & & & \\\hline

\multirow{2}{*}{BMII 2} & (2, 15), (2, 19), & \multirow{2}{*}{8} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top 5 \cite{Angus2014}} & \multirow{2}{*}{1155.87 (43.92)} & \multirow{2}{*}{-}\\ %\cline{2-2}
                       & (2, 15.5), (2, 18) & & & & & &\\\hline

\multirow{2}{*}{BMII 3} & (2, 15), (2, 23), & \multirow{2}{*}{8} & \multirow{2}{*}{2$^*$} & \multirow{2}{*}{10} &  \multirow{2}{*}{Top 1 \cite{Angus2014}} & \multirow{2}{*}{1159.33 (41.82)} & \multirow{2}{*}{-}\\ %\cline{2-2}
                       & (2, 15.5), (2, 22.5) & & & & & & \\\hline\hline

\multirow{4}{*}{GA2.0 1}   & (0.25, 17.5), (1.75, 17),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Tournament} & \multirow{4}{*}{1173.96 (37.80)} & \multirow{4}{*}{+(0.00)} \\ %\cline{4-4}
                       & (1.5, 20.5), (1.5, 15) & & & & & & \\ %\cline{4-4}
                       & (1.75, 16), (1.5, 18.5) & & & & & & \\ %\cline{4-4}
                       & (1.75, 15) & & & & & &\\ \hline
\multirow{4}{*}{GA2.0 2}   & (0.5, 11), (1.5, 18),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Simple} & \multirow{4}{*}{1172.55 (38.71)} & \multirow{4}{*}{+(0.00)} \\ %\cline{4-4}
                       & (1.75, 20), (1.75, 16.5) & & & & & & \\ %\cline{4-4}
                       & (1.75, 16.5), (1.5, 16.5) & & & & & & \\ %\cline{4-4}
                       & (1.25, 16) & & & & & &\\ \hline
\multirow{4}{*}{GA2.0 3}   & (0.25, 2), (0.25, 15.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Roulette} & \multirow{4}{*}{1172.44 (36.74)} & \multirow{4}{*}{+(0.00)} \\ %\cline{4-4}
                       & (1.5, 15.5), (1.5, 19) & & & & & & \\ %\cline{4-4}
                       & (1.5, 15), (1.75, 16.5) & & & & & & \\ %\cline{4-4}
                       & (1.75, 16), (1.5, 18.5) & & & & & &\\ \hline
    \end{tabular}
\end{table}

% \begin{table}[hp!]
%     \centering
%     \caption{\wadd{TODO: To remove in favour of new table 3?}
%     Best GA results, with constraints of max single dose equals to 1.25 or 2.0, and min time interval between doses to 12 and 24h. The p-value (compared to BMII) indicates the probability that the GA algorithm results (GA1-GA4) are from the same set as the BMII results. The formula used for p-value calculation is available in~\cite{SciPyTTest}.}
%     \label{tab:best_ga_results}
%     \small
%     \hspace*{-1cm}
%     \begin{tabular}{|l|c|c|c|c|c|c|c|}
% \hline
% Id & Protocol schedule (Dose,   & Total &  Max & Min time & Selection & GPU mean & P-value\\
%  & inter-fraction time-gap)  & dose  & dose & interval & type & fitness score (sd) & (compared \\
%  & ([Gy], [h]) & [Gy] & [Gy] & [h] & & & to BMII)\\ \hline\hline
 
 
%  \multirow{4}{*}{GA 1} & (0.25, 17.5), (1.75, 17),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Tournament} & \multirow{4}{*}{1173.96 (37.80)} & \multirow{4}{*}{0} \\ %\cline{4-4}
%                       & (1.5, 20.5), (1.5, 15) & & & & & & \\ %\cline{4-4}
%                       & (1.75, 16), (1.5, 18.5) & & & & & & \\ %\cline{4-4}
%                       & (1.75, 15) & & & & & & \\ \hline
% \multirow{4}{*}{GA 2} & (0.5, 11), (1.5, 18), & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} &  \multirow{4}{*}{Simple} & \multirow{4}{*}{1172.55 (38.71)} & \multirow{4}{*}{0} \\ %\cline{4-4}
%                       & (1.75, 20), (1.75, 16.5), & & & & & &\\ %\cline{4-4}
%                       & (1.75, 16.5), (1.5, 16.5), & & & & & &\\ %\cline{4-4}
%                       & (1.25, 16) & & & & & &\\ \hline
% \multirow{4}{*}{GA 3} & (0.25, 2), (0.25, 15.5), & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} &  \multirow{4}{*}{Roulette} & \multirow{4}{*}{1172.44 (36.66)} & \multirow{4}{*}{0} \\ %\cline{4-4}
%                       & (1.5, 16.5), (1.5, 19), & & & & & & \\ %\cline{4-4}
%                       & (1.5, 15), (1.75, 16.5), & & & & & & \\ %\cline{4-4}
%                       & (1.65, 16), (1.5, 18.5) & & & & & & \\ \hline
% \multirow{4}{*}{GA 4} & (0.5, 17.5), (1, 17.5), & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} &  \multirow{4}{*}{Roulette} & \multirow{4}{*}{1171.72 (36.74)} & \multirow{4}{*}{0} \\ %\cline{4-4}
%                       & (1.75, 18.5), (1.5, 15.5), & & & & & & \\ %\cline{4-4}
%                       & (1.75, 15.5), (1.75, 16), & & & & & & \\ %\cline{4-4}
%                       & (1.75, 17) & & & & & &\\ \hline
% \multirow{4}{*}{GA 5} & (0.25, 2), (0.25, 16.5), & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} &  \multirow{4}{*}{Roulette} & \multirow{4}{*}{1170.8 (36.69)}  & \multirow{4}{*}{0} \\ %\cline{4-4}
%                       & (1.5, 15.5), (1.5, 19), & & & & & &\\ %\cline{4-4}
%                       & (1.5, 15), (1.75, 16.5), & & & & & &\\ %\cline{4-4}
%                       & (1.75, 16), (1.5, 18.5), & & & & & &\\\hline
% \multirow{4}{*}{GA 6} & (0.75, 20), (1.75, 18), & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} &  \multirow{4}{*}{Tournament} & \multirow{4}{*}{1170.37 (39.19)}  & \multirow{4}{*}{0} \\ %\cline{4-4}
%                       & (1.75, 21), (1.75, 14.5), & & & & & &\\ %\cline{4-4}
%                       & (1.5, 14.5), (1.75, 16.5), & & & & & &\\ %\cline{4-4}
%                       & (0.75, 18.5) & & & & & & \\\hline
% \multirow{4}{*}{GA 7} & (0.25, 17.5), (1.75, 20.5), & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} &  \multirow{4}{*}{Tournament} & \multirow{4}{*}{1170.37 (37.57)} & \multirow{4}{*}{0}  \\ %\cline{4-4}
%                       & (1.5, 17), (1.5, 15), & & & & & &\\ %\cline{4-4}
%                       & (1.75, 16), (1.5, 18.5), & & & & & &\\ %\cline{4-4}
%                       & (1.75, 15), & & & & & &\\\hline
% \multirow{4}{*}{GA 8} & (0.5, 17.5), (1.25, 18.5), & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} &  \multirow{4}{*}{Roulette} & \multirow{4}{*}{1170.11 (38.62)} & \multirow{4}{*}{0}  \\ %\cline{4-4}
%                       & (1.5, 17.5), (1.5, 15.5), & & & & & &\\ %\cline{4-4}
%                       & (1.75, 15.5), (1.75, 16), & & & & & &\\ %\cline{4-4}
%                       & (1.75, 17) & & & & & &\\\hline
 
% \multirow{4}{*}{GA 9} & (0.25, 2), (0.25, 16.5),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Roulette} & \multirow{4}{*}{1169.87 (37.18)} & \multirow{4}{*}{0}  \\ %\cline{4-4}
%                       & (1.5, 15.5), (1.5, 17.5) & & & & & &\\ %\cline{4-4}
%                       & (1.5, 16.5), (1.75, 16.5) & & & & & &\\ %\cline{4-4}
%                       & (1.75, 16), (1.5, 18.5) & & & & & &\\ \hline
% \multirow{4}{*}{GA 10} & (0.75, 20), (1.75, 18),  & \multirow{4}{*}{10} & \multirow{4}{*}{2} & \multirow{4}{*}{12} & \multirow{4}{*}{Tournament} & \multirow{4}{*}{1169.06 (38.58)} & \multirow{4}{*}{0}  \\ %\cline{4-4}
%                       & (1.75, 21), (1.75, 14.5) & & & & & &\\ %\cline{4-4}
%                       & (1.5, 18), (1.75, 13) & & & & & &\\ %\cline{4-4}
%                       & (0.75, 15) & & & & & &\\ \hline
%     \end{tabular}
    

%     % \caption{Best protocols found using GA algorithm. For comparison purposes with protocols in \ref{tab:bench}, presented protocols have constraints of minimum 12 and 24~hours between doses and 1.25 and 2.0~Gy maximum single dose. Fitness scores are calculated based on fitness function~\eqref{eq:final_fit_fun} defined in Section~\ref{sec:GPU:implem} evaluated 100 times for each of 10 tumours from study library, for details see Section~\ref{sec:lib}}.
%     %\mon{consider to change it -- shall we include CPU, GPU results?}
% \end{table}


\begin{table}[htp!]
    \centering
    \caption{
    Best GA results, with constraints of max single dose between 2.5 and 4.5~Gy (see Figure~\ref{fig:benchmark_comaprison} C, and min time interval between doses to 12 and 24h. The p-value is calculated against BMII and indicates the probability that the GA algorithm results (GA2.5 1 - GA4.5 2) are from the same set as the BMII results.}
    \label{tab:best_ga_larger_dose_results}
    \small
    \hspace*{-1cm}
    \begin{tabular}{|l|c|c|c|c|c|c|c|}
\hline
Id & Protocol schedule (Dose,   & Total &  Max & Min time & Selection & GPU mean & P-value\\
 & inter-fraction time-gap)  & dose  & dose & interval & type & fitness score & (compared \\
 & ([Gy], [h]) & [Gy] & [Gy] & [h] & & & to BMII)\\ \hline\hline
 
 
 \multirow{3}{*}{GA2.5 1} & (0.25, 25.5), (1.75, 26.5),  & \multirow{3}{*}{10} & \multirow{3}{*}{2.5} & \multirow{3}{*}{12} & \multirow{3}{*}{Roulette} & \multirow{3}{*}{1193.99 (42.37)} & \multirow{3}{*}{+(0.00)} \\ %\cline{3-3}
                       & (2, 16), (2, 19) & & & & & & \\ %\cline{3-3}
                       & (2, 16.5), (2, 15.5) & & & & & & \\ \hline
 \multirow{3}{*}{GA2.5 2} & (2, 44.5), (2, 24.0),  & \multirow{3}{*}{10} & \multirow{3}{*}{2.5} & \multirow{3}{*}{12} & \multirow{3}{*}{Tournament} & \multirow{3}{*}{1192.59 (45.81)} & \multirow{3}{*}{+(0.00)} \\ %\cline{3-3}
                       & (2.25, 17.5), (2, 14.5) & & & & & & \\ %\cline{3-3}
                       & (1.75, 14) & & & & & & \\ \hline\hline
\multirow{3}{*}{GA3.0 1} & (0.25, 11.5), (3, 42), & \multirow{3}{*}{10} & \multirow{3}{*}{3} & \multirow{3}{*}{12} &  \multirow{3}{*}{Roulette} & \multirow{3}{*}{1213.25 (50.57)} & \multirow{3}{*}{+(0.00)} \\ %\cline{3-3}
                       & (2.5, 23.5), (2.25, 17), & & & & & &\\ %\cline{3-3}
                       & (2, 18.5) & & & & & &\\ \hline
\multirow{3}{*}{GA3.0 2} & (0.5, 22), (0.5, 14.5), & \multirow{3}{*}{10} & \multirow{3}{*}{3} & \multirow{3}{*}{12} &  \multirow{3}{*}{Tournament} & \multirow{3}{*}{1212.63 (50.92)} & \multirow{3}{*}{+(0.00)} \\ %\cline{3-3}
                       & (0.25, 17.5), (3, 16), & & & & & &\\ %\cline{3-3}
                       & (2.75, 23), (3, 20.5) & & & & & &\\ \hline\hline
\multirow{2}{*}{GA3.5 1} & (2.5, 60.5), (2.5, 21), & \multirow{2}{*}{10} & \multirow{2}{*}{3.5} & \multirow{2}{*}{12} &  \multirow{2}{*}{Tournament} & \multirow{2}{*}{1232.35 (60.85)} & \multirow{2}{*}{+(0.00)} \\ %\cline{2-2}
                       & (2.5, 17.5), (2.5, 16.5) & & & & & &\\ \hline
\multirow{3}{*}{GA3.5 2} & (0.25, 2.5), (0.75, 57), & \multirow{3}{*}{10} & \multirow{3}{*}{3.5} & \multirow{3}{*}{12} &  \multirow{3}{*}{Simple} & \multirow{3}{*}{1231.71 (52.88)} & \multirow{3}{*}{+(0.00)} \\ %\cline{3-3}
                       & (3, 13.5), (2.75, 26.5), & & & & & &\\ %\cline{3-3}
                       & (3.25, 19.5) & & & & & &\\ \hline\hline
\multirow{2}{*}{GA4.0 1} & (3.75, 75.5), (3.75, 24.5), & \multirow{2}{*}{10} & \multirow{2}{*}{4} & \multirow{2}{*}{12} &  \multirow{2}{*}{Simple} & \multirow{2}{*}{1273.7 (64.75)} & \multirow{2}{*}{+(0.00)} \\ %\cline{2-2}
                       & (2.5, 19) & & & & & &\\ \hline
\multirow{2}{*}{GA4.0 2} & (3.25, 60), (3.5, 24), & \multirow{2}{*}{10} & \multirow{2}{*}{4} & \multirow{2}{*}{12} &  \multirow{2}{*}{Simple} & \multirow{2}{*}{1248.25 (67.53)} & \multirow{2}{*}{+(0.00)} \\ %\cline{2-2}
                       & (3.25, 19.5) & & & & & &\\ \hline\hline
\multirow{2}{*}{GA4.5 1} & (0.25, 53.5), (3.5, 22.5), & \multirow{2}{*}{10} & \multirow{2}{*}{4.5} & \multirow{2}{*}{12} &  \multirow{2}{*}{Tournament} & \multirow{2}{*}{1264.54 (63.76)}  & \multirow{2}{*}{+(0.00)} \\ %\cline{2-2}
                       & (2.0, 24.5), (4.25, 15) & & & & & &\\\hline
\multirow{2}{*}{GA4.5 2} & (0.25, 3.5), (3.75, 69), & \multirow{2}{*}{10} & \multirow{2}{*}{4.5} & \multirow{2}{*}{12} &  \multirow{2}{*}{Simple} & \multirow{2}{*}{1264.07 (61.92)}  & \multirow{2}{*}{+(0.00)} \\ %\cline{2-2}
                       & (4.25, 26), (1.75, 20) & & & & & &\\\hline
    \end{tabular}
\end{table}
%-------------------------------------------------------------------------

%-------------------------------------------------------------------------
\subsection{GA results} general idea of GA approach, {\bf fig.2} representing the scheme (like in Plos One paper), estimation of the size of the possible protocol space(?) , GA performance with constraints (e.g. total dose < 10 Gy; fractional dose < Y Gy; + no more than X/fractions per day), presentation of the best protocols ({\bf table 1 continuation or table 2}), information about the convergence ({\bf supp fig.S1}) + {\bf fig.3 combined} presenting results

\begin{figure}
    \centering
    \includegraphics[scale=0.65]{figs/benchmark_comparison_top10.pdf}
    \caption{Comparison of the evaluation (GPU implementation) of clinical protocols (clinical), previously derived suboptimal protocols (with fixed single irradiation doses) reported in~\cite{Angus2014} (state of art) and results from this study, when we allow to vary the time and dose of each single administration of irradiation (this paper). In the case of the new results indicated values denote the maximal acceptable of fractional doses. (A)..(B).. and (C).. 
    \label{fig:benchmark_comaprison}}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[scale=0.7]{figs/density_composite.pdf}
    \caption{Histograms indicating the most often appearing the 10 best protocols (among all GA experiments, compare~Table~\ref{tab:best_protocols}) combinations of delay (time to the next administration) and fractional dose. (A)..(B).. (C) and (D)..  
    \label{fig:density_composite}}
\end{figure}

\begin{table}[htp!]
    \centering
\caption{Evaluations of hand-crafted (HC) protocols. Protocols were inspired by the results presented in~Figure~\ref{fig:density_composite}. Protocol \textit{HC 1} consists of doses with the fixed values of 1.4~Gy and inter-fraction time-gaps of 15.5~h, obtained as the densest point of histogram. Protocols \textit{HC 2} and \textit{HC 3} are the variations of the \textit{HC 1} with altered first and last doses, to match the second significant value from the histogram. The last \textit{HC 4} protocol is the dummy configuration of 7 equal dose values and time-gaps to fit into constraint of 120~h and 10~Gy per protocol.}
    \label{tab:manual_results}
    \small
    \hspace*{-1cm}
    \begin{tabular}{|l|c|c|c|c|}
\hline
Id & Protocol schedule (Dose, & Total dose [Gy] &  GPU mean & p-value\\
 & inter-fraction time-gap) ([Gy], [h]) & & fitness score (sd) & (compared to BMII) \\\hline
 \multirow{3}{*}{Manual 1} & (1.4, 15.5), (1.4, 15.5), (1.4, 15.5),  & \multirow{3}{*}{9.8} & \multirow{3}{*}{1147.05 (36.31)} & \multirow{3}{*}{0.32} \\ %\cline{3-}
                       & (1.4, 15.5), (1.4, 15.5), (1.4, 15.5), & & & \\ %\cline{3-3}
                       &  (1.4, 15.5) & & & \\ \hline
 \multirow{3}{*}{Manual 2} & (0.7, 0.5), (1.4, 15.5), (1.4, 15.5),  & \multirow{3}{*}{9.8} & \multirow{3}{*}{1123.14 (37.37)} & \multirow{3}{*}{0} \\ %\cline{3-}
                       & (1.4, 15.5), (1.4, 15.5), (1.4, 15.5), & & & \\ %\cline{3-3}
                       &  (1.4, 15.5) & & & \\ \hline
 \multirow{3}{*}{Manual 3} & (0.7, 0.5), (1.4, 15.5), (1.4, 15.5),  & \multirow{3}{*}{9.8} & \multirow{3}{*}{1148.79 (38.29)} & \multirow{3}{*}{0.95} \\ %\cline{3-}
                       & (1.4, 15.5), (1.4, 15.5), (1.4, 15.5), & & & \\ %\cline{3-3}
                       &  (1.4, 15.5), (0.7, 0.5) & & & \\ \hline
 \multirow{3}{*}{Manual 4} & (1.42, 17.1), (1.42, 17.1), (1.42, 17.1),  & \multirow{3}{*}{9.94} & \multirow{3}{*}{1157.83 (35.96)} & \multirow{3}{*}{0} \\ %\cline{3-}
                       & (1.42, 17.1), (1.42, 17.1), (1.42, 17.1), & & & \\ %\cline{3-3}
                       &  (1.42, 17.1) & & & \\ \hline
    \end{tabular}
\end{table}

%-------------------------------------------------------------------------
%\subsection{Evaluation (Robustness?) of the best protocols} robustness, skipped dose study, comparison for the best GA, benchmarks and best Plos One paper protocols ({\bf combined fig.4})

%-------------------------------------------------------------------------

\section{Discussion}

\mon{Include a Discussion that summarizes (but does not merely repeat) your conclusions and elaborates on their implications. There should be a paragraph outlining the limitations of your results and interpretation, as well as a discussion of the steps that need to be taken for the findings to be applied. Please avoid claims of priority.}

%-------------------------------------------------------------------------
\section{Materials and Methods}

\subsection{Tumour growth -- multi-fraction irradiation model}


In this work, we extend our high-fidelity numerical 
simulation approach to examine yet untried protocols within the vast space 
of available protocols to find candidate protocols that offer statistically significant improvements over benchmark protocols presently used in 
radiotherapy clinics. Our simulation model described in~\cite{Angus2014} takes into account several biologically important components of tumour growth and cellular response to irradiation such as: the diffusion of nutrients (oxygen, 
glucose) and metabolic waists; cell cycles with distinguishable cell phases; 
differentiated cellular metabolism (aerobic or anaerobic, proliferating or 
quiescent) which is responsive to the cell's environmental nutrient 
concentrations and local pH. The model was developed to replicate the 
biological dynamics of a~specific mouse cancer cell line (EMT6/Ro), a~widely 
used analogue of human breast cancer, and, owing to the abundant experimental 
studies on this cell line. Moreover, our model has been calibrated at each step of 
development, including its response to 18 independent multi-fraction 
irradiation protocols tested in laboratory. The calibration and validation of 
the model's performance has been accomplished with reference to a~broad 
spectrum of tumour characteristics including: number of cells; saturation size; 
tumour volume; tumour doubling time; thickens of the proliferating rim; cell 
phase population fraction; onset and progression of necrosis; and effective 
dose induced by multi-fraction irradiation. For details, see~\cite{Angus2014} 
and Supplementary Information within.

%\subsection{Benchmark protocols}
%-------------------------------------------------------------------------
\subsection{In silico tumour library characteristics}\label{sec:lib}

Due to the stochastic nature of the tumour growth and radiation response model to discover best performing sub-optimal protocols, we follow the approach presented in~\cite{Angus2014} and perform evaluation of all considered protocols across the case library. Library consists of a ten 10-day old tumours developed in-silico as described in~\cite{Angus:2013id} from an initial seed population of 200 cancer cells (10 cellular automata lattice grid sites) placed in a well-mixed replenished substrate (glucose concentration: 5.5 mM, oxygen concentration: 0.28 mM, with pH level maintained at 7.4). Tumours have been grown for 10 days without any radiation interference. For the full characterisation of the developed tumour we refer reader to~\cite[Supp. Inf., Table ~S2]{Angus2014}. The state of each tumour was saved, and stored in case library for the future use.


\subsection{Infrastructure used in experiments}
All the experiments with programs prepared in C++ were run using a computational infrastructure provided by the Interdisciplinary Centre for Mathematical and Computational Modelling of the University of Warsaw\cite{ICM}. Experiments requiring CPUs were run on $36$ cores of a Rysy cluster (at the Interdisciplinary Centre for Mathematical Modelling \cite{ICM}), with Intel(R) Xeon(R) Gold 6154/6252 CPU processors (3.7 GHz) and 380 GB RAM. Experiments requiring GPUs were run on 4 GPUs of graphical processing units NVIDIA V100 on the same cluster - Rysy.

\subsection{Advantage of GPU implementation} 
\mon{about execution time comparison of CPU and GPU models -- single experiment is faster over 4\,000 times,  fidelity of GPU implementation (comparing to Matlab CA), {\bf combined fig.1} (Rafals plots with the comparison of Matlab CA and GPU for variety of protocols, 1-2 panels), normality of the distribution + KS test results (1 panel)}


We developed a new implementation (in C++) of the original EMT6/Ro Matlab simulation,~\cite{Angus2014}, utilizing general-purpose computing on GPUs to accelerate the GA search. We achieved an average speed of 8\,0000 simulations per hour using 4 NVIDIA V100 GPUs compared to about 110 simulations per hour using 13 10-core CPUs reported in~\cite[Supp. Inf., Numerical Implementation: Details]{Angus2014}.

To ensure fidelity of the new GPU implementation, we used an intermediate version of code written in standard C++, which accurately copied the behaviour of the original MATLAB simulation but was fast enough to provide a convenient baseline to compare the GPU implementation with~\cite{SourceCPU}. 
A series of randomly generated protocols was evaluated with both CPU and GPU simulations and the results were statistically compared. For details, see ~Fig.~\ref{fig:CPUvsGPU_normal} \madd{(A) together with ~Fig.~\ref{fig:CPUvsGPU_normal} (B) where we report p-values of the two-tailed t-test comparing corresponding GPU and CPU distributions.} 

Using the fast GPU implementation of the model, we were able to test the normality of the distribution of simulation results. We chose the final number of occupied grid sites as a representative measure because it is used to calculate the GA fitness function. To check if a data set can be well described by a normal distribution, we randomly selected a single irradiation protocol and single tumour, ran 2\,000 evaluation of that protocol on selected tumour and plotted a histogram of the results (see Fig.~\ref{fig:CPUvsGPU_normal} \madd{(D)}), which together with the Kolmogorov-Smirnov test (p-value = 0.38) indicated that we cannot reject the hypothesis that the distribution is normal. \madd{In Fig.~\ref{fig:CPUvsGPU_normal} (C) we present p-values of Kolmogorov-Smirnov test for all randomly generated protocols.}


\begin{figure}[ht]
    \centering
    \includegraphics[scale=0.365]{figs/CPU_GPU_comparison_v2}
    \caption{(A) Comparison of relative deviations from the average number of occupied (life) sites for CPU (blue) and GPU implementation (orange) for 60 randomly generated protocols under assumption that sum of all doses is less or equal to 10 Gy. For each protocol, for each tumour (10 in total) and for each implementation, 100 (GPU) and 48 (CPU) simulations were performed. The bars represent a 95\% confidence interval of the mean result. \madd{(B) P-value of two-sided t-test comparison between GPU and CPU evaluation of protocols considered in (A), grey colour indicates protocols for which p<0.1, light grey colour indicates protocols for which p<0.05. Black and red lines indicate 0.1 and 0.05 significance levels, respectively. (C)  P-value of  Kolmogorov-Smirnov test (KS-test) for protocols considered in (A), grey colour indicates protocols for which p<0.1. Black and red lines indicate 0.1 and 0.05 significance levels, respectively.. (D)} Histogram of 2\,000  \madd{GPU evaluations of a single randomly selected protocol over} a single \madd{randomly selected tumour (final number of occupied sites) } compared to the normal distribution with the same mean and standard deviation.}
    \label{fig:CPUvsGPU_normal}
\end{figure}

%    Normality of the distribution, see Fig.~\ref{fig:normal_dist}.
%    \begin{figure}
%        \centering
%        \includegraphics{figs/normal_dist.png}
%        \includegraphics{figs/test result.PNG}
%        \caption{Result of 2\,000 simulations performed using GPU code for a single tumour and single protocol. Two-sided Kolmogorov-Smirnov test confirms that distribution is normal (p-value>0.3826).}
%        \label{fig:normal_dist}
 %   \end{figure}


%-------------------------------------------------------------------------



\subsection{GPU model implementation}\label{sec:GPU:implem}

To employ general purpose GPU (GPGPU) programming for increasing the performance of the simulation we chose C++ language and the CUDA programming model to rewrite the existing MATLAB implementation. Newly developed code was published under MIT License~\cite{SourceCUDA}. 
Usage of GPUs was a promising strategy to mightily improve the performance of the algorithms due to the parallelizable nature of cellular automata computations and the fact that evaluation of a radiation protocol requires multiple independent simulation runs.
The architectural idea of the new implementation was to use massive parallelism of the GPUs not only to parallelize the processing of single cells within a simulation but also to concurrently run a batch of simulations. This allowed gaining significant performance improvement over CPU simulations regarding the mean time of multiple protocol evaluations even though the speed of running a single simulation is lower on a GPU. 



%-------------------------------------------------------------------------
\subsection{Genetic algorithm approach}
{\bf to put here: definition of the fitness function, details regarding implemented operators ect., 
experimental setup, including initial population description}

In order to find heuristically optimal protocols, we use genetic algorithms (GAs) exploring the vast space of possible protocols. Tumour growth simulation model based on cellular automata (CA) is used to compute values of a fitness function for GAs. GPU implementation of CA allowed to significantly accelerate computations and perform hundreds of different experiments within reasonable time horizon. 

In the genetic algorithm approach, the protocols are encoded as genotypes represented as vectors. Then, the quality of genotypes is evaluated by a fitness function for which the values are calculated using computer simulations (see formula \ref{eq:final_fit_fun}). Then, best genotypes are selected using a selection operator, they are later used to create new genotypes using a crossover operator. The newly produced genotypes are then perturbed according to a mutation operator. This procedure is schematically presented in Figure~\ref{fig:GA_scheme}. The next subsections describe in detail the fitness function and operators that were used in our experiments.

\subsubsection{Fitness function}
\label{fitness_function}

One of the crucial elements of our experiments is to determine the value of a fitness function $f$, required for GAs as the fitness indicator. We use
\begin{equation}\label{eq:final_fit_fun}
   f(p_{i,j}) = 1\,500 - n_{i,j}, 
\end{equation}
where $p_{i,j}$ denotes considered protocol $i$ evaluated on a single tumour $j$, 1\,500 is the maximal number of possibly occupied sites, $n_{i,j}$ denotes the number of the occupied sites after testing protocol $i$ on tumour $j$. In the performed experiments, we pass the treatment protocol into the CA model and obtain fitness score. In order to assess robust results, we evaluate each protocol on 10 different tumours using 2 GPUs, 4 times per GPUs per tumour. This allows us to calculate 80 samples for every single treatment protocol. Thus, the fitness for the particular tested protocol is the mean from the fitnesses of all its evaluations.

\subsubsection{Setup of experiments}

A~single experiment starts with initial steps -- random protocols initialization and initial \textit{fitness} value calculation. Next, we perform the specified number of iterations through GA, each consisting of the creation of a new generation of protocols using \textit{selection} and \textit{crossover operators}, application of \textit{mutation operators} and  calculation of \textit{fitness} function value. Finally, all output data from the whole experiments are saved.

To create the first generation, we generated a list of 40 protocols to be tested and each of the protocols fulfills a set of constraints.

The CA model is designed with a resolution such that each hour between doses equals to 600 units of time. For our treatment protocols, we used a minimal step equal to 300 units, establishing the minimal time span between two doses in a protocol equals half of hour. The full treatment time is set to 5~days (120~hours), which entailed each protocol got 240~time-steps. The reason behind these settings was the limitation of the searching space for the time of dose. To further restrict the searching space for protocols, the number of possible options for dose values, we established some additional constraints.
First, we follow our previous work \cite{Angus2014} and assume the sum of all dose values for a~single protocol should not be higher than 10~Gy to ensure that total delivered dose, affecting both the tumour and the healthy tissues, do not exceed a standard clinical practice for low-dose, multi-fraction, irradiation schemes \cite{Rosenstein:2004vs, ORourke:2009, BoardoftheFacultyofClinicalOncology:2006wq}. Moreover, the value of an acceptable single dose should be at a minimum 0.25~Gy and could be increased by 0.25~Gy per step, up to the maximal single dose, between 2.0~Gy and 4.5~Gy, depending on a particular experimental setting.
In addition, to simulate realistic treatment protocols with doses given to the patient according to the scheduled intervals, we restricted the minimal time interval between doses to vary from 6 to 48 hours.

The above rules for creating new protocols hold for all experiments, however, for all generations except the first one, we additionally transferred 10\% of the best protocols from the previous generation.

After initializing the first generation, newly created protocols are passed into the CA model to compute a value of the fitness function. At this point, it is important to describe the representation of the protocols. For the genetic algorithm, each protocol is represented by a sparse vector of 240 dose values, where each position corresponds to a specific point in time. Zeroed values in this representation indicates lack of dose at a corresponding time.
\mon{remark: for the original CA model we had pairs (dose, time to the second dose), now we have (dose, time of application)}
For the CA model, a protocol is represented as a key-value pair of the time-step and the value of the dose given at this time-step. This is a difference comparing to the original CA model where time-step value indicated time between previous and the current dose, not the absolute time of application 
The reason behind using two different representations is that the CA model was created first with the key-value pair representation. Now, during development of the GA model, performing various types of selection, crossover, and mutation operations might be non-trivial for such a representation. Thus, the new vector representation was introduced, which is more similar to standard candidates representation as an array of bits. Because conversion between two representations is very fast and computationally inexpensive, the CA model remained untouched and we convert protocols in each experiment.

\subsubsection{\madd{Genetic algorithm evolution}}
The main loop of the genetic algorithm is initiated by performing \textit{selection} operations on 40 protocols evaluated according to the \textit{fitness} function (40 is the number of protocols evaluated in each generation). First, we retain 10\% of the best protocols (according to their fitness score) and do not apply crossover and mutation operations. Next, we select 18 of the best treatment protocols as parents and apply \textit{crossover} operator, obtaining the remaining 18 protocols. Then, we execute a set of \textit{mutation} operators on protocols and calculate the \textit{fitness} function value for the obtained generation.
After a prearranged number of iterations, the experiment is finished.

\begin{figure}[ht]
    \centering
    \includegraphics[scale=0.37]{figs/GA_scheme.pdf}
    \caption{GA scheme.}
    \label{fig:GA_scheme}
\end{figure}

\subsubsection{Selection}
Selection procedure is an important step for GA to obtain the convergence and desired results. During the process, a portion of the current population is selected and used as a parental population for a new generation. It ensures diversity of protocols, enhances exploration of the search space, helps in preventing overfitting to relatively good protocols, and leads to higher possibility for finding better results.

To empower results, we use 3~different selection algorithms, commonly applied in GA problems: \textit{simple selection}, \textit{roulette wheel selection} and \textit{tournament selection}, see Fig.~\ref{fig:GA_scheme} for details. 

The first method, \textit{simple selection}, sorts protocols by associated fitness values in descending order and selects half of them. This algorithm is very simple, however it might limit the searching space for new possible treatment protocols and result in being stuck in local minima. 

To prevent such situation, we use two alternative algorithms, which might mitigate the aforementioned risk. In the \textit{roulette wheel selection} algorithm, called also fitness proportionate selection, we associate fitness values with the probability of selection. The probability $p_i$ for each protocol is calculated as 
\[
p_i = \frac{f_i}{\sum^N_{j=1}f_j},
\]
where $f_i$ denotes a fitness value for $i$-th protocol and $N$ is the number of treatment protocols in the considered generation ($40$). Next, we calculate the cumulative sum of those probabilities, summing them all to 1. To illustrate \textit{roulette wheel selection}, we can use a roulette wheel, where the proportions of the wheel segments are represented by probability $p_i$. We order the segments according to their order on the wheel and for each segment, we calculate sum of probabilities of all previous segments. For choosing a single treatment protocol, we take a random value $q$ between 0 and 1 and select a protocol associated with the first segment for which the partial sum of probabilities is higher or equal to $q$. We run the \textit{roulette wheel selection} algorithm in the loop to obtain half (18) of the treatment protocols. This method promotes protocols with higher probabilities, however, it allows to select weaker protocols with a small chance. This non-zero chance for weak protocols is an advantage over the simple selection which may help in escaping from the local minima.

For a similar purpose, we use the third selection algorithm, the \textit{tournament selection}, where we randomly select $k$~protocols and run tournaments among them. The probability of each protocol is associated with their fitness value. To obtain the best protocols from a tournament, we test the following probabilities:

\begin{itemize}
    \item probability of the best protocol: $p_1 = q$
    \item probability of the second best protocol: $p_2 = q*(1-q)$
    \item probability of the $k$-th best protocol: $p_i = q*(1-q)^{(k-1)}$
\end{itemize}

We repeat selection of $k$~protocols and tournaments $20$ times to obtain half of the desired number of protocols for crossover. Similarly to the \textit{roulette wheel selection}, this algorithm also allows weaker  protocols to be selected. Comparing with the roulette selection, we can adjust the probability of selection by modifying the value of the parameter $q$.

For all three selection operators, we retain 10\% of the best protocols (according to their fitness score) using the simple selection. These retained protocols are not taken into account in crossover and mutation operations to prevent the loss of the best protocols while GA is being proceeded.

\subsubsection{Crossover}
\label{crossover}
The goal of the crossover operation is to produce the new generation from the selected parents. In our experiments, the newly generated child inherits genes (doses at time-steps) from two randomly taken parents. This operation is repeated until the size of new population is appropriate. We introduce three different types of crossover algorithms: \textit{single-point crossover}, \textit{two-point crossover} and \textit{uniform crossover}, see Fig.~\ref{fig:GA_scheme}. 

For crossover operator, we benefit from storing protocols as a sparse vectors with zeroes corresponding to the lack of dose and non-zeroes to a dose given at the particular time-step. This makes algorithms more conceptually straightforward and very similar to those presented in the literature.

In the \textit{single-point crossover}, we randomly choose two parent protocols and select the same crossover point for both of them. Next, we swap the parts of protocols after the crossing points and obtain two children protocols. Crossing point is selected randomly, although with restricted range of choice between 25 and 75~percent of the length of a protocol. This limitation is important in the case of sparse representation, because in a situation when only a few first or last positions are swapped, the protocols will remain unchanged.

Second algorithm, \textit{two-point crossover}, is a variation of the aforementioned operator. Instead of selecting a single crossing point, we randomly selects two points. The first and the last parts of parents protocols remains unchanged in children protocols, and only the middle section is swapped. In this case, we also introduce limitations for the location of crossover points positions. The range for the first point is set to between $25\%$ and $50\%$ of the length of a protocol, while for the second: between $50\%$ and $75\%$.

The last introduced crossover operation is the so-called \textit{uniform crossover}, which also takes two parent protocols and returns two children protocols. However, the child protocol is created by selecting single positions, one-by-one, from both parent protocols with equal probability. If one child receives a value from the first parent, then the other child receives a value from the second parent on this position.

Protocols generated by \textit{crossover operator} might initially not meet the constraints introduced regarding the minimal time interval between doses and the maximum sum of all dose values. To prevent the former issue, we designed a \textit{guard algorithm} that recognises doses that violate a constraint and tries to assign them at another time-steps. If all the non-zero doses are too close to each other to allocate a dose between them, the algorithm does not allocate this particular dose. The latter problem with exceeding the limit of 10~Gy by the sum of all doses was resolved by another algorithm. If the formula detects the larger sum, the algorithm starts subtracting the value of 0.25~Gy from the consecutive largest doses in a loop, until the sum of doses for a protocol is acceptable.

\subsubsection{Mutation}
Mutation is an essential genetic operator to sustain diversity of a population and to avoid a stuck of the GA in local minima. Mutation operators are applied with a low probability to reduce the possibility of random search and loss of good protocols. On the other hand, usage of too low mutation probability causes stuck of GA in local minima and lower exploratory abilities of GA. In the presented study, we implement four mutation operators: \textit{swap mutation}, \textit{split mutation}, \textit{dose time mutation} and \textit{dose value mutation}. For simplicity, in the case of mutation operators we use sparse vector representation of protocols, similarly as in the case of crossover operators.

\textit{Swap mutation} operator swaps two doses. We start by iterating over protocols and selecting (with a probability $p$) a random position on the chromosome to swap. Such operation might break the constraints regarding minimal time interval between two doses and thus we apply the same \textit{guard algorithm} as described in Sec.~\ref{crossover} to overcome that issue.

\textit{Split mutation} operator splits a larger dose into two smaller ones. With a probability of $p$, it randomly selects a single, non-zero dose from a protocol. If the dose is larger than the minimal value of 0.25~Gy, it is split into two smaller doses. Next, the algorithm removes the split dose and allocates the smaller doses at randomly selected positions. It does not replace doses but appends them to existing values, only if the new, combined dose, is not larger than a maximal allowed single dose value (chosen for the particular execution of GA). If the combined dose is larger than maximal allowed single dose, algorithm skips the operation and tries to find a better position for a dose. Similarly as in the case of the swap mutation, we used the \textit{guard algorithm} to meet the constraints of both minimal time interval between doses and the maximal value of a single dose.

The third operator, \textit{dose time mutation}, alters the position of a dose in a protocol. With a probability of $p$, it randomly selects the dose at a source position from a protocol and the new destination position for it. Next, we take the value of the source dose and appends it to the dose at the destination position. The dose at the source position is removed (zeroed). To meet the constraints, we clamp the dose at the destination position to maximal allowed single dose value. The remained dose from the destination position which is above the threshold, is allocated back at the source position. Using this algorithm, we avoid losing dose values from a protocol keeping the sum of doses for the protocol at the same level.

The last introduced operator, \textit{dose value mutation}, changes the value of selected dose. With a probability $p$ we allocate a new dose to the protocol. First, we calculate the difference between a sum of dose values and the maximum allowed sum of doses (10~Gy). If it is smaller or equal to zero we do nothing, otherwise, we randomly select an existing dose in the protocol and replace it with a larger dose, which does not break the maximum allowed sum limit. However, this might break a time interval constraints, so again we use the \textit{guard algorithm}. The benefit of this mutation operation is the increase of the sum of doses for protocol, alleviating the possible loss of dose values from split and swap mutations.

% Don't know where to place the section below
% mjp: it is a good place 

\subsection{GA metrics}
For every experiment we collect data to further evaluate the performance of protocols. For each genetic algorithm loop iteration we store the best fitness function value and associate the best protocol. Next, we use it to analyse and compare the best protocols from different iterations and experiments. In order to check whether our GA model does not overfit, for each iteration we store the average value of a fitness function for all protocols.

\subsection{\wadd{GA experiments} [WO + PG]}
Due to the limited resources and relatively long computation time, we organised the experiments in batches. Each batch was focusing on particular assumptions and hypotheses from previous runs, so results of experiments from previous batches helped us in making decisions on how to configure the next series of experiments. 

We ran $8$ series of experiments, their settings are described in Table \ref{tab:exp_settings}. However, in this section, we describe results obtained only in the last batch among all the performed experiments. The detailed analysis of each batch, with upgrades developed during research and comparison of the results is summarised in Appendix \ref{appendix_a}. In addition, prior to these batches, we tested various settings and configurations of GA algorithms. However, we decided to not include them in this article since the fitness function formula was also adjusted from experiment to experiment and thus the results are hard to compare. 


% \begin{table}[h!]
%     \centering
%     \begin{tabular}{|l|c|c|c|c|c|c|}
% \hline
% Id & Protocol schedule (Dose (Gy),    & Total. &  Max & Min time & Selection & GPU  \\
%  & inter-fraction time-gap (h))  & dose (Gy)  & dose (Gy) & interval (h) & type &  fitness score \\ \hline\hline
% \multirow{4}{*}{1}   & (1.25, 9.5), (1.25, 17.5),  & \multirow{4}{*}{10.0} & \multirow{4}{*}{1.25} & \multirow{4}{*}{12} & \multirow{2}{*}{Simple} & \multirow{4}{*}{1162.5375} \\ %\cline{4-4}
%                       & (1.25, 18.5), (1.25, 15.5) & & & & \multirow{2}{*}{selection} & \\ %\cline{4-4}
%                       & (1.25, 15.5), (1.25, 13.5) & & & & & \\ %\cline{4-4}
%                       & (1.25, 15), (1.25, 14) & & & & & \\ \hline
% \multirow{4}{*}{2}   & (0.25, 17.5), (1.75, 17.0),  & \multirow{4}{*}{10.0} & \multirow{4}{*}{2.0} & \multirow{4}{*}{12} & \multirow{2}{*}{Tournament} & \multirow{4}{*}{1182.3125} \\ %\cline{4-4}
%                       & (1.5, 20.5), (1.5, 15.0) & & & & \multirow{2}{*}{selection} & \\ %\cline{4-4}
%                       & (1.75, 16.0), (1.5, 18.5) & & & & & \\ %\cline{4-4}
%                       & (1.75, 15.0) & & & & & \\ \hline
% \multirow{4}{*}{3}   & (1.25, 4.5), (1.25, 26.0),  & \multirow{4}{*}{6.25} & \multirow{4}{*}{1.25} & \multirow{4}{*}{24} & \multirow{2}{*}{Tournament} & \multirow{4}{*}{1046.15} \\ %\cline{4-4}
%                       & (1.25, 26.0), (1.25, 26.5), & & & & \multirow{2}{*}{selection} & \\ %\cline{4-4}
%                       & (1.25, 30.5) & & & & & \\ %\cline{4-4}
%                       & & & & & & \\ \hline
% \multirow{4}{*}{4}   & (2.0, 13.0), (2.0, 24.5),  & \multirow{4}{*}{10.0} & \multirow{4}{*}{2.0} & \multirow{4}{*}{24} & \multirow{2}{*}{Simple} & \multirow{4}{*}{1182.3125} \\ %\cline{4-4}
%                       & (2.0, 25.0), (2.0, 25.0), & & & & \multirow{2}{*}{selection} & \\ %\cline{4-4}
%                       & (2.0, 28.0) & & & & & \\ %\cline{4-4}
%                       & & & & & & \\ \hline
%     \end{tabular}
    
%     \caption{Best protocols found using GA algorithm. For comparison purposes with protocols in \ref{tab:bench}, presented protocols have constraints of minimum 12 and 24~hours between doses and 1.25 and 2.0~Gy maximum single dose. Fitness scores are calculated based on fitness function~\eqref{eq:final_fit_fun} defined in Section~\ref{sec:GPU:implem} evaluated 100 times for each of 10 tumours from study library, for details see Section~\ref{sec:lib}}.
%     %\mon{consider to change it -- shall we include CPU, GPU results?}
%     \label{tab:ga_results}
% \end{table}


\begin{table}[htp]
    \centering
    \caption{Batches with configurations of GA algorithm experiments, described in the Appendix \ref{appendix_a}. \textit{Number of runs} represents the number of GA runs with for each configuration settings. Starting from series 6$^*$, we introduced the retaining best 10\% of protocols. Final evaluation on all configurations was conducted in the series 8$^{**}$, as least 3 times for each of the configuration for robustness.}
    \begin{tabular}{|c|c|c|c|c|c|c|}
    \hline
        Series & Number & \multirow{3}{*}{Selection} & \multirow{3}{*}{Crossover} & \multirow{3}{*}{Mutation} & Max & Time \\
        No. & of runs & & & & single & interval \\
        & & & & & dose [Gy] & [h] \\
        \hline
        \multirow{2}{*}{1} & \multirow{2}{*}{8} & simple, & single-point & dose time, & \multirow{2}{*}{2} & \multirow{2}{*}{3} \\
         & & tournament & uniform & dose value & & \\
        \hline
        \multirow{2}{*}{2} & \multirow{2}{*}{24} & simple, & \multirow{2}{*}{single-point} & dose time, & \multirow{2}{*}{2.5} & 6, \\
         & & tournament & & dose value & & 24 \\
        \hline
        \multirow{4}{*}{3} & \multirow{4}{*}{12} & \multirow{4}{*}{simple} & single-point, & dose time, & \multirow{4}{*}{2.5} & \multirow{4}{*}{24} \\
        & & & two-point, & dose value, & & \\
        & & & uniform & swap, & & \\
        & & & & split & & \\
        \hline
        \multirow{3}{*}{4} & \multirow{3}{*}{22} & \multirow{3}{*}{simple} & single-point, & dose time, & 3, 3.5, & \multirow{3}{*}{24} \\
        & & & two-point, & dose value & 4, 4.5 & \\
        & & & uniform & & & \\
        \hline
        \multirow{2}{*}{5} & \multirow{2}{*}{25} & \multirow{2}{*}{simple} & two-point, & dose time, & 2, 2.5, 3, & \multirow{2}{*}{48} \\
        & & & uniform & dose value & 3.5, 4, 4.5 & \\
        \hline
        \multirow{2}{*}{6*} & \multirow{2}{*}{25} & \multirow{2}{*}{simple} & \multirow{2}{*}{two-point} & dose time, & 2, 2.5, 3, & 24, \\
        & & & & dose value & 3.5, 4, 4.5 & 48 \\
        \hline
        \multirow{4}{*}{7} & \multirow{4}{*}{30} & \multirow{4}{*}{simple} & \multirow{4}{*}{two-point} & dose time, & 1.25, & 6, \\
        & & & & dose value & 2, 2.5, & 12, \\
        & & & & dose value & 3, 3.5, & 24, \\
        & & & & dose value & 4, 4.5 & 48 \\
        \hline
        \multirow{4}{*}{8**} & \multirow{4}{*}{272} & simple & single-point & dose time, & 1.25, & 6, \\
        & & tournament & two-point & dose value, & 2, 2.5 & 12, \\
        & & roulette wheel & uniform & swap, & 3, 3.5, & 24, \\
        & & & & split & 4, 4.5 & 48 \\
        \hline
    \end{tabular}
    \label{tab:exp_settings}
\end{table}


Based on previous experiment batches, as the eighth batch we decided to run the best performing configuration of operators for all available maximum doses and time intervals. We used: \emph{two-point crossover} operator and a combination of three mutation operators, namely: \textit{swap mutation}, \textit{split mutation}, and \textit{dose time mutation}. We conducted experiments with the three introduced selection operators, \textit{roulette wheel selection}, \textit{tournament selection} and \textit{simple selection}.
To capture all of the possible time intervals and maximum doses configurations, we decided to use intervals of 6, 12, 24, and 48~h and single admission doses limited to 1.25, 2.0, 2.5, 3.0, 3.5, 4.0 and 4.5~Gy. This gave us a total of 84 configurations, increasing a previous searching space for doses limited to 1.25~Gy with 24 and 48~h intervals and doses above 2.0~Gy with smaller 6 and 12~h intervals. Each individual configuration was run three times to ensure robustness of the results. Because our GA algorithm usually converged after 30 iterations we decide to use 50 iterations of GA per single run. During evaluation of results, we decided to conduct additional 3 runs up to 100 iterations of GA for \textit{roulette wheel selection}, because for some experiments, the variability of fitness function was higher. However, it did not affect the results. Each protocol was evaluated by simulations 80 times (see Section~\ref{fitness_function}), the average and standard deviations of evaluation values were calculated. For each configuration, we collected values for 3 best protocols from all iterations obtained during those 10 GA runs.

The results of these experiments for 3 best protocols obtained during GA runs are summarised in Table~\ref{tab:roulette-selection}. The superiority of \emph{simple selection} over \emph{tournament} and \emph{roulette wheel} is visible. Clearly, the results for \emph{simple selection} are dominant for the shortest time interval of 6~h.


\begin{table}[ht]
\centering
\caption{
Summary of the results for selection operators: \emph{simple selection}, \emph{roulette wheel selection} and \emph{tournament selection}. Reported values are means and standard deviations (for 3 best protocols). Bold values indicate the highest fitness function values for a particular time interval and maximum dose value out of available selection operators.}\label{tab:roulette-selection}
\begin{tabular}{|c|c|r|r|r|r|}
    \hline
    \multicolumn{2}{|c|}{\multirow{2}{*}{\textbf{Simple selection}}} & \multicolumn{4}{c|}{Min time interval [h]} \\
    \cline{3-6}
    \multicolumn{2}{|c|}{} & 6~h & 12~h & 24~h & 48~h \\
    \hline
    \multirow{7}{*}{Max single dose} & 1.25~Gy & \textbf{1142.55 (4.84)} & \textbf{1160.79 (3.85)} & \textbf{1046.52 (4.09)} & 982.75 (1.39) \\\cline{2-6}
    & 2.0~Gy & \textbf{1179.78 (6.49)} & \textbf{1181.75 (3.29)} & \textbf{1156.23 (2.7)} & 1034.06 (0.21) \\\cline{2-6}
    & 2.5~Gy & \textbf{1197.31 (16.73)} & 1198.58 (4.57) & \textbf{1179.41 (17.76)} & \textbf{1074.56 (4.55)} \\\cline{2-6}
    & 3.0~Gy & \textbf{1213.46 (21.42)} & \textbf{1220.67 (4.25)} & \textbf{1193.23 (11.04)} & \textbf{1116.95 (0.53)} \\\cline{2-6}
    & 3.5~Gy & \textbf{1234.54 (7.56)} & \textbf{1235.49 (14.59)} & 1240.3 (22.22) & \textbf{1153.63 (1.73)} \\\cline{2-6}
    & 4.0~Gy & \textbf{1248.37 (44.21)} & \textbf{1271.51 (26.07)} & \textbf{1259.14 (6.38)} & 1153.24 (6.83) \\\cline{2-6}
    & 4.5~Gy & \textbf{1268.23 (17.44)} & \textbf{1279.39 (7.78)} & 1271.97 (10.3) & \textbf{1189.35 (15.55)} \\\cline{2-6}
    \hline\hline
    \multicolumn{2}{|c|}{\multirow{2}{*}{\textbf{Tournament selection}}} & \multicolumn{4}{c|}{Min time interval} \\
    \cline{3-6}
    \multicolumn{2}{|c|}{} & 6~h & 12~h & 24~h & 48~h \\
    \hline
    \multirow{7}{*}{Max single dose} & 1.25~Gy & 1132.24 (6.39) & 1154.73 (1.48) & 1045.35 (0.88) & 988.55 (8.61) \\\cline{2-6}
    & 2.0~Gy & 1167.22 (1.93) & 1180.82 (5.39) & 1132.48 (22.7) & \textbf{1034.82 (1.95)} \\\cline{2-6}
    & 2.5~Gy & 1187.57 (8.01) & \textbf{1198.64 (3.99)} & 1163.35 (10.12) & 1074.41 (1.26) \\\cline{2-6}
    & 3.0~Gy & 1192.37 (4.47) & 1207.65 (22.55) & 1190.1 (6.96) & 1116.0 (4.31) \\\cline{2-6}
    & 3.5~Gy & 1210.81 (47.82) & 1226.08 (24.22) & \textbf{1249.1 (5.83)} & 1149.67 (4.89) \\\cline{2-6}
    & 4.0~Gy & 1233.58 (12.3) & 1250.71 (11.65) & 1253.17 (12.68) & \textbf{1158.9 (2.91)} \\\cline{2-6}
    & 4.5~Gy & 1251.98 (27.07) & 1277.99 (11.41) & \textbf{1275.11 (1.1)} & 1172.56 (13.56) \\\cline{2-6}
    \hline\hline
    \multicolumn{2}{|c|}{\multirow{2}{*}{\textbf{Roulette selection}}} & \multicolumn{4}{c|}{Min time interval} \\
    \cline{3-6}
    \multicolumn{2}{|c|}{} & 6~h & 12~h & 24~h & 48~h \\
    \hline
    \multirow{7}{*}{Max single dose} & 1.25~Gy & 1130.29 (5.84) & 1155.53 (3.01) & 1040.01 (1.67) & \textbf{991.99 (8.55)} \\\cline{2-6}
    & 2.0~Gy & 1161.11 (3.14) & 1178.66 (5.11) & 1138.92 (15.9) & 1031.1 (2.26) \\\cline{2-6}
    & 2.5~Gy & 1176.81 (4.16) & 1193.78 (15.9) & 1166.56 (14.12) & 1052.09 (15.31) \\\cline{2-6}
    & 3.0~Gy & 1202.3 (21.66) & 1208.35 (17.35) & 1189.1 (12.34) & 1107.09 (12.64) \\\cline{2-6}
    & 3.5~Gy & 1224.58 (20.92) & 1232.26 (1.42) & 1217.83 (23.16) & 1132.03 (11.45) \\\cline{2-6}
    & 4.0~Gy & 1227.4 (29.44) & 1227.82 (7.33) & 1238.54 (20.19) & 1158.25 (6.18) \\\cline{2-6}
    & 4.5~Gy & 1246.75 (25.58) & 1261.55 (11.44) & 1255.3 (13.21) & 1168.19 (21.0) \\\cline{2-6}
    \hline
\end{tabular} 
\medskip
\raggedright
\label{tab:results-table}
\end{table}

% Section about different experiments and setups we were running.

%-------------------------------------------------------------------------

\section{Acknowledgements}

This work was supported by grant no. 2015/19/B/ST1/01163 of National Science Centre, Poland, "Mathematical models and methods in description of tumour growth and its therapies". This research was carried out with the support of the Interdisciplinary Centre for Mathematical and
Computational Modelling (ICM) University of Warsaw under grants no G74-17 and GR79-29.


\mon{
Acknowledgments, when needed, should include the following information in the order listed below, a single paragraph, starting with the word "Acknowledgments:" in bold. Please use the subhead and boldface layout as shown below.).

General: Thank others for any contributions. 

Funding: Include all funding sources, including grant numbers and funding agencies. 

Author contributions: Describe the contributions of each author (use initials) to the paper. 

Competing interests: Include any financial interests of the authors that could be perceived as being a conflict of interest. Also include any awarded or filed patents pertaining to the results presented in the paper. If there are no competing interests, please state so.

Data and materials availability: If data are in an archive, include the accession number or a placeholder for it. Also include any materials that must be obtained through an MTA. 

Acknowledgments follow the references but are not numbered.}
%	\FloatBarrier
%-------------------------------------------------------------------------

\section{Supplementary Materials}

\mon{You may include up to a total of ten figures and/or tables (combined) throughout the supplemental text. Include supporting text (including supplementary materials and methods, tables, and figures) at the end of the main manuscript file, in a separate section titled Supplementary Materials, if this can be easily done and if the total file size does not exceed 6 MB.  Alternatively, Supplementary Materials can be included as a separate file that can be uploaded as the final figure file within the 6 MB upload limit. In that case, use one of the file types specified above (.doc or .docx preferred).

If you have any Supplemental Materials please list them by sections in the following order: supplementary materials and methods (if any), supplementary figures, supplementary tables, and other supplementary files (such as movies, data, interactive images, or database files). Be sure to submit all Supplementary Materials with the manuscript. Example:\\

Materials and Methods\\
Fig. S1. Title of the first supplementary figure.\\
Fig. S2. Title of the second supplementary figure.\\
Table S1. Title of the first supplementary table.\\
Data file S1. Title of the first supplementary data file.\\
Movie S1. Title of the first supplementary movie.}

%	\FloatBarrier

\appendix
\section{Appendix A}
\label{appendix_a}
% \input{chapters/appendix}

\madd{{\bf Description of Batches}}\\
In this section, we describe the most important batches of experiments with genetic algorithms we conducted.

\woj{Batch no.1, between 23-25.08.2020}
\madd{{\bf Batch No. 1}}

In the first batch of experiments we used the following fitness function
\begin{equation*}
   f(p_{i,j}) = 1 600 - n_{i,j}, 
\end{equation*}
where $p_{i,j}$ denotes considered protocol $i$ evaluated on a single tumour $j$, 1\,600 is the maximal number of possibly occupied sites, $n_{i,j}$ denotes the number of the occupied sites after testing protocol $i$ on tumour $j$.

Comparing to other batches in which formula ~\eqref{eq:final_fit_fun} was used to calculate the fitness function, this formula leads to increase of all fitness by $100$.

In this batch we implemented the \textit{annealing scheme} for mutation operation probabilities. Based on our preliminary experiments we decided to start with the \textit{dose time mutation} and \textit{dose value mutation} operators. Our goal was to scrutinise two type of crossover operators, namely \textit{single-point crossover} and \textit{uniform crossover}, with selection operators: \textit{simple selection} and \textit{tournament selection}. For tournament selection, we tested three different probability values: 0.099, 0.5 and 0.9. The maximum single dose was 2.0Gy, minimal time interval was equal to 3~h and the experiments were run till reached 300 iterations of GA.

The results showed a superiority of \textit{simple selection}, with a stable increase of fitness value. The results for \textit{tournament selection} were lower by ~about 20 points than for the \textit{simple selection}, with the significant drop in fitness value, especially for the lower probability. There was no notable difference between \textit{uniform crossover} and \textit{single-point crossover} operators. 

\woj{Batch no.2, between 07-16.09.2020}
\madd{{\bf Batch No. 2}}\\
For the second batch of experiments we used the final fitness function formula \eqref{eq:final_fit_fun}. We decided to increase a maximum single dose to 2.5~Gy, use only \emph{single-point crossover} operator and test more realistic time interval between doses, i.e., 6~h and 24~h. Moreover, we evaluated \textit{dose time mutation} and \textit{dose value mutation} operators, making it to a total number of 24 different experiment configurations.

The results indicated the key importance of time between doses interval, with the average superiority of 25 points in fitness function value for 6~h interval over 24~h interval. As in batch no. $1$, the results for \emph{tournament selection} were lower than for a \emph{simple selection} and the level of variance was significantly larger. Thus, in the further analysis, we focused on \emph{simple selection} to reduce the time required for every batch of experiments.


\woj{Batch no.3, between 17-21.09.2020}
\madd{{\bf Batch No. 3}}\\
The main contribution of this batch of experiments was the introduction of annealing probability of mutation in comparison to the prior fixed value. We tested two sets of mutation operators: in the first, we mutated both \textit{dose time mutation} and \textit{dose value mutation}, while in the second, we additionally used \emph{swap mutation} and \emph{split mutation} operators. Together with this improvement, we evaluated three types of \textit{crossover operators}. The rest of the experiment settings was the same as in batch no. 1, with \emph{simple selection}, time interval of 24~h, and maximum single dose of 2.5~Gy.

Fitness function values for the experiments with all four \emph{mutation operators} showed variability at the beginning of the experiments, finally reaching almost the same results as with \textit{dose time mutation} and \textit{dose value mutation} only. There was no significant difference between experiments with annealing mutation probability and experiments with fixed probability values. Finally, the only noteworthy difference occurred for the \emph{uniform crossover}, over the \emph{single-point crossover} and \emph{two-point crossover} operators, achieving up to 20 points increase.

\woj{Batch no.4, between 26.09-04.10.2020}
\madd{{\bf Batch No. 4}}\\
We decided to extend the experiments with the different maximum single doses of 3.0~Gy, 3.5~Gy, 4.0~Gy and 4.5~Gy. We applied \emph{mutation operators} to \textit{dose time mutation} and \textit{dose value mutation}, and all other settings remained the same as in the batch no.~3.

The results confirmed the negligible difference between \emph{single-point crossover} and \emph{two-point crossover} operators in all runs. Moreover, the superiority of the \emph{uniform crossover} operator, observed in batch no.~4, was no longer maintained for larger doses, and in consequence, the results for all \emph{crossover operators} remain almost identical. It is worth to underline that the huge improvement of the fitness function value was observed with the change of maximum dose value, with the lowest fitness function value of 1180 for 3.0~Gy and up to 1260 for 4.5~Gy.

\woj{Batch no.5, between 01-10.10.2020}

\madd{{\bf Batch No. 5}}\\
Having established the importance of the maximum single dose, we increased the time interval between doses from 24~h to 48~h. To capture larger number of possible maximum doses, we tested doses between 2.0~Gy up to 4.5~Gy, with 0.5~Gy steps. Finally, to limit the computational time, we investigated only \emph{two-point crossover} and \emph{uniform crossover} operators.

Experiment outcomes showed an interesting fact, namely for time interval of 48~h, the \emph{two-point crossover} operator results were higher by 10 points on average than the results for \emph{uniform crossover} operator, which is the opposite to the batch no.3 results, with time interval of 24~h.
Increasing the time interval to 48~h we considerably lowered the results, even up to 100 points, and the highest difference was observed for 2.5~Gy and 3.5~Gy maximum doses.

\woj{Batch no.6, between 10-21.10.2020 and batch 6.1 between 01-11.11.2020}
\madd{{\bf Batch No. 6}}\\
To achieve our goal of reaching the highest value of fitness function, we introduced retaining of the best protocols from the last iteration of the GA. Ten percent of the best parent protocols were passed unchanged to the next iteration, allowing us to maintain at lest the best results from the previous iteration. We evaluated this setup using only \emph{two-point crossover} operator, mutation operators acting on dose time and value with annealing probability and \emph{simple selection}. As in batch no.~5, we tested maximum doses between 2.0~Gy up to 4.5~Gy, with 0.5~Gy steps, but this time two time intervals of 24~h and 48~h.

The results obtained with the improved algorithm showed an increase in results stability, but the final value remained akin. The important conclusion from this batch is that we can reduce the number of iterations because the GA convergence is achievable earlier than 50\% of all the iterations.

For comparison with results from \cite{Angus2014}, we run additional experiments with time interval of 6~h and 12~h, and maximum single dose of 1.25~Gy. The results for 6~h time interval were lower by 10 points comparing to 12~h interval, which is in contrary to our previous experiments.
Overall, the results for additional experiments were similar to results for experiments with 2.0~Gy maximum dose \& 24~h time interval or 3.5~Gy \& 47~h time interval.

\woj{Batch no.7, between 24-29.11.2020 and batch 7.1, between 06-14.12.2020}
\madd{{\bf Batch No. 7}}\\
To establish the difference between the best and worst protocols, we introduced inversion of the GA. Our intention was to obtain the lower fitness function value and to analyse the protocols, searching for possible patterns. The experiments set up was similar to batch {no.~6}, testing both 24~h and 48~h time intervals, and maximum single doses between 2.0~Gy and 4.5~Gy. Similarly as in batch {no.~6}, we tested additional two configurations with maximum dose of 1.25~Gy and time intervals of 6~h and 12~h.

The GA algorithm was converging fast, however for obtained protocols we saw a significant variance. Most of the experiments were able to achieve the lowest fitness function value of about 910. 


%-------------------------------------------------------------------------
\bibliographystyle{plain}
\bibliography{refs,automaty}
\end{document}

// In terms of journal --

I think if the above plan was achievable, then at least /JTB/ if not trying for /Plos-CB/?

PLOS-CB is a tough but good review system .. quite fast. They have good exposure, and rate well.

I think the paper would need to demonstrate a sufficient level of novelty beyond the previous paper we did .. i.e. to show that the GPU effort was worth it -- significantly increasing the quality of results, and in various clinically feasible setups (hence, the constraints I suggest).

Or, one could try with /Science Advances/?

