\section{Introduction}
\label{sec:intro}

    \begin{figure}[ht!]
        \centering
        \centerline{\includegraphics[width=\linewidth]{figures/fig1.pdf}}
        \caption{\textbf{(a)} Overview of the VAE training pipeline. \textbf{(b)} Overview of the pipeline at inference. For generating UMAP plots, a similar patch sampling as training is used.}
        \label{fig:1}
    \end{figure}
    
    %Histopathological images are really large. The size of the dataset is in order of Petabytes which makes it difficult and time consuming for storage and processing. Therefore, we need to compress these images. In fact, image compression can help us remove useless information of images so that it can be processed and stored more efficiently. 
    Histopathological images derived from cross sectional tissue microscopy are used in the clinical setting for diagnosis of various diseases and conditions \cite{gurcan2009histopathological}. Hemotoxylin and Eosin (H\&E) staining, which introduce a contrast dye for the discernment of nuclear and cytoplasmic structures, has long been used to determine carcinomal regions of excised tissue from cancer patients \cite{he2012histology}. For this reason, databases of tumor patient slides, such as the NIH Genomic Data Commons (GDC), have been compiled for researchers to access tens of thousands of cancer patients' histopathological data. The GDC itself contains more than 30,000 Whole Slide Images (WSIs) which, with each slide representing over a billion pixels each, is stored on over 20 TB of data. Most purposes, from retrieval to transmission, local storage, and data analysis would benefit from efficient, indexable storage structures of this WSI data \cite{niazi2019pathological}.  This is especially applicable to image search algorithms for large whole slide image databases \cite{kalra2020yottixel}.
    
    Several solutions have been proposed for the efficient storage and indexing of cancer tissue image data. Classic compression formulas such as JPEG2000 can successfully reduce image size at a compression ratio of 32:1 before becoming unusable for histopathological classification of malignancy \cite{krupinski2012compressing}. Compression and scaling has also been found to adversely effect tissue segmentation up to ratios of 50:1 \cite{konsti2012effect}. In contrast to discrete cosine transformation models, neural networks have been proven to retain high efficiency and fidelity in the lossy compression of image data \cite{soliman2006neural}. While neural networks seek to store image data in latent space representations, not every network does this at equivalent efficiency or accuracy \cite{jamil2022learning}. Several studies have demonstrated that Variational Auto Encoders (VAEs) retain higher image quality and lower noise ratios at extreme compression ratios \cite{hu2021learning,lombardo2019deep,yilmaz2021self}. Tellez et al., \cite{tellez2019neural}  showed in a benchmark study that VAE compression of medical tissue images to a latent space of 128 ($>$5000:1 compression ratio) retained the most details of the original whole slide image compared to 4 other encoders. In the current study, we develop a VAE to compress and index images in latent space for fast complex search of whole slide H\&E cancer images.
    
    \begin{figure}[t]
        \centering
        \centerline{\includegraphics[width=\linewidth]{figures/fig2-new.pdf}}
        \caption{\textbf{(a)} Example of how normalization affects the performance of our pipeline. Both models are trained using the exact same hyper-parameters (\texttt{latent\_dim} = 64). \textbf{(b)} The effect of batch size and latent dimension of validation loss. For better visualization, early stopping is not used for these experiments.}
        \label{fig:2}
    \end{figure}
    
    
    %Neural networks VAE \cite{tellez2019neural}
    %non-VAE search \cite{kalra2020yottixel}
    %However, it is crucially needed to do this task with high accuracy in order to avoid losing important features of images. For this aim, autoencoders have responded very well. \cite{jamil2022learning} has examined image compression methods. 
    %Lombardi et al., \cite{lombardo2019deep} has used VAE for compressing temporal sequences of video data. Also, \cite{yilmaz2021self} used self-VAE to increase nonlinearity in order to compress images with more accuracy. There are some works for histopathology images. \cite{kwonautoencoder} explained that typical VAEs are unaccepted in terms of loss for diagnostic. In other words, they believed that these networks are not useful for medical images because the constructed images are not accepted by histopathologist and they need a lot of resources. Therefore, they proposed a lightweight network (VQ-VAE) which can use much less resources, while it preserves the important features of medical images. In other work, \cite{tellez2019neural} had take advantage of image compression method for gigapixel histopathology image and used a CNN network to predict image-level labels.
    
    
    % Several studies have been carried out to determine whether the VAE-based latent space can capture biologically relevant features. Ref. \cite{way_extracting_2018} has trained VAE for compressing TCGA pan-cancer RNA-seq data and found some specific features on the latent space. \cite{karlova2021molecular} presented a low dimensional latent representation of pharmacophoric features using $\beta$-VAE, that could be beneficial for downstream tasks like drug discovery, molecular similarity assesment. \cite{ternes2022multi} extracted biologically significant characteristics from single cell image analysis, including emergent features that were not immediately apparent from prior information, using multi-encoder VAE (ME-VAE).
    
