
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% This is a (brief) model paper using the achemso class
%% The document class accepts keyval options, which should include
%% the target journal and optionally the manuscript type.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\documentclass[journal=jacsat,manuscript=article]{achemso}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Place any additional packages needed here.  Only include packages
%% which are essential, to avoid problems later.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\usepackage{chemformula} % Formula subscripts using \ch{}
\usepackage[T1]{fontenc} % Use modern font encodings

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% If issues arise when submitting your manuscript, you may want to
%% un-comment the next line.  This provides information on the
%% version of every file you have used.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%\listfiles

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Place any additional macros here.  Please use \newcommand* where
%% possible, and avoid layout-changing macros (which are not used
%% when typesetting).
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newcommand*\mycommand[1]{\texttt{\emph{#1}}}

%%%%%%%%%%%%%%%%%%%
\SectionNumbersOn
\usepackage{hyperref}
\usepackage{url}
\hypersetup{
    colorlinks=true,
    linkcolor=blue,
    filecolor=gray,      
    urlcolor=blue,
    citecolor=blue,
}
\usepackage{breakurl}
\usepackage{bm}
\usepackage{pythonhighlight}
\usepackage{placeins}
\usepackage{graphicx}
\usepackage{layouts}
\usepackage{subfigure}
\usepackage{caption}

%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Meta-data block
%% ---------------
%% Each author should be given as a separate \author command.
%%
%% Corresponding authors should have an e-mail given after the author
%% name as an \email command. Phone and fax numbers can be given
%% using \phone and \fax, respectively; this information is optional.
%%
%% The affiliation of authors is given after the authors; each
%% \affiliation command applies to all preceding authors not already
%% assigned an affiliation.
%%
%% The affiliation takes an option argument for the short name.  This
%% will typically be something like "University of Somewhere".
%%
%% The \altaffiliation macro should be used for new address, etc.
%% On the other hand, \alsoaffiliation is used on a per author basis
%% when authors are associated with multiple institutions.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\author{Xiaojuan Hu}
\affiliation{Fritz-Haber-Institut der Max-Planck-Gesellschaft, Faradayweg 4-6, 14195 Berlin, Germany}
\email{xhu@fhi-berlin.mpg.de}
\author{Kazi S. Amin}
\affiliation{Centre for Molecular Simulation and Department of Biological Sciences, University of Calgary, 2500 University Drive NW, Calgary, Alberta T2N 1N4, Canada}
\email{kazi.amin@ucalgary.ca}
\author{Markus Schneider}
\affiliation{Fritz-Haber-Institut der Max-Planck-Gesellschaft, Faradayweg 4-6, 14195 Berlin, Germany}
%\author{Zhongkang Han}
%\affiliation{State Key Laboratory of Silicon Materials, School of Materials Science and Engineering, Zhejiang University, Hangzhou 310027, China}
\author{Carmay Lim}
\affiliation{Institute of Biomedical Sciences, Academia Sinica, Taipei 115, Taiwan}
\alsoaffiliation{Department of Chemistry, National Tsing Hua University, Hsinchu 300, Taiwan}
\author{Dennis Salahub}
\affiliation{Centre for Molecular Simulation and Department of Chemistry, University of Calgary, 2500 University Drive NW, Calgary, Alberta T2N 1N4, Canada}
\email{dsalahub@ucalgary.ca}
\author{Carsten Baldauf}
\affiliation{Fritz-Haber-Institut der Max-Planck-Gesellschaft, Faradayweg 4-6, 14195 Berlin, Germany}
\email{baldauf@fhi-berlin.mpg.de}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% The document title should be given as usual. Some journals require
%% a running title from the author: this should be supplied as an
%% optional argument to \title.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title[FFAFFURR]
  {System-specific parameter optimization for non-polarizable and polarizable force-fields}
  %\footnote{A footnote for the title}}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% User defined variables and commands
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\ksa}[1]{\textcolor{blue}{\textbf{ksa: }#1}}
\newcommand{\xhu}[1]{\textcolor{red}{\textbf{Xhu: }#1}}
\newcommand{\cba}[1]{\textcolor{orange}\textbf{cba: }#1}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% The manuscript does not need to include \maketitle, which is
%% executed automatically.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}
\begin{center}
\noindent
We dedicate this manuscript to Sergei Noskov, who initiated this work\\and whose much too early death shook us all.
\end{center}

\newpage

\begin{abstract}
\noindent
The accuracy of classical force-fields (FFs) has been shown to be limited for the simulation of cation-protein systems despite their importance in understanding the processes of life.
  Improvements can result from optimizing the parameters of classical FFs or by extending the FF formulation by terms describing charge transfer and polarization effects.
  In this work, we introduce our implementation of the CTPOL model in OpenMM, which extends the classical additive FF formula by adding charge transfer (CT) and polarization (POL).
  Furthermore, we present an open-source parameterization tool, called FFAFFURR that enables the (system specific) parameterization of OPLS-AA and CTPOL models.
  The performance of our workflow was evaluated by its ability to reproduce quantum chemistry energies and by molecular dynamics simulations of a Zinc-finger protein.
\end{abstract}


\newpage

\tableofcontents

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Start the main part of the manuscript here.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%This is a paragraph of text to fill the introduction of the
%demonstration file.  The demonstration file attempts to show the
%modifications of the standard \LaTeX\ macros that are implemented by
%the \textsf{achemso} class.  These are mainly concerned with content,
%as opposed to appearance.

Metal ions are essential in biological systems and are involved in physiological functions ranging from maintaining protein structure and stability to directly participating in catalytic activities. \cite{sarkar1987metal} 
Approximately one-third of all proteins contain metal ions. \cite{Peters2010} 
As an abundant cation in the human body, \cite{christianson1991structural} Zinc ions are known to play an important role in enzyme catalysis or protein folding$/$stability. 
In aqueous solutions, Zn$^{2+}$ normally coordinates with six water molecules in an octahedral coordination geometry.
However, in a protein environment, Zn$^{2+}$ is often observed to form a tetrahedral coordination structure with four ligating amino acid residues, \cite{patel2007analysis} commonly His and Cys.
Due to the nature of electrostatic interactions, Zn$^{2+}$ also tends to be close to negatively charged residues such as Asp or Glu.
Zn$^{2+}$ is involved in various biological functions by interacting with these residues.
For example, metallothioneins (MTs) \cite{Babu2014, https://doi.org/10.1002/cbic.200800511} are present in all living organisms and are involved in various diseases. \cite{Capdevila2012,Cherian2004Metallo, Durand2010} 
Under physiological conditions, the four mammalian MT isoforms have Zn$_3$Cys$_9$ clusters and Zn$_4$Cys$_{11}$ clusters in their centers as functional groups. 
Zinc-finger proteins are another well-studied class of Zinc-containing proteins. 
Multiple fingers can combine together to carry out many complex functions, 
such as regulating DNA/RNA transcription,\cite{Miller1985Repetitive,Wolfe2000DNA} 
protein folding and assembly, lipid binding, Zinc sensing,\cite{laity2001zinc} and even protein recognition.\cite{Gamsjaeger2007Sticky}
The most well characterized Zinc-finger proteins feature a binding domain with two Cys and two His residues.
The study of the classical Cys$_2$His$_2$ Zinc-finger structures is crucial for a better understanding of their broader functions.

Molecular dynamics (MD) simulations employing molecular mechanics (MM) are widely used in the study of complex biological processes, such as protein folding, protein dynamics, and enzyme catalysis because of their ability to model systems at atomic scales ranging in sizes from thousands to millions of atoms and time scales of milli-seconds. \cite{Mobley2012, Lemkul2016, Jorgensen2004} 
The majority of current MD studies employ classical force-fields (FFs) such as OPLS-AA, \cite{A.Kaminski2001} AMBER, \cite{doi:10.1002/wcms.1121} CHARMM\cite{Huang2017} and GROMOS. \cite{Reif2013} 
%The success of force-fields is likely to depend on two aspects: the physics behind the potential function, and parameter optimization on increasingly large \textit{ab initio} and experimental data sets. \cite{Zhu2012} 
It is a challenge for classical force-field models to describe metal–protein interactions due to the strong local electrostatic field and induction effect, \cite{Li2015,Li2017,Amin2020,doi:10.1002/qua.26369,schneider2018relative, wu2010polarizable} for example, computer simulation of Zinc-containing proteins has been a long-standing challenge that appears hard to tackle without explicit treatment of charge-transfer or polarization.

One approach to improve the accuracy of force-fields is to refine the parameters by fitting the model to more and more accurate experimental data or quantum mechanical (QM) calculations. 
For example, force-matching algorithms \cite{Akin-Ojo2008} were used to fit parameters to reproduce \textit{ab initio} forces. 
Empirical Continuum Correction (ECC) \cite{Duboue-Dijon2020,Martinek2018,LeBreton2020} force-fields scale the charges to implicitly take electronic polarization into account.
Several works \cite{li2015systematic, li2015parameterization} tune the Lennard-Jones (LJ) parameters or use a 12-6-4 LJ-type model to simulate charge-induced dipole interactions. 
These efforts have been successful to some extent, however, reparameterization is often time-consuming and labor-intensive. 
There are a few automatic parameterization tools, for example, CHARMM General force-field (CGenFF), \cite{Vanommeslaeghe2010} LigParGen, \cite{dodda2017ligpargen} and Antechamber. \cite{sousa2012acpype, Wang2004} 
These programs typically generate missing parameters for a given system based on analogies with atom types and the relevant parameters available in the corresponding FF or through parameter estimation algorithms. \cite{vassetti2019assessment} 
However, the accuracy of assigning approximate parameters to a specific system is limited, and parameters already present in a given FF may also have to be optimized.
FFparam \cite{kumar2020ffparam} and ForceBalance \cite{wang2014building} enable the tuning of existing FF parameters. 
All these parameterization tools share a common assumption of transferability, which assumes a set of parameters optimal for small organic molecules for a given atom type can be applied in a wide range of chemical and spatial contexts.
It is well known that the presence of electron donors and acceptors can significantly affect molecular properties by polarization effects. \cite{Jorgensen2007} 
LJ parameters are also sensitive to the local environment \cite{Tkatchenko2012, Tkatchenko2009} and long-range electrodynamic screening. \cite{Gobre2013} 
In this regard, a fundamentally different approach to derive environment-specific or molecule-specific parameters is proposed in references \cite{Horton2019, Cole2016, Grimme2014}. 
However, parameters still remain fixed despite structures and environments changing over the course of, e.g., MD simulations.

Another approach to improve FF accuracy in metalloprotein simulations is to introduce more physics in to the model. 
Including polarization effects is a significant step to improve force-fields. \cite{Borodin2009, Cieplak2009} 
There is growing evidence that polarizable force-fields describe ionic systems more accurately than classical force-fields. 
It has been found that the inclusion of polarization plays an important role in the simulation of ion channels, \cite{Allen2004} enzymatic catalysis, \cite{Boulanger2014} protein-ligand binding affinity \cite{Panel2018} and dynamic properties of proteins. \cite{Li2011}

At present, there are three main groups of polarizable force-fields, fluctuating charge, induced point-dipoles, and Drude oscillator models. \cite{Bedrov2019} 
The fluctuating charge models simulate polarization effects by allowing charge to flow through the molecule until the electronegativities of atoms become equalized, while keeping the total charge unchanged. \cite{Olano2005} 
%Fluctuating charge models include AACT (atom-atom charge transfer), \cite{chelli1999electrical} ABEEMsp (atom-bond electronegativity equalization model with s- and p-bonds) \cite{Yang2014} and CHARMM-FQ. \cite{Patel2004} 
One drawback of the fluctuating charge model is that it fails to capture out-of-plane polarization of planar or linear chemical groups. 
%The use of additional out-of-plane sites was proved to be inefficient for systems containing thousands of atoms. \cite{Yang2014} 
The fluctuating charge formula can also be used in conjunction with induced point-dipoles as a complementary approach to account for charge transfer (CT). \cite{soniat2012effects}
A notable model is SIBFA (Sum of Interactions Between Fragments \textit{Ab Initio} Computed). \cite{piquemal2007toward} 

The induced point-dipole models describe polarization energy as the interaction between static point charges and induced dipole moments.
Notable induced point-dipole models include OPLS/PFF, \cite{friesner2005modeling} AMBER ff02, \cite{cieplak2001molecular} and AMOEBA. \cite{ponder2003force, ren2003polarizable} 
The performance of the induced point-dipole models strongly depends on the accuracy of polarizability parameters. 

The Drude oscillator model simulates the distortion of the electron density by attaching additional charged particles (the oscillators) to each polarizable atom. 
Despite many successes of the Drude oscillator model, \cite{Li2015, ngo2015quantum, villa2018classical} it may be limited when charge transfer between cation and coordinating ligand atoms is significant, for example, Cys$^-$ coordinated to metal ions. \cite{Dudev2014} 
%A polarizable force-field may fail to describe systems in which a significant charge transfer is present. \cite{Ahlstrand2017} 
Ngo \textit{et al.} \cite{Ngo2015} and Dudev \textit{et al.} \cite{Dudev2003} showed that the charge located on the coordinating ligand is significantly perturbed due to the presence of Ca$^{2+}$. 
The effect exists not only in the first coordination shell, but also in the second shell. 
Thus, including the description of charge transfer is critical for the development of next-generation polarizable FFs. 

The CTPOL \cite{Sakharov2005,sakharov2009force} model incorporates charge transfer (CT) and polarization effects (POL) into classical force-fields. 
The inclusion of charge transfer reduces the amount of partial charge on cation and cation coordinating atoms. 
Thus, their charge/dipole–charge interactions are weakened. 
Local polarization energy between cation and coordinating ligands, which also depends on the partial charge, is introduced for compensation. 

Although numerous studies have shown that polarizable models perform better than classical force-fields in the simulation of metalloproteins, they have received only limited validation. 
Therefore, reparameterization may be necessary when applied to different systems. 
Our previous study \cite{Amin2020} has shown how QM data \cite{ropo2016first, hu2022better} drive the parameter development of the Drude and CTPOL models. 
However, most parameterization tools focus on classical force-field models.
%The reparameterization of polarizable models is even more time-consuming and tedious due to the increase in parameters. 
FFparam \cite{kumar2020ffparam} provides parameterization of Drude model; a CTPOL parameterization tool is not yet available.


%\xhu{Dennis: Needs polishing.} 
%In this work, we implemented the CTPOL model in OpenMM. The code of this implementation is shared on github. Furthermore, we present an open-source tool, FFAFFURR (Framework For Adjusting force-fields Using Regularized Regression), which facilitates the parameterization of OPLS-AA and CTPOL models for a specific system in question, e.g. peptide system or peptide-cation system. One advantage of FFAFFURR is the rapid construction of FFs for troublesome metal centers in metallo-proteins. In this work, the new parameters obtained from FFAFFURR are validated by the comparison of FF energies and QM potential energies and MD simulations in the condensed phase using a Zinc-finger protein as an example.

In this work, we fill this gap by (i) implementing the CTPOL model in OpenMM\cite{Eastman2017} and sharing this code \cite{Hu2023CTPOLMD} and (ii) publishing the Framework For Adjusting force-fields Using Regularized Regression (in short FFAFFURR ) an open-source tool, which facilitates the parameterization of OPLS-AA and CTPOL models for a specific system, e.g. a peptide system or a peptide-cation system.
A major advantage of FFAFFURR is the rapid construction of FFs for troublesome metal centers in metalloproteins.
In this work, the new parameters obtained from FFAFFURR were validated by the comparison of FF energies and QM energies in isolation and by assessing the stability of condensed phase MD simulations using a Zinc-finger protein as an example. 

\section{Methods}

\subsection{OPLS-AA functional form}

OPLS-AA is one of the major families of classical force-fields. 
It is used as the starting point for parameterization in this work. 
OPLS-AA uses the harmonic functional form to represent the potential energy shown in eq. \ref{additive_FF}.
\begin{equation}
E^{\mathrm{FF}} =  E_{\mathrm{bonds}} + E_{\mathrm{angles}} + E_{\mathrm{torsions}} + E_{\mathrm{improper}} + E_{\mathrm{vdW}} + E_{\mathrm{ele}} \label{additive_FF} 
\end{equation}
where $E^{\mathrm{FF}}$ is the potential energy of the system. 
$E_{\mathrm{bonds}}$, $E_{\mathrm{angles}}$, $E_{\mathrm{torsions}}$ and $E_{\mathrm{improper}}$ correspond to bonded or so-called covalent terms of bond stretching, bond-angle bending, dihedral-angle torsion, and improper dihedral-angle bending (or out-of-plane distortions) in the molecules.
$E_{\mathrm{vdW}}$ and $E_{\mathrm{ele}}$ are nonbonded terms. 
They describe van der Waals (vdW) and Coulomb (electrostatic) interactions, respectively.

The energy terms in eq. \ref{additive_FF} are depicted in detail in eq. \ref{additive_FF_detail}.

\begin{equation}
\begin{split}
E^{\mathrm{FF}} =  & \sum_{\mathrm{bonds}}^{\mathrm{1-2 atoms}} \frac{1}{2} K_{ij}^{r} \left (r_{ij}-r_{ij}^0 \right )^{2}+\sum_{\mathrm{angles}}^{\mathrm{1-3 atoms}} \frac{1}{2} K_{ij}^{\theta} \left (\theta_{ij}-\theta_{ij}^0 \right )^{2}+\sum_{\mathrm{dihedrals,n}}^{\mathrm{1-4 atoms}} V_n^{ij} \left ( 1 + cos \left( n\phi _{ij}-\phi_{ij}^0\right)\right )\\
& + \sum_{\mathrm{improper}}^{\mathrm{1-4 atoms}} V_{2imp}^{ij} \left ( 1 + cos \left ( 2\phi _{ij}-\phi_{ij}^0 \right) \right )+\sum_{i<j} 4 \varepsilon_{ij} \left [ \left ( \frac{\sigma_{ij}}{r_{ij}} \right)^{12} - \left (\frac{\sigma_{ij}}{r_{ij}} \right)^{6} \right ] f_{ij}+\sum_{i<j} \frac{q_iq_j}{r_{ij}}f_{ij} \label{additive_FF_detail}
\end{split}
\end{equation}

%\begin{eqnarray}
%E_{bonds} &=& \sum_{i<j}^{1-2 atoms} \frac{1}{2} K_{ij}^{r} \left (r_{ij}-r_{ij}^0 \right )^{2} \label{FF_bonds}\\
%E_{angles} &=&  \sum_{i<j}^{1-3 atoms} \frac{1}{2} K_{ij}^{\theta} \left (\theta_{ij}-\theta_{ij}^0 \right )^{2} %\label{FF_angles} \\
%E_{torsions} &=&  \sum_{i<j,n}^{1-4 atoms} V_n^{ij} \left ( 1 + cos \left( n\phi _{ij}-\phi_{ij}^0\right)\right ) \label{FF_torsions} \\
%E_{improper} &=&  \sum_{i<j}^{1-4 atoms} V_{2imp}^{ij} \left ( 1 + cos \left ( 2\phi _{ij}-\phi_{ij}^0 \right) \right ) \label{FF_improper} \\
%E_{vdW} &=& \sum_{i<j} 4 \epsilon_{ij} \left [ \left ( \frac{\sigma_{ij}}{r_{ij}} \right)^{12} - \left (\frac{\sigma_{ij}}{r_{ij}} \right)^{6} \right ] f_{ij} \label{FF_vdW}\\ 
%E_{ele} &=&  \sum_{i<j} \frac{q_iq_j}{r_{ij}}f_{ij} \label{FF_ele}
%\end{eqnarray}

where $K_{ij}^{r}$, $K_{ij}^{\theta}$, $V_n^{ij}$, and $V_{2imp}^{ij}$ are force constants, $r_{ij}^0$ and $\theta_{ij}^0$ are the reference bond length and bond angle, $r_{ij}$, $\theta_{ij}$ and $\phi_{ij}$ are current bond length, bond angle and dihedral angle, respectively, $n$ is the periodicity, $\phi_{ij}^0$ is the phase offset, $\sigma_{ij}$ is the distance at zero energy, $\varepsilon_{ij}$ sets the strength of the interaction, $q_i$ and $q_j$ are the charges of the two particles, and $f_{ij}$ is the scaling factor for short distances (i.e. ``1-4 pairs'') of nonbonded interaction. 
In OPLS-AA, the pairwise LJ parameters $\sigma_{ij}$ and $\varepsilon_{ij}$ are calculated as the geometric mean of those of individual atom types ($\sigma_{i}$ and $\varepsilon_{i}$). 

Classical force-field simulations were performed using OpenMM7, a high performance toolkit for molecular simulations. \cite{Eastman2017} 
%OpenMM7 does not provide the parameter file and LJ combination rules of OPLS-AA.
%Thus, we obtained the OPLS-AA parameter file for OpenMM through a python script \href{https://github.com/openmm/openmm/blob/master/devtools/forcefield-scripts/processTinkerForceField.py}{`processTinkerForceField.py'}. 
%The LJ combination rules for OPLS-AA were performed as decribed at \url{http://zarbi.chem.yale.edu/ligpargen/openMM_tutorial.html}.

\subsection{CTPOL model}

The CTPOL \cite{Sakharov2005, sakharov2009force} model introduces charge transfer and polarization effects into classical force fields. 
Instead of a fixed-charge model, CTPOL takes the charge transfer from a ligand atom $L$ (O, S, N) to a metal cation into account. 
The amount of transferred charge, $\Delta q_{\mathrm{L-Me}}$, is assumed to depend linearly on the inter-atomic distance, $r_{\mathrm{L-Me}}$
\begin{equation}\label{charge_transfer}
\Delta q_{\mathrm{L-Me}}  =  a_L r_{\mathrm{L-Me}} + b_L. 
\end{equation}

where $a_L$ and $b_L$ are parameters to be determined that are specific for pairs of ligand $L$ (O, S, N) and a metal cation.
The parameters $a_L$ and $b_L$ are of opposite sign, so that the magnitude of charge transfer decreases with distance.
The distance at which $\Delta q_{\mathrm{L-Me}}$ becomes $0$ is

\begin{equation}
    r_{\mathrm{L-Me}}^0 = -\frac{b_L}{a_L}
\end{equation}

Beyond this distance, we assume charge transfer to be $0$. 
This approximates real-life charge transfer, which is generally negligible at distances greater than the sum of the vdW radii of atoms $i$ and $j$, $r^{\mathrm{vdW}}_{ij}$.


Thus, charge on ligand atom $L$, $q_L$, can be calculated as
\begin{equation}
q_L  =  q_L^0 + \Delta q_{\mathrm{L-Me}},
\end{equation}
where $q_L^0$ refers to the charge on atom $L$ in a fixed-charge model.

Polarization energy, $E^{\mathrm{pol}}_r$, can be computed as
\begin{equation}
E^{\mathrm{pol}}_r = -\frac{1}{2}\sum_{i} \bm{\mu_i \cdot {E}^0_i},
\end{equation}
where \bm{$\mu_i$} is the induced dipole on atom $i$ and \bm{$E^0_i$} is the electrostatic field produced by the current charge distribution in the system at the polarizable site \textit{i}. 
The summation is over the metal and the metal-bonded residues. 
A cutoff distance $r^{\mathrm{cutoff}}$, which is equal to the sum of the vdW radii of atoms $i$ and $j$ scaled by a parameter $\gamma$ = 0.92, is introduced to avoid unphysically high induced dipoles at close distance. 
If the distance between atom $i$ and $j$, $r^{ij}$, is smaller than $r^{\mathrm{cutoff}}$, we set $r^{ij}$ equal to $r^{\mathrm{cutoff}}$.
The only parameter here is the atomic polarizability:
\begin{equation}\label{pol}
\bm{\mu_i} = \alpha_i \bm{E_i},  
\end{equation}
where $E_i$ is the total electrostatic field on atom $i$ due to the charges and induced dipoles in the system.

In this work, we have implemented the CTPOL model in OpenMM via a python script, which can be found at \url{https://github.com/XiaojuanHu/CTPOL_MD}. \cite{Hu2023CTPOLMD}
This represents a proof-of-concept implementation, which runs on CPUs. Further code optimization and a transfer to GPUs will likely speed up simulations substantially.

\subsection{Reference data set} \label{ref_stuctures}

To evaluate the performance of the parameterization protocol on dipeptide and dipeptide-cation systems, we created a quantum chemistry data set. 
The data set consists of six models: (1) AcAla$_{2}$NMe (231 conformers); (2) AcAla$_{2}$NMe+Na$^+$ (327 conformers); (3) deprotonated cysteine: AcCys$^-$NMe (77 conformers), which often acts as  interaction center in metalloproteins; (4) AcCys$^-$NMe+Zn$^{2+}$ (261 conformers); (5) AcCys$^{-}_{2}$NMe+Zn$^{2+}$ (475 conformers), and (6) AcHisDNMe+Zn$^{2+}$ (209 conformers). 
The structures and energy hierarchies are shown in Figure \ref{Reference_data}. 
The data set can be found on the NOMAD repository via the DOI: \url{10.17172/NOMAD/2023.02.03-1}. \cite{Hu2023NOMAD}

%\begin{table}
%  \caption{Number of conformers of reference data}
%  \label{tbl:numberOfConf}
%  \begin{tabular}{lr}
%    \hline
%    System  & Number of conformers  \\
%    \hline
%    AcAla$_{2}$NMe   & 231   \\
%    AcAla$_{2}$NMe+Na$^+$ & 327  \\
%    AcCys$^-$NMe  & 77  \\
%    AcCys$^-$NMe+Zn$^{2+}$ & 261 \\
%    AcCys$^{-}_{2}$NMe+Zn$^{2+}$ & 475 \\
%    AcHisDNMe+Zn$^{2+}$ & 209 \\
%    \hline
%  \end{tabular}
%\end{table}

\begin{figure}[htbp]
\vspace{0.0in}
\begin{center}
\includegraphics[width=0.75\textwidth]{./Figure_reference_data_sum_plus2.pdf}
\end{center}
\vspace{-0.2in}
\setlength{\belowcaptionskip}{0.2in}
\caption{Structures and energy hierarchies of reference data in this study. The numbers of conformers are stated at the top of the individual columns.}
%Structures in this figure are local minima of relevant system.
\label{Reference_data}
\vspace{-0.2in}
\end{figure}

All DFT calculations in this work were performed with the numeric atom-centered basis set all-electron code FHI-aims. \cite{BLUM20092175, HAVU20098367, Ren_2012}
The PBE \cite{PhysRevLett.77.3865} generalized-gradient exchange-correlation functional augmented by the correction of van der Waals interactions using the Tkatchenko-Scheffler formalism \cite{PhysRevLett.102.073005} (PBE+vdW$^{TS}$) was employed. 
The choice of functional has been validated in previous articles. \cite{ropo2016first, ropo2016trends} 
For each conformation, several types of partial charges were provided. 
Hirshfeld charges \cite{Bultinck2007} are derived based on the Hirshfeld partitioning scheme. \cite{Hirshfeld1977, Bultinck2007} 
ESP charges \cite{Bultinck2007, Singh1984} are derived by fitting partial charges to reproduce the electrostatic potential. 
RESP charges \cite{Bayly1993} are extracted by a two-stage restrained electrostatic potential (RESP) fitting procedure \cite{Bayly1993} within the Antechamber suite of the AmberTools package. \cite{doi:10.1002/wcms.1121} 
The electrostatic potential was evaluated on a set of grids in a fixed spatial region located in a cubic space around the molecule. 
The 5 radial-shells were generated in a radial region between 1.4 and 2.0 multiples of the atomic vdW radius.
The cubic space contains 35 points along $x$, $y$, and $z$ directions, respectively.

The conformers of AcAla$_{2}$NMe, AcAla$_{2}$NMe+Na$^+$, and AcHisDNMe+Zn$^{2+}$ were obtained by a conformational search algorithm as shown in the studies of Rossi \textit{et al.} \cite{Rossi2014} and Schneider \textit{et al}. \cite{schneider2018relative} 
First, a global conformational search was performed with the basin-hopping approach \cite{J.Wales1997, Wales1368} at the force-field level (OPLS-AA). \cite{Jorgensen1996} 
The scan program of the TINKER molecular modeling package \cite{doi:10.1002/jcc.540080710, Ren2003} was employed to perform the basin-hopping search strategy. 
An energy threshold of 100 kcal/mol for local minima and a convergence criterion for local geometry optimizations of 0.0001 kcal/mol were used. 
All obtained conformers were relaxed at PBE+vdW$^{TS}$ level with \textit{tier 1} basis set and \textit{light} setting employed. 
%Relaxation was accomplished using a trust radius method version of the Broyden-Fletcher-Goldfarb-Shanno (BFGS) optimization algorithm. \cite{Nocedal2006} 
A clustering scheme was then applied to exclude duplicates using the root-mean-square deviations (RMSD) of atomic positions. 
%Clustering was performed by applying the Unweighted Pair Group Method with Arithmetic Mean (UPGMA) \cite{sokal58} method implemented in Python’s SciPy \cite{article} library. 
Finally, further relaxation was accomplished at the PBE+vdW$^{TS}$ level using \textit{tier 2} basis set and \textit{tight} setting. 

%After clustering, 231, 327 and 209 conformers were obtained for AcAla$_{2}$NMe, AcAla$_{2}$NMe+Na$^+$, and AcHisDNMe+Zn$^{2+}$, respectively.

The conformers of AcCys$^-$NMe, AcCys$^-$NMe+Zn$^{2+}$, and AcCys$^{-}_{2}$NMe+Zn$^{2+}$ were obtained with the genetic algorithm (GA) package Fafoom.\cite{Supady2015} 
First, a GA search at the PBE+vdW$^{TS}$ level with \textit{light} basis set was employed for structure sampling. 
Then a clustering scheme with a clustering criterion of RMSD of 0.02 {\AA} for atomic positions and a relative energy of 0.02 kcal/mol was applied to remove duplicates. 
The obtained conformers were further relaxed with FHI-aims \cite{BLUM20092175, HAVU20098367, Ren_2012} at the PBE+vdW$^{TS}$ level with \textit{tight} basis set. 
Final conformers were obtained after clustering. 
%This procedure resulted in 77, 261, and 475 conformers for AcCys$^-$NMe, AcCys$^-$NMe+Zn$^{2+}$ and AcCys$^{-}_{2}$NMe+Zn$^{2+}$, respectively. 
Both conformational search protocols have been well validated. \cite{Rossi2014, Supady2015}

%Number of conformers of each system obtained from the conformational sampling described above are listed in Table \ref{tbl:numberOfConf}

%\subsection{Training set}

%To cover as many structural characters as possible, we employed farthest point sampling\cite{Gonzalez1985, Moenning2003} to select the training set. Following references \cite{De2017, doi:10.1002/qua.26369}, local atom-centered environments of the structures were represented with the smooth overlap of atomic positions (SOAP)\cite{Bartok2017} descriptors. SOAP descriptors were obtained with a cutoff of 5.0 \AA, a Gaussian broadening of $\sigma$ = 0.5 \AA  and an intermediate regularization parameter $\gamma$ =0.01. The similarity matrix between different conformers were obtained through the regularized entropy match kernel (REMatch)\cite{C6CP00415F}. SOAP kernels were only calculated for heavy atoms in the molecule (disregarding metal hydrogen atoms) and were obtained using the GLOSIM package\cite{C6CP00415F}.

\subsection{Parameter optimization} \label{Method-optimization}

Optimization methods used in this work include LASSO (least absolute shrinkage and selection operator) \cite{Tibshirani1996} regression, Ridge regression \cite{doi:10.1080/00401706.1970.10488634} and particle swarm optimization (PSO). \cite{Shi1945, Koh2007} 
%Although we did a throughout conformational search on the potential energy surface (PES), there is still a danger of overfitting.
If the parameters enter the force-field function in a quadratic way, e.g. $V_n^{ij}$, the optimization can be performed by solving a set of linear equations.
In this case, LASSO and Ridge regression were employed to treat the potential overfitting. 
The regularization parameter $\lambda$ in LASSO and Ridge regression was selected by 10-fold cross-validation. 
LASSO and Ridge regression were performed with Python’s scikit-learn \cite{pedregosa2011scikit} library. 
If the parameters can not be obtained by solving a set of linear equations, e.g. the charge transfer parameters $a_L$, PSO was employed.
PSO is a powerful population-based global optimization algorithm. 
It relies on a population of candidate solutions, called particles, and finds the optimal solution by moving these particles through a high-dimensional parameter space based on their position and velocity. 
PSO was performed with the python package pyswarm. \cite{Lee2014}

\subsection{FFAFFURR} \label{Method-FFAFFURR}
%\xhu{Dennis thinks this part was not clear.}
 
%FFAFFURR is written in python and acts as a “wrapper” between the molecular modeling package openMM and the \textit{ab initio} molecular simulations package FHI-aims.
% as shown in Figure \ref{framework_ffaffurr}. 
%The output of FFAFFURR is a file of optimized parameters that can be processed directly by openMM.
%\xhu{How to refine this paragraph? Or do we need this paragraph?} 

Force field parameterization is an optimization problem with three challenging aspects: \cite{Aktulga2015}
\begin{enumerate}
%\item [1)] Definition of the optimization problem (selection of reference data, objective of the optimization, and force-field parameters to adjust):  High quality QM data has been used for FF parameterization and is likely continue to be an essential part of next-generation FF development. \cite{wang2017building} FFAFFURR uses high quality QM data as described in section \ref{ref_stuctures} as the reference. In principle, the parameters of every energy term in a force-field have to be optimized since the parameters of all terms are interdependent, only adjusting one energy term may cause parameter inconsistency.  
%FFAFFURR supports full optimization of force-fields. 
%Users can choose which energy terms to tune according to specific problems.  OPLS-AA parameters are used as initial parameters. 
\item The optimization problem has to be defined, which consists of the objective of the optimization and, following this, the selection of training data as well as force-field parameters to adjust.

\item In order to perform the force-field parameterization, a preferably automated procedure has to be implemented. The framework and algorithms used in FFAFFURR are explained in this paper. 
\item The obtained set of force-field parameters has to be validated against other data than the training data. 
\end{enumerate}

\noindent
Regarding item 1, the parameters of every energy term in a force-field have to be optimized since energy terms and parameters are interdependent and only adjusting a subset may cause inconsistency. 
Items 1 and 3, training and validation, heavily rely on high-quality data. We use DFT data for comparing potential energies and further validate by MD simulation.

Some practical points were considered when establishing the FFAFFURR framework: (i) the framework should be straightforward to set up and use, (ii) it should be easy to extend with other FF parameters or functional forms, and (iii) the result should be immediately usable by a molecular simulation package. 
FFAFFURR acts as a “wrapper” between the molecular mechanics package openMM \cite{Eastman2017} and the \textit{ab initio} molecular simulation package FHI-aims. \cite{BLUM20092175, HAVU20098367, Ren_2012} 
The code reads QM data directly from the output of FHI-aims and the output itself is a parameter file that can be processed by openMM. 
FFAFFURR is designed as the next step of the genetic algorithm package Fafoom. \cite{Supady2015} 
Conformers obtained by Fafoom through global search can be directly parsed to FFAFFURR. 
FFAFFURR is an open source tool and can be found at \url{https://github.com/XiaojuanHu/ffaffurr-dev/releases/tag/version1.0}.
%In this work, the transferability of force-field is not our major goal since it's easy to derive parameters for different systems and to extend the framework to set up a transferable FF.

%\begin{figure}[htbp]
%\vspace{0.0in}
%\begin{center}
%\includegraphics[width=0.75\textwidth]{./Figure_ffaffurr.pdf}
%\end{center}
%\vspace{-0.2in}
%\setlength{\belowcaptionskip}{0.2in}
%\caption{The concept of FFAFFURR. CB: figure will eventually be recycled for creating the TOC picture.}
%\label{framework_ffaffurr}
%\vspace{-0.2in}
%\end{figure}


%The input to FFAFFURR consists of four files: \textbf{ffaffurr.input}, \textbf{OPLS-AA.xml}, \textbf{input.pdb} and \textbf{ffaffurr.input.FHI-aims-logfiles}. 
%The file \textbf{ffaffurr.input} contains the “switches” that control the behavior of the framework, for example, parameters of which energy term to be tuned or which fitting method to use. 
%One can choose to do a global optimization or tune several energy terms in ffaffurr.input. 
%\textbf{OPLS-AA.xml} provides original force-field parameters. \textbf{input.pdb} denotes the structural information of the system. \textbf{ffaffurr.input.FHI-aims-logfiles} contains a list of FHI-aims-specific output files of conformers in training set to provide DFT information ( conformational energies, partial charge, \textit{etc} ). QM information in \textbf{ffaffurr.input.FHI-aims-logfiles} can be parsed to FFAFFURR directly. 
%If the QM interaction energies is needed in the parameterization,  \textbf{energies\_inter\_DFT.dat} file which contains DFT interaction energies needs to be provided.
%The output of FFAFFURR is a file of optimized parameters that can be processed directly by openMM.

%\subsection{force-field parameterization}\label{parameterization}

%FFAFFURR supports a large number of functions for the parameterization. We will only focus on functions explained in this section. Further details of FFAFFURR’s other support functions will not be discussed here (see the documents in FFAFFURR github link: ....)

\subsubsection*{Bond and angle parameterization}

$K_{ij}^{r}$, $K_{ij}^{\theta}$, $r_{ij}^0$ and $\theta_{ij}^0$ are empirical parameters of bond-stretching and angle-bending terms. 
The “spring” parameters $K_{ij}^{r}$ and $K_{ij}^{\theta}$ are unaltered in FFAFFURR.
The focus simply lies on the “torsional” and “non-bonded” parameters.
Bond-stretching and angle-bending terms intend to model small displacements away from the lowest energy structure. 
We adjust $r_{ij}^0$ and $\theta_{ij}^0$ by simply taking the average of the respective bond or angle over all local minima in the quantum chemistry data set.

\subsubsection*{Torsion angle parameterization}

The torsion angle term represents a combination of the bonded and nonbonded interactions. 
It has been reported that torsional parameters fitted to gas phase QM data perform similarly to those fitted to the experimental data. \cite{wang2017building} 
Although torsional parameters can be derived from vibrational analysis or using vibrational spectra as target data, this approach is complicated and requires a more elaborate treatment. \cite{pulay1979systematic, kumar2020ffparam, mayne2013rapid} 
In the case of the torsion term, force constants $V_n^{ij}$ and $V_{2imp}^{ij}$ can be tuned by LASSO or Ridge regression to minimize the difference between the FF and QM torsional energies. 
The ``torsions contribution'' from QM $\widetilde E^{\mathrm{QM}}_{\mathrm{torsions}}$ is calculated as:
\begin{equation}
\widetilde E^{\mathrm{QM}}_{\mathrm{torsions}} = E^{\mathrm{QM}}_{\mathrm{total}} - E^{\mathrm{FF}}_{\mathrm{nonbonded}}  - E^{\mathrm{FF}}_{\mathrm{bond}} - E^{\mathrm{FF}}_{\mathrm{angle}},
\end{equation}
where $E^{\mathrm{QM}}_{\mathrm{total}}$ represents the total energy of a conformer from a QM calculation, $E^{\mathrm{FF}}_{\mathrm{nonbonded}}$, $E^{\mathrm{FF}}_{\mathrm{bond}}$ and $E^{\mathrm{FF}}_{\mathrm{angle}}$ represent energies of nonbonded terms, bonded terms, and angle terms from FF calculations, respectively.
%The scanning of dihedrals was performed on the QM data set. 
%Such a scan provides dihedral values, corresponding energies, and identifies missing torsional parameters in FF. 
%Users can choose to fit the missing torsional parameters with the existing torsinal parameters, or set missing torsional parameters to zero.

\subsubsection*{Electrostatic parameterization}\label{ele_param}

A key difference between FF parameter sets is the origin of the atomic partial charges. 
Deriving charges from QM data is widely used. 
The workflow of FFAFFURR tested three choices of partial charges: Hirshfeld, \cite{Hirshfeld1977, Bultinck2007} ESP \cite{Bultinck2007, Singh1984} and RESP \cite{Bayly1993} charges. 
%Independent of using the Hirshfeld, ESP, or RESP charges, partial charge is defined as the average of atoms with the same atom type in the quantum chemistry data set.
The charge of each atom type of the force-field is defined as the average value of QM charges.
The scaling factor $f_{ij}$ used to scale the electrostatic interactions between the third neighbors (1,4-interactions) can also be adjusted by fitting to minimize the difference between the FF and QM energies.

\subsubsection*{LJ parameterization} \label{LJ_para}

Pair-specific Lennard-Jones (LJ) interaction parameters (referred to as NBFIX in the CHARMM force-fields) have been proven to better describe the interaction between cations and carbonyl groups of a protein backbone. \cite{Li2015} 
FFAFFURR employs pairwise Lennard-Jones (LJ) parameters instead of values determined by the combination rule.

In recent years, progress has been made in the calculation of pairwise dispersion interaction strength from the ground-state electron density of molecules. \cite{klimevs2012perspective, reilly2015van, cole2016biomolecular} 
The interatomic pairwise parameter $\sigma_{ij}$ can be derived using the atomic Hirshfeld partitioning scheme, which has already been used in the pairwise Tkatchenko-Scheffler vdW model. 
With the concept of the vdW radius, the LJ energy can be written as
\begin{equation} \label{eq_FF_vdw}
E_{\mathrm{vdw}} = \sum_{i<j} \varepsilon_{ij} \left [ \left ( \frac{R^{\mathrm{min}}_{ij}}{r_{ij}} \right)^{12} - 2\left (\frac{R^{\mathrm{min}}_{ij}}{r_{ij}} \right)^{6} \right ] f_{ij}, 
\end{equation}
where $R^{\mathrm{min}}_{ij}$ refers to the atomic distance where the vdW potential is at its minimum. 
With the definition of the free and effective atomic volume $V^{\mathrm{free}}$ and $V^{\mathrm{eff}}$, $R^{\mathrm{min}}_{ij}$ is estimated as the sum of effective atomic van der Waals radii of atom $i$ and atom $j$. 
The effective vdW radius of an atom is given by
\begin{equation}\label{R_eff}
R_{\mathrm{eff}}^0 = \left ( \frac{V^{\mathrm{eff}}}{V^{\mathrm{free}}} \right ) ^{1/3}R_{\mathrm{free}}^0,
\end{equation}
where $R_{\mathrm{free}}^0$ is the free-atom vdW radii that correspond to the electron density contour value determined for the noble gas on the same period using its vdW radius by Bondi. \cite{Bondi1964}
Pairwise $\sigma_{ij}$ can be calculated as
\begin{equation}
\sigma_{ij} = 2^{-1/6}R^{\mathrm{min}}_{ij}.
\end{equation}
The $\varepsilon_{ij}$ parameter from eq. \ref{eq_FF_vdw} can be tuned by fitting FF LJ energies to reproduce QM vdW energies by LASSO or Ridge regression.  

\subsubsection*{Deriving charge transfer parameters}

In all Zinc-finger proteins and most enzymes, Zn$^{2+}$ coordinates to four ligands.
However, due to the setup of the QM data set with monomeric and dimeric peptides, the cations have coordination numbers (CNs) of one or two. 
%We tested the charge transfer from per sulfur atom to Zn$^{2+}$ using toy models as shown in SI \ref{Figure_models_CN}. 
%Figure SI \ref{CT_CN} shows that CN significantly affects the amount of transferred charge.
Therefore we added a correction factor for CN in eq. \ref{charge_transfer}
\begin{equation} \label{correction_ct}
\Delta q_{\mathrm{L-Me}}  = \frac{1}{\mathrm{CN}^k} ( a_L r_{\mathrm{Me-L}} + b_L ). 
\end{equation}
$k$, $a_L$, and $r^{\mathrm{cutoff}}$ can be adjusted by PSO. 
The target objective of fitting can be the QM potential energy, the QM interaction energy, or the electrostatic potential derived from electron densities. 
$b_L$ can be calculated with the assumption that charge transfer is zero at the cutoff distance.

\subsubsection*{Polarization energy}

To get the value of atomic polarizability $\alpha_i$ in eq. \ref{pol}, we use the definition of effective polarizability of an atom in a molecule, where the free-atom polarizabilitiy is scaled according to its close environment with a partitioning:
\begin{equation}
\alpha_{\mathrm{eff}} = \left (\frac{V^{\mathrm{eff}}}{V^{\mathrm{free}}} \right )\alpha^0_{\mathrm{free}},
\end{equation}
where $V^{\mathrm{eff}}$ and $V^{\mathrm{free}}$ are the same as in eq. \ref{R_eff}, and $\alpha^0_{\mathrm{free}}$ is the isotropic static polarizability. 
$\alpha_i$ is taken by averaging over all atoms with the same atom type in the quantum chemistry data set.
FFAFFURR also supports slightly adjusting $\alpha_i$ by fitting force-field energies to reproduce QM energies via PSO.

\subsubsection*{Boltzmann-type weighted fitting}

The quantum chemistry data set covers a wide range of relative energies.
By transitioning from, in our case, DFT to an additive force-field, even including charge transfer and polarization, we reduce dimensionality of the energy function and therewith the ability to correctly/fully represent the PES. Consequently, a force-field, describing, e.g., such a cation-protein system, cannot fully reproduce a DFT PES. Hence, it appears advisable to put focus on the accuracy of distinct areas of the PES.
RMSD between two surfaces is a common fitting criteria, but this approach gives more weight to areas of the energy surface with larger absolute values, while the real weight should more closely represent the Boltzmann weight of the energy surface. 
Consequently, we calculate Boltzmann-type weights and apply them as a scoring function.
The weighted RMSD, $wRMSD$, is given as:
\begin{equation}
wRMSD = \left [ \sum_{i=1}^{N} w_i ( E_i^{\mathrm{FF}} - \Delta E_i^{\mathrm{QM}}) ^2 \right ] ^{\frac{1}{2}},
\end{equation}
where RMSD is modified by including a Boltzmann-type factor,
\begin{equation}
w_i = A \exp \left [ \frac{-E_i^{\mathrm{QM}}}{\mathrm{RT}} \right],
\end{equation}
where A is the normalization constant (so that $\sum{w_i}$ = 1) and RT is a temperature factor that has no physical meaning in the context of this application, but affects the flatness of the distribution.
Our previous work \cite{Amin2020} has shown how Boltzmann-type weighted RMSD with an appropriate choice of RT can be utilized as the objective function for force-field parameter optimization. 
Therefore, we implemented Boltzmann-type weighted fitting in FFAFFURR by scaling the energies with the corresponding Boltzmann-type weights.

\subsection{Validation of new parameters}
\subsubsection*{Assessment of the energies} \label{Assess_energies}

To evaluate the performance of the parameterization, energies of conformers in the test set calculated with optimized parameters were compared to DFT energies by mean absolute errors (MAEs) and maximum errors (MEs). 
The MAE for the relative energies between FF energies and QM energies is calculated as
\begin{equation} 
\mathrm{MAE} = \frac{1}{N} \sum_{i=1}^{N} \arrowvert \Delta E_i^{\mathrm{FF}} - \Delta E_i^{\mathrm{QM}} + c \arrowvert \label{eq_MAE},
\end{equation}
where $N$ is the number of conformers in a given data set. 
$\Delta E_i$ refers to the energy difference between conformer $i$ and the lowest-energy conformer in the set. 
The adjustable parameter $c$ is used to shift the FF or QM energy hierarchies to one another to get the lowest MAE. 
ME is calculated as:
\begin{equation}
\mathrm{ME} =\max_{i \in N} \arrowvert \Delta E_i^{\mathrm{FF}} - \Delta E_i^{\mathrm{QM}} + c\arrowvert.
\end{equation}

\subsubsection*{Molecular dynamics simulations}

We performed MD simulations of the NMR structure 1ZNF \cite{lee1989three} with different parameter sets to evaluate the performance of FFAFFURR.
All MD simulations were performed using OpenMM7. \cite{Eastman2017} 
The structure of 1ZNF was placed in a cubic box of 68 {\AA} side length filled with TIP3P water. 
Four Cl$^{-}$ were added to neutralize the system. 
Then energy minimization was performed with the steepest descent minimization.
To equilibrate the solvent and ions around the protein, we continued 100 ps NVT and 100 ps NPT equilibration at a temperature of 300 K.
SHAKE constraints were applied to heavy atoms of the protein. 
Then independent MD simulations were performed with a time step of 2 fs. 
In all calculations, the long-range electrostatics beyond the cutoff of 12 {\AA} were treated with the Particle Mesh Ewald (PME) method. \cite{darden1993particle} The LJ cutoff was set to 12 {\AA}. 
The LJ and electrostatic interactions were computed every time step. 
For the simulations with the CTPOL model, charge transfer and induced dipoles were updated every 10 steps.
Covalent bonds and water angles were constrained. 


\section{Results and discussion}

To assess the performance of FFAFFURR and describe which protocol to use to create a parameter set, we optimized the parameters of OPLS-AA with FFAFFURR and extended the OPLS-AA model by the CTPOL model. 
The quality of optimized parameters was assessed by examining the structural stability of the Zinc-finger motif in MD simulations.
%This study focuses on several functions of FFAFFURR to assess the performance and robustness of FFAFFURR. 

\subsection{OPLS-AA parameterization}

Although studies have shown that it is difficult to implicitly incorporate the polarization effects into classical FFs, \cite{zhang2012modeling, Amin2020} fine-tuning parameters of fixed-charge models to describe cation-protein systems is still attractive due to its low computational cost and easier parameterization. 
Here we tested the performance of the fixed-charge model OPLS-AA parametrized using FFAFFURR. 
Five systems were tested: (1) AcAla$_{2}$NMe; (2) AcAla$_{2}$NMe+Na$^+$; (3) AcCys$^-$NMe; (4) AcCys$^-$NMe+Zn$^{2+}$; and (5) AcCys$^{-}_{2}$NMe+Zn$^{2+}$. 
AcAla$_{2}$NMe and AcAla$_{2}$NMe + Na$^+$ were used as reference models since the charge transfer and polarization effects caused by Na$^+$ are less than that of Zn$^{2+}$.
On the contrary, Cys$^{-}$ is one of the ligands that interact with Zn$^{2+}$ in proteins, and charge transfer between Cys$^{-}$ and Zn$^{2+}$ is significant.
For each system, 80 percent of the conformers were randomly selected as the training set, and the remaining 20 percent were used as the test set.

We first demonstrate the functionality of FFAFFURR on the example of OPLS-AA parameterization. The key steps of OPLS-AA parameterization are briefly described in Figure \ref{full-opt-oplsaa} (a). 
We showed the ability to reproduce PES by optimizing parameters of bonds, angles, electrostatic interactions, LJ interactions, and torsional interactions. 
Users can choose which energy items to adjust according to their needs. 
In Figure \ref{full-opt-oplsaa} (a), the parameters in blue boxes are derived from DFT calculations and the parameters in red boxes are fitted by LASSO or Ridge regression as described in Sections \ref{Method-optimization} and \ref{Method-FFAFFURR}. 
Here, we only tested RESP partial charges, the LASSO method in deriving $\varepsilon_{ij}$, and Ridge regression in deriving $V_{n}^{ij}$. 
%The parameterization protocol followed the order shown in Figure \ref{full-opt-oplsaa} (a).
The parameters derived from DFT calculations are tuned first because they are considered fixed with respect to changes of the other parameters, then different tuning orders of the parameters for the other energy terms in the FF formula are tested to choose the order that gives the smallest errors between DFT and FF energies. The final order of the protocol is shown in Figure \ref{full-opt-oplsaa} (a).

Figure \ref{full-opt-oplsaa} (b-f) shows the comparison of FF energies with optimized parameters after each step in Figure \ref{full-opt-oplsaa} (a).
Noticeably, charges for AcAla$_2$NMe, AcCys$^-$NMe and AcAla$_2$NMe+Na$^+$ were not altered since the original charges yielded errors lower than average RESP charges from QM calculations, while average RESP charges were employed for AcCys$^-$NMe+Zn$^{2+}$ and AcCys$^{-}_{2}$NMe+Zn$^{2+}$. 
Figure \ref{full-opt-oplsaa} (e) and (f) indicate that using average RESP charges significantly reduces absolute errors for AcCys$^-$NMe+Zn$^{2+}$ and AcCys$^{-}_{2}$NMe+Zn$^{2+}$. 
This could be due to the capture of charge transfer to some extent. 
In the case of AcAla$_2$NMe and AcCys$^-$NMe, the MAEs were improved from \mbox{2.72 kcal/mol} and 3.59 kcal/mol to 0.61 kcal/mol and 0.98 kcal/mol, respectively, which is well within the 1 kcal/mol chemical accuracy. 
In the case of AcAla$_2$NMe+Na$^+$, the MAE was improved from 3.99 kcal/mol to 1.67 kcal/mol. 
Although the optimized MAE is above the chemical accuracy, the maximum error is significantly reduced. 
However, in the cases of AcCys$^-$NMe+Zn$^{2+}$ and AcCys$^{-}_{2}$NMe+Zn$^{2+}$, the MAEs were improved from 51.75 kcal/mol and 43.47 kcal/mol to 16.8 kcal/mol and 16.59 kcal/mol, respectively. 
Although these are by numbers great improvements, the MAEs are much higher than for the other systems. Calculations based on parameters of such quality have no predictive power.
This confirms the necessity of explicitly including charge transfer and polarization effects to describe divalent ion-dipeptide systems.
We note that for dipeptides and dipeptides with monovalent cation systems, optimization of torsional parameters has the greatest impact on the improvement of the accuracy.
Previous studies by some of us \cite{baldauf2013cations, ropo2016trends} have shown that mono- and divalent cations strongly modify the preferences of torsion angles. 
While for dipeptides with divalent cations, apparently, the adjustment and treatment of charge interactions plays the most important role. 
This further confirms that the capture of charge transfer and polarization is crucial for the accurate description of systems with divalent cation. 
We also note that the maximum errors are greatly reduced after the parameterization of LJ interactions of the five systems.

\begin{figure}
\vspace{0.0in}
\begin{center}
\includegraphics[width=1.0\textwidth]{./Figure_OPLS-param2.pdf}
\end{center}
\vspace{-0.2in}
\setlength{\belowcaptionskip}{0.2in}
\caption{(a) Workflow of the parameterization of OPLS-AA in four major steps. Different colors represent different fitting methods. Parameters in blue boxes are derived from DFT calculation, and parameters in red boxes are tuned by LASSO or Ridge regression. (b-f) Box plots of absolute errors of OPLS-AA parameterization major steps (OPLS-AA, step 1, step 2, step 3, step 4) for the test set of (b) AcAla$_2$NMe, (c) AcCys$^-$NMe, (d) AcAla$_2$NMe+Na$^+$, (e) AcCys$^-$NMe+Zn$^{2+}$ and (f) AcCys$^{-}_{2}$NMe+Zn$^{2+}$. The upper and lower lines of the rectangles mark the 75\% and 25\% percentiles of the distribution, the horizontal line in the box indicates the median (50 percentile), internal colored dashed line indicate the mean value, and the upper and lower lines of the “error bars” depict the 99\% and 1\% percentiles. The crosses represent the outliers. Black dashed line indicates the chemical accuracy, which is 1 kcal/mol. Note the large differences in scales in subfigures (b) to (f).
}
\label{full-opt-oplsaa}
\vspace{-0.2in}
\end{figure}

\subsection{CTPOL parameterization}

The CTPOL model introduces both local polarization and charge-transfer effects into classical force-fields. 
We investigated the performance of the CTPOL model on the cation-dipeptide systems: AcAla$_2$NMe+Na$^+$, and two challenging systems AcCys$^-$NMe+Zn$^{2+}$ and AcCys$^{-}_{2}$NMe +Zn$^{2+}$. 
The major steps of the CTPOL parameterization workflow are depicted in Figure \ref{full-opt-ctpol} (a). 
Following the methodology of OPLS-AA optimization, parameters unaffected by others are adjusted first, then different orders are tested to choose the order that gives smallest errors between DFT and FF energies. Charges are taken from OPLS-AA in step 1 to step 3. In step 4, charge transfer was introduced.
As already mentioned, the parameters in blue boxes are derived from DFT calculations and the parameters in red boxes are fitted by LASSO or Ridge regression. 
The parameters in green boxes are obtained by PSO. 
Noticeably, $\alpha_i$ is tuned twice. 
In step 3, $\alpha_i$ is taken as the average effective polarizability calculated from the \textit{ab initio} method. 
In step 5, we tried to slightly tune $\alpha_i$ by PSO.
An additional round of parameterization from step 4 to step 5 can be performed to better optimize the FF parameters.

Absolute errors of each step in Figure \ref{full-opt-ctpol} (a) are illustrated in Figures \ref{full-opt-ctpol} (b-d). 
Absolute errors of optimized OPLS-AA (opt-opls) are also shown in Figure \ref{full-opt-ctpol} to compare the performance of FFAFFURR on OPLS-AA and CTPOL models. 
As shown in Figure \ref{full-opt-ctpol}, the introduction of polarization effects in step 3 didn't improve the accuracy much, and the errors of the AcAla$_2$NMe+Na$^+$ system even increased. 
This may be due to the fact that classical force-fields already take some account of polarization effects, since the charges come from fitting to reproduce quantum mechanical or experimental electrostatic field distributions. \cite{sakharov2009force} 
Including charge transfer from ligand atoms to the cation reduces atomic charges, therefore compensating for the electrostatic potential. 
Not surprisingly, errors are significantly reduced after including charge transfer as displayed in Figure \ref{full-opt-ctpol}. 
After the parameterization, the MAEs of AcAla$_2$NMe+Na$^+$, AcCys$^-$NMe+Zn$^{2+}$ and AcCys$^{-}_{2}$NMe+Zn$^{2+}$ reached 1.45 kcal/mol, 7.42 kcal/mol, and 8.12 kcal/mol, respectively. 
In contrast, the MAEs of the optimized OPLS-AA are 1.67 kcal/mol, 16.8 kcal/mol, and 16.59 kcal/mol, respectively. 
Apparently, the inclusion of charge transfer and polarization effects better describes systems involving cations than classical force-fields, especially for systems with divalent cations.

\begin{figure}
\vspace{0.0in}
\begin{center}
\includegraphics[width=1\textwidth]{./Figure_CTPOL-param2.pdf}
\end{center}
\vspace{-0.2in}
\setlength{\belowcaptionskip}{0.2in}
\caption{ (a) Workflow of full CTPOL parameterization in five major steps. Different colors represent different fitting methods. Parameters in blue boxes are derived from DFT calculation, parameters in red boxes are tuned by LASSO or Ridge regression, and parameters in green boxes are tuned by PSO. (b-d) Box plots of absolute errors of CTPOL parameterization major steps (OPLS-AA, step 1, step 2, step 3, step 4, step 5) and OPLS-AA with full optimized parameters (opt-opls) for test set of (b) AcAla$_2$NMe+Na$^+$, (c) AcCys$^-$NMe+Zn$^{2+}$ and (d) AcCys$^{-}_{2}$NMe+Zn$^{2+}$. The upper and lower lines of the rectangles mark the 75\% and 25\% percentiles of the distribution, the horizontal line in the box indicates the median (50 percentile), internal colored dashed line indicate the mean value, and the upper and lower lines of the “error bars” depict the 99\% and 1\% percentiles. The crosses represent the outliers. Black dashed line indicates the chemical accuracy, which is 1 kcal/mol.}
\label{full-opt-ctpol}
\vspace{-0.2in}
\end{figure}

%\subsection{Weighted fitting}

To focus the fitting on the low-energy part of the PES, we applied Boltzmann-type weights to the scoring function during the fitting of the charge transfer parameters. 
In Figure \ref{BW_Cys-Zn}, the AcCys$^-$NMe+Zn$^{2+}$ system is taken as an example. 
Figure \ref{SI_BW_Cys-Zn} shows the Boltzmann-type weights ($w_i$) along QM relative energies with different temperature factor (RT) values. 
For all RT, the weight decreases as the relative energy increases, 
but increasing RT decreases the weights on low-energy conformations. 
Figure \ref{BW_Cys-Zn} shows the difference in mean absolute errors between unweighted fitting and weighted fitting with RT = 16. 
In Figure \ref{BW_Cys-Zn}, the height of the bar represents the mean absolute error for conformers whose relative energies are smaller than the right node of the bar.
%Weighted fitting yielded lower errors than unweighted fit in low-energy regions.
Interestingly, the weighted fitting improves accuracy substantially in the low-energy region, while high-energy regions do not get worse.
%Thus, one can balance the errors on the PES with appropriate RT value.

\begin{figure}[htbp]
\vspace{0.0in}
\begin{center}
\includegraphics[width=0.75\textwidth]{./Figure_BW_ctpol_RT16_additive2.pdf}
\end{center}
\vspace{-0.2in}
\setlength{\belowcaptionskip}{0.2in}
\caption{Absolute errors of optimized FF energies with respect to QM energies by weighted/unweighted fitting of AcCys$^-$NMe+Zn$^{2+}$ system. The height of the bar represents the mean absolute error for conformers whose relative energies are smaller than the right node of the bar.}
\label{BW_Cys-Zn}
\vspace{-0.2in}
\end{figure}

\FloatBarrier

%\subsubsection{charge transfer}

%A Figure of PSO convergence


%\subsubsection{Subgroup discovery}

%To investigate the reason of high errors of Zn$^{2+}$ involved systems and hopefully further reduce the error, we analyzed the structure space of AcCys$^-$NMe-Zn$^{2+}$ and AcCys$^{-}_{2}$NMe-Zn$^{2+}$ and errors of OPLS-AA on these two systems. Data sets of these two systems have high diversity. Relative energies are up to about 100 kcal/mol. Figure \ref{structural_map} (c) and (d) shows that Zn$^{2+}$ ion is gradually released from sulfur with the increase of relative energies. According to this process, data sets can be briefly divided into subgroups based on the interactions between Zn$^{2+}$ and sulfur. In the case of AcCys$^-$NMe-Zn$^{2+}$, data set can be divided as (1) Zn$^{2+}$ bind with sulfur, which can be labeled as Zn-S; (2) Zn$^{2+}$ released from sulfur, which can be labeled as Zn-\-S. In the case of AcCys$^{-}_{2}$NMe-Zn$^{2+}$, data set can be divided as (1) Zn$^{2+}$ bind with two sulfur, which can be labeled as S-Zn-S; (2) Zn$^{2+}$ bind with one sulfur, which can be labeled as Zn-S. Here, structures in which Zn$^{2+}$ is released from sulfur are not included since their relative energies are very high. Furthermore, Figure \ref{structural_map} (a) and (b) displayed structural map of AcCys$^-$NMe-Zn$^{2+}$ and AcCys$^{-}_{2}$NMe-Zn$^{2+}$ which have Zn-S interactions according to distance between Zn$^{2+}$ and sulfur and sum of angles around nitrogen. Plots are coloured by absolute errors of OPLS-AA. Structural map of AcCys$^{-}_{2}$NMe-Zn$^{2+}$ which have S-Zn-S interactions are displayed in SI \ref{struc_map_S-Zn-S}. Conformers in this set are rather stable since Zn$^{2+}$ are fixed between to sulfur atoms. The sum of angles around nitrogen represents degree of out of plane deformations. The smaller the value, the higher the degree of deformation. The out-of-plane distortions of planar groups are treated by improper dihedral-angle bending terms. we noted that conformers in Figure \ref{structural_map} (a) and (b) can be divided into three subgroups according to the errors of OPLS-AA on it. Eventually, data sets of AcCys$^-$NMe-Zn$^{2+}$ and AcCys$^{-}_{2}$NMe-Zn$^{2+}$ are divided into four sets as shown in Figure \ref{structural_map} (c) and (d). Based on this, we tried to optimized parameters for each subgroups in order to further reduce FF errors. We showed two ways to define subgroups in Table \ref{tbl:subgroup}. In Method 1, we put set 3 and set 2 of AcCys$^-$NMe-Zn$^{2+}$ together since set 3 only includes 11 structures. One requirement that subgroup should satisfy is that the subgroup should not be too small to avoid too less information. We also put set 3 and set 4 of AcCys$^{-}_{2}$NMe-Zn$^{2+}$ together for the consistency. Method 2 divided data sets according to the number of sulfur atoms interacted with Zn$^{2+}$. Figure \ref{MAEs_subgroup} displayed MAEs of different sets of AcCys$^-$NMe-Zn$^{2+}$ and AcCys$^{-}_{2}$NMe-Zn$^{2+}$ with different subgroup dividing methods. We noted that the accuracy of CTPOL can be further improve with subgroup dividing.

%\begin{figure}
%\vspace{0.0in}
%\begin{center}
%\subfigure[AcCys$^-$NMe-Zn$^{2+}$]{\includegraphics[width=0.45\textwidth]{./Figure_cys-Zn_struc_map.pdf}}
%\subfigure[AcCys$^{-}_{2}$NMe-Zn$^{2+}$]{\includegraphics[width=0.45\textwidth]{./Figure_cys2-Zn_struc_map_coor1.pdf}}
%\subfigure[AcCys$^-$NMe-Zn$^{2+}$]{\includegraphics[width=0.45\textwidth]{./Figure_cys-Zn_subgroup_ene.pdf}}
%\subfigure[AcCys$^{-}_{2}$NMe-Zn$^{2+}$]{\includegraphics[width=0.45\textwidth]{./Figure_cys2-Zn_subgroup_ene.pdf}}
%\end{center}
%\vspace{-0.2in}
%\setlength{\belowcaptionskip}{0.2in}
%\caption{Structural map of conformers of (a) AcCys$^-$NMe-Zn$^{2+}$ and (b) AcCys$^{-}_{2}$NMe-Zn$^{2+}$ which have 'Zn-S' interactions. Points are coloured by absolute errors of OPLS-AA. X axis of (a) and (b) are the minimum of the sum of angles around nitrogen. (c) and (d) are subgroups of AcCys$^-$NMe-Zn$^{2+}$ and AcCys$^{-}_{2}$NMe-Zn$^{2+}$. Numbers in (c) and (d) are represent number of conformers of each set.}
%\label{structural_map}
%\vspace{-0.2in}
%\end{figure}

%\begin{table}
%  \caption{Subgroup discovery}
%  \label{tbl:subgroup}
%  \begin{tabular}{lll}
%    \hline
%      & AcCys$^-$NMe-Zn$^{2+}$ &  AcCys$^{-}_{2}$NMe-Zn$^{2+}$ \\
%    \hline
%    Method 1   & set 1, set 2\&set 3, set 4 & set 1, set 2, set 3\&set %4   \\
%    Method 2   & set 1\&set 2\&set 3, set 4 & set 1, set 2\&set 3\&set %4  \\
%    \hline
%  \end{tabular}
%\end{table}

%\begin{figure}
%\vspace{0.0in}
%\begin{center}
%\subfigure[AcCys$^-$NMe-Zn$^{2+}$]{\includegraphics[width=0.45\textwidth]{./cys-Zn_subgroup_MAE.pdf}}
%\subfigure[AcCys$^{-}_{2}$NMe-Zn$^{2+}$]{\includegraphics[width=0.45\textwidth]{./cys2-Zn_subgroup_MAE.pdf}}
%\end{center}
%\vspace{-0.2in}
%\setlength{\belowcaptionskip}{0.2in}
%\caption{MAEs of each subset with different subgroup definition. (a) AcCys$^-$NMe-Zn$^{2+}$ and (b) AcCys$^{-}_{2}$NMe-Zn$^{2+}$. Black dashed line indicates the chemical accuracy, which is 1 kcal/mol.}
%\label{MAEs_subgroup}
%\vspace{-0.2in}
%\end{figure}


\subsection{Validation with molecular dynamics simulations}
%\ksa{
%textwidth in in: \printinunitsof{in}\prntlen{\textwidth}
%}

% \begin{figure}[htbp]
% \vspace{0.0in}
% \begin{center}
% \includegraphics[width=0.5\textwidth]{./1znf.pdf}
% \end{center}
% \vspace{-0.2in}
% \setlength{\belowcaptionskip}{0.2in}
% \caption{(a) The full protein in backbone ribbon representation, with the binding site in atomistic detail. 
%     (b) Zoom in of the binding site only, with atoms relabelled for simplicity. 
%     The sulfurs S1 and S2 belong to Cys4 and Cys7, and both are of atom-type SG. 
%     The nitrogens N1 and N2 belong to His20 and His24, and both are of atom-type NE2. }

% \vspace{-0.2in}
% \end{figure}

% \begin{figure}[htbp]
%     \centering
% \subfigure[Protein backbone and \ch{Zn^{2+}} interaction center.]{
%       \includegraphics[width=.35\linewidth]{Images/protein_and_interaction.png}
% }%
% \subfigure[Close-up of interaction center]{
%       \includegraphics[width=.35\linewidth,]{Images/interaction_center_labelled.png}
% }%
% \caption{A view of the protein structure from the 1st model of the NMR structure 1ZNF. 
% The numbers in the atom names refer to the residue number. 
% The sulfurs are from Cys4 and Cys7, while the nitrogens are the NE2 nitrogens of His20 and His24.}
% \label{fig:1znf}
% \end{figure}

The 1ZNF PDB structure \cite{lee1989three} is one of the first Zinc-finger structures to be resolved experimentally. 
It is also the simplest, containing only 25 amino acids and one \ch{Cys2His2} Zn$^{2+}$ binding domain  where the Zinc ion is in a stable coordination geometry consisting of cysteine sulfurs and histidine nitrogens in the first coordination shell (see Figure \ref{fig:MD-rmsd}).
Due to its compact size, 
the 1ZNF structure provides an ideal case study for an MD validation of a FFAFFURR parameterization workflow.
One potential application of FFAFFURR to this system is to optimize selected parameters for the interaction center (Figure \ref{fig:MD-rmsd} bottom-left), since that is the region of most complexity.

%\xhu{ Cite MCPB.py: A Python Based Metal Center Parameter Builder \cite{li2016mcpb}. They optimize parameters for the interaction center and give them unique residue name to differ them from residues in the standard force-field library. We did the similar.}

In this paper, we used an approach similar to Li \emph{et al.}, \cite{li2016mcpb} giving the residues in the interaction center unique residue names to distinguish them from similar residues in the rest of the protein. 
This allows us to target only atom types within the binding domain for parameterization, without affecting the parameters of similar atom types away from the binding site.

Four parameter sets were tested with MD in this study, as described in Table \ref{tbl:parameter_sets}. For the unparameterized OPLS-AA force-field, we observed unbinding of the two histidine residues from the Zn$^{2+}$ interaction center, as shown in Figure \ref{structure-MD}, almost immediately after the start of the simulation.
To try and prevent this, we optimized pair-wise LJ parameters between atoms in HisD and Zn$^{2+}$. 
The parameters that are optimized are listed in Table \ref{tbl:opt_LJ_params_full}. The LJ parameters between atoms in Cys and Zn$^{2+}$ are kept untouched since we haven't seen strange behaviors between Cys and Zn$^{2+}$.
The optimized LJ parameters were used in opt-OPLS-AA and opt-CTPOL sets. In the CTPOL and opt-CTPOL models, charge transfer was introduced for S/N/O/Zn atoms in the binding site, and polarization effects between non-hydrogen atoms and Zn$^{2+}$ were added. Oxygen is included as some of the structures in the ab-initio dataset have Zn$^{2+}$ interacting with a peptide oxygen atom. 

\begin{table}
  \caption{Parameter sets used for MD simulation.
  The determination of LJ parameters from FFAFFURR is described in \ref{LJ_para}. 
    optimized parameters are listed in Table \ref{tbl:opt_LJ_params_full} and \ref{tbl:CTPOL_params}.}
  \label{tbl:parameter_sets}
  \begin{tabular}{lll}
    \hline
      Parameter set& Pair-wise LJ parameters of atoms in HisD and Zn$^{2+}$  &  CT + POL \\
    \hline
    OPLS-AA & original  &  No \\
    opt-OPLS-AA& from FFAFFURR  & No \\
    CTPOL  & same as OPLS-AA & Yes \\
    opt-CTPOL & from FFAFFURR  & Yes \\
    \hline
  \end{tabular}
\end{table}

\subsubsection*{Backbone structure and binding domain are better preserved with CTPOL}

We ran three 40 ns long simulations with each of the four models listed in Table \ref{tbl:parameter_sets}.
We also used the 37 experimental NMR structures of 1ZNF to compare structural features between our simulations and NMR observations. 
Figure \ref{fig:MD-rmsd} shows the RMSD of each of the parameter sets, using the first model of the NMR structures as a reference. 
In the same figure, we also plot the RMSD of the 37 NMR models with respect to the same first model to see how much variation occurs among those.

It is clear from Figure \ref{fig:MD-rmsd} that both, the overall structure and binding domain, are in better agreement with the NMR structures when charge transfer and polarizability are taken into account. 
With opt-OPLS-AA, there is a marginal but noticeable improvement over OPLS-AA, but in both OPLS-AA and opt-OPLS-AA force-fields the binding domain breaks apart. This is evident from the RMSD of the backbone, as shown in the bottom panel of Figure \ref{fig:MD-rmsd}. This is primarily due to the Histidines breaking away from the binding with Zn$^{2+}$, as supported by Figure \ref{SI:zn-n_v_time}.

% \begin{figure}[htbp]
% \subfigure[Protein backbone and \ch{Zn^{2+}} interaction center.]{
%       \includegraphics[width=.25\linewidth]{Images/protein_and_interaction.pdf}
% }%
% \subfigure[RMSD Plots]{
% \centering
%     \includegraphics[width=0.73\linewidth]{./Images/rmsd_density.pdf}      
% }%
% \caption{RMSDs of MD trajectories from the NMR structure of 1ZNF (Model 1), calculated for different parameter sets. 
% Top: the protein backbone atoms only. Bottom: the binding site containing Zn, S4, S7, N20, and N24, as shown in Figure \ref{fig:MD-rmsd}. 
% The densities of RMSD values are shown on the right, using Kernel Density Approximation \cite{parzen1962kde, rosenblatt1956kde}, where the dashed line is the RMSD distribution obtained from NMR data of 1ZNF with respect to the first model of the PDB.}
% \label{fig:MD-rmsd}
% \end{figure}

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{./Figure_protein_and_rmsd_distance_labelled.png}
\caption{\textit{Top-Left}: The protein structure of 1ZNF, with the backbone represented by a ribbon, and the Zn\ch{^{2+}}-binding site shown explicitly. \textit{Bottom left}: Zoom in of the site, with distances of the coordinating atoms relative to Zn\ch{^{2+}}. 
The sulfurs are from Cys4 and Cys7, while the nitrogens are the NE2 nitrogens of His20 and His24.
\textit{Right}: RMSDs of MD trajectories from the NMR structure of 1ZNF (Model 1), calculated for different parameter sets, for backone (top) and interaction site (bottom).  
The densities of RMSD values are shown on the right, using Kernel Density Approximation \cite{parzen1962kde, rosenblatt1956kde}, where the dashed line is the RMSD distribution obtained from NMR data of 1ZNF with respect to the first model of the PDB.}
\label{fig:MD-rmsd}
\end{figure}

\begin{figure}[htbp]
\centering
\subfigure[OPLS-AA]{
    \includegraphics[width=0.22\textwidth]{./Figure_snapshot-oplsaa.png}
}
\subfigure[opt-OPLS-AA]{
    \includegraphics[width=0.22\textwidth]{./Figure_snapshot-opt-oplsaa.png}
}
\subfigure[CTPOL]{
    \includegraphics[width=0.22\textwidth]{./Figure_snapshot-ctpol.png}
}
\subfigure[opt-CTPOL]{
    \includegraphics[width=0.22\textwidth]{./Figure_snapshot-opt-ctpol.png}
}
\caption{Snapshots showing the conformation of binding site after 40 ns of simulation.}
\label{structure-MD}
%\vspace{-0.2in}
%\setlength{\belowcaptionskip}{0.2in}
%\vspace{-0.2in}
\end{figure}
%Over the last few decades, several non-additive force-fields were developed to capture essential charge transfer dynamics or atomic polarizability, such as SIBFA\cite{gresh2011polarizable}, CTPOL\cite{sakharov2009force}, AMOEBA\cite{wu2010polarizable}, SLEF\cite{wu2011transferable}, and CHARMM-Drude\cite{Lemkul2016}.
%While improvements were made, they were not guaranteed due to difficulties in parameter fitting and choosing the right potential function.

The RMSDs of OPLS-AA and opt-OPLS-AA deviate far from the NMR model, particularly the RMSDs of the binding site only.
We observed in our simulations that with OPLS-AA, the two histidine residues in the binding site stray uncharacteristically far from \ch{Zn^{2+}}. 
Even  with optimization of the pair-wise LJ parameters of Zn$^{2+}$ and histidine (opt-OPLS-AA), we observed one of the histidines escaping the binding domain. Figure \ref{structure-MD} (a) and (b) shows snapshots of such conformations after 40 ns.
Similar problems with binding domain stability have been observed in previous studies, where the Zn$^{2+}$ escapes from the coordination center in non-polarizable FF simulations. \cite{zhang2012modeling, donini2000calculation}

However, both CTPOL and opt-CTPOL preserve the binding domain of \ch{Zn^{2+}}, with both histidines and both cysteines coordinating the \ch{Zn^{2+}} ion throughout the 40 ns simulations (snapshots of Figure \ref{structure-MD} (c) and (d)).
This emphasizes that explicitly including charge transfer and polarization effects is critical for a proper description of the binding domain, and hence the overall structure of Zinc-fingers. 


\FloatBarrier

\subsubsection*{LJ parameterization makes the CTPOL model more robust}

To evaluate the effect of optimized pair-wise LJ parameters we compared the CTPOL model without any LJ parameterization (CTPOL) to the CTPOL model with LJ parameterization (opt-CTPOL). 
From Figure \ref{fig:MD-rmsd}, it may appear that such optimization has little effect, and in fact may slightly worsen the overall structure due to the higher RMSD of the backbone. However,
while both models preserve the interaction center much better than OPLS-AA and opt-OPLS-AA,
opt-CTPOL appears to produce a much more stable binding domain than CTPOL. This can be seen when we recompute RMSD after varying the initial conditions. 
To test the impact of initial conditions, we ran 40 independent 1ns long simulations, with the initial frame randomly chosen from a 4 ns MD simulation and random initial velocities. These are reasonable initial conditions that should exhibit similar behavior, as they are taken from a simulation.  
Figure \ref{fig:ctpol_rmsd} shows that the 40 ns trajectory of CTPOL using the NMR structure as the starting point is more or less stable.
However, when running simulations from different initial conditions, this stability is not guaranteed, as seen from the spikes in RMSD. 
On the other hand, opt-CTPOL appears to be stable for all initial conditions.

\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{./Figure_rmsd_density_40x1.pdf}
    \caption{RMSD of CTPOL and opt-CTPOL vs 1st model of NMR, with 40 trajectories of 1 ns concatenated into one.
    The dotted lines represent concatenation boundaries of the trajectories.}
    \label{fig:ctpol_rmsd}
\end{figure}


\begin{figure}
    \centering
    \includegraphics[width=0.8\textwidth]{./Figure_ct_vs_time__ctpol_vs_opt.pdf}
    \caption{Charge transfer as a function of time for (left) a continuous 40 ns trajectory from one stable initial structure, 
    and (right) 40 independent 1 ns simulations concatenated together. The dashed vertical lines mark the concatenation boundaries.
    The $40 \times 1$ ns simulations were started from different initial conditions randomly chosen from a continuous MD simulation, with randomized velocities. }
    \label{fig:ct_vs_time}
\end{figure}


A reason for this is the abnormal charge transfers to Zn\ch{^{2+}} in CTPOL as seen in Figure \ref{fig:ct_vs_time}. This occurs around the same time as the binding domain fluctuations in Figure \ref{fig:ctpol_rmsd}. 
A closer inspection of the distances between Zn\ch{^{2+}} and coordinating nitrogens (Figure \ref{fig:ct_dists_v_time}) reveals that these fluctuations are perfectly correlated with these distances. 
As the binding site breaks down, the coordinating histidines containing these nitrogens move far away, as much as 9 {\AA} away, but the sulfurs remain in close proximity at all times. 
At such distances, the charge transfer contribution of the nitrogens drops to zero, and the only contribution is from the sulfurs, and hence the lower total charge transfer.
However, opt-CTPOL appears to have no such fluctuation in either the 40 ns or 40 $\times$ 1 ns trajectories. 
\begin{figure}
    \centering
    \textbf{Charge transfer and relevant distances in CTPOL}\par\medskip
    \includegraphics[width=0.8\textwidth]{./Figure_ct_dists_vs_time.pdf}
    \caption{Coordinating nitrogen and sulfur distances (left y-axis) and charge transfer (right y-axis) vs time for a continuous trajectory (left) and 40 independent concatenated trajectories.
    In cyan, we have the charge transfer, in green, the average of the distances of Zn-N20 and Zn-N24, and in yellow the average of the distances of Zn-S4 and Zn-S7. Out of the 40 independent simulations, the average distance of Zn-N20/24 rises above 3 {\AA} 8 times. 
    }
    \label{fig:ct_dists_v_time}
\end{figure}

These unfolding events within 1ns occur about 20\% of the time for CTPOL, thus making CTPOL without LJ-optimization unreliable. 

\FloatBarrier
%\subsubsection*{Radial distribution function and peak compositions are improved by opt-CTPOL}
\subsubsection*{opt-CTPOL shows improvement with a caveat to be addressed in the future}

\begin{figure}
    \centering
    \textbf{Continuous 40 ns trajectory}
    \subfigure[Radial Distribution Functions]{
    \includegraphics[width=0.8\textwidth]{./Figure_rdf_cdf_all.pdf}
    }\\
    \subfigure[Peak Breakdown]{
    \includegraphics[width=0.8\textwidth]{./Figure_Peak_breakdown_log.pdf}
    }
    \caption{\textbf{Coordination analyses of continuous 40 ns trajectory.} a) RDF and coordination number ($N(r)$) of all non-hydrogen protein atoms, with the distance ranges of selected peaks. The solid lines are the RDF (left $y$ axis), and the dashed lines are the corresponding $N(r)$ (right $y$ axis). Composition of each peak, where atoms of the same type and residue are lumped  together.
    The $y$ axis represents the average fraction of conformations in which each of the atoms appears within the peak range.
    }
    \label{fig:rdf_peaks}
\end{figure}

\begin{figure}
    \centering
    \textbf{$40\times 1$ns trajectory}
    \subfigure[Radial Distribution Functions]{
    \includegraphics[width=0.8\textwidth]{./Figure_rdf_cdf_all_40_1ns.pdf}
    }\\
    \subfigure[Peak Breakdown]{
    \includegraphics[width=0.8\textwidth]{./Figure_Peak_breakdown_log_40_1.pdf}
    }
    \caption{\textbf{Coordination analyses of $40\times1$ ns trajectory.} a) RDF and coordination number ($N(r)$) of all non-hydrogen protein atoms, with the distance ranges of selected peaks. Solid lines are RDF, and dashed lines are the $N(r)$. b) Composition of each peak, where atoms of the same type and residue are lumped  together. 
    The $y$ axis represents the average fraction of conformations in which each of the atoms appears within the peak range.
    }
    \label{fig:rdf_peaks_40_1}
\end{figure}

% \begin{figure}[htbp]
% \centering
% \subfigure[RDF of all]{
%     \includegraphics[width=0.75\textwidth]{Images/rdf_all.png}
% }
% \subfigure[Peak breakdown]{
%     \includegraphics[width=0.75\textwidth]{Images/Peak_breakdown_log.png}
% }
% \caption{a) RDF of all non-hydrogen protein atoms, with the distance ranges of selected peaks below. b) Composition of each of the peaks, where the atoms of same type and residues in multiple locations have been lumped together. The Y-axis represents the average fraction of conformations in which each of the atoms appear within the peak range.}
% \label{MD-rdf}
% \end{figure}

To evaluate how parameters affect the coordination of \ch{Zn^{2+}}, we plotted the radial distribution function of non-hydrogen protein atoms around the cation in Figure \ref{fig:rdf_peaks} (top).
We can see immediately that NMR and opt-CTPOL have a similar peak structure, but the distances are shorter in opt-CTPOL. 
In CTPOL, the first and second peaks, containing Nitrogens and Sulfurs respectively, overlap completely and are indistinguishable.
In the NMR models, the first and second peaks at 2.0 and 2.3 $\AA$ correspond to Zn$^{2+}$-N(His) and Zn$^{2+}$-S(Cys), respectively. 
In contrast to CTPOL, the opt-CTPOL peaks are distinct, with only a small percentage (< 2\%) of trajectories showing Nitrogens in the 2nd peak dominated by Sulfur. 
These features are also seen in similar analyses of the $40\times 1$ ns trajectories (Figure \ref{fig:rdf_peaks_40_1}). 
Based on the analyses in this manuscript we can conclude that CTPOL does not reproduce NMR binding domain as well as opt-CTPOL even for the stable continuous 40 ns trajectory.

After identifying the peaks, and selecting a range of distances (Figure \ref{fig:rdf_peaks} (top)), we determined which atoms comprise each peak and what fraction of the trajectory these atoms remain in that peak, as shown in Figure \ref{fig:rdf_peaks} (bottom). 
The 1st and 2nd peaks in CTPOL appear to be contaminated by other atom types which do not appear in NMR peaks at all. 
In the 40 $\times$ 1 ns trajectory, since CTPOL binding site has been shown to break apart in a few cases, it is no surprise that water also appears in Peak 1 of CTPOL (Figure \ref{fig:rdf_peaks_40_1} (bottom)).
The opt-CTPOL model has no other atom types in the first peak, and only relatively few others in the 2nd peak not present in NMR.

We should note that the NMR model we used does not contain any explicit water molecules. 
To determine if water could be present in the binding site, we looked at 15 Zinc-finger X-ray crystallography structures from the Protein Data Bank\cite{berman2000pdb} (PDB) website (\href{http://www.rcsb.org/pdb/}{http://www.rcsb.org/pdb/}) to find binding sites which are similar to this one (see \ref{SI:tbl:other_pdbs} for a full list). 
We looked at binding sites which had a total of 2 histidines and 2 cysteines, similar to 1ZNF.  
We found 8 binding sites from the 15 crystal structures, and the smallest water distance to \ch{Zn^{2+}} was 4.38 {\AA}, well outside even the 3rd peak range in the NMR models. 
We further relaxed the matching criterion for the binding site to any binding site that contains a total of 4 histidines or cysteines (i.e., the number of coordinating histidines and cysteines sum to 4, but does not have to be 2 each).
This resulted in a total of 60 binding sites.
From these, we found the smallest water distance to be 3.98 {\AA}, still beyond the peak 3 range.

Thus, the inclusion of water in the 1st and 2nd peaks, as is the case in CTPOL model, is uncharacteristic of Zn-finger binding sites of similar nature to 1ZNF. 
The opt-CTPOL model does a better job of keeping the water outside these peaks, with only a small fraction of water in the 2nd peak. 

\FloatBarrier
\subsubsection*{Angle and distance distributions}
\begin{figure}
  \begin{minipage}[c]{0.49\textwidth}
    \includegraphics[width=\textwidth]{./Figure_binding_site_labelled.png}
  \end{minipage}\hfill
\begin{minipage}[c]{0.49\textwidth}
    \caption{
    Binding site with \ch{Zn^{2+}} at the center (grey atom), the sulfurs from Cys7 (S7) and Cys4 (S4), the NE nitrogens from His20 (N20) and His24 (24). 
    Hydrogens have been removed for clarity. 
    The yellow triangle on top has vertices on Zn, S4 and S7, while the blue triangle at the bottom has vertices on Zn, N20, and N24. 
    The angle between the planes of these triangles are used for plotting the distributions in Figure \ref{fig:dihedral_bisector}.
    The red and blue arrows ($\vec{S}_{bi}$ and $\vec{N}_{bi}$) are vectors that bisect angles S7-Zn-S4 and N20-Zn-N24 respectively.
    The distributions of angle $\theta$ between these two bisectors are plotted on Figure \ref{fig:dihedral_bisector} (b).
    The distributions of some of the distances between the 5 atoms shown in this figure are shown on Figure \ref{fig:distances}, while the distributions for some of the angles are shown in Figure \ref{fig:angles}.
    } \label{fig:binding_angles}
  \end{minipage}
\end{figure}

To further evaluate the stability and accuracy of the binding domain in the CTPOL and opt-CTPOL frameworks, we analyzed a number of geometric quantities which are defined in Figure \ref{fig:binding_angles} and its caption. 
Here we only consider the 40 ns continuous trajectory for which the binding domain is stable for CTPOL, since these geometric quantities would not make sense for the $40\times1$ns trajectory where the binding domain destabilizes.

Figure \ref{fig:angles} shows the distribution of most of the angles that the coordinating atoms make with Zn$^{2+}$.
Additionally, Figure \ref{fig:dihedral_bisector} (a) shows the distributions of angles between the planes shown in Figure \ref{fig:binding_angles},
and Figure \ref{fig:dihedral_bisector} (b) shows the distributions of the angles between the bisectors, also defined in Figure \ref{fig:binding_angles}.
It is quite clear that opt-CTPOL reproduces the NMR distributions of angles as well or better than CTPOL.
The distribution of the S4-Zn-S7 angle appears to agree particularly well with NMR, as does the angle between the bisectors. 
While the CTPOL 40 ns trajectory showed a slightly better overall RMSD from Figure \ref{fig:MD-rmsd}, it is clearly not reproducing these angles as well as opt-CTPOL.
This implies that opt-CTPOL is maintaining the shape of the binding domain better, which is in accordance with the RDF distribution and peak analysis of Figure \ref{fig:rdf_peaks}.

\begin{figure}
    \centering
    \includegraphics[width=.7\linewidth]{./Figure_Angular_distributions.pdf}
    \caption{Probability distribution of angles over the continuous 40 ns trajectories of CTPOL (blue) and opt-CTPOL (orange) and over 37 NMR models (black dashed). 
    The corresponding atoms are depicted in Figure \ref{fig:binding_angles}. The distributions were calculated using Kernel Density Estimation \cite{parzen1962kde, rosenblatt1956kde}. The vertical lines represent the averages of each distribution. 
    %\xhu{The title of the forth figure is not correct.}\ksa{I checked, it is correct.}
    }
    \label{fig:angles}
\end{figure}

\begin{figure}
    \centering
\subfigure[Dihedral distribution]{
      \includegraphics[width=.35\linewidth]{./Figure_dihedral_between_SZnS_NZnN.pdf}
}%
\subfigure[Bisector angle distribution]{
      \includegraphics[width=.35\linewidth]{./Figure_bisector_angles.pdf}
}%
\caption{(a) Probability distributions of angle between the S7-Zn-S4 and N24-Zn-N20 planes as depicted in Figure \ref{fig:binding_angles}. (b) Angle between S4-Zn-S7 and N20-Zn-N24 bisectors, which are depicted in Figure \ref{fig:binding_angles} as $\vec{S}_{bi}$ and $\vec{N}_{bi}$, respectively.
}
\label{fig:dihedral_bisector}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=.8\linewidth]{./Figure_Distance_distributions.pdf}
    \caption{Probability distribution of distances (using Kernel Density Estimation \cite{parzen1962kde, rosenblatt1956kde}) over entire trajectory (for simulations) and over 37 models (for NMR data).}
    \label{fig:distances}
\end{figure}

Furthermore, we see from Figure \ref{fig:distances} that the distances of the opt-CTPOL binding domain are consistently shorter than those of the experimental NMR structures. 
This is in line with the RDF analysis of Figure \ref{fig:rdf_peaks}, where we see a similar peak structure of opt-CTPOL, but at shorter distances. 
On the other hand, CTPOL distances do not appear to have a consistent relation to the NMR distances. 
For instance, the distances of S*-Zn and N*-Zn (top left) show that opt-CTPOL distances trend the same way as NMR, i.e., the N*-Zn distances are significantly shorter than S*-Zn distances.
For CTPOL, it turns out to be almost the opposite, with plenty of overlap between the two distributions, and thus their 1st and 2nd peaks in Figure \ref{fig:rdf_peaks} also overlap. 

\FloatBarrier

\subsubsection*{Issues we observe and their possible origins}



%\ksa{\textbf{Discussion of QM distances}}
The main issue we face is clearly the contraction of distances in the Zn$^{2+}$-binding site in comparison to NMR data.
To investigate the source of this issue, we first analyzed the relevant distances in the \textit{ab initio} data set, as depicted in Figure \ref{fig:qm-distances}. 
We see that the Zn-N distances are mostly around the 2 {\AA} mark, and Zn-S distances are mostly around 2.3 \AA, as it is indicated by the maxima of the histograms.
In both cases, they cover a wide range of energies as well, suggesting that these are preferred distances for various conformations of the dipeptide.
However, these are (certainly almost exclusively) minima on the PES.

The larger distances in Figure \ref{fig:qm-distances} are due to coordination of Zn with other atoms (such as carbonyl oxygen). 
To make a meaningful comparison with the 1ZNF system, we further filtered the QM data down to where the nearest atom to Zn is atom SG or atom NE2, since that is the case in the 1ZNF system. 
From the filtered data, we computed the average distances, and compare them in Table \ref{tab:avg-distances}, where it is clear that the QM data agrees well with NMR, unlike opt-CTPOL, which systematically gives shorter distances.


% \begin{figure}[htb]
%     \centering
%     \subfigure[Zn$^{2+}$-N distance]{
%     \includegraphics[width=0.45\textwidth]{Images/Zn_NE2_distance_energy.pdf}
%     }
%     \subfigure[Zn$^{2+}$-S distance]{
%     \includegraphics[width=0.45\textwidth]{Images/Zn_SG_distance_energy.pdf}
%     }
%     \caption{Distribution of distances, with bars representing histograms of the numbers of conformers (left axis) over distance range, and points representing the distance and energy of individual conformers (right axis). }
%     \label{fig:qm-distances}
% \end{figure}

\begin{figure}
    \centering
    \includegraphics{./Figure_QM_Zn_ligand_distance_energy.pdf}
    \caption{Distribution of \textit{ab-initio} distances, with bars representing histograms of the numbers of conformers (left axis) over distance range, and points representing the distance and energy of individual conformers (right axis).
    The plot on the left takes distances between Zn$^{2+}$ and S(Cys) from AcCys$^-$-NMe + Zn$^{2+}$ \textit{ab-initio} conformations. 
    The plot on the right takes distances between Zn$^{2+}$ and N(His) from AcCys$^-$-His + Zn$^{2+}$ \textit{ab-initio} conformations.}
    \label{fig:qm-distances}
\end{figure}


\begin{table}[htb]
\begin{tabular}{r|cccc}

\textbf{Pair} & \textbf{QM} & \textbf{NMR} & \textbf{CTPOL} & \textbf{opt-CTPOL} \\ \hline
\textbf{N-Z}  & 1.95   & 1.99     & 2.01       & 1.73            \\ 
%\textbf{N-Z}  & 1.9540461   & 1.988557     & 2.014272       & 1.73238            \\ \hline
\textbf{S-Z}  & 2.28   & 2.31     & 1.89       & 2.01           \\ 
%\textbf{S-Z}  & 2.2756083   & 2.307921     & 1.889855       & 2.014202           \\ \hline
\end{tabular}
\caption{\label{tab:avg-distances}Average distance data.}
\end{table}

%\ksa{End discussion of QM distances}

Given that the shorter distances are not an artefact that was carried from the gas phase QM data, in the following we discuss a few likely causes for this behavior of our model:
\begin{itemize}
    \item  We demonstrate in this paper (see Figure \ref{full-opt-ctpol}) that our approach can properly match energy hierarchies. However, the focus on equilibrium geometries may result in insufficient treatment / parameterization of repulsive (off-equilibrium) geometries. The effect may even be amplified due to the Boltzmann-weighting used in the parameter optimization process.

    \item The Lennard-Jones-(12,6)-potential is intended to model pairwise  repulsive and non-Coulomb attractive interactions among atoms. However, due to the nature of classical force-field formulations, it may also contain aspects of polarization, dipole-dipole interactions etc. These aspects may be incorporated in parameters as well as in the functional form (e.g. in the 12-6 form).

    \item Although we are matching MM energies at the exact same conformations, we do not know if the MM minima themselves correspond to those conformations. It is possible that we are simply matching the MM energies at the QM minima, but the MM minima may be in different states altogether, in this case, those that result in shorter ligand distances.
    \item None of the \textit{ab initio} references contain both Sulfur and Nitrogen in the coordination of Zn$^{2+}$, or even specifically S-N, N-N, and S-S pairwise interactions. We only parameterize Zn$^{2+}$-ligand LJ interactions in this work as an example, but it is possible that optimizing ligand-ligand interactions will improve the result.

\end{itemize}

% \begin{figure}
%     \centering
%     \subfigure[Angle distribution]{
%         \includegraphics[width=0.45\textwidth]{Images/Distance_distributions_from_Zn.png}
%     }
%     \subfigure[Distance distribution]{
%         \includegraphics[width=0.45\textwidth]{Images/Angular_distributions.png}
%     }
%     \caption{Distribution a) distances and b) angles over entire trajectory (for simulations) and over 37 models (for NMR data) \ksa{Find a dihedral with CB of HIS and Zn as end points that captures orientation of His}
%     }
%     \label{fig:angles_distances}
% \end{figure}

\FloatBarrier
\section{Conclusion and outlook}
The availability of sufficiently accurate force-field parameters for cation-peptide systems is a major obstacle in metalloprotein simulations. 
One approach to facilitate the development of new force-field parameters is to construct tools to derive parameters from QM calculations.
The benefits of such an approach was shown in previous work \cite{Amin2020} on the QM-driven parameterization of Drude and CTPOL models  for ion-protein interactions in MD simulations.

Since the explicit Drude model includes polarization as a degree of freedom subject to forces, it requires shorter time steps and a dual thermostat and it introduces additional parameters such as the mass of Drude particles and spring constants.
CTPOL -- as an implicit model --  requires the minimization of dipole moments at every time step, but allows for normal time steps.
Both models exhibit  their own challenges and -- in particular our re-implementation of CTPOL -- their own room for improvements.

%On the other hand, the Drude model requires shorter time steps, which is not the case for CTPOL, although how this will affect the computing times has to be investigated.
%Furthermore, the Drude model requires a dual thermostat and introduces additional parameters such as the mass of Drude particles and additional spring constants.
%This makes both parameter optimization and implementation of the Drude model more difficult.
%Finally, charge transfer effects are not captured by the Drude model.

FFAFFURR is developed as a python tool to facilitate the parameterization of classical and polarizable CTPOL models. 
In this paper, we chose to parameterize OPLS-AA as an example, although the tool can be adjusted to work with other similar force-fields such as CHARMM and AMBER once the code is generalized. 
It automatically parses QM calculation outputs from FHI-aims and generates parameter files that can be directly processed by the molecular dynamics package OpenMM.
FFAFFURR also allows users to choose which energy terms to adjust.

We utilized an extensive data set of model peptide-cation conformers from DFT calculations. 
Structures cover an extended relative energy range and all required properties for the parameterization were extracted.  
Due to the sheer size of the utilized data set -- by means of number of conformers and energy range -- we think that we have covered sufficient individual structural diversity to properly derive parameters for the FF energy terms.
The performance of optimized parameters in each energy term was evaluated by comparing of FF energies and QM potential energies. 

We showed that the CTPOL model outperforms OPLS-AA in terms of the accuracy of reproducing the QM energy hierarchies for divalent-dipeptide systems. 

One potential usage of FFAFFURR is the rapid construction of FFs for troublesome metal centers in metalloproteins. 
We tested this function by performing MD simulations on the 1ZNF Zinc-finger protein\cite{lee1989three} and comparing simulation results with NMR models. 
With the parameters optimized from FFAFFURR, we found that CTPOL much better reproduces the overall structure of the protein, while with OPLS and opt-OPLS, the Zn$^{2+}$ binding site unravels. 
However, to better stabilize and reproduce structural features of the binding domain, LJ optimization (opt-CTPOL) was necessary, since CTPOL alone had some shortcomings in correctly reproducing the binding domain, or keeping it stable under various initial conditions.
The LJ optimization resulted in coordination composition and geometry that better agrees with the NMR models than CTPOL alone.

On the other hand, the optimization of LJ does lead to a shrunken binding domain. While we briefly discussed reasons for this behavior above, we focus here on possible remedies:
\begin{itemize}
    \item A possible lack of off-equilibrium geometries in the parameterization could be tested first by omitting the Boltzmann-weighting. This would give more impact to higher-energy structures that are more likely to feature at least partly more compressed structures. Maybe this already would lead to improved geometries.

    \item The repulsive part of the Lennard-Jones potential could be better captured by locally exploring the PES by adding points through dislocating atoms in the cation interaction site. This could either be realized by carefully altering Cartesian coordinates by small values in the $+$/$-$ space directions or through short DFT-based MD simulations. This would also capture whether or not the MM-minima are at those conformations.
    
    \item The functional form of the Van der Waals potential may be better generalized by alternatives to the Lennard-Jones form which allow for better tuning of the shape of the repulsive components. For example, the Mie potential is a generalized version of Lennard-Jones where the exponents and coefficients of each term can be varied. The Buckingham potential is another alternative where the repulsive term is replaced with an exponential decay with variable amplitude and decay rate. Such adjustments of the Van der Waals potential also requires a data base that contains sufficient off-equilibrium geometries.
    \item Lastly, the dataset could be expanded to contain models with more than one amino-acid side chain, in order to capture  complex coordination structures, allowing us to parameterize other pairwise interaction such as Sulfur and Nitrogen, and not just Zn$^{2+}$ and ligand atoms.

\end{itemize}

%There could be several reasons for this, such as (i) insufficient fitting criterion, (ii) overfitting a few specific parameters as opposed to implementing a holistic CTPOL model for the entire system, (iii) fundamental lack of transferability of gas phase-derived parameters to condensed phase. 
%Issues (i) and (ii) can be addressed using the current QM data set.
%However, if issue (iii) is true regardless of how much we improve the parameterization protocol, then we would need to improve on the data set by incorporating solvent affect to some degree.  
%Whether this is a major concern remains to be seen with further studies, such as calculations of relative binding affinities with other metals, or other macroscopic analyses of similar systems which could be verified experimentally. 

\noindent
In summary, FFAFFURR has a wide range of functions to facilitate and perform parameter optimization in peptide systems. 
It can provide better additive force-field parameters for systems with no cations, as seen from Figures \ref{full-opt-oplsaa}b and \ref{full-opt-oplsaa}c. 
Additionally, FFAFFURR and can provide almost all the functions required for the cation-peptide parameterization process, including for CTPOL force-fields. 
FFAFFURR helps users remove labor-intensive steps in force-field optimization. 

Despite the success of FFAFFURR in this study, we see several directions to discuss in future research. 
Note that only the parameters of the Zinc-finger protein interaction center were optimized with FFAFFURR in the MD simulation, while the standard OPLS-AA parameters were used for the rest of the protein. 
While our study indicates the compatibility of the optimized parameters with the standard FF parameters, this may have to be investigated in more detail in a future study. 
One characteristic of FFAFFURR is that it can be employed to derive parameters for a specific system. 
This helps to grasp the specific environment of the system. 
However, QM calculations are required when a new system is under investigation. 
We created a data set of cation-dipeptides containing several divalent cations, which can be automatically parsed to FFAFFURR. \cite{hu2022better} 
If the user's system goes beyond the scope of the dataset, an in-house genetic algorithm package Fafoom \cite{Supady2015} can be used to generate conformers and do the QM generation fast and automatically. 





 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% The "Acknowledgement" section can be given in all manuscript
%% classes.  This should be given within the "acknowledgement"
%% environment, which will make the correct section or running title.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{acknowledgement}


The authors thank:
\begin{description}
    \item [The China Scholarship Council] for providing X.H. with a doctoral fellowship;
    \item[The Federal Ministry of Education and Research of Germany] for providing funding for the project STREAM (“Semantische Repräsentation, Vernetzung und 333 Kuratierung von qualitätsgesicherten Materialdaten”, ID: 16QK11C); and
    \item[MITACS] for the MITACS Globalink Research Award which funded the visit of K.S.A. to the lab of X.H. and C.B.
    
\end{description}

\end{acknowledgement}

\section*{Available data and code}

%The links to data and code used in this study are summarized here:
The reference data can be found on the NOMAD repository via the DOI: \url{https://doi.org/10.17172/NOMAD/2023.02.03-1}.\cite{Hu2023NOMAD}
FFAFFURR can be found at: \url{https://github.com/XiaojuanHu/ffaffurr-dev/releases/tag/version1.0}.
Code for implementing CTPOL in OpenMM: \url{https://github.com/XiaojuanHu/CTPOL_MD}. \cite{Hu2023CTPOLMD}

%\section{Available data and codes}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% The appropriate \bibliography command should be placed here.
%% Notice that the class file automatically sets \bibliographystyle
%% and also names the section correctly.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\bibliography{Hu_FFAFFURR}

\providecommand{\latin}[1]{#1}
\makeatletter
\providecommand{\doi}
  {\begingroup\let\do\@makeother\dospecials
  \catcode`\{=1 \catcode`\}=2 \doi@aux}
\providecommand{\doi@aux}[1]{\endgroup\texttt{#1}}
\makeatother
\providecommand*\mcitethebibliography{\thebibliography}
\csname @ifundefined\endcsname{endmcitethebibliography}
  {\let\endmcitethebibliography\endthebibliography}{}
\begin{mcitethebibliography}{113}
\providecommand*\natexlab[1]{#1}
\providecommand*\mciteSetBstSublistMode[1]{}
\providecommand*\mciteSetBstMaxWidthForm[2]{}
\providecommand*\mciteBstWouldAddEndPuncttrue
  {\def\EndOfBibitem{\unskip.}}
\providecommand*\mciteBstWouldAddEndPunctfalse
  {\let\EndOfBibitem\relax}
\providecommand*\mciteSetBstMidEndSepPunct[3]{}
\providecommand*\mciteSetBstSublistLabelBeginEnd[3]{}
\providecommand*\EndOfBibitem{}
\mciteSetBstSublistMode{f}
\mciteSetBstMaxWidthForm{subitem}{(\alph{mcitesubitemcount})}
\mciteSetBstSublistLabelBeginEnd
  {\mcitemaxwidthsubitemform\space}
  {\relax}
  {\relax}

\bibitem[Sarkar(1987)]{sarkar1987metal}
Sarkar,~B. Metal protein interactions. \emph{Prog. Food Nutr. Sci.}
  \textbf{1987}, \emph{11}, 363--400\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Peters \latin{et~al.}(2010)Peters, Yang, Wang,
  F{\"{u}}sti-Moln{\'{a}}r, Weaver, and Merz]{Peters2010}
Peters,~M.~B.; Yang,~Y.; Wang,~B.; F{\"{u}}sti-Moln{\'{a}}r,~L.; Weaver,~M.~N.;
  Merz,~K.~M. {Structural survey of zinc-containing proteins and development of
  the zinc AMBER force field (ZAFF)}. \emph{J. Chem. Theory Comput.}
  \textbf{2010}, \emph{6}, 2935--2947\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Christianson(1991)]{christianson1991structural}
Christianson,~D.~W. Structural biology of zinc. \emph{Adv. Protein Chem.}
  \textbf{1991}, \emph{42}, 281--355\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Patel \latin{et~al.}(2007)Patel, Kumar, and Durani]{patel2007analysis}
Patel,~K.; Kumar,~A.; Durani,~S. Analysis of the structural consensus of the
  zinc coordination centers of metalloprotein structures. \emph{Biochim.
  Biophys. Acta - Proteins Proteom.} \textbf{2007}, \emph{1774},
  1247--1253\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Babu \latin{et~al.}(2014)Babu, Lee, Dudev, and Lim]{Babu2014}
Babu,~C.~S.; Lee,~Y.~M.; Dudev,~T.; Lim,~C. {Modeling Zn$^{2+}$ release from
  metallothionein}. \emph{J. Phys. Chem. A} \textbf{2014}, \emph{118},
  9244--9252\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bell and Vallee(2009)Bell, and
  Vallee]{https://doi.org/10.1002/cbic.200800511}
Bell,~S.~G.; Vallee,~B.~L. {The metallothionein/thionein system: An
  oxidoreductive metabolic zinc link}. \emph{ChemBioChem} \textbf{2009},
  \emph{10}, 55--62\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Capdevila \latin{et~al.}(2012)Capdevila, Bofill, Palacios, and
  Atrian]{Capdevila2012}
Capdevila,~M.; Bofill,~R.; Palacios,~O.; Atrian,~S. {State-of-the-art of
  metallothioneins at the beginning of the 21st century}. \emph{Coord. Chem.
  Rev.} \textbf{2012}, \emph{256}, 46--62\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Cherian \latin{et~al.}(2004)Cherian, Jayasurya, and
  Bay]{Cherian2004Metallo}
Cherian,~G.; Jayasurya,~A.; Bay,~B.-H. Metallothionein in human tumors and
  potential carcinogenesis. \emph{Mutat. Res.} \textbf{2004}, \emph{533},
  201--9\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Durand \latin{et~al.}(2010)Durand, Meloni, Talmard, Va{\v{s}}{\'{a}}k,
  and Faller]{Durand2010}
Durand,~J.; Meloni,~G.; Talmard,~C.; Va{\v{s}}{\'{a}}k,~M.; Faller,~P. {Zinc
  release of Zn$_{7}$-metallothionein-3 induces fibrillar type amyloid-$\beta$
  aggregates}. \emph{Metallomics} \textbf{2010}, \emph{2}, 741--744\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Miller \latin{et~al.}(1985)Miller, McLachlan, and
  Klug]{Miller1985Repetitive}
Miller,~J.; McLachlan,~A.~D.; Klug,~A. {Repetitive zinc-binding domains in the
  protein transcription factor IIIA from Xenopus oocytes.} \emph{EMBO J.}
  \textbf{1985}, \emph{4}, 1609--1614\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wolfe \latin{et~al.}(2000)Wolfe, Nekludova, and Pabo]{Wolfe2000DNA}
Wolfe,~S.~A.; Nekludova,~L.; Pabo,~C.~O. DNA recognition by
  (Cys$_{2}$His$_{2}$) zinc finger proteins. \emph{Annu. Rev. Bioph. Biom.}
  \textbf{2000}, \emph{29}, 183\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Laity \latin{et~al.}(2001)Laity, Lee, and Wright]{laity2001zinc}
Laity,~J.~H.; Lee,~B.~M.; Wright,~P.~E. Zinc finger proteins: new insights into
  structural and functional diversity. \emph{Curr. Opin. Struct. Biol.}
  \textbf{2001}, \emph{11}, 39--46\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Gamsjaeger \latin{et~al.}(2007)Gamsjaeger, Liew, Loughlin, Crossley,
  and Mackay]{Gamsjaeger2007Sticky}
Gamsjaeger,~R.; Liew,~C.~K.; Loughlin,~F.~E.; Crossley,~M.; Mackay,~J.~P.
  Sticky fingers: zinc-fingers as protein-recognition motifs. \emph{Trends
  Biochem. Sci.} \textbf{2007}, \emph{32}, 63--70\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Mobley and Klimovich(2012)Mobley, and Klimovich]{Mobley2012}
Mobley,~D.~L.; Klimovich,~P.~V. {Perspective: Alchemical free energy
  calculations for drug discovery}. \emph{J. Chem. Phys.} \textbf{2012},
  \emph{137}, 230901\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Lemkul \latin{et~al.}(2016)Lemkul, Huang, Roux, and
  MacKerell]{Lemkul2016}
Lemkul,~J.~A.; Huang,~J.; Roux,~B.; MacKerell,~A.~D. {An empirical polarizable
  force field based on the classical drude oscillator model: Development
  history and recent applications}. \emph{Chem. Rev.} \textbf{2016},
  \emph{116}, 4983--5013\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Jorgensen(2004)]{Jorgensen2004}
Jorgensen,~W.~L. {The many roles of computation in drug discovery}.
  \emph{Science} \textbf{2004}, \emph{303}, 1813 -- 1818\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[{A. Kaminski} \latin{et~al.}(2001){A. Kaminski}, {A. Friesner},
  Tirado-Rives, and {L. Jorgensen}]{A.Kaminski2001}
{A. Kaminski},~G.; {A. Friesner},~R.; Tirado-Rives,~J.; {L. Jorgensen},~W.
  {Evaluation and reparametrization of the OPLS-AA force field for proteins via
  comparison with accurate quantum chemical calculations on peptides}. \emph{J.
  Phys. Chem. B} \textbf{2001}, \emph{105}, 6474--6487\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Salomon-Ferrer \latin{et~al.}(2013)Salomon-Ferrer, Case, and
  Walker]{doi:10.1002/wcms.1121}
Salomon-Ferrer,~R.; Case,~D.~A.; Walker,~R.~C. {An overview of the Amber
  biomolecular simulation package}. \emph{WIREs Comput. Mol. Sci.}
  \textbf{2013}, \emph{3}, 198--210\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Huang \latin{et~al.}(2017)Huang, Rauscher, Nawrocki, Ran, Feig,
  de~Groot, Grubm{\"{u}}ller, and MacKerell]{Huang2017}
Huang,~J.; Rauscher,~S.; Nawrocki,~G.; Ran,~T.; Feig,~M.; de~Groot,~B.~L.;
  Grubm{\"{u}}ller,~H.; MacKerell,~A.~D. {CHARMM36m: an improved force field
  for folded and intrinsically disordered proteins}. \emph{Nat. Methods}
  \textbf{2017}, \emph{14}, 71--73\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Reif \latin{et~al.}(2013)Reif, Winger, and Oostenbrink]{Reif2013}
Reif,~M.~M.; Winger,~M.; Oostenbrink,~C. {Testing of the GROMOS force-field
  parameter set 54A8: Structural properties of electrolyte solutions, lipid
  bilayers, and proteins}. \emph{J. Chem. Theory Comput.} \textbf{2013},
  \emph{9}, 1247--1264\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li \latin{et~al.}(2015)Li, Ngo, {Da Silva}, Salahub, Callahan, Roux,
  and Noskov]{Li2015}
Li,~H.; Ngo,~V.; {Da Silva},~M.~C.; Salahub,~D.~R.; Callahan,~K.; Roux,~B.;
  Noskov,~S.~Y. {Representation of ion-protein interactions using the Drude
  polarizable force-field}. \emph{J. Phys. Chem. B} \textbf{2015}, \emph{119},
  9401--9416\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li and Merz(2017)Li, and Merz]{Li2017}
Li,~P.; Merz,~K.~M. {Metal ion modeling using classical mechanics}. \emph{Chem.
  Rev.} \textbf{2017}, \emph{117}, 1564--1686\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Amin \latin{et~al.}(2020)Amin, Hu, Salahub, Baldauf, Lim, and
  Noskov]{Amin2020}
Amin,~K.~S.; Hu,~X.; Salahub,~D.~R.; Baldauf,~C.; Lim,~C.; Noskov,~S.
  {Benchmarking polarizable and non-polarizable force fields for
  Ca$^{2+}$–peptides against a comprehensive QM dataset}. \emph{J. Chem.
  Phys.} \textbf{2020}, \emph{153}, 144102\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Maksimov \latin{et~al.}(2021)Maksimov, Baldauf, and
  Rossi]{doi:10.1002/qua.26369}
Maksimov,~D.; Baldauf,~C.; Rossi,~M. The conformational space of a flexible
  amino acid at metallic surfaces. \emph{Int. J. Quantum Chem.} \textbf{2021},
  \emph{121}, e26369\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Schneider and Baldauf(2018)Schneider, and
  Baldauf]{schneider2018relative}
Schneider,~M.; Baldauf,~C. Relative energetics of acetyl-histidine protomers
  with and without Zn$^{2+}$ and a benchmark of energy methods. \emph{arXiv
  preprint arXiv:1810.10596} \textbf{2018}, \relax
\mciteBstWouldAddEndPunctfalse
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wu \latin{et~al.}(2010)Wu, Piquemal, Chaudret, Reinhardt, and
  Ren]{wu2010polarizable}
Wu,~J.~C.; Piquemal,~J.-P.; Chaudret,~R.; Reinhardt,~P.; Ren,~P. Polarizable
  molecular dynamics simulation of Zn (II) in water using the AMOEBA force
  field. \emph{J. Chem. Theory Comput.} \textbf{2010}, \emph{6},
  2059--2070\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Akin-Ojo \latin{et~al.}(2008)Akin-Ojo, Song, and Wang]{Akin-Ojo2008}
Akin-Ojo,~O.; Song,~Y.; Wang,~F. {Developing ab initio quality force fields
  from condensed phase quantum-mechanics/molecular-mechanics calculations
  through the adaptive force matching method}. \emph{J. Chem. Phys.}
  \textbf{2008}, \emph{129}, 64108\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Dubou{\'{e}}-Dijon \latin{et~al.}(2020)Dubou{\'{e}}-Dijon, Javanainen,
  Delcroix, Jungwirth, and Martinez-Seara]{Duboue-Dijon2020}
Dubou{\'{e}}-Dijon,~E.; Javanainen,~M.; Delcroix,~P.; Jungwirth,~P.;
  Martinez-Seara,~H. {A practical guide to biologically relevant molecular
  simulations with charge scaling for electronic polarization}. \emph{J. Chem.
  Phys.} \textbf{2020}, \emph{153}, 50901\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Martinek \latin{et~al.}(2018)Martinek, Dubou{\'{e}}-Dijon, Timr,
  Mason, Baxov{\'{a}}, Fischer, Schmidt, Pluhařov{\'{a}}, and
  Jungwirth]{Martinek2018}
Martinek,~T.; Dubou{\'{e}}-Dijon,~E.; Timr,~{\v{S}}.; Mason,~P.~E.;
  Baxov{\'{a}},~K.; Fischer,~H.~E.; Schmidt,~B.; Pluhařov{\'{a}},~E.;
  Jungwirth,~P. {Calcium ions in aqueous solutions: Accurate force field
  description aided by ab initio molecular dynamics and neutron scattering}.
  \emph{J. Chem. Phys.} \textbf{2018}, \emph{148}, 222813\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[{Le Breton} and Joly(2020){Le Breton}, and Joly]{LeBreton2020}
{Le Breton},~G.; Joly,~L. {Molecular modeling of aqueous electrolytes at
  interfaces: Effects of long-range dispersion forces and of ionic charge
  rescaling}. \emph{J. Chem. Phys.} \textbf{2020}, \emph{152}, 241102\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li \latin{et~al.}(2015)Li, Song, and Merz~Jr]{li2015systematic}
Li,~P.; Song,~L.~F.; Merz~Jr,~K.~M. Systematic parameterization of monovalent
  ions employing the nonbonded model. \emph{J. Chem. Theory Comput.}
  \textbf{2015}, \emph{11}, 1645--1657\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li \latin{et~al.}(2015)Li, Song, and Merz~Jr]{li2015parameterization}
Li,~P.; Song,~L.~F.; Merz~Jr,~K.~M. Parameterization of highly charged metal
  ions using the 12-6-4 LJ-type nonbonded model in explicit water. \emph{J.
  Phys. Chem. B} \textbf{2015}, \emph{119}, 883--895\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Vanommeslaeghe \latin{et~al.}(2010)Vanommeslaeghe, Hatcher, Acharya,
  Kundu, Zhong, Shim, Darian, Guvench, Lopes, Vorobyov, and {Mackerell
  Jr.}]{Vanommeslaeghe2010}
Vanommeslaeghe,~K.; Hatcher,~E.; Acharya,~C.; Kundu,~S.; Zhong,~S.; Shim,~J.;
  Darian,~E.; Guvench,~O.; Lopes,~P.; Vorobyov,~I.; {Mackerell Jr.},~A.~D.
  {CHARMM general force field: A force field for drug-like molecules compatible
  with the CHARMM all-atom additive biological force fields}. \emph{J. Comput.
  Chem.} \textbf{2010}, \emph{31}, 671--690\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Dodda \latin{et~al.}(2017)Dodda, Cabeza~de Vaca, Tirado-Rives, and
  Jorgensen]{dodda2017ligpargen}
Dodda,~L.~S.; Cabeza~de Vaca,~I.; Tirado-Rives,~J.; Jorgensen,~W.~L. LigParGen
  web server: an automatic OPLS-AA parameter generator for organic ligands.
  \emph{Nucleic Acids Res.} \textbf{2017}, \emph{45}, W331--W336\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Sousa~da Silva and Vranken(2012)Sousa~da Silva, and
  Vranken]{sousa2012acpype}
Sousa~da Silva,~A.~W.; Vranken,~W.~F. ACPYPE-Antechamber python parser
  interface. \emph{BMC Res. Notes} \textbf{2012}, \emph{5}, 1--8\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wang \latin{et~al.}(2004)Wang, Wolf, Caldwell, Kollman, and
  Case]{Wang2004}
Wang,~J.; Wolf,~R.~M.; Caldwell,~J.~W.; Kollman,~P.~A.; Case,~D.~A.
  {Development and testing of a general amber force field}. \emph{J. Comput.
  Chem.} \textbf{2004}, \emph{25}, 1157--1174\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Vassetti \latin{et~al.}(2019)Vassetti, Pagliai, and
  Procacci]{vassetti2019assessment}
Vassetti,~D.; Pagliai,~M.; Procacci,~P. Assessment of GAFF2 and OPLS-AA general
  force fields in combination with the water models TIP3P, SPCE, and OPC3 for
  the solvation free energy of druglike organic molecules. \emph{J. Chem.
  Theory Comput.} \textbf{2019}, \emph{15}, 1983--1995\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Kumar \latin{et~al.}(2020)Kumar, Yoluk, and
  MacKerell~Jr]{kumar2020ffparam}
Kumar,~A.; Yoluk,~O.; MacKerell~Jr,~A.~D. FFParam: Standalone package for
  CHARMM additive and Drude polarizable force field parametrization of small
  molecules. \emph{J. Comput. Chem.} \textbf{2020}, \emph{41}, 958--970\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wang \latin{et~al.}(2014)Wang, Martinez, and Pande]{wang2014building}
Wang,~L.-P.; Martinez,~T.~J.; Pande,~V.~S. Building force fields: An automatic,
  systematic, and reproducible approach. \emph{J. Phys. Chem. Lett.}
  \textbf{2014}, \emph{5}, 1885--1891\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Jorgensen \latin{et~al.}(2007)Jorgensen, Jensen, and
  Alexandrova]{Jorgensen2007}
Jorgensen,~W.~L.; Jensen,~K.~P.; Alexandrova,~A.~N. {Polarization effects for
  hydrogen-bonded complexes of substituted phenols with water and chloride
  ion}. \emph{J. Chem. Theory Comput.} \textbf{2007}, \emph{3},
  1987--1992\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Tkatchenko \latin{et~al.}(2012)Tkatchenko, DiStasio~Jr, Car, and
  Scheffler]{Tkatchenko2012}
Tkatchenko,~A.; DiStasio~Jr,~R.~A.; Car,~R.; Scheffler,~M. Accurate and
  efficient method for many-body van der Waals interactions. \emph{Phys. Rev.
  Lett.} \textbf{2012}, \emph{108}, 236402\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Tkatchenko and Scheffler(2009)Tkatchenko, and
  Scheffler]{Tkatchenko2009}
Tkatchenko,~A.; Scheffler,~M. {Accurate molecular van der Waals interactions
  from ground-state electron density and free-atom reference data}. \emph{Phys.
  Rev. Lett.} \textbf{2009}, \emph{102}, 6--9\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Gobre and Tkatchenko(2013)Gobre, and Tkatchenko]{Gobre2013}
Gobre,~V.~V.; Tkatchenko,~A. {Scaling laws for van der Waals interactions in
  nanostructured materials}. \emph{Nat. Commun.} \textbf{2013}, \emph{4},
  2341\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Horton \latin{et~al.}(2019)Horton, Allen, Dodda, and Cole]{Horton2019}
Horton,~J.~T.; Allen,~A.~E.; Dodda,~L.~S.; Cole,~D.~J. {QUBEKit: Automating the
  derivation of force field parameters from quantum mechanics}. \emph{J. Chem.
  Inf. Model.} \textbf{2019}, \emph{59}, 1366--1381\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Cole \latin{et~al.}(2016)Cole, Vilseck, Tirado-Rives, Payne, and
  Jorgensen]{Cole2016}
Cole,~D.~J.; Vilseck,~J.~Z.; Tirado-Rives,~J.; Payne,~M.~C.; Jorgensen,~W.~L.
  {Biomolecular force field parameterization via atoms-in-molecule electron
  density partitioning}. \emph{J. Chem. Theory Comput.} \textbf{2016},
  \emph{12}, 2312--2323\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Grimme(2014)]{Grimme2014}
Grimme,~S. {A general quantum mechanically derived force field (QMDFF) for
  molecules and condensed phase simulations}. \emph{J. Chem. Theory Comput.}
  \textbf{2014}, \emph{10}, 4497--4514\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Borodin(2009)]{Borodin2009}
Borodin,~O. {Polarizable force field development and molecular dynamics
  simulations of ionic liquids}. \emph{J. Phys. Chem. B} \textbf{2009},
  \emph{113}, 11463--11478\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Cieplak \latin{et~al.}(2009)Cieplak, Dupradeau, Duan, and
  Wang]{Cieplak2009}
Cieplak,~P.; Dupradeau,~F.-Y.; Duan,~Y.; Wang,~J. Polarization effects in
  molecular mechanical force fields. \emph{J. Phys. Condens. Matter}
  \textbf{2009}, \emph{21}, 333102\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Allen \latin{et~al.}(2004)Allen, Andersen, and Roux]{Allen2004}
Allen,~T.~W.; Andersen,~O.~S.; Roux,~B. {Energetics of ion conduction through
  the gramicidin channel}. \emph{Proc. Natl. Acad. Sci.} \textbf{2004},
  \emph{101}, 117 -- 122\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Boulanger and Thiel(2014)Boulanger, and Thiel]{Boulanger2014}
Boulanger,~E.; Thiel,~W. {Toward QM/MM simulation of enzymatic reactions with
  the Drude oscillator polarizable force field}. \emph{J. Chem. Theory Comput.}
  \textbf{2014}, \emph{10}, 1795--1809\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Panel \latin{et~al.}(2018)Panel, Villa, Fuentes, and
  Simonson]{Panel2018}
Panel,~N.; Villa,~F.; Fuentes,~E.~J.; Simonson,~T. {Accurate PDZ/peptide
  binding specificity with additive and polarizable free energy simulations}.
  \emph{Biophys. J.} \textbf{2018}, \emph{114}, 1091--1102\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li \latin{et~al.}(2011)Li, Mei, Zhang, Xie, and Zhang]{Li2011}
Li,~Y.~L.; Mei,~Y.; Zhang,~D.~W.; Xie,~D.~Q.; Zhang,~J. Z.~H. {Structure and
  dynamics of a dizinc metalloprotein: effect of charge transfer and
  polarization}. \emph{J. Phys. Chem. B} \textbf{2011}, \emph{115},
  10154--10162\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bedrov \latin{et~al.}(2019)Bedrov, Piquemal, Borodin, MacKerell, Roux,
  and Schr{\"{o}}der]{Bedrov2019}
Bedrov,~D.; Piquemal,~J.-P.; Borodin,~O.; MacKerell,~A.~D.; Roux,~B.;
  Schr{\"{o}}der,~C. {Molecular dynamics simulations of ionic liquids and
  electrolytes using polarizable force fields}. \emph{Chem. Rev.}
  \textbf{2019}, \emph{119}, 7940--7995\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Olano and Rick(2005)Olano, and Rick]{Olano2005}
Olano,~L.~R.; Rick,~S.~W. {Fluctuating charge normal modes: An algorithm for
  implementing molecular dynamics simulations with polarizable potentials}.
  \emph{J. Comput. Chem.} \textbf{2005}, \emph{26}, 699--707\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Soniat and Rick(2012)Soniat, and Rick]{soniat2012effects}
Soniat,~M.; Rick,~S.~W. The effects of charge transfer on the aqueous solvation
  of ions. \emph{J. Chem. Phys.} \textbf{2012}, \emph{137}, 044511\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Piquemal \latin{et~al.}(2007)Piquemal, Chevreau, and
  Gresh]{piquemal2007toward}
Piquemal,~J.-P.; Chevreau,~H.; Gresh,~N. Toward a separate reproduction of the
  contributions to the Hartree-Fock and DFT intermolecular interaction energies
  by polarizable molecular mechanics with the SIBFA potential. \emph{J. Chem.
  Theory Comput.} \textbf{2007}, \emph{3}, 824--837\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Friesner(2005)]{friesner2005modeling}
Friesner,~R.~A. Modeling polarization in proteins and protein--ligand
  complexes: Methods and preliminary results. \emph{Adv Protein Chem.}
  \textbf{2005}, \emph{72}, 79--104\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Cieplak \latin{et~al.}(2001)Cieplak, Caldwell, and
  Kollman]{cieplak2001molecular}
Cieplak,~P.; Caldwell,~J.; Kollman,~P. Molecular mechanical models for organic
  and biological systems going beyond the atom centered two body additive
  approximation: aqueous solution free energies of methanol and N-methyl
  acetamide, nucleic acid base, and amide hydrogen bonding and chloroform/water
  partition coefficients of the nucleic acid bases. \emph{J. Comput. Chem.}
  \textbf{2001}, \emph{22}, 1048--1057\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ponder and Case(2003)Ponder, and Case]{ponder2003force}
Ponder,~J.~W.; Case,~D.~A. \emph{Advances in protein chemistry}; Elsevier,
  2003; Vol.~66; pp 27--85\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ren and Ponder(2003)Ren, and Ponder]{ren2003polarizable}
Ren,~P.; Ponder,~J.~W. Polarizable atomic multipole water model for molecular
  mechanics simulation. \emph{J. Phys. Chem. B} \textbf{2003}, \emph{107},
  5933--5947\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ngo \latin{et~al.}(2015)Ngo, da~Silva, Kubillus, Li, Roux, Elstner,
  Cui, Salahub, and Noskov]{ngo2015quantum}
Ngo,~V.; da~Silva,~M.~C.; Kubillus,~M.; Li,~H.; Roux,~B.; Elstner,~M.; Cui,~Q.;
  Salahub,~D.~R.; Noskov,~S.~Y. Quantum effects in cation interactions with
  first and second coordination shell ligands in metalloproteins. \emph{J.
  Chem. Theory Comput.} \textbf{2015}, \emph{11}, 4992--5001\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Villa \latin{et~al.}(2018)Villa, MacKerell~Jr, Roux, and
  Simonson]{villa2018classical}
Villa,~F.; MacKerell~Jr,~A.~D.; Roux,~B.; Simonson,~T. Classical Drude
  polarizable force field model for methyl phosphate and its interactions with
  Mg$^{2+}$. \emph{J. Phys. Chem. A} \textbf{2018}, \emph{122},
  6147--6155\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Dudev and Lim(2014)Dudev, and Lim]{Dudev2014}
Dudev,~T.; Lim,~C. {Competition among metal ions for protein binding sites:
  Determinants of metal ion selectivity in proteins}. \emph{Chem. Rev.}
  \textbf{2014}, \emph{114}, 538--556\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ngo \latin{et~al.}(2015)Ngo, da~Silva, Kubillus, Li, Roux, Elstner,
  Cui, Salahub, and Noskov]{Ngo2015}
Ngo,~V.; da~Silva,~M.~C.; Kubillus,~M.; Li,~H.; Roux,~B.; Elstner,~M.; Cui,~Q.;
  Salahub,~D.~R.; Noskov,~S.~Y. {Quantum effects in cation interactions with
  first and second coordination shell ligands in metalloproteins}. \emph{J.
  Chem. Theory Comput.} \textbf{2015}, \emph{11}, 4992--5001\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Dudev \latin{et~al.}(2003)Dudev, Lin, Dudev, and Lim]{Dudev2003}
Dudev,~T.; Lin,~Y.-l.; Dudev,~M.; Lim,~C. First-second shell interactions in
  metal binding sites in proteins: A PDB survey and DFT/CDM calculations.
  \emph{J. Am. Chem. Soc.} \textbf{2003}, \emph{125}, 3168--3180\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Sakharov and Lim(2005)Sakharov, and Lim]{Sakharov2005}
Sakharov,~D.~V.; Lim,~C. {Zn protein simulations including charge transfer and
  local polarization effects}. \emph{J. Am. Chem. Soc.} \textbf{2005},
  \emph{127}, 4921--4929\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Sakharov and Lim(2009)Sakharov, and Lim]{sakharov2009force}
Sakharov,~D.~V.; Lim,~C. Force fields including charge transfer and local
  polarization effects: Application to proteins containing multi/heavy metal
  ions. \emph{J. Comput. Chem.} \textbf{2009}, \emph{30}, 191--202\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ropo \latin{et~al.}(2016)Ropo, Schneider, Baldauf, and
  Blum]{ropo2016first}
Ropo,~M.; Schneider,~M.; Baldauf,~C.; Blum,~V. First-principles data set of
  45,892 isolated and cation-coordinated conformers of 20 proteinogenic amino
  acids. \emph{Sci. Data} \textbf{2016}, \emph{3}, 1--13\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Hu \latin{et~al.}(2022)Hu, Lenz-Himmer, and Baldauf]{hu2022better}
Hu,~X.; Lenz-Himmer,~M.-O.; Baldauf,~C. Better force fields start with better
  data: A data set of cation dipeptide interactions. \emph{Sci. Data}
  \textbf{2022}, \emph{9}, 1--14\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Eastman \latin{et~al.}(2017)Eastman, Swails, Chodera, McGibbon, Zhao,
  Beauchamp, Wang, Simmonett, Harrigan, Stern, Wiewiora, Brooks, and
  Pande]{Eastman2017}
Eastman,~P.; Swails,~J.; Chodera,~J.~D.; McGibbon,~R.~T.; Zhao,~Y.;
  Beauchamp,~K.~A.; Wang,~L.-P.; Simmonett,~A.~C.; Harrigan,~M.~P.;
  Stern,~C.~D.; Wiewiora,~R.~P.; Brooks,~B.~R.; Pande,~V.~S. {OpenMM 7: Rapid
  development of high performance algorithms for molecular dynamics}.
  \emph{PLoS Comput. Biol.} \textbf{2017}, \emph{13}, e1005659\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Hu and Baldauf(2023)Hu, and Baldauf]{Hu2023CTPOLMD}
Hu,~X.; Baldauf,~C. {First release of CTPOL\_MD code}. \emph{zenodo}
  \textbf{2023}, \relax
\mciteBstWouldAddEndPunctfalse
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Hu and Baldauf(2023)Hu, and Baldauf]{Hu2023NOMAD}
Hu,~X.; Baldauf,~C. {FFAFFURR Data set}. \emph{NOMAD} \textbf{2023}, \relax
\mciteBstWouldAddEndPunctfalse
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Blum \latin{et~al.}(2009)Blum, Gehrke, Hanke, Havu, Havu, Ren, Reuter,
  and Scheffler]{BLUM20092175}
Blum,~V.; Gehrke,~R.; Hanke,~F.; Havu,~P.; Havu,~V.; Ren,~X.; Reuter,~K.;
  Scheffler,~M. {Ab initio molecular simulations with numeric atom-centered
  orbitals}. \emph{Comput. Phys. Commun.} \textbf{2009}, \emph{180},
  2175--2196\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Havu \latin{et~al.}(2009)Havu, Blum, Havu, and
  Scheffler]{HAVU20098367}
Havu,~V.; Blum,~V.; Havu,~P.; Scheffler,~M. {Efficient O(N) integration for
  all-electron electronic structure calculation using numeric basis functions}.
  \emph{J. Comput. Phys.} \textbf{2009}, \emph{228}, 8367--8379\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ren \latin{et~al.}(2012)Ren, Rinke, Blum, Wieferink, Tkatchenko,
  Sanfilippo, Reuter, and Scheffler]{Ren_2012}
Ren,~X.; Rinke,~P.; Blum,~V.; Wieferink,~J.; Tkatchenko,~A.; Sanfilippo,~A.;
  Reuter,~K.; Scheffler,~M. Resolution-of-identity approach to Hartree--Fock,
  hybrid density functionals, RPA, MP2 and GW with numeric atom-centered
  orbital basis functions. \emph{New J. Phys.} \textbf{2012}, \emph{14},
  053020\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Perdew \latin{et~al.}(1996)Perdew, Burke, and
  Ernzerhof]{PhysRevLett.77.3865}
Perdew,~J.~P.; Burke,~K.; Ernzerhof,~M. Generalized gradient approximation made
  simple. \emph{Phys. Rev. Lett.} \textbf{1996}, \emph{77}, 3865--3868\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Tkatchenko and Scheffler(2009)Tkatchenko, and
  Scheffler]{PhysRevLett.102.073005}
Tkatchenko,~A.; Scheffler,~M. Accurate molecular van der Waals interactions
  from ground-state electron density and free-atom reference data. \emph{Phys.
  Rev. Lett.} \textbf{2009}, \emph{102}, 073005\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ropo \latin{et~al.}(2016)Ropo, Blum, and Baldauf]{ropo2016trends}
Ropo,~M.; Blum,~V.; Baldauf,~C. Trends for isolated amino acids and dipeptides:
  Conformation, divalent ion binding, and remarkable similarity of binding to
  calcium and lead. \emph{Sci. Rep.} \textbf{2016}, \emph{6}, 1--11\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bultinck \latin{et~al.}(2007)Bultinck, {Van Alsenoy}, Ayers, and
  Carb{\'{o}}-Dorca]{Bultinck2007}
Bultinck,~P.; {Van Alsenoy},~C.; Ayers,~P.~W.; Carb{\'{o}}-Dorca,~R. {Critical
  analysis and extension of the Hirshfeld atoms in molecules}. \emph{J. Chem.
  Phys.} \textbf{2007}, \emph{126}\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Hirshfeld(1977)]{Hirshfeld1977}
Hirshfeld,~F.~L. {Bonded-atom fragments for describing molecular charge
  densities}. \emph{Theor. Chim. Acta} \textbf{1977}, \emph{44}, 129--138\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Singh and Kollman(1984)Singh, and Kollman]{Singh1984}
Singh,~U.~C.; Kollman,~P.~A. {An approach to computing electrostatic charges
  for molecules}. \emph{J. Comput. Chem.} \textbf{1984}, \emph{5}, 129\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bayly \latin{et~al.}(1993)Bayly, Cieplak, Cornell, and
  Kollman]{Bayly1993}
Bayly,~C.~I.; Cieplak,~P.; Cornell,~W.~D.; Kollman,~P.~A. {A well-behaved
  electrostatic potential based method using charge restraints for deriving
  atomic charges: The RESP model}. \emph{J. Phys. Chem.} \textbf{1993},
  \emph{97}, 10269--10280\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Rossi \latin{et~al.}(2014)Rossi, Chutia, Scheffler, and
  Blum]{Rossi2014}
Rossi,~M.; Chutia,~S.; Scheffler,~M.; Blum,~V. {Validation challenge of
  density-functional theory for peptides-example of
  Ac-Phe-Ala$_{5}$-LysH$^{+}$}. \emph{J. Phys. Chem. A} \textbf{2014},
  \emph{118}, 7349--7359\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wales and K.(1997)Wales, and K.]{J.Wales1997}
Wales,~D.~J.; K.,~D.~P. {Global optimization by basin-hopping and the lowest
  energy structures of Lennard-Jones clusters containing up to 110 atoms}.
  \emph{J. Phys. Chem. A} \textbf{1997}, \emph{101}, 5111--5116\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wales and Scheraga(1999)Wales, and Scheraga]{Wales1368}
Wales,~D.~J.; Scheraga,~H.~A. {Global optimization of clusters, crystals, and
  biomolecules}. \emph{Science} \textbf{1999}, \emph{285}, 1368--1372\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Jorgensen \latin{et~al.}(1996)Jorgensen, Maxwell, and
  Tirado-Rives]{Jorgensen1996}
Jorgensen,~W.~L.; Maxwell,~D.~S.; Tirado-Rives,~J. Development and testing of
  the OPLS all-atom force field on conformational energetics and properties of
  organic liquids. \emph{J. Am. Chem. Soc.} \textbf{1996}, \emph{118},
  11225--11236\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ponder and Richards(1987)Ponder, and
  Richards]{doi:10.1002/jcc.540080710}
Ponder,~J.~W.; Richards,~F.~M. {An efficient newton-like method for molecular
  mechanics energy minimization of large molecules}. \emph{J. Comput. Chem.}
  \textbf{1987}, \emph{8}, 1016--1024\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Ren and Ponder(2003)Ren, and Ponder]{Ren2003}
Ren,~P.; Ponder,~J.~W. Polarizable atomic multipole water model for molecular
  mechanics simulation. \emph{J. Phys. Chem. B} \textbf{2003}, \emph{107},
  5933--5947\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Supady \latin{et~al.}(2015)Supady, Blum, and Baldauf]{Supady2015}
Supady,~A.; Blum,~V.; Baldauf,~C. {First-principles molecular structure search
  with a genetic algorithm}. \emph{J. Chem. Inf. Model.} \textbf{2015},
  \emph{55}, 2338--2348\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Tibshirani(1996)]{Tibshirani1996}
Tibshirani,~R. {Regression shrinkage and selection via the lasso}. \emph{J. R.
  Stat. Soc. Ser. B Methodol.} \textbf{1996}, \emph{58}, 267--288\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Hoerl and Kennard(1970)Hoerl, and
  Kennard]{doi:10.1080/00401706.1970.10488634}
Hoerl,~A.~E.; Kennard,~R.~W. {Ridge regression: Biased estimation for
  nonorthogonal problems}. \emph{Technometrics} \textbf{1970}, \emph{12},
  55--67\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Shi and Eberhart(1999)Shi, and Eberhart]{Shi1945}
Shi,~Y.; Eberhart,~R.~C. Empirical study of particle swarm optimization.
  Proceedings of the 1999 congress on evolutionary computation-CEC99 (Cat. No.
  99TH8406). 1999; pp 1945--1950\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Koh \latin{et~al.}(2006)Koh, George, Haftka, and Fregly]{Koh2007}
Koh,~B.-I.; George,~A.~D.; Haftka,~R.~T.; Fregly,~B.~J. Parallel asynchronous
  particle swarm optimization. \emph{Int. J. Numer. Methods Eng.}
  \textbf{2006}, \emph{67}, 578--595\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Pedregosa \latin{et~al.}(2011)Pedregosa, Varoquaux, Gramfort, Michel,
  Thirion, Grisel, Blondel, Prettenhofer, Weiss, Dubourg, \latin{et~al.}
  others]{pedregosa2011scikit}
Pedregosa,~F.; Varoquaux,~G.; Gramfort,~A.; Michel,~V.; Thirion,~B.;
  Grisel,~O.; Blondel,~M.; Prettenhofer,~P.; Weiss,~R.; Dubourg,~V.; others
  Scikit-learn: Machine learning in Python. \emph{J. Mach. Learn. Res.}
  \textbf{2011}, \emph{12}, 2825--2830\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Lee(2014)]{Lee2014}
Lee,~A. 2014; \url{https://pythonhosted.org/pyswarm/}\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Dittner \latin{et~al.}(2015)Dittner, M{\"u}ller, Aktulga, and
  Hartke]{Aktulga2015}
Dittner,~M.; M{\"u}ller,~J.; Aktulga,~H.~M.; Hartke,~B. Efficient global
  optimization of reactive force-field parameters. \emph{J. Comput. Chem.}
  \textbf{2015}, \emph{36}, 1550--1561\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Wang \latin{et~al.}(2017)Wang, McKiernan, Gomes, Beauchamp,
  Head-Gordon, Rice, Swope, Mart{\'\i}nez, and Pande]{wang2017building}
Wang,~L.-P.; McKiernan,~K.~A.; Gomes,~J.; Beauchamp,~K.~A.; Head-Gordon,~T.;
  Rice,~J.~E.; Swope,~W.~C.; Mart{\'\i}nez,~T.~J.; Pande,~V.~S. Building a more
  predictive protein force field: a systematic and reproducible route to
  AMBER-FB15. \emph{J. Phys. Chem. B} \textbf{2017}, \emph{121},
  4023--4039\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Pulay \latin{et~al.}(1979)Pulay, Fogarasi, Pang, and
  Boggs]{pulay1979systematic}
Pulay,~P.; Fogarasi,~G.; Pang,~F.; Boggs,~J.~E. Systematic ab initio gradient
  calculation of molecular geometries, force constants, and dipole moment
  derivatives. \emph{J. Am. Chem. Soc.} \textbf{1979}, \emph{101},
  2550--2560\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Mayne \latin{et~al.}(2013)Mayne, Saam, Schulten, Tajkhorshid, and
  Gumbart]{mayne2013rapid}
Mayne,~C.~G.; Saam,~J.; Schulten,~K.; Tajkhorshid,~E.; Gumbart,~J.~C. Rapid
  parameterization of small molecules using the force field toolkit. \emph{J.
  Comput. Chem.} \textbf{2013}, \emph{34}, 2757--2770\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Klime{\v{s}} and Michaelides(2012)Klime{\v{s}}, and
  Michaelides]{klimevs2012perspective}
Klime{\v{s}},~J.; Michaelides,~A. Perspective: Advances and challenges in
  treating van der Waals dispersion forces in density functional theory.
  \emph{J. Chem. Phys.} \textbf{2012}, \emph{137}, 120901\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Reilly and Tkatchenko(2015)Reilly, and Tkatchenko]{reilly2015van}
Reilly,~A.~M.; Tkatchenko,~A. van der Waals dispersion interactions in
  molecular materials: beyond pairwise additivity. \emph{Chem. Sci.}
  \textbf{2015}, \emph{6}, 3289--3301\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Cole \latin{et~al.}(2016)Cole, Vilseck, Tirado-Rives, Payne, and
  Jorgensen]{cole2016biomolecular}
Cole,~D.~J.; Vilseck,~J.~Z.; Tirado-Rives,~J.; Payne,~M.~C.; Jorgensen,~W.~L.
  Biomolecular force field parameterization via atoms-in-molecule electron
  density partitioning. \emph{J. Chem. Theory Comput.} \textbf{2016},
  \emph{12}, 2312--2323\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Bondi(1964)]{Bondi1964}
Bondi,~A. {van der Waals volumes and radii}. \emph{J. Phys. Chem.}
  \textbf{1964}, \emph{68}, 441--451\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Lee \latin{et~al.}(1989)Lee, Gippert, Soman, Case, and
  Wright]{lee1989three}
Lee,~M.~S.; Gippert,~G.~P.; Soman,~K.~V.; Case,~D.~A.; Wright,~P.~E.
  Three-dimensional solution structure of a single zinc finger DNA-binding
  domain. \emph{Science} \textbf{1989}, \emph{245}, 635--637\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Darden \latin{et~al.}(1993)Darden, York, and
  Pedersen]{darden1993particle}
Darden,~T.; York,~D.; Pedersen,~L. Particle mesh Ewald: An N$\cdot$log(N)
  method for Ewald sums in large systems. \emph{J. Chem. Phys.} \textbf{1993},
  \emph{98}, 10089--10092\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Zhang \latin{et~al.}(2012)Zhang, Yang, Piquemal, and
  Ren]{zhang2012modeling}
Zhang,~J.; Yang,~W.; Piquemal,~J.-P.; Ren,~P. Modeling structural coordination
  and ligand binding in zinc proteins with a polarizable potential. \emph{J.
  Chem. Theory Comput.} \textbf{2012}, \emph{8}, 1314--1324\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Baldauf \latin{et~al.}(2013)Baldauf, Pagel, Warnke, Von~Helden,
  Koksch, Blum, and Scheffler]{baldauf2013cations}
Baldauf,~C.; Pagel,~K.; Warnke,~S.; Von~Helden,~G.; Koksch,~B.; Blum,~V.;
  Scheffler,~M. How cations change peptide structure. \emph{Chem. - Eur. J.}
  \textbf{2013}, \emph{19}, 11224--11234\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Li and Merz~Jr(2016)Li, and Merz~Jr]{li2016mcpb}
Li,~P.; Merz~Jr,~K.~M. MCPB.py: a python based metal center parameter builder.
  \emph{J. Chem. Inf. Model.} \textbf{2016}, \emph{56}, 599--604\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Parzen(1962)]{parzen1962kde}
Parzen,~E. {On Estimation of a Probability Density Function and Mode}.
  \emph{The Annals of Mathematical Statistics} \textbf{1962}, \emph{33}, 1065
  -- 1076\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Rosenblatt(1956)]{rosenblatt1956kde}
Rosenblatt,~M. {Remarks on Some Nonparametric Estimates of a Density Function}.
  \emph{The Annals of Mathematical Statistics} \textbf{1956}, \emph{27}, 832 --
  837\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Donini and Kollman(2000)Donini, and Kollman]{donini2000calculation}
Donini,~O.~A.; Kollman,~P.~A. Calculation and prediction of binding free
  energies for the matrix metalloproteinases. \emph{J. Med. Chem.}
  \textbf{2000}, \emph{43}, 4180--4188\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\bibitem[Berman \latin{et~al.}(2000)Berman, Westbrook, Feng, Gilliland, Bhat,
  Weissig, Shindyalov, and Bourne]{berman2000pdb}
Berman,~H.~M.; Westbrook,~J.; Feng,~Z.; Gilliland,~G.; Bhat,~T.~N.;
  Weissig,~H.; Shindyalov,~I.~N.; Bourne,~P.~E. {The Protein Data Bank}.
  \emph{Nucleic Acids Research} \textbf{2000}, \emph{28}, 235--242\relax
\mciteBstWouldAddEndPuncttrue
\mciteSetBstMidEndSepPunct{\mcitedefaultmidpunct}
{\mcitedefaultendpunct}{\mcitedefaultseppunct}\relax
\EndOfBibitem
\end{mcitethebibliography}


\newpage\null\thispagestyle{empty}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% The same is true for Supporting Information, which should use the
%% suppinfo environment.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{suppinfo}


\renewcommand\thefigure{S\arabic{figure}}

\setcounter{figure}{0}

\begin{figure}[htbp]
\vspace{0.0in}
\begin{center}
\includegraphics[width=0.75\textwidth]{./Figure_cys-Zn_BW_rel_energies.pdf}
\end{center}
\vspace{-0.2in}
\setlength{\belowcaptionskip}{0.2in}
\caption{Boltzmann-type weights vs. relative QM energies at various RTs of AcCys$^-$NMe+Zn$^{2+}$ system.}
\label{SI_BW_Cys-Zn}
\vspace{-0.2in}
\end{figure}

\renewcommand\thetable{S\arabic{table}}
\setcounter{table}{0}

\begin{table}
  \caption{Atom types in HisD+Zn$^{2+}$ and Cys$^{-}$+Zn$^{2+}$. }
  \label{tbl:atom_type_hisD}
  \begin{tabular}{lr | lr}
    \hline
    HisD+Zn$^{2+}$ & & Cys$^{-}$+Zn$^{2+}$ &  \\
    \hline
    Atom  & Atom type & Atom  & Atom type \\
    \hline
    C   & 2177  & C & 1177  \\
    CA & 2166  &  CA & 1166\\
    CB  & 2446 & CB & 1148 \\
    CD2 & 2448 & H & 1183\\
    CE1   & 2447  & HA & 1086 \\
    CG & 2449  & HB2 & 1085 \\
    H  & 2183  & HB3 & 1085 \\
    HA & 2086 & N & 1180 \\
    HB2   & 2085 & O & 1178  \\
    HB3 & 2085 & SG & 1142 \\
    HD1  & 2445 \\
    HD2 & 2091 \\
    HE1   & 2092   \\
    N & 2180  \\
    ND1  & 2444  \\
    NE2 & 2452 \\
    O &  2178 \\
    Zn & 834 \\
    \hline
  \end{tabular}
\end{table}

\begin{table}
  \caption{ The optimized LJ parameters. Epsilon = 0 means the LJ interaction is neglected. LASSO tends to focus on only important factors while neglecting insignificant ones. }
  \label{tbl:opt_LJ_params_full}
  \begin{tabular}{lrlr}
    \hline
    Type1  & Type2 & Sigma (nm)  & Epsilon (kJ/mol) \\
    \hline
    2178   & 834  & 0.31933 & 0.00024413  \\
    2448 & 834  &  0.331094 & 0 \\
    2183  & 834 & 0.32642 & 0.001277 \\
    2446 & 834 & 0.32934 & 0.03138\\
    2177   & 834  & 0.330867 & 0 \\
    2092 & 834  & 0.288564 & 0 \\
    2091 & 834  & 0.288564 & 0 \\
    2180  & 834  & 0.319954 & 0 \\
    2447 & 834 & 0.329767 & 0 \\
    2444   & 834 & 0.31992 & 0.25885  \\
    2445 & 834 & 0.29663 & 6.7250 \\
    2085  & 834 & 0.294726 & 0 \\
    2086  & 834 & 0.294726 & 0 \\
    2184 & 834 & 0.325209 & 0 \\
    2452 & 834 & 0.32598 & 0.00153   \\
    2166 & 834 & 0.331252 & 0.01205  \\
    2449 & 834 & 0.33125 & 0.01205  \\
    \hline
  \end{tabular}
\end{table}

\begin{table}
  \caption{ The CTPOL parameters. The $a$ and $b$ are parameters in eq. \ref{correction_ct}, $r$ is the cutoff distance. The correction factor $k$ in eq. \ref{correction_ct} is set as 3.418. }
  \label{tbl:CTPOL_params}
  \begin{tabular}{lllll}
    \hline
    Type & Polarizability (nm$^{3}$) & a & b & r (nm)  \\
    \hline
    1142 & 0.002668 & -1.037  & 0.323 & 0.312   \\
    1178 & 0.000729 & -0.246  &  0.072 & 0.294  \\
    1180 & 0.00093 & -0.478 & 0.129 & 0.270 \\
    2178 & 0.000721 & -2.667 & 0.722 & 0.271 \\
    2180 & 0.000901 & -0.635 & 0.172 & 0.270  \\
    2452 & 0.000952 & -0.593 & 0.193 & 0.325  \\
    2444 & 0.000879 & -2.424 & 0.843 & 0.348  \\
    444 & 0.000879 \\
    452 & 0.000952 \\
    834 & 0.004383 \\
    166/2166/1166 & 0.001454 \\
    447/2447 & 0.001341 \\
    448/2448 & 0.001416 \\
    80 & 0.001316 \\
    1177 & 0.001473 \\
    446/2446 & 0.001397 \\
    177/2177 & 0.001441 \\
    178 & 0.000721 \\
    184 & 0.001292 \\
    449/2449 & 0.001446 \\
    180 & 0.000901 \\
    1148 & 0.001475 \\
    96 & 0.000724 \\
    250 & 0.001394 \\
    246 & 0.000906 \\
    235 & 0.001339 \\
    81/82 & 0.001431 \\
    243 & 0.000864 \\
    94 & 0.001457 \\
    108 & 0.001497 \\
    214 & 0.000858 \\
    213 & 0.001685 \\
    230 & 0.000810 \\
    179 & 0.000904 \\
    165 & 0.001425 \\
    90 & 0.001471 \\
    251 & 0.001417 \\
    216 & 0.001504 \\
    109 & 0.000711 \\
    99 & 0.001439 \\
    245 & 0.001410 \\
    \hline
  \end{tabular}
\end{table}

\begin{figure}[htbp]
    \centering
    \includegraphics[width=0.8\textwidth]{./Figure_Zn-N_vs_Time.pdf}
    \caption{Average of the two Zn-N distances, where the N are the NE2 atoms of the two histidines in the binding site, as a function of time. }
    \label{SI:zn-n_v_time}
\end{figure}

\begin{table}[]
    \centering
 \begin{tabular}{|c|c|c|c|c|}
\hline
PDBid &  Zn\_sites &  N4HC &  N2H2C &  Min H2O dist \\
\hline
 1MEY &         8 &     7 &      7 &          4.38 \\
 4QF3 &         4 &     4 &      0 &          3.98 \\
 6UEI &         4 &     4 &      0 &          4.24 \\
 6UEJ &         4 &     4 &      0 &          4.30 \\
 2PUY &         4 &     4 &      0 &          4.35 \\
 6FI1 &         4 &     4 &      0 &          9.00 \\
 6FHQ &         4 &     4 &      0 &          4.04 \\
 5YC3 &         2 &     2 &      0 &          6.49 \\
 3T7L &         2 &     2 &      0 &          4.41 \\
 3U9G &         4 &     4 &      0 &          4.26 \\
 4Q6F &         8 &     8 &      0 &          4.32 \\
 3IUF &         1 &     1 &      1 &          5.42 \\
 4BBQ &         8 &     8 &      0 &          4.44 \\
 5YC4 &         2 &     2 &      0 &          6.60 \\
 5Y20 &         2 &     2 &      0 &          5.66 \\
\hline
\end{tabular}
    \caption{PDB ids of Zn fingers. 
    N4HC denotes number of Zn binding sites with 4 His and Cys residues, whereas N2H2C denotes the number of Zn binding sites with exactly 2 His and 2 Cys.
    The last column denotes the distance of the closest water molecule to the Zn ion.}
    \label{SI:tbl:other_pdbs}
\end{table}

\begin{figure}

      \includegraphics[width=0.8\textwidth]{./Figure_ang_between_his.pdf}

\caption{ Probability distributions of select dihedral angles. (top) The dihedral angle between the two coordinating histidine planes. 
    The planes were determined using the CG, CD, and CE atoms of histidine. 
    (bottom) The dihedral angle between plane defined by His1  CG, CD, and CE1 atoms, and plane defined by His1 CG, ND, and NE atoms.
    This is to check for internal distortion of the plane.
    The values are close to 180 (instead of 0) because one set of atoms goes clockwise, and the other counter clockwise, when defining the planes.}
    \label{SI:fig:his_ang}
\end{figure}

\end{suppinfo}

\end{document}
