\documentclass{article}


\usepackage{arxiv}
\usepackage{wrapfig}

\usepackage[utf8]{inputenc} % allow utf-8 input
\usepackage[T1]{fontenc}    % use 8-bit T1 fonts
\usepackage{hyperref}       % hyperlinks
\usepackage{url}            % simple URL typesetting
\usepackage{booktabs}       % professional-quality tables
\usepackage{amsfonts}       % blackboard math symbols
\usepackage{nicefrac}       % compact symbols for 1/2, etc.
\usepackage{microtype}      % microtypography
\usepackage{lipsum}
\usepackage{array}
\usepackage{graphics}
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{physics}
\graphicspath{ {./images/} }
\RequirePackage[authoryear]{natbib}
\title{Improvement of variables interpretability in kernel PCA}


\author{
 Mitja Briscik \\
  Institut de Mathématiques de Toulouse, UMR5219\\
  Université de Toulouse, CNRS, UPS, Cedex 9, 31062\\
  Toulouse, FR \\
  \texttt{mitja.briscik@math.univ-toulouse.fr} \\
  %% examples of more authors
    \And
 Sébastien Déjean \\
   Institut de Mathématiques de Toulouse, UMR5219\\
   Université de Toulouse, CNRS, UPS, Cedex 9, 31062\\
  Toulouse, FR \\
  \texttt{sebastien.dejean@math.univ-toulouse.fr} \\
   \And
 Marie-Agnès Dillies \\
  Institut Pasteur\\
  Université Paris Cité, \\
  Bioinformatics and Biostatistics Hub, F-75015\\
  Paris, FR \\
  \texttt{marie-agnes.dillies@pasteur.fr} \\
}
  %% \AND
  %% Coauthor \\
  %% Affiliation \\
  %% Address \\
  %% \texttt{email} \\
  %% \And
  %% Coauthor \\
  %% Affiliation \\
  %% Address \\
  %% \texttt{email} \\
  %% \And
  %% Coauthor \\
  %% Affiliation \\
  %% Address \\
  %% \texttt{email} \\


\begin{document}
\maketitle
\begin{abstract}
Kernel methods have been proven to be a powerful tool for the integration and analysis of high-throughput technologies generated data. Kernels offer a nonlinear version of any linear algorithm solely based on dot products. The kernelized version of \textit{Principal Component Analysis} is a valid nonlinear alternative to tackle the nonlinearity of biological sample spaces.
This paper proposes a novel methodology to obtain a data-driven feature importance based on the \textit{KPCA} representation of the data.
The proposed method, kernel PCA Interpretable Gradient (KPCA-IG), provides a data-driven feature importance that is computationally fast and based solely on linear algebra calculations. It has been compared with existing methods on three benchmark datasets. The accuracy obtained using KPCA-IG selected features is equal to or greater than the other methods' average. Also, the computational complexity required demonstrates the high efficiency of the method. 
An exhaustive literature search has been conducted on the selected genes from a publicly available Hepatocellular carcinoma dataset to validate the retained features from a biological point of view. The results once again remark on the appropriateness of the computed ranking. 
The black-box nature of kernel PCA needs new methods to interpret the original features. Our proposed methodology KPCA-IG proved to be a valid alternative to select influential variables in high-dimensional high-throughput datasets, potentially unravelling new biological and medical biomarkers. 

\end{abstract}


% keywords can be removed
\keywords{Kernel PCA \and Relevant variables \and Unsupervised learning \and Kernel methods}


\section{Introduction}
The recent advancement in high-throughput biotechnologies is making large multi-omics datasets easily available. Bioinformatics has recently entered the \textit{Big Data} era, offering researchers new perspectives to analyse biological systems to discover new genotype-phenotype interactions. \\
Consequently, new ad-hoc methods to optimise post-genomic data analysis are needed, considering the high complexity and heterogeneity involved. For instance, multi-omics datasets pose the additional difficulty of dealing with a multilayered framework making data integration extremely challenging.
\\ In this context, kernel methods offer a natural theoretical framework for the high dimensionality and heterogeneous nature of omics data, addressing their peculiar convoluted nature \citep{Schlkopf2003}. These methods facilitate the analysis and the integration of various types of omics data, such as vectors, sequences, networks, phylogenetic trees, and images, through a relevant kernel function. Using kernels enables the representation of the datasets in terms of pairwise similarities between sample points, which is helpful for handling high-dimensional sample spaces more efficiently than using Euclidean distance alone. Euclidean distance can be inadequate in complex scenarios, as stated in \citet{Duda2000}, but kernels can help overcome this limitation.
Moreover, kernel methods have the advantage of providing a nonlinear version of any linear algorithm which relies solely on dot products. For instance, Kernel Principal Component Analysis, \citep{Schlkopf1997}, Kernel Canonical Correlation Analysis \citep{Bach}, Kernel Discriminant Analysis \citep{RothS99} and Kernel Clustering \citep{Girolami2002} are all examples of nonlinear algorithms enabled by kernel transformations.\\
This work will focus on the kernelised version of Principal Component Analysis, KPCA,  that provides a nonlinear alternative to the standard PCA to reduce the sample space dimensions.
\\
However, one of the drawbacks of KPCA and kernel methods, in general,  is that they pose new challenges in interpretability. The so-called \textit{pre-image} problem arises since data points are only addressed through the kernel function, causing the original features to be lost during the data embedding process. The initial information contained in the original variables is summarised in the pairwise kernel similarity scores among data sample points.
Thus, retrieving the original input dimensions is highly challenging when it comes to identifying the most prominent features.
Even if it is possible for certain specific kernels to solve the pre-image problem through a fixed-point iteration method, the provided solution is typically numerically unstable since it involves a non-convex optimisation problem \citep{Scholkopf98}. Moreover, in most cases, the exact pre-image does not even exist \citep{Mika1998KernelPA}.\\
However, it is possible to find works that aim at finding the pre-image problem solution, like the pre-image based on distance constraints in the feature space in \citet{Kwok2004} or local isomorphism as in \citet{5995685}.

Instead, this article concentrates on unsupervised feature selection based on kernel PCA.
More specifically, we propose a method to identify the most influential variables for the kernel principal components that account for the majority of the variability of the data.
This procedure provides a computationally fast and stable feature ranking to identify the most prominent original variables. Unimportant descriptors can be thus ignored, refining the kernel procedure whose similarity measure can be influenced by irrelevant dimensions \citep{Brouard2022}.



\subsection{Existing approaches to facilitate feature interpretability in the unsupervised setting}

The literature on unsupervised feature selection is generally less extensive than its supervised learning counterpart. One of the main reasons for this disparity is that the selection is made without a specific prediction goal, making it difficult to evaluate the quality of a particular solution.
In the same way, the unsupervised feature selection field that takes advantage of the kernel framework has been found to be less explored than kernel applications with classification purposes.
As mentioned earlier, interpreting kernel PCA requires additional attention as the kernel principal component axes themselves are only defined by the similarity scores of the sample points.
 However, the literature has limited attempts to explain how to interpret these axes after the kernel transformation. Therefore, feature selection methods based on KPCA are rare.

Among others, \citet{Reverter2014} proposed a method to visualize the original variables into the $2D$ kernel PCs plot. For every sample point projected in the KPCA axes, they propose to display the original variables as arrows representing the vector field of the direction of maximum growth for each input variable or combination of them. This algorithm does not provide variable importance ranking, requiring previous knowledge about which variables to display. 

On the contrary, \citet{Mariette2017} proposed a variable importance selection method to identify the most influential variables for every principal component based on random permutation. The procedure is performed for all variables, selecting the ones that result in the largest Crone-Crosby \citep{Crone1995} distance between kernel matrices, i.e. the variables whose permutations of the observations lead to a significant change in the kernel Gram matrix values. 
However, the method does not come with a variable representation and can be computationally expensive. This method will be denoted as \textbf{KPCA-permute} in the rest of the article.\\
Another method that takes advantage of the kernel framework is the unsupervised method \textbf{UKFS} with its extension \textbf{UKFS-KPCA} in \citet{Brouard2022} where the authors propose to select important features through a non-convex optimization problem with a $\ell_1$ penalty for a Frobenius norm distortion measure.
\\ 
As exhaustively described in the overview presented in \citet{Li2017}, there are different approaches to assess variable importance in an unsupervised setting not based on the kernel framework. Among others, we can mention two methodologies that are based on the computation of a score,  the Laplacian Score \textbf{lapl} in \citet{NIPS2005_b5b03f06}  and its extension Spectral Feature Selection \textbf{SPEC} in \citet{Zhao2007}. 
Other alternatives are the Multi-Cluster Feature Selection \textbf{MCFS} in \citet{Cai2010UnsupervisedFS}, the Nonnegative Discriminative Feature Selection \textbf{NDFS} in \citet{Li2021}, and the Unsupervised Discriminative Feature Selection \textbf{UDFS} in \citet{Yang2011l21R}. These methods aim to select features by keeping only the ones that best represent the implicit nature of the clustered data.
Then, Convex Principal Feature Selection \textbf{CPFS} \citet{Masaeli2010ConvexPF} adopts a distinct approach to feature selection, focusing on selecting a subset of features that can best reconstruct the projection of the data on the initial axes of the Principal Component Analysis. 
\\ As mentioned, the present study introduces a novel contribution to the interpretability of variables in kernel PCA, assuming that the first kernel PC axes contain the most relevant information about the data. The newly proposed method follows and extends the idea proposed by  \citet{Reverter2014}, with the fundamental difference that it gives a data-driven features importance ranking. Moreover, contrarily \textbf{KPCA-permute} in \citet{Mariette2017}, it does not have a random nature while being considerably faster.
From now on, it will be referred to as \textbf{KPCA-IG}, which stands for KPCA Interpretable Gradient.

\section{Methods}
This section presents the formulation behind our proposed method \textbf{KPCA-IG}, starting with describing the kernel framework.    
 \subsection{Kernel PCA}
  
 Given a dataset of $n$ observations $\boldsymbol{x}_1, \ldots, \boldsymbol{x}_n$ with $\boldsymbol{x}_i \in$ \scalebox{1.5}{$\chi$}, a function k defined as k: \scalebox{1.5}{$\chi$} $\times$ \scalebox{1.5}{$\chi$} $\longrightarrow {\rm I\!R}$ is a valid kernel if it is symmetric and positive semi-definite i.e. $k(\boldsymbol{x}_i, \boldsymbol{x}_j) = k(\boldsymbol{x}_j, \boldsymbol{x}_i)$ and  c'$\boldsymbol{K}$c $\geqslant$ 0, $\forall c \in {\rm I\!R} $, where $\boldsymbol{K}$ is the  $n \times n$ kernel matrix containing all the data pairwise similarities $\boldsymbol{K} = k(x_i,x_j) $.
The input set \scalebox{1.5}{$\chi$} does not require any assumption. In this work  we consider it to be \scalebox{1.5}{$\chi$}$= {\rm I\!R}^d $. \\
  Every kernel function is associated with an implicit function $\phi$:   $\scalebox{1.5}{$\chi$}$ $\longrightarrow$ $\mathcal{H}$ which maps the input points into a generic feature space $\mathcal{H}$,  with possibly  an infinite dimensionality,   with the expression
$k(\boldsymbol{x}_i, \boldsymbol{x}_j) = \langle \phi (\boldsymbol{x}_i), \phi (\boldsymbol{x}_j)\rangle$. This relation allows to compute the dot products in the feature space, implicitly applying the kernel function to the input objects,  without explicitly computing the mapping function $\phi$. 
\\ Principal Component Analysis is a well-established linear algorithm to extract the data structure in an unsupervised setting \citep{Friedman2009}.
However, it is commonly accepted that in specific fields, such as bioinformatics, assuming a linear sample space may not help to capture the data manifold adequately \citep{Reverter2014}. In other words, the relationships between the variables may be nonlinear, making linear methods unsuitable. Hence, with high-dimensional data such as genomic data, where the number of features is usually much larger than the number of samples, nonlinear methods like kernel methods can provide a valid alternative for data analysis. 
\\ A compelling approach to overcome this challenge is through kernel PCA, which was introduced in \citet{Schlkopf1997}. Kernel PCA applies PCA in the feature space generated by the kernel, and as PCA relies on solving an eigenvalue problem, its kernelized version operates under the same principle. The algorithm requires the data to be centered in the feature space, and the diagonalization of the centered covariance matrix in the feature space $\mathcal{H}$ is equivalent to the eigendecomposition of the kernel matrix $\boldsymbol{K}$. 
The data coordinates in the feature space are unknown as $\phi$ is not explicitly computed. Consequently, the required centering of variables in the feature space cannot be done explicitly. However, it is possible to compute the centered Gram matrix $\boldsymbol{\tilde{K}}$ as $\boldsymbol{\tilde{K}} = ( \boldsymbol{K} - \boldsymbol{1}_n\boldsymbol{K}  - \boldsymbol{K} \boldsymbol{1}_n  +  \boldsymbol{1}_{n}' \boldsymbol{K} \boldsymbol{1}_n )$ with $\boldsymbol{1}_n$ a vector with length n and $1$ for all entries. 
If we express the eigenvalues of  $\boldsymbol{\tilde{K}}$ with $\lambda_1 \ge \lambda_2 \ge \ldots \ge \lambda_n$ and the corresponding set of eigenvectors $\boldsymbol{a}^1,\ldots,\boldsymbol{a}^n$, the principal component axes  can be expressed as $  \boldsymbol{v}^k = \sum_{i=1}^{n} a_{i}^k \phi(\boldsymbol{x}_i)$ with $\boldsymbol{v}^k$ and $\boldsymbol{a}^k$ orthonormal in  $\mathcal{H}$ with $k = 1, \dots, n$.
Thus solving  $n \boldsymbol{\lambda} \boldsymbol{a} = \boldsymbol{\tilde{K}}\boldsymbol{a}$, it is possible to compute the projection of the points into the subspace of the feature space spanned by the eigenvectors. The projection of a test point $\boldsymbol{x}$ into the $k$-th eigenvector becomes then 
$ \rho_k :=  \langle \boldsymbol{v}_k, \phi(\boldsymbol{x}) \rangle = \sum_{i=1}^{n} a_{i}^k k(\boldsymbol{x}, \boldsymbol{x}_i ) $.
Likewise, utilizing the concise, explicit form of the centered gram matrix $\boldsymbol{\tilde{K}}$, it is possible to express the projection of an arbitrary point $\boldsymbol{x}$ onto the $k$-th kernel principal component as follows:

\begin{eqnarray}\label{Xprojection}
\rho_k &=& \Bigg( k(\boldsymbol{x}, \boldsymbol{x}_i)^T - \frac{1}{n} \boldsymbol{1}_n^T K \Bigg) \Bigg( \boldsymbol{I}_n - \frac{1}{n}  \boldsymbol{1}_n \boldsymbol{1}_n^T \Bigg) \Tilde{\boldsymbol{v}}
\nonumber\\
\end{eqnarray}
As we observe, the kernel PCA algorithm can be mathematically represented using only the entries of the kernel matrix. This means that the algorithm operates entirely on the original input data without requiring the computation of new data coordinates in the feature space. This technique effectively resolves the issue of potentially high computational complexity by allowing the input points to be implicitly mapped into the feature space. However, it also introduces new challenges in terms of interpretation.
Determining which input variables have the most significant impact on the kernel principal components can be highly challenging, making it difficult to interpret them in terms of the original features. In other words, since the kernel function maps the data to a higher-dimensional feature space, it can be hard to understand how the original features contribute to the newly obtained kernel principal components.
In the previous section, we have mentioned the few techniques available in the literature that can be used to gain insight into the original input variables that had the most influence on the KPCA solution. The following section presents our contribution to providing practitioners with a data-driven and faster variable ranking methodology.

\subsection{Improvement of variable interpretability in KPCA}



It is known that gradient descent is one of the most common algorithms for the training phase of most neural networks \citep{Ruder2016}. In this framework, the norm of the cost function gradient plays a crucial role as it determines the step size of the update at each iteration. Together with the direction, its magnitude is a measure of the steepness of the cost function at a particular point in the parameter space. If the gradient magnitude is high, the cost function changes rapidly; thus, the parameters must be updated with a larger step size. Conversely, if the gradient magnitude is low, it indicates that the cost function is changing slowly, and therefore, the parameters need to be adjusted with a smaller step size.
In the same way, we propose to compute at each sample point the norm of the partial derivative of every feature curve projected into the eigenspace of the Gram matrix. When the norm of the partial derivative of a variable is high, it means that the variable substantially affects the position of the sample points in the kernel PC axes. Conversely, when the norm of the partial derivative of a variable is small, the variable can be deemed negligible for the kernel principal axes.
Thus, the main idea is to compute the lengths of the gradient vectors for every variable at each sample point as they represent how steep the direction given by the partial derivative of the induced curve is.
Some works in the neuroimaging and earth system sciences domain have also shown that kernel derivatives may indicate the influence carried by the original variables as in \citet{Rasmussen2011} and \citet{Johnson2020}.
Followingly in this work, we apply this intuition to the KPCA framework.
\\ Analytically, \citet{Reverter2014} and \citet{Sanz2018} showed how every variable $j = 1, \dots, p$ can be represented in the input space with a real-valued function $\boldsymbol{f}(\boldsymbol{x})^j$ representing the position of the every sample point $\boldsymbol{x}$ in the input space \scalebox{1.5}{$\chi$}$= {\rm I\!R}^d $. This function will be simply addressed as $\boldsymbol{f}^j$ for the rest of the work.\\
We are interested in the projection of these $j$ functions onto the linear subspace of the feature space induced by the kernel. First, we can express the projection of $\boldsymbol{f}^j$ in the feature space through the implicit map $\phi$ as $\boldsymbol{h}^j$.\\
In the previous section, we showed how to represent the projection of every mapped sample point  $\phi(\boldsymbol{x})$ into the subspace spanned by the eigenvectors of $\boldsymbol{\tilde{K}}$ in Equation (\ref{Xprojection}).\\ 
Similarly, every function $\boldsymbol{h}^j$ can be further projected from the feature space into the subspace of the kernel PCA.

\begin{eqnarray}\label{hprojection}
 \varphi_{1 \times q}^j = \Bigg(  k(\boldsymbol{h^{j}}, \boldsymbol{x}_i)^T - \frac{1}{n}  \boldsymbol{1}_n^T \boldsymbol{K} \Bigg) \Bigg( \boldsymbol{I}_n - \frac{1}{n}  \boldsymbol{1}_n \boldsymbol{1}_n^T \Bigg) \Tilde{\boldsymbol{v}}
\nonumber\\
\end{eqnarray}
with $q$ the number of kernel PCs.
In order to assess the influence of the $j$-th variable on the coordinates of the data points into the kernel principal axes, we first compute the derivative of $\boldsymbol{h}^j$ with respect to the $j$-th variable at each sample point.\\
Formally, the partial derivative of the projected curves of $\boldsymbol{h}^j$ computed at the generic point $\boldsymbol{x}$ can be defined as follows:

\begin{eqnarray}\label{eqexpmuts}
 w^j &=&  \pdv{\varphi^j}{x^j} =  
  \pdv{k(\boldsymbol{x},\boldsymbol{x}_i)}{x^j}
    \Bigg( \boldsymbol{I}_n - \frac{1}{n}  \boldsymbol{1}_n \boldsymbol{1}_n^T \Bigg) \Tilde{\boldsymbol{v}} 
    \nonumber\\
    &=&  
    \Big[ \pdv{k(\boldsymbol{x},\boldsymbol{x}_1)}{x^j}, ... , \pdv{k(\boldsymbol{x},\boldsymbol{x}_n)}{x^j}\Big]^T  \Bigg( \boldsymbol{I}_n - \frac{1}{n}  \boldsymbol{1}_n \boldsymbol{1}_n^T \Bigg) \Tilde{\boldsymbol{v}} 
    \nonumber\\
    \end{eqnarray}

As said, the direction of maximal variation associated with the variable $j$ is given by this partial derivative of the projected curve as exploited in \citep{Reverter2014}. 

If $\varphi^j$ represents the input variable $j$ locally in the kernel PC axes, $w^j$ is the $n \times q$ matrix giving the direction and length of the gradient associated with the $j$-th variable for each input point.
The mean value of the norm of this partial derivative computed for all the sample points suggests this variable's influence.
Analytically:

\begin{eqnarray}\label{scores}
 r^j = \frac{1}{n}\sqrt{ \sum_{i=1}^{n} \Bigg( \pdv{\varphi^j}{x_i^j}\Bigg)^2} &=&  
  \frac{1}{n} \sum_{i=1}^{n}\sqrt{\sum_{t=1}^{q}\Big(w_{it}^j\Big)^2}
    \nonumber\\
\end{eqnarray}
Thus, $r^j$ is the mean of the norm vectors of the partial derivative of $\varphi^j$ among all the $n$ sample points, giving an indication of the overall influence of the $j$-th variable on the points.\\
Finally, we can group all the $j$ means in a sorted vector representing the ranking of the original features proposed by the method, i.e. $\boldsymbol{r} = (r^1, \dots, r^p)$. Every entry of $\boldsymbol{r}$ is a score that indicates the impact of every variable on the kernel PCA representation of the data, from the most influential to the least important. \\
The method is non-iterative, and it only requires linear algebra. Thus, it is not susceptible to numerical instability or local minimum problems.
It is computationally very fast, and it can be applied to any kernel function that admits a first-order derivative.
The described procedure has been implemented on R, and the code can be available upon request to the authors.


\section{Results}


We conducted experiments on three benchmark datasets from the biological domain to assess the accuracy of the proposed unsupervised approach for feature selection. These datasets include two microarray datasets, named \textit{Carcinom} and \textit{Glioma}, which are available in the Python package scikit-feature \citet{Li2017} and the gene expression data from normal and prostate tumour tissues \citet{Chandran2007},  \textit{GPL93}  from the GEO, a public functional genomics data repository.
 \textit{Glioma} contains the gene expression of $4434$ for $50$ patients, while \textit{Carcinom} $9182$ for $174$ individuals. Both datasets have already been used as a benchmark in numerous studies including several methods comparisons, such as  \citet{Li2017} and \citet{Brouard2022}. Then, the dataset \textit{GPL93}  contains the expression of $12626$ genes for $165$ patients, and it has been chosen for its complexity and higher dimensionality.
\\

The idea is to compare the proposed methodology \textbf{KPCA-IG} with existing unsupervised feature selection methods from diverse frameworks, as conducted in \citet{Li2017} and \citet{Brouard2022}:
\begin{itemize}
    \item From  \citet{NIPS2005_b5b03f06}, \textbf{lapl} to include one method that relies on the computation of a score 
    \item \textbf{NDFS} \citet{Li2021}, to add one of the methods primarily designed for clustering. It is based on the implicit assumption that samples are structured into subgroups and demands the a priori definition of the number of clusters.
    \item \textbf{KPCA-permute} in \citet{Mariette2017} available in the \textbf{mixKernel} R package to include another methodology from the context of kernel PCA.
    \end{itemize}

To evaluate the selected features provided by the four methods, we measured the overall accuracy (ACC) and normalized mutual information (NMI) \citep{NMI} based on k-means cluster performance. For each method, the k-means clustering ACC and NMI have been obtained using several subsets with a different number $d$ of selected features, with $d \in \{10,20, \dots,290, 300\}$. Thus, the relevance of the selected values has been estimated according to their ability to reconstruct the clustered nature of the data. More specifically, the three datasets \textit{Glioma}, \textit{Carcinom} and \textit{GPL93} are characterized by $4$, $11$ and $4$ groups respectively. Thus, the k-means clustering was computed using the correct number of clusters in the datasets to obtain a metric for the capability of the selected features to keep this nature.
\begin{table}[h!]
\caption{Comparison of the different methods in terms of mean ACC and NMI over $20$ runs of a k-means clustering for several subsets with a different number d of selected features. CPU represents the computational time in seconds required by the four methods only to find the most influential features.}
      \begin{tabular}{ccccc}
      \hline
      &   &   &  & \\
           & lapl  & NDFS  & KPCA-permute & KPCA-IG\\  &   &   &  & \\\hline
           &   &   &  & \\
            \textit{Glioma} ($n=50, p =4434)$& && \\ &   &   &  & \\\hline
            &   &   &  & \\
        ACC(10) & $\boldsymbol{0.50}$ (0.02) & 0.37 (0.04)  &  0.48 (0.03) & 0.42 (0.01)  \\
        NMI(10) & $\boldsymbol{0.34}$ (0.02) & 0.13 (0.03)  & 0.31 (0.02)   & 0.21 (0.01)  \\
        ACC(150) & $\boldsymbol{0.56}$ (0.03) & 0.53 (0.04)   & 0.54 (0.04)   & $\boldsymbol{0.56 (0.05)}$  \\
        NMI(150) & $\boldsymbol{0.50}$ (0.02) & 0.41 (0.03)  & 0.48 (0.02)   & 0.36  (0.05) \\
        ACC(300) & 0.54 (0.04) & 0.55 (0.04)   & 0.52 (0.03)   & $\boldsymbol{0.57}$ (0.05)  \\
        NMI(300) & $\boldsymbol{0.48}$ (0.03) & 0.41 (0.03)  & 0.45 (0.02)   & 0.35 (0.05)  \\
        CPU time & 0.4 & 84.6  & 620.9   & 2.9  \\
        &   &   &  & \\
        \hline &   &   &  & \\
         \textit{Carcinom} ($n=174, p =9182)$& && \\ &   &   &  & \\\hline
         &   &   &  & \\
        ACC(10) & 0.27 (0.02) & 0.47 (0.04)  &  0.48 (0.02) & $\boldsymbol{0.51}$ (0.01)  \\
        NMI(10) & 0.23 (0.01) & 0.48 (0.03)  & 0.43 (0.01)   & $\boldsymbol{0.49}$ (0.02)  \\
        ACC(150) & 0.61 (0.03) & 0.68 (0.04)   & 0.67 (0.03)   & $\boldsymbol{0.70}$ (0.02)  \\
        NMI(150) & 0.62 (0.03) & $\boldsymbol{0.72}$ (0.03)  & 0.69 (0.03)   & 0.70  (0.03) \\
        ACC(300) & 0.69 (0.04) & 0.69 (0.04)   & $\boldsymbol{0.70}$ (0.048)   & 0.69 (0.03)  \\
        NMI(300) & 0.73 (0.03) & $\boldsymbol{0.73}$ (0.03)  & 0.71 (0.03)   & 0.70 (0.02)  \\
        CPU time & 1.4 & 391.8  & 7937.6  & 30.5   \\
        &   &   &  & \\
          \hline &   &   &  & \\
         \textit{GPL93} ($n=165, p =12626)$& && \\ &   &   &  & \\\hline
         &   &   &  & \\
        ACC(10) & 0.38 (0.01) & 0.41 (0.01)  &  $\boldsymbol{0.42}$ (0.01) & 0.40 (0.01)  \\
        NMI(10) & 0.08 (0.01) & 0.109 (0.07)  & 0.11 (0.01)   & $\boldsymbol{0.15}$ (0.01)  \\
        ACC(150) & 0.38 (0.01) & 0.45 (0.01)   & $\boldsymbol{0.60}$ (0.02)   & 0.58 (0.01)  \\
        NMI(150) & 0.07 (0.01) & 0.18 (0.02)  & $\boldsymbol{0.39}$ (0.01)   & 0.29 (0.01) \\
        ACC(300) & 0.37 (0.01) & 0.49 (0.03)   & 0.56 (0.05)   & $\boldsymbol{0.56}$ (0.01)  \\
        NMI(300) & 0.07 (0.01) & 0.22 (0.02)  & 0.22 (0.01)   & $\boldsymbol{0.31}$ (0.01)  \\
        CPU time & 2.1 & 1277.4  & 17691.4   & 39.8   \\
        &   &   &  & \\
      \hline
      \end{tabular}
      \label{Table1}
\end{table}
Note that only the \textbf{NDFS} method is implemented to explicitly obtain an optimum solution in terms of clustering, also requiring in advance the number of groups in the data.
For each method, the k-means clustering was run 20 times to obtain a mean of the overall accuracy and normalized mutual information for each of the $30$ subsets of selected features.
Both our novel method \textbf{KPCA-IG} and \textbf{KPCA-permute} have been employed with a Gaussian kernel with a sigma value depending on the dataset. The selected features are, in both cases, based on the first 3, 5 and 3 kernel PC axes for  \textit{Glioma}, \textit{Carcinom} and \textit{GPL93}, respectively.
The CPU time in seconds required to obtain the feature ranking for all the methods has also been observed. The experiment was conducted on a standard laptop Intel Core $i5$ with $16GB$ RAM.
\subsection{Evaluation on benchmarks datasets}

In Table \ref{Table1}, we can see the results in terms of mean Accuracy and NMI over $20$ runs for different numbers of retained features $d$. For the first dataset \textit{Glioma} \textbf{lapl} seems to show the best performance in terms of NMI and AUC, except when $d=300$ where the Accuracy obtained with \textbf{KPCA-IG} is the highest, even if all the methods seem to behave very similarly in terms of ACC.
Analyzing the results for the other two datasets \textit{Carcinom} and \textit{GPL93} that are considerably bigger and possibly more complex in terms of sample space manifold, the two methods based on the kernel framework exhibit to surpass the \textbf{lapl} and \textbf{NDFS} approaches, especially in the \textit{GPL93} datasets. The comparison of the different approaches in terms of NMI and ACC of these two datasets can also be observed in Figure \ref{Figure1} and Figure \ref{Figure2}.

\begin{figure}[h!]
\centering
  \includegraphics[width=0.91\textwidth]{Figure1.png}
  \caption{ \textit{Carcinom} ACC and NMI:
     Comparison of the performance of the four methods on the \textit{Carcinom} dataset in terms of Accuracy (left) and Normalized Mutual Information (right) as a function of the number of selected features $d$. ACC and NMI are computed for the k-means results using only the d selected features.}
      \label{Figure1}
  \end{figure}



Moreover, as shown in \citet{Brouard2022} \textbf{NDFS} and the other cluster-based methods like \textbf{MCFS} and \textbf{UDFS} suffer if the user selects an incorrect decision for the a priori number of clusters. In our case, we show that the proposed methodology behaves similarly or even better to a method like \textbf{NDFS} that is specifically optimized for this cluster setting.

\begin{figure}[h!]
\centering
  \includegraphics[width=0.92\textwidth]{Figure2.png}
  \caption{\textit{GPL93} ACC and NMI:
     Comparison of the performance of the four methods on the \textit{GPL93} dataset in terms of Accuracy (left) and Normalized Mutual Information (right) as a function of the number of selected features $d$. ACC and NMI are computed for the k-means results using only the d selected features.}
       \label{Figure2}
  \end{figure}
The two kernel-based approaches, namely \textbf{KPCA-permute} and our novel method \textbf{KPCA-IG}, reveal an excellent performance in this setting, once again displaying the appropriateness of the kernel framework in the context of complex biological datasets. 
However, \textbf{KPCA-IG} can provide these above-average performances with a considerably lower CPU time.

Only \textbf{lapl} seems as fast as \textbf{KPCA-IG} while showing  poorer results in the more complex scenario represented in this case by the \textit{GPL93} dataset.
Other methods, such as the concrete autoencoder in \citet{abid2019}, have proven successful in this context. 
The results obtained with the concrete autoencoder, as demonstrated in \citet{Brouard2022}, were comparable or even inferior in terms of accuracy and NMI. Furthermore, the computational time required to achieve these results was on the order of days. As a result, we opted not to include it in our simulations.
\newpage

\section{Application on Hepatocellular carcinoma dataset}


\begin{figure}[h!]
  \centering
\includegraphics[width=0.6\textwidth]{Figure3.jpeg}
  \caption{Kernel PCA on HCC (GSE102079) Dataset}
      \label{Figure3}
\end{figure}
Liver cancer is a global health challenge, and it is estimated that there will be over $1$ million cases by $2025$. Hepatocellular carcinoma (HCC) is the most common type of liver cancer, accounting for around $90\%$ of cases \citep{Llovet2021}.
The most significant risk factors associated with  HCC are, among others, chronic hepatitis B and C infections, nonalcoholic fatty disease, and chronic alcohol abuse \citep{Morse2019}.
To analyse the use of the \textbf{KPCA-IG} method, we used the expression profiling by array of an HCC dataset from the Gene expression Omnibus (series GSE102079).





It contains the gene expression microarray profiles of $3$ groups of patients. First, $152$ patients with HCC who were treated with hepatic resection between $2006$ and $2011$ at Tokyo Medical and Dental University Hospital. Then, the gene expression of normal liver tissues of $14$ patients as control \citep{Chiyonobu2018}.
The third group contains the gene expression of $91$ patients with liver cancer but of non-tumorous liver tissue. 
The total expression matrix for the $257$ patients contains the expression of $54613$ genes, and the data has been normalised by robust multichip analysis (RMA) as in \citet{affy} and scaled and centered before applying KPCA.
To show the potentiality of KPCA-IG, we first perform kernel PCA with radial basis kernel with $\sigma = 0.00001$, which was set heuristically to maximize the explained variance and obtain a clear two dimension data representation.

In Figure \ref{Figure3}, it is possible to see the application of kernel Principal Component on the HCC dataset. 
Even if detecting groups is not the optimization criterion of kernel PCA, it is possible to see that the algorithm catches the dataset's clustered structure.
For this reason, applying a method like the proposed KPCA-IG can enlighten the kernel component axes, possibly giving an interpretation of the genes' influence on the sample points representation.
The KPCA-IG provides a feature ranking based on the KPCA solution, in this case, based on the first $2$ kernel Principal Components.
\begin{figure}[h!]
\centering
  \includegraphics[width=0.5\textwidth]{Figure4.jpeg}
   \caption{Distribution of the scores for the ordered $54613$ genes, from a maximum of $0.428$ to a minimum of $0.05\times 10^3$}
       \label{Figure4}
  \end{figure}
As mentioned before, one of the main advantages of the proposed method is the fast computational time required, as with this high-dimensional dataset, the CPU time was $654.1$ seconds.
Table \ref{Table2} presents the first $25$ genes and Figure \ref{Figure4} the distribution of the $54613$ variables scores.

One possible way to assess the relevance of the obtained ranking is first to visualize the genes with the method proposed by \citet{Reverter2014} to see if the bio-medical community has already found the retained genes. 
For instance, Figure \ref{Figure5} displays the representation of the variable  237350\_at (TTC36), the second gene in the ranking provided by KPCA-IG. 
The direction of the arrows suggests an upper expression of the gene towards the cluster of patients that do not have liver cancer or patients whose liver tissue is not tumorous. 

\begin{table}[h!]
\centering
\begin{tabular}{rrr}
 \hline
      &  \\
Genes & Score & Symbol  \\ 
&  & \\
  \hline
   & & \\
1555797\_a\_at & 0.427972 & ARPC5\\ 
  237350\_at & 0.426140 & TTC36\\ 
  1559573\_at & 0.424048 &  LINC01093 \\ 
  230478\_at & 0.420690 &  OIT3\\ 
  203213\_at & 0.417682 &  	CDK1 \\ 
  205019\_s\_at & 0.417597 & VIPR1 \\ 
  1559065\_a\_at & 0.417234 &  	CLEC4G \\ 
  205984\_at & 0.417234 & CRHBP \\ 
   220114\_s\_at & 0.416410 & STAB2 \\ 
  202604\_x\_at & 0.416228 & ADAM10 \\ 
  220496\_at & 0.415608 & CLEC1B \\ 
  205866\_at & 0.414893 & FCN3\\ 
  214895\_s\_at & 0.414887 & ADAM10\\ 
  240963\_x\_at & 0.413698 & PLXDC1 \\ 
  234304\_s\_at & 0.413574 & IPO11 \\ 
  222077\_s\_at & 0.412939 & RACGAP1\\ 
  223341\_s\_at & 0.411044 & SCOC \\ 
  214710\_s\_at & 0.410616 & CCNB1 \\ 
  218009\_s\_at & 0.410610 & PRC1\\ 
  219918\_s\_at & 0.410460 &  ASPM\\ 
  226524\_at & 0.410119 & C3orf38\\ 
  201890\_at & 0.410097 & RRM2 \\ 
  207804\_s\_at & 0.409962 & FCN2 \\ 
  210481\_s\_at & 0.409839 & CLEC4M \\ 
  209470\_s\_at &   0.409759  & GPM6A\\
  \dots & \dots \\ 
  229461\_x\_at & 0.0520878 & NEGR1 \\
   230538\_at & 0.0520119 & SHC4\\
    206145\_at & 0.0507935 & RHAG\\
 \hline
   &  \\
\end{tabular}\caption{The $25$ most relevant genes and the last $3$ out of the total number of $54613$ according to the proposed KPCA-IG method. The original scores have been multiplied by $10^3$ for a better visualization}
\label{Table2}
\end{table}
 \begin{figure}[h!]
 \centering
  \includegraphics[width=0.8\textwidth]{Figure5.jpeg}
   \caption{Gene 237350\_at (TTC36): Gene visualization obtained using the procedure described in \citet{Reverter2014}}
       \label{Figure5}
  \end{figure}

To validate the procedure, we selected relevant literature about the gene TTC36. This gene, also known as HBP21, is a protein encoder gene. It has been shown that this gene's encoded protein may function as a tumour suppressor in hepatocellular carcinoma (HCC) since it promotes apoptosis while it has been proven to be downregulated in HCC cases \citep{Jiang2015}.
\
 \\ Another gene that shows differential expression in the two groups is 203213\_at (CDK1). In this case, Figure \ref{Figure6} suggests that this gene seems to be upregulated in the presence of cancer tissue. 
 \begin{figure}[h!]
  \centering
  \includegraphics[width=0.8\textwidth]{Figure6.jpeg}
   \caption{Gene 203213\_at (CDK1): Gene visualization obtained using the procedure described in \citet{Reverter2014} }
       \label{Figure6}
  \end{figure}
The indication found in multiple studies is that the increased expression of this gene is indeed linked with a poorer prognosis or outcome, such as high tumour grade, invasion of lymphovascular or muscularis propria, and the presence of distant metastasis \citet{Heo2022}, \citet{Li2020}, \citet{Sofi2022}, \citet{JLi2020}. \citet{Liu2022}.
In the same way, another of the most critical genes, according to KPCA-IG, that seems to be prominent in the case of an HCC patient reflecting the same indication in the medical literature, is $202604\_x\_at$ (ADAM10), known to be involved in the RIPing and shedding of numerous substrates leading to cancer progression and inflammatory disease  \citet{Krzystanek2016}, and indicated as a target for cancer therapy \citet{Moss2008}, \citet{Crawford2009}, while being upregulated in metastasis cancers
\citet{Lee2010},   \citet{Gavert2007}.
Rac GTPase activating protein $1$ gene RACGAP1 (222077\_s\_at) selected by KPCA-IG shares a similar behaviour with ADAM10 and CDK1. The literature concerning this gene is also broad, where it has been marked as a potential prognostic and immunological biomarker in different types of cancer, such as gastric cancer \citet{Saigusa2014}, uterine carcinosarcoma \citet{Mi2016}, breast cancer \citet{Pliarchopoulou2012} or colorectal cancer \citet{Imaoka2015} among many others.
CCNB1 (202604\_x\_at) has also been indicated to be an oncogenic factor in the proliferation of HCC cells \citet{Chai2018}, showing a significant impact on the patient's survival time \citet{Ding2014}, \citet{Zhuang2018}  and thus has been targeted for cancer treatments \citet{Fang2014}. PRC1 has revealed upper expression in other cancer tissues such as, among others, invasive cervical carcinomas \citet{Santin2005},  papillary renal cell carcinoma \citet{Yang2005}, 
pediatric adrenocortical tumour \citet{West2007}, while yet not being studied in depth as compared to  CCNB1, ADAM10 or RACGAP1. \\
ASPM (219918\_s\_at) was known initially as a gene involved in the control of the human brain development and in the cerebral cortical size   \citet{Zhang2003}, \citet{Bond2002} whose mutations may lead to primary autosomal recessive microcephaly \citet{Kouprina2005}, more recently its overexpression has also been linked with tumour progression as in \citet{Wang2013}, \citet{Xu2018}.
\\ Lastly, for the group of upregulated genes in HCC, RRM2 (201890\_at) has also been linked with low overall survival \citet{Chen2019}, \citet{Jin2020}, 
leading to exhaustive cancer research suggesting targeting its inhibition for different types of tumour treatments \citet{Rahman2012},\citet{Wang2014_2}, \citet{Osako2019}, \citet{Ohmura2021}.
\\ On the other hand, the selected genes that manifest down-regulation in cancerous HCC tissues are LINC01093, OIT3, VIPR1, CLEC4G, CRHBP, STAB2, CLEC1B,  FCN3, FCN2 and CLEC4M.
The literature regarding these genes indicates that they work as suppressors in different cancerous situations, once again endorsing the selection provided by KPCA-IG for the upregulated genes.





The few genes in the first 25 selected by KPCA-IG that do not exhibit differential expression using \citet{Reverter2014} method (ARPC5, IPO11, C3orf38, SCOC) are potential genes that explain much variability in the data or that share a possibly nonlinear interaction with the differential expressed genes. Since the ultimate goal of KPCA is not to discriminate groups, it is expected that some of the variables found by the novel method are not linked with a classification benefit. However, further follow-up on the function of these genes may be done in cooperation with an expert in the field.

\section{Conclusion}

We have seen how the unsupervised feature selection literature is narrower than its supervised counterpart. Moreover, algorithms that use the kernel Principal Component Analysis for feature selection are reduced to a few works.
In the present work, we have introduced a novel method to enhance variables' interpretability in kernel PCA. Using benchmark datasets, we have proven the comparability in terms of accuracy with already existing and recognized methods, where the efficiency of KPCA-IG has proven to be competitive. The application on the real-life Hepatocellular carcinoma dataset and the validation obtained from the comparison of the selected variables by the method with the bio-medical literature have confirmed the effectiveness and strengths of the proposed methodology.\\  
In future works, further in-depth analysis will be realized to assess the impact of the choice of the kernel function on the feature ranking obtained by KPCA-IG. Moreover, the method will be adapted to other linear algorithms that are solely based on dot-products hence supporting a kernelized version, such as kernel Discriminant Analysis or kernel Partial Least-Squares Discriminant Analysis.



\paragraph{Abbreviations}
KPCA: Kernel principal component analysis; KPCA-IG: Kernel principal component analysis Interpretable Gradient; HCC: Hepatocellular carcinoma; SPEC: Spectral Feature Selection; MCFS: Multi-Cluster Feature Selection; NDFS: Nonnegative Discriminative Feature
Selection; UDFS: Unsupervised Discriminative Feature Selection; CPFS: Convex Principal Feature Selection; lapl: Laplacian score
\paragraph{Funding}
This work was funded by e-MUSE MSCA-ITN-2020 European Training Network under the Marie Sklodowska-Curie grant agreement No 956126.
 The funding did not influence the study's design, the interpretation of data, or the writing of the manuscript.

\paragraph{Availability of data and materials}
Glioma and Carcinoma datasets are freely available at \url{https://github.com/jundongl/scikit-feature/tree/master/skfeature/data}, \\
GPL93 is freely available at \url{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL93}\\
HCC datatset is freely available at \url{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE102079 }

\bibliographystyle{bmc-mathphys} 
%\bibliography{references} 
\nocite{label}%%% Remove comment to use the external .bib file (using bibtex).
%%% and comment out the ``thebibliography'' section.
\begin{thebibliography}{67}
% BibTex style file: bmc-mathphys.bst (version 2.1), 2014-07-24
\ifx \bisbn   \undefined \def \bisbn  #1{ISBN #1}\fi
\ifx \binits  \undefined \def \binits#1{#1}\fi
\ifx \bauthor  \undefined \def \bauthor#1{#1}\fi
\ifx \batitle  \undefined \def \batitle#1{#1}\fi
\ifx \bjtitle  \undefined \def \bjtitle#1{#1}\fi
\ifx \bvolume  \undefined \def \bvolume#1{\textbf{#1}}\fi
\ifx \byear  \undefined \def \byear#1{#1}\fi
\ifx \bissue  \undefined \def \bissue#1{#1}\fi
\ifx \bfpage  \undefined \def \bfpage#1{#1}\fi
\ifx \blpage  \undefined \def \blpage #1{#1}\fi
\ifx \burl  \undefined \def \burl#1{\textsf{#1}}\fi
\ifx \doiurl  \undefined \def \doiurl#1{\textsf{#1}}\fi
\ifx \betal  \undefined \def \betal{\textit{et al.}}\fi
\ifx \binstitute  \undefined \def \binstitute#1{#1}\fi
\ifx \binstitutionaled  \undefined \def \binstitutionaled#1{#1}\fi
\ifx \bctitle  \undefined \def \bctitle#1{#1}\fi
\ifx \beditor  \undefined \def \beditor#1{#1}\fi
\ifx \bpublisher  \undefined \def \bpublisher#1{#1}\fi
\ifx \bbtitle  \undefined \def \bbtitle#1{#1}\fi
\ifx \bedition  \undefined \def \bedition#1{#1}\fi
\ifx \bseriesno  \undefined \def \bseriesno#1{#1}\fi
\ifx \blocation  \undefined \def \blocation#1{#1}\fi
\ifx \bsertitle  \undefined \def \bsertitle#1{#1}\fi
\ifx \bsnm \undefined \def \bsnm#1{#1}\fi
\ifx \bsuffix \undefined \def \bsuffix#1{#1}\fi
\ifx \bparticle \undefined \def \bparticle#1{#1}\fi
\ifx \barticle \undefined \def \barticle#1{#1}\fi
\ifx \bconfdate \undefined \def \bconfdate #1{#1}\fi
\ifx \botherref \undefined \def \botherref #1{#1}\fi
\ifx \url \undefined \def \url#1{\textsf{#1}}\fi
\ifx \bchapter \undefined \def \bchapter#1{#1}\fi
\ifx \bbook \undefined \def \bbook#1{#1}\fi
\ifx \bcomment \undefined \def \bcomment#1{#1}\fi
\ifx \oauthor \undefined \def \oauthor#1{#1}\fi
\ifx \citeauthoryear \undefined \def \citeauthoryear#1{#1}\fi
\ifx \endbibitem  \undefined \def \endbibitem {}\fi
\ifx \bconflocation  \undefined \def \bconflocation#1{#1}\fi
\ifx \arxivurl  \undefined \def \arxivurl#1{\textsf{#1}}\fi
\csname PreBibitemsHook\endcsname

%%% 1
\bibitem[\protect\citeauthoryear{Abid et~al.}{2019}]{abid2019}
\begin{botherref}
\oauthor{\bsnm{Abid}, \binits{A.}},
\oauthor{\bsnm{Balin}, \binits{M.F.}},
\oauthor{\bsnm{Zou}, \binits{J.}}:
Concrete Autoencoders for Differentiable Feature Selection and Reconstruction.
arXiv
(2019).
doi:\doiurl{10.48550/ARXIV.1901.09346}.
\url{https://arxiv.org/abs/1901.09346}
\end{botherref}
\endbibitem

%%% 2
\bibitem[\protect\citeauthoryear{Bach and Jordan}{2003}]{Bach}
\begin{barticle}
\bauthor{\bsnm{Bach}, \binits{F.}},
\bauthor{\bsnm{Jordan}, \binits{M.}}:
\batitle{Kernel independent component analysis}.
\bjtitle{Journal of Machine Learning Research}
\bvolume{3},
\bfpage{1}--\blpage{48}
(\byear{2003}).
doi:\doiurl{10.1162/153244303768966085}
\end{barticle}
\endbibitem

%%% 3
\bibitem[\protect\citeauthoryear{Bond et~al.}{2002}]{Bond2002}
\begin{barticle}
\bauthor{\bsnm{Bond}, \binits{J.}},
\bauthor{\bsnm{Roberts}, \binits{E.}},
\bauthor{\bsnm{Mochida}, \binits{G.H.}},
\bauthor{\bsnm{Hampshire}, \binits{D.J.}},
\bauthor{\bsnm{Scott}, \binits{S.}},
\bauthor{\bsnm{Askham}, \binits{J.M.}},
\bauthor{\bsnm{Springell}, \binits{K.}},
\bauthor{\bsnm{Mahadevan}, \binits{M.}},
\bauthor{\bsnm{Crow}, \binits{Y.J.}},
\bauthor{\bsnm{Markham}, \binits{A.F.}},
\bauthor{\bsnm{Walsh}, \binits{C.A.}},
\bauthor{\bsnm{Woods}, \binits{C.G.}}:
\batitle{{ASPM} is a major determinant of cerebral cortical size}.
\bjtitle{Nature Genetics}
\bvolume{32}(\bissue{2}),
\bfpage{316}--\blpage{320}
(\byear{2002}).
doi:\doiurl{10.1038/ng995}
\end{barticle}
\endbibitem

%%% 4
\bibitem[\protect\citeauthoryear{Brouard et~al.}{2022}]{Brouard2022}
\begin{botherref}
\oauthor{\bsnm{Brouard}, \binits{C.}},
\oauthor{\bsnm{Mariette}, \binits{J.}},
\oauthor{\bsnm{Flamary}, \binits{R.}},
\oauthor{\bsnm{Vialaneix}, \binits{N.}}:
Feature selection for kernel methods in systems biology.
{NAR} Genomics and Bioinformatics
\textbf{4}(1)
(2022).
doi:\doiurl{10.1093/nargab/lqac014}
\end{botherref}
\endbibitem

%%% 5
\bibitem[\protect\citeauthoryear{Cai et~al.}{2010}]{Cai2010UnsupervisedFS}
\begin{botherref}
\oauthor{\bsnm{Cai}, \binits{D.}},
\oauthor{\bsnm{Zhang}, \binits{C.}},
\oauthor{\bsnm{He}, \binits{X.}}:
Unsupervised feature selection for multi-cluster data.
Proceedings of the 16th ACM SIGKDD international conference on Knowledge
  discovery and data mining
(2010)
\end{botherref}
\endbibitem

%%% 6
\bibitem[\protect\citeauthoryear{Chai et~al.}{2018}]{Chai2018}
\begin{barticle}
\bauthor{\bsnm{Chai}, \binits{N.}},
\bauthor{\bsnm{Xie}, \binits{H.-h.}},
\bauthor{\bsnm{Yin}, \binits{J.-p.}},
\bauthor{\bsnm{Sa}, \binits{K.-d.}},
\bauthor{\bsnm{Guo}, \binits{Y.}},
\bauthor{\bsnm{Wang}, \binits{M.}},
\bauthor{\bsnm{Liu}, \binits{J.}},
\bauthor{\bsnm{Zhang}, \binits{X.-f.}},
\bauthor{\bsnm{Zhang}, \binits{X.}},
\bauthor{\bsnm{Yin}, \binits{H.}},
\bauthor{\bsnm{Nie}, \binits{Y.-z.}},
\bauthor{\bsnm{Wu}, \binits{K.-c.}},
\bauthor{\bsnm{Yang}, \binits{A.-g.}},
\bauthor{\bsnm{Zhang}, \binits{R.}}:
\batitle{{FOXM}1 promotes proliferation in human hepatocellular carcinoma cells
  by transcriptional activation of {CCNB}1}.
\bjtitle{Biochemical and Biophysical Research Communications}
\bvolume{500}(\bissue{4}),
\bfpage{924}--\blpage{929}
(\byear{2018}).
doi:\doiurl{10.1016/j.bbrc.2018.04.201}
\end{barticle}
\endbibitem

%%% 7
\bibitem[\protect\citeauthoryear{Chandran et~al.}{2007}]{Chandran2007}
\begin{botherref}
\oauthor{\bsnm{Chandran}, \binits{U.R.}},
\oauthor{\bsnm{Ma}, \binits{C.}},
\oauthor{\bsnm{Dhir}, \binits{R.}},
\oauthor{\bsnm{Bisceglia}, \binits{M.}},
\oauthor{\bsnm{Lyons-Weiler}, \binits{M.}},
\oauthor{\bsnm{Liang}, \binits{W.}},
\oauthor{\bsnm{Michalopoulos}, \binits{G.}},
\oauthor{\bsnm{Becich}, \binits{M.}},
\oauthor{\bsnm{Monzon}, \binits{F.A.}}:
Gene expression profiles of prostate cancer reveal involvement of multiple
  molecular pathways in the metastatic process.
{BMC} Cancer
\textbf{7}(1)
(2007).
doi:\doiurl{10.1186/1471-2407-7-64}
\end{botherref}
\endbibitem

%%% 8
\bibitem[\protect\citeauthoryear{Chen et~al.}{2019}]{Chen2019}
\begin{botherref}
\oauthor{\bsnm{Chen}, \binits{W.}},
\oauthor{\bsnm{Yang}, \binits{L.-g.}},
\oauthor{\bsnm{Xu}, \binits{L.-y.}},
\oauthor{\bsnm{Cheng}, \binits{L.}},
\oauthor{\bsnm{Qian}, \binits{Q.}},
\oauthor{\bsnm{Sun}, \binits{L.}},
\oauthor{\bsnm{Zhu}, \binits{Y.-l.}}:
Bioinformatics analysis revealing prognostic significance of rrm2 gene in
  breast cancer.
Bioscience Reports
\textbf{39}(4)
(2019).
doi:\doiurl{10.1042/bsr20182062}
\end{botherref}
\endbibitem

%%% 9
\bibitem[\protect\citeauthoryear{Chiyonobu et~al.}{2018}]{Chiyonobu2018}
\begin{barticle}
\bauthor{\bsnm{Chiyonobu}, \binits{N.}},
\bauthor{\bsnm{Shimada}, \binits{S.}},
\bauthor{\bsnm{Akiyama}, \binits{Y.}},
\bauthor{\bsnm{Mogushi}, \binits{K.}},
\bauthor{\bsnm{Itoh}, \binits{M.}},
\bauthor{\bsnm{Akahoshi}, \binits{K.}},
\bauthor{\bsnm{Matsumura}, \binits{S.}},
\bauthor{\bsnm{Ogawa}, \binits{K.}},
\bauthor{\bsnm{Ono}, \binits{H.}},
\bauthor{\bsnm{Mitsunori}, \binits{Y.}},
\bauthor{\bsnm{Ban}, \binits{D.}},
\bauthor{\bsnm{Kudo}, \binits{A.}},
\bauthor{\bsnm{Arii}, \binits{S.}},
\bauthor{\bsnm{Suganami}, \binits{T.}},
\bauthor{\bsnm{Yamaoka}, \binits{S.}},
\bauthor{\bsnm{Ogawa}, \binits{Y.}},
\bauthor{\bsnm{Tanabe}, \binits{M.}},
\bauthor{\bsnm{Tanaka}, \binits{S.}}:
\batitle{Fatty acid binding protein 4 ({FABP}4) overexpression in intratumoral
  hepatic stellate cells within hepatocellular carcinoma with metabolic risk
  factors}.
\bjtitle{The American Journal of Pathology}
\bvolume{188}(\bissue{5}),
\bfpage{1213}--\blpage{1224}
(\byear{2018}).
doi:\doiurl{10.1016/j.ajpath.2018.01.012}
\end{barticle}
\endbibitem

%%% 10
\bibitem[\protect\citeauthoryear{Crawford et~al.}{2009}]{Crawford2009}
\begin{barticle}
\bauthor{\bsnm{Crawford}, \binits{H.}},
\bauthor{\bsnm{Dempsey}, \binits{P.}},
\bauthor{\bsnm{Brown}, \binits{G.}},
\bauthor{\bsnm{Adam}, \binits{L.}},
\bauthor{\bsnm{Moss}, \binits{M.}}:
\batitle{{ADAM}10 as a therapeutic target for cancer and inflammation}.
\bjtitle{Current Pharmaceutical Design}
\bvolume{15}(\bissue{20}),
\bfpage{2288}--\blpage{2299}
(\byear{2009}).
doi:\doiurl{10.2174/138161209788682442}
\end{barticle}
\endbibitem

%%% 11
\bibitem[\protect\citeauthoryear{Crone and Crosby}{1995}]{Crone1995}
\begin{barticle}
\bauthor{\bsnm{Crone}, \binits{L.J.}},
\bauthor{\bsnm{Crosby}, \binits{D.S.}}:
\batitle{Statistical applications of a metric on subspaces to satellite
  meteorology}.
\bjtitle{Technometrics}
\bvolume{37}(\bissue{3}),
\bfpage{324}--\blpage{328}
(\byear{1995}).
doi:\doiurl{10.1080/00401706.1995.10484338}
\end{barticle}
\endbibitem

%%% 12
\bibitem[\protect\citeauthoryear{Danon et~al.}{2005}]{NMI}
\begin{botherref}
\oauthor{\bsnm{Danon}, \binits{L.}},
\oauthor{\bsnm{Duch}, \binits{J.}},
\oauthor{\bsnm{Diaz-Guilera}, \binits{A.}},
\oauthor{\bsnm{Arenas}, \binits{A.}}:
Comparing community structure identification
(2005).
doi:\doiurl{10.48550/ARXIV.COND-MAT/0505245}
\end{botherref}
\endbibitem

%%% 13
\bibitem[\protect\citeauthoryear{Ding et~al.}{2014}]{Ding2014}
\begin{barticle}
\bauthor{\bsnm{Ding}, \binits{K.}},
\bauthor{\bsnm{Li}, \binits{W.}},
\bauthor{\bsnm{Zou}, \binits{Z.}},
\bauthor{\bsnm{Zou}, \binits{X.}},
\bauthor{\bsnm{Wang}, \binits{C.}}:
\batitle{{CCNB}1 is a prognostic biomarker for {ER}+ breast cancer}.
\bjtitle{Medical Hypotheses}
\bvolume{83}(\bissue{3}),
\bfpage{359}--\blpage{364}
(\byear{2014}).
doi:\doiurl{10.1016/j.mehy.2014.06.013}
\end{barticle}
\endbibitem

%%% 14
\bibitem[\protect\citeauthoryear{Duda et~al.}{2000}]{Duda2000}
\begin{botherref}
\oauthor{\bsnm{Duda}, \binits{R.O.}},
\oauthor{\bsnm{Hart}, \binits{P.E.}},
\oauthor{\bsnm{Stork}, \binits{D.G.}},
2nd edn.
(2000)
\end{botherref}
\endbibitem

%%% 15
\bibitem[\protect\citeauthoryear{Fang et~al.}{2014}]{Fang2014}
\begin{barticle}
\bauthor{\bsnm{Fang}, \binits{Y.}},
\bauthor{\bsnm{Yu}, \binits{H.}},
\bauthor{\bsnm{Liang}, \binits{X.}},
\bauthor{\bsnm{Xu}, \binits{J.}},
\bauthor{\bsnm{Cai}, \binits{X.}}:
\batitle{Chk1-induced {CCNB}1 overexpression promotes cell proliferation and
  tumor growth in human colorectal cancer}.
\bjtitle{Cancer Biology {\&} Therapy}
\bvolume{15}(\bissue{9}),
\bfpage{1268}--\blpage{1279}
(\byear{2014}).
doi:\doiurl{10.4161/cbt.29691}
\end{barticle}
\endbibitem

%%% 16
\bibitem[\protect\citeauthoryear{Gautier et~al.}{2004}]{affy}
\begin{barticle}
\bauthor{\bsnm{Gautier}, \binits{L.}},
\bauthor{\bsnm{Cope}, \binits{L.}},
\bauthor{\bsnm{Bolstad}, \binits{B.M.}},
\bauthor{\bsnm{Irizarry}, \binits{R.A.}}:
\batitle{{affy—analysis of Affymetrix GeneChip data at the probe level}}.
\bjtitle{Bioinformatics}
\bvolume{20}(\bissue{3}),
\bfpage{307}--\blpage{315}
(\byear{2004}).
doi:\doiurl{10.1093/bioinformatics/btg405}
\end{barticle}
\endbibitem

%%% 17
\bibitem[\protect\citeauthoryear{Gavert et~al.}{2007}]{Gavert2007}
\begin{barticle}
\bauthor{\bsnm{Gavert}, \binits{N.}},
\bauthor{\bsnm{Sheffer}, \binits{M.}},
\bauthor{\bsnm{Raveh}, \binits{S.}},
\bauthor{\bsnm{Spaderna}, \binits{S.}},
\bauthor{\bsnm{Shtutman}, \binits{M.}},
\bauthor{\bsnm{Brabletz}, \binits{T.}},
\bauthor{\bsnm{Barany}, \binits{F.}},
\bauthor{\bsnm{Paty}, \binits{P.}},
\bauthor{\bsnm{Notterman}, \binits{D.}},
\bauthor{\bsnm{Domany}, \binits{E.}},
\bauthor{\bsnm{Ben-Ze{\textquotesingle}ev}, \binits{A.}}:
\batitle{Expression of l1-{CAM} and {ADAM}10 in human colon cancer cells
  induces metastasis}.
\bjtitle{Cancer Research}
\bvolume{67}(\bissue{16}),
\bfpage{7703}--\blpage{7712}
(\byear{2007}).
doi:\doiurl{10.1158/0008-5472.can-07-0991}
\end{barticle}
\endbibitem

%%% 18
\bibitem[\protect\citeauthoryear{Girolami}{2002}]{Girolami2002}
\begin{barticle}
\bauthor{\bsnm{Girolami}, \binits{M.}}:
\batitle{Mercer kernel-based clustering in feature space}.
\bjtitle{{IEEE} Transactions on Neural Networks}
\bvolume{13}(\bissue{3}),
\bfpage{780}--\blpage{784}
(\byear{2002}).
doi:\doiurl{10.1109/tnn.2002.1000150}
\end{barticle}
\endbibitem

%%% 19
\bibitem[\protect\citeauthoryear{Hastie et~al.}{2009}]{Friedman2009}
\begin{bbook}
\bauthor{\bsnm{Hastie}, \binits{T.}},
\bauthor{\bsnm{Tibshirani}, \binits{R.}},
\bauthor{\bsnm{Friedman}, \binits{J.H.}}:
\bbtitle{The Elements of Statistical Learning: Data Mining, Inference, and
  Prediction}.
\bsertitle{Springer series in statistics},
(\byear{2009}).
\burl{https://books.google.co.uk/books?id=eBSgoAEACAAJ}
\end{bbook}
\endbibitem

%%% 20
\bibitem[\protect\citeauthoryear{He et~al.}{2005}]{NIPS2005_b5b03f06}
\begin{bchapter}
\bauthor{\bsnm{He}, \binits{X.}},
\bauthor{\bsnm{Cai}, \binits{D.}},
\bauthor{\bsnm{Niyogi}, \binits{P.}}:
\bctitle{Laplacian score for feature selection}.
In: \bbtitle{Proceedings of the 18th International Conference on Neural
  Information Processing Systems}.
\bsertitle{NIPS'05},
pp. \bfpage{507}--\blpage{514}.
\bpublisher{MIT Press},
\blocation{Cambridge, MA, USA}
(\byear{2005})
\end{bchapter}
\endbibitem

%%% 21
\bibitem[\protect\citeauthoryear{Heo et~al.}{2022}]{Heo2022}
\begin{barticle}
\bauthor{\bsnm{Heo}, \binits{J.}},
\bauthor{\bsnm{Lee}, \binits{J.}},
\bauthor{\bsnm{Nam}, \binits{Y.J.}},
\bauthor{\bsnm{Kim}, \binits{Y.}},
\bauthor{\bsnm{Yun}, \binits{H.}},
\bauthor{\bsnm{Lee}, \binits{S.}},
\bauthor{\bsnm{Ju}, \binits{H.}},
\bauthor{\bsnm{Ryu}, \binits{C.-M.}},
\bauthor{\bsnm{Jeong}, \binits{S.M.}},
\bauthor{\bsnm{Lee}, \binits{J.}},
\bauthor{\bsnm{Lim}, \binits{J.}},
\bauthor{\bsnm{Cho}, \binits{Y.M.}},
\bauthor{\bsnm{Jeong}, \binits{E.M.}},
\bauthor{\bsnm{Hong}, \binits{B.}},
\bauthor{\bsnm{Son}, \binits{J.}},
\bauthor{\bsnm{Shin}, \binits{D.-M.}}:
\batitle{The {CDK}1/{TFCP}2l1/{ID}2 cascade offers a novel combination therapy
  strategy in a preclinical model of bladder cancer}.
\bjtitle{Experimental \& Molecular Medicine}
\bvolume{54}(\bissue{6}),
\bfpage{801}--\blpage{811}
(\byear{2022}).
doi:\doiurl{10.1038/s12276-022-00786-0}
\end{barticle}
\endbibitem

%%% 22
\bibitem[\protect\citeauthoryear{Huang et~al.}{2011}]{5995685}
\begin{bchapter}
\bauthor{\bsnm{Huang}, \binits{D.}},
\bauthor{\bsnm{Tian}, \binits{Y.}},
\bauthor{\bparticle{De~la} \bsnm{Torre}, \binits{F.}}:
\bctitle{Local isomorphism to solve the pre-image problem in kernel methods}.
In: \bbtitle{CVPR 2011},
pp. \bfpage{2761}--\blpage{2768}
(\byear{2011}).
doi:\doiurl{10.1109/CVPR.2011.5995685}
\end{bchapter}
\endbibitem

%%% 23
\bibitem[\protect\citeauthoryear{Imaoka et~al.}{2015}]{Imaoka2015}
\begin{barticle}
\bauthor{\bsnm{Imaoka}, \binits{H.}},
\bauthor{\bsnm{Toiyama}, \binits{Y.}},
\bauthor{\bsnm{Saigusa}, \binits{S.}},
\bauthor{\bsnm{Kawamura}, \binits{M.}},
\bauthor{\bsnm{Kawamoto}, \binits{A.}},
\bauthor{\bsnm{Okugawa}, \binits{Y.}},
\bauthor{\bsnm{Hiro}, \binits{J.}},
\bauthor{\bsnm{Tanaka}, \binits{K.}},
\bauthor{\bsnm{Inoue}, \binits{Y.}},
\bauthor{\bsnm{Mohri}, \binits{Y.}},
\bauthor{\bsnm{Kusunoki}, \binits{M.}}:
\batitle{{RacGAP}1 expression, increasing tumor malignant potential, as a
  predictive biomarker for lymph node metastasis and poor prognosis in
  colorectal cancer}.
\bjtitle{Carcinogenesis}
\bvolume{36}(\bissue{3}),
\bfpage{346}--\blpage{354}
(\byear{2015}).
doi:\doiurl{10.1093/carcin/bgu327}
\end{barticle}
\endbibitem

%%% 24
\bibitem[\protect\citeauthoryear{Jiang et~al.}{2015}]{Jiang2015}
\begin{barticle}
\bauthor{\bsnm{Jiang}, \binits{L.}},
\bauthor{\bsnm{Kwong}, \binits{D.L.-W.}},
\bauthor{\bsnm{Li}, \binits{Y.}},
\bauthor{\bsnm{Liu}, \binits{M.}},
\bauthor{\bsnm{Yuan}, \binits{Y.-F.}},
\bauthor{\bsnm{Li}, \binits{Y.}},
\bauthor{\bsnm{Fu}, \binits{L.}},
\bauthor{\bsnm{Guan}, \binits{X.-Y.}}:
\batitle{{HBP}21, a chaperone of heat shock protein 70, functions as a tumor
  suppressor in hepatocellular carcinoma}.
\bjtitle{Carcinogenesis}
\bvolume{36}(\bissue{10}),
\bfpage{1111}--\blpage{1120}
(\byear{2015}).
doi:\doiurl{10.1093/carcin/bgv116}
\end{barticle}
\endbibitem

%%% 25
\bibitem[\protect\citeauthoryear{Jin et~al.}{2020}]{Jin2020}
\begin{barticle}
\bauthor{\bsnm{Jin}, \binits{C.-Y.}},
\bauthor{\bsnm{Du}, \binits{L.}},
\bauthor{\bsnm{Nuerlan}, \binits{A.-H.}},
\bauthor{\bsnm{Wang}, \binits{X.-L.}},
\bauthor{\bsnm{Yang}, \binits{Y.-W.}},
\bauthor{\bsnm{Guo}, \binits{R.}}:
\batitle{High expression of {RRM}2 as an independent predictive factor of poor
  prognosis in patients with lung adenocarcinoma}.
\bjtitle{Aging}
\bvolume{13}(\bissue{3}),
\bfpage{3518}--\blpage{3535}
(\byear{2020}).
doi:\doiurl{10.18632/aging.202292}
\end{barticle}
\endbibitem

%%% 26
\bibitem[\protect\citeauthoryear{Johnson et~al.}{2020}]{Johnson2020}
\begin{barticle}
\bauthor{\bsnm{Johnson}, \binits{J.E.}},
\bauthor{\bsnm{Laparra}, \binits{V.}},
\bauthor{\bsnm{P{\'{e}}rez-Suay}, \binits{A.}},
\bauthor{\bsnm{Mahecha}, \binits{M.D.}},
\bauthor{\bsnm{Camps-Valls}, \binits{G.}}:
\batitle{Kernel methods and their derivatives: Concept and perspectives for the
  earth system sciences}.
\bjtitle{{PLOS} {ONE}}
\bvolume{15}(\bissue{10}),
\bfpage{0235885}
(\byear{2020}).
doi:\doiurl{10.1371/journal.pone.0235885}
\end{barticle}
\endbibitem

%%% 27
\bibitem[\protect\citeauthoryear{Kouprina et~al.}{2005}]{Kouprina2005}
\begin{barticle}
\bauthor{\bsnm{Kouprina}, \binits{N.}},
\bauthor{\bsnm{Pavlicek}, \binits{A.}},
\bauthor{\bsnm{Collins}, \binits{N.K.}},
\bauthor{\bsnm{Nakano}, \binits{M.}},
\bauthor{\bsnm{Noskov}, \binits{V.N.}},
\bauthor{\bsnm{Ohzeki}, \binits{J.-I.}},
\bauthor{\bsnm{Mochida}, \binits{G.H.}},
\bauthor{\bsnm{Risinger}, \binits{J.I.}},
\bauthor{\bsnm{Goldsmith}, \binits{P.}},
\bauthor{\bsnm{Gunsior}, \binits{M.}},
\bauthor{\bsnm{Solomon}, \binits{G.}},
\bauthor{\bsnm{Gersch}, \binits{W.}},
\bauthor{\bsnm{Kim}, \binits{J.-H.}},
\bauthor{\bsnm{Barrett}, \binits{J.C.}},
\bauthor{\bsnm{Walsh}, \binits{C.A.}},
\bauthor{\bsnm{Jurka}, \binits{J.}},
\bauthor{\bsnm{Masumoto}, \binits{H.}},
\bauthor{\bsnm{Larionov}, \binits{V.}}:
\batitle{The microcephaly {ASPM} gene is expressed in proliferating tissues and
  encodes for a mitotic spindle protein}.
\bjtitle{Human Molecular Genetics}
\bvolume{14}(\bissue{15}),
\bfpage{2155}--\blpage{2165}
(\byear{2005}).
doi:\doiurl{10.1093/hmg/ddi220}
\end{barticle}
\endbibitem

%%% 28
\bibitem[\protect\citeauthoryear{Krzystanek et~al.}{2016}]{Krzystanek2016}
\begin{botherref}
\oauthor{\bsnm{Krzystanek}, \binits{M.}},
\oauthor{\bsnm{Moldvay}, \binits{J.}},
\oauthor{\bsnm{Sz\"{u}ts}, \binits{D.}},
\oauthor{\bsnm{Szallasi}, \binits{Z.}},
\oauthor{\bsnm{Eklund}, \binits{A.C.}}:
A robust prognostic gene expression signature for early stage lung
  adenocarcinoma.
Biomarker Research
\textbf{4}(1)
(2016).
doi:\doiurl{10.1186/s40364-016-0058-3}
\end{botherref}
\endbibitem

%%% 29
\bibitem[\protect\citeauthoryear{Kwok and Tsang}{2004}]{Kwok2004}
\begin{barticle}
\bauthor{\bsnm{Kwok}, \binits{J.T.-Y.}},
\bauthor{\bsnm{Tsang}, \binits{I.W.-H.}}:
\batitle{The pre-image problem in kernel methods}.
\bjtitle{{IEEE} Transactions on Neural Networks}
\bvolume{15}(\bissue{6}),
\bfpage{1517}--\blpage{1525}
(\byear{2004}).
doi:\doiurl{10.1109/tnn.2004.837781}
\end{barticle}
\endbibitem

%%% 30
\bibitem[\protect\citeauthoryear{Lee et~al.}{2010}]{Lee2010}
\begin{barticle}
\bauthor{\bsnm{Lee}, \binits{S.B.}},
\bauthor{\bsnm{Schramme}, \binits{A.}},
\bauthor{\bsnm{Doberstein}, \binits{K.}},
\bauthor{\bsnm{Dummer}, \binits{R.}},
\bauthor{\bsnm{Abdel-Bakky}, \binits{M.S.}},
\bauthor{\bsnm{Keller}, \binits{S.}},
\bauthor{\bsnm{Altevogt}, \binits{P.}},
\bauthor{\bsnm{Oh}, \binits{S.T.}},
\bauthor{\bsnm{Reichrath}, \binits{J.}},
\bauthor{\bsnm{Oxmann}, \binits{D.}},
\bauthor{\bsnm{Pfeilschifter}, \binits{J.}},
\bauthor{\bsnm{Mihic-Probst}, \binits{D.}},
\bauthor{\bsnm{Gutwein}, \binits{P.}}:
\batitle{{ADAM}10 is upregulated in melanoma metastasis compared with primary
  melanoma}.
\bjtitle{Journal of Investigative Dermatology}
\bvolume{130}(\bissue{3}),
\bfpage{763}--\blpage{773}
(\byear{2010}).
doi:\doiurl{10.1038/jid.2009.335}
\end{barticle}
\endbibitem

%%% 31
\bibitem[\protect\citeauthoryear{Li et~al.}{2020}]{JLi2020}
\begin{botherref}
\oauthor{\bsnm{Li}, \binits{J.}},
\oauthor{\bsnm{Wang}, \binits{Y.}},
\oauthor{\bsnm{Wang}, \binits{X.}},
\oauthor{\bsnm{Yang}, \binits{Q.}}:
{CDK}1 and {CDC}20 overexpression in patients with colorectal cancer are
  associated with poor prognosis: evidence from integrated bioinformatics
  analysis.
World Journal of Surgical Oncology
\textbf{18}(1)
(2020).
doi:\doiurl{10.1186/s12957-020-01817-8}
\end{botherref}
\endbibitem

%%% 32
\bibitem[\protect\citeauthoryear{Li et~al.}{2017}]{Li2017}
\begin{barticle}
\bauthor{\bsnm{Li}, \binits{J.}},
\bauthor{\bsnm{Cheng}, \binits{K.}},
\bauthor{\bsnm{Wang}, \binits{S.}},
\bauthor{\bsnm{Morstatter}, \binits{F.}},
\bauthor{\bsnm{Trevino}, \binits{R.P.}},
\bauthor{\bsnm{Tang}, \binits{J.}},
\bauthor{\bsnm{Liu}, \binits{H.}}:
\batitle{Feature selection feature selection: a data perspective}.
\bjtitle{{ACM} Computing Surveys}
\bvolume{50}(\bissue{6}),
\bfpage{1}--\blpage{45}
(\byear{2017}).
doi:\doiurl{10.1145/3136625}
\end{barticle}
\endbibitem

%%% 33
\bibitem[\protect\citeauthoryear{Li et~al.}{2020}]{Li2020}
\begin{barticle}
\bauthor{\bsnm{Li}, \binits{M.}},
\bauthor{\bsnm{He}, \binits{F.}},
\bauthor{\bsnm{Zhang}, \binits{Z.}},
\bauthor{\bsnm{Xiang}, \binits{Z.}},
\bauthor{\bsnm{Hu}, \binits{D.}}:
\batitle{{CDK}1 serves as a potential prognostic biomarker and target for lung
  cancer}.
\bjtitle{Journal of International Medical Research}
\bvolume{48}(\bissue{2}),
\bfpage{030006051989750}
(\byear{2020}).
doi:\doiurl{10.1177/0300060519897508}
\end{barticle}
\endbibitem

%%% 34
\bibitem[\protect\citeauthoryear{Li et~al.}{2021}]{Li2021}
\begin{barticle}
\bauthor{\bsnm{Li}, \binits{Z.}},
\bauthor{\bsnm{Yang}, \binits{Y.}},
\bauthor{\bsnm{Liu}, \binits{J.}},
\bauthor{\bsnm{Zhou}, \binits{X.}},
\bauthor{\bsnm{Lu}, \binits{H.}}:
\batitle{Unsupervised feature selection using nonnegative spectral analysis}.
\bjtitle{Proceedings of the {AAAI} Conference on Artificial Intelligence}
\bvolume{26}(\bissue{1}),
\bfpage{1026}--\blpage{1032}
(\byear{2021}).
doi:\doiurl{10.1609/aaai.v26i1.8289}
\end{barticle}
\endbibitem

%%% 35
\bibitem[\protect\citeauthoryear{Liu et~al.}{2022}]{Liu2022}
\begin{barticle}
\bauthor{\bsnm{Liu}, \binits{X.}},
\bauthor{\bsnm{Wu}, \binits{H.}},
\bauthor{\bsnm{Liu}, \binits{Z.}}:
\batitle{An integrative human pan-cancer analysis of cyclin-dependent kinase 1
  ({CDK}1)}.
\bjtitle{Cancers}
\bvolume{14}(\bissue{11}),
\bfpage{2658}
(\byear{2022}).
doi:\doiurl{10.3390/cancers14112658}
\end{barticle}
\endbibitem

%%% 36
\bibitem[\protect\citeauthoryear{Llovet et~al.}{2021}]{Llovet2021}
\begin{botherref}
\oauthor{\bsnm{Llovet}, \binits{J.M.}},
\oauthor{\bsnm{Kelley}, \binits{R.K.}},
\oauthor{\bsnm{Villanueva}, \binits{A.}},
\oauthor{\bsnm{Singal}, \binits{A.G.}},
\oauthor{\bsnm{Pikarsky}, \binits{E.}},
\oauthor{\bsnm{Roayaie}, \binits{S.}},
\oauthor{\bsnm{Lencioni}, \binits{R.}},
\oauthor{\bsnm{Koike}, \binits{K.}},
\oauthor{\bsnm{Zucman-Rossi}, \binits{J.}},
\oauthor{\bsnm{Finn}, \binits{R.S.}}:
Hepatocellular carcinoma.
Nature Reviews Disease Primers
\textbf{7}(1)
(2021).
doi:\doiurl{10.1038/s41572-020-00240-3}
\end{botherref}
\endbibitem

%%% 37
\bibitem[\protect\citeauthoryear{Mariette and
  Villa-Vialaneix}{2017}]{Mariette2017}
\begin{barticle}
\bauthor{\bsnm{Mariette}, \binits{J.}},
\bauthor{\bsnm{Villa-Vialaneix}, \binits{N.}}:
\batitle{Unsupervised multiple kernel learning for heterogeneous data
  integration}.
\bjtitle{Bioinformatics}
\bvolume{34}(\bissue{6}),
\bfpage{1009}--\blpage{1015}
(\byear{2017}).
doi:\doiurl{10.1093/bioinformatics/btx682}
\end{barticle}
\endbibitem

%%% 38
\bibitem[\protect\citeauthoryear{Masaeli et~al.}{2010}]{Masaeli2010ConvexPF}
\begin{bchapter}
\bauthor{\bsnm{Masaeli}, \binits{M.}},
\bauthor{\bsnm{Yan}, \binits{Y.}},
\bauthor{\bsnm{Cui}, \binits{Y.}},
\bauthor{\bsnm{Fung}, \binits{G.}},
\bauthor{\bsnm{Dy}, \binits{J.G.}}:
\bctitle{Convex principal feature selection}.
In: \bbtitle{SDM}
(\byear{2010})
\end{bchapter}
\endbibitem

%%% 39
\bibitem[\protect\citeauthoryear{Mi et~al.}{2016}]{Mi2016}
\begin{barticle}
\bauthor{\bsnm{Mi}, \binits{S.}},
\bauthor{\bsnm{Lin}, \binits{M.}},
\bauthor{\bsnm{Brouwer-Visser}, \binits{J.}},
\bauthor{\bsnm{Heim}, \binits{J.}},
\bauthor{\bsnm{Smotkin}, \binits{D.}},
\bauthor{\bsnm{Hebert}, \binits{T.}},
\bauthor{\bsnm{Gunter}, \binits{M.J.}},
\bauthor{\bsnm{Goldberg}, \binits{G.L.}},
\bauthor{\bsnm{Zheng}, \binits{D.}},
\bauthor{\bsnm{Huang}, \binits{G.S.}}:
\batitle{{RNA}-seq identification of {RACGAP}1 as a metastatic driver in
  uterine carcinosarcoma}.
\bjtitle{Clinical Cancer Research}
\bvolume{22}(\bissue{18}),
\bfpage{4676}--\blpage{4686}
(\byear{2016}).
doi:\doiurl{10.1158/1078-0432.ccr-15-2116}
\end{barticle}
\endbibitem

%%% 40
\bibitem[\protect\citeauthoryear{Mika et~al.}{1998}]{Mika1998KernelPA}
\begin{bchapter}
\bauthor{\bsnm{Mika}, \binits{S.}},
\bauthor{\bsnm{Sch{\"o}lkopf}, \binits{B.}},
\bauthor{\bsnm{Smola}, \binits{A.}},
\bauthor{\bsnm{M{\"u}ller}, \binits{K.-R.}},
\bauthor{\bsnm{Scholz}, \binits{M.}},
\bauthor{\bsnm{R{\"a}tsch}, \binits{G.}}:
\bctitle{Kernel pca and de-noising in feature spaces}.
In: \bbtitle{NIPS}
(\byear{1998})
\end{bchapter}
\endbibitem

%%% 41
\bibitem[\protect\citeauthoryear{Morse et~al.}{2019}]{Morse2019}
\begin{barticle}
\bauthor{\bsnm{Morse}, \binits{M.A.}},
\bauthor{\bsnm{Sun}, \binits{W.}},
\bauthor{\bsnm{Kim}, \binits{R.}},
\bauthor{\bsnm{He}, \binits{A.R.}},
\bauthor{\bsnm{Abada}, \binits{P.B.}},
\bauthor{\bsnm{Mynderse}, \binits{M.}},
\bauthor{\bsnm{Finn}, \binits{R.S.}}:
\batitle{The role of angiogenesis in hepatocellular carcinoma}.
\bjtitle{Clinical Cancer Research}
\bvolume{25}(\bissue{3}),
\bfpage{912}--\blpage{920}
(\byear{2019}).
doi:\doiurl{10.1158/1078-0432.ccr-18-1254}
\end{barticle}
\endbibitem

%%% 42
\bibitem[\protect\citeauthoryear{Moss et~al.}{2008}]{Moss2008}
\begin{barticle}
\bauthor{\bsnm{Moss}, \binits{M.}},
\bauthor{\bsnm{Stoeck}, \binits{A.}},
\bauthor{\bsnm{Yan}, \binits{W.}},
\bauthor{\bsnm{Dempsey}, \binits{P.}}:
\batitle{{ADAM}10 as a target for anti-cancer therapy}.
\bjtitle{Current Pharmaceutical Biotechnology}
\bvolume{9}(\bissue{1}),
\bfpage{2}--\blpage{8}
(\byear{2008}).
doi:\doiurl{10.2174/138920108783497613}
\end{barticle}
\endbibitem

%%% 43
\bibitem[\protect\citeauthoryear{Ohmura et~al.}{2021}]{Ohmura2021}
\begin{botherref}
\oauthor{\bsnm{Ohmura}, \binits{S.}},
\oauthor{\bsnm{Marchetto}, \binits{A.}},
\oauthor{\bsnm{Orth}, \binits{M.F.}},
\oauthor{\bsnm{Li}, \binits{J.}},
\oauthor{\bsnm{Jabar}, \binits{S.}},
\oauthor{\bsnm{Ranft}, \binits{A.}},
\oauthor{\bsnm{Vinca}, \binits{E.}},
\oauthor{\bsnm{Ceranski}, \binits{K.}},
\oauthor{\bsnm{Carre{\~{n}}o-Gonzalez}, \binits{M.J.}},
\oauthor{\bsnm{Romero-P{\'{e}}rez}, \binits{L.}},
\oauthor{\bsnm{Wehweck}, \binits{F.S.}},
\oauthor{\bsnm{Musa}, \binits{J.}},
\oauthor{\bsnm{Bestvater}, \binits{F.}},
\oauthor{\bsnm{Knott}, \binits{M.M.L.}},
\oauthor{\bsnm{H\"{o}lting}, \binits{T.L.B.}},
\oauthor{\bsnm{Hartmann}, \binits{W.}},
\oauthor{\bsnm{Dirksen}, \binits{U.}},
\oauthor{\bsnm{Kirchner}, \binits{T.}},
\oauthor{\bsnm{Cidre-Aranaz}, \binits{F.}},
\oauthor{\bsnm{Gr\"{u}newald}, \binits{T.G.P.}}:
Translational evidence for {RRM}2 as a prognostic biomarker and therapeutic
  target in ewing sarcoma.
Molecular Cancer
\textbf{20}(1)
(2021).
doi:\doiurl{10.1186/s12943-021-01393-9}
\end{botherref}
\endbibitem

%%% 44
\bibitem[\protect\citeauthoryear{Osako et~al.}{2019}]{Osako2019}
\begin{barticle}
\bauthor{\bsnm{Osako}, \binits{Y.}},
\bauthor{\bsnm{Yoshino}, \binits{H.}},
\bauthor{\bsnm{Sakaguchi}, \binits{T.}},
\bauthor{\bsnm{Sugita}, \binits{S.}},
\bauthor{\bsnm{Yonemori}, \binits{M.}},
\bauthor{\bsnm{Nakagawa}, \binits{M.}},
\bauthor{\bsnm{Enokida}, \binits{H.}}:
\batitle{Potential tumor‑suppressive role of {microRNA}‑99a‑3p in
  sunitinib‑resistant renal cell carcinoma cells through the regulation of
  {RRM}2}.
\bjtitle{International Journal of Oncology}
(\byear{2019}).
doi:\doiurl{10.3892/ijo.2019.4736}
\end{barticle}
\endbibitem

%%% 45
\bibitem[\protect\citeauthoryear{Pliarchopoulou
  et~al.}{2012}]{Pliarchopoulou2012}
\begin{barticle}
\bauthor{\bsnm{Pliarchopoulou}, \binits{K.}},
\bauthor{\bsnm{Kalogeras}, \binits{K.T.}},
\bauthor{\bsnm{Kronenwett}, \binits{R.}},
\bauthor{\bsnm{Wirtz}, \binits{R.M.}},
\bauthor{\bsnm{Eleftheraki}, \binits{A.G.}},
\bauthor{\bsnm{Batistatou}, \binits{A.}},
\bauthor{\bsnm{Bobos}, \binits{M.}},
\bauthor{\bsnm{Soupos}, \binits{N.}},
\bauthor{\bsnm{Polychronidou}, \binits{G.}},
\bauthor{\bsnm{Gogas}, \binits{H.}},
\bauthor{\bsnm{Samantas}, \binits{E.}},
\bauthor{\bsnm{Christodoulou}, \binits{C.}},
\bauthor{\bsnm{Makatsoris}, \binits{T.}},
\bauthor{\bsnm{Pavlidis}, \binits{N.}},
\bauthor{\bsnm{Pectasides}, \binits{D.}},
\bauthor{\bsnm{Fountzilas}, \binits{G.}}:
\batitle{Prognostic significance of {RACGAP}1 {mRNA} expression in high-risk
  early breast cancer: a study in primary tumors of breast cancer patients
  participating in a randomized hellenic cooperative oncology group trial}.
\bjtitle{Cancer Chemotherapy and Pharmacology}
\bvolume{71}(\bissue{1}),
\bfpage{245}--\blpage{255}
(\byear{2012}).
doi:\doiurl{10.1007/s00280-012-2002-z}
\end{barticle}
\endbibitem

%%% 46
\bibitem[\protect\citeauthoryear{Rahman et~al.}{2012}]{Rahman2012}
\begin{barticle}
\bauthor{\bsnm{Rahman}, \binits{M.A.}},
\bauthor{\bsnm{Amin}, \binits{A.R.M.R.}},
\bauthor{\bsnm{Wang}, \binits{X.}},
\bauthor{\bsnm{Zuckerman}, \binits{J.E.}},
\bauthor{\bsnm{Choi}, \binits{C.H.J.}},
\bauthor{\bsnm{Zhou}, \binits{B.}},
\bauthor{\bsnm{Wang}, \binits{D.}},
\bauthor{\bsnm{Nannapaneni}, \binits{S.}},
\bauthor{\bsnm{Koenig}, \binits{L.}},
\bauthor{\bsnm{Chen}, \binits{Z.}},
\bauthor{\bsnm{Chen}, \binits{Z.G.}},
\bauthor{\bsnm{Yen}, \binits{Y.}},
\bauthor{\bsnm{Davis}, \binits{M.E.}},
\bauthor{\bsnm{Shin}, \binits{D.M.}}:
\batitle{Systemic delivery of {siRNA} nanoparticles targeting {RRM}2 suppresses
  head and neck tumor growth}.
\bjtitle{Journal of Controlled Release}
\bvolume{159}(\bissue{3}),
\bfpage{384}--\blpage{392}
(\byear{2012}).
doi:\doiurl{10.1016/j.jconrel.2012.01.045}
\end{barticle}
\endbibitem

%%% 47
\bibitem[\protect\citeauthoryear{Rasmussen et~al.}{2011}]{Rasmussen2011}
\begin{barticle}
\bauthor{\bsnm{Rasmussen}, \binits{P.M.}},
\bauthor{\bsnm{Madsen}, \binits{K.H.}},
\bauthor{\bsnm{Lund}, \binits{T.E.}},
\bauthor{\bsnm{Hansen}, \binits{L.K.}}:
\batitle{Visualization of nonlinear kernel models in neuroimaging by
  sensitivity maps}.
\bjtitle{{NeuroImage}}
\bvolume{55}(\bissue{3}),
\bfpage{1120}--\blpage{1131}
(\byear{2011}).
doi:\doiurl{10.1016/j.neuroimage.2010.12.035}
\end{barticle}
\endbibitem

%%% 48
\bibitem[\protect\citeauthoryear{Reverter et~al.}{2014}]{Reverter2014}
\begin{botherref}
\oauthor{\bsnm{Reverter}, \binits{F.}},
\oauthor{\bsnm{Vegas}, \binits{E.}},
\oauthor{\bsnm{Oller}, \binits{J.M.}}:
Kernel-{PCA} data integration with enhanced interpretability.
{BMC} Systems Biology
\textbf{8}(S2)
(2014).
doi:\doiurl{10.1186/1752-0509-8-s2-s6}
\end{botherref}
\endbibitem

%%% 49
\bibitem[\protect\citeauthoryear{Roth and Steinhage}{1999}]{RothS99}
\begin{bchapter}
\bauthor{\bsnm{Roth}, \binits{V.}},
\bauthor{\bsnm{Steinhage}, \binits{V.}}:
\bctitle{Nonlinear discriminant analysis using kernel functions}.
In: \bbtitle{NIPS},
pp. \bfpage{568}--\blpage{574}
(\byear{1999}).
\burl{http://papers.nips.cc/paper/1736-nonlinear-discriminant-analysis-using-kernel-functions}
\end{bchapter}
\endbibitem

%%% 50
\bibitem[\protect\citeauthoryear{Ruder}{2016}]{Ruder2016}
\begin{botherref}
\oauthor{\bsnm{Ruder}, \binits{S.}}:
An overview of gradient descent optimization algorithms.
arXiv
(2016).
doi:\doiurl{10.48550/ARXIV.1609.04747}.
\url{https://arxiv.org/abs/1609.04747}
\end{botherref}
\endbibitem

%%% 51
\bibitem[\protect\citeauthoryear{Saigusa et~al.}{2014}]{Saigusa2014}
\begin{barticle}
\bauthor{\bsnm{Saigusa}, \binits{S.}},
\bauthor{\bsnm{Tanaka}, \binits{K.}},
\bauthor{\bsnm{Mohri}, \binits{Y.}},
\bauthor{\bsnm{Ohi}, \binits{M.}},
\bauthor{\bsnm{Shimura}, \binits{T.}},
\bauthor{\bsnm{Kitajima}, \binits{T.}},
\bauthor{\bsnm{Kondo}, \binits{S.}},
\bauthor{\bsnm{Okugawa}, \binits{Y.}},
\bauthor{\bsnm{Toiyama}, \binits{Y.}},
\bauthor{\bsnm{Inoue}, \binits{Y.}},
\bauthor{\bsnm{Kusunoki}, \binits{M.}}:
\batitle{Clinical significance of {RacGAP}1 expression at the invasive front of
  gastric cancer}.
\bjtitle{Gastric Cancer}
\bvolume{18}(\bissue{1}),
\bfpage{84}--\blpage{92}
(\byear{2014}).
doi:\doiurl{10.1007/s10120-014-0355-1}
\end{barticle}
\endbibitem

%%% 52
\bibitem[\protect\citeauthoryear{Santin et~al.}{2005}]{Santin2005}
\begin{barticle}
\bauthor{\bsnm{Santin}, \binits{A.D.}},
\bauthor{\bsnm{Zhan}, \binits{F.}},
\bauthor{\bsnm{Bignotti}, \binits{E.}},
\bauthor{\bsnm{Siegel}, \binits{E.R.}},
\bauthor{\bsnm{Can{\'{e}}}, \binits{S.}},
\bauthor{\bsnm{Bellone}, \binits{S.}},
\bauthor{\bsnm{Palmieri}, \binits{M.}},
\bauthor{\bsnm{Anfossi}, \binits{S.}},
\bauthor{\bsnm{Thomas}, \binits{M.}},
\bauthor{\bsnm{Burnett}, \binits{A.}},
\bauthor{\bsnm{Kay}, \binits{H.H.}},
\bauthor{\bsnm{Roman}, \binits{J.J.}},
\bauthor{\bsnm{O{\textquotesingle}Brien}, \binits{T.J.}},
\bauthor{\bsnm{Tian}, \binits{E.}},
\bauthor{\bsnm{Cannon}, \binits{M.J.}},
\bauthor{\bsnm{Shaughnessy}, \binits{J.}},
\bauthor{\bsnm{Pecorelli}, \binits{S.}}:
\batitle{Gene expression profiles of primary {HPV}16- and {HPV}18-infected
  early stage cervical cancers and normal cervical epithelium: identification
  of novel candidate molecular markers for cervical cancer diagnosis and
  therapy}.
\bjtitle{Virology}
\bvolume{331}(\bissue{2}),
\bfpage{269}--\blpage{291}
(\byear{2005}).
doi:\doiurl{10.1016/j.virol.2004.09.045}
\end{barticle}
\endbibitem

%%% 53
\bibitem[\protect\citeauthoryear{Sanz et~al.}{2018}]{Sanz2018}
\begin{botherref}
\oauthor{\bsnm{Sanz}, \binits{H.}},
\oauthor{\bsnm{Valim}, \binits{C.}},
\oauthor{\bsnm{Vegas}, \binits{E.}},
\oauthor{\bsnm{Oller}, \binits{J.M.}},
\oauthor{\bsnm{Reverter}, \binits{F.}}:
{SVM}-{RFE}: selection and visualization of the most relevant features through
  non-linear kernels.
{BMC} Bioinformatics
\textbf{19}(1)
(2018).
doi:\doiurl{10.1186/s12859-018-2451-4}
\end{botherref}
\endbibitem

%%% 54
\bibitem[\protect\citeauthoryear{Sch{\"o}lkopf et~al.}{1998}]{Scholkopf98}
\begin{bchapter}
\bauthor{\bsnm{Sch{\"o}lkopf}, \binits{B.}},
\bauthor{\bsnm{Knirsch}, \binits{P.}},
\bauthor{\bsnm{Smola}, \binits{A.}},
\bauthor{\bsnm{Burges}, \binits{C.}}:
\bctitle{Fast approximation of support vector kernel expansions, and an
  interpretation of clustering as approximation in feature spaces.}
In: \bbtitle{Mustererkennung 1998}.
\bsertitle{Informatik aktuell},
pp. \bfpage{125}--\blpage{132}.
\bpublisher{Springer},
\blocation{Berlin, Germany}
(\byear{1998}).
\bcomment{Max-Planck-Gesellschaft}
\end{bchapter}
\endbibitem

%%% 55
\bibitem[\protect\citeauthoryear{Sch{\"o}lkopf et~al.}{1997}]{Schlkopf1997}
\begin{bchapter}
\bauthor{\bsnm{Sch{\"o}lkopf}, \binits{B.}},
\bauthor{\bsnm{Smola}, \binits{A.}},
\bauthor{\bsnm{M{\"u}ller}, \binits{K.-R.}}:
\bctitle{Kernel principal component analysis}.
In: \bbtitle{Artificial Neural Networks --- ICANN'97},
pp. \bfpage{583}--\blpage{588}.
\bpublisher{Springer},
\blocation{Berlin, Heidelberg}
(\byear{1997})
\end{bchapter}
\endbibitem

%%% 56
\bibitem[\protect\citeauthoryear{Sch{\"o}lkopf et~al.}{2003}]{Schlkopf2003}
\begin{bchapter}
\bauthor{\bsnm{Sch{\"o}lkopf}, \binits{B.}},
\bauthor{\bsnm{Tsuda}, \binits{K.}},
\bauthor{\bsnm{Vert}, \binits{J.-P.}}:
\bctitle{Kernel methods in computational biology}.
(\byear{2003})
\end{bchapter}
\endbibitem

%%% 57
\bibitem[\protect\citeauthoryear{Sofi et~al.}{2022}]{Sofi2022}
\begin{botherref}
\oauthor{\bsnm{Sofi}, \binits{S.}},
\oauthor{\bsnm{Mehraj}, \binits{U.}},
\oauthor{\bsnm{Qayoom}, \binits{H.}},
\oauthor{\bsnm{Aisha}, \binits{S.}},
\oauthor{\bsnm{Almilaibary}, \binits{A.}},
\oauthor{\bsnm{Alkhanani}, \binits{M.}},
\oauthor{\bsnm{Mir}, \binits{M.A.}}:
Targeting cyclin-dependent kinase 1 ({CDK}1) in cancer: molecular docking and
  dynamic simulations of potential {CDK}1 inhibitors.
Medical Oncology
\textbf{39}(9)
(2022).
doi:\doiurl{10.1007/s12032-022-01748-2}
\end{botherref}
\endbibitem

%%% 58
\bibitem[\protect\citeauthoryear{Wang et~al.}{2014}]{Wang2014_2}
\begin{barticle}
\bauthor{\bsnm{Wang}, \binits{N.}},
\bauthor{\bsnm{Zhan}, \binits{T.}},
\bauthor{\bsnm{Ke}, \binits{T.}},
\bauthor{\bsnm{Huang}, \binits{X.}},
\bauthor{\bsnm{Ke}, \binits{D.}},
\bauthor{\bsnm{Wang}, \binits{Q.}},
\bauthor{\bsnm{Li}, \binits{H.}}:
\batitle{Increased expression of {RRM}2 by human papillomavirus e7 oncoprotein
  promotes angiogenesis in cervical cancer}.
\bjtitle{British Journal of Cancer}
\bvolume{110}(\bissue{4}),
\bfpage{1034}--\blpage{1044}
(\byear{2014}).
doi:\doiurl{10.1038/bjc.2013.817}
\end{barticle}
\endbibitem

%%% 59
\bibitem[\protect\citeauthoryear{Wang et~al.}{2013}]{Wang2013}
\begin{barticle}
\bauthor{\bsnm{Wang}, \binits{W.}},
\bauthor{\bsnm{Hsu}, \binits{C.}},
\bauthor{\bsnm{Wang}, \binits{T.}},
\bauthor{\bsnm{Li}, \binits{C.}},
\bauthor{\bsnm{Hou}, \binits{Y.}},
\bauthor{\bsnm{Chu}, \binits{J.}},
\bauthor{\bsnm{Lee}, \binits{C.}},
\bauthor{\bsnm{Liu}, \binits{M.}},
\bauthor{\bsnm{Su}, \binits{J.J.{\textendash}.}},
\bauthor{\bsnm{Jian}, \binits{K.}},
\bauthor{\bsnm{Huang}, \binits{S.}},
\bauthor{\bsnm{Jiang}, \binits{S.}},
\bauthor{\bsnm{Shan}, \binits{Y.}},
\bauthor{\bsnm{Lin}, \binits{P.}},
\bauthor{\bsnm{Shen}, \binits{Y.}},
\bauthor{\bsnm{Lee}, \binits{M.T.{\textendash}.}},
\bauthor{\bsnm{Chan}, \binits{T.}},
\bauthor{\bsnm{Chang}, \binits{C.}},
\bauthor{\bsnm{Chen}, \binits{C.}},
\bauthor{\bsnm{Chang}, \binits{I.}},
\bauthor{\bsnm{Lee}, \binits{Y.}},
\bauthor{\bsnm{Chen}, \binits{L.}},
\bauthor{\bsnm{Tsai}, \binits{K.K.}}:
\batitle{A gene expression signature of epithelial tubulogenesis and a role for
  {ASPM} in pancreatic tumor progression}.
\bjtitle{Gastroenterology}
\bvolume{145}(\bissue{5}),
\bfpage{1110}--\blpage{1120}
(\byear{2013}).
doi:\doiurl{10.1053/j.gastro.2013.07.040}
\end{barticle}
\endbibitem

%%% 60
\bibitem[\protect\citeauthoryear{West et~al.}{2007}]{West2007}
\begin{barticle}
\bauthor{\bsnm{West}, \binits{A.N.}},
\bauthor{\bsnm{Neale}, \binits{G.A.}},
\bauthor{\bsnm{Pounds}, \binits{S.}},
\bauthor{\bsnm{Figueredo}, \binits{B.C.}},
\bauthor{\bsnm{Galindo}, \binits{C.R.}},
\bauthor{\bsnm{Pianovski}, \binits{M.A.D.}},
\bauthor{\bsnm{Filho}, \binits{A.G.O.}},
\bauthor{\bsnm{Malkin}, \binits{D.}},
\bauthor{\bsnm{Lalli}, \binits{E.}},
\bauthor{\bsnm{Ribeiro}, \binits{R.}},
\bauthor{\bsnm{Zambetti}, \binits{G.P.}}:
\batitle{Gene expression profiling of childhood adrenocortical tumors}.
\bjtitle{Cancer Research}
\bvolume{67}(\bissue{2}),
\bfpage{600}--\blpage{608}
(\byear{2007}).
doi:\doiurl{10.1158/0008-5472.can-06-3767}
\end{barticle}
\endbibitem

%%% 61
\bibitem[\protect\citeauthoryear{Xu et~al.}{2018}]{Xu2018}
\begin{barticle}
\bauthor{\bsnm{Xu}, \binits{Z.}},
\bauthor{\bsnm{Zhang}, \binits{Q.}},
\bauthor{\bsnm{Luh}, \binits{F.}},
\bauthor{\bsnm{Jin}, \binits{B.}},
\bauthor{\bsnm{Liu}, \binits{X.}}:
\batitle{Overexpression of the {ASPM} gene is associated with aggressiveness
  and poor outcome in bladder cancer}.
\bjtitle{Oncology Letters}
(\byear{2018}).
doi:\doiurl{10.3892/ol.2018.9762}
\end{barticle}
\endbibitem

%%% 62
\bibitem[\protect\citeauthoryear{Yang et~al.}{2005}]{Yang2005}
\begin{barticle}
\bauthor{\bsnm{Yang}, \binits{X.J.}},
\bauthor{\bsnm{Tan}, \binits{M.-H.}},
\bauthor{\bsnm{Kim}, \binits{H.L.}},
\bauthor{\bsnm{Ditlev}, \binits{J.A.}},
\bauthor{\bsnm{Betten}, \binits{M.W.}},
\bauthor{\bsnm{Png}, \binits{C.E.}},
\bauthor{\bsnm{Kort}, \binits{E.J.}},
\bauthor{\bsnm{Futami}, \binits{K.}},
\bauthor{\bsnm{Furge}, \binits{K.A.}},
\bauthor{\bsnm{Takahashi}, \binits{M.}},
\bauthor{\bsnm{Kanayama}, \binits{H.-o.}},
\bauthor{\bsnm{Tan}, \binits{P.H.}},
\bauthor{\bsnm{Teh}, \binits{B.S.}},
\bauthor{\bsnm{Luan}, \binits{C.}},
\bauthor{\bsnm{Wang}, \binits{K.}},
\bauthor{\bsnm{Pins}, \binits{M.}},
\bauthor{\bsnm{Tretiakova}, \binits{M.}},
\bauthor{\bsnm{Anema}, \binits{J.}},
\bauthor{\bsnm{Kahnoski}, \binits{R.}},
\bauthor{\bsnm{Nicol}, \binits{T.}},
\bauthor{\bsnm{Stadler}, \binits{W.}},
\bauthor{\bsnm{Vogelzang}, \binits{N.G.}},
\bauthor{\bsnm{Amato}, \binits{R.}},
\bauthor{\bsnm{Seligson}, \binits{D.}},
\bauthor{\bsnm{Figlin}, \binits{R.}},
\bauthor{\bsnm{Belldegrun}, \binits{A.}},
\bauthor{\bsnm{Rogers}, \binits{C.G.}},
\bauthor{\bsnm{Teh}, \binits{B.T.}}:
\batitle{A molecular classification of papillary renal cell carcinoma}.
\bjtitle{Cancer Research}
\bvolume{65}(\bissue{13}),
\bfpage{5628}--\blpage{5637}
(\byear{2005}).
doi:\doiurl{10.1158/0008-5472.can-05-0533}
\end{barticle}
\endbibitem

%%% 63
\bibitem[\protect\citeauthoryear{Yang et~al.}{2011}]{Yang2011l21R}
\begin{bchapter}
\bauthor{\bsnm{Yang}, \binits{Y.}},
\bauthor{\bsnm{Shen}, \binits{H.T.}},
\bauthor{\bsnm{Ma}, \binits{Z.}},
\bauthor{\bsnm{Huang}, \binits{Z.}},
\bauthor{\bsnm{Zhou}, \binits{X.}}:
\bctitle{l2, 1-norm regularized discriminative feature selection for
  unsupervised learning}.
In: \bbtitle{International Joint Conference on Artificial Intelligence}
(\byear{2011})
\end{bchapter}
\endbibitem

%%% 64
\bibitem[\protect\citeauthoryear{Zhang}{2003}]{Zhang2003}
\begin{barticle}
\bauthor{\bsnm{Zhang}, \binits{J.}}:
\batitle{Evolution of the human aspm gene, a major determinant of brain size}.
\bjtitle{Genetics}
\bvolume{165}(\bissue{4}),
\bfpage{2063}--\blpage{2070}
(\byear{2003}).
doi:\doiurl{10.1093/genetics/165.4.2063}
\end{barticle}
\endbibitem

%%% 65
\bibitem[\protect\citeauthoryear{Zhao and Liu}{2007}]{Zhao2007}
\begin{bchapter}
\bauthor{\bsnm{Zhao}, \binits{Z.}},
\bauthor{\bsnm{Liu}, \binits{H.}}:
\bctitle{Spectral feature selection for supervised and unsupervised learning}.
In: \bbtitle{Proceedings of the 24th International Conference on Machine
  Learning}.
\bsertitle{ICML '07},
pp. \bfpage{1151}--\blpage{1157}.
\bpublisher{Association for Computing Machinery},
\blocation{New York, NY, USA}
(\byear{2007}).
doi:\doiurl{10.1145/1273496.1273641}
\end{bchapter}
\endbibitem

%%% 66
\bibitem[\protect\citeauthoryear{Zhuang et~al.}{2018}]{Zhuang2018}
\begin{barticle}
\bauthor{\bsnm{Zhuang}, \binits{L.}},
\bauthor{\bsnm{Yang}, \binits{Z.}},
\bauthor{\bsnm{Meng}, \binits{Z.}}:
\batitle{Upregulation of {BUB}1b, {CCNB}1, {CDC}7, {CDC}20, and {MCM}3 in tumor
  tissues predicted worse overall survival and disease-free survival in
  hepatocellular carcinoma patients}.
\bjtitle{{BioMed} Research International}
\bvolume{2018},
\bfpage{1}--\blpage{8}
(\byear{2018}).
doi:\doiurl{10.1155/2018/7897346}
\end{barticle}
\endbibitem

\end{thebibliography}

\newcommand{\BMCxmlcomment}[1]{}

\BMCxmlcomment{

<refgrp>

<bibl id="B1">
</bibl>

<bibl id="B2">
  <title><p>Concrete Autoencoders for Differentiable Feature Selection and
  Reconstruction</p></title>
  <aug>
    <au><snm>Abid</snm><fnm>A</fnm></au>
    <au><snm>Balin</snm><fnm>MF</fnm></au>
    <au><snm>Zou</snm><fnm>J</fnm></au>
  </aug>
  <publisher>arXiv</publisher>
  <pubdate>2019</pubdate>
  <url>https://arxiv.org/abs/1901.09346</url>
</bibl>

<bibl id="B3">
  <title><p>Kernel Independent Component Analysis</p></title>
  <aug>
    <au><snm>Bach</snm><fnm>F</fnm></au>
    <au><snm>Jordan</snm><fnm>M</fnm></au>
  </aug>
  <source>Journal of Machine Learning Research</source>
  <pubdate>2003</pubdate>
  <volume>3</volume>
  <fpage>1</fpage>
  <lpage>48</lpage>
</bibl>

<bibl id="B4">
  <title><p>{ASPM} is a major determinant of cerebral cortical size</p></title>
  <aug>
    <au><snm>Bond</snm><fnm>J</fnm></au>
    <au><snm>Roberts</snm><fnm>E</fnm></au>
    <au><snm>Mochida</snm><fnm>GH</fnm></au>
    <au><snm>Hampshire</snm><fnm>DJ</fnm></au>
    <au><snm>Scott</snm><fnm>S</fnm></au>
    <au><snm>Askham</snm><fnm>JM</fnm></au>
    <au><snm>Springell</snm><fnm>K</fnm></au>
    <au><snm>Mahadevan</snm><fnm>M</fnm></au>
    <au><snm>Crow</snm><fnm>YJ</fnm></au>
    <au><snm>Markham</snm><fnm>AF</fnm></au>
    <au><snm>Walsh</snm><fnm>CA</fnm></au>
    <au><snm>Woods</snm><fnm>CG</fnm></au>
  </aug>
  <source>Nature Genetics</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2002</pubdate>
  <volume>32</volume>
  <issue>2</issue>
  <fpage>316</fpage>
  <lpage>-320</lpage>
  <url>https://doi.org/10.1038/ng995</url>
</bibl>

<bibl id="B5">
  <title><p>Feature selection for kernel methods in systems biology</p></title>
  <aug>
    <au><snm>Brouard</snm><fnm>C</fnm></au>
    <au><snm>Mariette</snm><fnm>J</fnm></au>
    <au><snm>Flamary</snm><fnm>R</fnm></au>
    <au><snm>Vialaneix</snm><fnm>N</fnm></au>
  </aug>
  <source>{NAR} Genomics and Bioinformatics</source>
  <publisher>Oxford University Press ({OUP})</publisher>
  <pubdate>2022</pubdate>
  <volume>4</volume>
  <issue>1</issue>
  <url>https://doi.org/10.1093/nargab/lqac014</url>
</bibl>

<bibl id="B6">
  <title><p>Unsupervised feature selection for multi-cluster data</p></title>
  <aug>
    <au><snm>Cai</snm><fnm>D</fnm></au>
    <au><snm>Zhang</snm><fnm>C</fnm></au>
    <au><snm>He</snm><fnm>X</fnm></au>
  </aug>
  <source>Proceedings of the 16th ACM SIGKDD international conference on
  Knowledge discovery and data mining</source>
  <pubdate>2010</pubdate>
</bibl>

<bibl id="B7">
  <title><p>{FOXM}1 promotes proliferation in human hepatocellular carcinoma
  cells by transcriptional activation of {CCNB}1</p></title>
  <aug>
    <au><snm>Chai</snm><fnm>N</fnm></au>
    <au><snm>Xie</snm><fnm>H</fnm></au>
    <au><snm>Yin</snm><fnm>J</fnm></au>
    <au><snm>Sa</snm><fnm>K</fnm></au>
    <au><snm>Guo</snm><fnm>Y</fnm></au>
    <au><snm>Wang</snm><fnm>M</fnm></au>
    <au><snm>Liu</snm><fnm>J</fnm></au>
    <au><snm>Zhang</snm><fnm>X</fnm></au>
    <au><snm>Zhang</snm><fnm>X</fnm></au>
    <au><snm>Yin</snm><fnm>H</fnm></au>
    <au><snm>Nie</snm><fnm>Y</fnm></au>
    <au><snm>Wu</snm><fnm>K</fnm></au>
    <au><snm>Yang</snm><fnm>A</fnm></au>
    <au><snm>Zhang</snm><fnm>R</fnm></au>
  </aug>
  <source>Biochemical and Biophysical Research Communications</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2018</pubdate>
  <volume>500</volume>
  <issue>4</issue>
  <fpage>924</fpage>
  <lpage>-929</lpage>
  <url>https://doi.org/10.1016/j.bbrc.2018.04.201</url>
</bibl>

<bibl id="B8">
  <title><p>Gene expression profiles of prostate cancer reveal involvement of
  multiple molecular pathways in the metastatic process</p></title>
  <aug>
    <au><snm>Chandran</snm><fnm>UR</fnm></au>
    <au><snm>Ma</snm><fnm>C</fnm></au>
    <au><snm>Dhir</snm><fnm>R</fnm></au>
    <au><snm>Bisceglia</snm><fnm>M</fnm></au>
    <au><snm>Lyons Weiler</snm><fnm>M</fnm></au>
    <au><snm>Liang</snm><fnm>W</fnm></au>
    <au><snm>Michalopoulos</snm><fnm>G</fnm></au>
    <au><snm>Becich</snm><fnm>M</fnm></au>
    <au><snm>Monzon</snm><fnm>FA</fnm></au>
  </aug>
  <source>{BMC} Cancer</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2007</pubdate>
  <volume>7</volume>
  <issue>1</issue>
  <url>https://doi.org/10.1186/1471-2407-7-64</url>
</bibl>

<bibl id="B9">
  <title><p>Bioinformatics analysis revealing prognostic significance of RRM2
  gene in breast cancer</p></title>
  <aug>
    <au><snm>Chen</snm><fnm>W</fnm></au>
    <au><snm>Yang</snm><fnm>L</fnm></au>
    <au><snm>Xu</snm><fnm>L</fnm></au>
    <au><snm>Cheng</snm><fnm>L</fnm></au>
    <au><snm>Qian</snm><fnm>Q</fnm></au>
    <au><snm>Sun</snm><fnm>L</fnm></au>
    <au><snm>Zhu</snm><fnm>Y</fnm></au>
  </aug>
  <source>Bioscience Reports</source>
  <publisher>Portland Press Ltd.</publisher>
  <pubdate>2019</pubdate>
  <volume>39</volume>
  <issue>4</issue>
  <url>https://doi.org/10.1042/bsr20182062</url>
</bibl>

<bibl id="B10">
  <title><p>Fatty Acid Binding Protein 4 ({FABP}4) Overexpression in
  Intratumoral Hepatic Stellate Cells within Hepatocellular Carcinoma with
  Metabolic Risk Factors</p></title>
  <aug>
    <au><snm>Chiyonobu</snm><fnm>N</fnm></au>
    <au><snm>Shimada</snm><fnm>S</fnm></au>
    <au><snm>Akiyama</snm><fnm>Y</fnm></au>
    <au><snm>Mogushi</snm><fnm>K</fnm></au>
    <au><snm>Itoh</snm><fnm>M</fnm></au>
    <au><snm>Akahoshi</snm><fnm>K</fnm></au>
    <au><snm>Matsumura</snm><fnm>S</fnm></au>
    <au><snm>Ogawa</snm><fnm>K</fnm></au>
    <au><snm>Ono</snm><fnm>H</fnm></au>
    <au><snm>Mitsunori</snm><fnm>Y</fnm></au>
    <au><snm>Ban</snm><fnm>D</fnm></au>
    <au><snm>Kudo</snm><fnm>A</fnm></au>
    <au><snm>Arii</snm><fnm>S</fnm></au>
    <au><snm>Suganami</snm><fnm>T</fnm></au>
    <au><snm>Yamaoka</snm><fnm>S</fnm></au>
    <au><snm>Ogawa</snm><fnm>Y</fnm></au>
    <au><snm>Tanabe</snm><fnm>M</fnm></au>
    <au><snm>Tanaka</snm><fnm>S</fnm></au>
  </aug>
  <source>The American Journal of Pathology</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2018</pubdate>
  <volume>188</volume>
  <issue>5</issue>
  <fpage>1213</fpage>
  <lpage>-1224</lpage>
  <url>https://doi.org/10.1016/j.ajpath.2018.01.012</url>
</bibl>

<bibl id="B11">
  <title><p>{ADAM}10 as a Therapeutic Target for Cancer and
  Inflammation</p></title>
  <aug>
    <au><snm>Crawford</snm><fnm>H</fnm></au>
    <au><snm>Dempsey</snm><fnm>P</fnm></au>
    <au><snm>Brown</snm><fnm>G</fnm></au>
    <au><snm>Adam</snm><fnm>L</fnm></au>
    <au><snm>Moss</snm><fnm>M</fnm></au>
  </aug>
  <source>Current Pharmaceutical Design</source>
  <publisher>Bentham Science Publishers Ltd.</publisher>
  <pubdate>2009</pubdate>
  <volume>15</volume>
  <issue>20</issue>
  <fpage>2288</fpage>
  <lpage>-2299</lpage>
  <url>https://doi.org/10.2174/138161209788682442</url>
</bibl>

<bibl id="B12">
  <title><p>Statistical Applications of a Metric on Subspaces to Satellite
  Meteorology</p></title>
  <aug>
    <au><snm>Crone</snm><fnm>L. J.</fnm></au>
    <au><snm>Crosby</snm><fnm>D. S.</fnm></au>
  </aug>
  <source>Technometrics</source>
  <publisher>Informa {UK} Limited</publisher>
  <pubdate>1995</pubdate>
  <volume>37</volume>
  <issue>3</issue>
  <fpage>324</fpage>
  <lpage>-328</lpage>
  <url>https://doi.org/10.1080/00401706.1995.10484338</url>
</bibl>

<bibl id="B13">
  <title><p>Comparing community structure identification</p></title>
  <aug>
    <au><snm>Danon</snm><fnm>L</fnm></au>
    <au><snm>Duch</snm><fnm>J</fnm></au>
    <au><snm>Diaz Guilera</snm><fnm>A</fnm></au>
    <au><snm>Arenas</snm><fnm>A</fnm></au>
  </aug>
  <publisher>arXiv</publisher>
  <pubdate>2005</pubdate>
  <url>https://arxiv.org/abs/cond-mat/0505245</url>
</bibl>

<bibl id="B14">
  <title><p>{CCNB}1 is a prognostic biomarker for {ER}+ breast
  cancer</p></title>
  <aug>
    <au><snm>Ding</snm><fnm>K</fnm></au>
    <au><snm>Li</snm><fnm>W</fnm></au>
    <au><snm>Zou</snm><fnm>Z</fnm></au>
    <au><snm>Zou</snm><fnm>X</fnm></au>
    <au><snm>Wang</snm><fnm>C</fnm></au>
  </aug>
  <source>Medical Hypotheses</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2014</pubdate>
  <volume>83</volume>
  <issue>3</issue>
  <fpage>359</fpage>
  <lpage>-364</lpage>
  <url>https://doi.org/10.1016/j.mehy.2014.06.013</url>
</bibl>

<bibl id="B15">
  <aug>
    <au><snm>Duda</snm><fnm>RO</fnm></au>
    <au><snm>Hart</snm><fnm>PE</fnm></au>
    <au><snm>Stork</snm><fnm>DG</fnm></au>
  </aug>
  <edition>2</edition>
  <pubdate>2000</pubdate>
</bibl>

<bibl id="B16">
  <title><p>Chk1-induced {CCNB}1 overexpression promotes cell proliferation and
  tumor growth in human colorectal cancer</p></title>
  <aug>
    <au><snm>Fang</snm><fnm>Y</fnm></au>
    <au><snm>Yu</snm><fnm>H</fnm></au>
    <au><snm>Liang</snm><fnm>X</fnm></au>
    <au><snm>Xu</snm><fnm>J</fnm></au>
    <au><snm>Cai</snm><fnm>X</fnm></au>
  </aug>
  <source>Cancer Biology {\&} Therapy</source>
  <publisher>Informa {UK} Limited</publisher>
  <pubdate>2014</pubdate>
  <volume>15</volume>
  <issue>9</issue>
  <fpage>1268</fpage>
  <lpage>-1279</lpage>
  <url>https://doi.org/10.4161/cbt.29691</url>
</bibl>

<bibl id="B17">
  <title><p>{affy—analysis of Affymetrix GeneChip data at the probe
  level}</p></title>
  <aug>
    <au><snm>Gautier</snm><fnm>L</fnm></au>
    <au><snm>Cope</snm><fnm>L</fnm></au>
    <au><snm>Bolstad</snm><fnm>BM</fnm></au>
    <au><snm>Irizarry</snm><fnm>RA</fnm></au>
  </aug>
  <source>Bioinformatics</source>
  <pubdate>2004</pubdate>
  <volume>20</volume>
  <issue>3</issue>
  <fpage>307</fpage>
  <lpage>315</lpage>
  <url>https://doi.org/10.1093/bioinformatics/btg405</url>
</bibl>

<bibl id="B18">
  <title><p>Expression of L1-{CAM} and {ADAM}10 in Human Colon Cancer Cells
  Induces Metastasis</p></title>
  <aug>
    <au><snm>Gavert</snm><fnm>N</fnm></au>
    <au><snm>Sheffer</snm><fnm>M</fnm></au>
    <au><snm>Raveh</snm><fnm>S</fnm></au>
    <au><snm>Spaderna</snm><fnm>S</fnm></au>
    <au><snm>Shtutman</snm><fnm>M</fnm></au>
    <au><snm>Brabletz</snm><fnm>T</fnm></au>
    <au><snm>Barany</snm><fnm>F</fnm></au>
    <au><snm>Paty</snm><fnm>P</fnm></au>
    <au><snm>Notterman</snm><fnm>D</fnm></au>
    <au><snm>Domany</snm><fnm>E</fnm></au>
    <au><snm>Ben Ze{\textquotesingle}ev</snm><fnm>A</fnm></au>
  </aug>
  <source>Cancer Research</source>
  <publisher>American Association for Cancer Research ({AACR})</publisher>
  <pubdate>2007</pubdate>
  <volume>67</volume>
  <issue>16</issue>
  <fpage>7703</fpage>
  <lpage>-7712</lpage>
  <url>https://doi.org/10.1158/0008-5472.can-07-0991</url>
</bibl>

<bibl id="B19">
  <title><p>Mercer kernel-based clustering in feature space</p></title>
  <aug>
    <au><snm>Girolami</snm><fnm>M.</fnm></au>
  </aug>
  <source>{IEEE} Transactions on Neural Networks</source>
  <publisher>Institute of Electrical and Electronics Engineers
  ({IEEE})</publisher>
  <pubdate>2002</pubdate>
  <volume>13</volume>
  <issue>3</issue>
  <fpage>780</fpage>
  <lpage>-784</lpage>
  <url>https://doi.org/10.1109/tnn.2002.1000150</url>
</bibl>

<bibl id="B20">
  <title><p>The Elements of Statistical Learning: Data Mining, Inference, and
  Prediction</p></title>
  <aug>
    <au><snm>Hastie</snm><fnm>T.</fnm></au>
    <au><snm>Tibshirani</snm><fnm>R.</fnm></au>
    <au><snm>Friedman</snm><fnm>J.H.</fnm></au>
  </aug>
  <series><title><p>Springer series in statistics</p></title></series>
  <pubdate>2009</pubdate>
  <url>https://books.google.co.uk/books?id=eBSgoAEACAAJ</url>
</bibl>

<bibl id="B21">
  <title><p>Laplacian Score for Feature Selection</p></title>
  <aug>
    <au><snm>He</snm><fnm>X</fnm></au>
    <au><snm>Cai</snm><fnm>D</fnm></au>
    <au><snm>Niyogi</snm><fnm>P</fnm></au>
  </aug>
  <source>Proceedings of the 18th International Conference on Neural
  Information Processing Systems</source>
  <publisher>Cambridge, MA, USA: MIT Press</publisher>
  <series><title><p>NIPS'05</p></title></series>
  <pubdate>2005</pubdate>
  <fpage>507–514</fpage>
</bibl>

<bibl id="B22">
  <title><p>The {CDK}1/{TFCP}2L1/{ID}2 cascade offers a novel combination
  therapy strategy in a preclinical model of bladder cancer</p></title>
  <aug>
    <au><snm>Heo</snm><fnm>J</fnm></au>
    <au><snm>Lee</snm><fnm>J</fnm></au>
    <au><snm>Nam</snm><fnm>YJ</fnm></au>
    <au><snm>Kim</snm><fnm>Y</fnm></au>
    <au><snm>Yun</snm><fnm>H</fnm></au>
    <au><snm>Lee</snm><fnm>S</fnm></au>
    <au><snm>Ju</snm><fnm>H</fnm></au>
    <au><snm>Ryu</snm><fnm>CM</fnm></au>
    <au><snm>Jeong</snm><fnm>SM</fnm></au>
    <au><snm>Lee</snm><fnm>J</fnm></au>
    <au><snm>Lim</snm><fnm>J</fnm></au>
    <au><snm>Cho</snm><fnm>YM</fnm></au>
    <au><snm>Jeong</snm><fnm>EM</fnm></au>
    <au><snm>Hong</snm><fnm>B</fnm></au>
    <au><snm>Son</snm><fnm>J</fnm></au>
    <au><snm>Shin</snm><fnm>DM</fnm></au>
  </aug>
  <source>Experimental \& Molecular Medicine</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2022</pubdate>
  <volume>54</volume>
  <issue>6</issue>
  <fpage>801</fpage>
  <lpage>-811</lpage>
  <url>https://doi.org/10.1038/s12276-022-00786-0</url>
</bibl>

<bibl id="B23">
  <title><p>Local isomorphism to solve the pre-image problem in kernel
  methods</p></title>
  <aug>
    <au><snm>Huang</snm><fnm>D</fnm></au>
    <au><snm>Tian</snm><fnm>Y</fnm></au>
    <au><snm>Torre</snm><fnm>F</fnm></au>
  </aug>
  <source>CVPR 2011</source>
  <pubdate>2011</pubdate>
  <fpage>2761</fpage>
  <lpage>2768</lpage>
</bibl>

<bibl id="B24">
  <title><p>{RacGAP}1 expression, increasing tumor malignant potential, as a
  predictive biomarker for lymph node metastasis and poor prognosis in
  colorectal cancer</p></title>
  <aug>
    <au><snm>Imaoka</snm><fnm>H</fnm></au>
    <au><snm>Toiyama</snm><fnm>Y</fnm></au>
    <au><snm>Saigusa</snm><fnm>S</fnm></au>
    <au><snm>Kawamura</snm><fnm>M</fnm></au>
    <au><snm>Kawamoto</snm><fnm>A</fnm></au>
    <au><snm>Okugawa</snm><fnm>Y</fnm></au>
    <au><snm>Hiro</snm><fnm>J</fnm></au>
    <au><snm>Tanaka</snm><fnm>K</fnm></au>
    <au><snm>Inoue</snm><fnm>Y</fnm></au>
    <au><snm>Mohri</snm><fnm>Y</fnm></au>
    <au><snm>Kusunoki</snm><fnm>M</fnm></au>
  </aug>
  <source>Carcinogenesis</source>
  <publisher>Oxford University Press ({OUP})</publisher>
  <pubdate>2015</pubdate>
  <volume>36</volume>
  <issue>3</issue>
  <fpage>346</fpage>
  <lpage>-354</lpage>
  <url>https://doi.org/10.1093/carcin/bgu327</url>
</bibl>

<bibl id="B25">
  <title><p>{HBP}21, a chaperone of heat shock protein 70, functions as a tumor
  suppressor in hepatocellular carcinoma</p></title>
  <aug>
    <au><snm>Jiang</snm><fnm>L</fnm></au>
    <au><snm>Kwong</snm><fnm>DLW</fnm></au>
    <au><snm>Li</snm><fnm>Y</fnm></au>
    <au><snm>Liu</snm><fnm>M</fnm></au>
    <au><snm>Yuan</snm><fnm>YF</fnm></au>
    <au><snm>Li</snm><fnm>Y</fnm></au>
    <au><snm>Fu</snm><fnm>L</fnm></au>
    <au><snm>Guan</snm><fnm>XY</fnm></au>
  </aug>
  <source>Carcinogenesis</source>
  <publisher>Oxford University Press ({OUP})</publisher>
  <pubdate>2015</pubdate>
  <volume>36</volume>
  <issue>10</issue>
  <fpage>1111</fpage>
  <lpage>-1120</lpage>
  <url>https://doi.org/10.1093/carcin/bgv116</url>
</bibl>

<bibl id="B26">
  <title><p>High expression of {RRM}2 as an independent predictive factor of
  poor prognosis in patients with lung adenocarcinoma</p></title>
  <aug>
    <au><snm>Jin</snm><fnm>CY</fnm></au>
    <au><snm>Du</snm><fnm>L</fnm></au>
    <au><snm>Nuerlan</snm><fnm>AH</fnm></au>
    <au><snm>Wang</snm><fnm>XL</fnm></au>
    <au><snm>Yang</snm><fnm>YW</fnm></au>
    <au><snm>Guo</snm><fnm>R</fnm></au>
  </aug>
  <source>Aging</source>
  <publisher>Impact Journals, {LLC}</publisher>
  <pubdate>2020</pubdate>
  <volume>13</volume>
  <issue>3</issue>
  <fpage>3518</fpage>
  <lpage>-3535</lpage>
  <url>https://doi.org/10.18632/aging.202292</url>
</bibl>

<bibl id="B27">
  <title><p>Kernel methods and their derivatives: Concept and perspectives for
  the earth system sciences</p></title>
  <aug>
    <au><snm>Johnson</snm><fnm>JE</fnm></au>
    <au><snm>Laparra</snm><fnm>V</fnm></au>
    <au><snm>P{\'{e}}rez Suay</snm><fnm>A</fnm></au>
    <au><snm>Mahecha</snm><fnm>MD</fnm></au>
    <au><snm>Camps Valls</snm><fnm>G</fnm></au>
  </aug>
  <source>{PLOS} {ONE}</source>
  <publisher>Public Library of Science ({PLoS})</publisher>
  <editor>Qichun Zhang</editor>
  <pubdate>2020</pubdate>
  <volume>15</volume>
  <issue>10</issue>
  <fpage>e0235885</fpage>
  <url>https://doi.org/10.1371/journal.pone.0235885</url>
</bibl>

<bibl id="B28">
  <title><p>The microcephaly {ASPM} gene is expressed in proliferating tissues
  and encodes for a mitotic spindle protein</p></title>
  <aug>
    <au><snm>Kouprina</snm><fnm>N</fnm></au>
    <au><snm>Pavlicek</snm><fnm>A</fnm></au>
    <au><snm>Collins</snm><fnm>NK</fnm></au>
    <au><snm>Nakano</snm><fnm>M</fnm></au>
    <au><snm>Noskov</snm><fnm>VN</fnm></au>
    <au><snm>Ohzeki</snm><fnm>JI</fnm></au>
    <au><snm>Mochida</snm><fnm>GH</fnm></au>
    <au><snm>Risinger</snm><fnm>JI</fnm></au>
    <au><snm>Goldsmith</snm><fnm>P</fnm></au>
    <au><snm>Gunsior</snm><fnm>M</fnm></au>
    <au><snm>Solomon</snm><fnm>G</fnm></au>
    <au><snm>Gersch</snm><fnm>W</fnm></au>
    <au><snm>Kim</snm><fnm>JH</fnm></au>
    <au><snm>Barrett</snm><fnm>JC</fnm></au>
    <au><snm>Walsh</snm><fnm>CA</fnm></au>
    <au><snm>Jurka</snm><fnm>J</fnm></au>
    <au><snm>Masumoto</snm><fnm>H</fnm></au>
    <au><snm>Larionov</snm><fnm>V</fnm></au>
  </aug>
  <source>Human Molecular Genetics</source>
  <publisher>Oxford University Press ({OUP})</publisher>
  <pubdate>2005</pubdate>
  <volume>14</volume>
  <issue>15</issue>
  <fpage>2155</fpage>
  <lpage>-2165</lpage>
  <url>https://doi.org/10.1093/hmg/ddi220</url>
</bibl>

<bibl id="B29">
  <title><p>A robust prognostic gene expression signature for early stage lung
  adenocarcinoma</p></title>
  <aug>
    <au><snm>Krzystanek</snm><fnm>M</fnm></au>
    <au><snm>Moldvay</snm><fnm>J</fnm></au>
    <au><snm>Sz\"{u}ts</snm><fnm>D</fnm></au>
    <au><snm>Szallasi</snm><fnm>Z</fnm></au>
    <au><snm>Eklund</snm><fnm>AC</fnm></au>
  </aug>
  <source>Biomarker Research</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2016</pubdate>
  <volume>4</volume>
  <issue>1</issue>
  <url>https://doi.org/10.1186/s40364-016-0058-3</url>
</bibl>

<bibl id="B30">
  <title><p>The Pre-Image Problem in Kernel Methods</p></title>
  <aug>
    <au><snm>Kwok</snm><fnm>J.T. Y.</fnm></au>
    <au><snm>Tsang</snm><fnm>I.W. H.</fnm></au>
  </aug>
  <source>{IEEE} Transactions on Neural Networks</source>
  <publisher>Institute of Electrical and Electronics Engineers
  ({IEEE})</publisher>
  <pubdate>2004</pubdate>
  <volume>15</volume>
  <issue>6</issue>
  <fpage>1517</fpage>
  <lpage>-1525</lpage>
  <url>https://doi.org/10.1109/tnn.2004.837781</url>
</bibl>

<bibl id="B31">
  <title><p>{ADAM}10 Is Upregulated in Melanoma Metastasis Compared with
  Primary Melanoma</p></title>
  <aug>
    <au><snm>Lee</snm><fnm>SB</fnm></au>
    <au><snm>Schramme</snm><fnm>A</fnm></au>
    <au><snm>Doberstein</snm><fnm>K</fnm></au>
    <au><snm>Dummer</snm><fnm>R</fnm></au>
    <au><snm>Abdel Bakky</snm><fnm>MS</fnm></au>
    <au><snm>Keller</snm><fnm>S</fnm></au>
    <au><snm>Altevogt</snm><fnm>P</fnm></au>
    <au><snm>Oh</snm><fnm>ST</fnm></au>
    <au><snm>Reichrath</snm><fnm>J</fnm></au>
    <au><snm>Oxmann</snm><fnm>D</fnm></au>
    <au><snm>Pfeilschifter</snm><fnm>J</fnm></au>
    <au><snm>Mihic Probst</snm><fnm>D</fnm></au>
    <au><snm>Gutwein</snm><fnm>P</fnm></au>
  </aug>
  <source>Journal of Investigative Dermatology</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2010</pubdate>
  <volume>130</volume>
  <issue>3</issue>
  <fpage>763</fpage>
  <lpage>-773</lpage>
  <url>https://doi.org/10.1038/jid.2009.335</url>
</bibl>

<bibl id="B32">
  <title><p>{CDK}1 and {CDC}20 overexpression in patients with colorectal
  cancer are associated with poor prognosis: evidence from integrated
  bioinformatics analysis</p></title>
  <aug>
    <au><snm>Li</snm><fnm>J</fnm></au>
    <au><snm>Wang</snm><fnm>Y</fnm></au>
    <au><snm>Wang</snm><fnm>X</fnm></au>
    <au><snm>Yang</snm><fnm>Q</fnm></au>
  </aug>
  <source>World Journal of Surgical Oncology</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2020</pubdate>
  <volume>18</volume>
  <issue>1</issue>
  <url>https://doi.org/10.1186/s12957-020-01817-8</url>
</bibl>

<bibl id="B33">
  <title><p>Feature Selection Feature selection: a data perspective</p></title>
  <aug>
    <au><snm>Li</snm><fnm>J</fnm></au>
    <au><snm>Cheng</snm><fnm>K</fnm></au>
    <au><snm>Wang</snm><fnm>S</fnm></au>
    <au><snm>Morstatter</snm><fnm>F</fnm></au>
    <au><snm>Trevino</snm><fnm>RP</fnm></au>
    <au><snm>Tang</snm><fnm>J</fnm></au>
    <au><snm>Liu</snm><fnm>H</fnm></au>
  </aug>
  <source>{ACM} Computing Surveys</source>
  <publisher>Association for Computing Machinery ({ACM})</publisher>
  <pubdate>2017</pubdate>
  <volume>50</volume>
  <issue>6</issue>
  <fpage>1</fpage>
  <lpage>-45</lpage>
  <url>https://doi.org/10.1145/3136625</url>
</bibl>

<bibl id="B34">
  <title><p>{CDK}1 serves as a potential prognostic biomarker and target for
  lung cancer</p></title>
  <aug>
    <au><snm>Li</snm><fnm>M</fnm></au>
    <au><snm>He</snm><fnm>F</fnm></au>
    <au><snm>Zhang</snm><fnm>Z</fnm></au>
    <au><snm>Xiang</snm><fnm>Z</fnm></au>
    <au><snm>Hu</snm><fnm>D</fnm></au>
  </aug>
  <source>Journal of International Medical Research</source>
  <publisher>{SAGE} Publications</publisher>
  <pubdate>2020</pubdate>
  <volume>48</volume>
  <issue>2</issue>
  <fpage>030006051989750</fpage>
  <url>https://doi.org/10.1177/0300060519897508</url>
</bibl>

<bibl id="B35">
  <title><p>Unsupervised Feature Selection Using Nonnegative Spectral
  Analysis</p></title>
  <aug>
    <au><snm>Li</snm><fnm>Z</fnm></au>
    <au><snm>Yang</snm><fnm>Y</fnm></au>
    <au><snm>Liu</snm><fnm>J</fnm></au>
    <au><snm>Zhou</snm><fnm>X</fnm></au>
    <au><snm>Lu</snm><fnm>H</fnm></au>
  </aug>
  <source>Proceedings of the {AAAI} Conference on Artificial
  Intelligence</source>
  <publisher>Association for the Advancement of Artificial Intelligence
  ({AAAI})</publisher>
  <pubdate>2021</pubdate>
  <volume>26</volume>
  <issue>1</issue>
  <fpage>1026</fpage>
  <lpage>-1032</lpage>
  <url>https://doi.org/10.1609/aaai.v26i1.8289</url>
</bibl>

<bibl id="B36">
  <title><p>An Integrative Human Pan-Cancer Analysis of Cyclin-Dependent Kinase
  1 ({CDK}1)</p></title>
  <aug>
    <au><snm>Liu</snm><fnm>X</fnm></au>
    <au><snm>Wu</snm><fnm>H</fnm></au>
    <au><snm>Liu</snm><fnm>Z</fnm></au>
  </aug>
  <source>Cancers</source>
  <publisher>{MDPI} {AG}</publisher>
  <pubdate>2022</pubdate>
  <volume>14</volume>
  <issue>11</issue>
  <fpage>2658</fpage>
  <url>https://doi.org/10.3390/cancers14112658</url>
</bibl>

<bibl id="B37">
  <title><p>Hepatocellular carcinoma</p></title>
  <aug>
    <au><snm>Llovet</snm><fnm>JM</fnm></au>
    <au><snm>Kelley</snm><fnm>RK</fnm></au>
    <au><snm>Villanueva</snm><fnm>A</fnm></au>
    <au><snm>Singal</snm><fnm>AG</fnm></au>
    <au><snm>Pikarsky</snm><fnm>E</fnm></au>
    <au><snm>Roayaie</snm><fnm>S</fnm></au>
    <au><snm>Lencioni</snm><fnm>R</fnm></au>
    <au><snm>Koike</snm><fnm>K</fnm></au>
    <au><snm>Zucman Rossi</snm><fnm>J</fnm></au>
    <au><snm>Finn</snm><fnm>RS</fnm></au>
  </aug>
  <source>Nature Reviews Disease Primers</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2021</pubdate>
  <volume>7</volume>
  <issue>1</issue>
  <url>https://doi.org/10.1038/s41572-020-00240-3</url>
</bibl>

<bibl id="B38">
  <title><p>Unsupervised multiple kernel learning for heterogeneous data
  integration</p></title>
  <aug>
    <au><snm>Mariette</snm><fnm>J</fnm></au>
    <au><snm>Villa Vialaneix</snm><fnm>N</fnm></au>
  </aug>
  <source>Bioinformatics</source>
  <publisher>Oxford University Press ({OUP})</publisher>
  <editor>Jonathan Wren</editor>
  <pubdate>2017</pubdate>
  <volume>34</volume>
  <issue>6</issue>
  <fpage>1009</fpage>
  <lpage>-1015</lpage>
  <url>https://doi.org/10.1093/bioinformatics/btx682</url>
</bibl>

<bibl id="B39">
  <title><p>Convex Principal Feature Selection</p></title>
  <aug>
    <au><snm>Masaeli</snm><fnm>M</fnm></au>
    <au><snm>Yan</snm><fnm>Y</fnm></au>
    <au><snm>Cui</snm><fnm>Y</fnm></au>
    <au><snm>Fung</snm><fnm>G</fnm></au>
    <au><snm>Dy</snm><fnm>JG</fnm></au>
  </aug>
  <source>SDM</source>
  <pubdate>2010</pubdate>
</bibl>

<bibl id="B40">
  <title><p>{RNA}-seq Identification of {RACGAP}1 as a Metastatic Driver in
  Uterine Carcinosarcoma</p></title>
  <aug>
    <au><snm>Mi</snm><fnm>S</fnm></au>
    <au><snm>Lin</snm><fnm>M</fnm></au>
    <au><snm>Brouwer Visser</snm><fnm>J</fnm></au>
    <au><snm>Heim</snm><fnm>J</fnm></au>
    <au><snm>Smotkin</snm><fnm>D</fnm></au>
    <au><snm>Hebert</snm><fnm>T</fnm></au>
    <au><snm>Gunter</snm><fnm>MJ</fnm></au>
    <au><snm>Goldberg</snm><fnm>GL</fnm></au>
    <au><snm>Zheng</snm><fnm>D</fnm></au>
    <au><snm>Huang</snm><fnm>GS</fnm></au>
  </aug>
  <source>Clinical Cancer Research</source>
  <publisher>American Association for Cancer Research ({AACR})</publisher>
  <pubdate>2016</pubdate>
  <volume>22</volume>
  <issue>18</issue>
  <fpage>4676</fpage>
  <lpage>-4686</lpage>
  <url>https://doi.org/10.1158/1078-0432.ccr-15-2116</url>
</bibl>

<bibl id="B41">
  <title><p>Kernel PCA and De-Noising in Feature Spaces</p></title>
  <aug>
    <au><snm>Mika</snm><fnm>S</fnm></au>
    <au><snm>Sch{\"o}lkopf</snm><fnm>B</fnm></au>
    <au><snm>Smola</snm><fnm>A</fnm></au>
    <au><snm>M{\"u}ller</snm><fnm>KR</fnm></au>
    <au><snm>Scholz</snm><fnm>M</fnm></au>
    <au><snm>R{\"a}tsch</snm><fnm>G</fnm></au>
  </aug>
  <source>NIPS</source>
  <pubdate>1998</pubdate>
</bibl>

<bibl id="B42">
  <title><p>The Role of Angiogenesis in Hepatocellular Carcinoma</p></title>
  <aug>
    <au><snm>Morse</snm><fnm>MA</fnm></au>
    <au><snm>Sun</snm><fnm>W</fnm></au>
    <au><snm>Kim</snm><fnm>R</fnm></au>
    <au><snm>He</snm><fnm>AR</fnm></au>
    <au><snm>Abada</snm><fnm>PB</fnm></au>
    <au><snm>Mynderse</snm><fnm>M</fnm></au>
    <au><snm>Finn</snm><fnm>RS</fnm></au>
  </aug>
  <source>Clinical Cancer Research</source>
  <publisher>American Association for Cancer Research ({AACR})</publisher>
  <pubdate>2019</pubdate>
  <volume>25</volume>
  <issue>3</issue>
  <fpage>912</fpage>
  <lpage>-920</lpage>
  <url>https://doi.org/10.1158/1078-0432.ccr-18-1254</url>
</bibl>

<bibl id="B43">
  <title><p>{ADAM}10 as a Target for Anti-Cancer Therapy</p></title>
  <aug>
    <au><snm>Moss</snm><fnm>M</fnm></au>
    <au><snm>Stoeck</snm><fnm>A</fnm></au>
    <au><snm>Yan</snm><fnm>W</fnm></au>
    <au><snm>Dempsey</snm><fnm>P</fnm></au>
  </aug>
  <source>Current Pharmaceutical Biotechnology</source>
  <publisher>Bentham Science Publishers Ltd.</publisher>
  <pubdate>2008</pubdate>
  <volume>9</volume>
  <issue>1</issue>
  <fpage>2</fpage>
  <lpage>-8</lpage>
  <url>https://doi.org/10.2174/138920108783497613</url>
</bibl>

<bibl id="B44">
  <title><p>Translational evidence for {RRM}2 as a prognostic biomarker and
  therapeutic target in Ewing sarcoma</p></title>
  <aug>
    <au><snm>Ohmura</snm><fnm>S</fnm></au>
    <au><snm>Marchetto</snm><fnm>A</fnm></au>
    <au><snm>Orth</snm><fnm>MF</fnm></au>
    <au><snm>Li</snm><fnm>J</fnm></au>
    <au><snm>Jabar</snm><fnm>S</fnm></au>
    <au><snm>Ranft</snm><fnm>A</fnm></au>
    <au><snm>Vinca</snm><fnm>E</fnm></au>
    <au><snm>Ceranski</snm><fnm>K</fnm></au>
    <au><snm>Carre{\~{n}}o Gonzalez</snm><fnm>MJ</fnm></au>
    <au><snm>Romero P{\'{e}}rez</snm><fnm>L</fnm></au>
    <au><snm>Wehweck</snm><fnm>FS</fnm></au>
    <au><snm>Musa</snm><fnm>J</fnm></au>
    <au><snm>Bestvater</snm><fnm>F</fnm></au>
    <au><snm>Knott</snm><fnm>MML</fnm></au>
    <au><snm>H\"{o}lting</snm><fnm>TLB</fnm></au>
    <au><snm>Hartmann</snm><fnm>W</fnm></au>
    <au><snm>Dirksen</snm><fnm>U</fnm></au>
    <au><snm>Kirchner</snm><fnm>T</fnm></au>
    <au><snm>Cidre Aranaz</snm><fnm>F</fnm></au>
    <au><snm>Gr\"{u}newald</snm><fnm>TGP</fnm></au>
  </aug>
  <source>Molecular Cancer</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2021</pubdate>
  <volume>20</volume>
  <issue>1</issue>
  <url>https://doi.org/10.1186/s12943-021-01393-9</url>
</bibl>

<bibl id="B45">
  <title><p>Potential tumor‑suppressive role of {microRNA}‑99a‑3p in
  sunitinib‑resistant renal cell carcinoma cells through the regulation of
  {RRM}2</p></title>
  <aug>
    <au><snm>Osako</snm><fnm>Y</fnm></au>
    <au><snm>Yoshino</snm><fnm>H</fnm></au>
    <au><snm>Sakaguchi</snm><fnm>T</fnm></au>
    <au><snm>Sugita</snm><fnm>S</fnm></au>
    <au><snm>Yonemori</snm><fnm>M</fnm></au>
    <au><snm>Nakagawa</snm><fnm>M</fnm></au>
    <au><snm>Enokida</snm><fnm>H</fnm></au>
  </aug>
  <source>International Journal of Oncology</source>
  <publisher>Spandidos Publications</publisher>
  <pubdate>2019</pubdate>
  <url>https://doi.org/10.3892/ijo.2019.4736</url>
</bibl>

<bibl id="B46">
  <title><p>Prognostic significance of {RACGAP}1 {mRNA} expression in high-risk
  early breast cancer: a study in primary tumors of breast cancer patients
  participating in a randomized Hellenic Cooperative Oncology Group
  trial</p></title>
  <aug>
    <au><snm>Pliarchopoulou</snm><fnm>K.</fnm></au>
    <au><snm>Kalogeras</snm><fnm>K. T.</fnm></au>
    <au><snm>Kronenwett</snm><fnm>R.</fnm></au>
    <au><snm>Wirtz</snm><fnm>R. M.</fnm></au>
    <au><snm>Eleftheraki</snm><fnm>A. G.</fnm></au>
    <au><snm>Batistatou</snm><fnm>A.</fnm></au>
    <au><snm>Bobos</snm><fnm>M.</fnm></au>
    <au><snm>Soupos</snm><fnm>N.</fnm></au>
    <au><snm>Polychronidou</snm><fnm>G.</fnm></au>
    <au><snm>Gogas</snm><fnm>H.</fnm></au>
    <au><snm>Samantas</snm><fnm>E.</fnm></au>
    <au><snm>Christodoulou</snm><fnm>C.</fnm></au>
    <au><snm>Makatsoris</snm><fnm>T.</fnm></au>
    <au><snm>Pavlidis</snm><fnm>N.</fnm></au>
    <au><snm>Pectasides</snm><fnm>D.</fnm></au>
    <au><snm>Fountzilas</snm><fnm>G.</fnm></au>
  </aug>
  <source>Cancer Chemotherapy and Pharmacology</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2012</pubdate>
  <volume>71</volume>
  <issue>1</issue>
  <fpage>245</fpage>
  <lpage>-255</lpage>
  <url>https://doi.org/10.1007/s00280-012-2002-z</url>
</bibl>

<bibl id="B47">
  <title><p>Systemic delivery of {siRNA} nanoparticles targeting {RRM}2
  suppresses head and neck tumor growth</p></title>
  <aug>
    <au><snm>Rahman</snm><fnm>MA</fnm></au>
    <au><snm>Amin</snm><fnm>AR</fnm></au>
    <au><snm>Wang</snm><fnm>X</fnm></au>
    <au><snm>Zuckerman</snm><fnm>JE</fnm></au>
    <au><snm>Choi</snm><fnm>CHJ</fnm></au>
    <au><snm>Zhou</snm><fnm>B</fnm></au>
    <au><snm>Wang</snm><fnm>D</fnm></au>
    <au><snm>Nannapaneni</snm><fnm>S</fnm></au>
    <au><snm>Koenig</snm><fnm>L</fnm></au>
    <au><snm>Chen</snm><fnm>Z</fnm></au>
    <au><snm>Chen</snm><fnm>ZG</fnm></au>
    <au><snm>Yen</snm><fnm>Y</fnm></au>
    <au><snm>Davis</snm><fnm>ME</fnm></au>
    <au><snm>Shin</snm><fnm>DM</fnm></au>
  </aug>
  <source>Journal of Controlled Release</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2012</pubdate>
  <volume>159</volume>
  <issue>3</issue>
  <fpage>384</fpage>
  <lpage>-392</lpage>
  <url>https://doi.org/10.1016/j.jconrel.2012.01.045</url>
</bibl>

<bibl id="B48">
  <title><p>Visualization of nonlinear kernel models in neuroimaging by
  sensitivity maps</p></title>
  <aug>
    <au><snm>Rasmussen</snm><fnm>PM</fnm></au>
    <au><snm>Madsen</snm><fnm>KH</fnm></au>
    <au><snm>Lund</snm><fnm>TE</fnm></au>
    <au><snm>Hansen</snm><fnm>LK</fnm></au>
  </aug>
  <source>{NeuroImage}</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2011</pubdate>
  <volume>55</volume>
  <issue>3</issue>
  <fpage>1120</fpage>
  <lpage>-1131</lpage>
  <url>https://doi.org/10.1016/j.neuroimage.2010.12.035</url>
</bibl>

<bibl id="B49">
  <title><p>Kernel-{PCA} data integration with enhanced
  interpretability</p></title>
  <aug>
    <au><snm>Reverter</snm><fnm>F</fnm></au>
    <au><snm>Vegas</snm><fnm>E</fnm></au>
    <au><snm>Oller</snm><fnm>JM</fnm></au>
  </aug>
  <source>{BMC} Systems Biology</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2014</pubdate>
  <volume>8</volume>
  <issue>S2</issue>
  <url>https://doi.org/10.1186/1752-0509-8-s2-s6</url>
</bibl>

<bibl id="B50">
  <title><p>Nonlinear Discriminant Analysis Using Kernel Functions</p></title>
  <aug>
    <au><snm>Roth</snm><fnm>V</fnm></au>
    <au><snm>Steinhage</snm><fnm>V</fnm></au>
  </aug>
  <source>NIPS</source>
  <pubdate>1999</pubdate>
  <fpage>568</fpage>
  <lpage>574</lpage>
  <url>http://papers.nips.cc/paper/1736-nonlinear-discriminant-analysis-using-kernel-functions</url>
</bibl>

<bibl id="B51">
  <title><p>An overview of gradient descent optimization algorithms</p></title>
  <aug>
    <au><snm>Ruder</snm><fnm>S</fnm></au>
  </aug>
  <publisher>arXiv</publisher>
  <pubdate>2016</pubdate>
  <url>https://arxiv.org/abs/1609.04747</url>
</bibl>

<bibl id="B52">
  <title><p>Clinical significance of {RacGAP}1 expression at the invasive front
  of gastric cancer</p></title>
  <aug>
    <au><snm>Saigusa</snm><fnm>S</fnm></au>
    <au><snm>Tanaka</snm><fnm>K</fnm></au>
    <au><snm>Mohri</snm><fnm>Y</fnm></au>
    <au><snm>Ohi</snm><fnm>M</fnm></au>
    <au><snm>Shimura</snm><fnm>T</fnm></au>
    <au><snm>Kitajima</snm><fnm>T</fnm></au>
    <au><snm>Kondo</snm><fnm>S</fnm></au>
    <au><snm>Okugawa</snm><fnm>Y</fnm></au>
    <au><snm>Toiyama</snm><fnm>Y</fnm></au>
    <au><snm>Inoue</snm><fnm>Y</fnm></au>
    <au><snm>Kusunoki</snm><fnm>M</fnm></au>
  </aug>
  <source>Gastric Cancer</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2014</pubdate>
  <volume>18</volume>
  <issue>1</issue>
  <fpage>84</fpage>
  <lpage>-92</lpage>
  <url>https://doi.org/10.1007/s10120-014-0355-1</url>
</bibl>

<bibl id="B53">
  <title><p>Gene expression profiles of primary {HPV}16- and {HPV}18-infected
  early stage cervical cancers and normal cervical epithelium: identification
  of novel candidate molecular markers for cervical cancer diagnosis and
  therapy</p></title>
  <aug>
    <au><snm>Santin</snm><fnm>AD</fnm></au>
    <au><snm>Zhan</snm><fnm>F</fnm></au>
    <au><snm>Bignotti</snm><fnm>E</fnm></au>
    <au><snm>Siegel</snm><fnm>ER</fnm></au>
    <au><snm>Can{\'{e}}</snm><fnm>S</fnm></au>
    <au><snm>Bellone</snm><fnm>S</fnm></au>
    <au><snm>Palmieri</snm><fnm>M</fnm></au>
    <au><snm>Anfossi</snm><fnm>S</fnm></au>
    <au><snm>Thomas</snm><fnm>M</fnm></au>
    <au><snm>Burnett</snm><fnm>A</fnm></au>
    <au><snm>Kay</snm><fnm>HH</fnm></au>
    <au><snm>Roman</snm><fnm>JJ</fnm></au>
    <au><snm>O{\textquotesingle}Brien</snm><fnm>TJ</fnm></au>
    <au><snm>Tian</snm><fnm>E</fnm></au>
    <au><snm>Cannon</snm><fnm>MJ</fnm></au>
    <au><snm>Shaughnessy</snm><fnm>J</fnm></au>
    <au><snm>Pecorelli</snm><fnm>S</fnm></au>
  </aug>
  <source>Virology</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2005</pubdate>
  <volume>331</volume>
  <issue>2</issue>
  <fpage>269</fpage>
  <lpage>-291</lpage>
  <url>https://doi.org/10.1016/j.virol.2004.09.045</url>
</bibl>

<bibl id="B54">
  <title><p>{SVM}-{RFE}: selection and visualization of the most relevant
  features through non-linear kernels</p></title>
  <aug>
    <au><snm>Sanz</snm><fnm>H</fnm></au>
    <au><snm>Valim</snm><fnm>C</fnm></au>
    <au><snm>Vegas</snm><fnm>E</fnm></au>
    <au><snm>Oller</snm><fnm>JM</fnm></au>
    <au><snm>Reverter</snm><fnm>F</fnm></au>
  </aug>
  <source>{BMC} Bioinformatics</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2018</pubdate>
  <volume>19</volume>
  <issue>1</issue>
  <url>https://doi.org/10.1186/s12859-018-2451-4</url>
</bibl>

<bibl id="B55">
  <title><p>Fast approximation of support vector kernel expansions, and an
  interpretation of clustering as approximation in feature spaces.</p></title>
  <aug>
    <au><snm>Sch{\"o}lkopf</snm><fnm>B.</fnm></au>
    <au><snm>Knirsch</snm><fnm>P.</fnm></au>
    <au><snm>Smola</snm><fnm>AJ.</fnm></au>
    <au><snm>Burges</snm><fnm>C.</fnm></au>
  </aug>
  <source>Mustererkennung 1998</source>
  <publisher>Biologische Kybernetik</publisher>
  <series><title><p>Informatik aktuell</p></title></series>
  <pubdate>1998</pubdate>
  <fpage>125</fpage>
  <lpage>132</lpage>
</bibl>

<bibl id="B56">
  <title><p>Kernel principal component analysis</p></title>
  <aug>
    <au><snm>Sch{\"o}lkopf</snm><fnm>B</fnm></au>
    <au><snm>Smola</snm><fnm>A</fnm></au>
    <au><snm>M{\"u}ller</snm><fnm>KR</fnm></au>
  </aug>
  <source>Artificial Neural Networks --- ICANN'97</source>
  <publisher>Berlin, Heidelberg: Springer Berlin Heidelberg</publisher>
  <pubdate>1997</pubdate>
  <fpage>583</fpage>
  <lpage>-588</lpage>
</bibl>

<bibl id="B57">
  <title><p>Kernel Methods in Computational Biology</p></title>
  <aug>
    <au><snm>Sch{\"o}lkopf</snm><fnm>B</fnm></au>
    <au><snm>Tsuda</snm><fnm>K</fnm></au>
    <au><snm>Vert</snm><fnm>JP</fnm></au>
  </aug>
  <pubdate>2003</pubdate>
</bibl>

<bibl id="B58">
  <title><p>Targeting cyclin-dependent kinase 1 ({CDK}1) in cancer: molecular
  docking and dynamic simulations of potential {CDK}1 inhibitors</p></title>
  <aug>
    <au><snm>Sofi</snm><fnm>S</fnm></au>
    <au><snm>Mehraj</snm><fnm>U</fnm></au>
    <au><snm>Qayoom</snm><fnm>H</fnm></au>
    <au><snm>Aisha</snm><fnm>S</fnm></au>
    <au><snm>Almilaibary</snm><fnm>A</fnm></au>
    <au><snm>Alkhanani</snm><fnm>M</fnm></au>
    <au><snm>Mir</snm><fnm>MA</fnm></au>
  </aug>
  <source>Medical Oncology</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2022</pubdate>
  <volume>39</volume>
  <issue>9</issue>
  <url>https://doi.org/10.1007/s12032-022-01748-2</url>
</bibl>

<bibl id="B59">
  <title><p>Increased expression of {RRM}2 by human papillomavirus E7
  oncoprotein promotes angiogenesis in cervical cancer</p></title>
  <aug>
    <au><snm>Wang</snm><fnm>N</fnm></au>
    <au><snm>Zhan</snm><fnm>T</fnm></au>
    <au><snm>Ke</snm><fnm>T</fnm></au>
    <au><snm>Huang</snm><fnm>X</fnm></au>
    <au><snm>Ke</snm><fnm>D</fnm></au>
    <au><snm>Wang</snm><fnm>Q</fnm></au>
    <au><snm>Li</snm><fnm>H</fnm></au>
  </aug>
  <source>British Journal of Cancer</source>
  <publisher>Springer Science and Business Media {LLC}</publisher>
  <pubdate>2014</pubdate>
  <volume>110</volume>
  <issue>4</issue>
  <fpage>1034</fpage>
  <lpage>-1044</lpage>
  <url>https://doi.org/10.1038/bjc.2013.817</url>
</bibl>

<bibl id="B60">
  <title><p>A Gene Expression Signature of Epithelial Tubulogenesis and a Role
  for {ASPM} in Pancreatic Tumor Progression</p></title>
  <aug>
    <au><snm>Wang</snm><fnm>W</fnm></au>
    <au><snm>Hsu</snm><fnm>C</fnm></au>
    <au><snm>Wang</snm><fnm>T</fnm></au>
    <au><snm>Li</snm><fnm>C</fnm></au>
    <au><snm>Hou</snm><fnm>Y</fnm></au>
    <au><snm>Chu</snm><fnm>J</fnm></au>
    <au><snm>Lee</snm><fnm>C</fnm></au>
    <au><snm>Liu</snm><fnm>M</fnm></au>
    <au><snm>Su</snm><fnm>JJ</fnm></au>
    <au><snm>Jian</snm><fnm>K</fnm></au>
    <au><snm>Huang</snm><fnm>S</fnm></au>
    <au><snm>Jiang</snm><fnm>S</fnm></au>
    <au><snm>Shan</snm><fnm>Y</fnm></au>
    <au><snm>Lin</snm><fnm>P</fnm></au>
    <au><snm>Shen</snm><fnm>Y</fnm></au>
    <au><snm>Lee</snm><fnm>MT</fnm></au>
    <au><snm>Chan</snm><fnm>T</fnm></au>
    <au><snm>Chang</snm><fnm>C</fnm></au>
    <au><snm>Chen</snm><fnm>C</fnm></au>
    <au><snm>Chang</snm><fnm>I</fnm></au>
    <au><snm>Lee</snm><fnm>Y</fnm></au>
    <au><snm>Chen</snm><fnm>L</fnm></au>
    <au><snm>Tsai</snm><fnm>KK</fnm></au>
  </aug>
  <source>Gastroenterology</source>
  <publisher>Elsevier {BV}</publisher>
  <pubdate>2013</pubdate>
  <volume>145</volume>
  <issue>5</issue>
  <fpage>1110</fpage>
  <lpage>-1120</lpage>
  <url>https://doi.org/10.1053/j.gastro.2013.07.040</url>
</bibl>

<bibl id="B61">
  <title><p>Gene Expression Profiling of Childhood Adrenocortical
  Tumors</p></title>
  <aug>
    <au><snm>West</snm><fnm>AN</fnm></au>
    <au><snm>Neale</snm><fnm>GA</fnm></au>
    <au><snm>Pounds</snm><fnm>S</fnm></au>
    <au><snm>Figueredo</snm><fnm>BC</fnm></au>
    <au><snm>Galindo</snm><fnm>CR</fnm></au>
    <au><snm>Pianovski</snm><fnm>MAD</fnm></au>
    <au><snm>Filho</snm><fnm>AGO</fnm></au>
    <au><snm>Malkin</snm><fnm>D</fnm></au>
    <au><snm>Lalli</snm><fnm>E</fnm></au>
    <au><snm>Ribeiro</snm><fnm>R</fnm></au>
    <au><snm>Zambetti</snm><fnm>GP</fnm></au>
  </aug>
  <source>Cancer Research</source>
  <publisher>American Association for Cancer Research ({AACR})</publisher>
  <pubdate>2007</pubdate>
  <volume>67</volume>
  <issue>2</issue>
  <fpage>600</fpage>
  <lpage>-608</lpage>
  <url>https://doi.org/10.1158/0008-5472.can-06-3767</url>
</bibl>

<bibl id="B62">
  <title><p>Overexpression of the {ASPM} gene is associated with aggressiveness
  and poor outcome in bladder cancer</p></title>
  <aug>
    <au><snm>Xu</snm><fnm>Z</fnm></au>
    <au><snm>Zhang</snm><fnm>Q</fnm></au>
    <au><snm>Luh</snm><fnm>F</fnm></au>
    <au><snm>Jin</snm><fnm>B</fnm></au>
    <au><snm>Liu</snm><fnm>X</fnm></au>
  </aug>
  <source>Oncology Letters</source>
  <publisher>Spandidos Publications</publisher>
  <pubdate>2018</pubdate>
  <url>https://doi.org/10.3892/ol.2018.9762</url>
</bibl>

<bibl id="B63">
  <title><p>A Molecular Classification of Papillary Renal Cell
  Carcinoma</p></title>
  <aug>
    <au><snm>Yang</snm><fnm>XJ</fnm></au>
    <au><snm>Tan</snm><fnm>MH</fnm></au>
    <au><snm>Kim</snm><fnm>HL</fnm></au>
    <au><snm>Ditlev</snm><fnm>JA</fnm></au>
    <au><snm>Betten</snm><fnm>MW</fnm></au>
    <au><snm>Png</snm><fnm>CE</fnm></au>
    <au><snm>Kort</snm><fnm>EJ</fnm></au>
    <au><snm>Futami</snm><fnm>K</fnm></au>
    <au><snm>Furge</snm><fnm>KA</fnm></au>
    <au><snm>Takahashi</snm><fnm>M</fnm></au>
    <au><snm>Kanayama</snm><fnm>H</fnm></au>
    <au><snm>Tan</snm><fnm>PH</fnm></au>
    <au><snm>Teh</snm><fnm>BS</fnm></au>
    <au><snm>Luan</snm><fnm>C</fnm></au>
    <au><snm>Wang</snm><fnm>K</fnm></au>
    <au><snm>Pins</snm><fnm>M</fnm></au>
    <au><snm>Tretiakova</snm><fnm>M</fnm></au>
    <au><snm>Anema</snm><fnm>J</fnm></au>
    <au><snm>Kahnoski</snm><fnm>R</fnm></au>
    <au><snm>Nicol</snm><fnm>T</fnm></au>
    <au><snm>Stadler</snm><fnm>W</fnm></au>
    <au><snm>Vogelzang</snm><fnm>NG</fnm></au>
    <au><snm>Amato</snm><fnm>R</fnm></au>
    <au><snm>Seligson</snm><fnm>D</fnm></au>
    <au><snm>Figlin</snm><fnm>R</fnm></au>
    <au><snm>Belldegrun</snm><fnm>A</fnm></au>
    <au><snm>Rogers</snm><fnm>CG</fnm></au>
    <au><snm>Teh</snm><fnm>BT</fnm></au>
  </aug>
  <source>Cancer Research</source>
  <publisher>American Association for Cancer Research ({AACR})</publisher>
  <pubdate>2005</pubdate>
  <volume>65</volume>
  <issue>13</issue>
  <fpage>5628</fpage>
  <lpage>-5637</lpage>
  <url>https://doi.org/10.1158/0008-5472.can-05-0533</url>
</bibl>

<bibl id="B64">
  <title><p>l2, 1-Norm Regularized Discriminative Feature Selection for
  Unsupervised Learning</p></title>
  <aug>
    <au><snm>Yang</snm><fnm>Y</fnm></au>
    <au><snm>Shen</snm><fnm>HT</fnm></au>
    <au><snm>Ma</snm><fnm>Z</fnm></au>
    <au><snm>Huang</snm><fnm>Z</fnm></au>
    <au><snm>Zhou</snm><fnm>X</fnm></au>
  </aug>
  <source>International Joint Conference on Artificial Intelligence</source>
  <pubdate>2011</pubdate>
</bibl>

<bibl id="B65">
  <title><p>Evolution of the Human ASPM Gene, a Major Determinant of Brain
  Size</p></title>
  <aug>
    <au><snm>Zhang</snm><fnm>J</fnm></au>
  </aug>
  <source>Genetics</source>
  <publisher>Oxford University Press ({OUP})</publisher>
  <pubdate>2003</pubdate>
  <volume>165</volume>
  <issue>4</issue>
  <fpage>2063</fpage>
  <lpage>-2070</lpage>
  <url>https://doi.org/10.1093/genetics/165.4.2063</url>
</bibl>

<bibl id="B66">
  <title><p>Spectral Feature Selection for Supervised and Unsupervised
  Learning</p></title>
  <aug>
    <au><snm>Zhao</snm><fnm>Z</fnm></au>
    <au><snm>Liu</snm><fnm>H</fnm></au>
  </aug>
  <source>Proceedings of the 24th International Conference on Machine
  Learning</source>
  <publisher>New York, NY, USA: Association for Computing Machinery</publisher>
  <series><title><p>ICML '07</p></title></series>
  <pubdate>2007</pubdate>
  <fpage>1151–1157</fpage>
</bibl>

<bibl id="B67">
  <title><p>Upregulation of {BUB}1B, {CCNB}1, {CDC}7, {CDC}20, and {MCM}3 in
  Tumor Tissues Predicted Worse Overall Survival and Disease-Free Survival in
  Hepatocellular Carcinoma Patients</p></title>
  <aug>
    <au><snm>Zhuang</snm><fnm>L</fnm></au>
    <au><snm>Yang</snm><fnm>Z</fnm></au>
    <au><snm>Meng</snm><fnm>Z</fnm></au>
  </aug>
  <source>{BioMed} Research International</source>
  <publisher>Hindawi Limited</publisher>
  <pubdate>2018</pubdate>
  <volume>2018</volume>
  <fpage>1</fpage>
  <lpage>-8</lpage>
  <url>https://doi.org/10.1155/2018/7897346</url>
</bibl>

</refgrp>
} % end of \BMCxmlcomment

%%% Comment out this section when you \bibliography{references} is enabled.



\end{document}
